{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h1> NLP Project </h1>\n",
    "<h2 style =\"color:blue\"> Search Engine and Literature Clustering</h2>\n",
    "<p>This notebook utilizes two major topics in natural language processing to tackle and answer questions related to COVID-19. The first one is: a <i>search engine</i> which allows the user to enter a query and fetches for them the most relevant papers related to that query (max 5). The other topic is <i>literature clustering</i> where PCA algorithm was used to reduce the dimension of the data, followed by K-means which was used to cluster the papers.</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package stopwords to\n",
      "[nltk_data]     C:\\Users\\Sarah\\AppData\\Roaming\\nltk_data...\n",
      "[nltk_data]   Package stopwords is already up-to-date!\n",
      "[nltk_data] Downloading package punkt to\n",
      "[nltk_data]     C:\\Users\\Sarah\\AppData\\Roaming\\nltk_data...\n",
      "[nltk_data]   Package punkt is already up-to-date!\n",
      "[nltk_data] Downloading package wordnet to\n",
      "[nltk_data]     C:\\Users\\Sarah\\AppData\\Roaming\\nltk_data...\n",
      "[nltk_data]   Package wordnet is already up-to-date!\n"
     ]
    }
   ],
   "source": [
    "# basic libraries\n",
    "import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\n",
    "import numpy as np\n",
    "from tqdm import tqdm\n",
    "import re #import the regular expression library\n",
    "import string\n",
    "import os\n",
    "import json\n",
    "import matplotlib.pyplot as plt \n",
    "\n",
    "# sklearn libraries used for tfidf and cosine similarity\n",
    "from sklearn.feature_extraction.text import CountVectorizer, TfidfVectorizer\n",
    "from sklearn.metrics.pairwise import cosine_similarity\n",
    "from sklearn.datasets import load_files\n",
    "\n",
    "# nltk libraries for stop word and puncuation removal.\n",
    "# nltk libraries for word lemmitization\n",
    "import nltk\n",
    "from nltk import pos_tag\n",
    "from nltk.tokenize import RegexpTokenizer\n",
    "from nltk.corpus import stopwords #import the stopwords from the ntlk.corpus library\n",
    "nltk.download('stopwords')\n",
    "from nltk.tokenize import word_tokenize #import the word_tokenize method, which is used to turn sentences into words\"\n",
    "nltk.download('punkt')\n",
    "from collections import Counter\n",
    "from nltk.stem import WordNetLemmatizer, PorterStemmer\n",
    "nltk.download('wordnet')\n",
    "\n",
    "import spacy\n",
    "from spacy.lang.en.stop_words import STOP_WORDS\n",
    "# !pip install scispacy scipy\n",
    "# !pip install https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.2.4/en_core_sci_lg-0.2.4.tar.gz\n",
    "import en_core_sci_lg  \n",
    "plt.style.use('ggplot')\n",
    "\n",
    "from sklearn.manifold import TSNE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h4>STEP 1: DATA PREPROCESSING</h4>\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<p> Before applying the NLP techniques on the data, data must be preprocessed, cleaned, structured,etc. In the following cell, the folder(s) containing the data in the json format are accessed. Each json file has, then, three majors information extracted from it:<br/>\n",
    " <ol>\n",
    "     <li>The title of the paper</li>\n",
    "     <li>The abstract of the paper</li>\n",
    "     <li>The body/full text of the paper</li>\n",
    "</ol>\n",
    "These are then added to the list which will contain all the papers with those 3 features or attributes. The list is then converted to a dataframe which will be used later in this notebook."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "_cell_guid": "b1076dfc-b9ad-4769-8c92-a6c4dae69d19",
    "_uuid": "8f2839f25d086af736a60e9eeb907d3b93b6e0e5"
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████████████████████████████████████| 1342/1342 [01:08<00:00, 19.55it/s]\n"
     ]
    }
   ],
   "source": [
    "FullPaper=[]#a list which will save all the papers \n",
    "directories = [\"biorxiv_medrxiv\"] #the folder name which contains the papers\n",
    "for directory in directories: #for each of the folders carrying the json format of different research papers\n",
    "    for file in tqdm(os.listdir(f\"{directory}/{directory}/pdf_json\")): #for every json file\n",
    "        file_path=f\"{directory}/{directory}/pdf_json/{file}\" #set the file path to the file_path variable \n",
    "        paper = json.load(open(file_path,\"rb\")) #load the json version of the file\n",
    "        title = paper['metadata']['title'] #set the title variable to the file's title\n",
    "        try:\n",
    "            abstract = paper['abstract']#set the abstract variable to the file's abstract\n",
    "        except:\n",
    "            abstarct=\"\"   #set the abstract variable to the an empty string if the file does not have an abstract             \n",
    "        full_text=\"\" #create an empty string which will hold the body of the paper (file)    \n",
    "        \n",
    "        for text in paper['body_text']: #for every word in the body of the file\n",
    "            full_text += (text['text'] +'\\n') #concatenate the word to the variable full_text\n",
    "        FullPaper.append([title,abstract,full_text]) #since the 3 variable: title, abstract, full_text, now have a value, a paper can be created and added to the FullPaper list\n",
    "        \n",
    "FullPaperDataframe=pd.DataFrame(FullPaper,columns=['title','abstract','full_text'])#create a dataframe which will hold all the papers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h5>Data Cleaning</h5>\n",
    "<p>In the below cell, each body text for each paper is stripped off stop words, punctuation, and is lemmatized. The <i>en_core_sci_lg</i> model was used as it works well on biomedical and scientific papers </p>\n",
    "<h6 style = \"color:green\"> PROS Data Cleaning </h6>\n",
    "<p>Data Cleaning is an essential step in producing high quality data to be further used. By removing stopwords, unnessary puncuations and lemmitization we are able to refine the data avaliable in the dataset to produce high quality results in our project.</p>\n",
    "<p>We choose to use lemmitization instead of stemming as stemming removes only the suffixes of the word rather than search for its root in a dictionary which is done using lemmtization.</p>\n",
    "<h6 style = \"color:red\"> CONS associated with data cleaning </h6>\n",
    "<p> Some semantic meaning of the data may be lost during the cleaning process.</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████████████████████████████████████| 1342/1342 [46:26<00:00,  2.08s/it]\n",
      "100%|█████████████████████████████████████████████████████████████████████████| 1342/1342 [1:54:33<00:00,  5.12s/it]\n",
      "IOPub data rate exceeded.\n",
      "The notebook server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--NotebookApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "NotebookApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "NotebookApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "full_text = FullPaperDataframe['full_text']\n",
    "\n",
    "clean_text = [] #a list which will hold the clean paper text (after data processing)\n",
    "dirty_text = []# a list which will hold the dirty paper text (before data processing)\n",
    "# Intializing parser lemmtizer\n",
    "lemmatizer = WordNetLemmatizer()\n",
    "\n",
    "# Using the scispacy library to get the biomedical terms \n",
    "parser=en_core_sci_lg.load(disable=[\"tagger\", \"ner\"])\n",
    "parser.max_length = 7000000\n",
    "\n",
    "# extending and customizing the stopwords list (Since we found these words in the text before adding the our_stopwords list)\n",
    "all_stopwords = stopwords.words('english')\n",
    "# 'the','we','it','they',\n",
    "our_stopwords=['copyright','https','et','al','preprint','preprints',\n",
    "               'this','these','also','however','although','among','in','medrxiv',\n",
    "                'biorxiv','license','without','fig','figure','doi','www','author','authors']\n",
    "all_stopwords.extend(our_stopwords)\n",
    "\n",
    "pattern = \"\"\"!\"#$%&'()*+,.:;<=>?@[\\]^`{|}~\"\"\"\n",
    "\n",
    "#  removing all stop words and extracting puctuations\n",
    "for val in tqdm(full_text):\n",
    "    val = val.lower()\n",
    "    body_tokens = word_tokenize(val)\n",
    "    paper_body_without_stopwords =  [token for token in body_tokens if not token in all_stopwords]#a list which will hold the tokens stripped off stop words and words that aren't pronouns\n",
    "    dirty_paper_tokens = [token for token in body_tokens] #will hold the tokens as is\n",
    "    clean_string = ' '.join(paper_body_without_stopwords) #convert the list into string\n",
    "    clean_string = re.sub(rf\"[{pattern}]\", '', clean_string) #Strip punctuation from the text\n",
    "    clean_string = re.sub(r'\\b[a-zA-Z]\\b', '', clean_string) # from single letters from the text\n",
    "    clean_string = re.sub(r'^\\d+\\s|\\s\\d+\\s|\\s\\d+$', '', clean_string) #remove digits that aren't associated with a word from the text\n",
    "    clean_text.append(clean_string)\n",
    "    dirty_string=' '.join(dirty_paper_tokens)#convert the list of the dirty tokens into a string\n",
    "    dirty_text.append(dirty_string) #this list will be used later for analysis\n",
    "lemmatized_text=[]    # a list which will carry the lemmatized text\n",
    "\n",
    "# applying the parserer on the dataset\n",
    "for val in tqdm(clean_text):\n",
    "    lemmatized_string=[]\n",
    "    words=val.split(' ')\n",
    "    for word in words:\n",
    "        if(len(word)>1):\n",
    "            token = parser(word)\n",
    "            lemm_token = lemmatizer.lemmatize(token.text)\n",
    "            lemmatized_string.append(lemm_token)\n",
    "    if(len(lemmatized_string)>0):\n",
    "        lemmatized_text.append(lemmatized_string)\n",
    "        \n",
    "for i in range(len(lemmatized_text)):\n",
    "    lemmatized_text[i]=\" \".join(lemmatized_text[i])\n",
    "    \n",
    "print(lemmatized_text)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h5> Corpus Word Count Graph Before Vs. After Data Cleaning </h5>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<p>To demonstrate how effective the data cleaning is, below is a bar chart comparing the corpus word count before and after the data cleaning.</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "26548830\n",
      "17344151\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAccAAAFRCAYAAAABw2bsAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAgAElEQVR4nO3de7xnU/348ddixiUSOZXGbSq3UImI8hVFqW9fSlmRlL4yqXSj+kliKNK3i4QS8pUUrYqaylff+rp2U5qkJKWQMQzjfmfYvz/WOmbPms8553Nun3NmvJ6Px3nM7LXX3nvt/dl7v/dae+29Q9M0SJKkhZaZ6AJIkjTZGBwlSaoYHCVJqhgcJUmqGBwlSaoYHCVJqhgcn8RCCKeHEH4+0eWYDDptixDC+0MIc0IIj4cQZpa03UMI/wghPBZCOH0iyrokWpq3WwhhegihCSFsO0bzOzqEMK/Mc5+xmOdYCSHMDCFcO0HL7un5alIHxxDC6iGE/wohXBNCeCiEcGsI4ZIQwttDCFMmunwjEULYN4TwaAjhqVX6lYOkf6O3pVxUCGGbEMI55YB9qJzkzgwhbD4BZVmrnDS27yLvRSVvU7btrSGEi0MIB4YQVqyyfxDYvTXtNOBLwGeANYHPhxCWBU4DErBOmWap1Drh9/8tCCHcGEL4Sr2PdjGvJ812G0oIYfkQwvwQwoMhhL4O418KfByYATwb+E4I4dQQwkU9KNtTQgiHlnPOAyGEO0IIl5WLxKeM9/K7sMgxOt4mbXAMIawFzAbeBBwJbA68HPg68BFg01HMe7mxKOMI/RyYAryiP6EcJJsAN3dI3xT4v5EubLTrGkJ4J3Ap8CiwF/B84C3A9cBxo5l3j3ybfJKZDuwEfA84CJgdQnhmf6amae5umubO1nTPJR8fs5qmublpmvvKfFYGzmua5qamae4eSYFCCMuUgLEk2JWF22+/MvylYc5jrLbb1BBCGMm0k8ibgBuAC4F9OoxfH3i8aZofNk1zS9M0D47lwgc6H4QQVgF+CbwfOBF4GbAF8HkgAq8ey3KMRIdjdNwXOCn/gB8BtwBP6zBuKrBS6//HADcBjwB/Ad5a5W+AD5BPlHcD3yUf7A2wNzn4PAhcB+zVmq4/z7bV/K4FZraG3wVcDTwE3A5cAqw1yLpdCxzbGo7AH4HjO6Q3wJqttHeUdXwYmAN8GpjSGn8R+QLiU+Rge1tJXw34DnA/MK9M9w3g54OUc1pZp5MGGL9a6/8bAj8B7it/PwLWa43fB1hQTb9WWb/ty/D2ZXinsg0fKOv6muq3bP9dP0j5LwJO7ZC+FnAncFor7fT+bQHM7LCcfTqk9Zd7C+B/y3rfBpwDrNua98zym78F+CuwANi0jNsDuKJs5+uBL1L27fY6AJ8kHw93lLKuVK3TW4Dfs3Af/J/q93l/WfZDwN+BT7T3mw7baDqd9/0vAH+s0gZc/yG22+tKmR8GbgW+Uq376eSLyfeXbfM4sPII1ycApwD/IB/r/wSOBpbv8DvtWuZ9PzmIPa+aVyz5HgJ+BezSaVsNUI6LS9kjcE017vQO22pmh7R9Sv6VyReoN5GPlT8Au3X4DfcCzivr8/kBynV82S7PGWDbrdreRtX4nciB9cFSlv8GVm+N35y8P95a9pHfATtX87ieXAk6jryPzyMH5mU7HaPV/jGDfMFxD/BD4BnVvD9EPlc+APyUfM5vGOQc3TTN5AyOwNOBx4BDu8j7OfLJYHdgA+AQ8kH0qlaepuR5P/C8kq9/x5lbdp4NyQHjceAlQ5wgngiO5BPDAuDtwLrAC8jBcrDg+DXgytbwScCxwBs7pF/dGv73sl0+XtbhLeST/KdaeS4C7i3Tbgy8oKSfW8r9SnIt9cyyMw0WHD/U1U4EK5ad8//K9tiCfFK5Fliu5NmH7oPjH4GdyVfRZwB3sfDgfHHJsxuwRn0gVPO/iA7BsYw7gXyhtEx94JFPOruV5by4LGclYMuStktJW65s4/uAI4CNyu//XeBvwAqtE8oD5BPj1uW3e2rZJneSD9bnAtsBVwLfrNbhrrJ/bFS2y13AEa087yTX7D9ZyvNCchNUX2v5N5D3r+eQg9K/2vtNh+0znWrfL2W8CvhqK23Q9S/7Rqft9kLycXMsuTXitaVM7XU/nbyPngtsVuY9ZYTrswz5+H5pWbddyBeP7e04kxxAzifvwy8iB5yLW3leTD5HfIZ8ztiNfFE9ZHAs2+dhoA9YnhwEtm+Nf1r53RaU7bQGeV/8FjkI96etSA5YF5b9Y9vy28wgVxBeVf2Gc4C3lTydgt8y5PNjx2OlyjuTVnAkn08eIJ9b1y+/9YXki9vQOq7fUfaVDcrv8AiwQWs+15OPhYPLfN5StsM7q/2hDo53A2eRW9heVvaLb7Ty7Fbm88Ey333I5/wlNjhuVQq/2xD5nlJ2tvdW6ecCF7SGG+DrAxz8n6rSfwWcOdAJoqS3g+Mbyw+0yjDWL5IPsGeW4b+RD9anlx+ynX58a7pLgVTN64PkK7b+IHRRmW6ZVp71ynrs1EpbjnyVN1hw/Apwdxfrs285QPpaac8q5Xp7Gd6H7oNj++p3jZL2mk7TDFGuixg4OO5f5tO/rU9n0QOvvyxrtdIW2x/KdGdX816+bI83lOGZ5fdep8p3PbB/lbZdWcZqrXW4sspzEvDr1vC/gBMGOUYeYPEr9bcDdw2y7frX9QFy8HuoDF9Aqb0NY/07bbdvAr+tptu1bKd1W/O+q1reiNZngHX8MPD31vBM8vH3jFbaHqVM/Rc6ZwK/quZzQL1+AyzvWODc6vj6dpVnHxY/Tk4FLqrSti+/ydOq9NOAH1Tb/ZNDlOuZJd+BXWyzmSwaHC8CjqnyrFPmt9kg8/kj8InqWJhV5TkfOKva1+rgeBuL1v4PBm5uDf+S1gVXSTuGLoLjZO3U0n9foRki33rAcjvttNOnYozvTSn134e8mFy7avttjPFYYAeAnXfeeZXzzz8f4NdVvl8CrxpGWX9GbqK5LoTwM/LJ45ymaeYPMs0F5d9XhRAuJV/RXdw0zd0hhCtb6esDH21Ntwm5abTtYvIV+vPITbsAv2+a5vFWno3Lv7/qT2ia5pEQwu/IV6YD6fb+zibAX9rr3DTNvBDCNWXccF3Rms8tIYTHyMF2LHW7jw1lS2C9EMJ9VfoK5N+v37ymaf71xMJDeAa5peGLIYTPdyjXeuTmJ2htj+Imyj2gct90bXKzZiebkGsa3w8htNd1WWCFEMIzmqa5bZD1eye56XMZ8sn208CsEMKOZR/rdv07leuCKu1i8vpvTK4BQG45ua+abkTrE0LYj9yqM53cEjCFxftdzK2mv6mU6Znki5CNWbwPwC8GWMf2slcgB/B9W8mnA5eEEFZvmub2oeZR2ZJygVvdhl2O3Mzc9tuhilf+HcmxsCWwdQjhgA7j1geuKPv6EeRa5hrk7b4Cef9v67SfP2eI5V/dNM3D1TTtc8XG5NtpbfU5v6PJGhz/Tr5a24RcCxzUnDlz3r3qqqvObCUFFv+h708pfbh/YPvttz+cfBVUa+9pj3dIg3yfE4Cmae4LIbyE3FloR3KN5L9CCK9qmub3ncrbNM38EMIfyUF4OWB2s7CTwoWt9MfIV2aLTD5Aedvp9w+yTsNxDbBKCGGtpmnmDJG304HV/h0e7zB+aoc0yE0utbHuPLYpuVYy3JNSbRlyLeiYDuPa865/k/71+SD5N6+1t3e9PRoW3x4Dndj68+1OblGo3THAdP1uapqmv+v+30II95IvsrYnB7du17+Tgco82L48ovUJIexO7mhyMDkI31PmcVSVtdO2bi+307mlG28mtwx9rwpmy5KbHL84zPktQ26x2rLDuHod6m1Yu43cpDmSC9llgM+S94HaLeXf08m1yY+Rm6AfBM4mn+PautnPa52mqc93I7oAnpTBsWmaO0II/wMcEEI4vql6t4UQppI37LXAwzfeeONGm266sPPq1KlTX7vyyitPjTH+ntwEs5gVVlhhl/Lfrck3q/ttw8IaWP8V5LTWsp9J7trfLu9j5Db2S0IIh1M6BZGvuAfyf+Sea1NZ9Ar6QvLN8anA5dW6X0XuzXpiK207FnYwGMhV5d+XkWu6/b3WtmThunbyXfJJ71By0F9ECGG1JvceuwrYP4TQ1197DCE8i3x/ob9WdCuwbAjhWU3TzCtpI3kUpP9gGHFvz9ITei/gu1UNeyQuJ98/+0dT2my6UWrWNwIbNk1zykgX3jTNrSGEOcBryJ2galeRm9+e2zTNeR3GD9eC8m9/1/4RrT8L9+W2V5BPZH8ZYrqRrM92wB+apnkiCIUQpg9j+vbyX16l1cOdvJscJL5Qpe9N7gU8WHB8hMX398uBVcnNvX/uYvkDaprm8RDCt4F9QwhHNU1zXXt86SG8Sn0ebpVjk9YFVCfbAR9rmmZWmd9K5NayUZW7S38hn9O/0krbupsJJ+2jHMB7yZ0Mfh9CeGsIYeMQwnohhLeRf5D1m6Z5APjyfffdd9D111+/Sghh/RDCIY8++uhOyy+//IdTSluQH/tYRIxx3WWWWWbtMrhvmf8GIYQjyRvySwBN7kb9S+BjIYQXhRC2IHcQeaIaH0LYNYTw4RDCFiGEdYA3kJu5BjvAIQfH6eR7lu3geEmZ/o0s3nzzGeBNIYSDS3kjufb7haZpOtW2KOtxLTALODGEsEMIYWPyfYxBn1drmuYm8v2U/UIIZ4cQXlWef9s8hHAEuWcY5GaL28jPZG1ettPZ5CaO/mbg35I7Ch1TfqedgcMGW/4A5pPvgb06hLBGCGG1IfKvWPJNCyG8sDT/XFbKVje9j8TR5A4lZ4YQtgohPKds4+NCCM8dYtpPAB8oz5ZtGkLYMITwhhDC14ZZhiOAd4cQPhlCeH4IYZMQwgHlYuW+UsajS9qGZfweIYTPdjHvp7e23zbkk/itLGyiH+n6fw7YPITwxRDCRmV/OB74Vrv5uTaK9bkGeEE5Xp8XQvggubPGcB0LbBNCOKocg28kPxo0oHK8bUvuHf3n9h+5c95GIYTtBpnFdSXPJiGEvhDC8uRzxs+Bc0IIbwwhPLecg95fmo+H6xPkFrvfhBBmlPPdc8r6XUy5HdXBYcCuIYRjQwiblW27cwjh62Hhs8TXAHuFEF4QQtiM3IGmV48yfQHYo2yX9UIIbyc3b8NQNcqhbsBO5B/wjLJyfyNfLd5K/qHeRum2DUxdYYUVvrr88ss/Sr7CunqrrbZ6ZPfdd7+i/69shLf1z3f33Xf/fzvuuOPpLHyU4yIWdqXfuyrDBmWZ95N3nt1YtEPOduQd9TYWdis/mNJTa5B1W6mU9xHgKdW4y0rZdugw3TvItb1HyCf4o1j8UY5Ojy+sTn4I+/5S1s8wxKMcrWm3BX5Qtv/D5FrqGbRuuJN77p3Hwkc5fkzrUY6S599L2fsvOl5D5w45a1XTLaB0Xy/DbyefMB5l6Ec5mvL3KDmwXgwcCKxY5T2dEXTIKekvIF8o3FnW7VrgZODpZfxMqu7vrWnfQL4H8gC5qe8K4LDBfk9yTf76Km0vcieHh8nNmT+h9PAt4/dl4SMjd5Z97D2DbLv+de3/e5zcTHYu5TGUYaz/QNut/SjHbcBX6fAoxwDlG+76TCUHojvKdv42pSNNK89ivxN532+A6a20PciPhDxclrtrp/Vr5e9/3KLjOYF8b7m/E+A+LN4h5+nkY+tuFn2UY0Vyy8515PPBLeROLK8cbLsPcU46nFyje7C1Xd9HOV4G2Eb/Rg7U95LPL1eTKxj95+gXkC+mHiSfY99b8p/emsf1VE8nUHVEqveHTvsHOTY0VdqHy/Z/kPwox4yyXVYfbHv0d7VdosUYpwM/TiltGmNcBbgmpfTsQfL/4brrrvvU5Zdf/n3g35qmGfKGuiRpyRdCOAz4YNM0qw+WbzI3q45ISuke4LoY4+4AMcYQY3xR//gY44bAarNnz549UWWUJI2/kN+qdHC5pfK8EMK7yE8ADHmff1J2yBmOGONZ5CawvhjjHHKzwF7AV2OMh5KbU84mNzkB7Amc/fjjo+2HIUma5BpyfDiI3MfiOvI9688NNeFS0awqSdJYWuqaVSVJGi2DoyRJlSX9nqNtwpKkTkb1ebMlPTgyd+7ciS7Ck1pfXx/z5w/2GlnpycVjYuJNmzZt6ExDsFlVkqSKwVGSpIrBUZKkisFRkqSKwVGSpIrBUZKkisFRkqSKwVGSpIrBUZKkisFRkqSKwVGSpMoS/27VsfDYfrtMdBGWWPMmugBLsGVPmTXRRZA0AGuOkiRVDI6SJFUMjpIkVQyOkiRVDI6SJFUMjpIkVQyOkiRVDI6SJFUMjpIkVQyOkiRVevL6uBjj2sAZwBrA48DJKaXjqjzbAz8EritJ56SUjuxF+SRJauvVu1UXAAellGbHGJ8K/D7G+LOU0l+qfJemlF7fozJJktRRT5pVU0o3p5Rml//fC1wNrNmLZUuSNFw9/ypHjHE68GLgsg6jt4kx/hGYC3wkpXRVL8smSRL0ODjGGFcGvg98KKV0TzV6NrBuSum+GOPrgB8A63eYxwxgBkBKib6+vlGXy88uaSKMxb6ryWfKlCn+tkuB0DRNTxYUY5wK/Bj4aUrpi13kvx54SUpp/iDZmrlz5466bH7PURPB7zkunfr6+pg/f7DTlsbbtGnTAMJo5tGTe44xxgB8Hbh6oMAYY1yj5CPGuFUp2+29KJ8kSW29alZ9ObA38KcY4xUl7RBgHYCU0knAm4H3xBgXAA8Ce6SUelOtlSSppWfNquPEZlUtsWxWXTrZrDrxlphmVUmSliQGR0mSKgZHSZIqBkdJkioGR0mSKgZHSZIqBkdJkioGR0mSKgZHSZIqBkdJkioGR0mSKgZHSZIqBkdJkioGR0mSKgZHSZIqBkdJkioGR0mSKgZHSZIqBkdJkioGR0mSKgZHSZIqBkdJkioGR0mSKgZHSZIqBkdJkioGR0mSKgZHSZIqBkdJkioGR0mSKgZHSZIqBkdJkioGR0mSKgZHSZIqBkdJkioGR0mSKgZHSZIqBkdJkioGR0mSKgZHSZIqBkdJkipTerGQGOPawBnAGsDjwMkppeOqPAE4Dngd8ACwT0ppdi/KJ0lSW69qjguAg1JKzwe2Bt4XY9y4yvNaYP3yNwP4ao/KJknSInoSHFNKN/fXAlNK9wJXA2tW2XYFzkgpNSml3wCrxhif3YvySZLU1vN7jjHG6cCLgcuqUWsCN7aG57B4AJUkadz15J5jvxjjysD3gQ+llO6pRocOkzQd5jGD3OxKSom+vr5Rl2veqOcgDd9Y7LuafKZMmeJvuxToWXCMMU4lB8ZvpZTO6ZBlDrB2a3gtYG6dKaV0MnByGWzmz58/1kWVesJ9d+nU19fnbzvBpk2bNup59Kq3agC+DlydUvriANlmAQfEGM8GXgrcnVK6uRflkySprVc1x5cDewN/ijFeUdIOAdYBSCmdBJxHfozjWvKjHO/sUdkkSVpEaJrFbustSZq5cxdreR22x/bbZQyKIg3PsqfMmugiaBzYrDrxSrNqp34sXfMNOZIkVQyOkiRVDI6SJFUMjpIkVQyOkiRVDI6SJFUMjpIkVQyOkiRVDI6SJFUMjpIkVQyOkiRVDI6SJFUMjpIkVQyOkiRVDI6SJFUMjpIkVQyOkiRVDI6SJFUMjpIkVQyOkiRVugqOMcY7Bki/dWyLI0nSxJvSZb6pdUKMcSqw7NgWR9Jk8Nh+u0x0EZZY8ya6AEuwZU+ZNdFFeMKgwTHGeCnQACvEGC+pRq8F/Gq8CiZJ0kQZquZ4KhCALYGvt9Ib8gXSBeNULkmSJsygwTGl9A2AGONvUkp/7U2RJEmaWF3dc0wp/TXG+GpgM2Dlatxh41EwSZImSlfBMcZ4AhCBC4EHWqOa8SiUJEkTqdveqnsCm6WUbhzPwkiSNBl0+xKA24G7xrMgkiRNFt3WHL8AfCvG+Bmqx3hSSv8c81JJkjSBug2OXy3/vr5Kb/BFAJKkpUy3vVV9B6sk6UnDoCdJUqXbRzn6XyO3mJTSdmNaIkmSJli39xxPrYbXAPYFzhzb4kiSNPG6vef4jTotxvh94L+BI8e6UJIkTaTR3HO8CXjhWBVEkqTJott7jv9ZJT0F2A34zZiXSJKkCdbtPce9q+H7yd9yPHZsiyNJ0sTr9p7jDuNdEEmSJotua47EGNcnv4B8TfL9xrNSSn/vctrTyG/XuTWltGmH8dsDPwSuK0nnpJTs6CNJmhDd3nP8D+BbwI+BG4ANgctjjHunlGZ1MYvTgROAMwbJc2lKqX49nSRJPddtzfFoYNeU0oX9CaW2dwIwZHBMKV0SY5w+kgJKktRr3QbHtYBLq7RflPSxsk2M8Y/AXOAjKaWrxnDekiR1rdvgeAVwEPDZVtqBJX0szAbWTSndF2N8HfADYP1OGWOMM4AZACkl+vr6Rr3weUNnkcbcWOy748VjQhNhMh0T3QbH9wA/ijF+ELgRWJv8OMcuY1GIlNI9rf+fF2P8SoyxL6U0v0Pek4GTy2Azf/5iWaQlgvuutKixOiamTZs26nl09YaclNJfgecDkfzh4whsnFK6etQlAGKMa8QYQ/n/VqVct4/FvCVJGq5Ba44xxqcDW6WUzk8pLSDfZ+wft3OM8bKU0p1DLSTGeBawPdAXY5wDHA5MBUgpnQS8GXhPjHEB8CCwR0qp41dAJEkab0M1qx5KrsGd32Hci4EdgY8MtZCU0p5DjD+B3PNVkqQJN1Sz6uuBrw0w7mRg17EtjiRJE2+o4LhGp04xxR3As8a4PJIkTbihguOdMcYNBxi3AXDXGJdHkqQJN1RwPBf4coxxxXZiGT4W+N54FUySpIkyVIecTwIXAP+MMZ4P3Aw8G3gN+XnHw8e3eJIk9d6gNceU0r3Ay8hBcgXgJeXfTwL/VsZLkrRUGfINOSmlR4FTy58kSUu9rt6QI0nSk4nBUZKkisFRkqSKwVGSpMqAHXJijN8Ehnz5d0rp7WNaIkmSJthgNcdrgX+Uv7uBNwDLAnPKdLviG3IkSUuhAWuOKaUj+v8fY/wp8O8ppUtbaduSn3eUJGmp0u09x62B31RplwHbjG1xJEmaeN0Gxz8AR/e/Y7X8exRwxXgVTJKkidJtcNwHeDlwd4xxHvke5LaAnXEkSUudIV8fF2MM5CC6Hfml49OAm1NK/xrnskmSNCG6ebdqE2P8E/DUlNKN5K9xSJK01BrOPccNxrMgkiRNFkPWHIuLgPNjjKeTa45PvBwgpXTa2BdLkqSJ021wfDlwHfCKKr0BDI6SpKVKV8ExpbTDeBdEkqTJotuaIzHG1YD/ANYEbgJ+lFK6c7wKJknSROmqQ06McRvyO1b3B14IvBv4R0mXJGmp0m3N8UvAe1NKZ/cnxBjfAnwZ2HI8CiZJ0kTp9lGODYBUpX0PWG9siyNJ0sTrNjj+HdijStud3NQqSdJSpdtm1Q8BP44xfgC4AZgOrA+8fpzKJUnShOmq5phS+hXwPOAE4PfA8cB6JV2SpKXKoDXHGGMELkkp3VIe2zizN8WSJGniDNWs+mngeTHGfwCXABeTg+UN414ySZImyKDNqimlDcifqPoE8CBwEPn5xhtijN+MMb6rB2WUJKmnuvlk1Tzgu+WPGOOqwAzgQOCtwKnjWUBJknqt248db0b+2PF2wMuAueTnHi8d19JJkjQBhuqQ82Ngc+Aa4BfAycA+KaV7e1A2SZImxFCPcmwIPEz+XNU/gGsNjJKkpd2gNceU0voxxmexsEn1QzHGPuCX5CbVX6SUrhj/YkqS1Duj6ZBzKPAMYNnxLKAkSb02kg452wKrApcDp3WzkBjjaeRXzd2aUtp0gGUcB7wOeIB8X3N2l+sgSdKYGqpDzk/IvVOXAy4jvwTgBODXKaWHhrGc08t0Zwww/rXkd7WuD7wU+Gr5V5Kknhuq5ngpcBTwu5TSoyNdSErpkhjj9EGy7AqckVJqgN/EGFeNMT47pXTzSJcpSdJIDdUh55gelWNN4MbW8JySZnCUJPVct5+sGm+hQ1rTKWOMcQa5QxApJfr6+ka98HmjnoM0fGOx744XjwlNhMl0TEyW4DgHWLs1vBb5LTyLSSmdTH4ZAUAzf/78cS6aND7cd6VFjdUxMW3atFHPY7IEx1nAATHGs8kdce72fqMkaaL0JDjGGM8Ctgf6YoxzgMOBqQAppZOA88iPcVxLfpTjnb0olyRJnfQkOKaU9hxifAO8rxdlkSRpKEO9W1WSpCcdg6MkSRWDoyRJFYOjJEkVg6MkSRWDoyRJFYOjJEkVg6MkSRWDoyRJFYOjJEkVg6MkSRWDoyRJFYOjJEkVg6MkSRWDoyRJFYOjJEkVg6MkSRWDoyRJFYOjJEkVg6MkSRWDoyRJFYOjJEkVg6MkSRWDoyRJFYOjJEkVg6MkSRWDoyRJFYOjJEkVg6MkSRWDoyRJFYOjJEkVg6MkSRWDoyRJFYOjJEkVg6MkSRWDoyRJFYOjJEkVg6MkSRWDoyRJFYOjJEmVKb1aUIxxZ+A4YFng1JTSMdX4fYDPATeVpBNSSqf2qnySJPXrSXCMMS4LnAjsBMwBfhdjnJVS+kuV9TsppQN6USZJkgbSq2bVrYBrU0r/TCk9ApwN7NqjZUuSNCy9alZdE7ixNTwHeGmHfG+KMW4H/A34cErpxg55JEkaV70KjqFDWlMN/wg4K6X0cIxxf+AbwCvriWKMM4AZACkl+vr6Rl24eaOegzR8Y7HvjhePCU2EyXRM9Co4zgHWbg2vBcxtZ0gp3d4aPAX4bKcZpZROBk4ug838+fPHsJhS77jvSosaq2Ni2rRpo55Hr+45/g5YP8b4nBjjcsAewKx2hhjjs1uDuwBX96hskiQtoic1x5TSghjjAcBPyY9ynJZSuirGeCRweUppFvCBGOMuwALgDmCfXpRNkqRaaJr61t8SpZk7d+7QuYbw2H67jEFRpOFZ9pRZQ2eaIB4TmghjdUyUZtVOfV265htyJEmqGBwlSaoYHCVJqhgcJUmqGBwlSaoYHCVJqhgcJUmqGBwlSaoYHCVJqhgcJUmqGBwlSaoYHCVJqhgcJUmqGBwlSaoYHCVJqhgcJUmqGBwlSf7iztYAAAiFSURBVKoYHCVJqhgcJUmqGBwlSaoYHCVJqhgcJUmqGBwlSaoYHCVJqhgcJUmqGBwlSaoYHCVJqhgcJUmqGBwlSaoYHCVJqhgcJUmqGBwlSaoYHCVJqhgcJUmqGBwlSaoYHCVJqhgcJUmqGBwlSaoYHCVJqhgcJUmqTOnVgmKMOwPHAcsCp6aUjqnGLw+cAWwB3A68JaV0fa/KJ0lSv57UHGOMywInAq8FNgb2jDFuXGXbF7gzpbQecCzw2V6UTZKkWq+aVbcCrk0p/TOl9AhwNrBrlWdX4Bvl/98DXhVjDD0qnyRJT+hVcFwTuLE1PKekdcyTUloA3A2s3pPSSZLU0qt7jp1qgM0I8hBjnAHMAEgpMW3atNGX7ieXj34e0tLEY0JPcr0KjnOAtVvDawFzB8gzJ8Y4BXgacEc9o5TSycDJ41RODVOM8fKU0ksmuhzSZOExsXToVXD8HbB+jPE5wE3AHsBbqzyzgHcAvwbeDFyQUlqs5ihJ0njryT3Hcg/xAOCnwNU5KV0VYzwyxrhLyfZ1YPUY47XAgcDBvSibJEm1nj3nmFI6DzivSjus9f+HgN17VR6NGZu4pUV5TCwFQtPYcilJUpuvj5MkqdKzZlVNTjHGx4A/kR+leQw4IKX0qyGm+QDwHmB2Smmv8S+lNPFijG8EzgGen1L6a0n7HPA68i2jXwJ/Syn9ZeJKqbFizVEPppQ2Sym9CPg48Jkupnkv8LpuA2N5NEda0u0J/ILc277fu4HNU0ofBd5Afj1m1zw2Ji9/GLWtAtzZPxBj/CgQgeWBc1NKh8cYTwKeC8yKMZ5GfuXfaSXtAWBGSunKGONMYBowHZgfY9wbOAbYvszvxJTS13q0XtKoxBhXBl4O7EB+7GxmjHEWsBJwWYzxXGAX4BUxxkOBN5VJTwSeQT429ksp/TXGeDr5Ge4XA7OBg3q5LuqONUetGGO8Isb4V+BU4FMAMcZXA+uT34u7GbBFjHG7lNL+5Bc47JBSOhY4AvhDSumFwCHkL6v02wLYNaX0VvKL5e9OKW0JbAnsV557lZYEbwDOTyn9Dbgjxrh5SmkXFra8HEEOmh8tw/8g91p9f0ppC+AjwFda89sA2DGlZGCcpAyO6j+4NwJ2Bs4oL3x/dfn7A/nqdiNysKxtC3wTIKV0AflZ1aeVcbNSSg+W/78aeHuM8QrgMvJ7czvNT5qM9iR/MIHy756DZS41zZcB3y37/NeAZ7eyfDel9Nh4FFRjw2ZVPSGl9OsYYx+5GSgAn+mi6XOwd+LeX+V7f0rpp6MvqdQ7McbVgVcCm8YYG/I3aZsY48cGmWwZ4K6U0mYDjL9/gHRNEtYc9YQY40bkA/928tuM/rNcARNjXDPG+MwOk10C7FXybA/MTynd0yHfT4H3xBinlrwbxBhXGvu1kMbcm4EzUkrrppSmp5TWBq4jt5q03Qs8FaAcA9fFGHcHiDGGGOOLellojY41R61Ymn0g1+7eUZp7/jfG+Hzg1zFGgPuAtwG3VtPPBP47xngludPBOwZYzqnkzjmzS7PtbeT7ONJktye5M1nb91n8/dBnA6eUR53eTL5o/GrpoDO1jP/jOJdVY8Q35EiSVLFZVZKkisFRkqSKwVGSpIrBUZKkisFRkqSKwVGSpIrPOUqjEGN8K3Ag+fV69wJXAEellH4xoQUbQIzxKmDdMrgi8CiwoAwfnVI6ehTzvgV482Rdd2k4DI7SCMUYDwQOBvYnvwHoEfL7aXclf9poOPOaklJaMHTO0UkpbdJa5kXAmSmlU8d7udKSxuAojUB5ufqRwDtTSue0Rv2o/BFjXB74LPmzXwAJ+H8ppYfLq/bOBI4HPgz8LMb49ZL2FXJt9D7gEymlb5X5XUQrmMUY9wHelVLatrx16Ivkt7IsD9wAvDWl9OcRrNu7y/KfAfya/Bmym2KMOwDfAV6UUro5xrgl8L/kr698Fngm+c1KjwOHpJS+PNxlS5OF9xylkdkGWAE4d5A8nwC2Jn/y60Xkz38d2hq/BvB0cjPnjFZaH7Am+VV8J8cYN+yiPK8GtiN/CmlV4C3kd+QOS4xxD+BDwH8AzyJ/leVMgJTSheQvsJwWY3xK+f9HU0r/TCntTn614KtTSisbGLWkMzhKI7M6+SXrgzWF7gUcmVK6NaV0G/nbl3u3xj8OHJ5Serj1aS+AT5a0i4GfsLDmOZhHyS+93ggIKaWrU0o3D2eFincDn04p/S2l9Ggp87YxxmeV8YcAa5E/O3aNTbJaWhkcpZG5HeiLMQ52a2IauXmz3w0lrd9tKaWHqmnuTCndP8g0HZVvaZ5A/vL8vBjjyTHGVYaaroN1gZNijHfFGO8ivyB+ATkgklJ6mPxB602Bz49g/tISwXuO0sj8GniI/GWR7w2QZy452FxVhtcpaf06vfV/tRjjSq0AuQ7Qf9/wfuAprbxrtCcsTZlfLp8WS8BHgU92tTYL3UhuKv1+p5ExxunAx4FvAMfGGLdu1Z79ioGWGgZHaQRSSnfHGA8DTowxLiB3THkU2BHYIaX0MeAs4NAY4+/IgeMwyv27IRwRYzwEeCnweuDwkn4FsFuM8VRybXJfYB5A6RyzDDCbHEQfAkbypfmTSpn/nFK6Jsa4GvDKlNL3Y4zLkGuNx5ObWy8o63RYmXYe8FyG2VNXmoxsVpVGKKX0RXKvzkPJzY83AgcAPyhZPg1cDlwJ/IkcuD49xGxvAe4k1zC/BeyfUvprGXcs+XGReeSa27da060CnFKmvYHc7DvsZs+U0lnk5tlzYoz3kAPyTmX0R8nPRn4qpfQ4ucPQ+2KMLy3jjwKOKk2yBwx32dJk4vccpUmi//GOlNJaE10W6cnOmqMkSRWDoyRJFZtVJUmqWHOUJKlicJQkqWJwlCSpYnCUJKlicJQkqWJwlCSp8v8BqBispatf/eEAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 504x360 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "count1=0\n",
    "count2 =0\n",
    "\n",
    "for d in dirty_text: #get the word count in all the corpus before preprocessing\n",
    "    count1 +=len(d)\n",
    "\n",
    "for st in lemmatized_text:#get the word count in all the corpus after preprocessing\n",
    "    count2+=len(st)\n",
    "labels_x=[\"Before\", \"After\"]\n",
    "values = [count1, count2]\n",
    "print(count1)\n",
    "print(count2)\n",
    "plt.figure(figsize=(7, 5))\n",
    "plt.bar(labels_x, values)\n",
    "plt.xlabel('Corpus Text')\n",
    "plt.ylabel('Word Count')\n",
    "plt.title('Corpus Word Count Difference Before and After Cleaning')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h4>STEP 2: SEARCH ENGINE IMPLEMENTATION</h4>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h5>TF-IDF</h5>\n",
    "<p> After the data has been preprocessed, a TF-IDF matrix is to be created. This is a matrix which will hold the values of each word (bag of words of all the corpus) with respect to the document (paper).</p> \n",
    "<h6 style = \"color:green\"> PROS of using tf-idf</h6>\n",
    "<p>tf-idf matrix considers the unique descriptives words in a corpus and gives them high values. On the contrary, it gives low values to reptitive words in the corpus. Hence, it helps to achieve accurate results for a search engine. </p>\n",
    "<h6 style = \"color:red\"> CONS of using tf-idf</h6>\n",
    "<p>tf-idf matrix does not account for semantics at all, unlike word embeddings, for example.</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[0.         0.         0.         ... 0.         0.         0.        ]\n",
      " [0.02239898 0.         0.         ... 0.         0.         0.        ]\n",
      " [0.         0.         0.         ... 0.         0.         0.        ]\n",
      " ...\n",
      " [0.         0.         0.         ... 0.         0.         0.        ]\n",
      " [0.         0.04055696 0.         ... 0.         0.         0.        ]\n",
      " [0.         0.         0.         ... 0.         0.         0.        ]]\n",
      "(1342, 2375)\n"
     ]
    }
   ],
   "source": [
    "#create a tf-idf matrix for the words above\n",
    "v = TfidfVectorizer(min_df = 0.05,max_df = 0.8)\n",
    "#For the above tf-idf vectorizer:\n",
    "#The first parameter specifies that if a term appears in less than 5% of the docs, ignore it\n",
    "#The second parameter, considers word that do not occur in more than 80% of the corpus\n",
    "tfidf = v.fit_transform(lemmatized_text) #fit and transform the lemmatized texts which were previously cleaned.\n",
    "#print(sorted(v.vocabulary_.items(), key=lambda x : x[1]))\n",
    "print(tfidf.toarray())\n",
    "print(tfidf.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h5>Cosine Similarity</h5>\n",
    "<p> Cosine similarity is then utilized to get the documents which are closest to the query. The closer the cosine similairty value is to 1, the closer the document is to the query (The more likely it is going to be output). However, for this notebook, we choose to display the top 5 papers with cosine similarity greater than or equal <i>0.1</i>. This number was chosen after several trials to fetch the most relevant papers for different queries.</p>\n",
    "<h6 style = \"color:green\"> PROS of using cosine similarity</h6>\n",
    "<p>Cosine similarity is of a good use, as it can match related documents and queries quite well compared to euclidean distance. Euclidean distance can classify documents and queries as non-similar even if the distribution of terms is significantly alike</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Enter your query: Disease models, including animal models for infection, disease and transmission\n",
      "[0.         0.02176277 0.01242025 ... 0.12733774 0.03620335 0.00568204]\n",
      "1315\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#Next, the search engine will be implemented with the help of the tf-idf matrix created above and the cosine similarity\n",
    "flag = False # a flag to indicate whether or not a result is found\n",
    "pd.set_option('display.max_colwidth', -1)#to avoid dataframe column's data truncation\n",
    "query = input(\"Enter your query: \") #get the query from the user\n",
    "finalQuery=\"\" #create an empty string\n",
    "words = query.split(' ') #split the query on spaces\n",
    "for word in words: #for each word in the query\n",
    "    word = word.lower()\n",
    "    if(word!=\"\"): #if the word is not an empty string\n",
    "        if (word==\"coronaviruses\"):#since lemmatize functions do not recognize the word coronaviruses as the plural of coronavirus, a conidition was created\n",
    "            word =\"coronavirus\"\n",
    "        lem_query = lemmatizer.lemmatize(word)#lemmatize the word\n",
    "        finalQuery+=lem_query+\" \" #lemmatize the word and concat it to the string finalQuery\n",
    "query = [finalQuery.strip()]# remove any trailing spaces from the query\n",
    "query_tfidf = v.transform(query)\n",
    "cosineSimilarities = cosine_similarity(query_tfidf, tfidf).flatten() #get the cosine similarity list \n",
    "#IF NO RESULT\n",
    "print(cosineSimilarities)\n",
    "countzero_in2 = np.count_nonzero(cosineSimilarities) #get the number of non zero values in cosine similarity\n",
    "print(countzero_in2)\n",
    "if (countzero_in2==0):#if all cosine similarities are 0 (no results)\n",
    "    flag = True \n",
    "\n",
    "else:\n",
    "    if (countzero_in2>=5):#if there are 5 or more cosine similarities that are greater than 0\n",
    "        answer_indicies = sorted(range(len(cosineSimilarities)), key=lambda i: cosineSimilarities[i])[-5:] #get the top 5 indices with the highest cosine similarity\n",
    "    else: #else if they are less than 5, get all of them\n",
    "        answer_indicies = sorted(range(len(cosineSimilarities)), key=lambda i: cosineSimilarities[i])[-countzero_in2:] #get the top countzero_in2 indices with the highest cosine \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0.23581758156790616, 0.24175444821051353, 0.2571401811704759, 0.2573360580090854, 0.26376795744431764]\n",
      "[21, 183, 939, 1221, 535]\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAcMAAAFRCAYAAAAIKMaWAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAgAElEQVR4nO3deZwcVbn/8c9JAqLsMCIkbAFBDYsiu8giAsYFcONhEWTPRQVU9KcgXJAIGMAr5F5QiICAIvERBeIlEFFWFzRBBG5YNOxJIBDCTkhMqN8f5zSpdKZ7apLumump7/v16td0VZ2qek7XdD1dVadOhSzLEBERqbJBfR2AiIhIX1MyFBGRylMyFBGRylMyFBGRylMyFBGRylMyFBGRylMyFOkAIYS7QggXlLCe94YQshDCNu1eVyfFAhBCGJni6WrlclpZzxDCMSGEV5d1OVWkZNiB0hen2evxNq//mLr1PRNCuD6E8L52rredQggr1NXptRDC/4UQjluKZe2RlrF2C0P8BHBSC5e3TEIIK4UQvhtCuD+E8HoI4fkQwl9CCF8KIby9Rav5F7AO8I8WLa+pEMKoEMI9IYRXQggvhxCmhhAuzBW5JcXz/DKuqlXL6c4VwEa1gRDCUSGEN9qwngFnSF8HIEtlndz77YDr09+n0riFJcTwOrAxEIhfvguAm0III7Ise62E9TcUQlg+y7L5Szn70cD/Au8gJqCxIYSXsiy7smUBJiGE5bIs+3eRslmWzWn1+pdWCGF14E5gDeBU4G/Aq8C2wNeAx4CblnU9WZYtBJ5Z1uUUEUI4BvghcAIxWQVgBPDxXDzzWxFPq5aTF0IIwJAsy+YCc1u57MrIskyvDn4BHwYyYMNupq0GXArMBt4A/gp8JDf9vWneA4DbUplpwH49rPMY4NW6cR9Ny9o1DR8KTAZeBp4DJgAb93bdwFDg56kOLxN3wh/KTR+ZlvMx4C/APOAoYHXgZ8CstOwngO83qdMKaTmfrxs/FbiqaEy5euVfN6Vp44mJ9hspnjeJP0gD8HXgnynWh4FvAYNz67wLuCC9Py6te/m6uE4jJqKQi+V64CVgDjFBjaib5wvAo2m9dwKfTTFv0+Sz+gkx+a3bzbRBwKrpfSAezT4OzE/b9yt15T8P3Ev8cfVC2oab132W29QNfxa4Mc0zDdi/bpmrABcCTwOvAVOAvXv4n74J+HkPZWr/a111w3sRfxDMTX/fA7w/1eW19HfTJstZrJ5p3LnAQ6mOTwL/A6xU/x0k/t/fC/wb2IPcdzO3nvzrIuBLxO9k/f/P94GH+nqf1levPg9Ar2XcgM2T4W+BR9KXZATw47TT2yhNr30JpwP7py/xOcQjyy2arLO7ZPihtKw90vDRxCOrjYGt085rKvHXa6F1AysRT5VdDXwQ2AQ4Pe10Nk5lal/4qWl9w4nJalzaCW4LbJA+pyOa1GmxZEjcqY8kJtef5so1jQkYDOyXlrUFsDawepp3PPAK4MCWxB1mAMYQE9I+Kf69gZnAybn15pNhFzG5fKauDv8CTk/vhxET5lhg8/R5X0z8cVCLZ0diQj49ff5G3PE2TIbE5P1KLZYe/je/QUwGh6fP6dgU9xfS9PWBBcBXU71HAIcA76v7H6lPhv8iJsR3A+elbbRBbrv9Cfg98X9yY+ArxGTx4Sax/pT4Y2TjJmUaJcPJwK7pc54C/B24A9gN2IyYIG9vspzukuGpwE7AhsRkOw24uO47uID4A3fXVM81WTwZLk880n2D+H+4NvGHwsppG+6fW94Q4v/cN/p6n9ZXrz4PQK9l3IANkmH6EmbA7rlxgZg0fpSGa1/Ck+vmvRu4pMk6F0uGwLuAScSjj9UbzLNOWtfWRded1vMoMKiuzJ+BMel9bcdSf0Q5CbioF59jLRnOJf7i/ncafhnYqq7uPcW0R5p37boy44kJ6u25casSd+a71ZUdBTyTG34rGabh64Brc8O1HyPvTsNjgNvqljmI+OPjmDR8DfCHujLfpHkyXD9N/3KBz/Q5YHTduB8DD6T3tWS8ToP5GyXDL+fKLJ8+v0Nz/w+vASvWLesXwPgmsa6bPuMsbd+rgSPrtlWjZDgyV+aQNO6TuXEHpnou32A5SyTDbuI7EHil7v8wA7bt4bt5FPBGN8sbB/w+N7xv+hy7in5nBtpL1wwHrs2IX8A/1kZkWZaFEO5M0/L+Ujf8Z+LRXDMrplZrgXh9bSrw2SzLXgAIIWxN/HX7fuIv1pDm24CY8Iqse1vizvfleEnkLW8jngLL+1vd8AXAL0MIOwJ/IJ4GuzlL3/wm/l8qO5R4quryLMvuyU3vTUzduT+L13VqtiTu0G8IIeRjGwysEEJYOcuyV7pZzpXA1SGENbMse564E/5TlmXTcnHu1E3LwrcTj9IgHon9um76H2muVummn2MIYS3iEewddZNuB44KISxHPKK6HXg4hHAz8XT5b7Ism9FDDG81qMmybH4IYTbxBxnEer8dmFW3fZYH7m+0wCzLpgM7hBA2A3YGdiAedX47hLBD1vya7b2597VrgffVjQvEz2Nmk+W8JYSwP/F0+MbEI7na/8MauVgWEo9Cl8ZFwJQQwkZZlj1KPJNzbZZls5dyeR1PybB6Aj3syFi0w2vmdeADaVmzsix7a6cbQlgVuDm9DiWemlueuNNYvhfrHkTc8R3QTbn6RjqLDWdZ9tsQwvrEayofAX4JTA4hjMyy7M0m638mJZRpIYTPAA+FEO7NsuzPSxFTd+rL1Fp070O8jlh0mf9LPNV1QAjhJ8RTnPnWpoOAicRTlfVeTH+L/C/Um0E8cq7/QdVI/fLf2r5Zli0IIewObE88mj4AODuEsG+WZTc3WWZ946iMRZ/jIOBZ4hmTevN6DDbLphJ/2F0UQjiLeN3uaODsJrPlG0FlTcYVar0fQtiFeCQ7mnia80XiqdBxLP79eSOLjYx6Lcuyv4cQ7ib+MLmQeLS619Isa6BQMhy4phK/fB8mto6rtTjbifhrPG+HWplkR3puzp7ljkLqbU5swHJilmWPpXXv3qBss3VPIV4bmtPDL/NGAc4GrgKuCiH8HLiV+Ev7XwXnnxlCuAQ4n9hat2hMtZ314AKruY+44xyeZdkfisSVYpsfQvgl8EXiEek7iNcia2pxPpk1blk7lfj/kFc/XL/eBWm9h4UQxqQjqreEEAYBK2dZ9mwI4TniTjxfr12Af2apFW06Ur8rvc4IIdwGHEb8IbU0pgBrERsRFdrOTdQaFq21jMvprZ2B6VmWnV4bEUI4eCmXNZ/G/4cXA98j/v89Rvx+VJbuMxyg0i/c3wLj0n1v7wN+RGx08F91xb8UQrAQwqYhhDHEI76xy7D6x4hfsONDCBuFEPYinnLsTrN1X0E8xXRDqsOGIYQdQginhBA+2SyAEMLZIYRPp+W+h3jN5WXikU1vnAdsFUL4XC9iejz9/WQIYa0QwiqNFp5OK58L/CDdv7lpCGHzEMJBIYQze4jtCmKS/k9gQpZlL+amnU9s7PObEMJOKc6dQwhjwqKbu38IfCSEcFpa737A8QU+k28RG9r8LYRwZAhhixDC8BDC54mnWXdM5cYA3wghHB5C2CSEcCzxOtxZACGE3UII3wkhbBdCWD/9n4wAHigQQyM3phgmhBD2SXFtE0L4WgjhsEYzhRAuTdvwwyGEDUII2xJbI69AvD5bpoeBYSGEQ9L35wjitb+l8RgwJITwiRBCVwhhxdy0q4k/ok4kXqfv7VmCgaWvL1rqtWwvit9aMY/Gt1YcSGxW/wbx1/D+Paxzidak3ZQ5kNiS9Q3ir/Vd0roO6M26gXcSm/I/TfyVO53Y8KPW4nSxxgi5+b5HPPJ5jXia6RZghybxdntrRZr2M+BB0q0OPcWUypxCvD70JnW3VjT5TO9L22kO8VrqUbnpizWgyY1/kLoGG7lpG7Go0c48YpK+ElgvV+aLxB3mPOL12s/RQ2OONN/KxNN4U4mNjmox/wewQioTgO/Q4NYK4vXkm4in0WvxfZ8lWxzXN6DZpi6W6cSzELXhFYEfEE871+7pmwjs0qQ++wM3EH8szUvzTAL2zJVp1ICmK1dmicZTxFalGelWlG6WU1/PQDwt+xzx/3cCixrmrN3sO9jdeOKP4OfS/BfVTftx+ozeVcb+qj+/avcjSQWFEN5L3Jlum2XZlKqsW0SiEMIEYF6WZfv1dSx9TdcMRUQqJoSwBrFh2Sfp4TpxVSgZiohUzwPE64WnZ1l2V18H0x/oNKmIiFSeWpOKiEjlKRmKiEjlDdRrhjr3KyIi3em2h62BmgyZObNQF4Bv6erqYvbsanTLp7oOTFWpa1XqCaprqw0dOrThNJ0mFRGRylMyFBGRylMyFBGRylMyFBGRylMyFBGRylMyFBGRylMyFBGRylMyFBGRylMyFBGRylMyFBGRylMyFBGRylMyFOkjC4/ep69DEJFEyVBERCpPyVBERCpPyVBERCpPyVBERCpPyVBERCpPyVBEpIUWHr2PWgq3UFmfpZKhiIhUnpKhiIhUnpKhiIhU3pCyVmRmI4GxwGDgEncfUzf9BOAoYAHwHHCEuz+Rpi0E7k9Fn3R3nZAXEZGWKSUZmtlg4EJgT2A6MNnMJrj7A7li9wDbuPvrZvYl4Bxg/zRtrrt/oIxYRUSkeso6MtwOmObujwKY2XhgX+CtZOjut+bK3wUcXFJsIiJScWUlw2HAU7nh6cD2TcofCdyYG17BzKYQT6GOcffr6mcws1HAKAB3p6urq1cBDhkypNfzdCrVtX+YBS2NrT/XtZX6ez1npb+tiLG/17WVGtW11d+Thutv+xqi0M24rLuCZnYwsA2wa270+u4+08w2Am4xs/vd/ZH8fO4+DhhXW/bs2bN7FWBXVxe9nadTqa79Rytj6+91bZVOqWcrYuyUurZCs7q26jMYOnRow2lltSadDqyXG14XmFlfyMz2AE4G9nH3ebXx7j4z/X0UuA3Yqp3BiohItZR1ZDgZ2MTMhgMzgAOAg/IFzGwr4GJgpLs/mxu/OvC6u88zsy5gJ2LjGhERkZYo5cjQ3RcAxwKTgAfjKJ9qZqPNrHabxLnASsCvzOwfZjYhjX8fMMXM7gVuJV4zfAAREZEWKe0+Q3efCEysG3dq7v0eDeb7M7BFe6MTEZEqUw80IiJSeUqGIiJSeUqGIiJSeUqGA5ieqyYiUoySoYiIVJ6SoYiIVJ6SoYiIVJ6SoYiIVJ6SYRNqgCIiUg1KhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlKhiIiUnlDihY0sz2BA4C13H1vM9sGWMXdb2lbdCIiIiUodGRoZscBPwb+BeySRs8FzmhTXCIiIqUpepr0a8Ae7j4GeDONewh4T1uiEhERKVHRZLgy8FR6n6W/ywHzWx6RiIhIyYpeM7wDOBE4MzfueODWoisys5HAWGAwcEk6ysxPPwE4ClgAPAcc4e5PpGmHAqekome4+xVF1ysiItKTokeGxwGfMbPHgZXN7GFgP+CEIjOb2WDgQuDjwAjgQDMbUVfsHmAbd98SuAY4J827BnAasD2wHXCama1eMG4REZEeFToydPenzWxbYjJan3jK9G/u/mbzOd+yHTDN3R8FMLPxwL7AA7l15I8y7wIOTu8/Btzs7nPSvDcDI4GrC65bRESkqcK3Vrh7Bvw1vXprGIuuOQJMJx7pNXIkcGOTeYctRQwiIiLdKpQMzewpFjWcWYy7r19gEaGbcd0uz8wOBrYBdu3NvGY2ChiVYqKrq6tAWIsMGTJkiXlmpb+9XVZ/0Sj+7uo6UPXnus6itf9b/bmurdTf69nK/UZ/r2srNaprq78nDddfsNzBdcPrAF8FxhecfzqwXm54XWBmfSEz2wM4GdjV3efl5t2tbt7b6ud193HAuDSYzZ49u2BoUVdXF43m6e2y+pv6+JvVdaDp73VtZWz9va6t0in1bEWMnVLXVihjHzx06NCG04peM7y9fpyZ3QbcRGwh2pPJwCZmNhyYQezJ5qC65W0FXAyMdPdnc5MmAWflGs3sBZxUJG4REZEilqVv0nnA8CIF3X0BcCwxsT0YR/lUMxttZvukYucCKwG/MrN/mNmENO8c4HvEhDoZGF1rTCMiItIKRa8Zjq4b9Q7gEyxq5NIjd58ITKwbd2ru/R5N5r0MuKzoukRERHqj6DXD9eqGXwN+CPysteGIiIiUr+g1w8PbHYiIiEhfaZgMzWz3IgvQI5xERKTTNTsyvLTA/BmwUYtiERER6RMNk6G7F2opKiIi0umW5dYKERGRAaHorRWrAN8ldpHWRa6LtILdsYmIiPRbRY8MfwR8EBgNrEF8pNOTwHltiktERKQ0RZPhXsDn3P16YGH6uz9wSNsiExERKUnRZDgIeCm9f9XMVgOeBt7dlqhERERKVLQHmnuJ1wv/ANxJfGr9q8A/2xSXiIhIaYoeGR4NPJ7eHw/MBVYDvtiGmERERErV9MjQzEa4+wPu/mhtnLs/BxzV9shERERK0tNp0nvM7D7gcmC8uz/f/pBERETK1VMyHAocSDwd+l9mdiMxMd6QnlEoIiLS8Zomw3QkeAFwgZm9h3grxfnAJWY2HrjS3Se3P0wREZH2KdqaFHd/GDgFOMXM9gTGAV8GBrcpNhERkVIUToYAZrYD8ZSpEe87HN2OoERERMrUYzI0sw2Ip0cPAd4FXAN8xt3vbHNsIiIipejp1orbgR2BW4HTgWvdfW4ZgYmIiJSlpyPDG4GD3H1GGcGIiIj0hZ5ak44pKxAREZG+oof7iohI5SkZiohI5SkZiohI5RW6z9DM7gGuAK5291ntDUlERKRcRW+6PwP4AnCmmd0B/AzdZiEiIgNEodOk7v5rd/8ssB5wPbEbtqfN7DIz272dAYqIiLRbr64Zuvsc4ErgIuBJ4HPAODP7p5nt0Yb4RERE2q7oNcNBwJ7ELtk+BfwFGEM6VWpmnwN+DqzdrkBFRETapeg1w5nAbOJR4bfcfWZ+orv/2syObXVwIiIiZSiaDD/l7lOaFXD3j7QgHhERkdIVvWb4u+5GmtmzLYxFRESkTxRNhsvVjzCz5dCDfUVEZADo6RFOdwIZsEK6vzBvXeDP7QpMRESkLD1dM7wECMC2wKW58RkwC7ilTXGJiIiUpqdHOF0BYGZ3uftD5YQkIiJSrobJ0MwOcfefpcEPmdmHuivn7pe1JTIREZGSNDsyPJDYBynEm+27kwFKhiIi0tEaJkN3/wSAmQXgSOBJd19QVmAiIiJl6fGme3fPzOx+YOVlWZGZjQTGEm/HuMTdx9RN3wU4H9gSOMDdr8lNWwjcnwafdPd9liUWERGRvKI90NwDbAosVSMaMxsMXEjs33Q6MNnMJrj7A7liTwKHAd/sZhFz3f0DS7NuERGRnhRNhrcBN5nZ5cBTxGuFQOEGNNsB09z9UQAzGw/sC7yVDN398TTtzYIxiYiItETRZLgT8Biwa934og1ohhGTaM10YPuC64Z40/8UYAEwxt2v68W8IiIiTRVKhi3ohDt0My7rZlwj67v7TDPbCLjFzO5390fyBcxsFDAKwN3p6urqVYBDhgxZYp5Z6W9vl9VfNIq/u7oOVP25rrNo7f9Wf65rK/X3erZyv9Hf69pKjera6u9Jw/X3dobUuvSt5ObuRU5rTgfWyw2vS3wsVCG1R0a5+6NmdhuwFfBIXZlxwLg0mM2ePbvo4oH4YTeap7fL6m/q429W14Gmv9e1lbH197q2SqfUsxUxdkpdW6GMffDQoUMbTiv6cN9hwAXALsBqdZOLdNY9GdjEzIYDM4ADgIMKrnt14HV3n2dmXcRTtucUmVdERKSIok+tuAiYD3wUeBX4IDABOKbIzOn+xGOBScCDcZRPNbPRZrYPgJlta2bTgf2Ai81sapr9fcAUM7sXuJV4zfCBJdciIiKydIqeJv0Q8brda2aWufu9ZnYk8akVPymyAHefCEysG3dq7v1k4unT+vn+DGxRME4REZFeK3pkuJDYkhPgRTN7J/AasZWoiIhIRyuaDP8KfCK9nwT8EvgNMKUdQYmIiJSp6GnSQ1iUOL9G7CVmJWL3aSIiIh2t6H2GL+bezwW+17aIREREStbseYajiywg3whGRESkEzU7MlyvyTQREZEBo9nzDA8vMxAREZG+0uw06Ya5J0ls1Khc7UkUIiIinarZadL8A32nETvWru9wO6NYd2wiIiL9VrPTpCvn3he9H1FERKTjKMmJiEjlFX1qxfrAacRHJ62Un+bum7YhLhERkdIU7YHmV8BDwKnA3PaFIyIiUr6iyfC9wI4FH+QrIiLSUYpeM/wtsGs7AxEREekrRY8Mjwf+bGaPALPyE9z9iJZHJSIiUqKiyfCnxGcaPoiuGYqIyABTNBnuDgx191faGYyIiEhfKHrN8D5gzXYGIiIi0leKHhneAvzOzH7KktcML2t5VCIiIiUqmgw/DMwA9qobnwFKhiIi0tGKPun+I+0OREREpK80e4RTcPcsvW94bVE34ouISKdrdmT4ErBKer+AeEo0L6BHOImIyADQLBlulns/vN2BiIiI9JVmzzN8Kvf+ifw0M1vd3V9oZ2AiIiJladqAxsy+CMxy90lpeBvgWmComU0D9nH3h9sfpoiISPv0dNP9N4BncsPjgN8DW6a/57YpLhERkdL0lAzXB+4HMLP1gC2Ab7j7VOBEYPv2hiciItJ+PSXDBcDy6f2HgIfcfU4afh14e7sCExERKUtPyfB24Ewz2xI4jvhcw5r3svgpVBERkY7UUzL8KrAV8CfikeDZuWmHADe1KS4REZHSNG1N6u4ziI9v6m7aiW2JSEREpGRFH+EkIiIyYCkZiohI5SkZiohI5SkZiohI5RV9uC9mtidwALCWu++dumZbxd1vaVt0IiIiJSh0ZGhmxwE/Bv4F7JJGzwXOaFNcIiIipSl6mvRrwB7uPgaoPcz3IeA9bYlKRESkREVPk64M1B7pVHvI73LA/KIrMrORwFjiw4AvSYk1P30X4HxiJ+AHuPs1uWmHAqekwTPc/Yqi6xUREelJ0SPDO4gdc+cdD9xaZGYzGwxcCHwcGAEcaGYj6oo9CRwG/KJu3jWA04idgm8HnGZmqxeMW0REpEdFk+FxwGfM7HFgZTN7GNgPOKHg/NsB09z9UXefD4wH9s0XcPfH3f0+Fp2GrfkYcLO7z0kPFL4ZGFlwvSIiIj0qlAzd/WlgW2B/4CDgUGB7dy/aUfcwFp1mBZiexrV7XhERkR4VvrXC3TPgr2Y2uTbOzAa5e/2RXHdCN+OybsYt9bxmNgoYlWKlq6ur4OKjIUOGLDHPrPS3t8vqLxrF311dB6r+XNdZtPZ/qz/XtZX6ez1bud/o73VtpUZ1bfX3pOH6ixQysw8Sr/ltCayQRgdiUhpcYBHTgfVyw+sCMwvGOB3YrW7e2+oLufs4YFwazGbPnl1w8VFXVxeN5untsvqb+vib1XWg6e91bWVs/b2urdIp9WxFjJ1S11YoYx88dOjQhtOKHhleQXyW4RHERzn11mRgEzMbDswg3rx/UMF5JwFn5RrN7AWctBQxiIiIdKtoMtwAODmdKu01d19gZscSE9tg4DJ3n2pmo4Ep7j7BzLYFrgVWB/Y2s9PdfTN3n2Nm3yMmVIDR7j5naeIQERHpTtFkeC3xiGzS0q7I3ScCE+vGnZp7P5l4CrS7eS8DLlvadYuIiDRTNBmuAFxrZn8EFmtB6u5fbHlUIiIiJSqaDB9ILxERkQGnUDJ099PbHYiIiEhfaZgMzWwXd78jvd+9UTk9wklERDpdsyPDHwGbp/eXNiiTARu1NCIREZGSNUyG7r557v3wcsIREREpX9GOuhdjZh8xs51bHYyIiEhfKPqk+9vNbKf0/tvEp06MN7PvtDM4ERGRMhQ9MtwcuCu9P5rYV+gOwDFtiElERKRURe8zHARkZrYxENz9QQA9ZFdERAaCosnwj8AFwDrErtlIibEa3amLiMiAVvQ06WHAi8B9wHfTuPcCY1sfkoiISLmK9kDzPPCdunE3tCUiERGRkhV9uO9ywCnAIcBQ4oN5fwac6e7z2xeeiIhI+xW9ZngOsB2x9egTxOcb/iewCvD19oQmIiJSjqLJcD/g/el0KcDDZvZ34F6UDEVEpMMVbUATejleRESkYxQ9MvwV8FszOx14knia9BTA2xWYiIhIWYomw28Rk9+FxAY0M4hdsp3RprhERERKU/TWivnAqeklIiIyoDS9ZmhmO5nZ2Q2mjTGzHdoTloiISHl6akDzHeCOBtNuB05ubTgiIiLl6ykZfgC4qcG0m4GtWxuOiIhI+XpKhqsAyzeYthywcmvDERERKV9PyfAhYK8G0/ZK00VERDpaT61JzwMuNrPBwHXu/qaZDQI+TbzN4oR2BygiItJuTY8M3f0XxH5JrwDeMLOZwBvA5cA57n512yMUERFpsx67Y3P3HwLDgL2Bb6a/67r7eW2OTUREpBRFb7p/GZjU5lhERET6RNGOukVERAYsJUMREak8JUMREak8JUMREak8JUMREak8JUMREak8JUMREak8JUMREak8JUMREak8JUMREak8JUMREam8Qn2TtoKZjQTGAoOBS9x9TN30twFXAlsDzwP7u/vjZrYh8CDwcCp6l7sfU1bcIiIy8JWSDNPzEC8E9gSmA5PNbIK7P5ArdiTwgru/28wOAM4G9k/THnH3D5QRq4iIVE9Zp0m3A6a5+6PuPh8YD+xbV2Zf4nMTAa4BPmpmoaT4RESkwspKhsOAp3LD09O4bsu4+wLgJWDNNG24md1jZreb2c7tDlZERKqlrGuG3R3hZQXLPA2s7+7Pm9nWwHVmtll6xuJbzGwUMArA3enq6upVgEOGDFlinlnpb2+X1V80ir+7ug5U/bmus2jt/1Z/rmsr9fd6tnK/0d/r2kqN6trq70nD9bd9DdF0YN7aIzQAAA2VSURBVL3c8LrAzAZlppvZEGBVYI67Z8A8AHe/28weATYFpuRndvdxwLg0mM2ePbtXAXZ1ddFont4uq7+pj79ZXQea/l7XVsbW3+vaKp1Sz1bE2Cl1bYUy9sFDhw5tOK2sZDgZ2MTMhgMzgAOAg+rKTAAOBf4CfB64xd0zM3snMSkuNLONgE2AR0uKW0REKqCUa4bpGuCxwCTibRLu7lPNbLSZ7ZOKXQqsaWbTgBOAE9P4XYD7zOxeYsOaY9x9Thlxi4hINZR2n6G7TwQm1o07Nff+DWC/bub7NfDrtgcoIiKVpR5oRESk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8pQMRUSk8oaUtSIzGwmMBQYDl7j7mLrpbwOuBLYGngf2d/fH07STgCOBhcDx7j6prLhFRGTgK+XI0MwGAxcCHwdGAAea2Yi6YkcCL7j7u4HzgLPTvCOAA4DNgJHAj9LyREREWqKs06TbAdPc/VF3nw+MB/atK7MvcEV6fw3wUTMLafx4d5/n7o8B09LyREREWqKsZDgMeCo3PD2N67aMuy8AXgLWLDiviIjIUivrmmHoZlxWsEyReTGzUcAoAHdn6NChvY1xyXlumNLrZfQrTeJfms+nU/Xburbh/6vf1rXF+nU9W7xd+3VdW6zbupa0Hy7ryHA6sF5ueF1gZqMyZjYEWBWYU3Be3H2cu2/j7tsQE2ivXmZ299LM14kv1XVgvqpS16rUU3Vt26tbZR0ZTgY2MbPhwAxig5iD6spMAA4F/gJ8HrjF3TMzmwD8wsx+CAwFNgH+VlLcIiJSAaUcGaZrgMcCk4AH4yifamajzWyfVOxSYE0zmwacAJyY5p0KOPAAcBPwFXdfWEbcIiJSDaXdZ+juE4GJdeNOzb1/A9ivwbxnAme2NUAY1+bl9yeq68BUlbpWpZ6gupYmZNkSbVFEREQqRd2xiYhI5ZV2mrS/6qmbuE5jZusRu7VbG3gTGOfuY81sDeCXwIbA44C5+wupY4OxwCeA14HD3P3vfRH70ko9Ek0BZrj7p1JDrfHAGsDfgUPcfX6zLv86gZmtBlwCbE68vegI4GEG4HY1s68DRxHreT9wOLAOA2C7mtllwKeAZ9198zSu199PMzsUOCUt9gx3v6LMehTRoK7nAnsD84FHgMPd/cU0rduuN8vYT1f6yLBgN3GdZgHwDXd/H7AD8JVUpxOBP7j7JsAf0jDEum+SXqOAH5cf8jL7KrFhVs3ZwHmpri8Qv1zQoMu/DjIWuMnd3wu8n1jnAbddzWwYcDywTdqBDia2QB8o2/VyYteSeb3ajil5ngZsT+yR6zQzW73tkffe5SxZ15uBzd19S+CfwEnQuOvNsvbTlU6GFOsmrqO4+9O1X47u/gpxhzmMxbu7uwL4dHq/L3Clu2fufhewmpmtU3LYS83M1gU+STxiIv2S3p3YpR8sWdfaZ5Dv8q/fM7NVgF2Ira5x9/np1/SA3K7Es1ZvT/ccvwN4mgGyXd39DuI91Hm93Y4fA2529znu/gIxwdQnnT7XXV3d/XfpDgOAu4j3jkPjrjdL2U9XPRkO6K7ezGxDYCvgr8C73P1piAkTWCsV6/TP4HzgW8RTwhC78Hsx92XL16dRl3+dYCPgOeCnZnaPmV1iZisyALeru88AfgA8SUyCLwF3MzC3a01vt2PHbt86RwA3pvd9WteqJ8Pufj0OiOa1ZrYS8Gvga+7+cpOiHfsZmFntWsTdudHN6tOxdSUeKX0Q+LG7bwW8xqJTad3p2Lqm0337AsOJHW2sSDxFVm8gbNeeNKpbx9fZzE4mXta5Ko3q07pWPRkW6uqt05jZcsREeJW7/yaNnlU7TZb+PpvGd/JnsBOwj5k9Tjx1sjvxSHG1dHoNFq9Poy7/OsF0YLq7/zUNX0NMjgNxu+4BPObuz7n7v4HfAB9iYG7Xmt5ux07evrXGP58CvuDutcTWp3WtemvSIt3EdZR0reRS4EF3/2FuUq27uzHp7/W58cea2XjixfiXaqdr+jt3P4lFF993A77p7l8ws18Ru/Qbz5J1XaLLv7LjXhru/oyZPWVm73H3h4GPEntleoABtl2Jp0d3MLN3AHOJdZ0C3MoA2645vfp+mtkk4Kxco5m9SN+F/i61DP02sKu7v56b1KjrzUAJ++lKJ0N3X2BmtW7iBgOXpe7fOtlOwCHA/Wb2jzTuO8QvmZvZkcSdTa23n4nEZtvTiE23Dy833Lb4NjDezM4A7iE1Okl/f5a6/JtD/FJ1kuOAq8xseeBR4rYaxADbru7+VzO7hnj7xALiNhwH3MAA2K5mdjWwG9BlZtOJrUJ79f109zlm9j3iD3qA0e7e746GG9T1JOBtwM1mBnCXux+Tuuisdb25gFzXm2Xsp9UDjYiIVF7VrxmKiIgoGYqIiCgZiohI5SkZiohI5SkZiohI5SkZivQRM8vM7N0tXubOZvZwK5cpUgWVvs9QOkPqYeZdxMe6vEa89+o4d3+1j+M6DDjK3T/cl3HkufudwHuWZt5Un0uJN7pD7Av1NuD77v7Pgsu4nNhTzik9lW0w/3eBk4F5udGj3f2cpVmeSFE6MpROsbe7r0TsgmxbFj3HrRS5bsAGur+kz3lVYrdoc4G7zWzzEmP4pbuvlHstkQjNLJiZ9l/SMlX5gssA4e4zzOxG4gNuMbPDiU+tWJd4JHO2u1+cpu0G/Bz4EXAC8Cpwsrtflaa/DTgTMGKPGNcCX3f3ubl5/wf4OvEROYc0iy0dwV4AfBHYALgJONTd30jT/1+KI6MumfcQy7eBzwAfTr0mfQk4Fti6tuzccnYDfu7u6xaJqZHU88cjwJfNbH3gu8Suzkjd3e0MvB24F/hS6j1kFPAFIDOzrwG3uvveZnYicDTxSQxPEbfBtc3W3x0zuw34E7FHkw8CW5jZzvS8/f8b+CbxzMKXiA+VPR/oAn7g7mel8oPSso4GViM+V/CY/tizi7SefllJRzGz9YjdU92TRj1L7PB3FWJXVeeZ2Qdzs6xN3OkNI/b5OM7MaqcRzwY2BT4AvDuVObVu3jWISWRU0RCJz5UbDmwJHJbiHkncIe9J7HNxj7r5msVyLnEHfoqZbQKcBRzcU0LrKaZe+A0x+dXcmOqwFrHLtKsA3H1cen9OOqLbO5V/JM2/KnA68PNleLbiIcRtsTLwBMW2/wos+jx/AhwMbJ1iOtXMNkpljyc+R3BXYt+YLxAfKisVoCND6RTXmVntWXU3EBMC7n5DrsztZvY74k7u77nx/+nu89L0GwBL/VseDWxZ++VvZmcBv2BRh8dvAqeleYv6b3efmZb3W2Jyg5iQfuru/5emfRc4ML0PzWJx9zfN7IupTvsTk809FNcopqJmEn8UAODul9Xep3q8YGaruvtL3c3s7r/KDf7SzE4iPrD1+u7Kx8Xap3LDI2rxA5fX9UvZ0/b/N3Cmuy9MnV2PA8amB19PNbOpxB8IjwL/ARzr7tNzdXvSzA7JPUdRBiglQ+kUn3b339ePNLOPEzv/3ZR4puMdwP25Ii+4+2u54SeIv/rfmcrenToLhtg7/uBc2ed6cfRV80zu/etpXaS/+ecuPpF732Ms7v64md1KPCru7dFKo5iKGkZ6JJKZDSaezt0vxV17qHIX8YfKElIiPwHYMI1aKZVvxN394AbT8g95LbL9n6919syihkGzctPnpnggngG41szezE1fSGy8NaNJvDIAKBlKx0rX2X5NvB52vbv/28yuY/GHga5uZivmEuL6wP8Bs4k7ws3Sk9W708pe7J9m8WeyrZ9732MsZvYJYEfidaxziUcxZfkMcGd6fxDxwbt7AI8TT32+wKLPfLHPzMw2IJ6a/Cixcc7C9DSV7h7YWsRbyy+4/XvjKeAId//TUs4vHUzXDKWTLU9sbPIcsCAdJezVTbnTzWz51NjiU8Cv3P1N4k76PDNbC8DMhpnZx9oUqwOHmdmI9Jy+096a0EMsZtZFvOXhKOJ1z71TcmwbMxtsZsPN7H+IDVZOT5NWJt728DzxKOysullnARvlhlckJrDn0nIPJzV+aoGi27+oi4AzUwLHzN5pZvsue5jSCZQMpWOl6z7HExPNC8Sjlgl1xZ5J02YSG3cc4+4PpWnfJj4n7i4zexn4PUt5j16BWG8ktmC8Ja3zlroizWIZRzzymejuzwNHApeY2ZptCHVHM3sVeJl4j+EqwLbuXjv1eCXxFO8M4nPn7qqb/1JghJm9aGbXufsDwH8RH7w7C9iC2CJ0mRXc/r0xNs3/OzN7hVi37Zc1TukMep6hDFj1txmIiDSiI0MREak8JUMREak8nSYVEZHK05GhiIhUnpKhiIhUnpKhiIhUnpKhiIhUnpKhiIhUnpKhiIhU3v8Hp3XNvOWnnWIAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 504x360 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "if (flag == False):\n",
    "    values = [] # a list which will hold the cosine value of each paper outputted\n",
    "    for i in answer_indicies:\n",
    "        values.append(cosineSimilarities[i])\n",
    "    print(values)\n",
    "    print(answer_indicies)\n",
    "    plt.figure(figsize=(7, 5))\n",
    "    plt.bar(answer_indicies, values,width=6)\n",
    "    plt.xlabel('Paper Index in Data Frame')\n",
    "    plt.ylabel('Cosine Similarity Value')\n",
    "    plt.title('Top Papers Retreived Cosine Similarity')\n",
    "    plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<p>In the below cell, the code verifies whether or not the cosine similarity is >=0.1.  </p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>A spatial model of CoVID-19 transmission in England and Wales: early spread and peak timing</td>\n",
       "      <td>[{'text': 'An outbreak of a novel coronavirus, named CoVID-19, was first reported in China on 31 December 2019. As of 9 February 2020, cases have been reported in 25 countries, including probable cases of human-to-human transmission in England.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'We adapted an existing national-scale metapopulation model to capture the spread of CoVID-19 in England and Wales. We used 2011 census data to capture population sizes and population movement, together with parameter estimates from the current outbreak in China.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'We predict that a CoVID-19 outbreak will peak 126 to 147 days (~4 months) after the start of person-to-person transmission in England and Wales in the absence of controls, assuming biological parameters remain unchanged. Therefore, if person-to-person transmission persists from February, we predict the epidemic peak would occur in June. The starting location has minimal impact on peak timing, and model stochasticity varies peak timing by 10 days. Incorporating realistic parameter uncertainty leads to estimates of peak time ranging from 78 days to 241 days after person-to-person transmission has been established. Seasonal changes in transmission rate substantially impact the timing and size of the epidemic peak, as well as the total attack rate.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'We provide initial estimates of the potential course of CoVID-19 in England and Wales in the absence of control measures. These results can be refined with improved estimates of epidemiological parameters, and permit investigation of control measures and cost effectiveness analyses. Seasonal changes in transmission rate could shift the timing of the peak into winter months, which will have important implications for health-care capacity planning.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>An outbreak of a novel coronavirus, recently renamed CoVID-19, was first reported from Wuhan, China on 31 December 2019. During January 2020, the outbreak spread to multiple cities in China, and the first cases started appearing outside China. By the end of January 2020, 9,720 cases had been confirmed in China, with 106 confirmed cases outside China across 19 different countries (1) .\\nEpidemiological analysis of the outbreak was quickly used to start estimating epidemiologicallyrelevant parameters, such as the basic reproduction number, the serial interval, the incubation period and the case fatality rate (2) (3) (4) (5) (6) (7) . Initial estimates suggested that the reproduction number was between 2 and 3 and the case fatality rate was less than 4% (8) . Control of spread by contact tracing and isolation appears to be challenging, given what is currently known about the virus (9) .\\nMathematical models are useful tools for understanding and predicting the possible course of an outbreak, given a set of underlying assumptions. Here, we adapt a metapopulation model of disease transmission in England and Wales to capture the spread of CoVID-19 (10) . The aim is to provide predictions about the likely timing of the peak of the epidemic in England and Wales and spatial features of spread.\\nWe use an existing national-scale stochastic metapopulation model of disease transmission in England and Wales. The model structure is based on the metapopulation model described in detail in Danon et al (2009) (10) . In this model, the population is divided into electoral wards. Because of the changes in data availability, we restricted the model to England and Wales, whereas the original model covered Great Britain.\\nTransmission between wards occurs via the daily movement of individuals. For each ward, we assume that individuals contribute to the force of infection in their \"home\" ward during the night and their \"work\" ward during the day. See Danon et al 2009 for further details (10) .\\nData for population and movement of individuals come from the 2011 census of the United Kingdom. The population size of each of the 8,570 electoral wards is available directly from the Office of National Statistics (ONS) website. The number of individuals moving between locations is also available from the ONS website, but at the level of census output areas (OAs). We aggregated the data from OA level to electoral wards level. Spatial location of electoral ward centres are extracted from maps available from the ONS websites.\\nWe use a Susceptible-Exposed-Infectious-Infectious-Recovered (SEIIR) model within each ward to capture the progression of disease within an individual (figure 1). Initial analyses used SARS-like parameters for the incubation period and infectious period, which now appear to differ from CoVID-19(4). Li et al (2020) analysed data on 425 cases reported in Wuhan in China and fitted a log-normal distribution to the incubation period, and a gamma distribution for the serial interval (2) . The infectious period for SARS was estimated as the serial interval minus the . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint incubation period, but as Li et al did not report the correlation between incubation period and serial interval, we were not able to estimate the infectious period distribution from the data, but used a uniform distribution between 2 and 3 days, to give a mean serial interval of approximately 7-8, in line with current estimates. We used two infectious states to represent a mildly symptomatic or prodromal period and a period with more pronounced symptoms. In the absence of data on the relative magnitude of these two infections states, we assumed the same length of time in each infectious state and assumed that each state was equally infectious. We sampled from each of the distributions 100 times independently (Table 1 ). \\nThe census data are used to initialise the population sizes within each of the 8,570 wards. At the start of the model, all individuals are assumed to be susceptible to infection with no 3 e . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint underlying immunity in the population. We investigated a range of starting scenarios by seeding the infection in example wards London, Birmingham, Brighton, Sheffield and Cardiff. To seed infection in a ward, we move five individuals (non-commuters) from the susceptible compartment to the first infectious state.\\nWe investigated the impact of a seasonally affected transmission rate, to capture potential decreased transmission during the summer months. We captured seasonal transmission by replacing the constant transmission rate with a time-varying transmission rate given by:\\nis the magnitude of the seasonal difference in transmission, ranging from\\n(maximum seasonality with no transmission at the peak of the summer).\\nFrom the model, we extracted the total number of infections per day, as the number of individuals in both of the Infectious states, and the number of infected wards per day as the total number of wards with at least one individual in one of the two Infectious states. The spatial growth of the epidemic in England and Wales was visualised using interactive maps. We estimated the timing of the epidemic peak from the aggregated epidemic curve and calculated 95% prediction intervals from the model simulations.\\nThe model is coded in C and is available on github (http://github.com/ldanon/MetaWards). The data are freely available from the Office for National Statistics website, or can be downloaded with the code at the github repository.\\nWe predict that, in the absence of any interventions, a disease with \"best-guess\" CoVID-19-like parameters will peak a median of 133 days (range 126 -147 days) following the start of personto-person transmission in England and Wales. Intrinsic model stochasticity is responsible for variation between model runs. Using exactly the same parameters and seeding the infection in the same initial wards resulted in a difference in peak timing of +/-10 days (figure 2). The attack rate for best-guess parameters had a median of 45799874 (81.67% range 81.64-81.69), with a peak incidence median 1,116,692.\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint\\nThe initial location of cases had some, but limited impact on the timing of the epidemic in England and Wales. Epidemics seeded in Brighton, London, Birmingham and Sheffield resulted in synchronised epidemics in England, reaching urban areas first followed by rural areas. Epidemics started in Cardiff had a slower time to peak but still resulted in a generalised outbreak.\\nSpatially, some disaggregation between England and Wales regions is observed. An outbreak starting in Brighton, (South East England) peaks in London and the South East first, and North East England, Yorkshire and Humber and Wales last, with a ten-day lag between regions ( Figure 3 ).\\nModel predictions are highly sensitive to parameter values and incorporating parameter uncertainty increases the variability of model predictions. In the absence of any control measures, all predictions resulted in epidemics that peaked within a year from the start of person-to-person transmission in England and Wales. Estimates of peak time ranged from 78 days to 241 days (Figure 4 ).\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint However, seasonality in transmission has a large impact on epidemic timing, peak incidence, and final attack rates. Assuming no difference in transmission rate during the year leads to a single large epidemic after approximately 4 months (June time if transmission starts in February), as above. With a 25% reduction in transmission the epidemic is smaller and peaks later, reducing the overall attack rate by 20%. A 50% reduction in transmission results in a smaller epidemic before the summer, followed by a resurgence in cases in the following winter. The attack rate is 10% less than a non-seasonal epidemic. A 75% reduction in transmission over the summer resulted in a delayed large outbreak, but with a similar attack rate. If transmission decreases to zero over the summer then the resulting outbreak is much reduced, with an attack rate of less than 1% ( Figure 5 , Table 2 ). . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint Table 2 : Effect of seasonal variation on the timing (shown in days following initial seeding), the height of the peak, and the final attack rate.\\nWe predict that, in the absence of control measures and with no seasonality in transmission, the introduction of CoVID-19 in England and Wales has the potential to result in a synchronised outbreak that peaks at around 4 months following the start of person-to-person transmission. Our findings suggest that the height of the epidemic and the attack rate is highly dependent on seasonality of transmission and that even small changes in transmission risk can lead to large changes in attack rate due to the spatial disaggregation of the population at risk.\\nA combination of control measures and seasonal changes in transmission rate could shift the peak of the outbreak to the winter of 2020/21, with little effect on the final attack rate. If contact tracing and isolation efforts succeed in reducing transmission, but are unable to control the epidemic (9), an additional influx of severe CoVID-19 cases may exacerbate existing challenges with winter healthcare demand. A careful analysis of the impact of control measures on the timing of incidence of severe cases is warranted.\\nThe strength of this model lies in the spatial heterogeneity which tempers transmission. As a comparison, an equivalent non-spatial model results in the epidemic peaking after 34 days, nearly four times faster than this spatial model, and would be unable to capture the interaction between spatial transmission and seasonality. The estimated total number of people infected in the spatial model is marginally smaller than for a non-spatial model, as the infection has the opportunity to die-out in local parts of the country. As the model framework was developed and published in 2009, it was possible to re-deploy the model for these new circumstances; developing such a model from scratch during an outbreak would be a significant challenge.\\nA key element missing from our model is morbidity, mortality and the treatment of cases. The model in its current form predicts the total number of infections in the community rather than diagnosed cases. Observations from China suggest that many cases have mild symptoms and that only around 5% of cases have been reported and diagnosed (3) . The parameter estimates we used from China appear to be substantially different to previous coronaviruses (6) . Should CoVID-19 continue spreading the UK it will become possible to get UK-specific parameter estimates and improve prediction accuracy.\\nAs with all modelling, it is impossible to capture the full complexity of an epidemic. In this model, the major assumptions are that we have assumed that there is no change in behaviour during the course of the epidemic. In practice, as the epidemic starts spreading in England and Wales there may well be a systematic change in behaviour as was seen during the H1N1 influenza pandemic in 2009. We have not included any age-effects, such as differential mixing, susceptibility or infectiousness. That means that we are not able to investigate the impact of school closures or the impact of the summer holidays, which had a large impact on the H1N1 influenza pandemic in 2009.\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                          title  \\\n",
       "21  A spatial model of CoVID-19 transmission in England and Wales: early spread and peak timing   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               abstract  \\\n",
       "21  [{'text': 'An outbreak of a novel coronavirus, named CoVID-19, was first reported in China on 31 December 2019. As of 9 February 2020, cases have been reported in 25 countries, including probable cases of human-to-human transmission in England.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'We adapted an existing national-scale metapopulation model to capture the spread of CoVID-19 in England and Wales. We used 2011 census data to capture population sizes and population movement, together with parameter estimates from the current outbreak in China.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'We predict that a CoVID-19 outbreak will peak 126 to 147 days (~4 months) after the start of person-to-person transmission in England and Wales in the absence of controls, assuming biological parameters remain unchanged. Therefore, if person-to-person transmission persists from February, we predict the epidemic peak would occur in June. The starting location has minimal impact on peak timing, and model stochasticity varies peak timing by 10 days. Incorporating realistic parameter uncertainty leads to estimates of peak time ranging from 78 days to 241 days after person-to-person transmission has been established. Seasonal changes in transmission rate substantially impact the timing and size of the epidemic peak, as well as the total attack rate.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'We provide initial estimates of the potential course of CoVID-19 in England and Wales in the absence of control measures. These results can be refined with improved estimates of epidemiological parameters, and permit investigation of control measures and cost effectiveness analyses. Seasonal changes in transmission rate could shift the timing of the peak into winter months, which will have important implications for health-care capacity planning.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   full_text  \n",
       "21  An outbreak of a novel coronavirus, recently renamed CoVID-19, was first reported from Wuhan, China on 31 December 2019. During January 2020, the outbreak spread to multiple cities in China, and the first cases started appearing outside China. By the end of January 2020, 9,720 cases had been confirmed in China, with 106 confirmed cases outside China across 19 different countries (1) .\\nEpidemiological analysis of the outbreak was quickly used to start estimating epidemiologicallyrelevant parameters, such as the basic reproduction number, the serial interval, the incubation period and the case fatality rate (2) (3) (4) (5) (6) (7) . Initial estimates suggested that the reproduction number was between 2 and 3 and the case fatality rate was less than 4% (8) . Control of spread by contact tracing and isolation appears to be challenging, given what is currently known about the virus (9) .\\nMathematical models are useful tools for understanding and predicting the possible course of an outbreak, given a set of underlying assumptions. Here, we adapt a metapopulation model of disease transmission in England and Wales to capture the spread of CoVID-19 (10) . The aim is to provide predictions about the likely timing of the peak of the epidemic in England and Wales and spatial features of spread.\\nWe use an existing national-scale stochastic metapopulation model of disease transmission in England and Wales. The model structure is based on the metapopulation model described in detail in Danon et al (2009) (10) . In this model, the population is divided into electoral wards. Because of the changes in data availability, we restricted the model to England and Wales, whereas the original model covered Great Britain.\\nTransmission between wards occurs via the daily movement of individuals. For each ward, we assume that individuals contribute to the force of infection in their \"home\" ward during the night and their \"work\" ward during the day. See Danon et al 2009 for further details (10) .\\nData for population and movement of individuals come from the 2011 census of the United Kingdom. The population size of each of the 8,570 electoral wards is available directly from the Office of National Statistics (ONS) website. The number of individuals moving between locations is also available from the ONS website, but at the level of census output areas (OAs). We aggregated the data from OA level to electoral wards level. Spatial location of electoral ward centres are extracted from maps available from the ONS websites.\\nWe use a Susceptible-Exposed-Infectious-Infectious-Recovered (SEIIR) model within each ward to capture the progression of disease within an individual (figure 1). Initial analyses used SARS-like parameters for the incubation period and infectious period, which now appear to differ from CoVID-19(4). Li et al (2020) analysed data on 425 cases reported in Wuhan in China and fitted a log-normal distribution to the incubation period, and a gamma distribution for the serial interval (2) . The infectious period for SARS was estimated as the serial interval minus the . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint incubation period, but as Li et al did not report the correlation between incubation period and serial interval, we were not able to estimate the infectious period distribution from the data, but used a uniform distribution between 2 and 3 days, to give a mean serial interval of approximately 7-8, in line with current estimates. We used two infectious states to represent a mildly symptomatic or prodromal period and a period with more pronounced symptoms. In the absence of data on the relative magnitude of these two infections states, we assumed the same length of time in each infectious state and assumed that each state was equally infectious. We sampled from each of the distributions 100 times independently (Table 1 ). \\nThe census data are used to initialise the population sizes within each of the 8,570 wards. At the start of the model, all individuals are assumed to be susceptible to infection with no 3 e . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint underlying immunity in the population. We investigated a range of starting scenarios by seeding the infection in example wards London, Birmingham, Brighton, Sheffield and Cardiff. To seed infection in a ward, we move five individuals (non-commuters) from the susceptible compartment to the first infectious state.\\nWe investigated the impact of a seasonally affected transmission rate, to capture potential decreased transmission during the summer months. We captured seasonal transmission by replacing the constant transmission rate with a time-varying transmission rate given by:\\nis the magnitude of the seasonal difference in transmission, ranging from\\n(maximum seasonality with no transmission at the peak of the summer).\\nFrom the model, we extracted the total number of infections per day, as the number of individuals in both of the Infectious states, and the number of infected wards per day as the total number of wards with at least one individual in one of the two Infectious states. The spatial growth of the epidemic in England and Wales was visualised using interactive maps. We estimated the timing of the epidemic peak from the aggregated epidemic curve and calculated 95% prediction intervals from the model simulations.\\nThe model is coded in C and is available on github (http://github.com/ldanon/MetaWards). The data are freely available from the Office for National Statistics website, or can be downloaded with the code at the github repository.\\nWe predict that, in the absence of any interventions, a disease with \"best-guess\" CoVID-19-like parameters will peak a median of 133 days (range 126 -147 days) following the start of personto-person transmission in England and Wales. Intrinsic model stochasticity is responsible for variation between model runs. Using exactly the same parameters and seeding the infection in the same initial wards resulted in a difference in peak timing of +/-10 days (figure 2). The attack rate for best-guess parameters had a median of 45799874 (81.67% range 81.64-81.69), with a peak incidence median 1,116,692.\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint\\nThe initial location of cases had some, but limited impact on the timing of the epidemic in England and Wales. Epidemics seeded in Brighton, London, Birmingham and Sheffield resulted in synchronised epidemics in England, reaching urban areas first followed by rural areas. Epidemics started in Cardiff had a slower time to peak but still resulted in a generalised outbreak.\\nSpatially, some disaggregation between England and Wales regions is observed. An outbreak starting in Brighton, (South East England) peaks in London and the South East first, and North East England, Yorkshire and Humber and Wales last, with a ten-day lag between regions ( Figure 3 ).\\nModel predictions are highly sensitive to parameter values and incorporating parameter uncertainty increases the variability of model predictions. In the absence of any control measures, all predictions resulted in epidemics that peaked within a year from the start of person-to-person transmission in England and Wales. Estimates of peak time ranged from 78 days to 241 days (Figure 4 ).\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint However, seasonality in transmission has a large impact on epidemic timing, peak incidence, and final attack rates. Assuming no difference in transmission rate during the year leads to a single large epidemic after approximately 4 months (June time if transmission starts in February), as above. With a 25% reduction in transmission the epidemic is smaller and peaks later, reducing the overall attack rate by 20%. A 50% reduction in transmission results in a smaller epidemic before the summer, followed by a resurgence in cases in the following winter. The attack rate is 10% less than a non-seasonal epidemic. A 75% reduction in transmission over the summer resulted in a delayed large outbreak, but with a similar attack rate. If transmission decreases to zero over the summer then the resulting outbreak is much reduced, with an attack rate of less than 1% ( Figure 5 , Table 2 ). . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint Table 2 : Effect of seasonal variation on the timing (shown in days following initial seeding), the height of the peak, and the final attack rate.\\nWe predict that, in the absence of control measures and with no seasonality in transmission, the introduction of CoVID-19 in England and Wales has the potential to result in a synchronised outbreak that peaks at around 4 months following the start of person-to-person transmission. Our findings suggest that the height of the epidemic and the attack rate is highly dependent on seasonality of transmission and that even small changes in transmission risk can lead to large changes in attack rate due to the spatial disaggregation of the population at risk.\\nA combination of control measures and seasonal changes in transmission rate could shift the peak of the outbreak to the winter of 2020/21, with little effect on the final attack rate. If contact tracing and isolation efforts succeed in reducing transmission, but are unable to control the epidemic (9), an additional influx of severe CoVID-19 cases may exacerbate existing challenges with winter healthcare demand. A careful analysis of the impact of control measures on the timing of incidence of severe cases is warranted.\\nThe strength of this model lies in the spatial heterogeneity which tempers transmission. As a comparison, an equivalent non-spatial model results in the epidemic peaking after 34 days, nearly four times faster than this spatial model, and would be unable to capture the interaction between spatial transmission and seasonality. The estimated total number of people infected in the spatial model is marginally smaller than for a non-spatial model, as the infection has the opportunity to die-out in local parts of the country. As the model framework was developed and published in 2009, it was possible to re-deploy the model for these new circumstances; developing such a model from scratch during an outbreak would be a significant challenge.\\nA key element missing from our model is morbidity, mortality and the treatment of cases. The model in its current form predicts the total number of infections in the community rather than diagnosed cases. Observations from China suggest that many cases have mild symptoms and that only around 5% of cases have been reported and diagnosed (3) . The parameter estimates we used from China appear to be substantially different to previous coronaviruses (6) . Should CoVID-19 continue spreading the UK it will become possible to get UK-specific parameter estimates and improve prediction accuracy.\\nAs with all modelling, it is impossible to capture the full complexity of an epidemic. In this model, the major assumptions are that we have assumed that there is no change in behaviour during the course of the epidemic. In practice, as the epidemic starts spreading in England and Wales there may well be a systematic change in behaviour as was seen during the H1N1 influenza pandemic in 2009. We have not included any age-effects, such as differential mixing, susceptibility or infectiousness. That means that we are not able to investigate the impact of school closures or the impact of the summer holidays, which had a large impact on the H1N1 influenza pandemic in 2009.\\n  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>183</td>\n",
       "      <td>Running Title: Global Biogeography of Disease Emergence Title In search for the hotspots of Disease X: A biogeographic approach to mapping the predictive risk of WHO's Blueprint Priority Diseases</td>\n",
       "      <td>[{'text': 'Anthropization of natural habitats including climate change along with overpopulation and global travel have been contributing to emerging infectious diseases outbreaks. The recent COVID-19 outbreak in Wuhan, highlights such threats to human health, social stability and global trade and economy. We used species distribution modelling and environmental data from satellite imagery to model Blueprint Priority Diseases occurrences. We constructed classical regression and Support Vector Machine models based on environmental predictor variables such as landscape, tree cover loss, climatic covariates. Models were evaluated and a weighed mean was used to map the predictive risk of disease emergence. We mapped the predictive risk for filovirus, Nipah, Rift Valley Fever and coronavirus diseases. Elevation, tree cover loss and climatic covariates were found to significant factors influencing disease emergence. We also showed the relevance of disease biogeography and in the identification potential hotspots for Disease X in regions in Uganda and China.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>Changes to local land use and biodiversity, the increasing international connectivity through human transport and trade and climate change provide optimal conditions for the emergence of refers to a potential international epidemic caused by a currently unknown pathogen, for which WHO calls for preparedness in the event of a novel disease emergence. This list of diseases in need of urgent research was termed as Blueprint list of Priority Diseases (BPDs).\\nAll the BPDs in the current list can be classified as zoonotic viral infections. In the past four decades, over 70% of the emerging infections are or were zoonoses (3, 4) . The increasing unpredictability in the global climate and the decreasing distance between the local humananimal-ecosystem interactions play a major role in driving the emergence of infection in human populations. The rising mean temperatures have been reported to have a significant effect in the tick-borne CCHF incidence and mosquito-borne Zika sustainability in subtropical and temperate . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint regions (5, 6) . Bushmeat consumption and animal trading, arising from the growing demand of animal protein, causes significant changes at the human-animal reservoir interface (7). Among the BPDs, studies demonstrate that the SARS and EVD outbreaks were directly linked to the consumption of infected bushmeat (8, 9) . LF, MVD and EVD flourish in West and central Africa where the consumption of bushmeat is four times greater than the Amazon, which is richer in biodiversity (10) . Moreover, EIDs are also triggered by anthropogenic pressures on land use for agricultural expansion and livestock farming to meet the demand of a growing human population. The fruit-bat migration driven by the deforestation through forest fires in the islands of Sumatra lead to the emergence of Nipah disease in farmers and abattoir workers in Malaysia (11, 12) . Once sufficient infection cycles between human-animal without the sustainable transmission among humans termed as viral chatter is established, the emergence of human-tohuman transmission is inevitable (7). Mathematical modeling and prediction provides quick assessment for control and potential preventive efforts when time for epidemiological studies is scarce (13) . Modeling of infection dynamics analyses diseases outbreaks in animal population and estimates the rate of transmission as well as the potential chance of spillover. Yet, the increasing trends of EIDs risks surpass our capacity in the surveillance and detection of spillovers and outbreaks. Thus, the current public health response to EIDs is to \"get ahead of the curve\" and the growing resolution of satellite imagery has shifted the paradigm towards identifying potential environmental drivers such as deforestation, land fragmentation, biodiversity loss and climate change rather than the surveillance of the EIDs themselves.\\nHere we proposed mapping the predictive risk of the BPDs using Species Distribution Models (SDM) based on potential environmental drivers such as deforestation, landscape and climatic covariates derived from satellite imagery following the year 2000. The aim is to provide a global . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint perspective, measure predictive risks and evaluate the use of biogeography on predicting diseases outbreaks. We also used our approach to EIDs using disease biogeography as a tool to identify the potential hotspots for an unknown Disease X listed in the BPDs.\\nTo construct robust models and generate predictive risk data of the BPDs' distribution, we adapted the following steps from species distribution modelling: 1) compilation of the geographical coordinates of the BPDs emergence; 2) construction of spatial buffer around the spatial points; 3) generation of pseudo absence points; 4) extraction of environmental predictor covariates; 5) fitting of the model and 6) calculation of the predictive risk of each BPD.\\nWe extracted the distributional data on the global occurrence of the BPDs in humans across the years 2000 to 2019 from WHO archives, Promed mail and published studies (Appendix Table. 1).\\nIn cases where the origin of the BPDs were unclear, we narrow down to the general region or district of origin. In other words, the localization of BPDs emergence correspond to patient zero of each outbreak. We did not include the diseases endemic to countries such as CCHF and Lassa fever. The calculation of predictive risk for endemic diseases and pandemics such as Zika may not contribute to the existing evidence. Laboratory outbreaks and outbreak in domestic and wildlife were also excluded. We georeferenced the sites of origin or the centroids of the region of occurrence to the nearest 0.0001°. Spatial buffers were constructed around the geographical coordinates depending upon the mobility range of the respective pathogens reservoirs and intermediate buffers (Appendix table 2 ). The construction of buffers also mitigates the relatively . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint less precise geographical logical coordinates in cases where the exact point of origin was unclear.\\nPresence and Pseudo-absence points\\nWe constructed spatial buffers to mitigate any potential inaccuracies in the localization of the occurrence points. For the presence points, we generated 100 random points at a 100km spatial buffer around the point of origin of each BPDs outbreak. Then, a spatial bufferzone of 150km was used as a mask and we generated 1000 pseudo-absence points within the geographical extent of the reservoirs of each of the BPDs. We chose the buffer of radius 150km to account to the average flying range of the order Chiroptera, the reservoirs of most BPD.\\nWe extracted the climatic covariates such as monthly maximum and minimum temperatures and precipitation from 'Terra Climate' high-resolution global data from 2000-2017 (14) . Land cover data used was a composite of the 2017 MODIS satellite imagery and the deforestations mdata was derived from the Hansan's deforestation tree loss-year maps from 2000-2018. The topological data including altitude and hydrological models was extracted from one arc-second digital elevation model (DEM) of 30m resolution, derived from NASA Version 3.0 Shuttle Radar Topography Mission (SRTM) imagery available in the United States Geological Survey (USGS). We obtained the geographical distribution of the primary hosts and reservoir mammals from the IUCN red list (15) . The shapefiles were rasterized and resampled into rasters of the 5km resolution raster model. We used livestock distribution rasters (2010) from the Food and Agriculture Organisation (FAO) for diseases such as RVF and Nipah including cattle, goats and . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint pigs (16) . The raster layers were resampled to a fixed resolution of 5km and stacked to a raster brick.\\nThe disease distribution models were fitted using the r package \"discmo\". In our study, we used the main two methods for species distribution modelling (SDM) i) the classical generalized linear models (glm) using gaussian regression methods and ii) the machine learning method, support vector machine (SVM). We choose the glm models to analyze the influence of the environmental factors on the emergence of BPDs. SVM are popular in SDM using presence and pseudo-absence data. The models were evaluated using ROC curves and the area under curves (AUC) for the produced thresholds were calculated. Studies have critized the use of AUCs in the evaluation of SDMs, especially when the study involves large extent. We removed the \"spatial sorting bias\" through \"point-wise distance sampling\" as explained by Hijmanns, 2012. Model prediction was made using the \"predict\" function to map the predictive risk of the diseases based on the values of the independent variables extracted from the environmental predictor rasters. A weighed average of the glm and SVM models were calculated using the AUCs for each BPD model and a final composite prediction was made.\\nTo analyze the impact of deforestation on the emergence of the BPDs, we conducted a detailed spatiotemporal analysis using the R package \"gfcanalytics\" on the Hansan's tree cover loss maps. We used the presence points at a spatial buffer of 100km, used for the BPD models, for this analysis.\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint\\nWe modelled five out of nine BPDs. We used glm and SVM methods of SDM for each of the disease. The AUC was maintained at an average of 0.9323 for glm models and 0.9597 for the SVM models. We have tabulated the AUC for each model in Table 1 .\\nThe distribution of the filovirus diseases, EVD and MVD, was restricted to the African subcontinent (Figures 1 &amp; 2 \\nThe weighed means of the AUC from the BPD models were used to create a threshold and the potential hotspots for the emergence of disease X was mapped ( Figure 6 ).\\nOur study was the first to map the predictive risk of the disease emergence using species distribution models and to establish the impact of environmental factors on their emergence. We . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint also identified the potential hotspots for future emerging infectious diseases based on the predictive models of BPDs. The results of our study have shown that the disease emergence is spatially dependent on the bioclimatic factors such as elevation, tree cover loss and climatic covariates. These factors could thus be utilized to identify and predict the hotspots of Disease X.\\nWe found elevation to have a significant influence on the distribution of Nipah and coronavirus diseases. Studies have hypothesized that Nipah emergence in lower Gangetic plains and at the low-lying backwaters could be attributed to flooding, which causes destruction of mammal habitats (17) . Rapid changes to habitats lead to starvation and migration of the known reservoirs of the Nipah virus, the bats, with contamination of fruit trees near human dwelling and increased exposure to the pathogen. Our results confirm this hypothesis associated with low-lying plains, flooding and emergence of Nipah disease. Coronaviruses are also linked with bat reservoirs. We hypothesize that similar events leading to destruction of bat habitats due to flooding at the lowlying regions such the wadis near Jeddah and Zhujiang Basin at the Province of Guangdong triggered the emergence of MERS and SARS respectively.\\nA recent study found that the increase in surface temperature and the unpredictability in the seasonal rainfall due to climate change, has an indirect effect on disease emergence through the sudden changes to the reservoirs habitats, biodiversity loss and small mammal migration (18) .\\nOur results demonstrate that increase in the residual temperature also known as the minimum temperature has a direct influence on the emergence and distribution of the BPDs. This direct spatial dependence of disease emergence on minimum temperatures is worrying. With climate change, the increasing night minimum temperatures lengthens the freeze-free season in most mid-and high latitude regions (19) . These conditions with worsening climate change could . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint increase the potential latitudinal extent of disease emergence. This hypothesis has been established from the vector-borne BPDs emergence predictions like CCHF and Zika (5, 20) .\\nOur study observed that deforestation is another important factor driving the emergence of BPDs. Studies have established the impact of deforestation and migration of bats on the occurrence of EVD, Nipah and SARS diseases (11, 21, 22) . Most of the models showed tree cover loss at 100km around the occurrence of the BPDs (see for example EVD, Figure 7 ). Biodiversity loss, due to anthropogenic changes to their habitat such as deforestation, land fragmentation for agriculture, increase in demand for protein and climate We hypothesize that a decrease in predators could cause an imbalance in the natural predator-prey equilibrium leading to an increase in small mammal reservoirs and thus to viruses transmission via disease vectors such as ticks. Lesser species diversity and inter-species interactions makes spillover easier towards the accidental human hosts. This has been termed as the \"dilution effect\" (23).\\nAs with all mathematical models, our study has its limitations. The size of the environmental predictor raster layers limited our spatial extent of the models. We chose the quality of the satellite imagery and resolution over a global perspective with poor or outdated data. Our study does not have a temporal component in the form of times series, which would be interesting especially with the climatic covariates. We mitigated this by choosing recent raster data corresponding to the period of the study and linking the spatiotemporal presence and pseudoabsence points to corresponding climatic monthly covariates. Despite these limitations, our study is the first to confirm the validity and effectiveness of using SDMs and other mathematical models to predict and identify the potential hotspots for BPDs. The use of a biogeographic approach in disease modelling offers a wider perspective on the environmental drivers and highlights the importance of climate change in the context of disease emergence. Most of all, our . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint study results observed that the potential hotspots for an unknown disease X is located in Uganda and China ( Figure 6 ). It is interesting to note that the associated predictive risk map includes the region of Wuhan, the epicenter of the ongoing COVID-19 outbreak.\\nIn our study with the use of a biogeographic approach, satellite imagery and SDMs, we were able to identify Wuhan as a potential hotspot of disease emergence in the absence of COVID-19 data.\\nWe confirm that distribution of disease emergence in humans is spatially dependent on environmental factors such as landscape, tree cover loss and climatic covariates. The following are our recommendations to get ahead of the curve in the prediction and prevention of disease 3. Alternate solutions to deforestation, land fragmentation for livestock farming and bushmeat consumption to meet the growing protein demand.\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                                                   title  \\\n",
       "183  Running Title: Global Biogeography of Disease Emergence Title In search for the hotspots of Disease X: A biogeographic approach to mapping the predictive risk of WHO's Blueprint Priority Diseases   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     abstract  \\\n",
       "183  [{'text': 'Anthropization of natural habitats including climate change along with overpopulation and global travel have been contributing to emerging infectious diseases outbreaks. The recent COVID-19 outbreak in Wuhan, highlights such threats to human health, social stability and global trade and economy. We used species distribution modelling and environmental data from satellite imagery to model Blueprint Priority Diseases occurrences. We constructed classical regression and Support Vector Machine models based on environmental predictor variables such as landscape, tree cover loss, climatic covariates. Models were evaluated and a weighed mean was used to map the predictive risk of disease emergence. We mapped the predictive risk for filovirus, Nipah, Rift Valley Fever and coronavirus diseases. Elevation, tree cover loss and climatic covariates were found to significant factors influencing disease emergence. We also showed the relevance of disease biogeography and in the identification potential hotspots for Disease X in regions in Uganda and China.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             full_text  \n",
       "183  Changes to local land use and biodiversity, the increasing international connectivity through human transport and trade and climate change provide optimal conditions for the emergence of refers to a potential international epidemic caused by a currently unknown pathogen, for which WHO calls for preparedness in the event of a novel disease emergence. This list of diseases in need of urgent research was termed as Blueprint list of Priority Diseases (BPDs).\\nAll the BPDs in the current list can be classified as zoonotic viral infections. In the past four decades, over 70% of the emerging infections are or were zoonoses (3, 4) . The increasing unpredictability in the global climate and the decreasing distance between the local humananimal-ecosystem interactions play a major role in driving the emergence of infection in human populations. The rising mean temperatures have been reported to have a significant effect in the tick-borne CCHF incidence and mosquito-borne Zika sustainability in subtropical and temperate . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint regions (5, 6) . Bushmeat consumption and animal trading, arising from the growing demand of animal protein, causes significant changes at the human-animal reservoir interface (7). Among the BPDs, studies demonstrate that the SARS and EVD outbreaks were directly linked to the consumption of infected bushmeat (8, 9) . LF, MVD and EVD flourish in West and central Africa where the consumption of bushmeat is four times greater than the Amazon, which is richer in biodiversity (10) . Moreover, EIDs are also triggered by anthropogenic pressures on land use for agricultural expansion and livestock farming to meet the demand of a growing human population. The fruit-bat migration driven by the deforestation through forest fires in the islands of Sumatra lead to the emergence of Nipah disease in farmers and abattoir workers in Malaysia (11, 12) . Once sufficient infection cycles between human-animal without the sustainable transmission among humans termed as viral chatter is established, the emergence of human-tohuman transmission is inevitable (7). Mathematical modeling and prediction provides quick assessment for control and potential preventive efforts when time for epidemiological studies is scarce (13) . Modeling of infection dynamics analyses diseases outbreaks in animal population and estimates the rate of transmission as well as the potential chance of spillover. Yet, the increasing trends of EIDs risks surpass our capacity in the surveillance and detection of spillovers and outbreaks. Thus, the current public health response to EIDs is to \"get ahead of the curve\" and the growing resolution of satellite imagery has shifted the paradigm towards identifying potential environmental drivers such as deforestation, land fragmentation, biodiversity loss and climate change rather than the surveillance of the EIDs themselves.\\nHere we proposed mapping the predictive risk of the BPDs using Species Distribution Models (SDM) based on potential environmental drivers such as deforestation, landscape and climatic covariates derived from satellite imagery following the year 2000. The aim is to provide a global . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint perspective, measure predictive risks and evaluate the use of biogeography on predicting diseases outbreaks. We also used our approach to EIDs using disease biogeography as a tool to identify the potential hotspots for an unknown Disease X listed in the BPDs.\\nTo construct robust models and generate predictive risk data of the BPDs' distribution, we adapted the following steps from species distribution modelling: 1) compilation of the geographical coordinates of the BPDs emergence; 2) construction of spatial buffer around the spatial points; 3) generation of pseudo absence points; 4) extraction of environmental predictor covariates; 5) fitting of the model and 6) calculation of the predictive risk of each BPD.\\nWe extracted the distributional data on the global occurrence of the BPDs in humans across the years 2000 to 2019 from WHO archives, Promed mail and published studies (Appendix Table. 1).\\nIn cases where the origin of the BPDs were unclear, we narrow down to the general region or district of origin. In other words, the localization of BPDs emergence correspond to patient zero of each outbreak. We did not include the diseases endemic to countries such as CCHF and Lassa fever. The calculation of predictive risk for endemic diseases and pandemics such as Zika may not contribute to the existing evidence. Laboratory outbreaks and outbreak in domestic and wildlife were also excluded. We georeferenced the sites of origin or the centroids of the region of occurrence to the nearest 0.0001°. Spatial buffers were constructed around the geographical coordinates depending upon the mobility range of the respective pathogens reservoirs and intermediate buffers (Appendix table 2 ). The construction of buffers also mitigates the relatively . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint less precise geographical logical coordinates in cases where the exact point of origin was unclear.\\nPresence and Pseudo-absence points\\nWe constructed spatial buffers to mitigate any potential inaccuracies in the localization of the occurrence points. For the presence points, we generated 100 random points at a 100km spatial buffer around the point of origin of each BPDs outbreak. Then, a spatial bufferzone of 150km was used as a mask and we generated 1000 pseudo-absence points within the geographical extent of the reservoirs of each of the BPDs. We chose the buffer of radius 150km to account to the average flying range of the order Chiroptera, the reservoirs of most BPD.\\nWe extracted the climatic covariates such as monthly maximum and minimum temperatures and precipitation from 'Terra Climate' high-resolution global data from 2000-2017 (14) . Land cover data used was a composite of the 2017 MODIS satellite imagery and the deforestations mdata was derived from the Hansan's deforestation tree loss-year maps from 2000-2018. The topological data including altitude and hydrological models was extracted from one arc-second digital elevation model (DEM) of 30m resolution, derived from NASA Version 3.0 Shuttle Radar Topography Mission (SRTM) imagery available in the United States Geological Survey (USGS). We obtained the geographical distribution of the primary hosts and reservoir mammals from the IUCN red list (15) . The shapefiles were rasterized and resampled into rasters of the 5km resolution raster model. We used livestock distribution rasters (2010) from the Food and Agriculture Organisation (FAO) for diseases such as RVF and Nipah including cattle, goats and . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint pigs (16) . The raster layers were resampled to a fixed resolution of 5km and stacked to a raster brick.\\nThe disease distribution models were fitted using the r package \"discmo\". In our study, we used the main two methods for species distribution modelling (SDM) i) the classical generalized linear models (glm) using gaussian regression methods and ii) the machine learning method, support vector machine (SVM). We choose the glm models to analyze the influence of the environmental factors on the emergence of BPDs. SVM are popular in SDM using presence and pseudo-absence data. The models were evaluated using ROC curves and the area under curves (AUC) for the produced thresholds were calculated. Studies have critized the use of AUCs in the evaluation of SDMs, especially when the study involves large extent. We removed the \"spatial sorting bias\" through \"point-wise distance sampling\" as explained by Hijmanns, 2012. Model prediction was made using the \"predict\" function to map the predictive risk of the diseases based on the values of the independent variables extracted from the environmental predictor rasters. A weighed average of the glm and SVM models were calculated using the AUCs for each BPD model and a final composite prediction was made.\\nTo analyze the impact of deforestation on the emergence of the BPDs, we conducted a detailed spatiotemporal analysis using the R package \"gfcanalytics\" on the Hansan's tree cover loss maps. We used the presence points at a spatial buffer of 100km, used for the BPD models, for this analysis.\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint\\nWe modelled five out of nine BPDs. We used glm and SVM methods of SDM for each of the disease. The AUC was maintained at an average of 0.9323 for glm models and 0.9597 for the SVM models. We have tabulated the AUC for each model in Table 1 .\\nThe distribution of the filovirus diseases, EVD and MVD, was restricted to the African subcontinent (Figures 1 & 2 \\nThe weighed means of the AUC from the BPD models were used to create a threshold and the potential hotspots for the emergence of disease X was mapped ( Figure 6 ).\\nOur study was the first to map the predictive risk of the disease emergence using species distribution models and to establish the impact of environmental factors on their emergence. We . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint also identified the potential hotspots for future emerging infectious diseases based on the predictive models of BPDs. The results of our study have shown that the disease emergence is spatially dependent on the bioclimatic factors such as elevation, tree cover loss and climatic covariates. These factors could thus be utilized to identify and predict the hotspots of Disease X.\\nWe found elevation to have a significant influence on the distribution of Nipah and coronavirus diseases. Studies have hypothesized that Nipah emergence in lower Gangetic plains and at the low-lying backwaters could be attributed to flooding, which causes destruction of mammal habitats (17) . Rapid changes to habitats lead to starvation and migration of the known reservoirs of the Nipah virus, the bats, with contamination of fruit trees near human dwelling and increased exposure to the pathogen. Our results confirm this hypothesis associated with low-lying plains, flooding and emergence of Nipah disease. Coronaviruses are also linked with bat reservoirs. We hypothesize that similar events leading to destruction of bat habitats due to flooding at the lowlying regions such the wadis near Jeddah and Zhujiang Basin at the Province of Guangdong triggered the emergence of MERS and SARS respectively.\\nA recent study found that the increase in surface temperature and the unpredictability in the seasonal rainfall due to climate change, has an indirect effect on disease emergence through the sudden changes to the reservoirs habitats, biodiversity loss and small mammal migration (18) .\\nOur results demonstrate that increase in the residual temperature also known as the minimum temperature has a direct influence on the emergence and distribution of the BPDs. This direct spatial dependence of disease emergence on minimum temperatures is worrying. With climate change, the increasing night minimum temperatures lengthens the freeze-free season in most mid-and high latitude regions (19) . These conditions with worsening climate change could . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint increase the potential latitudinal extent of disease emergence. This hypothesis has been established from the vector-borne BPDs emergence predictions like CCHF and Zika (5, 20) .\\nOur study observed that deforestation is another important factor driving the emergence of BPDs. Studies have established the impact of deforestation and migration of bats on the occurrence of EVD, Nipah and SARS diseases (11, 21, 22) . Most of the models showed tree cover loss at 100km around the occurrence of the BPDs (see for example EVD, Figure 7 ). Biodiversity loss, due to anthropogenic changes to their habitat such as deforestation, land fragmentation for agriculture, increase in demand for protein and climate We hypothesize that a decrease in predators could cause an imbalance in the natural predator-prey equilibrium leading to an increase in small mammal reservoirs and thus to viruses transmission via disease vectors such as ticks. Lesser species diversity and inter-species interactions makes spillover easier towards the accidental human hosts. This has been termed as the \"dilution effect\" (23).\\nAs with all mathematical models, our study has its limitations. The size of the environmental predictor raster layers limited our spatial extent of the models. We chose the quality of the satellite imagery and resolution over a global perspective with poor or outdated data. Our study does not have a temporal component in the form of times series, which would be interesting especially with the climatic covariates. We mitigated this by choosing recent raster data corresponding to the period of the study and linking the spatiotemporal presence and pseudoabsence points to corresponding climatic monthly covariates. Despite these limitations, our study is the first to confirm the validity and effectiveness of using SDMs and other mathematical models to predict and identify the potential hotspots for BPDs. The use of a biogeographic approach in disease modelling offers a wider perspective on the environmental drivers and highlights the importance of climate change in the context of disease emergence. Most of all, our . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint study results observed that the potential hotspots for an unknown disease X is located in Uganda and China ( Figure 6 ). It is interesting to note that the associated predictive risk map includes the region of Wuhan, the epicenter of the ongoing COVID-19 outbreak.\\nIn our study with the use of a biogeographic approach, satellite imagery and SDMs, we were able to identify Wuhan as a potential hotspot of disease emergence in the absence of COVID-19 data.\\nWe confirm that distribution of disease emergence in humans is spatially dependent on environmental factors such as landscape, tree cover loss and climatic covariates. The following are our recommendations to get ahead of the curve in the prediction and prevention of disease 3. Alternate solutions to deforestation, land fragmentation for livestock farming and bushmeat consumption to meet the growing protein demand.\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.27.20044156 doi: medRxiv preprint\\n  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>939</td>\n",
       "      <td>A Note on COVID-19 Diagnosis Number Prediction Model in China</td>\n",
       "      <td>[{'text': 'Importance: To predict the diagnosed COVID-19 patients and the trend of the epidemic in China. It may give the public some scientific information to ease the fear of the epidemic. Objective: In December 2019, pneumonia infected with the novel coronavirus burst in Wuhan, China. We aimed to use a mathematical model to predict number of diagnosed patients in future to ease anxiety on the emergent situation. Design: According to all diagnosis number from WHO website and combining with the transmission mode of infectious diseases, the mathematical model was fitted to predict future trend of outbreak. Setting: Our model was based on the epidemic situation in China, which could provide referential significance for disease prediction in other countries, and provide clues for prevention and intervention of relevant health authorities.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>In December of 2019, the increasing number of patients with pneumonia of unknown cause emerged in Wuhan, the capital of Hubei province, Central China. Most of the early patients had directly or indirectly related to wild animal source in the seafood and wet animal wholesale market through epidemiological survey [1] [2] [3] [4] [5] . Then laboratory tests found that the infection was caused by a novel coronavirus (now named as COVID-19 6-7 ).\\nThe spread of COVID-19 is expeditious. According to WHO statistics, the total number of diagnosed people in China was 309 in Jan 21 8 , but rapidly grew to 4537 in Jan 27 9 . Considering a simple constant exponential growth rate model [10] [11] , the diagnosis number will be unimaginable soon. This triggered a major concern and motivated this study [12] [13] [14] . A mathematical model is required to predict the spread trend and emergent situation.\\nOver the last few decades, mathematical models of disease transmission have been useful to gain insights into the transmission dynamics of infectious diseases and the potential role of intervention strategies 15 . Epidemic transmission models focused on the spread of an infectious disease through a population. But this spread can be modeled in two fundamental ways: by looking at the entire population as a single group or set of linked subgroups (population-based) or by considering everyone that makes up the population and how interactions between unique individuals facilitate disease spread (individual-based). Common models of infectious disease included meta-population models, individual-based network models, and simple SIR-type models that incorporate the effects of reactive behavior changes or inhomogeneous mixing 16 .\\nConsidering the limited number of data points and the complexity of the real situation, a simple but robust model is expected to work better than sophisticated epidemiology models. In this paper, we propose a robust model for next day diagnosis number predict. This model had excellent performance on prediction of infected patients in past few days. The prediction is expected to provide practical significance on social and economic application. &amp; # and its median aggregates were calculated (Fig.1) . It was observed that &amp; # manifests a random fluctuation around a constant center. It is reasonable to assume that is a constant. The estimation of currently (Feb 10) is 0.904. A future diagnosis number formula is obtained based on this estimation:\\nPrevious numbers of diagnoses in China are used to predict the number for the next day. The prediction is started from Jan 29 as several data points are needed to build the model (Table 1) . We obtain nice concordance (&lt;2.5% error) with the real numbers next day. \\nWe predict the new diagnosis number in China according to our model as Fig.2 . The model predicts that new diagnosis will last to March.\\nOur model indicates that the COVID-19 disease infection rate decreases exponentially. This model converges to a maximum number as time increases, indicating a limited impact of COVID-19. The model also predict that the new diagnosis number will last to March, indicating that the COVID-19 disease will be well controlled.\\nWe only analyzed the total number of diagnoses in China. The province scale data is available but is not as complete as total number. Here, we did not show these details analyses.\\nOur model is attractive as it not only predicts data accurately but also is minimal.\\n. CC-BY-NC-ND 4.0 International license It is made available under a is the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint . https://doi.org/10.1101/2020.02.19.20025262 doi: medRxiv preprint\\nThe minimal number of parameters and robust statistics make our model robust in the real-world scenario. We develop a formula to predict the next day diagnosis number:\\n) \" ，and estimate that is 0.904( &lt;1) which shows that the diagnosis number of COVID-19 patients will decrease. It is very effective in controlling the spread of infectious diseases that the Chinese health authorities have taken appropriate health care actions. If the outbreak is no longer controlled, the value of will be around 1 indicating that the outbreak will wreak havoc. We obtain this elegant formula with both curve plot visualization and mathematical intuition. We plot ln (Ni/Ni-1) against time (Fig.3 ). First few observations showed a linear decay of spread rate. However, later data points realized us at Feb 4 that it is an exponential decay. Later data points further confirm our model. \\nDaily diagnosis number of China is download from WHO situation reports (https://www.who.int/emergencies/diseases/novel-coronavirus-2019/situation-reports). The data starts from Jan 21 and currently it ends at Feb 10.\\nThe following model is proposed based on our observations:\\nwhere Ni is the number of diagnosed patients in China according to WHO at the ith day, is a parameter modeling the decay of diagnosis growth rate. A simple estimation of at the ith day would be\\nAs we have multiple days of observations, historical &amp; # with the median statistic, which is well known for its robustness, can be aggregated.\\nThe source code of model estimation is at: https://github.com/wangyi-fudan/ncp_ model\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                             title  \\\n",
       "939  A Note on COVID-19 Diagnosis Number Prediction Model in China   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          abstract  \\\n",
       "939  [{'text': 'Importance: To predict the diagnosed COVID-19 patients and the trend of the epidemic in China. It may give the public some scientific information to ease the fear of the epidemic. Objective: In December 2019, pneumonia infected with the novel coronavirus burst in Wuhan, China. We aimed to use a mathematical model to predict number of diagnosed patients in future to ease anxiety on the emergent situation. Design: According to all diagnosis number from WHO website and combining with the transmission mode of infectious diseases, the mathematical model was fitted to predict future trend of outbreak. Setting: Our model was based on the epidemic situation in China, which could provide referential significance for disease prediction in other countries, and provide clues for prevention and intervention of relevant health authorities.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   full_text  \n",
       "939  In December of 2019, the increasing number of patients with pneumonia of unknown cause emerged in Wuhan, the capital of Hubei province, Central China. Most of the early patients had directly or indirectly related to wild animal source in the seafood and wet animal wholesale market through epidemiological survey [1] [2] [3] [4] [5] . Then laboratory tests found that the infection was caused by a novel coronavirus (now named as COVID-19 6-7 ).\\nThe spread of COVID-19 is expeditious. According to WHO statistics, the total number of diagnosed people in China was 309 in Jan 21 8 , but rapidly grew to 4537 in Jan 27 9 . Considering a simple constant exponential growth rate model [10] [11] , the diagnosis number will be unimaginable soon. This triggered a major concern and motivated this study [12] [13] [14] . A mathematical model is required to predict the spread trend and emergent situation.\\nOver the last few decades, mathematical models of disease transmission have been useful to gain insights into the transmission dynamics of infectious diseases and the potential role of intervention strategies 15 . Epidemic transmission models focused on the spread of an infectious disease through a population. But this spread can be modeled in two fundamental ways: by looking at the entire population as a single group or set of linked subgroups (population-based) or by considering everyone that makes up the population and how interactions between unique individuals facilitate disease spread (individual-based). Common models of infectious disease included meta-population models, individual-based network models, and simple SIR-type models that incorporate the effects of reactive behavior changes or inhomogeneous mixing 16 .\\nConsidering the limited number of data points and the complexity of the real situation, a simple but robust model is expected to work better than sophisticated epidemiology models. In this paper, we propose a robust model for next day diagnosis number predict. This model had excellent performance on prediction of infected patients in past few days. The prediction is expected to provide practical significance on social and economic application. & # and its median aggregates were calculated (Fig.1) . It was observed that & # manifests a random fluctuation around a constant center. It is reasonable to assume that is a constant. The estimation of currently (Feb 10) is 0.904. A future diagnosis number formula is obtained based on this estimation:\\nPrevious numbers of diagnoses in China are used to predict the number for the next day. The prediction is started from Jan 29 as several data points are needed to build the model (Table 1) . We obtain nice concordance (<2.5% error) with the real numbers next day. \\nWe predict the new diagnosis number in China according to our model as Fig.2 . The model predicts that new diagnosis will last to March.\\nOur model indicates that the COVID-19 disease infection rate decreases exponentially. This model converges to a maximum number as time increases, indicating a limited impact of COVID-19. The model also predict that the new diagnosis number will last to March, indicating that the COVID-19 disease will be well controlled.\\nWe only analyzed the total number of diagnoses in China. The province scale data is available but is not as complete as total number. Here, we did not show these details analyses.\\nOur model is attractive as it not only predicts data accurately but also is minimal.\\n. CC-BY-NC-ND 4.0 International license It is made available under a is the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint . https://doi.org/10.1101/2020.02.19.20025262 doi: medRxiv preprint\\nThe minimal number of parameters and robust statistics make our model robust in the real-world scenario. We develop a formula to predict the next day diagnosis number:\\n) \" ，and estimate that is 0.904( <1) which shows that the diagnosis number of COVID-19 patients will decrease. It is very effective in controlling the spread of infectious diseases that the Chinese health authorities have taken appropriate health care actions. If the outbreak is no longer controlled, the value of will be around 1 indicating that the outbreak will wreak havoc. We obtain this elegant formula with both curve plot visualization and mathematical intuition. We plot ln (Ni/Ni-1) against time (Fig.3 ). First few observations showed a linear decay of spread rate. However, later data points realized us at Feb 4 that it is an exponential decay. Later data points further confirm our model. \\nDaily diagnosis number of China is download from WHO situation reports (https://www.who.int/emergencies/diseases/novel-coronavirus-2019/situation-reports). The data starts from Jan 21 and currently it ends at Feb 10.\\nThe following model is proposed based on our observations:\\nwhere Ni is the number of diagnosed patients in China according to WHO at the ith day, is a parameter modeling the decay of diagnosis growth rate. A simple estimation of at the ith day would be\\nAs we have multiple days of observations, historical & # with the median statistic, which is well known for its robustness, can be aggregated.\\nThe source code of model estimation is at: https://github.com/wangyi-fudan/ncp_ model\\n  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1221</td>\n",
       "      <td>Public Exposure to Live Animals, Behavioural Change, and Support in Containment Measures in response to COVID-19 Outbreak: a population-based cross sectional survey in China</td>\n",
       "      <td>[{'text': 'Word count (main text): 3489 : medRxiv preprint', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'The public responded quickly to the outbreak, and reduced exposure to live animals, especially in Wuhan. With high public support in containment measures, better regulation of wet markets and healthy diets should be promoted.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>In response to the COVID-19 outbreak, we aimed to investigate behavioural change on exposure to live animals before and during the outbreak, and public support and confidence for governmental containment measures.\\nA population-based cross-sectional telephone survey via random dialing was conducted in Wuhan (the epicentre) and Shanghai (an affected city with imported cases) between 1 and 10 February, 2020. 510 residents in Wuhan and 501 residents in Shanghai were randomly sampled. Differences of outcome measures were compared before and during the outbreak, and between two cities.\\nProportion of respondents visiting wet markets at usual was 23.3% (119/510) in Wuhan and 20.4%\\n(102/501) in Shanghai. During the outbreak, it decreased to 3.1% (16) in Wuhan (p&lt;0·001), and 4.4% (22) in Shanghai (p&lt;0·001). Proportion of those consuming wild animal products declined from 10.2%\\n(52) to 0.6% (3) in Wuhan (p&lt;0·001), and from 5.2% (26) to 0.8% (4) in Shanghai (p&lt;0·001). 79.0%\\n(403) of respondents in Wuhan and 66.9% (335) of respondents in Shanghai supported permanent closure of wet markets (P&lt;0.001). 95% and 92% of respondents supported banning wild animal trade and quarantining Wuhan, and 75% were confident towards containment measures. Females and the more educated were more supportive for the above containment measures.\\nAll rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint 4\\nEvidence before this study On February 19, 2020, we searched PubMed for papers published after January 1, 2020, containing the following terms: \"2019 nCoV\" or \"COVID-19\". We identified 179 studies, most of which are research on clinical and epidemiological characteristics of COVID-19. To date there is no primary research to quantify public behavioural response and support in containment measures in response to the outbreak.\\nOnly four commentaries mentioned the influence of the outbreak on mental health. One commentary introduced the habit of consuming wild animal products in China. Another commentary briefly introduced isolation, quarantine, social distancing and community containment as public health measures in the outbreak. The Chinese government has introduced a series of strict containment measures, and societal acceptability of these measure is important for effective and sustained response.\\nEvidence is urgently needed to help policy makers understand public response to the outbreak and support for the containment measures, but no evidence available to date.\\nWe conducted a population-based cross-sectional telephone survey via random digital dialing in Wuhan (the epicentre) and Shanghai (an affected city with imported cases) between 1 and 10 February, 2020.\\nTo date, this is the only few analyses on behavioural response to the outbreak and societal acceptability of governmental containment measures, which has been listed as the current priority of China CDC. We provide an assessment of behavioural change on exposure to live animals during the outbreak, by comparison before and during the outbreak, and between two cities with diverse exposure intensities to COVID-19. We also provide evidence on public support in governmental containment measures, including strict regulation on wet markets to reduce animal-to-human transmission and city quarantine All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint 5 to reduce human transmission.\\nWe found that wild animal consumption was more prevalent in Wuhan (10.2%) than in Shanghai (5.2%). The public responded quickly to the outbreak, and significantly reduced exposure to live animals and stopped wild animal consumption, especially in Wuhan. They were very supportive of governmental containment measures. With high public support, wet markets should be better regulated, and healthy diets, including changing the traditional habit of eating wild animal products, should be promoted. This can inform policy makers in China and other countries to implement and adjust containment strategies in response to the outbreak in the future.\\nEarly in December 2019, several pneumonia cases caused by the novel coronavirus (now known as were first reported in Wuhan city, China. [1] [2] [3] The majority of the earliest cases were epidemiologically traced to the local Huanan (Southern China) Seafood Wholesale Market, where some wild animals were sold. [1] [2] [3] [4] On Jan 26, novel coronavirus was detected in environmental samples from this market by China CDC, suggesting that the virus possibly originated in wild animals sold in the market. [3] [4] This pathogenic virus has been identified as a new strain of coronavirus, which is in the same family as severe acute respiratory syndrome coronavirus (SARS-CoV) and Middle East respiratory syndrome coronavirus (MERS-CoV). [4] [5] The COVID-19 outbreak was declared by World Health Organization as a Public Health Emergency of International Concern (PHEIC) on Jan 30, 2020. 6 As of Feb 16, 2020, a total of 70,548 cases have been reported in mainland China, with 1,770 resulting deaths. 7 The outbreak has now spread beyond China to twenty-five other countries. 8 The SARS-CoV and MERS-CoV were proven to have originated from wild animals. [9] [10] [11] [12] [13] [14] The COVID-19 is also likely to have a zoonotic origin. The practice of consuming wild animal products in China dates back to prehistoric times and persists into today. 9 Trade and consumption of wild animals were not well regulated even after SARS. Until the amendment of Wild Animal Protection Act in 2016, the prohibition was firstly added, but only for a short list of the nationally protected animals. Previous studies explored population exposure to live poultry in China, [15] [16] [17] but information on exposure to wild animals is limited.\\nIn the efforts to contain the spread of the virus, the Chinese government introduced a series of All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint containment measures. 18 Specifically, to lower the risk of zoonotic virus transmission, China has issued strict regulations on wet markets where live and/or wild animals are traded, including closing live poultry markets (LPMs) and bans on wild animal transactions. Furthermore, Wuhan city has been quarantined to stop human-to-human transmission: starting on Jan 23, 2020, public transportation in Wuhan was closed, and all flights and trains leaving from Wuhan were cancelled.\\nThe COVID-19 outbreak and subsequent containment measures to lower the zoonotic virus transmission offer a unique opportunity to reassess the exposure to live animals, demand for wild animal consumption, and public readiness to further regulate wet markets. In this study, we aim to investigate behaviour patterns of population interaction to live animals, with a particular focus on differences before and during the outbreak, between Wuhan and Shanghai representing diverse exposure intensities to COVID-19. We also seek to understand levels of public support for containment measures and public confidence in their effectiveness to quell the outbreak. This study will help to clarify the public's response to the outbreak and has important policy implications regarding the social acceptability of infection containment measures and their application in different contexts.\\nA population-based cross-sectional telephone survey was conducted in Wuhan and Shanghai between Feb 1 and 10, 2020, one month since the outbreak. These two cities were selected to represent diverse exposure intensities to COVID-19. Wuhan is the site of origin of the COVID-19 outbreak. Shanghai was selected due to its status as a city that has been significantly affected by imported COVID-19 cases All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint 8 from Wuhan. The first imported case was confirmed in Shanghai on Jan 20. During the period of our survey, the number of confirmed cases rose rapidly in Wuhan, from 4,109 to 18,454, but maintained low epidemic rates in Shanghai, with reported cases rising from 182 to 303. Figure 1 illustrates the timeline of the outbreak progression and containment measures. This provides an opportunity to assess and compare public behavioural response and support for containment measures during the outbreak between two cities with different exposure intensities.\\nThe sample size (n=500 for each city) is sufficient to estimate the change of outcomes (p=0.5) with a margin of error of 0.05 and a 95% confidence interval, according to previous studies. [15] [16] Our target population was persons aged 18 or over who had been living in the selected city for at least 6 months in the year prior to the survey and were current residents at the time of the survey. In each city, the telephone survey was conducted through a computer-assisted interviewing system, which enabled the random generation of mobile phone numbers and systematic data collection. We attempted to contact each generated number three times a day at different hours; if no contact was established after the third call, this number was classified as invalid or unreachable. Random dials with proportional quota sampling were used to ensure that respondents were demographically representative of the general population in each city, with quotas based on sex and age. 19\\nThe outcomes of interest we studied were exposure to live animals before and during the COVID-19 outbreak, public support for containment measures that aim to reduce population exposure to live animals and human transmission, and confidence in these measures to quell the outbreak. Exposure to All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint live animals includes the following behaviours: visiting the live poultry or wild animal markets, buying live poultry or wild animals (i.e., frequency of purchases, practice of picking up live animals before purchasing, location where the purchased animals were slaughtered), eating wild animal products, raising pets, and touching stray animals. To measure behavioural change during the outbreak, each respondent was asked to report exposing behaviours both (1) prior to the outbreak: before Dec 31, 2019 when COVID-19 cases were first released by Wuhan Health Commission and (2) since the outbreak, between Jan 1 to Feb 10, 2020. For live animal consumption habits, where behaviour change cannot be measured in a short period of time, we assessed people's intention to: 1) buy live poultry or wild animals, and eat wild animal products (continue to buy/eat as usual, buy/eat less, not buy/eat any more) and 2) continue raising pets and touching stray animals (continue to raise/touch as usual, temporarily not during the outbreak, try or resolutely not even after the outbreak).\\nWe measured public support for two types of governmental containment measures in response to the outbreak; one is regulation on wet markets to reduce population exposure to live animals, and the other is quarantining measures meant to reduce human transmission, such as the closure of public transportation in Wuhan and the cancellation of flights and trains leaving from Wuhan. Each respondent was asked to rate the extent, on a five-point Likert scale, to which they supported 1) the permanent closure of LPMs, 2) a ban on the wild animal trade, and 3) the quarantining of Wuhan; respondents were also asked to report the extent of their confidence in the effectiveness of these measures to quell the outbreak.\\nCharacteristics of respondents, including gender, age, marital status, education, income, and All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint employment status, were collected in our survey. We also collected the history of fever or cough symptoms for respondents and their relatives or friends in the two weeks prior to the survey, and whether there were confirmed or suspected COVID-19 cases in the neighbourhood in which they live.\\nEach respondent was also asked at what point they started paying attention to this outbreak.\\nThe survey questionnaire was adjusted according to an instrument used during the outbreaks of SARS in 2003 20-21 and influenza A(H7N9) in 2013. 15 The survey instrument was pretested for face and content validity, length, and understandability by the public before our survey was administered.\\nDescriptive and univariate analysis were used to investigate changes in exposure to live animals during the outbreak, public support for containment measures, and confidence in containment measures to quell the outbreak, between Wuhan (the epicentre) and Shanghai (an affected city with imported cases).\\nChi-square tests or fisher's exact tests (when applicable) were used to compare differences in these measures between the two cities, and before and during the outbreak. Multivariate logistic regression was used to estimate factors associated with public support for containment measures and confidence in intervention effectiveness. The proportion and Odds Ratio with 95% confidence intervals (CI) were reported. All statistical analyses were performed using Stata 14.0 (StataCorp LP, College Station, TX).\\nA total of 18,576 telephone numbers were randomly selected via digital dialing system in Wuhan and Shanghai. After calling, 6,836 respondents were eligible to participate, of whom 5,640 declined and All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint 11 185 terminated before completing the interview. In total, 510 respondents in Wuhan and 501 respondents in Shanghai completed the interview, and the overall response rate was 14.8%\\n(1,011/6,836) (Appendix Figure 1 ).\\nRespondents in Wuhan and Shanghai were generally similar in terms of sex, marital status and occupation, but in Shanghai respondents tended to be older and had higher education and income levels than in Wuhan (Table 1 ). Our sample was consistent with characteristics of the general population in the two cities, as Shanghai is more developed and has an older population than Wuhan. Expectedly, respondents in Wuhan were much more intensely exposed to the COVID-19 outbreak and paid attention to the outbreak earlier than those in Shanghai.\\nWe first investigated patterns of usual exposure to live animals before the outbreak ( Table 2 Panel of 55, 58.2% in Shanghai, p=0.339); more than 90% of respondents always arranged for slaughter of purchased animals in wet markets, with no noticeable differences between the two cities. However, residents in Wuhan reported wild animal consumption at a higher rate than those in Shanghai; 47 (9.2%) of 510 respondents in Wuhan and 23 (4.6%) of 501 respondents in Shanghai reported eating All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is To investigate effects of the outbreak, we further assessed behavioural changes in exposure to live animals before and during the outbreak (Figure 2) , and intention to change after the outbreak ( Table 2 Panel B). We found that respondents dramatically reduced exposure to live animals during the outbreak, with respondents in Wuhan changing their exposure much more significantly than in Shanghai. Specifically, the frequency of visiting LPMs decreased from 23.3% (n=119) to 3.1% (n=16)\\nin Wuhan (p&lt;0·001), and from 20.4% (n=102) to 4.4%(n=22) in Shanghai (p&lt;0·001; Figure 2 ). The proportion of respondents buying live poultry or wild animals also decreased from around 11.0% to 2.6% in both cities (p&lt;0·001). We also asked whether respondents intended to change their habit of buying live poultry or wild animals since the outbreak among those exposing to live animals at usual ( (4/55) of respondents in Wuhan and 25.5% (14/55) in Shanghai stated they would continue to buy these products as usual.\\nThe proportion of respondents consuming wild animals declined significantly, 10.2%(n=52) to 0.6%(n=3) in Wuhan (p&lt;0·001) and 5.2%(n=26) to 0.8%(n=4) in Shanghai (p&lt;0·001) (Figure 2 ). Most All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is Wuhan, respectively, and 75% of respondents were confident that governmental containment measures would quell the outbreak. Table 3 presents socio-demographic factors associated with public support for the governmental containment measures implemented (i.e. closure of LPMs, banning the wild animal trade, and city quarantine) and confidence in the effectiveness of governmental response. There was evidence that females were more supportive towards all three containment measures and that there was a positive association between education level and the likelihood of supporting regulations on wet markets and trade of wild animals. Unsurprisingly, those who self-reported visiting a wet market and eating wild All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint animal products prior to the outbreak are much less likely to support closure of the market or a ban on the trade as containment measures (adjusted OR=0.58, 95%CI [0.41-0.82] and adjusted OR=0.31, 95%CI [0.14-0.70], respectively). We did not find any difference in population profiles of those who support Wuhan quarantine or show confidence in the effectiveness of governmental response.\\nThis study is the first to investigate behavioural changes of exposure to live animals during the COVID-19 outbreak, and public support for governmental containment measures in response to the outbreak. We found that population exposures to live animals were similar in Wuhan and Shanghai before the outbreak, except that wild animal consumption was more prevalent in Wuhan than in Shanghai. During the outbreak, residents in both cities significantly changed their behaviours and reduced exposures to live animals, and such changes were much more significant in Wuhan. Most respondents supported governmental containment measures implemented, and three in four had confidence on the effectiveness of these measures. Male, the less educated and those exposing to live animals at usual were less likely to support containment measures.\\nIn our survey, more than 20% of respondents visited wet markets at usual, which was similar in Wuhan and Shanghai, despite Shanghai being much more developed. Compared with the 30% level reported in 2013 in the same study sites, 15 the share of respondents visiting wet markets declined by 10 percentage points, but was still high. 10% of respondents consumed products of wild animals in Wuhan, twice that reported in Shanghai. In China, there exists a long tradition of eating the meat and products of wild animals, which persists into modern times due to a false but commonly held belief that these products All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint have therapeutic effects. [22] [23] These beliefs and preferences for fresh food are especially popular in southern China, where Wuhan is located, but were greatly challenged and changed during the 2003 SARS outbreak in China 9,12 . However, these practices have become popular again in recent years, due in part to the popularity of social media influencers who broadcast themselves eating wild animal products on livestream platforms. 22 The revival of this tradition may have set the socio-biological context for the occurrence and spread of COVID-19 in China.\\nWe found that respondents' risks of exposure to live animals declined significantly, reaching a very low level, and most respondents stopped wild animal consumption after this outbreak, especially in Wuhan.\\nThis indicates that the public responded quickly and adequately to the outbreak. In the survey, all respondents cared about the outbreak, and about one in five respondents in Wuhan reported starting to pay attention before Dec 31, 2019. The outbreak gradually gained intense nationwide attention, 24 and a series of policies were implemented by the Chinese government to contain the outbreak. As of Jan 30, all provinces had initiated first-level responses to major public health emergencies. [25] [26] These actions raised public attention further and induced significant behavioural changes. A national online survey on Jan 26 showed that the vast majority of people knew of the disease and related preventive measures, 66.5% reported that they would wear masks when outside their home, and 80% cancelled travel plans during the Spring Festival due to the outbreak. 27 The findings from both this survey and ours demonstrate behavioural change in terms of exposure to live animals or daily protection.\\nIn addition, more strict regulations were implemented on wet markets to lower the risk of disease transmission. Wuhan instituted prohibited the sale of live poultry and wild animals starting on Jan 22. All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint Shanghai started its seasonal closure of LPMs on Jan 25 this year. The Chinese Government further instituted nationwide bans on all forms of wild animal transactions on Jan 26. 28 These regulations may accelerate the decrease in exposing behaviours to live animals. However, the COVID-19 epidemic occurred prior to the Chinese lunar new year, when the demand for live animals is generally high. 23 This may produce a conflict between the public demand for live animals and regulations. Fortunately, we found that the public strongly supports the regulations on wet markets to control the outbreak.\\nAlthough permanent closure of LPMs was not fully backed by the public, it was still much higher than reported during the 2013 influenza A outbreak (27.3% in Wuhan, 39.3% in Shanghai). 15, 29 The strong support for this policy offers a great opportunity to improve regulations on wet markets and change the prevalent habit of eating wild animal products in China.\\nWe believe that the keys to quelling outbreaks of zoonotic viral diseases such as COVID-19 and preventing future outbreaks are changing the false beliefs and dangerous habits regarding eating wild animal products and strengthening regulations on wet markets. Firstly, healthy diets need to be vigorously promoted in China. Health China 2030 promotes healthy diets, but does not discuss the risks of eating wild animals. We also believe the internet and social media can be an effective channel to promote healthy eating habits and reduce the inappropriate and outdated tradition of eating wild animal products. 25, 30 Since the outbreak, spontaneous initiatives have emerged on the internet to explain the risks of and resist consuming wild animals. Secondly, live animal transactions should be more strictly regulated, and wild animal transactions should be even prohibited in the long term. While different animals should be sold separately in specific wet markets to lower risks of virus transmission, in practice, various kinds of live animals are sold together within one market, as found in Huanan Seafood All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint Wholesale Market. Laws should be strengthened to regulate and implement the strict separation of live animals from general markets. With the currently high public support, bans on the wild animal trade should be promoted, but the government must be aware of the risk of creating an unregulated black market by instituting a ban. For live poultry and non-wild animals, better regulations may be a better option than bans without the full support of the public. 23 The feasible measures include seasonal closure of wet markets, regular rest days, and banning leaving live animals in markets overnight. 10, 15 The above measures should be given priority in China and other countries to reduce current levels of COVID-19, as well as future threats from zoonotic diseases.\\nOur study has several limitations. First, there may be recall bias. We aimed to identify behavioral changes in a cross-sectional survey, and outcomes before the outbreak were measured based on participant recall. Second, our results may be affected by selection bias from telephone survey. We did attempt multiple calls to unanswered numbers to mitigate this bias, and quota sampling was introduced to get a representative sample. The anonymous phone interview addressed the concern of response bias due to fear of being seen as criticizing the government. Third, a cross-sectional design prevents us from assessing how long the effect on behavioral/intention change demonstrated in this study will last, when the outbreak is no longer fresh in people's minds. Thus, we plan to conduct a second-round survey at the end of the outbreak to track the long-term changes.\\nIn conclusion, exposure to live animals was common before the COVID-19 outbreak. The public responded quickly to this outbreak, and significantly reduced their exposure to live animals and stopped wild animal consumption, especially in Wuhan. They were very supportive and confident for All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint current mitigating measures. With high public support, wet markets should be better regulated, and healthy diets, including changing the habit of wild animal consumption, should be promoted, especially for male, the less educated and those exposing to live animals at usual. \\nThe study was approved by the Institutional Review Board of the School of Public Health, Fudan University (IRB#2020-TYSQ-01-1). Verbal informed consent was obtained from all participants.\\nH.Y. has received research funding from Sanofi Pasteur, GlaxoSmithKline, Yichang HEC Changjiang Pharmaceutical Company and Shanghai Roche Pharmaceutical Company. None of that research funding is related to 2019-nCov. All data are n (%). P value are from Chi square test or fisher's exact tests (when applicable).\\n*Fever or cough symptoms in the past 2 weeks.\\nAll rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint Resolutely not after outbreak 6(14.0) 3(5.2) All rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity. All rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity. Odds ratios with 95% CI were obtained from multivariate binary logistic regressions. Significance level: ** p&lt;0.01, * p&lt;0.05.\\nAll rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint WHO stands for World Health Organization; PHEIC stands for public health emergency of international concern.\\nAll rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint Notes: Proportions are shown, and error bars represent 95% CIs.\\nAll rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint\\nNotes: p value is shown between Wuhan and Shanghai.\\nAll rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint \\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                              title  \\\n",
       "1221  Public Exposure to Live Animals, Behavioural Change, and Support in Containment Measures in response to COVID-19 Outbreak: a population-based cross sectional survey in China   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                              abstract  \\\n",
       "1221  [{'text': 'Word count (main text): 3489 : medRxiv preprint', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'The public responded quickly to the outbreak, and reduced exposure to live animals, especially in Wuhan. With high public support in containment measures, better regulation of wet markets and healthy diets should be promoted.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             full_text  \n",
       "1221  In response to the COVID-19 outbreak, we aimed to investigate behavioural change on exposure to live animals before and during the outbreak, and public support and confidence for governmental containment measures.\\nA population-based cross-sectional telephone survey via random dialing was conducted in Wuhan (the epicentre) and Shanghai (an affected city with imported cases) between 1 and 10 February, 2020. 510 residents in Wuhan and 501 residents in Shanghai were randomly sampled. Differences of outcome measures were compared before and during the outbreak, and between two cities.\\nProportion of respondents visiting wet markets at usual was 23.3% (119/510) in Wuhan and 20.4%\\n(102/501) in Shanghai. During the outbreak, it decreased to 3.1% (16) in Wuhan (p<0·001), and 4.4% (22) in Shanghai (p<0·001). Proportion of those consuming wild animal products declined from 10.2%\\n(52) to 0.6% (3) in Wuhan (p<0·001), and from 5.2% (26) to 0.8% (4) in Shanghai (p<0·001). 79.0%\\n(403) of respondents in Wuhan and 66.9% (335) of respondents in Shanghai supported permanent closure of wet markets (P<0.001). 95% and 92% of respondents supported banning wild animal trade and quarantining Wuhan, and 75% were confident towards containment measures. Females and the more educated were more supportive for the above containment measures.\\nAll rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint 4\\nEvidence before this study On February 19, 2020, we searched PubMed for papers published after January 1, 2020, containing the following terms: \"2019 nCoV\" or \"COVID-19\". We identified 179 studies, most of which are research on clinical and epidemiological characteristics of COVID-19. To date there is no primary research to quantify public behavioural response and support in containment measures in response to the outbreak.\\nOnly four commentaries mentioned the influence of the outbreak on mental health. One commentary introduced the habit of consuming wild animal products in China. Another commentary briefly introduced isolation, quarantine, social distancing and community containment as public health measures in the outbreak. The Chinese government has introduced a series of strict containment measures, and societal acceptability of these measure is important for effective and sustained response.\\nEvidence is urgently needed to help policy makers understand public response to the outbreak and support for the containment measures, but no evidence available to date.\\nWe conducted a population-based cross-sectional telephone survey via random digital dialing in Wuhan (the epicentre) and Shanghai (an affected city with imported cases) between 1 and 10 February, 2020.\\nTo date, this is the only few analyses on behavioural response to the outbreak and societal acceptability of governmental containment measures, which has been listed as the current priority of China CDC. We provide an assessment of behavioural change on exposure to live animals during the outbreak, by comparison before and during the outbreak, and between two cities with diverse exposure intensities to COVID-19. We also provide evidence on public support in governmental containment measures, including strict regulation on wet markets to reduce animal-to-human transmission and city quarantine All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint 5 to reduce human transmission.\\nWe found that wild animal consumption was more prevalent in Wuhan (10.2%) than in Shanghai (5.2%). The public responded quickly to the outbreak, and significantly reduced exposure to live animals and stopped wild animal consumption, especially in Wuhan. They were very supportive of governmental containment measures. With high public support, wet markets should be better regulated, and healthy diets, including changing the traditional habit of eating wild animal products, should be promoted. This can inform policy makers in China and other countries to implement and adjust containment strategies in response to the outbreak in the future.\\nEarly in December 2019, several pneumonia cases caused by the novel coronavirus (now known as were first reported in Wuhan city, China. [1] [2] [3] The majority of the earliest cases were epidemiologically traced to the local Huanan (Southern China) Seafood Wholesale Market, where some wild animals were sold. [1] [2] [3] [4] On Jan 26, novel coronavirus was detected in environmental samples from this market by China CDC, suggesting that the virus possibly originated in wild animals sold in the market. [3] [4] This pathogenic virus has been identified as a new strain of coronavirus, which is in the same family as severe acute respiratory syndrome coronavirus (SARS-CoV) and Middle East respiratory syndrome coronavirus (MERS-CoV). [4] [5] The COVID-19 outbreak was declared by World Health Organization as a Public Health Emergency of International Concern (PHEIC) on Jan 30, 2020. 6 As of Feb 16, 2020, a total of 70,548 cases have been reported in mainland China, with 1,770 resulting deaths. 7 The outbreak has now spread beyond China to twenty-five other countries. 8 The SARS-CoV and MERS-CoV were proven to have originated from wild animals. [9] [10] [11] [12] [13] [14] The COVID-19 is also likely to have a zoonotic origin. The practice of consuming wild animal products in China dates back to prehistoric times and persists into today. 9 Trade and consumption of wild animals were not well regulated even after SARS. Until the amendment of Wild Animal Protection Act in 2016, the prohibition was firstly added, but only for a short list of the nationally protected animals. Previous studies explored population exposure to live poultry in China, [15] [16] [17] but information on exposure to wild animals is limited.\\nIn the efforts to contain the spread of the virus, the Chinese government introduced a series of All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint containment measures. 18 Specifically, to lower the risk of zoonotic virus transmission, China has issued strict regulations on wet markets where live and/or wild animals are traded, including closing live poultry markets (LPMs) and bans on wild animal transactions. Furthermore, Wuhan city has been quarantined to stop human-to-human transmission: starting on Jan 23, 2020, public transportation in Wuhan was closed, and all flights and trains leaving from Wuhan were cancelled.\\nThe COVID-19 outbreak and subsequent containment measures to lower the zoonotic virus transmission offer a unique opportunity to reassess the exposure to live animals, demand for wild animal consumption, and public readiness to further regulate wet markets. In this study, we aim to investigate behaviour patterns of population interaction to live animals, with a particular focus on differences before and during the outbreak, between Wuhan and Shanghai representing diverse exposure intensities to COVID-19. We also seek to understand levels of public support for containment measures and public confidence in their effectiveness to quell the outbreak. This study will help to clarify the public's response to the outbreak and has important policy implications regarding the social acceptability of infection containment measures and their application in different contexts.\\nA population-based cross-sectional telephone survey was conducted in Wuhan and Shanghai between Feb 1 and 10, 2020, one month since the outbreak. These two cities were selected to represent diverse exposure intensities to COVID-19. Wuhan is the site of origin of the COVID-19 outbreak. Shanghai was selected due to its status as a city that has been significantly affected by imported COVID-19 cases All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint 8 from Wuhan. The first imported case was confirmed in Shanghai on Jan 20. During the period of our survey, the number of confirmed cases rose rapidly in Wuhan, from 4,109 to 18,454, but maintained low epidemic rates in Shanghai, with reported cases rising from 182 to 303. Figure 1 illustrates the timeline of the outbreak progression and containment measures. This provides an opportunity to assess and compare public behavioural response and support for containment measures during the outbreak between two cities with different exposure intensities.\\nThe sample size (n=500 for each city) is sufficient to estimate the change of outcomes (p=0.5) with a margin of error of 0.05 and a 95% confidence interval, according to previous studies. [15] [16] Our target population was persons aged 18 or over who had been living in the selected city for at least 6 months in the year prior to the survey and were current residents at the time of the survey. In each city, the telephone survey was conducted through a computer-assisted interviewing system, which enabled the random generation of mobile phone numbers and systematic data collection. We attempted to contact each generated number three times a day at different hours; if no contact was established after the third call, this number was classified as invalid or unreachable. Random dials with proportional quota sampling were used to ensure that respondents were demographically representative of the general population in each city, with quotas based on sex and age. 19\\nThe outcomes of interest we studied were exposure to live animals before and during the COVID-19 outbreak, public support for containment measures that aim to reduce population exposure to live animals and human transmission, and confidence in these measures to quell the outbreak. Exposure to All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint live animals includes the following behaviours: visiting the live poultry or wild animal markets, buying live poultry or wild animals (i.e., frequency of purchases, practice of picking up live animals before purchasing, location where the purchased animals were slaughtered), eating wild animal products, raising pets, and touching stray animals. To measure behavioural change during the outbreak, each respondent was asked to report exposing behaviours both (1) prior to the outbreak: before Dec 31, 2019 when COVID-19 cases were first released by Wuhan Health Commission and (2) since the outbreak, between Jan 1 to Feb 10, 2020. For live animal consumption habits, where behaviour change cannot be measured in a short period of time, we assessed people's intention to: 1) buy live poultry or wild animals, and eat wild animal products (continue to buy/eat as usual, buy/eat less, not buy/eat any more) and 2) continue raising pets and touching stray animals (continue to raise/touch as usual, temporarily not during the outbreak, try or resolutely not even after the outbreak).\\nWe measured public support for two types of governmental containment measures in response to the outbreak; one is regulation on wet markets to reduce population exposure to live animals, and the other is quarantining measures meant to reduce human transmission, such as the closure of public transportation in Wuhan and the cancellation of flights and trains leaving from Wuhan. Each respondent was asked to rate the extent, on a five-point Likert scale, to which they supported 1) the permanent closure of LPMs, 2) a ban on the wild animal trade, and 3) the quarantining of Wuhan; respondents were also asked to report the extent of their confidence in the effectiveness of these measures to quell the outbreak.\\nCharacteristics of respondents, including gender, age, marital status, education, income, and All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint employment status, were collected in our survey. We also collected the history of fever or cough symptoms for respondents and their relatives or friends in the two weeks prior to the survey, and whether there were confirmed or suspected COVID-19 cases in the neighbourhood in which they live.\\nEach respondent was also asked at what point they started paying attention to this outbreak.\\nThe survey questionnaire was adjusted according to an instrument used during the outbreaks of SARS in 2003 20-21 and influenza A(H7N9) in 2013. 15 The survey instrument was pretested for face and content validity, length, and understandability by the public before our survey was administered.\\nDescriptive and univariate analysis were used to investigate changes in exposure to live animals during the outbreak, public support for containment measures, and confidence in containment measures to quell the outbreak, between Wuhan (the epicentre) and Shanghai (an affected city with imported cases).\\nChi-square tests or fisher's exact tests (when applicable) were used to compare differences in these measures between the two cities, and before and during the outbreak. Multivariate logistic regression was used to estimate factors associated with public support for containment measures and confidence in intervention effectiveness. The proportion and Odds Ratio with 95% confidence intervals (CI) were reported. All statistical analyses were performed using Stata 14.0 (StataCorp LP, College Station, TX).\\nA total of 18,576 telephone numbers were randomly selected via digital dialing system in Wuhan and Shanghai. After calling, 6,836 respondents were eligible to participate, of whom 5,640 declined and All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint 11 185 terminated before completing the interview. In total, 510 respondents in Wuhan and 501 respondents in Shanghai completed the interview, and the overall response rate was 14.8%\\n(1,011/6,836) (Appendix Figure 1 ).\\nRespondents in Wuhan and Shanghai were generally similar in terms of sex, marital status and occupation, but in Shanghai respondents tended to be older and had higher education and income levels than in Wuhan (Table 1 ). Our sample was consistent with characteristics of the general population in the two cities, as Shanghai is more developed and has an older population than Wuhan. Expectedly, respondents in Wuhan were much more intensely exposed to the COVID-19 outbreak and paid attention to the outbreak earlier than those in Shanghai.\\nWe first investigated patterns of usual exposure to live animals before the outbreak ( Table 2 Panel of 55, 58.2% in Shanghai, p=0.339); more than 90% of respondents always arranged for slaughter of purchased animals in wet markets, with no noticeable differences between the two cities. However, residents in Wuhan reported wild animal consumption at a higher rate than those in Shanghai; 47 (9.2%) of 510 respondents in Wuhan and 23 (4.6%) of 501 respondents in Shanghai reported eating All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is To investigate effects of the outbreak, we further assessed behavioural changes in exposure to live animals before and during the outbreak (Figure 2) , and intention to change after the outbreak ( Table 2 Panel B). We found that respondents dramatically reduced exposure to live animals during the outbreak, with respondents in Wuhan changing their exposure much more significantly than in Shanghai. Specifically, the frequency of visiting LPMs decreased from 23.3% (n=119) to 3.1% (n=16)\\nin Wuhan (p<0·001), and from 20.4% (n=102) to 4.4%(n=22) in Shanghai (p<0·001; Figure 2 ). The proportion of respondents buying live poultry or wild animals also decreased from around 11.0% to 2.6% in both cities (p<0·001). We also asked whether respondents intended to change their habit of buying live poultry or wild animals since the outbreak among those exposing to live animals at usual ( (4/55) of respondents in Wuhan and 25.5% (14/55) in Shanghai stated they would continue to buy these products as usual.\\nThe proportion of respondents consuming wild animals declined significantly, 10.2%(n=52) to 0.6%(n=3) in Wuhan (p<0·001) and 5.2%(n=26) to 0.8%(n=4) in Shanghai (p<0·001) (Figure 2 ). Most All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is Wuhan, respectively, and 75% of respondents were confident that governmental containment measures would quell the outbreak. Table 3 presents socio-demographic factors associated with public support for the governmental containment measures implemented (i.e. closure of LPMs, banning the wild animal trade, and city quarantine) and confidence in the effectiveness of governmental response. There was evidence that females were more supportive towards all three containment measures and that there was a positive association between education level and the likelihood of supporting regulations on wet markets and trade of wild animals. Unsurprisingly, those who self-reported visiting a wet market and eating wild All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint animal products prior to the outbreak are much less likely to support closure of the market or a ban on the trade as containment measures (adjusted OR=0.58, 95%CI [0.41-0.82] and adjusted OR=0.31, 95%CI [0.14-0.70], respectively). We did not find any difference in population profiles of those who support Wuhan quarantine or show confidence in the effectiveness of governmental response.\\nThis study is the first to investigate behavioural changes of exposure to live animals during the COVID-19 outbreak, and public support for governmental containment measures in response to the outbreak. We found that population exposures to live animals were similar in Wuhan and Shanghai before the outbreak, except that wild animal consumption was more prevalent in Wuhan than in Shanghai. During the outbreak, residents in both cities significantly changed their behaviours and reduced exposures to live animals, and such changes were much more significant in Wuhan. Most respondents supported governmental containment measures implemented, and three in four had confidence on the effectiveness of these measures. Male, the less educated and those exposing to live animals at usual were less likely to support containment measures.\\nIn our survey, more than 20% of respondents visited wet markets at usual, which was similar in Wuhan and Shanghai, despite Shanghai being much more developed. Compared with the 30% level reported in 2013 in the same study sites, 15 the share of respondents visiting wet markets declined by 10 percentage points, but was still high. 10% of respondents consumed products of wild animals in Wuhan, twice that reported in Shanghai. In China, there exists a long tradition of eating the meat and products of wild animals, which persists into modern times due to a false but commonly held belief that these products All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint have therapeutic effects. [22] [23] These beliefs and preferences for fresh food are especially popular in southern China, where Wuhan is located, but were greatly challenged and changed during the 2003 SARS outbreak in China 9,12 . However, these practices have become popular again in recent years, due in part to the popularity of social media influencers who broadcast themselves eating wild animal products on livestream platforms. 22 The revival of this tradition may have set the socio-biological context for the occurrence and spread of COVID-19 in China.\\nWe found that respondents' risks of exposure to live animals declined significantly, reaching a very low level, and most respondents stopped wild animal consumption after this outbreak, especially in Wuhan.\\nThis indicates that the public responded quickly and adequately to the outbreak. In the survey, all respondents cared about the outbreak, and about one in five respondents in Wuhan reported starting to pay attention before Dec 31, 2019. The outbreak gradually gained intense nationwide attention, 24 and a series of policies were implemented by the Chinese government to contain the outbreak. As of Jan 30, all provinces had initiated first-level responses to major public health emergencies. [25] [26] These actions raised public attention further and induced significant behavioural changes. A national online survey on Jan 26 showed that the vast majority of people knew of the disease and related preventive measures, 66.5% reported that they would wear masks when outside their home, and 80% cancelled travel plans during the Spring Festival due to the outbreak. 27 The findings from both this survey and ours demonstrate behavioural change in terms of exposure to live animals or daily protection.\\nIn addition, more strict regulations were implemented on wet markets to lower the risk of disease transmission. Wuhan instituted prohibited the sale of live poultry and wild animals starting on Jan 22. All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint Shanghai started its seasonal closure of LPMs on Jan 25 this year. The Chinese Government further instituted nationwide bans on all forms of wild animal transactions on Jan 26. 28 These regulations may accelerate the decrease in exposing behaviours to live animals. However, the COVID-19 epidemic occurred prior to the Chinese lunar new year, when the demand for live animals is generally high. 23 This may produce a conflict between the public demand for live animals and regulations. Fortunately, we found that the public strongly supports the regulations on wet markets to control the outbreak.\\nAlthough permanent closure of LPMs was not fully backed by the public, it was still much higher than reported during the 2013 influenza A outbreak (27.3% in Wuhan, 39.3% in Shanghai). 15, 29 The strong support for this policy offers a great opportunity to improve regulations on wet markets and change the prevalent habit of eating wild animal products in China.\\nWe believe that the keys to quelling outbreaks of zoonotic viral diseases such as COVID-19 and preventing future outbreaks are changing the false beliefs and dangerous habits regarding eating wild animal products and strengthening regulations on wet markets. Firstly, healthy diets need to be vigorously promoted in China. Health China 2030 promotes healthy diets, but does not discuss the risks of eating wild animals. We also believe the internet and social media can be an effective channel to promote healthy eating habits and reduce the inappropriate and outdated tradition of eating wild animal products. 25, 30 Since the outbreak, spontaneous initiatives have emerged on the internet to explain the risks of and resist consuming wild animals. Secondly, live animal transactions should be more strictly regulated, and wild animal transactions should be even prohibited in the long term. While different animals should be sold separately in specific wet markets to lower risks of virus transmission, in practice, various kinds of live animals are sold together within one market, as found in Huanan Seafood All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint Wholesale Market. Laws should be strengthened to regulate and implement the strict separation of live animals from general markets. With the currently high public support, bans on the wild animal trade should be promoted, but the government must be aware of the risk of creating an unregulated black market by instituting a ban. For live poultry and non-wild animals, better regulations may be a better option than bans without the full support of the public. 23 The feasible measures include seasonal closure of wet markets, regular rest days, and banning leaving live animals in markets overnight. 10, 15 The above measures should be given priority in China and other countries to reduce current levels of COVID-19, as well as future threats from zoonotic diseases.\\nOur study has several limitations. First, there may be recall bias. We aimed to identify behavioral changes in a cross-sectional survey, and outcomes before the outbreak were measured based on participant recall. Second, our results may be affected by selection bias from telephone survey. We did attempt multiple calls to unanswered numbers to mitigate this bias, and quota sampling was introduced to get a representative sample. The anonymous phone interview addressed the concern of response bias due to fear of being seen as criticizing the government. Third, a cross-sectional design prevents us from assessing how long the effect on behavioral/intention change demonstrated in this study will last, when the outbreak is no longer fresh in people's minds. Thus, we plan to conduct a second-round survey at the end of the outbreak to track the long-term changes.\\nIn conclusion, exposure to live animals was common before the COVID-19 outbreak. The public responded quickly to this outbreak, and significantly reduced their exposure to live animals and stopped wild animal consumption, especially in Wuhan. They were very supportive and confident for All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint current mitigating measures. With high public support, wet markets should be better regulated, and healthy diets, including changing the habit of wild animal consumption, should be promoted, especially for male, the less educated and those exposing to live animals at usual. \\nThe study was approved by the Institutional Review Board of the School of Public Health, Fudan University (IRB#2020-TYSQ-01-1). Verbal informed consent was obtained from all participants.\\nH.Y. has received research funding from Sanofi Pasteur, GlaxoSmithKline, Yichang HEC Changjiang Pharmaceutical Company and Shanghai Roche Pharmaceutical Company. None of that research funding is related to 2019-nCov. All data are n (%). P value are from Chi square test or fisher's exact tests (when applicable).\\n*Fever or cough symptoms in the past 2 weeks.\\nAll rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint Resolutely not after outbreak 6(14.0) 3(5.2) All rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity. All rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity. Odds ratios with 95% CI were obtained from multivariate binary logistic regressions. Significance level: ** p<0.01, * p<0.05.\\nAll rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint WHO stands for World Health Organization; PHEIC stands for public health emergency of international concern.\\nAll rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint Notes: Proportions are shown, and error bars represent 95% CIs.\\nAll rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint\\nNotes: p value is shown between Wuhan and Shanghai.\\nAll rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint \\n  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>535</td>\n",
       "      <td>Comparison of alternative models of human movement and the spread of disease</td>\n",
       "      <td>[{'text': 'Predictive models for the spatial spread of infectious diseases has received much attention in recent years as tools for the management of infectious diseas outbreaks. Prominently, various versions of the so-called gravity model, borrowed from transportation theory, have been used. However, the original literature suggests that the model has some potential misspecifications inasmuch as it fails to capture higher-order interactions among population centers. The fields of economics, geography and network sciences holds alternative formulations for the spatial coupling within and among conurbations. These includes Stouffer's rank model, Fotheringham's competing destinations model and the radiation model of Simini et al. Since the spread of infectious disease reflects mobility through the filter of age-specific susceptibility and infectivity and since, moreover, disease may alter spatial behavior, it is essential to confront with epidemiological data on spread. To study their relative merit we, accordingly, fit variants of these models to the uniquely detailed dataset of prevaccination measles in the 954 cities and towns of England and Wales over the years 1944-65 and compare them using a consistent likelihood framework. We find that while the gravity model is a reasonable first approximation, both Stouffer's rank model, an extended version of the radiation model and the Fotheringham competing destinations model provide significantly better fits, Stouffer's model being the best. Through a new method of spatially disaggregated likelihoods we identify areas of relatively poorer fit, and show that it is indeed in densely-populated conurbations that higher order spatial interactions are most important. Our main conclusion is that it is premature to narrow in on a single class of models for predicting spatial spread of infectious disease. The supplemental materials contain all code for reproducing the results and applying the methods to other data sets.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'The ability to predict how infectious disease will spread is of great importance in the 1 face of the numerous emergent and re-emergent pathogens that currently threatening 2 : bioRxiv preprint human well-being. We identified a variety of alternative models that predict human 3 mobility as as a function of population distribution across a landscape. These consider 4 some models that account for pair-wise interactions between population centers, as well 5 as some that allow for higher-order interactions. We trained the models using a 6 uniquely rich spatiotemporal data set on pre-vaccination measles in England and wales 7 (1944-65), which comprises more than a million records from 954 cities and towns.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Likelihood rankings of the different models reveal strong evidence for higher-order 9 interactions in the form of competition among cities as destinations for travelers and, 10 thus, dilution of spatial transmission. The currently most commonly used so-called 11 'gravity' models were far from the best in capturing spatial disease dynamics.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': '12 42 movement patterns [13] including a recent incarnation as the 'radiation' model [14].', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'While these models have proved useful in their original applications, their use in 44 describing infectious disease spread is complicated by two considerations. First, spatial 45 interaction models predict bulk migration fluxes between population centers, but the 46 impact of these movements on the dynamics of infectious disease can depend strongly 47 not only on the magnitude, but also on the composition, of the migrant pool. In 48 particular, the extent to which migrants are more or less likely to be susceptible to 49 infection, or actually infected, than the general population can be critically important, 50 as can the age profile of the migrant pool, due to pronounced age-specific patterns of 51 behavior, susceptibility, and infectiousness [5, 7, 15] . Second, infectious disease dynamics 52 and Wales using a spatial hazards approach [18] . We considered four main classes of 268 models, including the gravity, competing destinations, Stouffer rank, and radiation 269 models, in addition to pure diffusion and mean field models which are interesting special 270', 'cite_spans': [{'start': 754, 'end': 757, 'text': '[5,', 'ref_id': 'BIBREF4'}, {'start': 758, 'end': 760, 'text': '7,', 'ref_id': 'BIBREF6'}, {'start': 761, 'end': 764, 'text': '15]', 'ref_id': 'BIBREF13'}, {'start': 849, 'end': 853, 'text': '[18]', 'ref_id': 'BIBREF16'}], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>Accurately predicting the geographical spread of emerging, re-emerging, and recurrent 14 epidemics in an increasingly globalized world is a matter of international urgency in the 15 wake of outbreaks of emerging pathogens such as severe acute respiratory syndrome 16 (SARS), Middle East respiratory syndrome (MERS), and ebola virus disease (EBVD) as 17 well as re-emerging and resurgent pathogens such as influenza subtype A-H1N1, 18 whooping cough, and measles. In response to this challenge, various 'spatial interaction' 19 models describing human movement as a function of population distribution have been 20 proposed. Some of these borrow from economics and human geography while others 21 adapt models from movement ecology and the physics of reaction and diffusion on 22 heterogenous landscapes. 23 In recent years, the field has seen the widespread adoption of a family of so-called 24 'gravity' models from transportation theory and human geography (see, e.g., [1, 2] ). In 25 its most common form, the gravity model posits that the migration flux between a pair 26 of cities is log-linearly dependent on their respective sizes and on their separating 27 distance. The application of this simple model to disease spread was originally proposed 28 by Murray and Cliff [3] , but over the last decade, many studies have used it to explain 29 historical, or predict future, disease spread in a range of infections including measles [4] , 30 influenza [5] [6] [7] , cholera [8] , and yellow fever [9] . While its application has yielded 31 insights, the geography literature has highlighted a prominent shortcoming of the 32 gravity model. In particular, these models ignore the potential for competitive or 33 synergistic interactions among population centers ( Fig. 1 ; [10, 11] ). Thus, for example, 34 movement between Boston and Washington DC is assumed to be unaffected by the 35 presence of the intervening city of New York. 36 As it happens, there are distinct families of models, arising from economics and 37 geography, that predict human movement patterns while allowing for higher-order 38 interactions among cities. In particular, Stouffer's [12] 'law of intervening opportunities' 39 posits that \"the number of persons going a given distance is directly proportional to the 40 number of opportunities at that distance and inversely proportional to the number of 41 intervening opportunities\". This idea has given rise to alternative models for human can feed back onto migration. Most obviously, disease symptoms can influence 53 movement behavior, as seen, for example, in the fact that a mild cold may induce 54 minimal changes in movement behavior but severe hemorrhagic fever or acute paralytic 55 disease will typically slow the movement of the infected hosts. These two considerations 56 do not preclude the utility of spatial interaction models in the infectious disease context; 57 they do, however, complicate their use and the importance of understanding 58 transmission relevant migration fluxes, which will be some sort of 'effective average' of 59 movement as filtered through such aforementioned complications.\\nSeveral spatial interaction models have been parameterized and tested using various 61 mobility data such as commuter flows (e.g., [5, 14] ), mobile phone geolocations 62 (e.g., [16] ), social media (e.g., [13] ), and microsimulations (e.g., [17] ). However, in view 63 of the challenges just noted, the ultimate test of the models is against data on the actual 64 spread of infection rather than bulk movement of people or cell phones. Bjørnstad and 65 Grenfell [18] proposed that for acute immunizing infections, the spatio-temporal 66 patterns of fade-outs (i.e., local disease extinction) across metapopulations provide 67 valuable information on disease spread because these patterns reflect spatial 68 transmission unclouded by local transmission. In this paper, we extract information 69 from fade-out patterns and a consistent likelihood framework to compare and contrast 70 to a suite of models including (i) the gravity model [1] , (ii) Fotheringham's competing 71 destinations model [10] , (iii) Stouffer's rank model [12] , and (iv) the radiation model [14] . 72 . We confront these models weekly data on measles incidence from all 954 cities and 73 towns in England and Wales from 1944 to 1965 [4, 19] . Comparing fits and predictions, 74 we show that while the gravity model is a reasonable first approximation, Stouffer's 75 rank model, an extended version of the radiation model, and the competing destinations 76 model all provide significantly better fits, Stouffer's model performing the best.\\nData 79 Historical incidence of measles in England and Wales has been an influential testbed for 80 models and methods in disease dynamics since Bartlett's [20, 21] seminal work on its 81 recurrent epidemics. We use the spatially resolved weekly measles data across all 954 82 cities and towns of England and Wales from 1944, when notification was made 83 mandatory by the UK Registrar General (OPCS), until 1965, which saw subtle shifts in 84 political boundaries around London. Vaccination was not introduced in the UK until 85 1967, so that these data span a period where measles dynamics were unaffected by mass 86 immunization. The data set is complete except for a region-wide underreporting rate of 87 around 50% [22, 23] . Grenfell et al. [19] give a detailed description of the data; the 88 entire data set has been made available by Lau et al. [24] . 89 An important feature of the system is that, between 29% and 38% of the population 90 (c. 47M during this period) resided in a small number (13-28 depending on exact 91 definition) of communities above a critical community size (CCS) of c. 250-300k. Cities 92 larger than the CCS tend to sustain local chains of transmission. The remaining 60-70% 93 were distributed among the more than 900 communities smaller than the CCS where 94 local extinctions are more or less frequent (depending on population size and degree of 95 isolation) and, consequently, the rate of reintroduction of the pathogen via spatial 96 transmission is an important determinant of measles incidence. In ecological terms, the 97 prevaccination measles system represented an epidemic mainland-island metapopulation 98 (e.g., [25, 26] ). Our analysis exploits this fact, using the timing and spatial pattern and 99 timing of reintroductions to inform the parameters of each of the spatial interaction 100 models.\\nUsing a spatially-extended time series susceptible-infected-recovered (TSIR) framework 103 (e.g., [4, 27] ), we can calculate, for each population center, the probability that a spatial 104 interaction happens (i.e., a contact between a resident susceptible host and a 105 non-resident infectious host) and that the contact results in a new local chain of 106 transmission [18] . In previous analyses, we showed that the signature of such events will 107 be drowned out in the presence of endemic circulation (see [23] ). Following a local 108 extinction, however, the re-colonization rate contains critical information on the spatial 109 interactions.\\nThe spatially-extended TSIR model predicts that the expected number of new 111 infected hosts in community i in epidemic generation t + 1 will be\\nHere, β s(t) is the seasonal transmission rate as molded by opening and closing of schools 113 and the proportion of transmission that occurs within the school setting as the school 114 year progresses [23] ; s(t) = t mod 26 is the seasonality function; S i,t /N is the 115 probability that a local individual is susceptible; I i,t is the local number of infections; α 116 is a correction for the discrete-time approximation of the underlying continuous-time 117 process [28] ; and ι i,t is the sum of transmission-relevant spatial interactions i has with 118 the other communities in the epidemic metapopulation. Following a local extinction,\\nAssuming demographic stochasticity in transmission due to the underlying epidemic 121 birth-and-death process, the realized number of cases in the next generation will be [23] : 122\\nAs a consequence, the number of susceptibles in the next generation will be\\nwhere B i,t is the recruitment of local susceptibles through births during the generation 124 interval. The 'serial interval' for measles is 10-14 days [29] , so we follow previous TSIR 125 analyses in aggregating the weekly data in 2-week increments (though [23] investigate 126 shorter intervals). Using susceptible reconstruction methods [30] , we estimate all local 127 parameters using the method of Finkenstadt et al. [31] . The online supplement contains 128 full documentation of these analyses and code for exact replication of all results 129 presented in this paper.\\nConditioning on the local parameters and restricting analyses at each location to 131 biweeks in which measles incidence was zero, we use Eqs. 2 and 4 to form a negative 132 binomial likelihood to estimate all parameters of each of the candidate spatial 133 interaction models.\\nSpatial interaction models 135 When including special cases, such as pure diffusion and variants, we consider a total of 136 10 spatial interaction models (Fig. 1) . Each of these amounts to a specification of the ι 137 term in Eqs. 1 and 2.\\nGravity model. Under the gravity model, the spatial interaction between locations i 139 and j takes the form θN τ1 i N τ2 j d −ρ ij , where θ, τ 1 , τ 2 , and ρ are non-negative parameters. 140 This leads to the following formulation for disease-relevant spatial interactions:\\nwhere the sum is across all non-self potential donor communities, and I j /N j is the 142 fraction of infected individuals in donor community j. The gravity model has two 143 important special cases: ρ = 0, τ 1 = τ 2 = 1, which is a mean-field model, and 144 τ 1 = τ 2 = 0, which is simple spatial diffusion. The φ parameter represents background 145 spatial transmission that is not predictable on the basis of distance and size [24] . 146 Xia's model. In the original analysis of the spatiotemporal dynamics of measles, Xia et al. [4] used a formulation rooted in the gravity-model literature but with a slightly different formulation:\\nwhich is the same as Eq. 5 if τ 2 = 1, but not otherwise.\\nCompeting destinations model. Fotheringham [11] noted that gravity models may be misspecified because they only consider pairwise interactions among locations. He argued that nearby or intervening destinations may, in fact, make spatial interactions either more or less likely. A synergistic effect occurs, for example, if individuals are disproportionately inclined to do all their shopping in districts with many shops. Antagonistic effects may arise where travel between two cities, say Boston and Washington, is made less likely by the presence of an intervening city such as New York because individuals from either Boston or Washington can get their out-of-town needs fulfilled by visiting New York. Under Fotheringham's [11] competing destinations formulation the flux between i and j takes the form θ N τ1\\nAs in the gravity model, τ 1 and τ 2 control how 'eagerness to travel' and city 'attractiveness' scales with population size; and ρ measures how the likelihood of travel decays with distance. The parameter δ quantifies how destinations k, of various sizes, at distances d jk from the donor j, modulate the spatial interaction between recipient i and and donor j. In particular, δ &gt; 0 indicates a synergistic effect; δ &lt; 0, an antagonistic effect. The resultant formulation for disease-relevant spatial interactions between community i and everywhere else is:\\nwhere φ again represents background, spatially unpredictable spread. 148 Stouffer's rank model. Noulas et al. [13] recently proposed that human mobility 149 patterns should be studied using Stouffer's [12] rank (\"law of intervening author/funder. This article is a US Government work. It is not subject to copyright under 17 USC 105 and is also made available for use under\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2019.12.19.882175 doi: bioRxiv preprint population size as a proxy for \"opportunities\". Accordingly, we let Ω(i, j) be the 155 collection of towns located closer to town i than is town j:\\n156 The Stouffer model for disease-relevant spatial   157 interactions is, then\\nStouffer's original model was framed in the context of continuous distribution of 159 population across a landscape. In the present metapopulation context, we therefore 164\\nExtended radiation model. The radiation model proposed by Simini et al. [14] was derived independently but is in spirit related to the original ideas of Stouffer [12] . Our metapopulation version of this model with respect to disease-relevant spatial interactions is\\nAs with the Stouffer model, we consider the set Ω(i, j) to either exclude ('radiation\") or 166 include ('radiation variant\") community size i itself in the spatial interaction \\nOf the models considered, the gravity variants (including mean-field and pure 174 diffusion) include purely pairwise interactions whilst the other models allow for 175 higher-order interactions either explicitly (as in the competing destination model) or 176 implicitly (as in the Stouffer and radiation models).\\nStatistical inference 178 We use maximum likelihood to estimate the parameters of each spatial interaction 179 model and profile likelihoods to study correlation among parameters and possible 180 identifiability issues [27] . author/funder. This article is a US Government work. It is not subject to copyright under 17 USC 105 and is also made available for use under\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . Table 1 . MLE estimates. Models: SV -Stouffer variant, XR -extended radiation, CD -competing destinations, S -Stouffer, G -gravity, X -Xia, RV -radiation variant, R -radiation, MF -mean field, D -diffusion. Here, denotes the log likelihood,ĉ is the computed variance inflation factor, ∆QAIC is the QAIC relative to that of the best-fitting model, and the other parameters are as in the model equations. biweek of local disease absence [18] ) All calculations are documented in the online 189 supplement, which includes code for the detailed replication of all results.\\n190 Table 1 shows estimated parameters and likelihoods for each of the spatial interaction networks. To explore the drivers among model differences in fit, we therefore contrast 214 model predictions of disease import and export rates. Fig. 2 shows the mean predicted 215 export and import rates vs. population size for each of the models. The dependence of 216 mean export rate on population size contrasts somewhat among models ( Fig. 2A) , but 217 the importation rates follow similar patterns (Fig. 2B) . Despite their differences in 218 underlying mathematical equations, there must be more nuanced differences among the 219 models that contribute their relative fit/lack-of-fit. This begs the question: \"What are 220\\nDecember 16, 2019 7/15 a CC0 license.\\nauthor/funder. This article is a US Government work. It is not subject to copyright under 17 USC 105 and is also made available for use under\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2019.12.19.882175 doi: bioRxiv preprint these differences and are they biologically meaningful or spatially random?\" 221 A cursory review of the recent literature suggests that the gravity model is the most 222 prominent in the infectious disease context, followed by the radiation model. Stouffer's 223 model and the competing destinations model have rarely been applied in this field. It is 224 therefore useful to use the gravity model fit as a baseline and explore how the 225 better-fitting models diverge from this baseline. Comparing matrices with half a million 226 entries is very difficult, so we amploy a new 'spatial likelihood contrast' (SLiC) method 227 with which to study the relative merit of the different spatial interaction models. The 228 idea is to disaggregate the overall hazard likelihoods by individual cities to study how 229 particular communities contribute to improved or diminished relative fit of each model. 230 To do so we normalize each location's contribution to the overall likelihood by the 231 number of data points each conurbation contributes to the likelihood (the accumulated 232 stretches of measles absence) and then map the model-model differences onto the 233 landscape (Fig. 3) . Inspection of Fig. 3 suggests that the poorer fit of the gravity model 234 vs. the Stouffer or competing destinations models is primarily in the urban northwest. 235 The supplement provides the full set of pairwise SLiC contrasts among all models. To 236 test whether the apparent patterns are statistically significant, we compute local 237 indicators of spatial association (LISA; [34] ) statistics for all population centers with 238 fewer than 50k inhabitants (Fig. 3A, B) . The main failure of the gravity model is in (e.g. [35, 36] ). In the case of epidemics playing out within a population center, models 254 closer to the mean field model (i.e., with uniform connectivity) often do remarkably well 255 (see [37] for a social network perspective on this). However, such models generally do 256 not capture regional spread of human infections because of the complicated patterns of 257 human movement (the third wave of the 2009 influenza pandemic across the USA, 258 perhaps, being an unusually diffusive counterexample [7] ) . To elucidate these patterns, 259 many empirical studies have been performed and a variety of abstractions have been 260 proposed. The task of choosing among these abstractions for help in help predicting 261 spread of infectious disease is complicated by the fact that (i) model formulations may 262 differ with respect to how well they actually reproduce aggregate mobility patterns, and 263 (ii) model fit may be shaped by the manner in which mobility is filtered by transmission 264 and behavior.\\nTo advance the discussion of these issues, we considered a suite of candidate models 266 of spatial interactions and fit them to the pre-vaccination measles data from England cases. The gravity model-which has gained recent prominence in infectious disease As the gravity model has had a very rapid adoption in spatial disease epidemiology 284 during the last decade (though Noulas et al. [13] have noted the importance of 285 considering alternatives), we used this as a baseline. Despite its common usage it 286 significantly under-performs relative to the several other models. Combining SLiC and 287 LISA statistics we found that the greatest tension between the gravity model and the 288 preferred alternatives is in the Manchester-Liverpool-Leeds conurbation of northwestern 289 England, where many towns and villages commingle amongst several major cities. formulation) that spatial interaction, once discounted for higher-order interference, is 295 roughly proportional to the geometric mean of the community sizes. This curious 296 suggestion warrants future investigation.\\nIn this study we have used actual disease incidence data, as opposed to raw 298 measurments of human movement, to further our understanding of the principles 299 governing how human hosts spread ifectious disease across a populated landscape. We 300 have found both strong evidence of higher-order interactions among population centers, 301 and that candidate models differ significantly in their ability to capture the empirical 302 patterns. We hope our findings will help stimulate a systematic, data-driven discussion 303 of the relative merit of alternative predictive models for the probable path for spatial 304 spread of infectious disease. Clearly, the prospects are good for good for further 305 refinements and improved parameterizations. Spatial interaction models predict the flux of human movements between population centers (cities, towns, villages) as a function of the distribution of the population on the landscape. In this diagram, the relative magnitude of the fluxes from a focal town to other population centers are represented by the widths of the arrows. In the widely-employed gravity models (A), interactions among cities is strictly pairwise. Thus the addition of a new town (B) has no effect on the movement to other towns. In Fotheringham's competing destinations model (C), however, competition or synergy among nearby communities can reduce or augment fluxes. Stouffer's model of intervening opportunities and the radiation model (D) posit that movement from one city to another is diminished by the presence of opportunities in communities more proximal to the source city. author/funder. This article is a US Government work. It is not subject to copyright under 17 USC 105 and is also made available for use under\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2019.12.19.882175 doi: bioRxiv preprint\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                            title  \\\n",
       "535  Comparison of alternative models of human movement and the spread of disease   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        abstract  \\\n",
       "535  [{'text': 'Predictive models for the spatial spread of infectious diseases has received much attention in recent years as tools for the management of infectious diseas outbreaks. Prominently, various versions of the so-called gravity model, borrowed from transportation theory, have been used. However, the original literature suggests that the model has some potential misspecifications inasmuch as it fails to capture higher-order interactions among population centers. The fields of economics, geography and network sciences holds alternative formulations for the spatial coupling within and among conurbations. These includes Stouffer's rank model, Fotheringham's competing destinations model and the radiation model of Simini et al. Since the spread of infectious disease reflects mobility through the filter of age-specific susceptibility and infectivity and since, moreover, disease may alter spatial behavior, it is essential to confront with epidemiological data on spread. To study their relative merit we, accordingly, fit variants of these models to the uniquely detailed dataset of prevaccination measles in the 954 cities and towns of England and Wales over the years 1944-65 and compare them using a consistent likelihood framework. We find that while the gravity model is a reasonable first approximation, both Stouffer's rank model, an extended version of the radiation model and the Fotheringham competing destinations model provide significantly better fits, Stouffer's model being the best. Through a new method of spatially disaggregated likelihoods we identify areas of relatively poorer fit, and show that it is indeed in densely-populated conurbations that higher order spatial interactions are most important. Our main conclusion is that it is premature to narrow in on a single class of models for predicting spatial spread of infectious disease. The supplemental materials contain all code for reproducing the results and applying the methods to other data sets.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'The ability to predict how infectious disease will spread is of great importance in the 1 face of the numerous emergent and re-emergent pathogens that currently threatening 2 : bioRxiv preprint human well-being. We identified a variety of alternative models that predict human 3 mobility as as a function of population distribution across a landscape. These consider 4 some models that account for pair-wise interactions between population centers, as well 5 as some that allow for higher-order interactions. We trained the models using a 6 uniquely rich spatiotemporal data set on pre-vaccination measles in England and wales 7 (1944-65), which comprises more than a million records from 954 cities and towns.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Likelihood rankings of the different models reveal strong evidence for higher-order 9 interactions in the form of competition among cities as destinations for travelers and, 10 thus, dilution of spatial transmission. The currently most commonly used so-called 11 'gravity' models were far from the best in capturing spatial disease dynamics.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': '12 42 movement patterns [13] including a recent incarnation as the 'radiation' model [14].', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'While these models have proved useful in their original applications, their use in 44 describing infectious disease spread is complicated by two considerations. First, spatial 45 interaction models predict bulk migration fluxes between population centers, but the 46 impact of these movements on the dynamics of infectious disease can depend strongly 47 not only on the magnitude, but also on the composition, of the migrant pool. In 48 particular, the extent to which migrants are more or less likely to be susceptible to 49 infection, or actually infected, than the general population can be critically important, 50 as can the age profile of the migrant pool, due to pronounced age-specific patterns of 51 behavior, susceptibility, and infectiousness [5, 7, 15] . Second, infectious disease dynamics 52 and Wales using a spatial hazards approach [18] . We considered four main classes of 268 models, including the gravity, competing destinations, Stouffer rank, and radiation 269 models, in addition to pure diffusion and mean field models which are interesting special 270', 'cite_spans': [{'start': 754, 'end': 757, 'text': '[5,', 'ref_id': 'BIBREF4'}, {'start': 758, 'end': 760, 'text': '7,', 'ref_id': 'BIBREF6'}, {'start': 761, 'end': 764, 'text': '15]', 'ref_id': 'BIBREF13'}, {'start': 849, 'end': 853, 'text': '[18]', 'ref_id': 'BIBREF16'}], 'ref_spans': [], 'section': 'Abstract'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       full_text  \n",
       "535  Accurately predicting the geographical spread of emerging, re-emerging, and recurrent 14 epidemics in an increasingly globalized world is a matter of international urgency in the 15 wake of outbreaks of emerging pathogens such as severe acute respiratory syndrome 16 (SARS), Middle East respiratory syndrome (MERS), and ebola virus disease (EBVD) as 17 well as re-emerging and resurgent pathogens such as influenza subtype A-H1N1, 18 whooping cough, and measles. In response to this challenge, various 'spatial interaction' 19 models describing human movement as a function of population distribution have been 20 proposed. Some of these borrow from economics and human geography while others 21 adapt models from movement ecology and the physics of reaction and diffusion on 22 heterogenous landscapes. 23 In recent years, the field has seen the widespread adoption of a family of so-called 24 'gravity' models from transportation theory and human geography (see, e.g., [1, 2] ). In 25 its most common form, the gravity model posits that the migration flux between a pair 26 of cities is log-linearly dependent on their respective sizes and on their separating 27 distance. The application of this simple model to disease spread was originally proposed 28 by Murray and Cliff [3] , but over the last decade, many studies have used it to explain 29 historical, or predict future, disease spread in a range of infections including measles [4] , 30 influenza [5] [6] [7] , cholera [8] , and yellow fever [9] . While its application has yielded 31 insights, the geography literature has highlighted a prominent shortcoming of the 32 gravity model. In particular, these models ignore the potential for competitive or 33 synergistic interactions among population centers ( Fig. 1 ; [10, 11] ). Thus, for example, 34 movement between Boston and Washington DC is assumed to be unaffected by the 35 presence of the intervening city of New York. 36 As it happens, there are distinct families of models, arising from economics and 37 geography, that predict human movement patterns while allowing for higher-order 38 interactions among cities. In particular, Stouffer's [12] 'law of intervening opportunities' 39 posits that \"the number of persons going a given distance is directly proportional to the 40 number of opportunities at that distance and inversely proportional to the number of 41 intervening opportunities\". This idea has given rise to alternative models for human can feed back onto migration. Most obviously, disease symptoms can influence 53 movement behavior, as seen, for example, in the fact that a mild cold may induce 54 minimal changes in movement behavior but severe hemorrhagic fever or acute paralytic 55 disease will typically slow the movement of the infected hosts. These two considerations 56 do not preclude the utility of spatial interaction models in the infectious disease context; 57 they do, however, complicate their use and the importance of understanding 58 transmission relevant migration fluxes, which will be some sort of 'effective average' of 59 movement as filtered through such aforementioned complications.\\nSeveral spatial interaction models have been parameterized and tested using various 61 mobility data such as commuter flows (e.g., [5, 14] ), mobile phone geolocations 62 (e.g., [16] ), social media (e.g., [13] ), and microsimulations (e.g., [17] ). However, in view 63 of the challenges just noted, the ultimate test of the models is against data on the actual 64 spread of infection rather than bulk movement of people or cell phones. Bjørnstad and 65 Grenfell [18] proposed that for acute immunizing infections, the spatio-temporal 66 patterns of fade-outs (i.e., local disease extinction) across metapopulations provide 67 valuable information on disease spread because these patterns reflect spatial 68 transmission unclouded by local transmission. In this paper, we extract information 69 from fade-out patterns and a consistent likelihood framework to compare and contrast 70 to a suite of models including (i) the gravity model [1] , (ii) Fotheringham's competing 71 destinations model [10] , (iii) Stouffer's rank model [12] , and (iv) the radiation model [14] . 72 . We confront these models weekly data on measles incidence from all 954 cities and 73 towns in England and Wales from 1944 to 1965 [4, 19] . Comparing fits and predictions, 74 we show that while the gravity model is a reasonable first approximation, Stouffer's 75 rank model, an extended version of the radiation model, and the competing destinations 76 model all provide significantly better fits, Stouffer's model performing the best.\\nData 79 Historical incidence of measles in England and Wales has been an influential testbed for 80 models and methods in disease dynamics since Bartlett's [20, 21] seminal work on its 81 recurrent epidemics. We use the spatially resolved weekly measles data across all 954 82 cities and towns of England and Wales from 1944, when notification was made 83 mandatory by the UK Registrar General (OPCS), until 1965, which saw subtle shifts in 84 political boundaries around London. Vaccination was not introduced in the UK until 85 1967, so that these data span a period where measles dynamics were unaffected by mass 86 immunization. The data set is complete except for a region-wide underreporting rate of 87 around 50% [22, 23] . Grenfell et al. [19] give a detailed description of the data; the 88 entire data set has been made available by Lau et al. [24] . 89 An important feature of the system is that, between 29% and 38% of the population 90 (c. 47M during this period) resided in a small number (13-28 depending on exact 91 definition) of communities above a critical community size (CCS) of c. 250-300k. Cities 92 larger than the CCS tend to sustain local chains of transmission. The remaining 60-70% 93 were distributed among the more than 900 communities smaller than the CCS where 94 local extinctions are more or less frequent (depending on population size and degree of 95 isolation) and, consequently, the rate of reintroduction of the pathogen via spatial 96 transmission is an important determinant of measles incidence. In ecological terms, the 97 prevaccination measles system represented an epidemic mainland-island metapopulation 98 (e.g., [25, 26] ). Our analysis exploits this fact, using the timing and spatial pattern and 99 timing of reintroductions to inform the parameters of each of the spatial interaction 100 models.\\nUsing a spatially-extended time series susceptible-infected-recovered (TSIR) framework 103 (e.g., [4, 27] ), we can calculate, for each population center, the probability that a spatial 104 interaction happens (i.e., a contact between a resident susceptible host and a 105 non-resident infectious host) and that the contact results in a new local chain of 106 transmission [18] . In previous analyses, we showed that the signature of such events will 107 be drowned out in the presence of endemic circulation (see [23] ). Following a local 108 extinction, however, the re-colonization rate contains critical information on the spatial 109 interactions.\\nThe spatially-extended TSIR model predicts that the expected number of new 111 infected hosts in community i in epidemic generation t + 1 will be\\nHere, β s(t) is the seasonal transmission rate as molded by opening and closing of schools 113 and the proportion of transmission that occurs within the school setting as the school 114 year progresses [23] ; s(t) = t mod 26 is the seasonality function; S i,t /N is the 115 probability that a local individual is susceptible; I i,t is the local number of infections; α 116 is a correction for the discrete-time approximation of the underlying continuous-time 117 process [28] ; and ι i,t is the sum of transmission-relevant spatial interactions i has with 118 the other communities in the epidemic metapopulation. Following a local extinction,\\nAssuming demographic stochasticity in transmission due to the underlying epidemic 121 birth-and-death process, the realized number of cases in the next generation will be [23] : 122\\nAs a consequence, the number of susceptibles in the next generation will be\\nwhere B i,t is the recruitment of local susceptibles through births during the generation 124 interval. The 'serial interval' for measles is 10-14 days [29] , so we follow previous TSIR 125 analyses in aggregating the weekly data in 2-week increments (though [23] investigate 126 shorter intervals). Using susceptible reconstruction methods [30] , we estimate all local 127 parameters using the method of Finkenstadt et al. [31] . The online supplement contains 128 full documentation of these analyses and code for exact replication of all results 129 presented in this paper.\\nConditioning on the local parameters and restricting analyses at each location to 131 biweeks in which measles incidence was zero, we use Eqs. 2 and 4 to form a negative 132 binomial likelihood to estimate all parameters of each of the candidate spatial 133 interaction models.\\nSpatial interaction models 135 When including special cases, such as pure diffusion and variants, we consider a total of 136 10 spatial interaction models (Fig. 1) . Each of these amounts to a specification of the ι 137 term in Eqs. 1 and 2.\\nGravity model. Under the gravity model, the spatial interaction between locations i 139 and j takes the form θN τ1 i N τ2 j d −ρ ij , where θ, τ 1 , τ 2 , and ρ are non-negative parameters. 140 This leads to the following formulation for disease-relevant spatial interactions:\\nwhere the sum is across all non-self potential donor communities, and I j /N j is the 142 fraction of infected individuals in donor community j. The gravity model has two 143 important special cases: ρ = 0, τ 1 = τ 2 = 1, which is a mean-field model, and 144 τ 1 = τ 2 = 0, which is simple spatial diffusion. The φ parameter represents background 145 spatial transmission that is not predictable on the basis of distance and size [24] . 146 Xia's model. In the original analysis of the spatiotemporal dynamics of measles, Xia et al. [4] used a formulation rooted in the gravity-model literature but with a slightly different formulation:\\nwhich is the same as Eq. 5 if τ 2 = 1, but not otherwise.\\nCompeting destinations model. Fotheringham [11] noted that gravity models may be misspecified because they only consider pairwise interactions among locations. He argued that nearby or intervening destinations may, in fact, make spatial interactions either more or less likely. A synergistic effect occurs, for example, if individuals are disproportionately inclined to do all their shopping in districts with many shops. Antagonistic effects may arise where travel between two cities, say Boston and Washington, is made less likely by the presence of an intervening city such as New York because individuals from either Boston or Washington can get their out-of-town needs fulfilled by visiting New York. Under Fotheringham's [11] competing destinations formulation the flux between i and j takes the form θ N τ1\\nAs in the gravity model, τ 1 and τ 2 control how 'eagerness to travel' and city 'attractiveness' scales with population size; and ρ measures how the likelihood of travel decays with distance. The parameter δ quantifies how destinations k, of various sizes, at distances d jk from the donor j, modulate the spatial interaction between recipient i and and donor j. In particular, δ > 0 indicates a synergistic effect; δ < 0, an antagonistic effect. The resultant formulation for disease-relevant spatial interactions between community i and everywhere else is:\\nwhere φ again represents background, spatially unpredictable spread. 148 Stouffer's rank model. Noulas et al. [13] recently proposed that human mobility 149 patterns should be studied using Stouffer's [12] rank (\"law of intervening author/funder. This article is a US Government work. It is not subject to copyright under 17 USC 105 and is also made available for use under\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2019.12.19.882175 doi: bioRxiv preprint population size as a proxy for \"opportunities\". Accordingly, we let Ω(i, j) be the 155 collection of towns located closer to town i than is town j:\\n156 The Stouffer model for disease-relevant spatial   157 interactions is, then\\nStouffer's original model was framed in the context of continuous distribution of 159 population across a landscape. In the present metapopulation context, we therefore 164\\nExtended radiation model. The radiation model proposed by Simini et al. [14] was derived independently but is in spirit related to the original ideas of Stouffer [12] . Our metapopulation version of this model with respect to disease-relevant spatial interactions is\\nAs with the Stouffer model, we consider the set Ω(i, j) to either exclude ('radiation\") or 166 include ('radiation variant\") community size i itself in the spatial interaction \\nOf the models considered, the gravity variants (including mean-field and pure 174 diffusion) include purely pairwise interactions whilst the other models allow for 175 higher-order interactions either explicitly (as in the competing destination model) or 176 implicitly (as in the Stouffer and radiation models).\\nStatistical inference 178 We use maximum likelihood to estimate the parameters of each spatial interaction 179 model and profile likelihoods to study correlation among parameters and possible 180 identifiability issues [27] . author/funder. This article is a US Government work. It is not subject to copyright under 17 USC 105 and is also made available for use under\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . Table 1 . MLE estimates. Models: SV -Stouffer variant, XR -extended radiation, CD -competing destinations, S -Stouffer, G -gravity, X -Xia, RV -radiation variant, R -radiation, MF -mean field, D -diffusion. Here, denotes the log likelihood,ĉ is the computed variance inflation factor, ∆QAIC is the QAIC relative to that of the best-fitting model, and the other parameters are as in the model equations. biweek of local disease absence [18] ) All calculations are documented in the online 189 supplement, which includes code for the detailed replication of all results.\\n190 Table 1 shows estimated parameters and likelihoods for each of the spatial interaction networks. To explore the drivers among model differences in fit, we therefore contrast 214 model predictions of disease import and export rates. Fig. 2 shows the mean predicted 215 export and import rates vs. population size for each of the models. The dependence of 216 mean export rate on population size contrasts somewhat among models ( Fig. 2A) , but 217 the importation rates follow similar patterns (Fig. 2B) . Despite their differences in 218 underlying mathematical equations, there must be more nuanced differences among the 219 models that contribute their relative fit/lack-of-fit. This begs the question: \"What are 220\\nDecember 16, 2019 7/15 a CC0 license.\\nauthor/funder. This article is a US Government work. It is not subject to copyright under 17 USC 105 and is also made available for use under\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2019.12.19.882175 doi: bioRxiv preprint these differences and are they biologically meaningful or spatially random?\" 221 A cursory review of the recent literature suggests that the gravity model is the most 222 prominent in the infectious disease context, followed by the radiation model. Stouffer's 223 model and the competing destinations model have rarely been applied in this field. It is 224 therefore useful to use the gravity model fit as a baseline and explore how the 225 better-fitting models diverge from this baseline. Comparing matrices with half a million 226 entries is very difficult, so we amploy a new 'spatial likelihood contrast' (SLiC) method 227 with which to study the relative merit of the different spatial interaction models. The 228 idea is to disaggregate the overall hazard likelihoods by individual cities to study how 229 particular communities contribute to improved or diminished relative fit of each model. 230 To do so we normalize each location's contribution to the overall likelihood by the 231 number of data points each conurbation contributes to the likelihood (the accumulated 232 stretches of measles absence) and then map the model-model differences onto the 233 landscape (Fig. 3) . Inspection of Fig. 3 suggests that the poorer fit of the gravity model 234 vs. the Stouffer or competing destinations models is primarily in the urban northwest. 235 The supplement provides the full set of pairwise SLiC contrasts among all models. To 236 test whether the apparent patterns are statistically significant, we compute local 237 indicators of spatial association (LISA; [34] ) statistics for all population centers with 238 fewer than 50k inhabitants (Fig. 3A, B) . The main failure of the gravity model is in (e.g. [35, 36] ). In the case of epidemics playing out within a population center, models 254 closer to the mean field model (i.e., with uniform connectivity) often do remarkably well 255 (see [37] for a social network perspective on this). However, such models generally do 256 not capture regional spread of human infections because of the complicated patterns of 257 human movement (the third wave of the 2009 influenza pandemic across the USA, 258 perhaps, being an unusually diffusive counterexample [7] ) . To elucidate these patterns, 259 many empirical studies have been performed and a variety of abstractions have been 260 proposed. The task of choosing among these abstractions for help in help predicting 261 spread of infectious disease is complicated by the fact that (i) model formulations may 262 differ with respect to how well they actually reproduce aggregate mobility patterns, and 263 (ii) model fit may be shaped by the manner in which mobility is filtered by transmission 264 and behavior.\\nTo advance the discussion of these issues, we considered a suite of candidate models 266 of spatial interactions and fit them to the pre-vaccination measles data from England cases. The gravity model-which has gained recent prominence in infectious disease As the gravity model has had a very rapid adoption in spatial disease epidemiology 284 during the last decade (though Noulas et al. [13] have noted the importance of 285 considering alternatives), we used this as a baseline. Despite its common usage it 286 significantly under-performs relative to the several other models. Combining SLiC and 287 LISA statistics we found that the greatest tension between the gravity model and the 288 preferred alternatives is in the Manchester-Liverpool-Leeds conurbation of northwestern 289 England, where many towns and villages commingle amongst several major cities. formulation) that spatial interaction, once discounted for higher-order interference, is 295 roughly proportional to the geometric mean of the community sizes. This curious 296 suggestion warrants future investigation.\\nIn this study we have used actual disease incidence data, as opposed to raw 298 measurments of human movement, to further our understanding of the principles 299 governing how human hosts spread ifectious disease across a populated landscape. We 300 have found both strong evidence of higher-order interactions among population centers, 301 and that candidate models differ significantly in their ability to capture the empirical 302 patterns. We hope our findings will help stimulate a systematic, data-driven discussion 303 of the relative merit of alternative predictive models for the probable path for spatial 304 spread of infectious disease. Clearly, the prospects are good for good for further 305 refinements and improved parameterizations. Spatial interaction models predict the flux of human movements between population centers (cities, towns, villages) as a function of the distribution of the population on the landscape. In this diagram, the relative magnitude of the fluxes from a focal town to other population centers are represented by the widths of the arrows. In the widely-employed gravity models (A), interactions among cities is strictly pairwise. Thus the addition of a new town (B) has no effect on the movement to other towns. In Fotheringham's competing destinations model (C), however, competition or synergy among nearby communities can reduce or augment fluxes. Stouffer's model of intervening opportunities and the radiation model (D) posit that movement from one city to another is diminished by the presence of opportunities in communities more proximal to the source city. author/funder. This article is a US Government work. It is not subject to copyright under 17 USC 105 and is also made available for use under\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2019.12.19.882175 doi: bioRxiv preprint\\n  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#loop on the top 5 indicies and display the title, abstract, and fulltext of the paper that was fetched from the search\n",
    "if (flag == True):\n",
    "    print(\"Sorry, no results\")\n",
    "else:\n",
    "    counter =0\n",
    "    for index in answer_indicies:\n",
    "        if (cosineSimilarities[index]>=0.1): #get papers with cosine similarities greater than or equal 0.1\n",
    "            display(FullPaperDataframe.iloc[[index]])\n",
    "        else:\n",
    "            counter = counter+1\n",
    "    if (counter == len(cosineSimilarities)):\n",
    "        print(\"Sorry, no results\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h4>STEP 4: LITERATURE CLUSTERING</h4>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<p>The second part of this notebook attempts to before K-Means clustering on the papers, such that each group of papers related together via certain keywords are grouped together.</p> \n",
    "<h6 style = \"color:green\"> PROS of using K-means</h6>\n",
    "<p> The pros of using K-means is that many packages provide its functionality and it is easy to implement.</p>\n",
    "<h6 style = \"color:red\"> CONS of using K-means</h6>\n",
    "<p> When clustering a new paper, the distance between that paper and every other paper should be computed, which is time consuming. Besides, the number of clusters which are to be chosen from the optimum cluster number, must be chosen.</p>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h5>Dimensionality Reduction</h5>\n",
    "<p>At this stage, we have lots of features/words in the TIF-IDF matrix. However, to allow the K-means to perform better, dimensionality reduction is needed. Therefore, the PCA algorithm is chosen, as it is known that it performs well with K-Means </p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.decomposition import PCA #import PCA which will be used to reduce the dimensions of the dataset\n",
    "\n",
    "pca = PCA(n_components=0.90,random_state=30)#keep 90% of the variance, and start random centroid @ 30\n",
    "tf_idf_norm_kmeans= pca.fit_transform(tfidf.toarray())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h5>Elbow testing </h5>\n",
    "<p>In order to determine how many clusters are needed for the K-means. A graph is plotted, the point at which the elbow forms is then searched. That value should be used as the number of clusters. </p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████████████████████████████████████████| 12/12 [00:16<00:00,  1.36s/it]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 1440x1440 with 0 Axes>"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYgAAAEaCAYAAAAL7cBuAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAgAElEQVR4nO3dd5gURfrA8W8Nu4sgosCiCIiCior8TCiCCgJm5VTUe+XOcIbT09Mz3pkzp6ee8c5wp2KOr4CKooiimDHrmSOYMJAUySxTvz+qF4ZhZrd3d8Lu7Pt5nn12p7un37dne6amq6qrnPceY4wxJl2i2AkYY4xpnKyAMMYYk5EVEMYYYzKyAsIYY0xGVkAYY4zJyAoIY4wxGVkB0cw45/7POfeac26hc25qsfMpBufcYc65qhzu73bn3NO52p/JDefcIOecd851LXYuTZUVEPXknGvlnBvhnPvMObfAOTfTOfe6c+6EYudWi8uBOcDGwDbZNnLOHe6ce9M5N8c596tz7iPn3M0Fy7LInHOTog8X75xb4pz7yTn3nHPuFOdcq7TNTwR+W4d9VznnDstpwkXinDs5Ov/bZ1n/qHPuxejvhHPur865951z85xzPzvn3nXO/b2esQ92zj3vnPsl2t/7zrnLnXNdGnJMtcQ8pzl9sbICov5uBA4F/gb0AoYA1wNrFDOpGDYEnvPeT/XeT8+0QfTh9R/gNmBroA9wBtCiUEnWxAXlBQh1L7A2sB6wCzAKOBV4yzm3ZvVG3vtfvPezC5BPY3RH9PuQ9BXRB/UewE3RovOAs4FLgc2A7YFLgFXrGtQ5NxIYCTwfxegFnAB0IvyPGj3nXEWxc6iV995+6vED/AwcX8s2twNPpy07OLzsyx5fAHwOCPAZMB94GGgL7Ad8AvxK+HBavZZ4awP3R7ktACYBW0fr1gN82s8FWfbzMDAqxmsgUe4LgZeBvaP97hCtHxQ97pr2vCrgsJTHFwMfRcf+DaFwWj1l/WHRcwYDbwOLgaHRul2Al6Lj/Y5QqHVIea4DRgA/AXOj1+dkoKqWY5sE3JJheVdgNnBrtv8zsCnwZPR/mBcd2yHRuqnp/4doeTvgbuDr6Fg+IXzQufQ4wNHAV4QrwUeAjmk57gy8EL2evwDPAeunrB8OvBP936YCVwGrpqzfIXpNf41+3gV2q+G1ugt4P8Pyc6PXqlX0+B3gihy89/aPXrvhWda3y3T+1eF8PAv4ElgETI/+l62i8zDjewgoI7yXp0Sv6wfAn9LieEIhdm/0f3mwpngNfZ1y8VP0BJrqT/SmfwxoX8M2K3xwRMsyFRDzgHGEb1U7RifJBOBxYHNgAPAjcFkNsRzwavQm3AH4P+CB6A1aSfj234nwAXxp9HebLPu6Mfrg6FlDvC2BJPAPYCNCYTaF+hUQ50THuB6wE/AxcEfK+sOiWK8TrtR6AB2jv+cDfyFcGW0DPEv4Vumi554Yvb5/AHoCpxE+uOtVQETrrove4IlM/2fgf9GHQK8o1z1YXqB1jI7/xOh/0Cla3gk4HdgK6B6dJ3OBw9POp1+A+4DewHaEgiL1tdoZWApcE507GwNHAhunvJazCd/4ewADo3zvita3AGYRCo0No59hwIAaXqsdov9z/5Rliegc+lfKsiei/2GXBr73HgY+i7HdCudfnPORcB7PAX4DdAO2AE4iFBCtCO+db6r/d0Tvoeh/8z9g1+j/d2B0nh2ZEscDMwnn6/rR+Zg1XqE+y2p8DYudQFP9IVwefxW9Gf9HuIzehwzf+NKel6mAqAIqU5ZdH+23Y8qya4E3ashnp+gE7JWyrCXwPXBeyrKpwDm1HFsn4MVof1MJBc3RQOuUbe4GXk573vHUo4DIEH8Y4dtU9QfwYdF+BqRtNwm4NG1Zt2jbLaLH3wIXp20zioYVEMdEMdbM9H8mfIjXdHw1Hn/a//yptPNpOtAyZdkZwPcpj18AHqthn1OBY9KWDYyOp13044FBdXw/fMCKV1W7RfvpnbJsY+B9QmH/CaF66iCgrI6xPgTGxthuhfMvzvlIuLr8FCjPss9zgKlpy7pHx7Rx2vLzgHdSHntgZNo2NcYr9o+1QdST9/4lwreAAYQTfS1gNDDWOefquLvvvPczUh7/APzgV2wj+AFYk+w2BWZ67z9MyXER4api07ok473/wXu/A+Eb8D8I38AvBz5IqXvvRaiGSPViXeJUc87tFzU2TnPOzQXuASoIBVWq19MebwOc5JybW/1D+PAA2NA51xboQqj+anCeqSlHv32W9VcAt0QN3Rc457aqdYehAfcM59w7zrkZ0bEcA6ybtulH0f+12neEc69aH8LVZ6YYHaP9XZX2mj0RbbKBD20ptwBPOueeiHLaqLb8CV+QDoxec4CjgFe89+9Xb+C9/5hwZduHcBVWEcWanKHhvyaO7K99QylQDnwV9U47xDm3Wi3P2TrK6Y201/UswhVYqtdyEK9grIBoAO99lff+Ze/9ld77fQjfdIcSvpFB+FaRXlhkalxdkr7rLMtq+39letPU+83kvf/Ie/9f7/0RhCqlrsCxddhvMmXb8IdzLUg5DufctsCDhGqhYYQqlmOi1amNeEu99wvT9p8ALiNclqf+bEj40Kvtg7y+ehOqD2ZmWum9H0GoPtBo28kxeuqcCpwJ/JvQrrIF4cMzvSFzcXo4Vj7Hsh1v9et+Iiu+XpsTXrP3ovyPInyIP0Wo8nzfOfenWvK/M8rjIOfcWoT2qJvSN/LB2977f3vvf0c41j6E9qy4PiF8QamrWs9H7/13hCudIwjtVucCnzjn1qlhv9XP344VX9fehGrjVPNSH9QzXsFYAZFbH0W/q79l/wR0Ttum1m+T9fQBUOmcW/bGcc61BPpG6xpqKqG+v/rYPiBUs6VKf/xT9Dv1NdiCFT/QdgBmeO/P8d6/6r3/lFAQxfEGsKn3/vMMP3O9978QvmHXlmdsUZ/6gwiN+Mls23nvv/Te3+C9P4BQ1XBsyurFrNwjbCAw3ns/MvoA/ZyVv33G8SaheidTTj8S6s83yvKaLUzZ9n3v/VXe+z0IvYWOrilodOXxIOHK4TDCuaIx8k1/z8RxN7CBc254ppXOuXZZnhfnfMR7v8h7P957fxrhiqc1sG+0OtP/7s3od7cMr+kXtR1MLfGKqqzYCTRVzrnnCI2FbxDqhTcgdNn7mdBQCqHHyenOueMJ32iHULdvSnXxDOHy9V7n3HGEevBzgVUIjc6xOeduJFRpPUPoVVNJ+NbZltBACHA18Lpz7mJCFdumrNy98HNCO80FzrmTo/1cworfcD8BOjrnjiS8bjsAf46Z6nnABOfc1VEOvxI+VH9L6GG2ALgSGOGc+xiYTPhmu3PM/bdyznUifJGqJHyIn0kodM7M9ATnXBvCVc1oQqP9GsDuLK/6Ilo+2Dn3BLA4ql78BDjEOTc42v+hwLaEBuW6GAE84Zy7BriV0JbTn1Dd8wmhm+lI59zPhP/lEmATYA/v/Z+ccxsQPuQfJRQmnQnVqG/FiP1fQrXjesDd3vv5qSudc6MJ1X0vA9MI1X/nRDmMi7bpS7gaOdR7n14dA4D3fpRz7k7gDufcpoTOHN8R2gIOI7xmp2R4aq3nY3QeJgjvpZ8JbXursfz/NwXo5JzrT9Tr0Hv/uXPuVuBm59xpwCuErrt9CO2Il2V7wWLEK65iN4I01R9C4+ALhG8lCwkfpHeT0kgcbXc24eSdSyhQjiNDN9e052RqCDsD+LaWnNK7uT5H1M01ZZup1N5IvR8wltDAu4jQg+opwodI6nbDgS+ibV4lNNIva6SOttmW8A1rAaG75ABW7sU0Iooxj/Bm/120n/Wi9YeRpVE52t/ThMKhukvpNUQNn4Q33yXAjGj9KOJ3c/Usr+6bEb2ep5DWw4SURmpCgXwvy7s7/kRo5F8nZfvdozwXVZ8LwOqEb9xzCFVX10evy9RMcVKWrdDpIVq2G+FDagHhi8KzQI+U9ftG6+dH8d4h6sgQnUNjUv7304CbqaWLdcq+34tes80yrDsqOo++j/b9HaGQSu39NIiYjeSEnmkvRscwj9AAfimwdtq+uqY8p8bzkXDuv0woZOZH+0ztiVQe/X9nsWI31xaEHnIfE64yqs+X36Y81wMHZ3ivZY1X7J/qroDGNJhzbj3CB+MA731DG4KNMUVmbRDGGGMysgLCGGNMRlbFZIwxJiO7gjDGGJNRKXVztUshY4ypn4yjP5RSAcG0adPq/dzKykpmzJhR+4Y5UMhYpR6vlI+t0PFK+dgKHa8pHVvnzun38i5nVUzGGGMysgLCGGNMRlZAGGOMycgKCGOMMRlZAWGMMSajkurFVB/JyZPwD93Fj7NnQLtK3LBDSPQbVOy0jDGm6Jp1AZGcPAl/1/WwOJqga9Z0/F3XkwQrJIwxzV6zrmLyD921vHCotnhRWG6MMc1csy4gmJXlxpJsy40xphlp3gVE+8q6LTfGmGakWRcQbtghUNFyxYUVLcNyY4xp5pp1I3Wi3yCSRG0Rs6ZDIgEHHWsN1MYYQzMvICDqrdRvEG0+/4BfLjsTV9bsXxJjjAGaeRVTqpZ9B0CnLvjxo7FJlIwxxgqIZVwigdt1GHwzBT56p9jpGGNM0VkBkcL1Gwyrtyc5fkyxUzHGmKKzAiKFKy/H7fwb+Ohd/FefFzsdY4wpKisg0riBu0Or1ni7ijDGNHNWQKRxrVfFDdwd/+bL+J++L3Y6xhhTNFZAZOB23htaJPATHip2KsYYUzQF6/QvIicCRwEOuFlVrxGRB4CNok3WAH5W1S1EZD3gI+CTaN1kVT2mULm6Ndrj+g/BvzQRv/fvcG3bFSq0McY0GgUpIESkN6Fw6AssBsaLyDhVPTBlmyuBX1Ke9oWqblGI/DJxu+6Lf/Ep/MRxuGEHFysNY4wpmkJVMW1CuAqYr6pVwHPAsOqVIuIAAe4rUD61cp26whbb4ieNwy+cX+x0jDGm4ApVxfQ+cLGIdAAWAHsCb6SsHwD8qKqfpSzrLiJvA3OAc1T1hfSdisjRwNEAqkplZf1HYS0rK1vp+UuGH8ms04+i9Vsvs+rew+u97zix8qmU45XysRU6XikfW6HjlcqxFaSAUNWPROQy4ClgLvAuUJWyye9Y8erhe6Cbqs4UkT7AwyKyqarOSdvvTcBN0UM/Y0b953GorKxkpee3Xwt6bsrch+9lft8dcWXl9d5/rbHyqJTjlfKxFTpeKR9boeM1pWPr3Llz1nUF68WkqiNVdStVHQjMAj4DEJEyYD/ggZRtF6nqzOjvN4EvgJ6FyjVVYvf9YfYM/GsrXcAYY0xJK1gBISJrRr+7EQqE6iuGnYGPVfXblG07ikiL6O8ewIbAl4XKdQW9+0CXdfFPjsEnk0VJwRhjiqGQ90GMFpEPgUeB41R1drR8OCs3Tg8E/ici7wKjgGNUdVbhUl3OOYfbbT+Y9jW892YxUjDGmKIo2H0Qqjogy/LDMiwbDYzOd05xuW0G4B++m+STo2mx+TbFTscYYwrC7qSOwZWV4XbZBz77EP/Fx8VOxxhjCiLWFYSIDAamquoUEVkbuBRYCpylqj/kM8HGwu2wC/7R+0mOH02L484udjrGGJN3ca8gbiAUCABXAuWAZ3kX05LnVmmFG7IXvPMq/vtva3+CMcY0cXELiC6q+nXUJXU3ws1pxwLb5S2zRsgN3gvKK/BP2lDgxpjSF7eAmCMiawE7Ah+q6txoeW7uHGsiXNs1cNvvjJ88CT97ZrHTMcaYvIpbQPwbeB24B7g+WrY90OxabN2u+0IyiZ84ttipGGNMXsUqIFT1MsINbdur6v3R4u+AP+YrscbKdeyE23p7/HPj8fPn1v4EY4xpourSzXUK0FlEqofo/o5i3d1cZG73/WDhAvxz44udijHG5E2sAkJE/g/4FLgZGBkt3hG4NU95NWqu2/rQawv8xEfxSxYXOx1jjMmLuFcQNwLnqerGwJJo2XPADnnJqglI7LYf/DIb/8qzxU7FGGPyIm4BsSlwd/S3B1DVeUCrfCTVJGyyOXRbHz/hYXxyae3bG2NMExO3gJgK9EldICJ9gc9znVBT4ZzD7b4//PgdvPNqsdMxxpici1tAnAuME5ELgQoRORN4EDgnb5k1AW6r/tCxE8nxY/DeFzsdY4zJqbjdXB8D9gA6Etoe1gX2U9UJecyt0XMtWoT7IqZ8Cp9+UOx0jDEmp2IP962qbwF/zmMuTZLbbif82PvCIH4b9S52OsYYkzNxu7mOEZEBacsGiMio/KTVdLiKlrghQ+H9N/HfTil2OsYYkzNx2yB2BF5OW/YKMDi36TRNbvCe0HIV/JMPFTsVY4zJmbgFxEJg1bRlbVh+T0Sz5lZdDTdgN/xrz+Nn/lTsdIwxJifiFhBPAv8VkbYA0e/rABtrIuJ22Rucwz/1SLFTMcaYnIhbQJwKtAVmichPwCxgdeCkfCXW1Lj2HXF9B+JfmICfO6fY6RhjTIPF6sWkqrOBvUSkE7AO8E1dpxoVkROBowAH3Kyq14jIBdGy6dFmZ6nq49H2ZwJHEmayO0FVn6xLvGJwu+2Hf+VZ/LOP434zvNjpGGNMg8Tu5hpJAjOB1iLSA0BVax3RVUR6EwqCvsBiYLyIjItWX62qV6Rt3wsYThjiozPwtIj0VNVGPaaF67Iu/N/W+Gcew+86DNeyZbFTMsaYeovbzXV3EfkO+IEwvEb1z2cx42wCTFbV+apaRbjZblgN2+8D3K+qi1R1ShSrb8xYRZXYfX+YOwf/8tPFTsUYYxok7hXE9cAI4A5VXVCPOO8DF4tIB2ABsCfwBuFq5HgROTR6fGpUndUFmJzy/G+jZSsQkaMJ82OjqlRWVtYjtaCsrKxBz6/mOwxkds9NWfr0WDrsdxCuxcovca5ixVXK8Ur52Aodr5SPrdDxSuXY4hYQ7YD/qmq9BhxS1Y9E5DLgKWAu8C5QRRhGfARhhNgRwJXAEYR2inQrxVbVm4CbqtfPmDGjPukBUFlZSUOen8rvvA/JGy5h+pNjSfQdmNdYcZRyvFI+tkLHK+VjK3S8pnRsnTt3zroubgExEjicBkwQpKojo/0gIpcA36rqj9XrReRm4LHo4beExvBqXYFp9Y1dcJv3hU5d8ONH47cZgHOZyjtjjGnc4hYQ/YATROQMQjvEMqq68lfkDERkTVX9SUS6AfsB/UVkbVX9PtpkGKEqCmAscK+IXEVopN4QeC1mrkXnEgncrsPwd14HH74Dm25Z7JSMMabO4hYQt0Q/DTE6aoNYAhynqrNF5C4R2YJQfTQV+BOAqn4gIgp8SKiKOq6x92BK5/oNxo+9l+STY2hhBYQxpgmKex/EHQ0NpKoDMiw7pIbtLwYubmjcYnHl5bid98aPuh3/1ee4dTcodkrGGFMnse+DEJG1CF1NK0lpRFbVerdLlDo3YDf8OMU/MRp3zOnFTscYY+okVgEhIvsS5qT+jHDz2gdAb+BFGtBwXepc61VxO+6Bf/Ih/E/TcGtm7y1gjDGNTdyxmP4OHK6qWwLzot9HA2/mLbMS4Xb6DbRI4Cc8XOxUjDGmTuIWEN1U9cG0ZXcAh+Y4n5Lj1miP6z8E/9JE/JzZxU7HGGNii1tA/BS1QQBMFZH+wPpAi/ykVVrcrsNgaRV+4rjaNzbGmEYibgFxM7BD9PfVwLOEu6FvzEdSpcZ16gJb9sNPGodfOL/Y6RhjTCyxCghVvUxVR0d/3wn0BPqo6jn5TK6UJHbbD+bPwz8/odipGGNMLHFHc11hmjRV/ToaX2lMftIqPa7HRtCpK3707fy43/YsPf1IkpMnFTstY4zJKm4V0+AsywflKI+Sl5w8CWb8CMkkeA+zpuPvut4KCWNMo1XjfRAiclH0Z0XK39V6AF/lJasS5B+6C6qWrLhw8aKwvN+gouRkjDE1qe1GueoRVROsOLqqB74BLshDTqVpVpaheLMtN8aYIquxgFDVwwFE5GVVvbkwKZWo9pUwa3qG5R0Kn4sxxsQQdyyml0RkLVX9UUTaAH8DlgJXqKr124zBDTsEf9f1sHjRiivarI6vqsKV1XV6cGOMya+4jdT3AmtEf18BDAT6A//NR1KlKNFvEO6Q46B9R3Au/O4/GL7+An/bNfhkkxrN3BjTDMT92rqeqn4iIo4wsc+mhLmlp+QtsxKU6DcI+g1aYXrA5Nrd8GPugIqWcMhxuETcMtsYY/Ir7qfRIhFZjTDc9zeqOgNYBKySt8yaicQe++OGHoh/8Sn8A7fgfb2m/TbGmJyLewVxL/AMsBpwXbRsK+wKIifc3r+HRQvxTz0CLVvCsENtHmtjTNHFnVHuZBHZFViiqs9Gi5PAyXnLrBlxzsFvjwj3RTwxGipWwQ09sNhpGWOaudhdZ1R1QtrjN3KfTvPlnIPfHxMKiUfuIdlyFRK77FPstIwxzVjWAkJExqvq7tHfLxBujluJqg7MU27Njksk4A8nwOLFeB1JsqIliR13L3ZaxphmqqYriDtT/r6loYFE5ETgKMJ81jer6jUi8k/gN8Bi4AvCrHU/i8h6wEfAJ9HTJ6vqMQ3NoSlwLVrAH0/BL16Ev+fGUEj0zzYUljHG5E/WAkJV7035+46GBBGR3oTCoS+hMBgvIuOAp4AzVbVKRC4DzgROj572hapu0ZC4TZUrKydx7Bkk/z0Cf9u1+IoKXJ/ti52WMaaZqbUNQkRaAYcDA4D2wCzgeeB2VV0QM84mhKuA+dE+nwOGqerlKdtMBg6oQ+4lzZVXkDjubJLXnE/y5itIlFfgNtum2GkZY5oRV1O/exFpC7wEdCB8258GdAF2BmYC26vqnNqCiMgmwCOEu68XABOBN1T1LynbPAo8oKp3R1VMHwCfAnOAc1T1hQz7PRo4GkBV+yxevDjGIWdWVlZGVVVVvZ+fr1jJeXOZfcEJVH31JWuccwUtN9s6r/FyobG+lhav8cQq9XhN6dgqKiogVP2vvN9annsGMB3or6pzqxdG4zE9FK0/q7YEosmFLiMUMnMJ05UuOxoROTt6fE+06Hugm6rOFJE+wMMisml6YaSqNwE3RQ999d3J9ZF6d3O+1TWWP+4cuOJsfr74byROvhC3Qa+8xmuoxvxaWrzGEavU4zWlY+vcuXPWdbXdST0U+Ftq4QAQPT6D0MAci6qOVNWtol5Ps4DPAETkD1Gcg1TVR9suUtWZ0d9vEhqwe8aNVWpcm7YkTrkI2lWS/NdF+KmfFTslY0wzUFsBsS7wXpZ17wHrxQ0kImtGv7sB+wH3icjuhEbpvVNHhRWRjiLSIvq7B7Ah8GXcWKXItW1H4pQR0LoNyWsuwH87tdgpGWNKXK1jMalqxor9aHmyDrFGi8iHwKPAcao6mzBsx2rAUyLyjoj8J9p2IPA/EXkXGAUco6qz6hCrJLn2lSRO/TuUV5C8+jz8D98VOyVjTAmrrQ1ilQxTjaZqGTeQqg7IsGyDLNuOBkbH3Xdz4jp2InHKCJL/PJPkVeeSOO0fuMq1ip2WMaYE1XYFcS9hqtFsP/flNTuTkVu7a2iTWLSQ5JXn4GfPLHZKxpgSFGvKUdP4uK7dSZx0IcmrzglXEn+7BNd2jdqfaIwxMdnsNE2Y674hib+cB7N+Cm0S834tdkrGmBJiBUQT53puSuK4s+GHb0PvpgU2RbgxJjesgCgBrteWJI45A775kuS/L8IvWljslIwxJcAKiBLhNu+LO/JU+Pxjkjdcgl9S/2FHjDEGap4PYkicHajqM7lLxzREYpsdSC5ZhL/tWpL/vZzEMWfgymLPCWWMMSuo6dNjZNrjLoRJg2YSBu9zwLdAj/ykZuojsd1OJBctwt/7H5KXnwG/zObH2TOgXSVu2CEk+g0qdorGmCaipvkgulf/LSJnEQqFc1V1voi0Bi4iFBamkUkM3pOln38Irz2/fOGs6fi7ricJVkgYY2KJ2wZxMnBG9XhJ0e8zgVPylZhpoM8/WnnZ4kX4h+4qfC7GmCYpbgExjzAbXKptAOtT2VjNyjL0b7blxhiTJm4L5rmEaUIfBb4hDLMxFDguX4mZBmpfCbOmr7y85Sr4RYtwLWMPo2WMaaZiXUGo6l3AtsBHQFvgY6BftNw0Qm7YIVCRVggkWsCiBST/fhJ+yqfFScwY02TE7gOpqh+KyMfAWqr6fR5zMjmQ6DeIJIQ2h5ReTG71diRvv5bkpafh9hTcXmJdYY0xGcX6ZBCRNYAbgAOAJcCqIrI30FdVz8ljfqYBEv0GQb9BK01HmDj/3/j7b8I/dj/+f6+TOPJkXOduxUvUGNMoxW2k/g/wC2GGuepbdF8BDsxHUia/XOtVSRxxMoljz4BZ00mOOJnkU4/gk3WZ/8kYU+riFhA7ASdEVUvV80ZPB9bMV2Im/9xW25G48N+w6ZZ4HUnyqnPxM38qdlrGmEYibgHxC1CZuiCaW9raIpo417YdiePOxh12Anz1OckL/kLypYl474udmjGmyOIWELcQ5pQeDCREpD9wB6HqyTRxzjkS2+9M4vx/Qbf18bdfGwb8m/NzsVMzxhRR3O4rlwELgeuBcuBW4L/AtXEDiciJwFGEMZxuVtVrRKQ98ACwHjAVEFWdLSIu2veehJvxDlPVt+LGMvXjKtcicerf8U+PxT90F8kL/kLikONwW/YrdmrGmCKotYAQkRbAH4AbVfWa+gQRkd6EwqEvoZF7vIiMi5ZNVNVLReQM4AzgdGAPYMPoZ1vgxui3yTOXSOB23Re/6VYkb72K5A2X4PoPwQ0/Ctd61WKnZ4wpoFqrmFR1KXCVqi5qQJxNgMmqOl9Vq4DngGHAPoSqKqLf+0Z/7wPcqapeVScDa4jI2g2Ib+rIdelG4sx/4oYeiH91EskLT8B//L9ip2WMKaC4VUyPishvVPXResZ5H7hYRDoACwhVR2+QctOdqn4vItW9oroQhvSo9m20bIVGcRE5Gjg6ej6VlSu0o9dJWVlZg57fWGM1ON6RJ7JkwM78cu0Ill55Dq2HCm0OPrbGoTrstWya8Ur52Aodr1SOLW4BsQowSkReIXxwL+vioqqH1vZkVf1IRC4DngLmAu8CVTU8xWVYtlK3GlW9Cbipen3qzWB1lX4zWT4VMlZO4rVfC3/WlbgxdzD/MWX+Gy+Hm/mK7KIAACAASURBVOvW2zA/8eqgyb2WjTheKR9boeM1pWPr3Llz1nVxezG9D1wCPAt8DnyR8hOLqo5U1a1UdSAwC/gM+LG66ij6Xd0J/1vCgIDVugLT4sYyuedatiTxu6NJnHwRLFpI8h9/Izn2PnxVTeW8MaYpi3UFoaoXNjSQiKypqj9F90/sB/QHuhMawC+Nfj8SbT4WOF5E7ic0Tv9i4z81Dq7XFiQu+Bf+vpvwj94XDdVxCm7trsVOzRiTY7FHaRORCmAjwg1zy6qA6jAn9eioDWIJcFzUnfVSQEXkSOBr4LfRto8T2ik+J3RzPTxunib/XOs2uCNPwW+xLcm7byA54iTcfofiV10NHr7bpjg1pkTEHaxvB+BBoCVhuO85wGqE9ohYc1Kr6oAMy2YShvFIX+6xuSYaPddnexIb9CJ553X4B24B56D6Dmyb4tSYJi9uG8TVwOWq2h74Nfo9gjDCq2nG3OrtSBx/Dqy62vLCoZpNcWpMkxa3gOjJyndNX0qYq9o0c845mDc380qb4tSYJqsug/W1jf7+XkR6Ae2ANnnJyjQ97bP0wS4vx8/4sbC5GGNyIm4BMYbQaAwwktDd9U1Cu4Qxmac4bdECkktJnvtnko/cg1+0sDjJGWPqJW4315NS/r5SRF4jXD08ma/ETNOSdYrTnr3xo+/AP/YA/qWJuAMOw20zIFRLGWMatXpNRqyqL+Q6EdP0ZZvi1B11Kn7QHiTvvwl/8xX4SY+TGH40rlusDnDGmCKJ2831BTIMdQEQ3RltTI3chr1InH0l/sWnw1Difz8FN2BX3L4H41ZrW/sOjDEFF/cK4pa0x52AI4G7c5uOKWUu0QI3cDd8n+3DXdjPjsO/8QJu74Nwg/bAtWhR7BSNMSnitkHckb5MREYDtwEX5TopU9rcqm1ww4/CD9yN5P034++/Cf/8eBLDj8Jtsnmx0zPGROL2YsrkO2CzXCVimh/XuRuJky8i8eezYPEikledy9Ib/4Gf/kOxUzPGEL8N4oi0Ra0JA+5NznlGpllxzsGW/Uj03go/4WH84w+S/N8buN2G4fY4ANdylWKnaEyzFbcN4pC0x/OAlwlDcBjTYK68AreX4PsPCd1ixyn+5Wdw+/8B13egdYs1pgjitkEMzncixgC49pWhW+zgPUjedzP+livxk54g8bujcN3WL3Z6xjQrcauY4o7Y+mXD0jEmcBv0InH2FVm6xa5e7PSMaRbiVjF9zvL7IBwr3hNRfe3vAeunaHJmWbfYrbfHP3o//pnH8G+8iNv79/hWq8Ij99jcE8bkUdwC4khgZ+AC4CtgXeA8YKKq3p6XzIyJuNZtcAf+ET9gV5IP3IK//2ZW+J5ic08YkxdxC4gRwIaquiB6/JmI/An4FLg9H4kZk8517kbipAtJnnIIzJ2z4srquSesgDAmZ+LeB5EA1ktbti5WpWQKzDkHc3/NvHLW9MImY0yJi3sFcTXwjIjcRphmdB3gMKybqymG9pVZC4PkqNvD/ROr2lQlxjRUrCsIVf0ncDiwFrA3YSymI1T18jzmZkxGGeeeKK+A9TfBT3iI5FlHkRw/Gr94UXESNKZExB7uW1XHA+PrG0hETgb+SGhZfI9Q4DwFrBZtsibwmqruKyKDgEeAKdG6MapqYz4ZIPvcE4l+g/DfTCE55s5ws93Ex3B7/w633U42EKAx9VBjASEiuwNzVPXl6PH6wJ1Ab+AV4HBV/b62ICLSBTgB6KWqC0REgeGqOiBlm9GEQqHaC6o6tK4HZJqHrHNPrNOdFieej//kfZKjb8ffeR1+wsMkhh0CW/azO7KNqYPaqphGsOI9D7cS5qf+PWG4jSvqEKsMaCUiZYSxnKZVrxCR1YAhwMN12J8xWbmNepM4858kjj0TgOSN/yB56Wn4T94vcmbGNB3O+4zzAAEgIrOANVW1SkTWJHyor6uq34lIJfA/Ve0cJ5CInAhcDCwAJqjqQSnrDgX2VtUDoseDgNHAt1HMv6rqBxn2eTRwNICq9lm8eHGcVDIqKyujqqqq3s9vrLFKPV6cWH5pFQuffYK5999CcuZ0KrbqT5uDj6G8+4Z5iZdLje21tHiNL1ZD41VUVMDyG55X3G8d9tMfmKKq30WPZxLmpa6ViLQD9gG6Az8DD4rIwapaPeHQ71hxUqK3CAXRXBHZk3BlsdK7WVVvAm6KHvrUqoa6Sq+qyKdCxir1eLFjbdEfem2Fe+YxFj8xilmnHobbdkfc3r/HdeyU+3g50ihfS4vXqGI1NF7nztm/49dWxfQ6cIKItCU0MD+Rsq4HEDejnQmFy3RVXQKMAbYDEJEOQF9gXPXGqjpHVedGfz8OlEdXLMbUm6toSWL3/UlccjNut/3wb75M8tw/h0mLfv2l2OkZ0+jUdgVxMvAo8E/CeEx/Sll3CPB8zDhfA/1EpDWhimkn4I1o3W+Bx1R1YfXGItIJ+FFVvYj0JRRkM2PGMqZGbtU2uP3/gB8ydPnUpy8+Heag2GUf3Cqtip2iMY1CjQWEqn4IrC8iHVQ1/QP6GiBWpb+qvioiowhVR1XA2yyvGhoOXJr2lAOAY0WkilCgDFfV7I0lxtSDa9cBd+jx+F32Jfnw3fix9+KfHYcbeiBu4G64svJip2hMUdXYSN3E+GnTptW+VRZWP9k04+Uylv/yE5Jj7oRP3oOOnXD7HITbZgAusbwm1l5Li9fYYjU0XtQG0eBGamNKmuuxEYlT/w4fvEVy9J1hsqIJD5EYdijJuXPgobtseHHTrFgBYUwK5xz07kOi15b411/AP3w3yWsvAOfA2/DipnnJ2otJRI5P+XuDwqRjTOPgEgkS2+5IYsQN0LrN8sKhWvXw4saUsJq6uV6c8vdb+U7EmMbIlZXD/HmZV86ajl+4IPM6Y0pATVVMX4jIlcAHhPsQjsi0karempfMjGksahpe/LQjcNvvjBu8J27NtQucmDH5VVMBMRw4jXCXcznhvod0njA+kzElyw07BH/X9ZA6fHhFS9htGO7HafhnH8NPHAubbUNiyFDYZHMbFNCUhKwFhKp+Srh7GhGZqKo7FSwrYxqRmoYXB/C/PRz/3Hj8c+NJvvsarL0ObshQXP/BuJarFDV3YxoiVi8mVd0pGoV1O6ALYRC9V1S1cKNRGVNE2YYXB3BrdMDtcxB+Twk9n555DH/PjfiH7sTtsAtu0J51Gu/JmMYiVgEhIhsBjwGtWD7l6EIR+Y2qfpTH/IxpMlx5OW67Ifj+g+HLT/ATHw0/Tz0Sqp92+g1svJlVP5kmI+59EDcShsa4onrICxH5K3ADMDhPuRnTJDnnYP2NcetvjJ89E//cE/jnnwzVT527heqnfoOs+sk0erHmpAa2AK5KGw/pmmi5MSYL164DiX0PJnHZSNzhJ0JZOf7uG0iedjjJB2/Dz/ix2Ckak1XcK4hpwI7AMynLBpAyK5wxJjtXXoHbbid8/yHwxcehneLpR0L10+ZR7yerfjKNTNwC4ixgrIg8BnwFrAvsBRycr8SMKUXOOdhgE9wGm+BnRb2fnh9P8p1XQ/XTTkNx2w7Gv/0K3sZ+MkUWq4pJVccCWwHvA6tFv/uo6iN5zM2YkubaV5IYdjCJy2/FHXYilJXh77qB5MkH42+7Ntyc5/3ysZ8mTyp2yqaZiT1YX3RfxN/zmIsxzZIrr8BtvxN+uyHw+UckrzkfkktX3Kh67Ce7ijAFFLeR2hiTZ8453Ia9YHGWebhmTcd/+DY+vfAwJk9suG9jGptsYz85R/Lq82GN9rhtd8T1G4zrul7B0zPNhxUQxjQyWcd++v2fSKzSiuQrz+KfHot/8iHo2h3XfxCu7464NdoXL2lTkqyAMKaRqW3spxZ9tsf/+ksY1mPyJPyDt+FH3QG9Ng9XFVv2s5vwTE7EHWqjG3A+sCXQJnWdqvbMQ17GNGs1jf0E4FZbHTdkKAwZiv/hW/wrk/CvTsKPvArfshVuq/64/oNho964RIvCH4ApCXGvIB4EPgbOA+o1Q4qInEwYHdYD7wGHA/8h3ID3S7TZYar6jog44FpgT2B+tNwmLTImA9epK27Ywfh9fg+ffxiuKt54Ef/KM7BGh9Be0X8wrsu6xU7VNDFxC4iNgf6qmqxPEBHpApwA9FLVBSKihPkmAP6mqqPSnrIHsGH0sy1hLKht6xPbmObCJRLQszeuZ2/88KPw776On/xsuGP7yTHQrUeoguo7ELd6u2Kna5qAuAXEo4Rv+s82MFYrEVkCtKbmYTr2Ae6Mxn6aLCJriMjaqvp9A+Ib02y4ipa4bXaAbXYI7RWvvYB/5Rm8jsSPug16bREKiy362V3bJivn0ydjz0BE2gMvA18AK4wupqoZpyLNsI8TCfNcLwAmqOpBInI70B9YBEwEzlDVRdGQHpeq6ovRcycCp6vqG2n7PBo4Osqjz+Js/cdjKCsro6qqMNNbFDJWqccr5WPLR7yqb6eycNJ4Fjz/JMnpP0J5OSxdCsmUyoGWLWl77Bm02nG3nMXNpKm/lo0lVkPjVVRUAGQcBCzuFcRtwFLgI+rRBiEi7QhXBd2Bn4EHReRg4EzgB6CCMJz46cBFWZJdqSRT1Zui5wH4TI15cWVrDMyHQsYq9XilfGx5ibdKG9j9ANh1PxKffUDy3yNgyZIVt1m0iDl33sC8TfvkLm4GTf61bCSxGhqvc+fOWdfFLSCGAJ1V9dd6ZQA7A1NUdTqAiIwBtlPVu6P1i0TkNuCv0eNvCZMSVeuKjRxrTM64RAI2+j9YtCjzBrOm45PJsJ1ptuL+9/8HdGhAnK+BfiLSOuqhtBPwkYisDRAt25cwCCDAWOBQEXEi0g/4xdofjMmD9pVZVyVHnBwaumNUQ5vSFPcK4hlgQvQtP70N4tbanqyqr4rIKOAtoAp4m1A19ISIdCRUKb0DHBM95XFCF9fPCd1cD4+ZpzGmDrLetd1/CHz4NsnrRkCPjUjsezBuk82Ll6gpirgFxA7Ad8Cuacs9UGsBAaCq5xNutks1JMu2HjguZm7GmHqq6a5tX1WFf3ki/rEHSF51Lmy8WSgo1t+42GmbAolVQKiqzTttTInKdte2KyvDDdwN339wmFf78VEkLz0NNtsmFBTrdC9e0qYg4g61kbWtor43zxljmgZXXoHbeR/8DrviJz6Kn/AQyYtOxG0zALf373CduhY7RZMncauYqsjQzTRiA70Y0wy4VVrh9hL84D3xTz6MnzgW/8ZLuO0G44YOx1WuVewUTY7FLSDSryXXBs4g3GFtjGlGXOs2YeynnYbinxiNn/Q4fvJzuIG74vYUG3a8hMRtg/gqbdFXIvIH4HVgZM6zMsY0eq7tGrgDj8Tvsg9+3AP455/Ev/Q0bvBeuN33x7VpW+wUTQM15C6YtkDHXCVijGmaXPtKEoccR+KiG3BbbYef8DDJM48iOfZe/IL5xU7PNEDcRuq7WLENojUwELg78zOMMc2NW3Nt3JGn4Hc/gOTYe/CP3o9/Zhxuj/1xg/bCtWxZ7BRNHcVtg/g87fE84D+q+nSO8zHGNHGuSzdaHHsmfupnJB++Gz/qdvxTj+D2EtwOu+LffMlGj20i4rZBXJjvRIwxpcWttyEtTroQ/+kHJB++C3/vf/GP3AsLF8DSaOTRWdPxd11PEqyQaIRqLCBEpA+wSFXfjx53BK4BegOvAH9V1bl5z9IY02S5npuS+Ns/4IO3SV739+WFQ7XFi8Kd3FZANDq1NVJfA3RKeXwL0JMwjlJv4PI85WWMKSHOOVzvrcLcE5nMmk7ynv+QfPmZMMd20u6/bQxqq2LaBHgBQETWIEwF2ltVPxWRsYRJhP6c3xSNMSWjfSXMmr7y8rJy/ORnYdLjoTdM61VhvZ64Hj1x3XtC941wq1m32UKrrYAoA6qnaesH/KCqnwKo6jdRoWGMMbFkGz3WHXIcru8A+P47/Jcfw5RP8VM+xY97EO+jq4mOnXDdN4LqQmOdHrjy8uIcSDNRWwHxAfBbQIHhwLJeSyLSBfglf6kZY0pNTaPHAtClG65LNxgQBo72CxfAV1/gp3wSCoxP34PXngtXGWVloZDo3hN6bBR+d+yEcytOSJmcPMl6TdVTbQXE6cCjIvIfwpSjO6SsOxB4KV+JGWNKU7bRYzNxq7SCjXrjNuq9bJmfNSO6wogKjRefgmceC4VGm7bQPVxhuO49Sc78CXTk8isW6zVVJzUWEKr6ooh0IzRMf5o25eg44P58JmeMMelc+0poX4nrsx0AfulS+O4r/JRPYcon+C8/xb//ZvaZ8KzXVGy13gcRFQpvZlj+SV4yMsaYOnAtWkC3HrhuPWDH3QHw8+fBV5+HiY4ymVXzlYsJbEZyY0zJca1XDVOkts82XJwneed1+K+/LGheTY0VEMaYkuWGHRLm2E5VXg4b9ca/OonkiJNYeulpoSF7yZLiJNmIxR2LqcFE5GTgj4RB/94DDicMFb41sAR4DfiTqi4RkUHAI8CU6OljVPWiQuVqjCkNNc65PW8u/pWJ+GefwI+8Cq8jcTvsgttxD1wHG6gaClRARF1iTwB6qeoCEanuNnsPcHC02b2EAuTG6PELqjq0EPkZY0pX1jm3V20TplId8hv4+F2Szz6OHz8GP34MbL4NiUF7wiab4xLNt6KlYFcQUaxWIrKEMFz4NFWdUL1SRF4DbHJbY0xBuUQCem1Ji15b4mf+FCY+emECyXdehTU74wbvgeu/E27VNsVOteBc1q5gOSYiJwIXAwuACap6UMq6cuBV4ERVfSGqYhoNfAtMIwwK+EGGfR4NHA2gqn0WL16cvklsZWVlVFVV1b5hDhQyVqnHK+VjK3S8Uj62usbzSxaz8OVnWTB+DEs+fg8qWtJq4K602mM/yntslNNYudCQeBUVFQAu07qCFBAi0o7wgX8g8DPwIDBKVe+O1t8MzFPVk6LHbYGkqs4VkT2Ba1V1w1rC+GnTptU7xzg37eRKIWOVerxSPrZCxyvlY2tIPP/1l2He7VcnweLFsP7GuEF74PrskHWoj6ZybACdO3eGLAVEoSrXdgamqOp0VV0CjAG2AxCR8wlTl55SvbGqzqkeRlxVHwfKRaSyQLkaY8wyrlsPEoceT+Kft+MOPBJ+nYMfeTXJ048gOeZO/Myfip1i3hSqDeJroJ+ItCZUMe0EvCEifwR2A3ZS1WXj+4pIJ+BHVfUi0pdQkM0sUK7GGLMS1zq9UfuJ5Y3am21NYvBeJOf8DA/fXTLjPhWkgFDVV0VkFPAWUAW8TZhTYh7wFfCKiMDy7qwHAMeKSBWhQBmuqoVpLDHGmBqs2Kg9Hf/8+NCo/e5rhJqa6KOqBMZ9KlgvJlU9Hzg/TnxVvQ64Lu9JGWNMA7gOHcMQ5kOHkzztMJj764obLF4Uplpts1oYDqRtu6LkWV+F7OZqjDElyZWXw9wssy8vmEfy2gvD36u3C0OUr9M9jB21To8wRHkjvdfCCghjjMmFbLPltaskceQp+G++gK+n4L/5Ev/RO2EUWoCWrWCd9XDr9IB1uuO6rQ+duzWKyZCsgDDGmBzIOlvefofi0ue0WLIEpn2N/+ZL+GZK6Er78jOwaEFowWjRAjp1XXaV4br1gK7dV7pZL9+TIVkBYYwxOVDrbHkpXHk5rLs+bt31ly3zySTM+GF5gfHNFPyH78Irz7Ksh06HNZcVGH7BPJj0BCyJbhDOQ6O4FRDGGJMjdZktL51LJGDNzmF4jz7bL1vu58yOqqamwDdfhiqqd1+FTDc553gyJCsgjDGmEXNt20HvdrjeWy1b5hcuIPmXAzM/IYeTITXOpnNjjDFZuVVaZZ8MqX3uBp2wAsIYY5qgjJMhVbQMy3PEqpiMMaYJqkujeH1ZAWGMMU1UQxrFY+0/53s0xhhTEqyAMMYYk5EVEMYYYzKyAsIYY0xGVkAYY4zJqCBzUhdIyRyIMcYUWFHnpC4E15AfEXmzoftojLFKPV4pH5u9lk03XhM8toxKqYAwxhiTQ1ZAGGOMycgKiOVuKtFYpR6vlI+t0PFK+dgKHa8kjq2UGqmNMcbkkF1BGGOMycgKCGOMMRk169FcRWQd4E6gE5AEblLVawsQtwXwBvCdqg7NY5w1gFuA3oT7RI5Q1VfyGO9k4I9RrPeAw1V1YQ73fyswFPhJVXtHy9oDDwDrAVMBUdXZeYz3T+A3wGLgC8Ix/pyPWCnr/gr8E+ioqjkZsjNbPBH5C3A8UAWMU9XT8hVPRLYA/gOsEsX7s6q+loNYGd/X+TpXaoiX83Olts+sXJ8rzf0Kogo4VVU3AfoBx4lIrwLEPRH4qABxrgXGq+rGwOb5jCkiXYATgK2jD4AWwPAch7kd2D1t2RnARFXdEJgYPc5nvKeA3qq6GfApcGYeY1V/IOwCfJ2jOFnjichgYB9gM1XdFLgin/GAy4ELVXUL4LzocS5ke1/n61zJFi8f50rWz6x8nCvNuoBQ1e9V9a3o718JH6Bd8hlTRLoCexG+2eczTltgIDASQFUX5+Kbbi3KgFYiUga0Bqblcueq+jwwK23xPsAd0d93APvmM56qTlDVqujhZKBrvmJFrgZOI8cjBWSJdyxwqaouirb5Kc/xPNA2+nt1cnS+1PC+zsu5ki1ePs6VWj6zcn6uNOsCIpWIrAdsCbya51DXEP6JyTzH6QFMB24TkbdF5BYRWTVfwVT1O8I3zq+B74FfVHVCvuKlWEtVv49y+B5YswAxqx0BPJGvnYvI3oRqyHfzFSNNT2CAiLwqIs+JyDZ5jncS8E8R+YZw7uTqamyZtPd13s+VGj5Hcn6upMbK17liBQQgIm2A0cBJqjonj3Gq62DfzFeMFGXAVsCNqrolMI/cVr+sQETaEb6hdQc6A6uKyMH5ildsInI24XL/njztvzVwNqHqpVDKgHaEqou/ASoiWYdhyIFjgZNVdR3gZKKr3Vwp1Pu6tnj5OFdSY0X7zsu50uwLCBEpJ7zQ96jqmDyH2x7YW0SmAvcDQ0Tk7jzF+hb4VlWrv8mMIhQY+bIzMEVVp6vqEmAMsF0e41X7UUTWBoh+56xaJBsR+QOhwfUgVc3XjUTrEwrbd6PzpSvwloh0ylM8COfMGFX1UWNxEqjMY7w/EM4TgAeBvrnacZb3dd7OlWyfI/k4VzLEytu50tx7MTnCt5aPVPWqfMdT1TOJLqNFZBDwV1XNy7dsVf1BRL4RkY1U9RNgJ+DDfMSKfA30i775LojivZHHeNXGEj5oLo1+P5LPYCKyO3A6sKOqzs9XHFV9j5QqkOiNv3WuejFl8TAwBJgkIj2BCiCf8aYBOwKTorif5WKnNbyv83KuZIuXj3MlU6x8nivN+k5qEdkBeIHQJbO6TeAsVX28ALEHEQqIfHZz3YLQGF4BfEnoZpeTLqBZ4l0IHEi45H0b+GN1g2eO9n8fMIjwrfZH4HzCh5oC3QiF1G9VNVNjb67inQm0BGZGm01W1WPyEUtVR6asn0oOC4gsx3YXcCuwBaFr5l9V9Zk8xvuE0NOuDFhI6Oba4OrXbO9rQrtAzs+VGuL9ixyfK3E+s6yAMMYYk3fNvg3CGGNMZlZAGGOMycgKCGOMMRlZAWGMMSYjKyCMMcZk1KzvgzClQ0QuADbI130lpUxEngDuV9U7MqxbD5gClKeMK2SaCSsgTJMhIr8HTgE2Bn4F3gEuVtUXc7T/9cjDh6GIVBD6xR9EGIZkOvAMcJGqTm3gvicBd6tqvQd/VNU9GpKDKV1WxWSaBBE5hTDQ4SXAWoSbnW4gjP/UKESj2GYyCtgb+D1h1NLNgTcJd5sXKydjamUnj2n0RGR14CLCneCp42U9Gv2kbz+I8K26a8qyqYQ7u58Wkb6EwqUnYViQe1T1FOD5aPOfRQRgF1V9RUSOIAxe1wl4DThaVb+K9usJE+ycRHg/dU/LZWfCGP09VfWbaPEvwPVpx3cVsCfh7tjbCHdSLxWRwwiTME0GjgR+Jtxx/ISIXAwMIAxxcg1wu6oenyknEdmOcNdyT8LcBCeq6stR/EnR63VLNJnVZcBhwBzgypX+IabZsCsI0xT0J8w69lCO9nctcK2qtiUMdKbR8oHR7zVUtU1UOOxLqB7aD+hIGObgvrT97QtsC2SabGpn4LWUwiGTOwjDk2xAGL55V0KhUG1bwrAUlYRJdUaKiFPVs6N8jo/yPT5TTtFMauMIQz90IBRG40SkQ4ZcjiIMLrclsDVwQA15mxJnVxCmKegAzMhhu8ASYAMRqYzGq5lcw7Z/Av6hqh8BiMglwFkism71VUS0PtuYPh0I82NkJCJrAXsQCqUFwDwRuRo4GvhvtNlXqnpztP0dhKuftYAfash7WU4icgDwmareFa27T0ROIEyHeXt6SsA11QWaiPyDMIaSaYbsCsI0BTOByhzWpx9JqGr5WERej+bpyGZd4FoR+VlEfibMiuZYcebBmq4OZgJr17L/cuD7lBj/ZcXJbJYVBCmjgrapYZ/pOXUGvkpb/xWZZ0/snPbc9OeZZsSuIExT8AphtM99CQ2+tZlHmPIUgKhevWP1Y1X9DPidiCQIVUejouqWTCNXfkPoKVXTZC81jXj5NHCiiHRV1W+z7H8RUFnPK6RssVOXTyMURKm6AeMzPO97YJ207UwzZQWEafRU9RcROQ+4XkSqgAmEaqKdgcGqelraUz4FVhGRvaJtzyIMuwxANNPdk6o6PfrGDrCU0P00SZiu9dNo+X+AESLyjqp+EDUo76qqD8bM/WkReQp4SESOAd4FWhG6vC5W1VtFZAJwpYicC8wlNHR3VdXnYoT4Mcq3Jo8D/466CSuwP6G95LFMKQMniMhj5HkWQtP4WRWTaRKiyVFOAc4hfJB/Q+ip83CGbX8B/kyYC+M7wgdd6rf33YEPRGQuocF6uKoujKpvLgZeiqp7+qnqQ4RePfeLyBzgfUKbQV0cQPiQfoDQg+l9QgPw09H6QwlzdnwIzCZcJdVULZXqWuAAEZktIv/Kgjya/wAAAGFJREFUtIGqziQ0PJ9KqPI6DRiaZb6Am4EnCQXZWyyf8c00QzYfhDHGmIzsCsIYY0xGVkAYY4zJyAoIY4wxGVkBYYwxJiMrIIwxxmRkBYQxxpiMrIAwxhiTkRUQxhhjMvp/x0M9hPOBIqYAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 1440x1440 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from sklearn.cluster import KMeans #import kmeans which will be used to cluster the data\n",
    "\n",
    "\n",
    "max_number_of_clusters = 25 #clusters can range from 2-max_number_of_clusters\n",
    "max_number_of_clusters = max_number_of_clusters+1 # since the range method is exclusive, a 1 should be added.\n",
    "Ks = range(2, max_number_of_clusters, 2) #Hold the range of clusters k (from 2-max_number_of_clusters)\n",
    "    \n",
    "sum_of_squared_distances = [] #a list which will carry the sum of squared distances at each k\n",
    "for k in tqdm(Ks):\n",
    "    sum_of_squared_distances.append(KMeans(n_clusters=k,random_state=30).fit(tf_idf_norm_kmeans).inertia_)#calculate the sum of squared distance for each k.\n",
    "        \n",
    "  #plot the elbow graph      \n",
    "f, ax = plt.subplots(1, 1)\n",
    "ax.plot(Ks, sum_of_squared_distances, marker='o')\n",
    "ax.set_xlabel('Cluster Centroid')\n",
    "ax.set_xticks(Ks)\n",
    "ax.set_xticklabels(Ks)\n",
    "ax.set_ylabel('Sum of Squared Distances')\n",
    "ax.set_title('Sum of Squared Distances VS. Clusters')\n",
    "plt.figure(figsize=(20,20))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h5>Elbow Graph Analysis </h5>\n",
    "<p>As exhibited in the graph above the elbow forms at k = 15, as the graph starts to level off</p>\n",
    "<h1 style =\"color:red\"> CHANGE K WHEN YOU RUN ALL THE DATASET. The above is for 1934 papers</h1>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h5> K-Means clustering</h5>\n",
    "<p>Now that the number of clusters is known, k-means can be performed as shown in the cell below </p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "clusters = KMeans(n_clusters=15,random_state=30).fit_predict(tf_idf_norm_kmeans) #from the elbow graph we get the n_clusters\n",
    "df =FullPaperDataframe.copy(deep=True)  #create a new df which will hold the papers attribute along with the cluster number\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<p>In the below cell, the cluster to which the paper belongs is added to the dataframe</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "      <th>clusters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>The RNA pseudoknots in foot-and-mouth disease virus are dispensable for genome replication but essential for the production of infectious virus. 2 3</td>\n",
       "      <td>[{'text': 'word count: 194 22 Text word count: 5168 23 24 25 author/funder. All rights reserved. No reuse allowed without permission. Abstract 27 The positive stranded RNA genomes of picornaviruses comprise a single large open reading 28 frame flanked by 5′ and 3′ untranslated regions (UTRs). Foot-and-mouth disease virus (FMDV) 29 has an unusually large 5′ UTR (1.3 kb) containing five structural domains. These include the 30 internal ribosome entry site (IRES), which facilitates initiation of translation, and the cis-acting 31 replication element (cre). Less well characterised structures are a 5′ terminal 360 nucleotide 32 stem-loop, a variable length poly-C-tract of approximately 100-200 nucleotides and a series of 33 two to four tandemly repeated pseudoknots (PKs). We investigated the structures of the PKs 34 by selective 2′ hydroxyl acetylation analysed by primer extension (SHAPE) analysis and 35 determined their contribution to genome replication by mutation and deletion experiments. 36 SHAPE and mutation experiments confirmed the importance of the previously predicted PK 37 structures for their function. Deletion experiments showed that although PKs are not essential 38', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'for replication, they provide genomes with a competitive advantage. However, although 39 replicons and full-length genomes lacking all PKs were replication competent, no infectious 40 virus was rescued from genomes containing less than one PK copy. This is consistent with our 41 earlier report describing the presence of putative packaging signals in the PK region. 42 43 author/funder. All rights reserved. No reuse allowed without permission.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>VP3, and VP0 (which is further processed to VP2 and VP4 during virus assembly) (6). The P2 64 and P3 regions encode the non-structural proteins 2B and 2C and 3A, 3B (1-3) (VPg), 3C pro and 4 structural protein-coding region is replaced by reporter genes, allow the study of genome 68 replication without the requirement for high containment (9, 10) ( figure 1A ).\\nThe FMDV 5′ UTR is the largest known picornavirus UTR, comprising approximately 1300 71 nucleotides and containing several highly structured regions. The first 360 nucleotides at the 5′ 72 end are predicted to fold into a single large stem loop termed the S-fragment, followed by a The PKs were originally predicted in 1987 and consist of two to four tandem repeats of a ~48 86 nucleotide region containing a small stem loop and downstream interaction site (figure 1B) 87 (12). Due to the sequence similarity between the PKs (figure 1C), it is speculated that they 88 were formed by duplication events during viral replication, probably involving recombination. 89 Between two and four PKs are present in different virus isolates but no strain has been 90 identified with less than two PKs, emphasising their potential importance in the viral life cycle 91 (19, 20) . The presence of PKs has been reported in the 5′ UTR of other picornaviruses such as 92 author/funder. All rights reserved. No reuse allowed without permission. can occur in the absence of PKs at least one is required for wild-type (wt) replication. 104 Furthermore, competition experiments showed that extra copies of PKs conferred a replicative 105 advantage to genomes. Although replicons and full-length genomes lacking PKs were 106 replication-competent, no infectious virus was rescued from genomes containing less than one 107 PK copy. This is consistent with our earlier report describing the presence of putative 108 packaging signals in the PK region (22). 109 110 author/funder. All rights reserved. No reuse allowed without permission. Plasmid construction. 117 The FMDV replicon plasmids, pRep-ptGFP, and the replication-defective polymerase mutant 118 control, 3D-GNN, have already been described (10).\\nTo introduce mutations into the PK region, the pRep-ptGFP replicon plasmid was digested 121 with SpeI and KpnI and the resulting fragment inserted into a sub-cloning vector (pBluescript) 122 to create the pBluescript PK. PKs 3 and 4 were removed by digestion with HindIII and AatII 123 before insertion of a synthetic DNA sequence with PK 3 and 4 deleted. PKs 2, 3 and 4 were 124 deleted by PCR amplification using ΔPK 234 Forward primer and FMDV 1331-1311 reverse 125 primer, the resultant product was digested with HindIII and AatII and ligated into the 126 pBluescript PK vector. Complete PK deletion was achieved by introduction of an AflII site at 127 the 3′ end of the poly-C tract by PCR mutagenesis to create the sub-cloning vector, pBluescript 128 C11, which was then used to remove all the PKs by PCR mutagenesis using ΔPK 1234 forward 129 primer and FMDV 1331-1311 reverse primer. The modified PK sequences were removed from 130 the sub-cloning vectors and inserted into the pRep-ptGFP plasmid using NheI-HF and KpnI-131 HF.\\n132 133 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 7 Mutations to disrupt and reform PK structure were introduced using synthetic DNA by 134 digestion with AflII and AatII and ligation into a similarly digested pBluescript PK vector.\\nMutations were then introduced into the replicon plasmid as described above.\\nTo assess the effects of truncation of the poly-C-tract on replication the entire sequence was 137 removed. This was performed by PCR mutagenesis using primers C0 SpeI, and FMDV 1331- In vitro transcription. 143 In vitro transcription reactions for replicon assays were performed as described previously (28).\\nTranscription reactions to produce large amounts of RNA for SHAPE analysis were performed 145 with purified linear DNA as described above, and 1 μg of linearised DNA was then used in a 146 HiScribe T7 synthesis kit (NEB), before DNase treatment and purification using a PureLink FastQ files were quality checked using FastQC with poor quality reads filtered using the 225 Sickle algorithm. Host cell reads were removed using FastQ Screen algorithm and FMDV 226 reads assembled de novo into contigs using IDBA-UD (35). Contigs that matched the FMDV 227 library (identified using Basic Local ALighnment Search Tool (BLAST)) were assembled 228 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint into consensus sequences using SeqMan Pro software in the DNA STAR Lasergene 13 229 package (DNA STAR) (36). The SHAPE data largely agreed with the predicted structures with the stems of PK 1, 2 and 3, interacting nucleotides showed little to no reactivity, suggesting NMIA could not interact with 300 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 14 these nucleotides either due to the predicted base pairing or steric hindrance (figure 2B). The\\nNMIA reactivity for the interacting nucleotides in the stem-loops with downstream residues of 302 PK 1, 2 and 3 again largely agreed with the predicted structure, although the SHAPE data 303 suggests that there might be fewer interactions than previously predicted. However, differences 304 here could be due to heterogeneity in the formation of PKs in this experiment. The evidence 305 for loop-downstream interaction was weaker for PK4. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint\\norientation. 351 Since removal of all four PKs resulted in a significant decrease in replication, the minimal 352 requirements to maintain wt levels of replication were investigated. As near wt level of 353 replication was observed when only one PK was present, all further mutagenesis was 354 performed in a C11 replicon plasmid containing only PK 1. In addition, the orientation of PK 1 was reversed by \"flipping\" the nucleotide sequence to 367 potentially facilitate hybridisation of the loop with upstream rather than downstream sequences.\\nChanging the orientation of the PK reduced replicon replication to a similar level seen in the replication decreased until at passage three there is a 2.5 fold reduction compared to that of 398 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint passage 0 (figure 5B). Therefore, it appears that replicons with a single PK are at a competitive 399 disadvantage compared to those with two or more. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 20 of infectious virus despite being able to replicate after transfection into cells, is consistent with 448 a requirement for RNA structure within the PK region being required for virus assembly. The 5′ UTR of FMDV is unique amongst picornaviruses due to its large size and the presence 454 of multiple RNA elements, some of which still have unknown function. One of these features 455 is a series of repeated PKs varying in number from 2-4, depending on virus strain. In this study, 456 we sequentially deleted or mutated the PKs to help understand their role in the viral life cycle. 457 We also confirmed the predicted PK structures by SHAPE mapping, although there may be Although all viruses isolated to date contain at least two PKs, replicons or viruses containing a 464 single PK were still replication competent. However, replicons with more than a single PK 465 were found to have a competitive advantage over replicons with a single PK when sequentially 466 passaged. Replicons lacking all PKs displayed poor passaging potential even when co-467 transfected with yeast tRNA, reinforcing the observation of a significant impact in replication.\\nMoreover, viruses recovered from genomes with reduced numbers of PKs were slower growing 469 and produced smaller plaques. In addition, these differences were more pronounced in more PKs is functionally competent as no differences was seen between replicons congaing a single 472 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 21 copy of PK1 or PK4. This observation is consistent with a previous report of deletion of PK1, 473 along with the Poly-C-tract, with no adverse effect in viral replication (37). This also supports 474 our findings that the truncation of the Poly-C-tract to create the C11 construct had no effect on 475 replicon replication in the cell lines tested. As has been described with Mengo virus, it is 476 possible that the role of the poly-C-tract is essential in other aspects of the viral lifecycle which 477 cannot be recapitulated in a standard tissue culture system (39).\\nThe presence of at least two PKs in all viral isolates sequenced so far suggests that multiple 480 PKs confer a competitive advantage in replication. Here we showed by sequential passage that 481 replicons containing at least two PKs were maintained at a level similar to wt, but replicons 482 containing only one PK showed a persistent decline. It is unclear why some viral isolates 483 contain two, three or four PKs is still unknown, but this may be stochastic variation or may 484 reflect subtle effects of host range or geographical localisation. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint Significance is shown comparing the replication of C11 PK disrupt and C11 PK restore (Aii). Significance shown is compared to wt replicon. Error bars are calculated by SEM, n = 3, * P 673 &lt; 0.05, **** P &lt; 0.0001. 674 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 33 675 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint \\n</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Analysis Title: Regaining perspective on SARS-CoV-2 molecular tracing and its implications</td>\n",
       "      <td>[{'text': 'During the past three months, a new coronavirus (SARS-CoV-2) epidemic has been growing exponentially, affecting over 100 thousand people worldwide, and causing enormous distress to economies and societies of affected countries. A plethora of analyses based on viral sequences has already been published, in scientific journals as well as through non-peer reviewed channels, to investigate SARS-CoV-2 genetic heterogeneity and spatiotemporal dissemination. We examined all full genome sequences currently available to assess the presence of sufficient information for reliable phylogenetic and phylogeographic studies. Our analysis clearly shows severe limitations in the present data, in light of which any finding should be considered, at the very best, preliminary and hypothesis-generating. Hence the need for avoiding stigmatization based on partial information, and for continuing concerted efforts to increase number and quality of the sequences required for robust tracing of the epidemic.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': '. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'is the (which was not peer-reviewed) The copyright holder for this preprint .', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>In December 2019, a novel coronavirus, SARS-CoV-2, was identified in Wuhan, China, as the etiologic agent of coronavirus disease 2019 , which by March 2020 has already spread across more than 80 countries 1 . Common symptoms of infection include fever, cough, and shortness of breath, while severe cases are characterized by advanced respiratory distress and pneumonia, often resulting in death 2 . Soon after the first epidemiological data together with SARS-CoV-2 genetic sequences were made available, a glut of phylogeny-based analyses began to circulate discussing, in scientific papers as well as (social) media, countries that might have been fueling the spread. The implications of misunderstanding the real dynamic of the COVID-19 pandemic are extremely dangerous. Ethnic or social discrimination resulting from unsupported assumptions on viral contagion -often amplified by irresponsible, uncontrollable communications -can be highly damaging for people and countries.\\nDespite (social) media are often vehicle for fake news and boast news hype, it is also worth noting the tremendous effort of the scientific community to provide free, up-to-date information on ongoing studies, as well as critical evaluations. In particular, the US-based NextStrain 3 team has been posting real-time updates on the epidemic tracing by molecular analyses. Several discussions and evidence-based debates on controversial hypotheses on the epidemic have ensued -such as the number of untraced infections in the US, the putative virus introduction in Italy through Germany, and the alleged lineage diversification in China 4 later criticized 5 -aided by a concerted and productive effort of internationally-renowned research groups in Edinburgh, Glasgow, and Oxford, UK, among others. Recently, an editorial published on Science 6 has also highlighted how unsupported or misleading claims circulating in forums, social media, and even peer-reviewed articles, have been led by a substantial over interpretation of the available data.\\nHence, the urgency to reframe the current debate in more rigorous scientific terms, and quantitatively evaluate whether sufficient information for reliable phylogenetic and phylogeographic studies currently exists, or which gaps need to be addressed.\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint Before carrying out any phylogeny-based analysis of virus evolution and spatiotemporal spread, it is crucial to test the quality of sequence data, since uneven sampling, presence of phylogenetic noise, and absence of temporal signal can affect reliability of the results (e.g. ancestral state reconstructions, molecular clock calibrations) 7 . We downloaded all SARS-CoV-2 full genome sequences deposited on GISAID (https://www.gisaid.org/) 8 to compare the number of high quality genomes sampled per country with the number of confirmed cases at the time of sampling, as well as the country's total population ( Figure 1 ). As March 10 th 2020, 331 genome sequences from 29 countries were available, almost twice as much than just one week earlier (March 3 rd 2020, 169 genome sequences from 22 countries). Yet, the sampling per country was inhomogeneous, with confirmed cases clearly uncorrelated to the number of genomes (Spearman correlation = 0.03). Moreover, correlation could only be investigated with confirmed cases (as proxy), since not all affected countries have made publicly available the total number of coronavirus testing performed. Even within the same country, sequenced genomes were usually sampled from few hotspots, not necessarily representative of the whole epidemic in that country.\\nItaly, for example, the second country per confirmed cases, has uploaded so far only two genomes from Lombardy and three from Lazio (Table S1) , i.e. only two of the twenty Italian regions, all with confirmed cases 9 . China and USA contributed with 123 and 63 genomes respectively (56% combined), while 13 countries contribute with only one or two samples (6.3% combined). Notably, three countries with less than 400 confirmed cases have provided a substantial number of genomes (Netherlands 28, Australia 18, Singapore 11; 17% combined) ( Figure 1 ).\\nThe effect of inhomogeneous sampling and missing data on phylogeography reconstructions, like the ones recently rushed through news and (social) media to claim specific dissemination routes of SARS-CoV-2 among countries, can be quite dramatic. An instructive example is the putative introduction of SARS-CoV-2 in Italy from Germany. The maximum likelihood (ML) tree, inferred from the full genome viral sequences available on March 3 rd , 2020 (Supplementary Figure S1 ), showed a well-supported cluster of European and Asian sequences (reported in Figure 2a ), which contained a subclade (subclade A, Figure 2a ) including a sequence isolated in Germany that . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint appears to be paraphyletic (with strong bootstrap support) to an Italian sequence clustering, in turn, with sequences from Finland, Mexico, Germany and Switzerland (Figure 2a ).\\nBased on this observation (which was available on NextStrain), a heated discussion circulated on social media about a transmission event from Germany to Italy followed by further spread from Italy the other countries. However, in a new tree inferred just one week later, when more than 135 new full genome sequences were made available on GISAID 8 ( Figure S2 ), the direct link between Germany and Italy has disappeared due to the additional clustering of previously unsampled sequences from Portugal, Brazil, Wales and Netherland ( Figure 2b ). Most importantly, there is no sufficient evidence of a higher likelihood (Shimodaira-Hasgawa test, Table 1 ) for tree topologies generated arbitrarily switching branches in the tree (see arrows in Figure 2b) compared to the ML tree inferred from the real data. In other words, it is not possible with the present data to decide which branching pattern (and, therefore, phylogeographic reconstruction)\\nis the most likely.\\nLack of resolution and uncertainty in the SARS-CoV-2 phylogenetic tree is to be expected, considering that relatively little genetic diversity can be accumulated during the first three months of an epidemic, even for an exponentially spreading and fast-evolving RNA virus. Plots of pairwise transitions/transversions vs. genetic distance 10 showed, indeed, low genetic diversity among SARS-CoV-2 genomes, considerable scattering, and overlap in transitions and transversions among closely related sequences (Figure 3a ). The 331 full genomes available contained 477 out of 29,564 variable sites (1.6%), of which 377 were singleton and only 100 (0.34%) parsimony informative, indicating a substantial lack of phylogenetic information, and raising the possibility of a number of sequencing errors. Further assessment of phylogenetic signal in the dataset was carried out using likelihood mapping analysis 11 , which estimates the likelihood of each of possible tree topology for any group of four sequences (quartet), randomly . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint chosen from an alignment, and reports them inside an equilateral triangle (the likelihood map)\\nwhere the corners represent distinct tree topologies and the center star-like trees. Quartets are considered \"resolved\" when the three likelihoods are significantly different (phylogenetic signal, most dots equally distributed in the corners, i.e. data are suitable for robust phylogeny inference), unresolved or partially resolved, when all three likelihood values or two of them are not significantly different (phylogenetic noise, most dots in sides or center areas, i.e. data may not be sufficient for robust phylogeny inference). Extensive simulation studies have shown that, for sequences to be considered robust in terms of phylogenetic signal, side/center areas of the likelihood mapping must include &lt;40% of the unresolved quartets 12 , way lower than the 52.9% unresolved quartets detected in the SARS-CoV-2 full genomes alignment ( Figure 3b ). It is important to keep in mind that such a lack of phylogenetic signal makes the overall topology of any SARS-CoV-2 tree unreliable, and even clades with high bootstrap values should be interpreted with extreme caution. Finally, the mean genetic distance of each sequence from the root of a ML tree indicated that the available genomic data lack temporal signal (no significant linear correlation between accumulated genetic distance and sampling time, Figure 3c ). This is, again, expected in genomes obtained over a very short period of time (~ three months) since the beginning of the outbreak. Bayesian analysis 13 , which infers phylogenetic and phylogeographic patterns from a posterior distribution of trees, might facilitate comparisons about different evolutionary scenarios, help in retrieving the correct topology, and estimate an accurate evolutionary rate. With more genome sequences sampled from different time points and geographic areas made available, literally, every day, Bayesian analysis may soon be a viable option. However, as long the new data do not increase phylogenetic and temporal signal, Bayesian phylogenetic and phylogeography results would remain equally questionable.\\nTaken together, our observations suggest that any conclusion drawn, at present, about existing lineages and direction of viral spread, based on phylogenetic analysis of SARS-CoV-2 sequence data, is potentially misleading. Nowadays, published scientific data and media are very easily . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint .\\naccessible to a worldwide audience, thus it is important more than ever to weigh the information being shared. We believe that current molecular epidemiology data are not solid enough to provide a scientifically sound analysis of SARS-CoV-2 spread. Any inference based on phylogenetic trees should be considered at the very best preliminary, and hypothesis-generating.\\nHence the need for 1) avoiding stigmatization based on partial information and 2) continuing the concerted scientific effort to increase the number and quality of available data. The evolutionary dynamics of SARS-CoV-2 spread will unveil an unprecedented amount of information, which is essential to make policy decisions. The whole of humanity is threatened by the current pandemic, and policy makers need to adjust their mitigation measures while the pandemic itself is developing. Some of the urgent answers required lie in the timely availability of abundant, high quality sequence data not only from countries experiencing a high number of reported cases, such as Italy and South Korea, but also from others that seem to be experiencing, at least for now, a lower number of infections.\\nGISAID was accessed on March 3 rd and 10 th , 2020 (final dataset information reported in Table   S1 ). After quality control of sequences that were not full genomes or contained extensive stretches of unknown nucleotides, the first dataset (March 3 rd 2020) consisted of 169 genomes from 22 countries, while the second (March 10 th , 2020) 331 from 29 countries. Confirmed cases were retrieved for the same day from interactive web-based dashboard developed by Johns Hopkins University 14 .\\nTransition/transversions vs. genetic distance plot were calculated using DAMBE6 10 . Evaluation of the presence of phylogenetic signal satisfying resolved phylogenetic relationships among sequences was carried out with IQ-TREE, allowing the software to search for all possible quartets . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint using the best-fitting nucleotide substitution model 11 . ML tree reconstruction was performed in IQ-TREE based on the best-fit model chosen according to Bayesian Information Criterion (BIC) 15, 16 . Topology was tested performing the Shimodaira-Hasegawa 17 on \"Subclade A\" by comparing the inferred tree to three trees where two branches: (Tree 1) Italy (EPI_ISL_412973) switched with Wales (EPI_ISL_413555), (Tree 2) Germany (EPI_ISL_406862 with) Brazil (EPI_ISL_412964), (Tree 3) and Portugal (EPI_ISL_413648) with Brazil (EPI_ISL_412964), (Tree 4) Germany (EPI_ISL_406862) with Portugal (EPI_ISL_413648). All tests performed 10,000 resamplings using the using the RELL approximation method 18 . Position of these genomes within \"Subclade A\" is indicated in Figure 1 with an arrow.\\nExploration of temporal structure, i.e. presence of molecular clock in the data, was assessed by regression of divergence -root-to-tip genetic distance-against sampling time using TempEst 19 . In this case, absence of a linear trend indicates that the data does not contain temporal signal and that the data is not appropriate for phylogenetic inference using molecular clock models.\\nAncestral state reconstruction of full datasets reported in Figures S1 and S2 were based on the ML tree obtained with IQ-TREE and performed using TreeTime 20 . We choose to show the \"European subclade\" and \"Subclade A\" as cladograms in Figure 1 as there are no phylogenetic and temporal signals in the datasets: scaling phylogenies in time or nucleotide substitution in this case is misleading. . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint Each bi-directional arrow, and corresponding number, connects two tip branches that were switched to generate an alternative tree topology to be tested (see Table 1 , Methods). CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity. EPI_ISL_412971_BetaCoV_Finland_FIN-25_2020 EPI_ISL_406533_BetaCoV_Guangzhou_20SF206_2020 EPI_ISL_407896_BetaCoV_Australia_QLD02_2020 EPI_ISL_408478_BetaCoV_Chongqing_YC01_2020\\nEPI_ISL_412979__BetaCoV_Wuhan_HBCDC-HB-03_2020\\nEPI_ISL_411220_BetaCoV_France_IDF0386-islP3_2020 EPI_ISL_410044_BetaCoV_USA_CA6_2020 EPI_ISL_413020_BetaCoV_Switzerland_1000477377_2020\\nEPI_ISL_406030_BetaCoV_Shenzhen_HKU-SZ-002_2020\\nEPI_ISL_408669_BetaCoV_Japan_KY-V-029_2020\\nEPI_ISL_408515_BetaCoV_Wuhan_IVDC-HB-envF13-21_2020 EPI_ISL_410713_BetaCoV_Singapore_7_2020 EPI_ISL_412873_BetaCoV_Korea_KCDC24_2020\\nEPI_ISL_402120_BetaCoV_Wuhan_IVDC-HB-04_2020\\nEPI_ISL_406970_BetaCoV_Hangzhou_HZ-1_2020 EPI_ISL_406973_BetaCoV_Singapore_1_2020 EPI_ISL_412029_BetaCoV_Hong_Kong_VM20001988_2020\\nEPI_ISL_410720_BetaCoV_France_IDF0372-isl_2020 EPI_ISL_411951_BetaCoV_Sweden_01_2020 EPI_ISL_408488_BetaCoV_Jiangsu_IVDC-JS-001_2020\\nEPI_ISL_402123_BetaCoV_Wuhan_IPBCAMS-WH-01_2019 EPI_ISL_407894_BetaCoV_Australia_QLD01_2020 EPI_ISL_411955_BetaCoV_USA_CA8_2020 EPI_ISL_403962_BetaCoV_Nonthaburi_61_2020 EPI_ISL_406801_BetaCov_Wuhan_WH04_2020 EPI_ISL_408978_BetaCoV_Wuhan_WH05_2020\\nEPI_ISL_407084_BetaCoV_Japan_AI_I-004_2020 EPI_ISL_407215_BetaCoV_USA_WA1-F6_2020 EPI_ISL_411060_BetaCoV_Fujian_8_2020 EPI_ISL_410984_BetaCoV_France_IDF0515-isl_2020\\nEPI_ISL_412980__BetaCoV_Wuhan_HBCDC-HB-04_2020 EPI_ISL_410546_BetaCoV_Italy_INMI1-cs_2020 EPI_ISL_408977_BetaCoV_Sydney_3_2020 EPI_ISL_412982__BetaCoV_Wuhan_HBCDC-HB-06_2020 EPI_ISL_407071_BetaCoV_England_01_2020 EPI_ISL_412912_BetaCoV_Germany_Baden-Wuerttemberg-1_2020 EPI_ISL_411927_BetaCoV_Taiwan_4_2020 EPI_ISL_410536_BetaCoV_Singapore_5_2020 EPI_ISL_406538_BetaCoV_Guangdong_20SF201_2020 EPI_ISL_407988_BetaCoV_Singapore_3_2020 EPI_ISL_406036_BetaCoV_USA_CA2_2020 EPI_ISL_411954_BetaCoV_USA_CA7_2020 EPI_ISL_412028_BetaCoV_Hong_Kong_VM20001061_2020 EPI_ISL_406862_BetaCoV_Germany_BavPat1_2020 EPI_ISL_406597_BetaCoV_France_IDF0373_2020 EPI_ISL_408009_BetaCoV_USA_CA4_2020 EPI_ISL_410716_BetaCoV_Singapore_10_2020 EPI_ISL_403933_BetaCoV_Guangdong_20SF013_2020 EPI_ISL_402129_BetaCoV_Wuhan_WIV06_2019\\nEPI_ISL_406594_BetaCoV_Shenzhen_SZTH-003_2020 EPI_ISL_407987_BetaCoV_Singapore_2_2020 EPI_ISL_413015_BetaCoV_Canada_ON_VIDO-01_2020 EPI_ISL_407976_BetaCoV_Belgium_GHB-03021_2020 EPI_ISL_406531_BetaCoV_Guangdong_20SF174_2020 EPI_ISL_410715_BetaCoV_Singapore_9_2020\\nEPI_ISL_404227_BetaCoV_Zhejiang_WZ-01_2020 EPI_ISL_408479_BetaCoV_Chongqing_ZX01_2020 EPI_ISL_406535_BetaCoV_Foshan_20SF210_2020 EPI_ISL_408976_BetaCoV_Sydney_2_2020 EPI_ISL_411066_BetaCoV_Fujian_13_2020 EPI_ISL_407073_BetaCoV_England_02_2020\\nEPI_ISL_403929_BetaCoV_Wuhan_IPBCAMS-WH-04_2019 EPI_ISL_410301_BetaCoV_Nepal_61_2020 EPI_ISL_403931_BetaCoV_Wuhan_IPBCAMS-WH-02_2019\\nEPI_ISL_410531_BetaCoV_Japan_NA-20-05-1_2020 EPI_ISL_403963_BetaCoV_Nonthaburi_74_2020 EPI_ISL_402125_BetaCoV_Wuhan-Hu-1_2019\\nEPI_ISL_408514_BetaCoV_Wuhan_IVDC-HB-envF13-20_2020\\nEPI_ISL_402119_BetaCoV_Wuhan_IVDC-HB-01_2019 EPI_ISL_411956_BetaCoV_USA_TX1_2020 EPI_ISL_406596_BetaCoV_France_IDF0372_2020 EPI_ISL_412967_BetaCoV_China_IQTC02_2020 EPI_ISL_408431_BetaCov_France_IDF0626_2020\\nEPI_ISL_410532_BetaCoV_Japan_OS-20-07-1_2020\\nEPI_ISL_411957_BetaCoV_China_WH-09_2020 EPI_ISL_412970_BetaCoV_USA_WA2_2020 EPI_ISL_412978__BetaCoV_Wuhan_HBCDC-HB-02_2020 EPI_ISL_412869_BetaCoV_Korea_KCDC05_2020 EPI_ISL_412965_BetaCoV_Canada_BC_37_0-2_2020 EPI_ISL_412871_BetaCoV_Korea_KCDC07_2020 EPI_ISL_408008_BetaCoV_USA_CA3_2020\\nEPI_ISL_406593_BetaCoV_Shenzhen_SZTH-002_2020 EPI_ISL_406536_BetaCoV_Foshan_20SF211_2020 EPI_ISL_412459_BetaCoV_Jingzhou_HBCDC-HB-01_2020 EPI_ISL_410718_BetaCoV_Australia_QLD04_2020 EPI_ISL_412862_BetaCoV_USA_CA9_2020 EPI_ISL_413019_BetaCoV_Switzerland_1000477102_2020\\nEPI_ISL_412981__BetaCoV_Wuhan_HBCDC-HB-05_2020\\nEPI_ISL_408668_BetaCoV_Vietnam_VR03-38142_2020 EPI_ISL_410535_BetaCoV_Singapore_4_2020 EPI_ISL_408665_BetaCoV_Japan_TY-WK-012_2020 EPI_ISL_406716_BetaCoV_China_WHU01_2020 EPI_ISL_402127_BetaCoV_Wuhan_WIV02_2019 EPI_ISL_411926_BetaCoV_Taiwan_3_2020 EPI_ISL_406223_BetaCoV_USA_AZ1_2020 EPI_ISL_407214_BetaCoV_USA_WA1-A12_2020 EPI_ISL_406717_BetaCoV_China_WHU02_2020 EPI_ISL_404228_BetaCoV_Zhejiang_WZ-02_2020\\nEPI_ISL_412898__BetaCoV_Wuhan_HBCDC-HB-02_2019 EPI_ISL_413017_BetaCoV_South_Korea_KUMC01_2020 EPI_ISL_408480_BetaCoV_Yunnan_IVDC-YN-003_2020 EPI_ISL_403936_BetaCoV_Guangdong_20SF028_2020 EPI_ISL_408667_BetaCoV_Japan_TY-WK-521_2020 EPI_ISL_411218_BetaCoV_France_IDF0571_2020 EPI_ISL_405839_BetaCoV_Shenzhen_HKU-SZ-005_2020 EPI_ISL_408010_BetaCoV_USA_CA5_2020 EPI_ISL_406031_BetaCoV_Taiwan_2_2020 EPI_ISL_406844_BetaCoV_Australia_VIC01_2020 EPI_ISL_411950_BetaCoV_Jiangsu_JS01_2020 EPI_ISL_412974_BetaCoV_Italy_SPL1_2020\\nEPI_ISL_403928_BetaCoV_Wuhan_IPBCAMS-WH-05_2020 EPI_ISL_412870_BetaCoV_Korea_KCDC06_2020 EPI_ISL_404253_BetaCoV_USA_IL1_2020 EPI_ISL_410045_BetaCoV_USA_IL2_2020 EPI_ISL_403932_BetaCoV_Guangdong_20SF012_2020\\nEPI_ISL_408484_BetaCoV_Sichuan_IVDC-SC-001_2020 EPI_ISL_410537_BetaCoV_Singapore_6_2020 EPI_ISL_412872_BetaCoV_Korea_KCDC12_2020 EPI_ISL_407079_BetaCoV_Finland_1_2020\\nEPI_ISL_410545_BetaCoV_Italy_INMI1-isl_2020 EPI_ISL_408670_BetaCoV_USA_WI1_2020 EPI_ISL_403937_BetaCoV_Guangdong_20SF040_2020 EPI_ISL_403934_BetaCoV_Guangdong_20SF014_2020\\nEPI_ISL_413014_BetaCoV_Canada_ON-PHL2445_2020\\nEPI_ISL_411915_BetaCoV_Taiwan_CGMH-CGU-01_2020 EPI_ISL_410717_BetaCoV_Australia_QLD03_2020 EPI_ISL_412026_BetaCoV_Hefei_2_2020 EPI_ISL_407313_BetaCoV_Hangzhou_HZCDC0001_2020 EPI_ISL_412972_BetaCoV_Mexico_CDMX_InDRE_01_2020 EPI_ISL_406034_BetaCoV_USA_CA1_2020 EPI_ISL_411952_BetaCoV_Jiangsu_JS02_2020 EPI_ISL_411902_BetaCoV_Cambodia_0012_2020\\nEPI_ISL_413016_BetaCov_Brazil_SPBR-02_2020 EPI_ISL_413018_BetaCoV_South_Korea_KUMC02_2020 EPI_ISL_403930_BetaCoV_Wuhan_IPBCAMS-WH-03_2019\\nEPI_ISL_412899__BetaCoV_Wuhan_HBCDC-HB-03_2019 EPI_ISL_408489_BetaCoV_Taiwan_NTU01_2020 EPI_ISL_409067_BetaCoV_USA_MA1_2020 EPI_ISL_404895_BetaCoV_USA_WA1_2020\\nEPI_ISL_408666_BetaCoV_Japan_TY-WK-501_2020 EPI_ISL_410218_BetaCov_Taiwan_NTU02_2020 EPI_ISL_407893_BetaCoV_Australia_NSW01_2020 EPI_ISL_407193_BetaCoV_South_Korea_KCDC03_2020\\nEPI_ISL_412983_BetaCoV_Tianmen_HBCDC-HB-07_2020 EPI_ISL_410714_BetaCoV_Singapore_8_2020 EPI_ISL_403935_BetaCoV_Guangdong_20SF025_2020 EPI_ISL_406798_BetaCov_Wuhan_WH01_2019 EPI_ISL_411953_BetaCoV_Jiangsu_JS03_2020 EPI_ISL_412030_BetaCoV_Hong_Kong_VB20026565_2020 EPI_ISL_412968_BetaCoV_Japan_Hu_DP_Kng_19-020_2020 EPI_ISL_412973_BetaCoV_Italy_CDG1_2020 EPI_ISL_402132_BetaCoV_Wuhan_HBCDC-HB-01_2019\\nEPI_ISL_408486_BetaCoV_Jiangxi_IVDC-JX-002_2020 EPI_ISL_408430_BetaCoV_France_IDF0515_2020 EPI_ISL_410486_BetaCoV_France_RA739_2020 EPI_ISL_412966_BetaCoV_China_IQTC01_2020\\nEPI_ISL_412969_BetaCoV_Japan_Hu_DP_Kng_19-027_2020\\nEPI_ISL_411219_BetaCoV_France_IDF0386-islP1_2020\\nEPI_ISL_408481_BetaCoV_Chongqing_IVDC-CQ-001_2020 EPI_ISL_412975_BetaCoV_Australia_NSW05_2020 EPI_ISL_406534_BetaCoV_Foshan_20SF207_2020 . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint \\n</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Healthcare-resource-adjusted vulnerabilities towards the 2019-nCoV epidemic across China</td>\n",
       "      <td>[]</td>\n",
       "      <td>The 2019-nCoV epidemic has spread across China and 24 other countries 1-3 as of February 8, 2020 . The mass quarantine measure implemented in the epidemic's epicenter, Wuhan, Hubei Province, China, on January 23, 2020, has stopped the human migration between Wuhan and other places. However, according to the local authorities, over five million people had already traveled from Wuhan to other parts of China in the weeks leading to January 23, 2020, with the majority (79.96%) to the other cities within Hubei.\\nHealthcare resources have become a major bottleneck in the response to this public health crisis. The sudden onset of the epidemic and the sheer size of the infection have caused significant shortage in healthcare resources across Hubei Province. So far, there have been over 11,000 non-local healthcare professionals, mobilized by the public health authorities from other parts of China, fighting against the epidemic in Wuhan. However, this is far from adequate for Wuhan given the severity of the epidemic. In addition to Wuhan, other cities in Hubei are in need of urgent external help as well with respect to health resources, as multiple localized outbreaks are taking place. On February 7, 2020, the National Health Commission of China (NHCC) initiated the \"One City Covered by One Province\" program, explicitly assigning a specific province to aid one city in Hubei. Two critical questions of operational nature remain, which cities, both in Hubei and other provinces, are most vulnerable to localized outbreaks, and how the public health authorities should balance the healthcare resources under stress to best fight the epidemic and minimize the risks? This study is aimed to shed light on these questions by developing a vulnerability analysis framework integrating the human movement and healthcare resource data.\\nThis study makes use of four public datasets: (a) the daily counts of confirmed 2019-nCoV cases, collected from NHCC 4 ; (b) the list of designated hospitals with the number of reserved beds, retrieved from the websites of the provincial and municipal health commissions; (c) the information of the hospitals designated for 2019-nCoV, retrieved from 99.com.cn, a public hospital database; and (d) the human movement data, provided by the Baidu Migration (http://qianxi.baidu.com/), a public site offering real-time migration information across mainland China.\\nRecent studies indicated that the confirmed case data for Hubei Province, in particular, Wuhan, was subject to the under-reporting bias, while the data from places outside of Hubei is more reliable for inferring the actual epidemic size 5 . Following Wu et al. 5 , we developed a susceptible-exposed-infectious-recovered (SEIR)-based metapopulation model, which has 100 sub-populations representing the 100 cities that have the greatest number of people traveling from Wuhan. Note that we adopted the same epidemiological parameters as Wu et al. 5 . This model was then used to estimate IDi, the number of infected individuals in each city. Lastly, healthcare-resource-adjusted vulnerability indexes to the 2019-nCoV outbreak for city i, Vi were obtained as follows: Vi = IDi / Bi, where Bi represents the total number of reserved beds in the designated hospitals.\\nThe Figure presents the estimated number of infected individuals, as well as the healthcareresource-adjusted vulnerability to the 2019-nCoV outbreak across mainland China, as of January 29, 2020. The epidemiological parameters may have changed after January 23, 2020, when the lockdown measure was implemented. Thus, we only use the data by January 23, 2020 and produce the estimation till January 29, 2020 to accommodate the six-day incubation period identified by Wu et al. 5 .\\nThe ten most vulnerable non-Hubei cities are also listed in the figure. Note that the cities in Hubei (which are being aided by other provinces) are not presented here because of their significantly higher vulnerability as compared to others. Among the ten most vulnerable non-Hubei cities, Changsha, Beijing, Chongqing and Shanghai are major metropolises with a large number of estimated infected individuals. Xinyang, Jiujiang, Nanchang and Yiyang are medium-sized cities neighboring Hubei. In particular, Xinyang appears to be in a particular vulnerable situation because of the large number of people traveling from Wuhan and its insufficient healthcare resources.\\nThe reported vulnerability analysis informs public health response to the 2019-nCoV epidemic in multiple ways. For the identified cities with high vulnerability, it is important to increase the reserve of healthcare resources or have contingent plans. In particular, some of these cities already contributed significant healthcare resources to Hubei, leaving themselves inadequately prepared for the localized outbreak. Our analysis revealed that several medium-sized cities, particularly those neighboring Hubei, are highly vulnerable due to the lack of healthcare resources. \\n</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>CHEER: hierarCHical taxonomic classification for viral mEtagEnomic data via deep leaRning</td>\n",
       "      <td>[{'text': 'The fast accumulation of viral metagenomic data has contributed significantly to new RNA virus discovery. However, the short read size, complex composition, and large data size can all make taxonomic analysis difficult. In particular, commonly used alignment-based methods are not ideal choices for detecting new viral species. In this work, we present a novel hierarchical classification model named CHEER, which can conduct read-level taxonomic classification from order to genus for new species. By combining k-mer embedding-based encoding, hierarchically organized CNNs, and carefully trained rejection layer, CHEER is able to assign correct taxonomic labels for reads from new species. We tested CHEER on both simulated and real sequencing data. The results show that CHEER can achieve higher accuracy than popular alignmentbased and alignment-free taxonomic assignment tools. The source code, scripts, and pre-trained parameters for CHEER are available via GitHub: https://github.com/KennthShang/CHEER.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>Metagenomic sequencing, which allows us to directly obtain total genomic DNAs from host-associated and environmental samples, has led to important findings in many areas, such as digestive health [1] . While bacteria are the main focus of most metagenomic sequencing projects, there are fast accumulation of viral metagenomic data with sequencing viruses as the main purpose [2] . There are different types of viruses. We are mainly concerned with RNA viruses because many RNA viruses are notorious human pathogens, such as Influenza A, Human immunodeficiency virus (HIV), Ebola, SARS-CoV, and recently identified 2019-nCoV causing Wuhan pneumonia. Unlike DNA viruses, RNA viruses, which contain RNA genomes, lack faithful proofreading mechanisms during replication and thus can produce a group of related but different viral strains infecting the same host. This high genetic diversity within and across different hosts poses a great challenge for designing long-term protection strategies against these infectious diseases. For example, as the circulating strains can change every year, flu vaccine has to be administered every year.\\nAdvances in viral metagenomics have contributed significantly to new RNA virus discovery. According to a survey by Woolhouse et. al, the number of newly identified RNA viruses is changing from 1,899 in 2005 to 5,561 in 2018, which is 3 times of increase [3] . Largescale RNA virus sequencing projects using nextgeneration sequencing technologies have been conducted for different species. For example, Shi et al. have discovered a large number of new RNA viruses by sequencing samples from invertebrate and vertebrate animals [4, 5] . Claire et al. have discovered 25 new RNA viruses by sequencing samples associated with different drosophilid [6] . Bolduc et al. sequenced RNA viruses from Archaeal and bacterial samples [7] . Given fast accumulation of viral metagenomic data and expected discovery of new viruses, a key step is to conduct composition analysis for these data and assign taxonomic groups for possibly new species.\\nComposition analysis can be conducted at the read level or contig level. There is no doubt that contigs can provide more information than short reads in taxonomic classification. However, metagenomic assembly is still one of the most challenging computational problems in bioinformatics. Unlike single genome assembly, metagenomic assembly is more likely to produce chimeric contigs that can contain regions from different species. Thus, metagenomic assembly is often preceded by read binning that groups reads with the same or similar taxonomic labels [8] in order to achieve better performance.\\nIn this work, we design a read level taxonomic classification tool (named as \"CHEER\") for labeling new species in viral metagenomic data. To detect known species with available reference genomes, alignment-based methods can provide sufficient information for both classification and abundance estimation. This function has been incorporated in existing pipelines [9, 10, 11] and is thus not the focus of our tool. Instead, our tool is designed to handle the challenging cases of assigning taxonomic labels for reads of new species, which have not been observed before. This problem was clearly formulated in metagenomic phylogenetic classification tools such as Phymm and PhymmBL [8] . We first define the classification problem following the formulation in Phymm: the classification problem is the assignment of a phylogenetic group label to each read in input data sets. As we are only interested in the hard case of classifying reads of new species, which have not been sequenced and thus no label exists for these species, the expected labels for these reads are the higher-rank taxonomic groups such as genus, family, or order. All our data and experiments are designed so that the test species are masked in the reference database and the trained models.\\nPrevious experiments have shown that nucleotide-level homology search is not sensitive enough for assigning higher rank taxonomic groups to reads from new species [9, 10] . Protein-based homology search is thus preferred for phylogenetic classification [11, 12] . In particular, a recent tool, VirusSeeker [13] utilized BLASTx to classify reads into bacteria, phages, and other viruses. It is not clear whether the protein homology search is able to generate accurate classification at lower ranks. We will evaluate this in our experiments.\\nThere are also alignment-free methods for phylogenetic classification. Many of those tools implemented different machine learning models and were mainly designed for bacteria. For example, RDP [14] adopted a Naïve Bayes based classifier that can automatically learn features from the reads and assigned taxonomic labels to sequences from bacterial and archaeal 16S genes and fungal 28S genes. The NBC classifier [15, 16] encoded metagenomic reads using k-mers and implemented a bag-of-word model to achieve the readlevel taxonomic classification. Phymm [8] utilized interpolated Markov models (IMMs) for metagenomic read classification. These learning-based models enable users to label reads originating from new species.\\nThe previous works have shown that sequence composition is still an important and informative feature for phylogenetic classification. Recent applications of deep learning models have achieved better performance than conventional machine learning methods in different sequence classification problems [17] . In particular, the convolution filters in Convolutional Neural Network models (CNN) can represent degenerated sequence motifs and thus help CNN models to learn abstract composition-based features. For example, frequently activated convolution filters learned by DeepFam often represented the conserved motifs in different protein domain families [18] . The successful applications of deep learning in sequence classification motivated us to design a novel deep learning-based classification model for assigning taxonomic groups for new species in viral metagenomic data.\\nOur main contributions are summarized below. First, our deep learning model can assign higher rank phylogenetic group labels (such as genus) for reads from new species. To achieve this goal, our model organizes multiple CNN-based classifiers in a tree for hierarchical taxonomic group assignment. Second, to enable our model to capture as much information as possible from short reads, we implemented and compared two encoding methods: one hot vs. embedding. Third, as viral metagenomic data usually contains contaminations from either the host genomes or other microbes, we formulated the pre-processing step as an open set problem in targeted image classification and rejected non-viral reads by choosing appropriate negative training sets. We tested our model on both simulated metagenomic data and also real sequencing data. The results show that our model competes favorably with other popular methods.\\nIn this section, we will introduce our method for viral metagenomic taxonomic classification. First, we will show the architecture of CHEER, which is a hierarchical classification model from order to genus. Second, classifiers at each level will be described. Third, we will introduce DNA sequence encoding using skipgram based word embedding and compare it with onehot encoding. Forth, we will describe the rejection layer, following the idea of the open set problem, which is adopted to filter reads not belonging to RNA viruses. Finally, we will detail our dataset used in training and validation.\\nThe architecture of our model is sketched in Fig. 1. The key component is a tree model that consists of multiple classifiers from order to genus. In order to conduct phylogenetic classification for reads from new species, or even new genus, our classification is conducted using a top-down approach from the root to the leaf node. The top layer is a trained CNN that can reject reads not belonging to RNA viruses, which is an important preprocessing step for viral metagenomic data because of the contamination from the host genome or other microbes. Then, the filtered reads (i.e. mostly RNA viral reads) will be fed into the hierarchical classification model, which is referred to as the tree model hereafter. Specifically, the order-level classifier is responsible for classifying the reads into their originating orders. Then each order has a separately trained classifier for assigning input reads to families within that order (O1 to O3 in Fig. 1 ). For all the reads assigned into one family, the family classifier (F1-F7) will assign the reads to different genus within that family. As an example, a path for classifying a read from a species in genus 7 is highlighted in Fig. 1 . CHEER also implemented early stop functions at each level for stopping the classification path at higher ranks. This function can accommodate taxonomic classification for species from new genus or even higher ranks, which is possible for RNA viruses. It is convenient to add fine-grained classifiers within each genus so that we can assign species-level labels. However, as our focus is to conduct phylogenetic classification for reads from new species that do not have a species labels in the training data, we did not include the species-level assignment in this work.\\nAlternative designs of hierarchical classification work, like [19, 20] , either build a classifier for each taxonomic rank or build a binary classifier for each class. We implemented a structure that built just one classifier for each taxonomic rank and compared different structures' performance. According to the result shown in [Supplementary file 1], the structure shown in Fig.1 is better than one classifier per rank. Although one classifier per rank requires fewer classifiers, it needs to train many more parameters because the number of labels for each classifier increases by times. As a result, we observe overfitting more frequently.\\nEach classifier in the tree model is implemented using CNN, which has achieved superior performance in various sequence classification problems [18] . The convolution filters used in CNN resemble positionspecific weight matrix of motifs in genomic sequences. Frequently activated convolution filters can represent well-conserved motifs among sequences in the same class. Thus, a large number of convolution filters will be used in our CNN in order to learn well-conserved sequence features in different classes. Deep CNNs in fields such as computer vision often have multiple convolution layers in order to obtain more abstract features from training images. Here, we use a wide convolution design rather than a deep one, following the idea in [18] . Fig. 2 shows the classifier with the default hyper-parameters. First, reads will be encoded in matrices as input to the convolution layer using two methods. We will describe and compare these two encoding methods later in the Section 2.3. Then, multiple convolution filters of different sizes are utilized to learn the conserved sequence features. According to our empirical study, we choose 3, 7, 11, 15 as the filter sizes, with 256 filters for each size respectively (see Fig. 2 ). Max pooling is then applied to each convolution filter's output so that only the highest convolution value is kept. After the max-pooling layer, we have a 256-dimentional vector as the output of each convolution layer. We concatenate these four vectors into one 1024-dimentional vector and feed it to one fully connected layer with 512 hidden units. Dropout is applied to mitigate overfitting. The CrossEntropyLoss function supplied by Pytorch is adopted as an error estimation function. Due to the unbalanced size of each training class, we use the reciprocal of the proportion of each class as their error weight to help the model fairly learn from each class. And we utilize Adam optimizer with a 0.001 learning rate to update the parameters. The model is trained on HPCC with the 2080Ti GPU unit. \\nIn previous deep learning-based sequence classification models [21, 22] , one hot encoding was commonly adopted to convert a sequence into a matrix, which will then be used as input to convolution layers. There also exist some methods using k-mer composition [23] and frequency [15, 16] to encode the DNA sequence, similar to the bag-of-words model in the field of natural language processing. However, these k-mers representation cannot give complete sequential information about the original reads.\\nTo incorporate both the k-mer composition and also their ordering information, we build a Skip-Gram [24] based embedding layer that can learn which k-mers tend to occur close to each other. Embedding is widely used in natural language processing to learn the semantic and syntactic relationships from sentences. A neural network with one hidden layer is trained to map a word to an n-dimensional vector so that the words that usually occur together will be closer in the n-dimension space. For our DNA read classification task, k-mers are the words and the embedding layer will map proximate kmers into vectors of high similarity. The main hyper-parameter when training the skip-gram model is the size of the training context m, which defines the maximum context location at which the furthest k-mer is taken for training. Specifically, for a kmer at position i as the input to the skip-gram model, the output k-mers are its neighbors located at i+jk, where −m ≤ ≤ m. Larger m results in more training samples with a cost of training time. In our model, the default value of m is 1. As shown in Fig. 3 , we will sample 3 k-mers as our dataset at a time with the middle one as the input and the surrounding two as the output.\\nAs proven in [24] , Skip-Gram is able to automatically learn the relationship between proximate words and thus is expected to help learn the order between k-mers in genomic data. We implemented a hidden layer with 100 hidden units shown in Fig. 3 . After training the model, the co-occurrence features are embedded in the output of the hidden layer. Thus, a k-mer in a read will be represented as a 100-dimension vector, where each element in the vector is the output of the hidden unit in the hidden layer.\\nWhen we use this pre-trained embedding layer to encode the reads, we first split each read into sequential k-mers. Second, each k-mer is fed to the embedding layer and the output is a 100-dimension vector. Third, all these vectors are combined into a large matrix according to the original position of each k-mers. Finally, this matrix will be fed to the convolutional layer.\\nBased on our large-scale experiments, we found that classifiers are easy to overfit with increase of k. Thus, we choose 100-dimensional feature vectors with k being 3 as our default embedding layer parameter set based on our empirical analysis.\\nDue to the small genome size and limitations in library construction, it is not rare that many viral metagenomic datasets are still contaminated with the genomes of the host and other microbes. Viral read screening is thus a key pre-processing step before the downstream composition analysis. Specifically, our tree model should only be applied to viral reads. However, the close set nature of the classifier forces it to assign a known label to each input without being able to distinguish viral reads from others. In this work, we will reject non-viral reads by formulating it as an open set problem in targeted image classification [25] .\\nIn a majority of neural network models [26, 27] , the logit of the last layer will be fed into the SoftMax function, which takes a vector of n real numbers as input and output a normalized probability vector for all labels/classes. However, for an irrelevant input from an unknown clade, all classes in the model tend to have low probabilities and thus applying a threshold on uncertainty can be used to reject unknown classes [25] . Along with this idea, serval approaches [28, 29] have been proposed to solve the open set problem.\\nIn this work, we incorporate the pre-processing step in our hierarchical classification model by integrating a trained CNN and a SoftMax threshold. Contaminations such as bacteria will be rejected using empirically chosen SoftMax threshold τ. Our experiments show that SoftMax outputs a nearly uniform probability distribution of all labels if the input is not relevant. Thus, the SoftMax threshold τ is used to reject reads not belonging to either DNA or RNA viruses. However, using this thresholding method is not able to distinguish DNA viral reads from RNA viral reads due to their similar sequence composition. To accommodate this challenge, we designed a rejection classifier in the top layer using DNA viral reads as the negative class in the training (see Fig. 1 ).\\nAs the focus of our work is to assign labels for new species, the hierarchical classification ends at genuslevel assignment, which is conducted by the bottom layer in Fig. 1 . Given the fast accumulation of viral metagenomic data, there are cases where a new species does not belong to any existing genus and thus the label assignment should stop at the higher ranks. Thus, we allow our model to stop at a higher rank in the tree during the top-down classification. Towards this goal, we still apply SoftMax threshold in our early stop function, which stops reads from further classification by the child classifiers. This early stop function also serves as a quality control measure for preventing wrong classifications if the confidence is low. For example, when a read arrives, the model may be able to identify its order and family labels with high reliability but cannot decide which genus it belongs to. In this case, the model should only assign order and family label to this read rather than classifying it into a potentially wrong genus. To achieve this goal, SoftMax threshold is applied for every classifier in the tree. Only reads with a score higher than the threshold will be assigned to the child class.\\nAn example of early stop is highlighted by the yellow path in Fig. 1 . The model can classify the read into order 2 (O2) and then family 4 (F4) with high reliability but cannot decide which genus it belongs to.\\nAll the viruses used in the experiments were obtained from the RefSeq in the NCBI virus database. The The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.26.009001 doi: bioRxiv preprint viruses' taxonomic classification was downloaded from Virus Taxonomy 2018b Release [30] published by the International Committee on Taxonomic of Virus (ICVT). Because training deep learning classifiers required enough samples, we removed the orders with only one family, families with only one genus, and genera with less than 3 species from our dataset. Finally, there are 6 orders, 23 families, and 55 genera remained. The RefSeq genomes are available at the following URL:https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/fi nd-data/virus. The information of the taxa can be found in [Supplementary file 1]. As CHEER conducts read-level taxonomic classification, the training is also conducted using reads. In order to train our model on each complete virus genome, we generated simulated reads by uniformly extracting substrings of 250 bp with 200bp overlaps. All these reads formed the training set so that the model can learn the features from the whole genome. The validation set, however, is generated by read simulation tool WgSim [31] with specified error rate so that we can have a better estimation of the model's performance on real sequencing data. Some existing deep learning tools for taxonomic classification, like [21, 23] , generate reads first, then randomly choose some of them as training set and others as testing set. This strategy cannot guarantee that the training and testing data sets have no overlaps from the same species. In order to test whether our model can learn the label of new species, we use test species masking in the training set. Specifically, we split our RNA virus database into two independent sets. For any test species, their reads are not used in the training data. Thus, the model will only use the knowledge of known species for the new species classification.\\nIn this section, we will first give details of the hyperparameters of CHEER. Then we present the results of applying CHEER to both simulated and mock sequencing data. For the simulated data by WgSim, we evaluated how sequencing error rate affects the performance of CHEER and compared the taxonomic classification results with state-of-the-art. For the mock data, we tested CHEER's performance in both taxonomic classification and its ability in distinguishing viral reads from others.\\nIn our experiments, the minimum number of species within a genus is 3. We thus constructed three test sets, each containing reads from one species that is randomly chosen from each genus. The three test sets contain three different species, respectively. In total, 165 species from 55 genera are chosen to be \"new\" species for testing and the information of these species can be found in [Supplementary file 1].\\nWe used WgSim to simulate reads from testing species. In order to evaluate the impact of sequencing error on CHEER, we simulated reads with an error rate of 0, 0.01, 0.02, and 0.05, respectively. For each classifier (CNN), we computed the classification accuracy, which is defined in equation (1). To ensure that the computation of the accuracy is not affected by the number of reads from different species, we constructed a balanced test read set by simulating the same number of reads for each test species (1,000 reads for each). (1) For each component classifier in the tree model, we record the average accuracy for multiple test species. For each taxa level, we calculate the average accuracy of all the classifiers in the same level.\\nWe present the results of CHEER at each level in this section. Because CHEER is designed for new species, we benchmark it against alignment-free models. Based on the results shown by Wood et al. [7] , Naïve Bayes Classifier (NBC) outperformed other alignment-free models. Thus, we compared the performance of CHEER and NBC. NBC in [14] was designed for bacteria taxonomic classification and thus, we retrained NBC on our RNA virus dataset. We implemented the NBC model following [14] with the same parameters. Since each family has a separately trained classifier, which conducts genus classification within that family, there are totally 14 classifiers at this level. Fig. 5 shows the accuracy for labeling families. There are totally 6 classifiers that assign reads to families within each of the 6 orders. Fig. 6 shows the mean accuracy at each rank. The family and genus level accuracy are the average accuracy of all classifiers in Fig. 4 and Fig. 5 . There is only one classifier for order level classification.\\nAs shown in Fig. 4, Fig. 5, and Fig. 6 , CHEER has better performance than NBC across different ranks. We also compared the performance of using one-hot encoding and embedding layer in CHEER. The performance comparison at each rank shows that for most of the classifiers, the embedding layer improves the learning ability. Fig. 4, Fig. 5, and Fig. 6 also revealed the robustness of CHEER. Although the classification accuracy decreases as the error rate increases, the decrease is smaller than 2%. The accuracy is almost identical for many classifiers. Thus, CHEER is not very sensitive to sequencing errors. We also conducted experiments for reads with higher error rates (0.02 and 0.05). The results can be found in [Supplementary file 1].\\nThe next experiment is to evaluate our model on a real sequencing dataset and test CHEER's performance in both taxonomic classification and non-viral read rejection. We built a mock dataset by mixing Ebolavirus reads with gut bacterial reads from a gut amplicon sequencing dataset, which is chosen to ensure that we only use reads from bacteria. The taxonomic classification of Zaire Ebolavirus in the tree model is highlighted in Fig. 7 . We chose one of the Ebola species named Zaire Ebolavirus as our testing species because it is the most dangerous of the five Ebolavirus species and was responsible for the 2014 West African Ebola outbreak. We downloaded the dataset from NCBI with accession number SRR1930021. These reads were sequenced by Illumina using whole-genome amplification strategy. Since Zaire Ebolavirus usually causes fatal hemorrhagic fever and abdominal pain, it is possible that this Ebola virus occurs with other gut bacteria. Thus, we created a mock viral metagenomic data by mixing these Ebola reads with a gut bacteria amplicon sequencing data in order to test whether CHEER can reject the bacterial reads at the top rejection layer. The gut bacteria dataset is downloaded from NCBI with accession number: SRR10714026. There are totally 2,737 Ebola virus reads and 51,280 gut bacterial reads in the mock data. After the rejection layer, the Ebola reads will be used to evaluate our tree model.\\nTo treat Zaire Ebolavirus as a \"new\" species, we removed its genome from our training set and re-trained the whole pipeline. As shown in Fig. 7 , after removing this species, we still have four Ebola species in our training set. However, our read mapping experiment using Bowtie2 [32] showed that none of the reads from Zaire Ebolavirus can be mapped to other Ebola species with the default parameters, indicating that using usual alignment-based methods will have difficulty to conduct genus-level assignment.\\nIn our work, CHEER will first use rejection layer to remove the reads not belonging to the RNA virus. Second, the \"Order classifier\" will be applied to evaluate how many reads can be classified into Mononegavirales order. Third, \"Mononegavirales classifier\" takes all these Mononegavirales reads as input to predict how many reads will be assigned to the Filoviridae family. Finally, \"Filoviridae classifier\" is adopted to check how many reads belong to the Ebola genus. Fig. 8 show the ROC curve of the rejection layer. True positive rate reveals how many Ebolavirus reads pass the rejection layer. The false positive rate reveals how many gut bacterial reads pass the rejection layer. We generated thresholds from 0 to 1 with step size 0.01 and used each threshold for the rejection layer. Then for each threshold, we recorded the true positive rate and false positive rate for plotting the ROC curve. As shown in Fig. 8 , with the increase of the threshold, more bacterial reads are filtered by the rejection layer, at the expense of missing more real Ebola reads. The default threshold for the rejection layer is 0.6 in CHEER. User can adjust the threshold according to their needs.\\nAfter evaluating the rejection layer, we continued to test the hierarchical classification performance at each rank. We first compared our tree model with NBC. Then we also benchmarked CHEER with the method using protein-level alignment in Section 3.3.3. The performance of the complete hierarchical classification pipeline of NBC and CHEER is shown in Fig. 9 . From order to genus, each bar shows the accuracy of correctly classified reads by the parent classifier. For example, NBC correctly classified 1580 at the order level. Then, 1564 out of the 1580 reads are correctly labeled at the family level. While Fig. 4 , 5, and 6 have shown the performance of CHEER and NBC at each taxonomic rank using the same set of reads as input for each level of classifiers, this experiment focused on comparing the hierarchical classification pipeline using only correctly predicted reads by the parent classifier. The results demonstrated that CHEER can classify Ebola reads with better performance. NBC can only classify 1,564 (57.1%) reads into Ebolavirus. CHEER, however, had totally 2407 (87.9%) reads classified correctly.\\nAlthough nucleotide-level alignment models are not ideal choices for predicting new species, as mentioned in [8] , protein-level alignment such as BLASTx is still frequently used for homology search because of the conservation among homologous protein sequences. We followed VirusSeeker [10] , which is a BLASTx based model, to align translated reads to virus-only protein database (VIRUSDB NR) provided by NCBI. Then, we kept alignments with E-value cutoff 0.001, which is used in VirusSeeker. The output of BLASTx shows that a read can often align to multiple proteins, making the phylogenetic classification ambiguous (see Fig. 11 ). Also, existing tools such as VirusSeeker often use the best alignment only, which does not always yield the highest accuracy. To optimize the usage of the alignment-based method, we implemented k-Nearest Neighbor (kNN) algorithm to record the best k alignments rather than only using the best one. The label of the input read is determined by the majority vote in the labels of the top k alignments. According to Fig. 10 , most of the reads have hit numbers between 11 to 15 and 6 to10. Thus, we chose 1, 5, 10, 15 as our k when running kNN. There are also 512 reads that cannot be aligned with any VIRUSDB protein sequence. All these reads will be regarded as wrong prediction. Fig. 11 shows the classification results of kNN. When we use the best hit as the prediction the accuracy is only 64%. The best performance of kNN will be 74% with 2,026 correct predictions when k is 5. But the accuracy sharply decreases if k is larger than 5. Fig. 10 and Fig. 11 also revealed that it is hard for users to choose a proper k in real classification tasks. The most common value of the hit number is 11-15 shown in Fig. 11 . The best kNN result, however, has a k equal to 5. Thus, comparing to this alignment-based method, CHEER, with 2,407 (87.9%) reads classified correctly, is more practical and reliable.\\nThe two main purposes of the early stop function are: 1) to handle the cases of a new species from an unseen genus or higher ranks; 2) to serve as a quality control measure for preventing wrong classifications in child classifiers. As Fig. 6 shows, the higher rank classification tends to be more accurate than lower ranks. Because the genus of Ebolavirus exists in our classification tree, the evaluation of the early stop function focuses on the second purpose. Fig. 12 shows the percentage of how many reads are classified into Ebolavirus genus correctly for different classifiers. By allowing early stop, the number of wrong predictions decreased sharply. In short, although there are fewer reads being labelled at the genus level, the total number of reads with wrong classification is decreased. Thus, this is still a useful feature provided by CHEER.\\nAnother significant advantage of CHEER is the high classification speed. Although training the model uses heavy computing resources, the inference procedure of this algorithm is fast. And with the help of GPU acceleration, the training time also decreases a lot.\\nIn the experiments shown in Section 3.4, it takes around 46 minutes to run BLASTx against the virus protein database. Our method, however, only takes 7 second for the preprocessing step to convert the raw sequence into a vector, and then takes less than 35 second to make the prediction for all 2,737 reads.\\nIn this work, all training data are RNA virus genomes downloaded from the NCBI RefSeq database. The training reads are extracted uniformly from the genome to make sure that classifiers can learn the features from the whole genome. To evaluate the model on new species detection, all species in the test set are excluded from the training data. Error rates are also added to evaluate the robustness of the model. The results reveal that the model is tolerant to the sequencing error. In addition, a SoftMax threshold rejection layer is applied to filter reads from non-RNA viruses. The SoftMax threshold in each classifier will also help early stop classification with low confidence and thus decreases the wrong predication rate. We show a case study on real-world data to identify how this threshold will influence the rejection classifier and early stop function. The results show that CHEER competes favorably with state-of-the-art tools.\\n</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Real-time, MinION-based, amplicon sequencing for lineage typing of infectious bronchitis virus from upper respiratory samples</td>\n",
       "      <td>[{'text': 'Infectious bronchitis (IB) causes significant economic losses in the global poultry industry. Control of infectious bronchitis is hindered by the genetic diversity of the causative agent, infectious bronchitis virus (IBV), which has led to the emergence of several serotypes that lack complete serologic cross-protection. While serotyping by definition requires immunologic characterization, genotyping is an efficient means to identify IBVs detected in samples. Sanger sequencing of the S1 subunit of the spike gene is currently used to genotype IBV; however, the universal S1 PCR was created to work from cultured IBV and it is inefficient at detecting mixed isolates. This paper describes a MinION-based AmpSeq method that genetically typed IBV from clinical samples, including samples with multiple isolates. Total RNA was extracted from fifteen tracheal scrapings and choanal cleft swab samples, randomly reverse transcribed, and PCR amplified using modified S1-targeted primers. Amplicons were barcoded to allow for pooling of samples, processed per manufacturer's instructions into a 1D MinION sequencing library, and sequenced on the MinION. The AmpSeq method detected IBV in 13 of 14 IBV-positive samples.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'AmpSeq accurately detected and genotyped both IBV lineages in three of five samples containing two IBV lineages. Additionally, one sample contained three IBV lineages, and AmpSeq accurately detected two of the three. Strain identification, including detection of different strains from the same lineage, was also possible with this AmpSeq method. The results demonstrate the feasibility of using MinION-based AmpSeq for rapid and accurate identification and lineage typing of IBV from oral swab samples.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Infectious bronchitis virus is an enveloped, pleomorphic gammacoronavirus with an unsegmented, single-stranded, positive-sense, 26-27.8 Kb, RNA genome that encodes the nonstructural polyprotein,1a and 1b, and several structural proteins: spike (S), envelop (E), membrane (M) and nucleocapsid (N). 38,41 In addition, two accessory genes, expressing 3a, 3b and 5a and 5b, respectively, have also been described. 6, 14, 38 The S protein is highly glycosylated and post-translational cleavage leads to two subunits: S1 and S2. 10,48 Besides acting as the viral attachment protein, the S1 protein is a major target of neutralizing antibodies. 7 As with many attachment proteins that are targets of virus-neutralizing antibodies, the S1 subunit is highly diverse with almost 50% of the amino acids differing among IBV serotypes. 2,21,38 Such variation leads to important biological differences between IBV serotypes and the emergence of novel variants. More than 60 serotypes of IBV have been reported, but the most common serotypes of All rights reserved. No reuse allowed without permission.', 'cite_spans': [{'start': 413, 'end': 416, 'text': '14,', 'ref_id': 'BIBREF7'}, {'start': 417, 'end': 419, 'text': '38', 'ref_id': 'BIBREF25'}], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Massachusetts. 16 This genetic diversity leads to emergence of new serotypes and a lack of complete cross-serotype protection by vaccines. 29 The correlation of IBV genotypes and serotypes of IBV has been reported 45 ; therefore, accurate genotypic identification of IBV is an important step to diagnose IBV in clinical respiratory cases, ensure selection of proper IBV vaccines for use in vaccination programs, and to better understand the epidemiology of this global virus.', 'cite_spans': [{'start': 15, 'end': 17, 'text': '16', 'ref_id': 'BIBREF9'}], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'One of the recent comprehensive classification schemes for IBV uses S1 gene sequencebased phylogeny of IBV and identified six genotypes (I-VI), 32 sub-genotypic lineages, and a number of inter-lineage recombinants in global strains of IBV. 44 Among the six genotypes, genotype I (G-I) is the most diverse group of viruses, with 27 unique lineages. 44 As such, sequencing of the S1 subunit provides important information regarding the classification of IBV within samples.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'The genetic classification of IBV has relied on genotype-specific RT-qPCR assays, serotype-specific S1 RT-qPCR 32,40 and/or pan-IBV S1 RT-PCR assays coupled with Sanger sequencing. 8,31,49 Genotype-specific RT-qPCRs are limited to short fragments, which may miss important changes in the S1 (~1.6 kb) 44 outside the short target. Additionally, since the target is short, the primers lie within the variable regions of the S1 and may require a different assay for each genotype. Lastly, the pan-IBV S1 primers 1, 17, 22, 23, 25, 33 , while only requiring one reaction, have relatively low sensitivity and are typically used only on egg-cultured virus, which adds an additional, time consuming step and many diagnostic labs do not have SPF embryos readily available. Lastly, cloning of PCR products has been used to detect multiple IBV sub-populations in the sample 30 ; however, this is inefficient when potentially dealing with multiple (three or four) All rights reserved. No reuse allowed without permission.', 'cite_spans': [{'start': 509, 'end': 511, 'text': '1,', 'ref_id': 'BIBREF0'}, {'start': 512, 'end': 515, 'text': '17,', 'ref_id': 'BIBREF10'}, {'start': 516, 'end': 519, 'text': '22,', 'ref_id': 'BIBREF13'}, {'start': 520, 'end': 523, 'text': '23,', 'ref_id': 'BIBREF14'}, {'start': 524, 'end': 527, 'text': '25,', 'ref_id': 'BIBREF16'}, {'start': 528, 'end': 530, 'text': '33', 'ref_id': 'BIBREF20'}], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint IBV types. As such, the development of a pan-IBV sequencing method to rapidly determine the genotype(s) in samples would aid in lineage typing and tracking of IBV genotypes circulating in poultry flocks.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Recently, third generation sequencing technology has been used for detection of viral nucleotides and sequencing ultra-long DNA molecules. 34 The MinION Nanopore sequencer, a new DNA sequencing technology that allows for rapid, in-house, real-time technique detection and differentiation of IBV lineages, may be cost effective and useful in the field. 19 Ampliconbased sequencing has also been used to amplify specific regions of Newcastle disease 3,13 , infectious laryngotracheitis virus 39 Zika 34 , Ebola 35 and avian influenza viruses 46 , by simple RT-PCR and then sequencing on the MinION device. Real-time data analysis, the lack of significant start-up cost investment or maintenance expenses, simultaneous and sequential multiplexing unique to MinION, and the ability to sequence long DNA molecules so that primers are in conserved regions while the product contains the variable region are the features that make this technology highly feasible to be used for disease diagnostics.', 'cite_spans': [{'start': 139, 'end': 141, 'text': '34', 'ref_id': 'BIBREF21'}], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Accurate identification of IBV genotypes from samples, including detection of multiple isolates from a single sample, is crucial to respiratory disease diagnosis, selection of appropriate IBV vaccines, and epidemiologic studies. Therefore, the aim of the current study is to create a single, amplicon-based protocol to sequence IBV-S1 gene and develop a sequence analysis workflow to identify IBV isolates from clinical swab samples. This method provides a useful assay for IBV and a model for the development of future amplicon sequencing (AmpSeq) based assays.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'All rights reserved. No reuse allowed without permission.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'A universal S1 primer set, 27 tailed with the MinION universal adapter sequence of 22 nucleotides (underlined, Table 2 ) to allow barcoding of amplicons, was used for targeted amplification of the IBV S1 gene for IBV. The PCR reaction mixture (Expand High Fidelity PCR system, Roche Diagnostics, Mannheim, Germany) was composed of 10 µl of cDNA, 1 µl of 10 µM forward primer, 1 µl of 10 µM reverse primer, 2.5 µl of 10× Expand high fidelity buffer, 1 µl of Expand High Fidelity Enzyme Mix, and 1 µl of 10mM dNTPs, to a final volume to 25 µl with 9 µl of nuclease-free water. The following thermocycling conditions were used for amplicon synthesis: denaturation at 95 °C for 2 min; 30 cycles of 94 °C for 1 min, 45 °C for 2 min, and 72 °C for 2 minutes, and final extension at 72 °C for 10 min. See Table 2 for details about all three primer sets used in this study. PCR products were electrophoresed in 1.5% agarose with SYBR Safe DNA gel stain (Invitrogen) for visual evaluation. Amplified DNA was purified by Agencourt AMPure XP beads (Beckman Coulter, USA) at 1.6:1 (volumetric bead:DNA) and quantified using the dsDNA High Sensitivity Assay Kit on a Qubit® Fluorimeter 3.0.', 'cite_spans': [], 'ref_spans': [{'start': 111, 'end': 118, 'text': 'Table 2', 'ref_id': None}, {'start': 798, 'end': 805, 'text': 'Table 2', 'ref_id': None}], 'section': 'Abstract'}, {'text': 'The amplicons obtained from the tailed S1 primer set were then used to prepare MinIONcompatible DNA libraries. Briefly, each of the amplicons were diluted to 0.5 nM and amplified using barcoding primers (1D PCR Barcoding Amplicon Kit, Oxford Nanopore Technologies, [ONT], UK) and LongAmp Taq 2X Master Mix (New England Biolabs, MA, USA) with the following conditions: 95 °C for 3 min; 15 cycles of 95 °C for 15 seconds, 62 °C for 15 seconds, 65 °C for 80 seconds; and 65 °C for 80 seconds. The barcoded amplicons were bead purified (1:1.4; bead:solution), pooled (Run1; pool of 10 samples, Run2; pool of 5 samples) into a single All rights reserved. No reuse allowed without permission.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Centrifuge 24 , Kraken 50 , QIIME 5 and Mothur 37 have been used to identify and profile microbial species; however, the high error rate 20 of Nanopore reads translates to poor classification accuracy for many of these tools. Alternatively, de novo 3,26 or reference-based 18 assembly methods have been used in MinION and other deep sequencing platforms. Using a strategy similar to other unbiased laboratory tests (e.g., standard bacterial cultures), an approach was developed to maximize usage of reads (i.e., reads are not discarded based on pre-set length or abundance requirements, similar to how a single colony may be interpreted as a significant result). This approach uses read-based classification against a database containing an equal number of representative sequences per lineage to detect and classify the reads based on their lineage, before conducting read-based classification to the isolate level by using all available IBV sequences in a separate database. Finally, lineage-clustered isolate-level read assignments are interpreted by a veterinary diagnostician to result in reads available for consensus building.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'The use of a final consensus alignment helps to overcome the individual error rate of MinION sequencing. 3 Similarly, the absence of predetermined metrics used in de novo assembly allows for the informed decision as to how many consensus sequences to build, a bioinformatics problem when dealing with clinical cases that can contain more than one isolate (e.g., similar to how there is not a predetermined number of significant bacterial colonies). Confirmatory followup tests (e.g., RT-PCR) may be needed when dealing with low number of reads in clinical samples; however, future testing of this new technology will allow for creating standards for such confirmatory testing.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Taken together, these results suggest the application of MinION-based AmpSeq, specifically for detecting IBV isolates from clinical samples within a few days, compared to several days to weeks and multiple diagnostic assays to culture and detect multiple IBV All rights reserved. No reuse allowed without permission.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>Infectious bronchitis (IB), which is caused by infectious bronchitis virus (IBV), is one of the most important diseases of poultry, causing severe economic losses worldwide. 8 Clinical signs of disease are diverse and include respiratory distress, severe ocular discharge, poor body weight gain, decreased egg production, flushing (renal disease), and occasionally mortality in chickens. 7 IB is often complicated by secondary bacterial (e.g., E. coli, Mycoplasma sp.) and viral infections (e.g., avian influenza virus, Newcastle disease virus, and infectious laryngotracheitis virus). 43 Lack of cross protection among IBV serotypes is a challenge to controlling IB 15 ; therefore, control of IB relies heavily on serotype-specific live attenuated vaccines. 8 Collectively, the presence of multiple IBVs in a single sample, emergence of variant IBVs, and the high genetic diversity of IBV can complicate the diagnosis of IB and illustrate the need for enhanced diagnostics. 15 \\nClinical swab samples (n = 15) were obtained from samples submitted to the Poultry Diagnostic &amp; Research Center, University of Georgia. Detailed information about the samples is provided in Table 1 .\\nTotal RNA was extracted from each of the swab samples using QIAamp Viral RNA Mini kit (Qiagen, Hilden, Germany) as per manufacturer's instructions, aliquoted, and stored at −80 ºC until further use. These samples were previously tested for IBV and GA08 serotype of IBV.\\nBriefly, a pan IBV RT-qPCR assay 4 was used to detect IBV in general and then IBV strain specific RT-qPCR assay 36 was used to detect GA08 in the samples using the AgPath ID, One step RT-PCR kit (Thermo Fisher Scientific, Waltham, MA, USA) on the Applied Biosystems 7500 FAST following the previously described protocols.\\nFor MinION cDNA synthesis, RNA was extracted from 500 µl of each clinical sample using Trizol-LS (Thermo Fisher Scientific, Waltham, MA, USA), per manufacturer's directions.\\nA reaction mixture of 8 µl of total RNA, 1 µl of random primers and 1 µl of dNTPs was incubated at 65 °C for 5 min, chilled on ice for at least 1 min followed by addition of 10 µl of cDNA synthesis mix including SuperScript III (Thermo Fisher Scientific, Waltham, MA, USA) according to the manufacture's instruction. The reaction was incubated at 25 °C for 10 min, then at 50 °C for 50 min for cDNA synthesis. The reaction was terminated at 85 °C for 5 min, and then chilled on ice. To remove residual RNA, the cDNA solution was incubated with RNase H at 37 °C for 20 min. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. sequencing protocol was initiated using the MinKNOW software v.5.12. Sequencing was allowed to continue for 2 hours until 42,940 reads were obtained for Run1 and 156,000 reads were obtained for Run 2 in 6 hours.\\nFirst, a lineage-typing database, containing 32 IBV S1 gene sequences (one sequence from each of 32 lineages in the 6 genotypes 44 ) and the chicken genome (GCF_000002315.4_Gallus_gallus-5.0), was constructed (referred to as \"IBV-lineage-typing database\" in this study). A second database was constructed with all the Avian Coronaviruses S1 sequences (n = 7328) available in NCBI (as of: 09-08-2017) and the chicken genome (referred to as \"All-IBV database\" in this study). Prior to database construction, all sequences were dustmasked (NCBI C++ ToolKit), and each isolate was assigned a unique taxonomy ID as a species hierarchically under the genus Gammacoronavirus to allow easy sorting. The local BLAST databases were compiled using default settings. The IBV-lineage-typing database was All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint used to assign IBV lineages to each read, as appropriate. The All-IBV database was used to assign IBV isolates taxonomic ID to each of IBV reads from within IBV lineage read clusters.\\nThe IBV-lineage-typing database, which contains only one sequence per lineage, is required as centrifuge divides the score for any given read by the number of hits that have an equal score.\\nSince some lineages are overrepresented in GenBank (e.g., Mass strains), this results in the scores of those reads (e.g., Mass reads) being divided by a large number, effectively reducing the score for any single alignment. However, because there is only one read per lineage in the IBVlineage-typing database, it has insufficient diversity to determine if there is more than one strain of the same lineage present in a sample. Thus, the All-IBV database provides the diversity required for that analysis.\\nA schematic diagram of the workflow of MinION data analysis is presented in Fig. 1 . For Nanopore sequencing data, pre-processing steps were performed to prepare data for downstream analysis. Briefly, Nanopore reads (FAST5) were basecalled using Albacore v 2.02 (https://github.com/Albacore/albacore) with following parameters (read_fast5_basecaller.py -i /Input_reads.fast5 --recursive -t 4 -s /Output_files --flowcell FLO-MIN107 --kit SQK-LSK108o fastq). Porechop (https://github.com/rrwick/Porechop) was used for adapter trimming, (default setting), barcode-based demultiplexing (default settings), and to trim an additional 21 nucleotides, representing the S1 primer sequences (porechop -i Input_file.fastq --extra_end_trim 21 -b ./output_demultiplexed files.fastq). After barcode and adapter removal, reads were analyzed with a script-based, 2-step data analysis protocol, which includes Centrifuge-kreport as taxonomic read classifier 24 using the sequences in the above mentioned BLAST databases.\\nBriefly, basecalled reads (FASTQ) from individual barcoded samples were used as an input. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint First, the basecalled reads were aligned to the IBV-lineage-typing database using BLASTn and reads were clustered based on the read sequence alignment to the respective prototype sequence of IBV lineage. These read clusters were used to interpret presence of IBV genotypes and lineages in the samples. For isolate level identification of IBV, the lineage-based read clusters were individually aligned to the All-IBV database, which produced sub-clusters of reads. Each of these read sub-clusters potentially represents a different isolate and was further used for interpretation. Knowing that MinION sequencing has a high sequencing error rate in individual reads, further steps were added in the data analysis algorithm to obtain a more accurate consensus sequence. Therefore, these read sub-clusters were mapped, using Geneious mapper in Geneious software v11.1.3. (Biomatters, Ltd, Auckland, New Zealand), to the IBV-lineagetyping database (a reference FASTA file) to obtain consensus sequences from each sub-cluster.\\nA minimum threshold for number of reads per consensus sequence was not set as in two samples (sample #4 in Run1 and sample #12 in Run 2), only 1 and 5 IBV reads were obtained, respectively; therefore, consensus sequences were built from all the available IBV reads per subcluster. The consensus sequences were compared to GenBank using BLASTn. To select the \"top hit\" from BLASTn output, sequence search results were ordered by \"sequence identity\" and then sequence alignments were evaluated for \"minimum mismatches\" and \"coverage\" of query or subject sequence. All isolates with the highest query or subject coverage and the fewest mismatches were used as \"top hit\" in the final results.\\nFor Sanger sequencing amplicons synthesis, cDNA from MinION library preparation was amplified using the following primer sets (Table 2) . Firstly, a primer set based on Connecticut sequence NCBI FJ904716.1 was used to amplify Genotype I-lineage 1 viruses (e.g., Connecticut All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint and Massachusetts serotypes). Secondly, a primer set based on GA98 sequence NCBI AF274439.1 was used to amplify Genotype IV-lineage 1 viruses (e.g., Georgia 1998 and Delaware 072 serotypes). Primer sets were designed using NCBI Primer-BLAST. 51 The PCR reaction mixture is composed of 10 µl of cDNA, 0.5 µl of 10 µM forward primer, 0.5 µl of 10 µM reverse primer, 2.5 µl of 10× buffer, 0.5 µl of enzyme mix, and 0.5 µl of 10mM dNTPs, then made the final volume to 25 µl with nuclease-free water. The following thermocycling conditions were used for amplicon synthesis: denaturation at 95 °C for 1 min, followed by \\nFor Sanger sequencing data, the chromatogram from each of the samples was manually checked and primer sequences were trimmed in MEGA 6.0. 42 Forward and reverse sequences from multiple clones were aligned using MEGA 6.0 software and consensus sequences were All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint compared to GenBank using Basic Local Alignment Tool (BLAST) (https://blast.ncbi.nlm.nih.gov/) (as of: 12-03-2018) and the top hit from BLASTn output was selected, as described above for MinION sequencing data, to identify the IBV isolate in samples.\\nA pairwise nucleotide identity comparison on the partial S1 sequences of IBVs from each of the samples sequenced with Sanger sequencing and MinION sequencing was performed.\\nBriefly, the final AmpSeq consensus sequences from identical IBV types in each of the sample were aligned using ClustalW in MEGA6 42 and this alignment was used for pairwise nucleotide identity using MEGA6. 42 All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint\\nFive samples from Run 1 were tested with a pan-IBV RT-qPCR assay 4 , and four of those were positive for IBV. Samples with pathogenic IBV strains (Run 2, n = 5) were tested with the pan-IBV RT-qPCR and all five samples were IBV positive (Table 1) . Next, a GA08 serotype specific primer set (in house validated set of primers used for diagnostics) was used on all samples from Run 1 (including five samples that were previously tested by the pan-IBV PCR) and four of 10 samples were positive. Sample #3 and #4, which were positive with the pan-IBV RT-qPCR assay, were negative with GA08 specific RT-qPCR assay indicating that these samples contained IBV serotypes other than GA08 and required further testing. Sample #5 tested negative with both assays (Table 1) .\\nA total of 10 clinical swab samples (Run 1) were processed for Sanger sequencing to confirm the presence of IBV lineages. Based on the MinION results, primers targeting GIV-L1 and GI-L1 (GA98 and Conn and primer sets, respectively) were created. The PCR products were cloned and multiple (6-24) clones from each sample were submitted for Sanger sequencing and results are summarized in Table 3 . Using GIV-L1 serotype primers showed that 9 out of 10 samples were positive with 1 sample (#5) negative for IBV (which is consistent with the pan-IBV RT-qPCR results for this sample). Consensus sequences from multiple clones obtained by using GIV-L1 primers showed the presence of GA98 (GA/A9dvaccinated) in 7 of 9 IBV-positive samples. DE072 was detected in samples #1 and #9 as the lone GIV-1 isolate but was also detected in three samples (#2, and #10) that also contained GA98 (Table 4 ). By using the GI-L1 primer set, which amplified Conn and Mass type IBVs, 3 of 9 IBV-positive samples were All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. (Table 4 ). Based on the two conventional RT-PCRs and the GA08 RT-qPCR, it was determined that only samples #8 and #9 contained a single IBV isolate, while the other 7 samples were positive for 2-4 types of IBV (Table 3) .\\nPCR amplicons obtained directly from 10 clinical swab samples with vaccine IBV serotypes were barcoded, pooled together and sequenced on MinION device (Run 1). A total of 38,661 reads were successfully basecalled from the entire sequencing run (total reads = 42,940).\\nAfter demultiplexing, reads per barcode ranged from 831 to 4114. A total of 14,845 reads were not assigned to any of the used barcodes and 128 reads were discarded due to middle adapters.\\nThe Nanopore reads were queried against IBV-lineage-typing database to determine if the samples contained IBV (Figure 1 ). This AmpSeq protocol detected IBV reads in 8 of 10 samples.\\nThe number of IBV reads per sample (IBV positive sample) ranged from 56-944. The MinIONnegative samples included sample #5, which was consistently negative with RT-qPCR and RT-PCR assays (and thus interpreted as IBV negative), and sample #1, which had the highest Cq values in the pan-IBV RT-qPCR (Table 1) .\\nSequencing data was further analyzed to determine if multiple IBV genotypes or lineages could be detected. Lineage-1 and lineage-27 from Genotype I (GI-L1, GI-L27), and lineage-1 from Genotype IV (GIV-L1) were the detected lineages in the samples. Results from individual RT-qPCR assay for GA08 (GI-L27), and RT-PCR assays for GIV-L1and GI-L1 confirmed the presence of multiple IBV genotypes and lineages in 6 of the 9 IBV-positive samples (#1, #2, #3, #4, #6, #7) and AmpSeq results confirmed multiple genotypes and lineages in 4 (#2, #4, #6, #7) of those 6 samples (AmpSeq failed to detect any IBV in #1, and failed to detect GIV-L1 in #3). All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint\\nThe presence of a single lineage was confirmed by PCR-based assays and AmpSeq in three samples (#8, # 9 and #10) ( Table 3) .\\nIt was observed that the single read cluster composed of less than 5 reads (Run1, sample #4, GA98) yielded a poor-quality consensus sequence, consistent with the known individual read error rate of MinION sequencing. High-quality consensus sequences were obtained from the other read clusters (&gt;5 reads per cluster). Each of the obtained consensus sequences were compared to GenBank databases using BLAST, which revealed &gt;99% sequence identity to respective IBV isolates Table 4 ). As described in the Sanger Sequencing section, two samples (#8 and #9) contained only one IBV isolate per the non-MinION assays, and the AmpSeq results were consistent with these findings. Non-MinION assays showed that seven samples contained two or more isolates per sample. Six samples (#1, #3, #4, #6, #7, and #10) contained two isolates per sample (Table 4 ). Of those six samples, AmpSeq detected both isolates in four samples (#4, #6, #7, and #10), one isolate in one sample (#3), and no IBV in one sample (#1; as mentioned above, sample #1 was the sample with the highest Cq value). Of note, sample #10 contained two isolates from the same lineage and AmpSeq was able to identify both isolates within this sample.\\nFinally, the seventh multi-isolate sample (#2) contained four isolates per the non-MinION results, and the AmpSeq detected two of those four isolates. These data show that a co-infection of multiple IBV lineages existed in the above-mentioned samples but a single RT-qPCR, or Sanger sequencing of a single clone may not have detected these co-infecting IBV lineages and required multiple assays to detect all the IBV types. However, the AmpSeq protocol accurately detected multiple IBV lineages in 4 of 7 samples, with partial detection in two of the remaining three samples (Table 4 ). In samples which had the same IBV type, Blast search of consensus All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint sequences obtained from AmpSeq and Sanger sequencing identified the same (or highly related isolate for sample #6 [GA98] and sample #9) in the NCBI nt database as per parameters described in the methods section.\\nTo evaluate the utility of this protocol on clinical swab samples with pathogenic IBV variants, IBV-S1 gene was amplified directly from five clinical tracheal scrapings and the PCR amplicons were used to create MinION libraries. A total of 146,540 reads were successfully basecalled from the entire sequencing run (total reads = 156,000). After demultiplexing, reads per barcode ranged from 4,285 to 41,131. A total of 24,297 reads were not assigned to any of the used barcodes and 8,109 reads were discarded due to middle adapters in the basecalled reads.\\nReal-time analysis of MinION data, which was obtained within 10 minutes of the sequencing, was sufficient for detection of IBV. However, the sequencing data obtained from entire sequencing run was processed with the same protocol as described above. This AmpSeq protocol detected IBV reads in all of the 5 tested samples using all basecalled reads. The number of IBV reads per sample ranged from 5-4,956. Additionally, the sequencing data analysis showed that the IBV reads belonged to GI-17 lineage and typed the isolate as DMV1639. After MinION sequencing results, these samples were later tested to confirm the presence of IBV variant by MDL_DMV1639 IBV variant specific RT-qPCR assay. All 5 samples were positive for IBV MDL_DMV1639 variant of IBV (Table 5) .\\nIn all of the samples where a matching IBV genotype/lineage was identified by both AmpSeq and Sanger sequencing, the AmpSeq data showed high concordance with Sanger sequencing data, In one sample (#4), only a single MinION read of GA98 was detected by All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint sequencing, consistent with the reported single-read accuracy of MinION sequencing. All other samples showed 99-100% pairwise identity across the partial S1 fragment generated by Sanger Sequencing. (Table 4 ).\\nAll rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint\\nThis study describes a sequencing approach for rapid and accurate identification of IBV genotypes/lineages utilizing the MinION Nanopore sequencer. Additionally, a sequence analysis workflow is described here. The workflow analysis is designed to deal with the multiple, highly similar sequences contained within GenBank that are due to numerous sequencing experiments of IBV (especially vaccine viruses). This protocol demonstrates the feasibility of real-time-based sequencing to provide isolate-level resolution of IBV directly from clinical swab samples, including samples that contain multiple isolates.\\nThe proper diagnosis of IBV as the cause of clinical respiratory disease is contingent on virus typing and differentiating live vaccine viruses from field strains. 11 Previously, it was reported that IBV genotypes correlate well with the serotypes of IBV 45 ; therefore, accurate genotypic identification of IBV will be useful to diagnose vaccine and variant viruses in clinical samples. The rapid pan-IBV and serotype specific RT-qPCR assays 4,23,36 have been used for serotype identification; however, positive results from RT-qPCR is insufficient to determine the IBV genotype and obtain isolate-level resolution of IBV; thus, sequence analysis of IBV-S1 gene is required. Previously, use of partial S1 gene sequences (450 bp) to type IBV has been described. 29 However, increasing the length of sequenced S1 gene (~1620 bp) results in more data to be used for genotyping as more of the hypervariable region is covered. 17, 27, 45 For example, the AmpSeq protocol was able to differentiate two genetically very similar (99.5% at S1 gene of IBV genome) but serotypically different GIV-L1 isolates, DE072 and GA98, within a single sample. Thus, the detection of highly diverse IBV genotypes, via the S1 subunit, with a single and rapid sequencing protocol is desirable and the ability to detect multiple isolates in a single sample would improve IBV detection in clinical samples. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint Currently, pan-IBV RT-qPCR is used to rapidly detect IBV from clinical samples for screening purposes, 4 but this requires additional serotype-specific RT-qPCRs to genotype positive samples, including samples containing more than one genotype. Additionally, if unidentified IBVs are in the sample, then current detection and characterization may also require egg culture followed by various PCRs. 8 The AmpSeq method described in this report detected IBV in 13 of 14 IBV-positive and detected all the mixed isolates in four of seven samples containing two or more isolates. While mixed isolates were not 100% identified (0/2 in #1, 2/4 in #2, and ½ in #3) in three samples with the AmpSeq method, detection of those mixed isolates required several non-AmpSeq assays. Although, there is room for improving this new AmpSeq method, it represents a promising, single-step assay that can be used without egg culture. Thus, it is another tool for the detection of IBV, especially in cases when multiple isolates may be present or when genotyping is especially important.\\nOne area that is problematic for a test such as AmpSeq is to detect all the isolates that may be present. While AmpSeq did detect all isolates in 4 samples, in 3 cases not all of the isolates were detected. A potential explanation to partial detection of multiple IBV genotypes could be the relative abundance of IBV genotypes in these clinical samples. It could also be that amplification of these IBV genotypes by serotype-specific RT-qPCR assays is more efficient due to small targeted fragment size (e.g., 120 bp for PCR and ~1600 bp for AmpSeq) and better primer alignment to the target (e.g., degenerate bases are used in the S1 primers used for AmpSeq). One complicating factor of the current AmpSeq protocol is that the IBV target sequence 27 was not originally designed for high specificity, especially from clinical samples. As such, a high proportion of total reads were non-IBV reads (e.g., often mapping to the chicken genome, data not shown), consistent with the extra bands visible in the original report for these All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint target sequences. 28 Additionally, it is possible that certain genotypes are better complemented to the S1 primers than others and may outcompete those isolates for amplification in the AmpSeq protocol. Lastly, increasing the total number of reads collected by AmpSeq may improve the ability to detect all genotypes in sample by AmpSeq. It is possible to allow the sequencing to continue longer to obtain more reads per sample. Overall, AmpSeq is a feasible test for IBV characterization, and work is ongoing to improve this new type of assay.\\nFor example, cost and time efficiency of a sequencing protocol can be improved through multiplexing of more samples in a single sequencing run. 3, 47 In this study, samples (Run 1 = 10, Run 2 = 5) were simultaneously multiplexed (i.e., pooled and then sequenced in one run) while maintaining IBV genotyping from data collected. Since the MinION flow cells were not exhausted, and can be washed and re-used, the AmpSeq method also has the potential for sequential multiplexing. This would decrease the need to hold samples for weeks while waiting for the cost-optimal number of samples for simultaneous multiplexing. Alternatively, if detection of all isolates within a given sample is preferred, the sequencing can be run longer to increase sensitivity. The single protocol nature of AmpSeq, the ability to obtain S1 gene sequence results, real-time data analysis, and flexibility of testing design makes MinION-based AmpSeq a viable sequencing protocol for typing of IBV isolates.\\nThe advent of real-time, in-house, third generation sequencing represents a transformative opportunity for diagnostic laboratories, by offering the ability to more fully characterize PCR reactions beyond confirming amplicon size (e.g., routine electrophoresis), Sanger sequencing RT-PCR products, or by confirming a partial sequence through probe hybridization (e.g., probe-based qPCR). However, the interpretation of such large data sets represents a challenge to veterinary diagnosticians. Read-based classification software such as genotypes from single sample. Thus, the MinION-based AmpSeq coupled with the data analysis workflow for identification, differentiation, and accurate prediction of IBV genotypes from clinical swab samples can be used as an adjunct to other established rapid diagnostic assays 9,12 until extensive testing of this protocol is done to improve and validate AmpSeq for IBV diagnostics. Furthermore, AmpSeq-based assays can be, and are being applied to other viral pathogens, 3,37 demonstrating the power and utility of this method in the age of molecular diagnostics. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint \\n</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                  title  \\\n",
       "0  The RNA pseudoknots in foot-and-mouth disease virus are dispensable for genome replication but essential for the production of infectious virus. 2 3   \n",
       "1  Analysis Title: Regaining perspective on SARS-CoV-2 molecular tracing and its implications                                                             \n",
       "2  Healthcare-resource-adjusted vulnerabilities towards the 2019-nCoV epidemic across China                                                               \n",
       "3  CHEER: hierarCHical taxonomic classification for viral mEtagEnomic data via deep leaRning                                                              \n",
       "4  Real-time, MinION-based, amplicon sequencing for lineage typing of infectious bronchitis virus from upper respiratory samples                          \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          abstract  \\\n",
       "0  [{'text': 'word count: 194 22 Text word count: 5168 23 24 25 author/funder. All rights reserved. No reuse allowed without permission. Abstract 27 The positive stranded RNA genomes of picornaviruses comprise a single large open reading 28 frame flanked by 5′ and 3′ untranslated regions (UTRs). Foot-and-mouth disease virus (FMDV) 29 has an unusually large 5′ UTR (1.3 kb) containing five structural domains. These include the 30 internal ribosome entry site (IRES), which facilitates initiation of translation, and the cis-acting 31 replication element (cre). Less well characterised structures are a 5′ terminal 360 nucleotide 32 stem-loop, a variable length poly-C-tract of approximately 100-200 nucleotides and a series of 33 two to four tandemly repeated pseudoknots (PKs). We investigated the structures of the PKs 34 by selective 2′ hydroxyl acetylation analysed by primer extension (SHAPE) analysis and 35 determined their contribution to genome replication by mutation and deletion experiments. 36 SHAPE and mutation experiments confirmed the importance of the previously predicted PK 37 structures for their function. Deletion experiments showed that although PKs are not essential 38', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'for replication, they provide genomes with a competitive advantage. However, although 39 replicons and full-length genomes lacking all PKs were replication competent, no infectious 40 virus was rescued from genomes containing less than one PK copy. This is consistent with our 41 earlier report describing the presence of putative packaging signals in the PK region. 42 43 author/funder. All rights reserved. No reuse allowed without permission.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
       "1  [{'text': 'During the past three months, a new coronavirus (SARS-CoV-2) epidemic has been growing exponentially, affecting over 100 thousand people worldwide, and causing enormous distress to economies and societies of affected countries. A plethora of analyses based on viral sequences has already been published, in scientific journals as well as through non-peer reviewed channels, to investigate SARS-CoV-2 genetic heterogeneity and spatiotemporal dissemination. We examined all full genome sequences currently available to assess the presence of sufficient information for reliable phylogenetic and phylogeographic studies. Our analysis clearly shows severe limitations in the present data, in light of which any finding should be considered, at the very best, preliminary and hypothesis-generating. Hence the need for avoiding stigmatization based on partial information, and for continuing concerted efforts to increase number and quality of the sequences required for robust tracing of the epidemic.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': '. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'is the (which was not peer-reviewed) The copyright holder for this preprint .', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
       "2  []                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
       "3  [{'text': 'The fast accumulation of viral metagenomic data has contributed significantly to new RNA virus discovery. However, the short read size, complex composition, and large data size can all make taxonomic analysis difficult. In particular, commonly used alignment-based methods are not ideal choices for detecting new viral species. In this work, we present a novel hierarchical classification model named CHEER, which can conduct read-level taxonomic classification from order to genus for new species. By combining k-mer embedding-based encoding, hierarchically organized CNNs, and carefully trained rejection layer, CHEER is able to assign correct taxonomic labels for reads from new species. We tested CHEER on both simulated and real sequencing data. The results show that CHEER can achieve higher accuracy than popular alignmentbased and alignment-free taxonomic assignment tools. The source code, scripts, and pre-trained parameters for CHEER are available via GitHub: https://github.com/KennthShang/CHEER.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
       "4  [{'text': 'Infectious bronchitis (IB) causes significant economic losses in the global poultry industry. Control of infectious bronchitis is hindered by the genetic diversity of the causative agent, infectious bronchitis virus (IBV), which has led to the emergence of several serotypes that lack complete serologic cross-protection. While serotyping by definition requires immunologic characterization, genotyping is an efficient means to identify IBVs detected in samples. Sanger sequencing of the S1 subunit of the spike gene is currently used to genotype IBV; however, the universal S1 PCR was created to work from cultured IBV and it is inefficient at detecting mixed isolates. This paper describes a MinION-based AmpSeq method that genetically typed IBV from clinical samples, including samples with multiple isolates. Total RNA was extracted from fifteen tracheal scrapings and choanal cleft swab samples, randomly reverse transcribed, and PCR amplified using modified S1-targeted primers. Amplicons were barcoded to allow for pooling of samples, processed per manufacturer's instructions into a 1D MinION sequencing library, and sequenced on the MinION. The AmpSeq method detected IBV in 13 of 14 IBV-positive samples.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'AmpSeq accurately detected and genotyped both IBV lineages in three of five samples containing two IBV lineages. Additionally, one sample contained three IBV lineages, and AmpSeq accurately detected two of the three. Strain identification, including detection of different strains from the same lineage, was also possible with this AmpSeq method. The results demonstrate the feasibility of using MinION-based AmpSeq for rapid and accurate identification and lineage typing of IBV from oral swab samples.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Infectious bronchitis virus is an enveloped, pleomorphic gammacoronavirus with an unsegmented, single-stranded, positive-sense, 26-27.8 Kb, RNA genome that encodes the nonstructural polyprotein,1a and 1b, and several structural proteins: spike (S), envelop (E), membrane (M) and nucleocapsid (N). 38,41 In addition, two accessory genes, expressing 3a, 3b and 5a and 5b, respectively, have also been described. 6, 14, 38 The S protein is highly glycosylated and post-translational cleavage leads to two subunits: S1 and S2. 10,48 Besides acting as the viral attachment protein, the S1 protein is a major target of neutralizing antibodies. 7 As with many attachment proteins that are targets of virus-neutralizing antibodies, the S1 subunit is highly diverse with almost 50% of the amino acids differing among IBV serotypes. 2,21,38 Such variation leads to important biological differences between IBV serotypes and the emergence of novel variants. More than 60 serotypes of IBV have been reported, but the most common serotypes of All rights reserved. No reuse allowed without permission.', 'cite_spans': [{'start': 413, 'end': 416, 'text': '14,', 'ref_id': 'BIBREF7'}, {'start': 417, 'end': 419, 'text': '38', 'ref_id': 'BIBREF25'}], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Massachusetts. 16 This genetic diversity leads to emergence of new serotypes and a lack of complete cross-serotype protection by vaccines. 29 The correlation of IBV genotypes and serotypes of IBV has been reported 45 ; therefore, accurate genotypic identification of IBV is an important step to diagnose IBV in clinical respiratory cases, ensure selection of proper IBV vaccines for use in vaccination programs, and to better understand the epidemiology of this global virus.', 'cite_spans': [{'start': 15, 'end': 17, 'text': '16', 'ref_id': 'BIBREF9'}], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'One of the recent comprehensive classification schemes for IBV uses S1 gene sequencebased phylogeny of IBV and identified six genotypes (I-VI), 32 sub-genotypic lineages, and a number of inter-lineage recombinants in global strains of IBV. 44 Among the six genotypes, genotype I (G-I) is the most diverse group of viruses, with 27 unique lineages. 44 As such, sequencing of the S1 subunit provides important information regarding the classification of IBV within samples.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'The genetic classification of IBV has relied on genotype-specific RT-qPCR assays, serotype-specific S1 RT-qPCR 32,40 and/or pan-IBV S1 RT-PCR assays coupled with Sanger sequencing. 8,31,49 Genotype-specific RT-qPCRs are limited to short fragments, which may miss important changes in the S1 (~1.6 kb) 44 outside the short target. Additionally, since the target is short, the primers lie within the variable regions of the S1 and may require a different assay for each genotype. Lastly, the pan-IBV S1 primers 1, 17, 22, 23, 25, 33 , while only requiring one reaction, have relatively low sensitivity and are typically used only on egg-cultured virus, which adds an additional, time consuming step and many diagnostic labs do not have SPF embryos readily available. Lastly, cloning of PCR products has been used to detect multiple IBV sub-populations in the sample 30 ; however, this is inefficient when potentially dealing with multiple (three or four) All rights reserved. No reuse allowed without permission.', 'cite_spans': [{'start': 509, 'end': 511, 'text': '1,', 'ref_id': 'BIBREF0'}, {'start': 512, 'end': 515, 'text': '17,', 'ref_id': 'BIBREF10'}, {'start': 516, 'end': 519, 'text': '22,', 'ref_id': 'BIBREF13'}, {'start': 520, 'end': 523, 'text': '23,', 'ref_id': 'BIBREF14'}, {'start': 524, 'end': 527, 'text': '25,', 'ref_id': 'BIBREF16'}, {'start': 528, 'end': 530, 'text': '33', 'ref_id': 'BIBREF20'}], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint IBV types. As such, the development of a pan-IBV sequencing method to rapidly determine the genotype(s) in samples would aid in lineage typing and tracking of IBV genotypes circulating in poultry flocks.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Recently, third generation sequencing technology has been used for detection of viral nucleotides and sequencing ultra-long DNA molecules. 34 The MinION Nanopore sequencer, a new DNA sequencing technology that allows for rapid, in-house, real-time technique detection and differentiation of IBV lineages, may be cost effective and useful in the field. 19 Ampliconbased sequencing has also been used to amplify specific regions of Newcastle disease 3,13 , infectious laryngotracheitis virus 39 Zika 34 , Ebola 35 and avian influenza viruses 46 , by simple RT-PCR and then sequencing on the MinION device. Real-time data analysis, the lack of significant start-up cost investment or maintenance expenses, simultaneous and sequential multiplexing unique to MinION, and the ability to sequence long DNA molecules so that primers are in conserved regions while the product contains the variable region are the features that make this technology highly feasible to be used for disease diagnostics.', 'cite_spans': [{'start': 139, 'end': 141, 'text': '34', 'ref_id': 'BIBREF21'}], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Accurate identification of IBV genotypes from samples, including detection of multiple isolates from a single sample, is crucial to respiratory disease diagnosis, selection of appropriate IBV vaccines, and epidemiologic studies. Therefore, the aim of the current study is to create a single, amplicon-based protocol to sequence IBV-S1 gene and develop a sequence analysis workflow to identify IBV isolates from clinical swab samples. This method provides a useful assay for IBV and a model for the development of future amplicon sequencing (AmpSeq) based assays.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'All rights reserved. No reuse allowed without permission.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'A universal S1 primer set, 27 tailed with the MinION universal adapter sequence of 22 nucleotides (underlined, Table 2 ) to allow barcoding of amplicons, was used for targeted amplification of the IBV S1 gene for IBV. The PCR reaction mixture (Expand High Fidelity PCR system, Roche Diagnostics, Mannheim, Germany) was composed of 10 µl of cDNA, 1 µl of 10 µM forward primer, 1 µl of 10 µM reverse primer, 2.5 µl of 10× Expand high fidelity buffer, 1 µl of Expand High Fidelity Enzyme Mix, and 1 µl of 10mM dNTPs, to a final volume to 25 µl with 9 µl of nuclease-free water. The following thermocycling conditions were used for amplicon synthesis: denaturation at 95 °C for 2 min; 30 cycles of 94 °C for 1 min, 45 °C for 2 min, and 72 °C for 2 minutes, and final extension at 72 °C for 10 min. See Table 2 for details about all three primer sets used in this study. PCR products were electrophoresed in 1.5% agarose with SYBR Safe DNA gel stain (Invitrogen) for visual evaluation. Amplified DNA was purified by Agencourt AMPure XP beads (Beckman Coulter, USA) at 1.6:1 (volumetric bead:DNA) and quantified using the dsDNA High Sensitivity Assay Kit on a Qubit® Fluorimeter 3.0.', 'cite_spans': [], 'ref_spans': [{'start': 111, 'end': 118, 'text': 'Table 2', 'ref_id': None}, {'start': 798, 'end': 805, 'text': 'Table 2', 'ref_id': None}], 'section': 'Abstract'}, {'text': 'The amplicons obtained from the tailed S1 primer set were then used to prepare MinIONcompatible DNA libraries. Briefly, each of the amplicons were diluted to 0.5 nM and amplified using barcoding primers (1D PCR Barcoding Amplicon Kit, Oxford Nanopore Technologies, [ONT], UK) and LongAmp Taq 2X Master Mix (New England Biolabs, MA, USA) with the following conditions: 95 °C for 3 min; 15 cycles of 95 °C for 15 seconds, 62 °C for 15 seconds, 65 °C for 80 seconds; and 65 °C for 80 seconds. The barcoded amplicons were bead purified (1:1.4; bead:solution), pooled (Run1; pool of 10 samples, Run2; pool of 5 samples) into a single All rights reserved. No reuse allowed without permission.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Centrifuge 24 , Kraken 50 , QIIME 5 and Mothur 37 have been used to identify and profile microbial species; however, the high error rate 20 of Nanopore reads translates to poor classification accuracy for many of these tools. Alternatively, de novo 3,26 or reference-based 18 assembly methods have been used in MinION and other deep sequencing platforms. Using a strategy similar to other unbiased laboratory tests (e.g., standard bacterial cultures), an approach was developed to maximize usage of reads (i.e., reads are not discarded based on pre-set length or abundance requirements, similar to how a single colony may be interpreted as a significant result). This approach uses read-based classification against a database containing an equal number of representative sequences per lineage to detect and classify the reads based on their lineage, before conducting read-based classification to the isolate level by using all available IBV sequences in a separate database. Finally, lineage-clustered isolate-level read assignments are interpreted by a veterinary diagnostician to result in reads available for consensus building.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'The use of a final consensus alignment helps to overcome the individual error rate of MinION sequencing. 3 Similarly, the absence of predetermined metrics used in de novo assembly allows for the informed decision as to how many consensus sequences to build, a bioinformatics problem when dealing with clinical cases that can contain more than one isolate (e.g., similar to how there is not a predetermined number of significant bacterial colonies). Confirmatory followup tests (e.g., RT-PCR) may be needed when dealing with low number of reads in clinical samples; however, future testing of this new technology will allow for creating standards for such confirmatory testing.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Taken together, these results suggest the application of MinION-based AmpSeq, specifically for detecting IBV isolates from clinical samples within a few days, compared to several days to weeks and multiple diagnostic assays to culture and detect multiple IBV All rights reserved. No reuse allowed without permission.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               full_text  \\\n",
       "0  VP3, and VP0 (which is further processed to VP2 and VP4 during virus assembly) (6). The P2 64 and P3 regions encode the non-structural proteins 2B and 2C and 3A, 3B (1-3) (VPg), 3C pro and 4 structural protein-coding region is replaced by reporter genes, allow the study of genome 68 replication without the requirement for high containment (9, 10) ( figure 1A ).\\nThe FMDV 5′ UTR is the largest known picornavirus UTR, comprising approximately 1300 71 nucleotides and containing several highly structured regions. The first 360 nucleotides at the 5′ 72 end are predicted to fold into a single large stem loop termed the S-fragment, followed by a The PKs were originally predicted in 1987 and consist of two to four tandem repeats of a ~48 86 nucleotide region containing a small stem loop and downstream interaction site (figure 1B) 87 (12). Due to the sequence similarity between the PKs (figure 1C), it is speculated that they 88 were formed by duplication events during viral replication, probably involving recombination. 89 Between two and four PKs are present in different virus isolates but no strain has been 90 identified with less than two PKs, emphasising their potential importance in the viral life cycle 91 (19, 20) . The presence of PKs has been reported in the 5′ UTR of other picornaviruses such as 92 author/funder. All rights reserved. No reuse allowed without permission. can occur in the absence of PKs at least one is required for wild-type (wt) replication. 104 Furthermore, competition experiments showed that extra copies of PKs conferred a replicative 105 advantage to genomes. Although replicons and full-length genomes lacking PKs were 106 replication-competent, no infectious virus was rescued from genomes containing less than one 107 PK copy. This is consistent with our earlier report describing the presence of putative 108 packaging signals in the PK region (22). 109 110 author/funder. All rights reserved. No reuse allowed without permission. Plasmid construction. 117 The FMDV replicon plasmids, pRep-ptGFP, and the replication-defective polymerase mutant 118 control, 3D-GNN, have already been described (10).\\nTo introduce mutations into the PK region, the pRep-ptGFP replicon plasmid was digested 121 with SpeI and KpnI and the resulting fragment inserted into a sub-cloning vector (pBluescript) 122 to create the pBluescript PK. PKs 3 and 4 were removed by digestion with HindIII and AatII 123 before insertion of a synthetic DNA sequence with PK 3 and 4 deleted. PKs 2, 3 and 4 were 124 deleted by PCR amplification using ΔPK 234 Forward primer and FMDV 1331-1311 reverse 125 primer, the resultant product was digested with HindIII and AatII and ligated into the 126 pBluescript PK vector. Complete PK deletion was achieved by introduction of an AflII site at 127 the 3′ end of the poly-C tract by PCR mutagenesis to create the sub-cloning vector, pBluescript 128 C11, which was then used to remove all the PKs by PCR mutagenesis using ΔPK 1234 forward 129 primer and FMDV 1331-1311 reverse primer. The modified PK sequences were removed from 130 the sub-cloning vectors and inserted into the pRep-ptGFP plasmid using NheI-HF and KpnI-131 HF.\\n132 133 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 7 Mutations to disrupt and reform PK structure were introduced using synthetic DNA by 134 digestion with AflII and AatII and ligation into a similarly digested pBluescript PK vector.\\nMutations were then introduced into the replicon plasmid as described above.\\nTo assess the effects of truncation of the poly-C-tract on replication the entire sequence was 137 removed. This was performed by PCR mutagenesis using primers C0 SpeI, and FMDV 1331- In vitro transcription. 143 In vitro transcription reactions for replicon assays were performed as described previously (28).\\nTranscription reactions to produce large amounts of RNA for SHAPE analysis were performed 145 with purified linear DNA as described above, and 1 μg of linearised DNA was then used in a 146 HiScribe T7 synthesis kit (NEB), before DNase treatment and purification using a PureLink FastQ files were quality checked using FastQC with poor quality reads filtered using the 225 Sickle algorithm. Host cell reads were removed using FastQ Screen algorithm and FMDV 226 reads assembled de novo into contigs using IDBA-UD (35). Contigs that matched the FMDV 227 library (identified using Basic Local ALighnment Search Tool (BLAST)) were assembled 228 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint into consensus sequences using SeqMan Pro software in the DNA STAR Lasergene 13 229 package (DNA STAR) (36). The SHAPE data largely agreed with the predicted structures with the stems of PK 1, 2 and 3, interacting nucleotides showed little to no reactivity, suggesting NMIA could not interact with 300 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 14 these nucleotides either due to the predicted base pairing or steric hindrance (figure 2B). The\\nNMIA reactivity for the interacting nucleotides in the stem-loops with downstream residues of 302 PK 1, 2 and 3 again largely agreed with the predicted structure, although the SHAPE data 303 suggests that there might be fewer interactions than previously predicted. However, differences 304 here could be due to heterogeneity in the formation of PKs in this experiment. The evidence 305 for loop-downstream interaction was weaker for PK4. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint\\norientation. 351 Since removal of all four PKs resulted in a significant decrease in replication, the minimal 352 requirements to maintain wt levels of replication were investigated. As near wt level of 353 replication was observed when only one PK was present, all further mutagenesis was 354 performed in a C11 replicon plasmid containing only PK 1. In addition, the orientation of PK 1 was reversed by \"flipping\" the nucleotide sequence to 367 potentially facilitate hybridisation of the loop with upstream rather than downstream sequences.\\nChanging the orientation of the PK reduced replicon replication to a similar level seen in the replication decreased until at passage three there is a 2.5 fold reduction compared to that of 398 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint passage 0 (figure 5B). Therefore, it appears that replicons with a single PK are at a competitive 399 disadvantage compared to those with two or more. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 20 of infectious virus despite being able to replicate after transfection into cells, is consistent with 448 a requirement for RNA structure within the PK region being required for virus assembly. The 5′ UTR of FMDV is unique amongst picornaviruses due to its large size and the presence 454 of multiple RNA elements, some of which still have unknown function. One of these features 455 is a series of repeated PKs varying in number from 2-4, depending on virus strain. In this study, 456 we sequentially deleted or mutated the PKs to help understand their role in the viral life cycle. 457 We also confirmed the predicted PK structures by SHAPE mapping, although there may be Although all viruses isolated to date contain at least two PKs, replicons or viruses containing a 464 single PK were still replication competent. However, replicons with more than a single PK 465 were found to have a competitive advantage over replicons with a single PK when sequentially 466 passaged. Replicons lacking all PKs displayed poor passaging potential even when co-467 transfected with yeast tRNA, reinforcing the observation of a significant impact in replication.\\nMoreover, viruses recovered from genomes with reduced numbers of PKs were slower growing 469 and produced smaller plaques. In addition, these differences were more pronounced in more PKs is functionally competent as no differences was seen between replicons congaing a single 472 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 21 copy of PK1 or PK4. This observation is consistent with a previous report of deletion of PK1, 473 along with the Poly-C-tract, with no adverse effect in viral replication (37). This also supports 474 our findings that the truncation of the Poly-C-tract to create the C11 construct had no effect on 475 replicon replication in the cell lines tested. As has been described with Mengo virus, it is 476 possible that the role of the poly-C-tract is essential in other aspects of the viral lifecycle which 477 cannot be recapitulated in a standard tissue culture system (39).\\nThe presence of at least two PKs in all viral isolates sequenced so far suggests that multiple 480 PKs confer a competitive advantage in replication. Here we showed by sequential passage that 481 replicons containing at least two PKs were maintained at a level similar to wt, but replicons 482 containing only one PK showed a persistent decline. It is unclear why some viral isolates 483 contain two, three or four PKs is still unknown, but this may be stochastic variation or may 484 reflect subtle effects of host range or geographical localisation. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint Significance is shown comparing the replication of C11 PK disrupt and C11 PK restore (Aii). Significance shown is compared to wt replicon. Error bars are calculated by SEM, n = 3, * P 673 < 0.05, **** P < 0.0001. 674 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 33 675 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint \\n                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
       "1  In December 2019, a novel coronavirus, SARS-CoV-2, was identified in Wuhan, China, as the etiologic agent of coronavirus disease 2019 , which by March 2020 has already spread across more than 80 countries 1 . Common symptoms of infection include fever, cough, and shortness of breath, while severe cases are characterized by advanced respiratory distress and pneumonia, often resulting in death 2 . Soon after the first epidemiological data together with SARS-CoV-2 genetic sequences were made available, a glut of phylogeny-based analyses began to circulate discussing, in scientific papers as well as (social) media, countries that might have been fueling the spread. The implications of misunderstanding the real dynamic of the COVID-19 pandemic are extremely dangerous. Ethnic or social discrimination resulting from unsupported assumptions on viral contagion -often amplified by irresponsible, uncontrollable communications -can be highly damaging for people and countries.\\nDespite (social) media are often vehicle for fake news and boast news hype, it is also worth noting the tremendous effort of the scientific community to provide free, up-to-date information on ongoing studies, as well as critical evaluations. In particular, the US-based NextStrain 3 team has been posting real-time updates on the epidemic tracing by molecular analyses. Several discussions and evidence-based debates on controversial hypotheses on the epidemic have ensued -such as the number of untraced infections in the US, the putative virus introduction in Italy through Germany, and the alleged lineage diversification in China 4 later criticized 5 -aided by a concerted and productive effort of internationally-renowned research groups in Edinburgh, Glasgow, and Oxford, UK, among others. Recently, an editorial published on Science 6 has also highlighted how unsupported or misleading claims circulating in forums, social media, and even peer-reviewed articles, have been led by a substantial over interpretation of the available data.\\nHence, the urgency to reframe the current debate in more rigorous scientific terms, and quantitatively evaluate whether sufficient information for reliable phylogenetic and phylogeographic studies currently exists, or which gaps need to be addressed.\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint Before carrying out any phylogeny-based analysis of virus evolution and spatiotemporal spread, it is crucial to test the quality of sequence data, since uneven sampling, presence of phylogenetic noise, and absence of temporal signal can affect reliability of the results (e.g. ancestral state reconstructions, molecular clock calibrations) 7 . We downloaded all SARS-CoV-2 full genome sequences deposited on GISAID (https://www.gisaid.org/) 8 to compare the number of high quality genomes sampled per country with the number of confirmed cases at the time of sampling, as well as the country's total population ( Figure 1 ). As March 10 th 2020, 331 genome sequences from 29 countries were available, almost twice as much than just one week earlier (March 3 rd 2020, 169 genome sequences from 22 countries). Yet, the sampling per country was inhomogeneous, with confirmed cases clearly uncorrelated to the number of genomes (Spearman correlation = 0.03). Moreover, correlation could only be investigated with confirmed cases (as proxy), since not all affected countries have made publicly available the total number of coronavirus testing performed. Even within the same country, sequenced genomes were usually sampled from few hotspots, not necessarily representative of the whole epidemic in that country.\\nItaly, for example, the second country per confirmed cases, has uploaded so far only two genomes from Lombardy and three from Lazio (Table S1) , i.e. only two of the twenty Italian regions, all with confirmed cases 9 . China and USA contributed with 123 and 63 genomes respectively (56% combined), while 13 countries contribute with only one or two samples (6.3% combined). Notably, three countries with less than 400 confirmed cases have provided a substantial number of genomes (Netherlands 28, Australia 18, Singapore 11; 17% combined) ( Figure 1 ).\\nThe effect of inhomogeneous sampling and missing data on phylogeography reconstructions, like the ones recently rushed through news and (social) media to claim specific dissemination routes of SARS-CoV-2 among countries, can be quite dramatic. An instructive example is the putative introduction of SARS-CoV-2 in Italy from Germany. The maximum likelihood (ML) tree, inferred from the full genome viral sequences available on March 3 rd , 2020 (Supplementary Figure S1 ), showed a well-supported cluster of European and Asian sequences (reported in Figure 2a ), which contained a subclade (subclade A, Figure 2a ) including a sequence isolated in Germany that . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint appears to be paraphyletic (with strong bootstrap support) to an Italian sequence clustering, in turn, with sequences from Finland, Mexico, Germany and Switzerland (Figure 2a ).\\nBased on this observation (which was available on NextStrain), a heated discussion circulated on social media about a transmission event from Germany to Italy followed by further spread from Italy the other countries. However, in a new tree inferred just one week later, when more than 135 new full genome sequences were made available on GISAID 8 ( Figure S2 ), the direct link between Germany and Italy has disappeared due to the additional clustering of previously unsampled sequences from Portugal, Brazil, Wales and Netherland ( Figure 2b ). Most importantly, there is no sufficient evidence of a higher likelihood (Shimodaira-Hasgawa test, Table 1 ) for tree topologies generated arbitrarily switching branches in the tree (see arrows in Figure 2b) compared to the ML tree inferred from the real data. In other words, it is not possible with the present data to decide which branching pattern (and, therefore, phylogeographic reconstruction)\\nis the most likely.\\nLack of resolution and uncertainty in the SARS-CoV-2 phylogenetic tree is to be expected, considering that relatively little genetic diversity can be accumulated during the first three months of an epidemic, even for an exponentially spreading and fast-evolving RNA virus. Plots of pairwise transitions/transversions vs. genetic distance 10 showed, indeed, low genetic diversity among SARS-CoV-2 genomes, considerable scattering, and overlap in transitions and transversions among closely related sequences (Figure 3a ). The 331 full genomes available contained 477 out of 29,564 variable sites (1.6%), of which 377 were singleton and only 100 (0.34%) parsimony informative, indicating a substantial lack of phylogenetic information, and raising the possibility of a number of sequencing errors. Further assessment of phylogenetic signal in the dataset was carried out using likelihood mapping analysis 11 , which estimates the likelihood of each of possible tree topology for any group of four sequences (quartet), randomly . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint chosen from an alignment, and reports them inside an equilateral triangle (the likelihood map)\\nwhere the corners represent distinct tree topologies and the center star-like trees. Quartets are considered \"resolved\" when the three likelihoods are significantly different (phylogenetic signal, most dots equally distributed in the corners, i.e. data are suitable for robust phylogeny inference), unresolved or partially resolved, when all three likelihood values or two of them are not significantly different (phylogenetic noise, most dots in sides or center areas, i.e. data may not be sufficient for robust phylogeny inference). Extensive simulation studies have shown that, for sequences to be considered robust in terms of phylogenetic signal, side/center areas of the likelihood mapping must include <40% of the unresolved quartets 12 , way lower than the 52.9% unresolved quartets detected in the SARS-CoV-2 full genomes alignment ( Figure 3b ). It is important to keep in mind that such a lack of phylogenetic signal makes the overall topology of any SARS-CoV-2 tree unreliable, and even clades with high bootstrap values should be interpreted with extreme caution. Finally, the mean genetic distance of each sequence from the root of a ML tree indicated that the available genomic data lack temporal signal (no significant linear correlation between accumulated genetic distance and sampling time, Figure 3c ). This is, again, expected in genomes obtained over a very short period of time (~ three months) since the beginning of the outbreak. Bayesian analysis 13 , which infers phylogenetic and phylogeographic patterns from a posterior distribution of trees, might facilitate comparisons about different evolutionary scenarios, help in retrieving the correct topology, and estimate an accurate evolutionary rate. With more genome sequences sampled from different time points and geographic areas made available, literally, every day, Bayesian analysis may soon be a viable option. However, as long the new data do not increase phylogenetic and temporal signal, Bayesian phylogenetic and phylogeography results would remain equally questionable.\\nTaken together, our observations suggest that any conclusion drawn, at present, about existing lineages and direction of viral spread, based on phylogenetic analysis of SARS-CoV-2 sequence data, is potentially misleading. Nowadays, published scientific data and media are very easily . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint .\\naccessible to a worldwide audience, thus it is important more than ever to weigh the information being shared. We believe that current molecular epidemiology data are not solid enough to provide a scientifically sound analysis of SARS-CoV-2 spread. Any inference based on phylogenetic trees should be considered at the very best preliminary, and hypothesis-generating.\\nHence the need for 1) avoiding stigmatization based on partial information and 2) continuing the concerted scientific effort to increase the number and quality of available data. The evolutionary dynamics of SARS-CoV-2 spread will unveil an unprecedented amount of information, which is essential to make policy decisions. The whole of humanity is threatened by the current pandemic, and policy makers need to adjust their mitigation measures while the pandemic itself is developing. Some of the urgent answers required lie in the timely availability of abundant, high quality sequence data not only from countries experiencing a high number of reported cases, such as Italy and South Korea, but also from others that seem to be experiencing, at least for now, a lower number of infections.\\nGISAID was accessed on March 3 rd and 10 th , 2020 (final dataset information reported in Table   S1 ). After quality control of sequences that were not full genomes or contained extensive stretches of unknown nucleotides, the first dataset (March 3 rd 2020) consisted of 169 genomes from 22 countries, while the second (March 10 th , 2020) 331 from 29 countries. Confirmed cases were retrieved for the same day from interactive web-based dashboard developed by Johns Hopkins University 14 .\\nTransition/transversions vs. genetic distance plot were calculated using DAMBE6 10 . Evaluation of the presence of phylogenetic signal satisfying resolved phylogenetic relationships among sequences was carried out with IQ-TREE, allowing the software to search for all possible quartets . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint using the best-fitting nucleotide substitution model 11 . ML tree reconstruction was performed in IQ-TREE based on the best-fit model chosen according to Bayesian Information Criterion (BIC) 15, 16 . Topology was tested performing the Shimodaira-Hasegawa 17 on \"Subclade A\" by comparing the inferred tree to three trees where two branches: (Tree 1) Italy (EPI_ISL_412973) switched with Wales (EPI_ISL_413555), (Tree 2) Germany (EPI_ISL_406862 with) Brazil (EPI_ISL_412964), (Tree 3) and Portugal (EPI_ISL_413648) with Brazil (EPI_ISL_412964), (Tree 4) Germany (EPI_ISL_406862) with Portugal (EPI_ISL_413648). All tests performed 10,000 resamplings using the using the RELL approximation method 18 . Position of these genomes within \"Subclade A\" is indicated in Figure 1 with an arrow.\\nExploration of temporal structure, i.e. presence of molecular clock in the data, was assessed by regression of divergence -root-to-tip genetic distance-against sampling time using TempEst 19 . In this case, absence of a linear trend indicates that the data does not contain temporal signal and that the data is not appropriate for phylogenetic inference using molecular clock models.\\nAncestral state reconstruction of full datasets reported in Figures S1 and S2 were based on the ML tree obtained with IQ-TREE and performed using TreeTime 20 . We choose to show the \"European subclade\" and \"Subclade A\" as cladograms in Figure 1 as there are no phylogenetic and temporal signals in the datasets: scaling phylogenies in time or nucleotide substitution in this case is misleading. . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint Each bi-directional arrow, and corresponding number, connects two tip branches that were switched to generate an alternative tree topology to be tested (see Table 1 , Methods). CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity. EPI_ISL_412971_BetaCoV_Finland_FIN-25_2020 EPI_ISL_406533_BetaCoV_Guangzhou_20SF206_2020 EPI_ISL_407896_BetaCoV_Australia_QLD02_2020 EPI_ISL_408478_BetaCoV_Chongqing_YC01_2020\\nEPI_ISL_412979__BetaCoV_Wuhan_HBCDC-HB-03_2020\\nEPI_ISL_411220_BetaCoV_France_IDF0386-islP3_2020 EPI_ISL_410044_BetaCoV_USA_CA6_2020 EPI_ISL_413020_BetaCoV_Switzerland_1000477377_2020\\nEPI_ISL_406030_BetaCoV_Shenzhen_HKU-SZ-002_2020\\nEPI_ISL_408669_BetaCoV_Japan_KY-V-029_2020\\nEPI_ISL_408515_BetaCoV_Wuhan_IVDC-HB-envF13-21_2020 EPI_ISL_410713_BetaCoV_Singapore_7_2020 EPI_ISL_412873_BetaCoV_Korea_KCDC24_2020\\nEPI_ISL_402120_BetaCoV_Wuhan_IVDC-HB-04_2020\\nEPI_ISL_406970_BetaCoV_Hangzhou_HZ-1_2020 EPI_ISL_406973_BetaCoV_Singapore_1_2020 EPI_ISL_412029_BetaCoV_Hong_Kong_VM20001988_2020\\nEPI_ISL_410720_BetaCoV_France_IDF0372-isl_2020 EPI_ISL_411951_BetaCoV_Sweden_01_2020 EPI_ISL_408488_BetaCoV_Jiangsu_IVDC-JS-001_2020\\nEPI_ISL_402123_BetaCoV_Wuhan_IPBCAMS-WH-01_2019 EPI_ISL_407894_BetaCoV_Australia_QLD01_2020 EPI_ISL_411955_BetaCoV_USA_CA8_2020 EPI_ISL_403962_BetaCoV_Nonthaburi_61_2020 EPI_ISL_406801_BetaCov_Wuhan_WH04_2020 EPI_ISL_408978_BetaCoV_Wuhan_WH05_2020\\nEPI_ISL_407084_BetaCoV_Japan_AI_I-004_2020 EPI_ISL_407215_BetaCoV_USA_WA1-F6_2020 EPI_ISL_411060_BetaCoV_Fujian_8_2020 EPI_ISL_410984_BetaCoV_France_IDF0515-isl_2020\\nEPI_ISL_412980__BetaCoV_Wuhan_HBCDC-HB-04_2020 EPI_ISL_410546_BetaCoV_Italy_INMI1-cs_2020 EPI_ISL_408977_BetaCoV_Sydney_3_2020 EPI_ISL_412982__BetaCoV_Wuhan_HBCDC-HB-06_2020 EPI_ISL_407071_BetaCoV_England_01_2020 EPI_ISL_412912_BetaCoV_Germany_Baden-Wuerttemberg-1_2020 EPI_ISL_411927_BetaCoV_Taiwan_4_2020 EPI_ISL_410536_BetaCoV_Singapore_5_2020 EPI_ISL_406538_BetaCoV_Guangdong_20SF201_2020 EPI_ISL_407988_BetaCoV_Singapore_3_2020 EPI_ISL_406036_BetaCoV_USA_CA2_2020 EPI_ISL_411954_BetaCoV_USA_CA7_2020 EPI_ISL_412028_BetaCoV_Hong_Kong_VM20001061_2020 EPI_ISL_406862_BetaCoV_Germany_BavPat1_2020 EPI_ISL_406597_BetaCoV_France_IDF0373_2020 EPI_ISL_408009_BetaCoV_USA_CA4_2020 EPI_ISL_410716_BetaCoV_Singapore_10_2020 EPI_ISL_403933_BetaCoV_Guangdong_20SF013_2020 EPI_ISL_402129_BetaCoV_Wuhan_WIV06_2019\\nEPI_ISL_406594_BetaCoV_Shenzhen_SZTH-003_2020 EPI_ISL_407987_BetaCoV_Singapore_2_2020 EPI_ISL_413015_BetaCoV_Canada_ON_VIDO-01_2020 EPI_ISL_407976_BetaCoV_Belgium_GHB-03021_2020 EPI_ISL_406531_BetaCoV_Guangdong_20SF174_2020 EPI_ISL_410715_BetaCoV_Singapore_9_2020\\nEPI_ISL_404227_BetaCoV_Zhejiang_WZ-01_2020 EPI_ISL_408479_BetaCoV_Chongqing_ZX01_2020 EPI_ISL_406535_BetaCoV_Foshan_20SF210_2020 EPI_ISL_408976_BetaCoV_Sydney_2_2020 EPI_ISL_411066_BetaCoV_Fujian_13_2020 EPI_ISL_407073_BetaCoV_England_02_2020\\nEPI_ISL_403929_BetaCoV_Wuhan_IPBCAMS-WH-04_2019 EPI_ISL_410301_BetaCoV_Nepal_61_2020 EPI_ISL_403931_BetaCoV_Wuhan_IPBCAMS-WH-02_2019\\nEPI_ISL_410531_BetaCoV_Japan_NA-20-05-1_2020 EPI_ISL_403963_BetaCoV_Nonthaburi_74_2020 EPI_ISL_402125_BetaCoV_Wuhan-Hu-1_2019\\nEPI_ISL_408514_BetaCoV_Wuhan_IVDC-HB-envF13-20_2020\\nEPI_ISL_402119_BetaCoV_Wuhan_IVDC-HB-01_2019 EPI_ISL_411956_BetaCoV_USA_TX1_2020 EPI_ISL_406596_BetaCoV_France_IDF0372_2020 EPI_ISL_412967_BetaCoV_China_IQTC02_2020 EPI_ISL_408431_BetaCov_France_IDF0626_2020\\nEPI_ISL_410532_BetaCoV_Japan_OS-20-07-1_2020\\nEPI_ISL_411957_BetaCoV_China_WH-09_2020 EPI_ISL_412970_BetaCoV_USA_WA2_2020 EPI_ISL_412978__BetaCoV_Wuhan_HBCDC-HB-02_2020 EPI_ISL_412869_BetaCoV_Korea_KCDC05_2020 EPI_ISL_412965_BetaCoV_Canada_BC_37_0-2_2020 EPI_ISL_412871_BetaCoV_Korea_KCDC07_2020 EPI_ISL_408008_BetaCoV_USA_CA3_2020\\nEPI_ISL_406593_BetaCoV_Shenzhen_SZTH-002_2020 EPI_ISL_406536_BetaCoV_Foshan_20SF211_2020 EPI_ISL_412459_BetaCoV_Jingzhou_HBCDC-HB-01_2020 EPI_ISL_410718_BetaCoV_Australia_QLD04_2020 EPI_ISL_412862_BetaCoV_USA_CA9_2020 EPI_ISL_413019_BetaCoV_Switzerland_1000477102_2020\\nEPI_ISL_412981__BetaCoV_Wuhan_HBCDC-HB-05_2020\\nEPI_ISL_408668_BetaCoV_Vietnam_VR03-38142_2020 EPI_ISL_410535_BetaCoV_Singapore_4_2020 EPI_ISL_408665_BetaCoV_Japan_TY-WK-012_2020 EPI_ISL_406716_BetaCoV_China_WHU01_2020 EPI_ISL_402127_BetaCoV_Wuhan_WIV02_2019 EPI_ISL_411926_BetaCoV_Taiwan_3_2020 EPI_ISL_406223_BetaCoV_USA_AZ1_2020 EPI_ISL_407214_BetaCoV_USA_WA1-A12_2020 EPI_ISL_406717_BetaCoV_China_WHU02_2020 EPI_ISL_404228_BetaCoV_Zhejiang_WZ-02_2020\\nEPI_ISL_412898__BetaCoV_Wuhan_HBCDC-HB-02_2019 EPI_ISL_413017_BetaCoV_South_Korea_KUMC01_2020 EPI_ISL_408480_BetaCoV_Yunnan_IVDC-YN-003_2020 EPI_ISL_403936_BetaCoV_Guangdong_20SF028_2020 EPI_ISL_408667_BetaCoV_Japan_TY-WK-521_2020 EPI_ISL_411218_BetaCoV_France_IDF0571_2020 EPI_ISL_405839_BetaCoV_Shenzhen_HKU-SZ-005_2020 EPI_ISL_408010_BetaCoV_USA_CA5_2020 EPI_ISL_406031_BetaCoV_Taiwan_2_2020 EPI_ISL_406844_BetaCoV_Australia_VIC01_2020 EPI_ISL_411950_BetaCoV_Jiangsu_JS01_2020 EPI_ISL_412974_BetaCoV_Italy_SPL1_2020\\nEPI_ISL_403928_BetaCoV_Wuhan_IPBCAMS-WH-05_2020 EPI_ISL_412870_BetaCoV_Korea_KCDC06_2020 EPI_ISL_404253_BetaCoV_USA_IL1_2020 EPI_ISL_410045_BetaCoV_USA_IL2_2020 EPI_ISL_403932_BetaCoV_Guangdong_20SF012_2020\\nEPI_ISL_408484_BetaCoV_Sichuan_IVDC-SC-001_2020 EPI_ISL_410537_BetaCoV_Singapore_6_2020 EPI_ISL_412872_BetaCoV_Korea_KCDC12_2020 EPI_ISL_407079_BetaCoV_Finland_1_2020\\nEPI_ISL_410545_BetaCoV_Italy_INMI1-isl_2020 EPI_ISL_408670_BetaCoV_USA_WI1_2020 EPI_ISL_403937_BetaCoV_Guangdong_20SF040_2020 EPI_ISL_403934_BetaCoV_Guangdong_20SF014_2020\\nEPI_ISL_413014_BetaCoV_Canada_ON-PHL2445_2020\\nEPI_ISL_411915_BetaCoV_Taiwan_CGMH-CGU-01_2020 EPI_ISL_410717_BetaCoV_Australia_QLD03_2020 EPI_ISL_412026_BetaCoV_Hefei_2_2020 EPI_ISL_407313_BetaCoV_Hangzhou_HZCDC0001_2020 EPI_ISL_412972_BetaCoV_Mexico_CDMX_InDRE_01_2020 EPI_ISL_406034_BetaCoV_USA_CA1_2020 EPI_ISL_411952_BetaCoV_Jiangsu_JS02_2020 EPI_ISL_411902_BetaCoV_Cambodia_0012_2020\\nEPI_ISL_413016_BetaCov_Brazil_SPBR-02_2020 EPI_ISL_413018_BetaCoV_South_Korea_KUMC02_2020 EPI_ISL_403930_BetaCoV_Wuhan_IPBCAMS-WH-03_2019\\nEPI_ISL_412899__BetaCoV_Wuhan_HBCDC-HB-03_2019 EPI_ISL_408489_BetaCoV_Taiwan_NTU01_2020 EPI_ISL_409067_BetaCoV_USA_MA1_2020 EPI_ISL_404895_BetaCoV_USA_WA1_2020\\nEPI_ISL_408666_BetaCoV_Japan_TY-WK-501_2020 EPI_ISL_410218_BetaCov_Taiwan_NTU02_2020 EPI_ISL_407893_BetaCoV_Australia_NSW01_2020 EPI_ISL_407193_BetaCoV_South_Korea_KCDC03_2020\\nEPI_ISL_412983_BetaCoV_Tianmen_HBCDC-HB-07_2020 EPI_ISL_410714_BetaCoV_Singapore_8_2020 EPI_ISL_403935_BetaCoV_Guangdong_20SF025_2020 EPI_ISL_406798_BetaCov_Wuhan_WH01_2019 EPI_ISL_411953_BetaCoV_Jiangsu_JS03_2020 EPI_ISL_412030_BetaCoV_Hong_Kong_VB20026565_2020 EPI_ISL_412968_BetaCoV_Japan_Hu_DP_Kng_19-020_2020 EPI_ISL_412973_BetaCoV_Italy_CDG1_2020 EPI_ISL_402132_BetaCoV_Wuhan_HBCDC-HB-01_2019\\nEPI_ISL_408486_BetaCoV_Jiangxi_IVDC-JX-002_2020 EPI_ISL_408430_BetaCoV_France_IDF0515_2020 EPI_ISL_410486_BetaCoV_France_RA739_2020 EPI_ISL_412966_BetaCoV_China_IQTC01_2020\\nEPI_ISL_412969_BetaCoV_Japan_Hu_DP_Kng_19-027_2020\\nEPI_ISL_411219_BetaCoV_France_IDF0386-islP1_2020\\nEPI_ISL_408481_BetaCoV_Chongqing_IVDC-CQ-001_2020 EPI_ISL_412975_BetaCoV_Australia_NSW05_2020 EPI_ISL_406534_BetaCoV_Foshan_20SF207_2020 . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint \\n                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
       "2  The 2019-nCoV epidemic has spread across China and 24 other countries 1-3 as of February 8, 2020 . The mass quarantine measure implemented in the epidemic's epicenter, Wuhan, Hubei Province, China, on January 23, 2020, has stopped the human migration between Wuhan and other places. However, according to the local authorities, over five million people had already traveled from Wuhan to other parts of China in the weeks leading to January 23, 2020, with the majority (79.96%) to the other cities within Hubei.\\nHealthcare resources have become a major bottleneck in the response to this public health crisis. The sudden onset of the epidemic and the sheer size of the infection have caused significant shortage in healthcare resources across Hubei Province. So far, there have been over 11,000 non-local healthcare professionals, mobilized by the public health authorities from other parts of China, fighting against the epidemic in Wuhan. However, this is far from adequate for Wuhan given the severity of the epidemic. In addition to Wuhan, other cities in Hubei are in need of urgent external help as well with respect to health resources, as multiple localized outbreaks are taking place. On February 7, 2020, the National Health Commission of China (NHCC) initiated the \"One City Covered by One Province\" program, explicitly assigning a specific province to aid one city in Hubei. Two critical questions of operational nature remain, which cities, both in Hubei and other provinces, are most vulnerable to localized outbreaks, and how the public health authorities should balance the healthcare resources under stress to best fight the epidemic and minimize the risks? This study is aimed to shed light on these questions by developing a vulnerability analysis framework integrating the human movement and healthcare resource data.\\nThis study makes use of four public datasets: (a) the daily counts of confirmed 2019-nCoV cases, collected from NHCC 4 ; (b) the list of designated hospitals with the number of reserved beds, retrieved from the websites of the provincial and municipal health commissions; (c) the information of the hospitals designated for 2019-nCoV, retrieved from 99.com.cn, a public hospital database; and (d) the human movement data, provided by the Baidu Migration (http://qianxi.baidu.com/), a public site offering real-time migration information across mainland China.\\nRecent studies indicated that the confirmed case data for Hubei Province, in particular, Wuhan, was subject to the under-reporting bias, while the data from places outside of Hubei is more reliable for inferring the actual epidemic size 5 . Following Wu et al. 5 , we developed a susceptible-exposed-infectious-recovered (SEIR)-based metapopulation model, which has 100 sub-populations representing the 100 cities that have the greatest number of people traveling from Wuhan. Note that we adopted the same epidemiological parameters as Wu et al. 5 . This model was then used to estimate IDi, the number of infected individuals in each city. Lastly, healthcare-resource-adjusted vulnerability indexes to the 2019-nCoV outbreak for city i, Vi were obtained as follows: Vi = IDi / Bi, where Bi represents the total number of reserved beds in the designated hospitals.\\nThe Figure presents the estimated number of infected individuals, as well as the healthcareresource-adjusted vulnerability to the 2019-nCoV outbreak across mainland China, as of January 29, 2020. The epidemiological parameters may have changed after January 23, 2020, when the lockdown measure was implemented. Thus, we only use the data by January 23, 2020 and produce the estimation till January 29, 2020 to accommodate the six-day incubation period identified by Wu et al. 5 .\\nThe ten most vulnerable non-Hubei cities are also listed in the figure. Note that the cities in Hubei (which are being aided by other provinces) are not presented here because of their significantly higher vulnerability as compared to others. Among the ten most vulnerable non-Hubei cities, Changsha, Beijing, Chongqing and Shanghai are major metropolises with a large number of estimated infected individuals. Xinyang, Jiujiang, Nanchang and Yiyang are medium-sized cities neighboring Hubei. In particular, Xinyang appears to be in a particular vulnerable situation because of the large number of people traveling from Wuhan and its insufficient healthcare resources.\\nThe reported vulnerability analysis informs public health response to the 2019-nCoV epidemic in multiple ways. For the identified cities with high vulnerability, it is important to increase the reserve of healthcare resources or have contingent plans. In particular, some of these cities already contributed significant healthcare resources to Hubei, leaving themselves inadequately prepared for the localized outbreak. Our analysis revealed that several medium-sized cities, particularly those neighboring Hubei, are highly vulnerable due to the lack of healthcare resources. \\n                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
       "3  Metagenomic sequencing, which allows us to directly obtain total genomic DNAs from host-associated and environmental samples, has led to important findings in many areas, such as digestive health [1] . While bacteria are the main focus of most metagenomic sequencing projects, there are fast accumulation of viral metagenomic data with sequencing viruses as the main purpose [2] . There are different types of viruses. We are mainly concerned with RNA viruses because many RNA viruses are notorious human pathogens, such as Influenza A, Human immunodeficiency virus (HIV), Ebola, SARS-CoV, and recently identified 2019-nCoV causing Wuhan pneumonia. Unlike DNA viruses, RNA viruses, which contain RNA genomes, lack faithful proofreading mechanisms during replication and thus can produce a group of related but different viral strains infecting the same host. This high genetic diversity within and across different hosts poses a great challenge for designing long-term protection strategies against these infectious diseases. For example, as the circulating strains can change every year, flu vaccine has to be administered every year.\\nAdvances in viral metagenomics have contributed significantly to new RNA virus discovery. According to a survey by Woolhouse et. al, the number of newly identified RNA viruses is changing from 1,899 in 2005 to 5,561 in 2018, which is 3 times of increase [3] . Largescale RNA virus sequencing projects using nextgeneration sequencing technologies have been conducted for different species. For example, Shi et al. have discovered a large number of new RNA viruses by sequencing samples from invertebrate and vertebrate animals [4, 5] . Claire et al. have discovered 25 new RNA viruses by sequencing samples associated with different drosophilid [6] . Bolduc et al. sequenced RNA viruses from Archaeal and bacterial samples [7] . Given fast accumulation of viral metagenomic data and expected discovery of new viruses, a key step is to conduct composition analysis for these data and assign taxonomic groups for possibly new species.\\nComposition analysis can be conducted at the read level or contig level. There is no doubt that contigs can provide more information than short reads in taxonomic classification. However, metagenomic assembly is still one of the most challenging computational problems in bioinformatics. Unlike single genome assembly, metagenomic assembly is more likely to produce chimeric contigs that can contain regions from different species. Thus, metagenomic assembly is often preceded by read binning that groups reads with the same or similar taxonomic labels [8] in order to achieve better performance.\\nIn this work, we design a read level taxonomic classification tool (named as \"CHEER\") for labeling new species in viral metagenomic data. To detect known species with available reference genomes, alignment-based methods can provide sufficient information for both classification and abundance estimation. This function has been incorporated in existing pipelines [9, 10, 11] and is thus not the focus of our tool. Instead, our tool is designed to handle the challenging cases of assigning taxonomic labels for reads of new species, which have not been observed before. This problem was clearly formulated in metagenomic phylogenetic classification tools such as Phymm and PhymmBL [8] . We first define the classification problem following the formulation in Phymm: the classification problem is the assignment of a phylogenetic group label to each read in input data sets. As we are only interested in the hard case of classifying reads of new species, which have not been sequenced and thus no label exists for these species, the expected labels for these reads are the higher-rank taxonomic groups such as genus, family, or order. All our data and experiments are designed so that the test species are masked in the reference database and the trained models.\\nPrevious experiments have shown that nucleotide-level homology search is not sensitive enough for assigning higher rank taxonomic groups to reads from new species [9, 10] . Protein-based homology search is thus preferred for phylogenetic classification [11, 12] . In particular, a recent tool, VirusSeeker [13] utilized BLASTx to classify reads into bacteria, phages, and other viruses. It is not clear whether the protein homology search is able to generate accurate classification at lower ranks. We will evaluate this in our experiments.\\nThere are also alignment-free methods for phylogenetic classification. Many of those tools implemented different machine learning models and were mainly designed for bacteria. For example, RDP [14] adopted a Naïve Bayes based classifier that can automatically learn features from the reads and assigned taxonomic labels to sequences from bacterial and archaeal 16S genes and fungal 28S genes. The NBC classifier [15, 16] encoded metagenomic reads using k-mers and implemented a bag-of-word model to achieve the readlevel taxonomic classification. Phymm [8] utilized interpolated Markov models (IMMs) for metagenomic read classification. These learning-based models enable users to label reads originating from new species.\\nThe previous works have shown that sequence composition is still an important and informative feature for phylogenetic classification. Recent applications of deep learning models have achieved better performance than conventional machine learning methods in different sequence classification problems [17] . In particular, the convolution filters in Convolutional Neural Network models (CNN) can represent degenerated sequence motifs and thus help CNN models to learn abstract composition-based features. For example, frequently activated convolution filters learned by DeepFam often represented the conserved motifs in different protein domain families [18] . The successful applications of deep learning in sequence classification motivated us to design a novel deep learning-based classification model for assigning taxonomic groups for new species in viral metagenomic data.\\nOur main contributions are summarized below. First, our deep learning model can assign higher rank phylogenetic group labels (such as genus) for reads from new species. To achieve this goal, our model organizes multiple CNN-based classifiers in a tree for hierarchical taxonomic group assignment. Second, to enable our model to capture as much information as possible from short reads, we implemented and compared two encoding methods: one hot vs. embedding. Third, as viral metagenomic data usually contains contaminations from either the host genomes or other microbes, we formulated the pre-processing step as an open set problem in targeted image classification and rejected non-viral reads by choosing appropriate negative training sets. We tested our model on both simulated metagenomic data and also real sequencing data. The results show that our model competes favorably with other popular methods.\\nIn this section, we will introduce our method for viral metagenomic taxonomic classification. First, we will show the architecture of CHEER, which is a hierarchical classification model from order to genus. Second, classifiers at each level will be described. Third, we will introduce DNA sequence encoding using skipgram based word embedding and compare it with onehot encoding. Forth, we will describe the rejection layer, following the idea of the open set problem, which is adopted to filter reads not belonging to RNA viruses. Finally, we will detail our dataset used in training and validation.\\nThe architecture of our model is sketched in Fig. 1. The key component is a tree model that consists of multiple classifiers from order to genus. In order to conduct phylogenetic classification for reads from new species, or even new genus, our classification is conducted using a top-down approach from the root to the leaf node. The top layer is a trained CNN that can reject reads not belonging to RNA viruses, which is an important preprocessing step for viral metagenomic data because of the contamination from the host genome or other microbes. Then, the filtered reads (i.e. mostly RNA viral reads) will be fed into the hierarchical classification model, which is referred to as the tree model hereafter. Specifically, the order-level classifier is responsible for classifying the reads into their originating orders. Then each order has a separately trained classifier for assigning input reads to families within that order (O1 to O3 in Fig. 1 ). For all the reads assigned into one family, the family classifier (F1-F7) will assign the reads to different genus within that family. As an example, a path for classifying a read from a species in genus 7 is highlighted in Fig. 1 . CHEER also implemented early stop functions at each level for stopping the classification path at higher ranks. This function can accommodate taxonomic classification for species from new genus or even higher ranks, which is possible for RNA viruses. It is convenient to add fine-grained classifiers within each genus so that we can assign species-level labels. However, as our focus is to conduct phylogenetic classification for reads from new species that do not have a species labels in the training data, we did not include the species-level assignment in this work.\\nAlternative designs of hierarchical classification work, like [19, 20] , either build a classifier for each taxonomic rank or build a binary classifier for each class. We implemented a structure that built just one classifier for each taxonomic rank and compared different structures' performance. According to the result shown in [Supplementary file 1], the structure shown in Fig.1 is better than one classifier per rank. Although one classifier per rank requires fewer classifiers, it needs to train many more parameters because the number of labels for each classifier increases by times. As a result, we observe overfitting more frequently.\\nEach classifier in the tree model is implemented using CNN, which has achieved superior performance in various sequence classification problems [18] . The convolution filters used in CNN resemble positionspecific weight matrix of motifs in genomic sequences. Frequently activated convolution filters can represent well-conserved motifs among sequences in the same class. Thus, a large number of convolution filters will be used in our CNN in order to learn well-conserved sequence features in different classes. Deep CNNs in fields such as computer vision often have multiple convolution layers in order to obtain more abstract features from training images. Here, we use a wide convolution design rather than a deep one, following the idea in [18] . Fig. 2 shows the classifier with the default hyper-parameters. First, reads will be encoded in matrices as input to the convolution layer using two methods. We will describe and compare these two encoding methods later in the Section 2.3. Then, multiple convolution filters of different sizes are utilized to learn the conserved sequence features. According to our empirical study, we choose 3, 7, 11, 15 as the filter sizes, with 256 filters for each size respectively (see Fig. 2 ). Max pooling is then applied to each convolution filter's output so that only the highest convolution value is kept. After the max-pooling layer, we have a 256-dimentional vector as the output of each convolution layer. We concatenate these four vectors into one 1024-dimentional vector and feed it to one fully connected layer with 512 hidden units. Dropout is applied to mitigate overfitting. The CrossEntropyLoss function supplied by Pytorch is adopted as an error estimation function. Due to the unbalanced size of each training class, we use the reciprocal of the proportion of each class as their error weight to help the model fairly learn from each class. And we utilize Adam optimizer with a 0.001 learning rate to update the parameters. The model is trained on HPCC with the 2080Ti GPU unit. \\nIn previous deep learning-based sequence classification models [21, 22] , one hot encoding was commonly adopted to convert a sequence into a matrix, which will then be used as input to convolution layers. There also exist some methods using k-mer composition [23] and frequency [15, 16] to encode the DNA sequence, similar to the bag-of-words model in the field of natural language processing. However, these k-mers representation cannot give complete sequential information about the original reads.\\nTo incorporate both the k-mer composition and also their ordering information, we build a Skip-Gram [24] based embedding layer that can learn which k-mers tend to occur close to each other. Embedding is widely used in natural language processing to learn the semantic and syntactic relationships from sentences. A neural network with one hidden layer is trained to map a word to an n-dimensional vector so that the words that usually occur together will be closer in the n-dimension space. For our DNA read classification task, k-mers are the words and the embedding layer will map proximate kmers into vectors of high similarity. The main hyper-parameter when training the skip-gram model is the size of the training context m, which defines the maximum context location at which the furthest k-mer is taken for training. Specifically, for a kmer at position i as the input to the skip-gram model, the output k-mers are its neighbors located at i+jk, where −m ≤ ≤ m. Larger m results in more training samples with a cost of training time. In our model, the default value of m is 1. As shown in Fig. 3 , we will sample 3 k-mers as our dataset at a time with the middle one as the input and the surrounding two as the output.\\nAs proven in [24] , Skip-Gram is able to automatically learn the relationship between proximate words and thus is expected to help learn the order between k-mers in genomic data. We implemented a hidden layer with 100 hidden units shown in Fig. 3 . After training the model, the co-occurrence features are embedded in the output of the hidden layer. Thus, a k-mer in a read will be represented as a 100-dimension vector, where each element in the vector is the output of the hidden unit in the hidden layer.\\nWhen we use this pre-trained embedding layer to encode the reads, we first split each read into sequential k-mers. Second, each k-mer is fed to the embedding layer and the output is a 100-dimension vector. Third, all these vectors are combined into a large matrix according to the original position of each k-mers. Finally, this matrix will be fed to the convolutional layer.\\nBased on our large-scale experiments, we found that classifiers are easy to overfit with increase of k. Thus, we choose 100-dimensional feature vectors with k being 3 as our default embedding layer parameter set based on our empirical analysis.\\nDue to the small genome size and limitations in library construction, it is not rare that many viral metagenomic datasets are still contaminated with the genomes of the host and other microbes. Viral read screening is thus a key pre-processing step before the downstream composition analysis. Specifically, our tree model should only be applied to viral reads. However, the close set nature of the classifier forces it to assign a known label to each input without being able to distinguish viral reads from others. In this work, we will reject non-viral reads by formulating it as an open set problem in targeted image classification [25] .\\nIn a majority of neural network models [26, 27] , the logit of the last layer will be fed into the SoftMax function, which takes a vector of n real numbers as input and output a normalized probability vector for all labels/classes. However, for an irrelevant input from an unknown clade, all classes in the model tend to have low probabilities and thus applying a threshold on uncertainty can be used to reject unknown classes [25] . Along with this idea, serval approaches [28, 29] have been proposed to solve the open set problem.\\nIn this work, we incorporate the pre-processing step in our hierarchical classification model by integrating a trained CNN and a SoftMax threshold. Contaminations such as bacteria will be rejected using empirically chosen SoftMax threshold τ. Our experiments show that SoftMax outputs a nearly uniform probability distribution of all labels if the input is not relevant. Thus, the SoftMax threshold τ is used to reject reads not belonging to either DNA or RNA viruses. However, using this thresholding method is not able to distinguish DNA viral reads from RNA viral reads due to their similar sequence composition. To accommodate this challenge, we designed a rejection classifier in the top layer using DNA viral reads as the negative class in the training (see Fig. 1 ).\\nAs the focus of our work is to assign labels for new species, the hierarchical classification ends at genuslevel assignment, which is conducted by the bottom layer in Fig. 1 . Given the fast accumulation of viral metagenomic data, there are cases where a new species does not belong to any existing genus and thus the label assignment should stop at the higher ranks. Thus, we allow our model to stop at a higher rank in the tree during the top-down classification. Towards this goal, we still apply SoftMax threshold in our early stop function, which stops reads from further classification by the child classifiers. This early stop function also serves as a quality control measure for preventing wrong classifications if the confidence is low. For example, when a read arrives, the model may be able to identify its order and family labels with high reliability but cannot decide which genus it belongs to. In this case, the model should only assign order and family label to this read rather than classifying it into a potentially wrong genus. To achieve this goal, SoftMax threshold is applied for every classifier in the tree. Only reads with a score higher than the threshold will be assigned to the child class.\\nAn example of early stop is highlighted by the yellow path in Fig. 1 . The model can classify the read into order 2 (O2) and then family 4 (F4) with high reliability but cannot decide which genus it belongs to.\\nAll the viruses used in the experiments were obtained from the RefSeq in the NCBI virus database. The The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.26.009001 doi: bioRxiv preprint viruses' taxonomic classification was downloaded from Virus Taxonomy 2018b Release [30] published by the International Committee on Taxonomic of Virus (ICVT). Because training deep learning classifiers required enough samples, we removed the orders with only one family, families with only one genus, and genera with less than 3 species from our dataset. Finally, there are 6 orders, 23 families, and 55 genera remained. The RefSeq genomes are available at the following URL:https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/fi nd-data/virus. The information of the taxa can be found in [Supplementary file 1]. As CHEER conducts read-level taxonomic classification, the training is also conducted using reads. In order to train our model on each complete virus genome, we generated simulated reads by uniformly extracting substrings of 250 bp with 200bp overlaps. All these reads formed the training set so that the model can learn the features from the whole genome. The validation set, however, is generated by read simulation tool WgSim [31] with specified error rate so that we can have a better estimation of the model's performance on real sequencing data. Some existing deep learning tools for taxonomic classification, like [21, 23] , generate reads first, then randomly choose some of them as training set and others as testing set. This strategy cannot guarantee that the training and testing data sets have no overlaps from the same species. In order to test whether our model can learn the label of new species, we use test species masking in the training set. Specifically, we split our RNA virus database into two independent sets. For any test species, their reads are not used in the training data. Thus, the model will only use the knowledge of known species for the new species classification.\\nIn this section, we will first give details of the hyperparameters of CHEER. Then we present the results of applying CHEER to both simulated and mock sequencing data. For the simulated data by WgSim, we evaluated how sequencing error rate affects the performance of CHEER and compared the taxonomic classification results with state-of-the-art. For the mock data, we tested CHEER's performance in both taxonomic classification and its ability in distinguishing viral reads from others.\\nIn our experiments, the minimum number of species within a genus is 3. We thus constructed three test sets, each containing reads from one species that is randomly chosen from each genus. The three test sets contain three different species, respectively. In total, 165 species from 55 genera are chosen to be \"new\" species for testing and the information of these species can be found in [Supplementary file 1].\\nWe used WgSim to simulate reads from testing species. In order to evaluate the impact of sequencing error on CHEER, we simulated reads with an error rate of 0, 0.01, 0.02, and 0.05, respectively. For each classifier (CNN), we computed the classification accuracy, which is defined in equation (1). To ensure that the computation of the accuracy is not affected by the number of reads from different species, we constructed a balanced test read set by simulating the same number of reads for each test species (1,000 reads for each). (1) For each component classifier in the tree model, we record the average accuracy for multiple test species. For each taxa level, we calculate the average accuracy of all the classifiers in the same level.\\nWe present the results of CHEER at each level in this section. Because CHEER is designed for new species, we benchmark it against alignment-free models. Based on the results shown by Wood et al. [7] , Naïve Bayes Classifier (NBC) outperformed other alignment-free models. Thus, we compared the performance of CHEER and NBC. NBC in [14] was designed for bacteria taxonomic classification and thus, we retrained NBC on our RNA virus dataset. We implemented the NBC model following [14] with the same parameters. Since each family has a separately trained classifier, which conducts genus classification within that family, there are totally 14 classifiers at this level. Fig. 5 shows the accuracy for labeling families. There are totally 6 classifiers that assign reads to families within each of the 6 orders. Fig. 6 shows the mean accuracy at each rank. The family and genus level accuracy are the average accuracy of all classifiers in Fig. 4 and Fig. 5 . There is only one classifier for order level classification.\\nAs shown in Fig. 4, Fig. 5, and Fig. 6 , CHEER has better performance than NBC across different ranks. We also compared the performance of using one-hot encoding and embedding layer in CHEER. The performance comparison at each rank shows that for most of the classifiers, the embedding layer improves the learning ability. Fig. 4, Fig. 5, and Fig. 6 also revealed the robustness of CHEER. Although the classification accuracy decreases as the error rate increases, the decrease is smaller than 2%. The accuracy is almost identical for many classifiers. Thus, CHEER is not very sensitive to sequencing errors. We also conducted experiments for reads with higher error rates (0.02 and 0.05). The results can be found in [Supplementary file 1].\\nThe next experiment is to evaluate our model on a real sequencing dataset and test CHEER's performance in both taxonomic classification and non-viral read rejection. We built a mock dataset by mixing Ebolavirus reads with gut bacterial reads from a gut amplicon sequencing dataset, which is chosen to ensure that we only use reads from bacteria. The taxonomic classification of Zaire Ebolavirus in the tree model is highlighted in Fig. 7 . We chose one of the Ebola species named Zaire Ebolavirus as our testing species because it is the most dangerous of the five Ebolavirus species and was responsible for the 2014 West African Ebola outbreak. We downloaded the dataset from NCBI with accession number SRR1930021. These reads were sequenced by Illumina using whole-genome amplification strategy. Since Zaire Ebolavirus usually causes fatal hemorrhagic fever and abdominal pain, it is possible that this Ebola virus occurs with other gut bacteria. Thus, we created a mock viral metagenomic data by mixing these Ebola reads with a gut bacteria amplicon sequencing data in order to test whether CHEER can reject the bacterial reads at the top rejection layer. The gut bacteria dataset is downloaded from NCBI with accession number: SRR10714026. There are totally 2,737 Ebola virus reads and 51,280 gut bacterial reads in the mock data. After the rejection layer, the Ebola reads will be used to evaluate our tree model.\\nTo treat Zaire Ebolavirus as a \"new\" species, we removed its genome from our training set and re-trained the whole pipeline. As shown in Fig. 7 , after removing this species, we still have four Ebola species in our training set. However, our read mapping experiment using Bowtie2 [32] showed that none of the reads from Zaire Ebolavirus can be mapped to other Ebola species with the default parameters, indicating that using usual alignment-based methods will have difficulty to conduct genus-level assignment.\\nIn our work, CHEER will first use rejection layer to remove the reads not belonging to the RNA virus. Second, the \"Order classifier\" will be applied to evaluate how many reads can be classified into Mononegavirales order. Third, \"Mononegavirales classifier\" takes all these Mononegavirales reads as input to predict how many reads will be assigned to the Filoviridae family. Finally, \"Filoviridae classifier\" is adopted to check how many reads belong to the Ebola genus. Fig. 8 show the ROC curve of the rejection layer. True positive rate reveals how many Ebolavirus reads pass the rejection layer. The false positive rate reveals how many gut bacterial reads pass the rejection layer. We generated thresholds from 0 to 1 with step size 0.01 and used each threshold for the rejection layer. Then for each threshold, we recorded the true positive rate and false positive rate for plotting the ROC curve. As shown in Fig. 8 , with the increase of the threshold, more bacterial reads are filtered by the rejection layer, at the expense of missing more real Ebola reads. The default threshold for the rejection layer is 0.6 in CHEER. User can adjust the threshold according to their needs.\\nAfter evaluating the rejection layer, we continued to test the hierarchical classification performance at each rank. We first compared our tree model with NBC. Then we also benchmarked CHEER with the method using protein-level alignment in Section 3.3.3. The performance of the complete hierarchical classification pipeline of NBC and CHEER is shown in Fig. 9 . From order to genus, each bar shows the accuracy of correctly classified reads by the parent classifier. For example, NBC correctly classified 1580 at the order level. Then, 1564 out of the 1580 reads are correctly labeled at the family level. While Fig. 4 , 5, and 6 have shown the performance of CHEER and NBC at each taxonomic rank using the same set of reads as input for each level of classifiers, this experiment focused on comparing the hierarchical classification pipeline using only correctly predicted reads by the parent classifier. The results demonstrated that CHEER can classify Ebola reads with better performance. NBC can only classify 1,564 (57.1%) reads into Ebolavirus. CHEER, however, had totally 2407 (87.9%) reads classified correctly.\\nAlthough nucleotide-level alignment models are not ideal choices for predicting new species, as mentioned in [8] , protein-level alignment such as BLASTx is still frequently used for homology search because of the conservation among homologous protein sequences. We followed VirusSeeker [10] , which is a BLASTx based model, to align translated reads to virus-only protein database (VIRUSDB NR) provided by NCBI. Then, we kept alignments with E-value cutoff 0.001, which is used in VirusSeeker. The output of BLASTx shows that a read can often align to multiple proteins, making the phylogenetic classification ambiguous (see Fig. 11 ). Also, existing tools such as VirusSeeker often use the best alignment only, which does not always yield the highest accuracy. To optimize the usage of the alignment-based method, we implemented k-Nearest Neighbor (kNN) algorithm to record the best k alignments rather than only using the best one. The label of the input read is determined by the majority vote in the labels of the top k alignments. According to Fig. 10 , most of the reads have hit numbers between 11 to 15 and 6 to10. Thus, we chose 1, 5, 10, 15 as our k when running kNN. There are also 512 reads that cannot be aligned with any VIRUSDB protein sequence. All these reads will be regarded as wrong prediction. Fig. 11 shows the classification results of kNN. When we use the best hit as the prediction the accuracy is only 64%. The best performance of kNN will be 74% with 2,026 correct predictions when k is 5. But the accuracy sharply decreases if k is larger than 5. Fig. 10 and Fig. 11 also revealed that it is hard for users to choose a proper k in real classification tasks. The most common value of the hit number is 11-15 shown in Fig. 11 . The best kNN result, however, has a k equal to 5. Thus, comparing to this alignment-based method, CHEER, with 2,407 (87.9%) reads classified correctly, is more practical and reliable.\\nThe two main purposes of the early stop function are: 1) to handle the cases of a new species from an unseen genus or higher ranks; 2) to serve as a quality control measure for preventing wrong classifications in child classifiers. As Fig. 6 shows, the higher rank classification tends to be more accurate than lower ranks. Because the genus of Ebolavirus exists in our classification tree, the evaluation of the early stop function focuses on the second purpose. Fig. 12 shows the percentage of how many reads are classified into Ebolavirus genus correctly for different classifiers. By allowing early stop, the number of wrong predictions decreased sharply. In short, although there are fewer reads being labelled at the genus level, the total number of reads with wrong classification is decreased. Thus, this is still a useful feature provided by CHEER.\\nAnother significant advantage of CHEER is the high classification speed. Although training the model uses heavy computing resources, the inference procedure of this algorithm is fast. And with the help of GPU acceleration, the training time also decreases a lot.\\nIn the experiments shown in Section 3.4, it takes around 46 minutes to run BLASTx against the virus protein database. Our method, however, only takes 7 second for the preprocessing step to convert the raw sequence into a vector, and then takes less than 35 second to make the prediction for all 2,737 reads.\\nIn this work, all training data are RNA virus genomes downloaded from the NCBI RefSeq database. The training reads are extracted uniformly from the genome to make sure that classifiers can learn the features from the whole genome. To evaluate the model on new species detection, all species in the test set are excluded from the training data. Error rates are also added to evaluate the robustness of the model. The results reveal that the model is tolerant to the sequencing error. In addition, a SoftMax threshold rejection layer is applied to filter reads from non-RNA viruses. The SoftMax threshold in each classifier will also help early stop classification with low confidence and thus decreases the wrong predication rate. We show a case study on real-world data to identify how this threshold will influence the rejection classifier and early stop function. The results show that CHEER competes favorably with state-of-the-art tools.\\n   \n",
       "4  Infectious bronchitis (IB), which is caused by infectious bronchitis virus (IBV), is one of the most important diseases of poultry, causing severe economic losses worldwide. 8 Clinical signs of disease are diverse and include respiratory distress, severe ocular discharge, poor body weight gain, decreased egg production, flushing (renal disease), and occasionally mortality in chickens. 7 IB is often complicated by secondary bacterial (e.g., E. coli, Mycoplasma sp.) and viral infections (e.g., avian influenza virus, Newcastle disease virus, and infectious laryngotracheitis virus). 43 Lack of cross protection among IBV serotypes is a challenge to controlling IB 15 ; therefore, control of IB relies heavily on serotype-specific live attenuated vaccines. 8 Collectively, the presence of multiple IBVs in a single sample, emergence of variant IBVs, and the high genetic diversity of IBV can complicate the diagnosis of IB and illustrate the need for enhanced diagnostics. 15 \\nClinical swab samples (n = 15) were obtained from samples submitted to the Poultry Diagnostic & Research Center, University of Georgia. Detailed information about the samples is provided in Table 1 .\\nTotal RNA was extracted from each of the swab samples using QIAamp Viral RNA Mini kit (Qiagen, Hilden, Germany) as per manufacturer's instructions, aliquoted, and stored at −80 ºC until further use. These samples were previously tested for IBV and GA08 serotype of IBV.\\nBriefly, a pan IBV RT-qPCR assay 4 was used to detect IBV in general and then IBV strain specific RT-qPCR assay 36 was used to detect GA08 in the samples using the AgPath ID, One step RT-PCR kit (Thermo Fisher Scientific, Waltham, MA, USA) on the Applied Biosystems 7500 FAST following the previously described protocols.\\nFor MinION cDNA synthesis, RNA was extracted from 500 µl of each clinical sample using Trizol-LS (Thermo Fisher Scientific, Waltham, MA, USA), per manufacturer's directions.\\nA reaction mixture of 8 µl of total RNA, 1 µl of random primers and 1 µl of dNTPs was incubated at 65 °C for 5 min, chilled on ice for at least 1 min followed by addition of 10 µl of cDNA synthesis mix including SuperScript III (Thermo Fisher Scientific, Waltham, MA, USA) according to the manufacture's instruction. The reaction was incubated at 25 °C for 10 min, then at 50 °C for 50 min for cDNA synthesis. The reaction was terminated at 85 °C for 5 min, and then chilled on ice. To remove residual RNA, the cDNA solution was incubated with RNase H at 37 °C for 20 min. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. sequencing protocol was initiated using the MinKNOW software v.5.12. Sequencing was allowed to continue for 2 hours until 42,940 reads were obtained for Run1 and 156,000 reads were obtained for Run 2 in 6 hours.\\nFirst, a lineage-typing database, containing 32 IBV S1 gene sequences (one sequence from each of 32 lineages in the 6 genotypes 44 ) and the chicken genome (GCF_000002315.4_Gallus_gallus-5.0), was constructed (referred to as \"IBV-lineage-typing database\" in this study). A second database was constructed with all the Avian Coronaviruses S1 sequences (n = 7328) available in NCBI (as of: 09-08-2017) and the chicken genome (referred to as \"All-IBV database\" in this study). Prior to database construction, all sequences were dustmasked (NCBI C++ ToolKit), and each isolate was assigned a unique taxonomy ID as a species hierarchically under the genus Gammacoronavirus to allow easy sorting. The local BLAST databases were compiled using default settings. The IBV-lineage-typing database was All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint used to assign IBV lineages to each read, as appropriate. The All-IBV database was used to assign IBV isolates taxonomic ID to each of IBV reads from within IBV lineage read clusters.\\nThe IBV-lineage-typing database, which contains only one sequence per lineage, is required as centrifuge divides the score for any given read by the number of hits that have an equal score.\\nSince some lineages are overrepresented in GenBank (e.g., Mass strains), this results in the scores of those reads (e.g., Mass reads) being divided by a large number, effectively reducing the score for any single alignment. However, because there is only one read per lineage in the IBVlineage-typing database, it has insufficient diversity to determine if there is more than one strain of the same lineage present in a sample. Thus, the All-IBV database provides the diversity required for that analysis.\\nA schematic diagram of the workflow of MinION data analysis is presented in Fig. 1 . For Nanopore sequencing data, pre-processing steps were performed to prepare data for downstream analysis. Briefly, Nanopore reads (FAST5) were basecalled using Albacore v 2.02 (https://github.com/Albacore/albacore) with following parameters (read_fast5_basecaller.py -i /Input_reads.fast5 --recursive -t 4 -s /Output_files --flowcell FLO-MIN107 --kit SQK-LSK108o fastq). Porechop (https://github.com/rrwick/Porechop) was used for adapter trimming, (default setting), barcode-based demultiplexing (default settings), and to trim an additional 21 nucleotides, representing the S1 primer sequences (porechop -i Input_file.fastq --extra_end_trim 21 -b ./output_demultiplexed files.fastq). After barcode and adapter removal, reads were analyzed with a script-based, 2-step data analysis protocol, which includes Centrifuge-kreport as taxonomic read classifier 24 using the sequences in the above mentioned BLAST databases.\\nBriefly, basecalled reads (FASTQ) from individual barcoded samples were used as an input. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint First, the basecalled reads were aligned to the IBV-lineage-typing database using BLASTn and reads were clustered based on the read sequence alignment to the respective prototype sequence of IBV lineage. These read clusters were used to interpret presence of IBV genotypes and lineages in the samples. For isolate level identification of IBV, the lineage-based read clusters were individually aligned to the All-IBV database, which produced sub-clusters of reads. Each of these read sub-clusters potentially represents a different isolate and was further used for interpretation. Knowing that MinION sequencing has a high sequencing error rate in individual reads, further steps were added in the data analysis algorithm to obtain a more accurate consensus sequence. Therefore, these read sub-clusters were mapped, using Geneious mapper in Geneious software v11.1.3. (Biomatters, Ltd, Auckland, New Zealand), to the IBV-lineagetyping database (a reference FASTA file) to obtain consensus sequences from each sub-cluster.\\nA minimum threshold for number of reads per consensus sequence was not set as in two samples (sample #4 in Run1 and sample #12 in Run 2), only 1 and 5 IBV reads were obtained, respectively; therefore, consensus sequences were built from all the available IBV reads per subcluster. The consensus sequences were compared to GenBank using BLASTn. To select the \"top hit\" from BLASTn output, sequence search results were ordered by \"sequence identity\" and then sequence alignments were evaluated for \"minimum mismatches\" and \"coverage\" of query or subject sequence. All isolates with the highest query or subject coverage and the fewest mismatches were used as \"top hit\" in the final results.\\nFor Sanger sequencing amplicons synthesis, cDNA from MinION library preparation was amplified using the following primer sets (Table 2) . Firstly, a primer set based on Connecticut sequence NCBI FJ904716.1 was used to amplify Genotype I-lineage 1 viruses (e.g., Connecticut All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint and Massachusetts serotypes). Secondly, a primer set based on GA98 sequence NCBI AF274439.1 was used to amplify Genotype IV-lineage 1 viruses (e.g., Georgia 1998 and Delaware 072 serotypes). Primer sets were designed using NCBI Primer-BLAST. 51 The PCR reaction mixture is composed of 10 µl of cDNA, 0.5 µl of 10 µM forward primer, 0.5 µl of 10 µM reverse primer, 2.5 µl of 10× buffer, 0.5 µl of enzyme mix, and 0.5 µl of 10mM dNTPs, then made the final volume to 25 µl with nuclease-free water. The following thermocycling conditions were used for amplicon synthesis: denaturation at 95 °C for 1 min, followed by \\nFor Sanger sequencing data, the chromatogram from each of the samples was manually checked and primer sequences were trimmed in MEGA 6.0. 42 Forward and reverse sequences from multiple clones were aligned using MEGA 6.0 software and consensus sequences were All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint compared to GenBank using Basic Local Alignment Tool (BLAST) (https://blast.ncbi.nlm.nih.gov/) (as of: 12-03-2018) and the top hit from BLASTn output was selected, as described above for MinION sequencing data, to identify the IBV isolate in samples.\\nA pairwise nucleotide identity comparison on the partial S1 sequences of IBVs from each of the samples sequenced with Sanger sequencing and MinION sequencing was performed.\\nBriefly, the final AmpSeq consensus sequences from identical IBV types in each of the sample were aligned using ClustalW in MEGA6 42 and this alignment was used for pairwise nucleotide identity using MEGA6. 42 All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint\\nFive samples from Run 1 were tested with a pan-IBV RT-qPCR assay 4 , and four of those were positive for IBV. Samples with pathogenic IBV strains (Run 2, n = 5) were tested with the pan-IBV RT-qPCR and all five samples were IBV positive (Table 1) . Next, a GA08 serotype specific primer set (in house validated set of primers used for diagnostics) was used on all samples from Run 1 (including five samples that were previously tested by the pan-IBV PCR) and four of 10 samples were positive. Sample #3 and #4, which were positive with the pan-IBV RT-qPCR assay, were negative with GA08 specific RT-qPCR assay indicating that these samples contained IBV serotypes other than GA08 and required further testing. Sample #5 tested negative with both assays (Table 1) .\\nA total of 10 clinical swab samples (Run 1) were processed for Sanger sequencing to confirm the presence of IBV lineages. Based on the MinION results, primers targeting GIV-L1 and GI-L1 (GA98 and Conn and primer sets, respectively) were created. The PCR products were cloned and multiple (6-24) clones from each sample were submitted for Sanger sequencing and results are summarized in Table 3 . Using GIV-L1 serotype primers showed that 9 out of 10 samples were positive with 1 sample (#5) negative for IBV (which is consistent with the pan-IBV RT-qPCR results for this sample). Consensus sequences from multiple clones obtained by using GIV-L1 primers showed the presence of GA98 (GA/A9dvaccinated) in 7 of 9 IBV-positive samples. DE072 was detected in samples #1 and #9 as the lone GIV-1 isolate but was also detected in three samples (#2, and #10) that also contained GA98 (Table 4 ). By using the GI-L1 primer set, which amplified Conn and Mass type IBVs, 3 of 9 IBV-positive samples were All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. (Table 4 ). Based on the two conventional RT-PCRs and the GA08 RT-qPCR, it was determined that only samples #8 and #9 contained a single IBV isolate, while the other 7 samples were positive for 2-4 types of IBV (Table 3) .\\nPCR amplicons obtained directly from 10 clinical swab samples with vaccine IBV serotypes were barcoded, pooled together and sequenced on MinION device (Run 1). A total of 38,661 reads were successfully basecalled from the entire sequencing run (total reads = 42,940).\\nAfter demultiplexing, reads per barcode ranged from 831 to 4114. A total of 14,845 reads were not assigned to any of the used barcodes and 128 reads were discarded due to middle adapters.\\nThe Nanopore reads were queried against IBV-lineage-typing database to determine if the samples contained IBV (Figure 1 ). This AmpSeq protocol detected IBV reads in 8 of 10 samples.\\nThe number of IBV reads per sample (IBV positive sample) ranged from 56-944. The MinIONnegative samples included sample #5, which was consistently negative with RT-qPCR and RT-PCR assays (and thus interpreted as IBV negative), and sample #1, which had the highest Cq values in the pan-IBV RT-qPCR (Table 1) .\\nSequencing data was further analyzed to determine if multiple IBV genotypes or lineages could be detected. Lineage-1 and lineage-27 from Genotype I (GI-L1, GI-L27), and lineage-1 from Genotype IV (GIV-L1) were the detected lineages in the samples. Results from individual RT-qPCR assay for GA08 (GI-L27), and RT-PCR assays for GIV-L1and GI-L1 confirmed the presence of multiple IBV genotypes and lineages in 6 of the 9 IBV-positive samples (#1, #2, #3, #4, #6, #7) and AmpSeq results confirmed multiple genotypes and lineages in 4 (#2, #4, #6, #7) of those 6 samples (AmpSeq failed to detect any IBV in #1, and failed to detect GIV-L1 in #3). All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint\\nThe presence of a single lineage was confirmed by PCR-based assays and AmpSeq in three samples (#8, # 9 and #10) ( Table 3) .\\nIt was observed that the single read cluster composed of less than 5 reads (Run1, sample #4, GA98) yielded a poor-quality consensus sequence, consistent with the known individual read error rate of MinION sequencing. High-quality consensus sequences were obtained from the other read clusters (>5 reads per cluster). Each of the obtained consensus sequences were compared to GenBank databases using BLAST, which revealed >99% sequence identity to respective IBV isolates Table 4 ). As described in the Sanger Sequencing section, two samples (#8 and #9) contained only one IBV isolate per the non-MinION assays, and the AmpSeq results were consistent with these findings. Non-MinION assays showed that seven samples contained two or more isolates per sample. Six samples (#1, #3, #4, #6, #7, and #10) contained two isolates per sample (Table 4 ). Of those six samples, AmpSeq detected both isolates in four samples (#4, #6, #7, and #10), one isolate in one sample (#3), and no IBV in one sample (#1; as mentioned above, sample #1 was the sample with the highest Cq value). Of note, sample #10 contained two isolates from the same lineage and AmpSeq was able to identify both isolates within this sample.\\nFinally, the seventh multi-isolate sample (#2) contained four isolates per the non-MinION results, and the AmpSeq detected two of those four isolates. These data show that a co-infection of multiple IBV lineages existed in the above-mentioned samples but a single RT-qPCR, or Sanger sequencing of a single clone may not have detected these co-infecting IBV lineages and required multiple assays to detect all the IBV types. However, the AmpSeq protocol accurately detected multiple IBV lineages in 4 of 7 samples, with partial detection in two of the remaining three samples (Table 4 ). In samples which had the same IBV type, Blast search of consensus All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint sequences obtained from AmpSeq and Sanger sequencing identified the same (or highly related isolate for sample #6 [GA98] and sample #9) in the NCBI nt database as per parameters described in the methods section.\\nTo evaluate the utility of this protocol on clinical swab samples with pathogenic IBV variants, IBV-S1 gene was amplified directly from five clinical tracheal scrapings and the PCR amplicons were used to create MinION libraries. A total of 146,540 reads were successfully basecalled from the entire sequencing run (total reads = 156,000). After demultiplexing, reads per barcode ranged from 4,285 to 41,131. A total of 24,297 reads were not assigned to any of the used barcodes and 8,109 reads were discarded due to middle adapters in the basecalled reads.\\nReal-time analysis of MinION data, which was obtained within 10 minutes of the sequencing, was sufficient for detection of IBV. However, the sequencing data obtained from entire sequencing run was processed with the same protocol as described above. This AmpSeq protocol detected IBV reads in all of the 5 tested samples using all basecalled reads. The number of IBV reads per sample ranged from 5-4,956. Additionally, the sequencing data analysis showed that the IBV reads belonged to GI-17 lineage and typed the isolate as DMV1639. After MinION sequencing results, these samples were later tested to confirm the presence of IBV variant by MDL_DMV1639 IBV variant specific RT-qPCR assay. All 5 samples were positive for IBV MDL_DMV1639 variant of IBV (Table 5) .\\nIn all of the samples where a matching IBV genotype/lineage was identified by both AmpSeq and Sanger sequencing, the AmpSeq data showed high concordance with Sanger sequencing data, In one sample (#4), only a single MinION read of GA98 was detected by All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint sequencing, consistent with the reported single-read accuracy of MinION sequencing. All other samples showed 99-100% pairwise identity across the partial S1 fragment generated by Sanger Sequencing. (Table 4 ).\\nAll rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint\\nThis study describes a sequencing approach for rapid and accurate identification of IBV genotypes/lineages utilizing the MinION Nanopore sequencer. Additionally, a sequence analysis workflow is described here. The workflow analysis is designed to deal with the multiple, highly similar sequences contained within GenBank that are due to numerous sequencing experiments of IBV (especially vaccine viruses). This protocol demonstrates the feasibility of real-time-based sequencing to provide isolate-level resolution of IBV directly from clinical swab samples, including samples that contain multiple isolates.\\nThe proper diagnosis of IBV as the cause of clinical respiratory disease is contingent on virus typing and differentiating live vaccine viruses from field strains. 11 Previously, it was reported that IBV genotypes correlate well with the serotypes of IBV 45 ; therefore, accurate genotypic identification of IBV will be useful to diagnose vaccine and variant viruses in clinical samples. The rapid pan-IBV and serotype specific RT-qPCR assays 4,23,36 have been used for serotype identification; however, positive results from RT-qPCR is insufficient to determine the IBV genotype and obtain isolate-level resolution of IBV; thus, sequence analysis of IBV-S1 gene is required. Previously, use of partial S1 gene sequences (450 bp) to type IBV has been described. 29 However, increasing the length of sequenced S1 gene (~1620 bp) results in more data to be used for genotyping as more of the hypervariable region is covered. 17, 27, 45 For example, the AmpSeq protocol was able to differentiate two genetically very similar (99.5% at S1 gene of IBV genome) but serotypically different GIV-L1 isolates, DE072 and GA98, within a single sample. Thus, the detection of highly diverse IBV genotypes, via the S1 subunit, with a single and rapid sequencing protocol is desirable and the ability to detect multiple isolates in a single sample would improve IBV detection in clinical samples. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint Currently, pan-IBV RT-qPCR is used to rapidly detect IBV from clinical samples for screening purposes, 4 but this requires additional serotype-specific RT-qPCRs to genotype positive samples, including samples containing more than one genotype. Additionally, if unidentified IBVs are in the sample, then current detection and characterization may also require egg culture followed by various PCRs. 8 The AmpSeq method described in this report detected IBV in 13 of 14 IBV-positive and detected all the mixed isolates in four of seven samples containing two or more isolates. While mixed isolates were not 100% identified (0/2 in #1, 2/4 in #2, and ½ in #3) in three samples with the AmpSeq method, detection of those mixed isolates required several non-AmpSeq assays. Although, there is room for improving this new AmpSeq method, it represents a promising, single-step assay that can be used without egg culture. Thus, it is another tool for the detection of IBV, especially in cases when multiple isolates may be present or when genotyping is especially important.\\nOne area that is problematic for a test such as AmpSeq is to detect all the isolates that may be present. While AmpSeq did detect all isolates in 4 samples, in 3 cases not all of the isolates were detected. A potential explanation to partial detection of multiple IBV genotypes could be the relative abundance of IBV genotypes in these clinical samples. It could also be that amplification of these IBV genotypes by serotype-specific RT-qPCR assays is more efficient due to small targeted fragment size (e.g., 120 bp for PCR and ~1600 bp for AmpSeq) and better primer alignment to the target (e.g., degenerate bases are used in the S1 primers used for AmpSeq). One complicating factor of the current AmpSeq protocol is that the IBV target sequence 27 was not originally designed for high specificity, especially from clinical samples. As such, a high proportion of total reads were non-IBV reads (e.g., often mapping to the chicken genome, data not shown), consistent with the extra bands visible in the original report for these All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint target sequences. 28 Additionally, it is possible that certain genotypes are better complemented to the S1 primers than others and may outcompete those isolates for amplification in the AmpSeq protocol. Lastly, increasing the total number of reads collected by AmpSeq may improve the ability to detect all genotypes in sample by AmpSeq. It is possible to allow the sequencing to continue longer to obtain more reads per sample. Overall, AmpSeq is a feasible test for IBV characterization, and work is ongoing to improve this new type of assay.\\nFor example, cost and time efficiency of a sequencing protocol can be improved through multiplexing of more samples in a single sequencing run. 3, 47 In this study, samples (Run 1 = 10, Run 2 = 5) were simultaneously multiplexed (i.e., pooled and then sequenced in one run) while maintaining IBV genotyping from data collected. Since the MinION flow cells were not exhausted, and can be washed and re-used, the AmpSeq method also has the potential for sequential multiplexing. This would decrease the need to hold samples for weeks while waiting for the cost-optimal number of samples for simultaneous multiplexing. Alternatively, if detection of all isolates within a given sample is preferred, the sequencing can be run longer to increase sensitivity. The single protocol nature of AmpSeq, the ability to obtain S1 gene sequence results, real-time data analysis, and flexibility of testing design makes MinION-based AmpSeq a viable sequencing protocol for typing of IBV isolates.\\nThe advent of real-time, in-house, third generation sequencing represents a transformative opportunity for diagnostic laboratories, by offering the ability to more fully characterize PCR reactions beyond confirming amplicon size (e.g., routine electrophoresis), Sanger sequencing RT-PCR products, or by confirming a partial sequence through probe hybridization (e.g., probe-based qPCR). However, the interpretation of such large data sets represents a challenge to veterinary diagnosticians. Read-based classification software such as genotypes from single sample. Thus, the MinION-based AmpSeq coupled with the data analysis workflow for identification, differentiation, and accurate prediction of IBV genotypes from clinical swab samples can be used as an adjunct to other established rapid diagnostic assays 9,12 until extensive testing of this protocol is done to improve and validate AmpSeq for IBV diagnostics. Furthermore, AmpSeq-based assays can be, and are being applied to other viral pathogens, 3,37 demonstrating the power and utility of this method in the age of molecular diagnostics. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint \\n                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "\n",
       "   clusters  \n",
       "0  3         \n",
       "1  7         \n",
       "2  2         \n",
       "3  9         \n",
       "4  9         "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['clusters'] = clusters\n",
    "#save the dataframe\n",
    "df.to_csv('clusters_df.csv')\n",
    "df.head() #display the head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h5>Visualization of clusters</h5>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To get a better sense of the clusters resulting from applying kmeans, we'll plot the clusters.\n",
    "\n",
    "After applying PCA to the cleaned data, we still have too many dimensions to be able to visualize the clusters. This is why further dimensionality reduction is applied on this highly dimesnional data. We chose to use the TSNE technique on top of PCA."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tsne = TSNE(random_state=30).fit_transform(tf_idf_norm_kmeans)\n",
    "    \n",
    "f, ax = plt.subplots(1, figsize=(9, 6))\n",
    "    \n",
    "cmap = {0: 'lime', 1: 'red', 2: 'blue', 3: 'green', 4: 'yellow', 5: 'orange', 6: 'pink', 7: 'purple', 8: 'brown', \n",
    "        9: 'black', 10: 'lightgreen', 11: 'lightblue', 12: 'deepskyblue', 13: 'violet', 14: 'indigo'}\n",
    "\n",
    "for c in np.unique(clusters):\n",
    "    ix = np.where(clusters == c)\n",
    "    ax.scatter(tsne[ix, 0], tsne[ix, 1], c=cmap[c])\n",
    "        \n",
    "ax.set_title('TSNE Cluster Plot')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The previous plot gives us more insight into how the papers are clustered/grouped. The algorithms used may help us find connections amongst different papers that may have been initially unapparent to us. This can help in the extraction of important information from the papers and advancing research.\n",
    "\n",
    "Next we apply topic modeling to find the most significant keywords that distinguish the clusters. This will allow us to obtain more meaning about the topics/themes of the different clusters."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h1>Topic Modeling</h1> \n",
    "<p>Group the documnents according to their cluster</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.decomposition import LatentDirichletAllocation\n",
    "\n",
    "clean_text = np.array(lemmatized_text)\n",
    "\n",
    "#grouped_docs is a dictionary that holds the docs indices in a cluster eg. {0: [5,20,31], 1: [7,8,11]...}\n",
    "grouped_docs = df.groupby('cluster').indices"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<p>Topic modeling for the set documents within each cluster </p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in range(0,20):\n",
    "    \n",
    "    # get the tf-idf matrix for the documents within this cluster\n",
    "    v = TfidfVectorizer(sublinear_tf = True, min_df = 0.05,max_df = 0.8)\n",
    "    current_tf_idf = v.fit_transform(clean_text[grouped_docs[i]])\n",
    "    feature_names = v.get_feature_names()\n",
    "    \n",
    "    # get the top words for 2 topics within each cluster\n",
    "    lda_model = LatentDirichletAllocation(n_components= 2, max_iter=5,\n",
    "                                    learning_method='online',\n",
    "                                    learning_offset=50.,\n",
    "                                    random_state=30)\n",
    "    lda_model.fit(current_tf_idf)\n",
    "    lda_model.transform(current_tf_idf)\n",
    "    \n",
    "    n_top_words = 10\n",
    "    print (\"cluster {}: \".format(i))\n",
    "    \n",
    "    for topic_idx, topic in enumerate(lda_model.components_):\n",
    "        message = \"Topic #%d: \" % topic_idx\n",
    "        message += \" \".join([feature_names[i]+\",\"\n",
    "                             for i in topic.argsort()[:-n_top_words - 1:-1]])\n",
    "        print(message[:-1])\n",
    "    print()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
