{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package stopwords to\n",
      "[nltk_data]     C:\\Users\\HP\\AppData\\Roaming\\nltk_data...\n",
      "[nltk_data]   Package stopwords is already up-to-date!\n",
      "[nltk_data] Downloading package punkt to\n",
      "[nltk_data]     C:\\Users\\HP\\AppData\\Roaming\\nltk_data...\n",
      "[nltk_data]   Package punkt is already up-to-date!\n",
      "[nltk_data] Downloading package wordnet to\n",
      "[nltk_data]     C:\\Users\\HP\\AppData\\Roaming\\nltk_data...\n",
      "[nltk_data]   Package wordnet is already up-to-date!\n"
     ]
    }
   ],
   "source": [
    "import nltk\n",
    "import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\n",
    "import re #import the regular expression library\n",
    "import string\n",
    "import os\n",
    "import json\n",
    "from sklearn.feature_extraction.text import CountVectorizer, TfidfVectorizer\n",
    "from sklearn.metrics.pairwise import cosine_similarity\n",
    "from sklearn.datasets import load_files\n",
    "from nltk import pos_tag\n",
    "from nltk.tokenize import RegexpTokenizer\n",
    "from tqdm import tqdm\n",
    "from nltk.corpus import stopwords #import the stopwords from the ntlk.corpus library\n",
    "nltk.download('stopwords')\n",
    "from nltk.tokenize import word_tokenize #import the word_tokenize method, which is used to turn sentences into words\"\n",
    "nltk.download('punkt')\n",
    "from collections import Counter\n",
    "from nltk.stem import WordNetLemmatizer, PorterStemmer\n",
    "nltk.download('wordnet')\n",
    "import numpy as np\n",
    "import spacy\n",
    "from spacy.lang.en.stop_words import STOP_WORDS\n",
    "import en_core_sci_lg  # model downloaded in previous step"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "_cell_guid": "b1076dfc-b9ad-4769-8c92-a6c4dae69d19",
    "_uuid": "8f2839f25d086af736a60e9eeb907d3b93b6e0e5"
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 1934/1934 [01:02<00:00, 31.13it/s]\n"
     ]
    }
   ],
   "source": [
    "FullPaper=[]\n",
    "directories = [\"biorxiv_medrxiv\"]\n",
    "for directory in directories: #for each of the three folders carrying the json format of different research papers\n",
    "    for file in tqdm(os.listdir(f\"{directory}/{directory}/pdf_json\")): #for every json file\n",
    "        file_path=f\"{directory}/{directory}/pdf_json/{file}\" #set the file path to the file_path variable \n",
    "        paper = json.load(open(file_path,\"rb\"))\n",
    "        title = paper['metadata']['title'] \n",
    "        try:\n",
    "            abstract = paper['abstract']\n",
    "        except:\n",
    "            abstarct=\"\"                \n",
    "        full_text=\"\"     \n",
    "        \n",
    "        for text in paper['body_text']:\n",
    "            full_text += text['text'] +'\\n\\n' \n",
    "        FullPaper.append([title,abstract,full_text])\n",
    "        \n",
    "FullPaperDataframe=pd.DataFrame(FullPaper,columns=['title','abstract','full_text'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "> Our aim is to address these points:\n",
    "* Capabilities to discover a therapeutic (not vaccine) for the disease, and clinical effectiveness studies to discover therapeutics, to include antiviral agents."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 100/100 [07:55<00:00,  4.75s/it]\n"
     ]
    }
   ],
   "source": [
    "full_text = FullPaperDataframe['full_text'].head(100)\n",
    "clean_text = [] #a list which will hold all the bodies of the papers after being stripped out of stopwords\n",
    "wnl = WordNetLemmatizer()\n",
    "porter = PorterStemmer()\n",
    "# punctuation_regex = string.punctuation\n",
    "# punctuation_regex = punctuation_regex.replace(\"-\", \"\") # keep the hyphens\n",
    "# pattern = r\"[{}]\".format(punctuation_regex) # generate the regex pattern\n",
    "pattern = \"\"\"!\"#$%&'()*+,.:;<=>?@[\\]^`{|}~\"\"\" #the pattern which will account for punctuation, and will later be used to remove them\n",
    "#cleaning the data and removing stop words\n",
    "for val in  tqdm(full_text):\n",
    "        body_tokens = word_tokenize(val)\n",
    "        paper_body_without_stopwords = [token.lower() for token in body_tokens if not token.lower() in stopwords.words('english') and nltk.pos_tag([token.lower()])!='PRP'] #remove the stop words in the body and return a list\n",
    "        clean_string = ' '.join(paper_body_without_stopwords).lower() #convert the list into string\n",
    "        clean_string = re.sub(rf\"[{pattern}]\", '', clean_string)#remove punctuation except for hyphens\n",
    "        clean_string = re.sub(r\"\\bthe\\b\", r\"\", clean_string) #remove the\n",
    "        clean_string = re.sub(r'\\bwe\\b', '',clean_string ) #remove the pronoun we\n",
    "        clean_string = re.sub(r'\\bit\\b', '',clean_string )#remove the pronoun it\n",
    "        clean_string = re.sub(r'\\bthey\\b', '',clean_string )#remove the pronoun they\n",
    "        clean_string = re.sub(r'\\bcopyright\\b', '',clean_string )#remove the word copyright\n",
    "        clean_string = re.sub(r'\\bhttps\\b', '',clean_string )#remove the word https\n",
    "        clean_string = re.sub(r'\\bet\\b', '',clean_string )#remove the word et\n",
    "        clean_string = re.sub(r'\\bal\\b', '',clean_string )#remove the word al\n",
    "        clean_string = re.sub(r'\\bpreprint\\b', '',clean_string )#remove the word preprint\n",
    "        clean_string = re.sub(r'\\bthis\\b', '',clean_string )#remove the word this\n",
    "        clean_string = re.sub(r'\\bthese\\b', '',clean_string )#remove the word these\n",
    "        clean_string = re.sub(r'\\balso\\b', '',clean_string )#remove the word also\n",
    "        clean_string = re.sub(r'\\bhowever\\b', '',clean_string )#remove the word however\n",
    "        clean_string = re.sub(r'\\balthough\\b', '',clean_string )#remove the word although\n",
    "        clean_string = re.sub(r'\\bamong\\b', '',clean_string )#remove the word among\n",
    "        clean_string = re.sub(r'\\bin\\b', '',clean_string )#remove the preposition in\n",
    "        clean_string = re.sub(r'\\bmedrxiv\\b', '',clean_string )#remove the word medrxiv\n",
    "        clean_string = re.sub(r'\\bbiorxiv\\b', '',clean_string )#remove the word mbiorxiv\n",
    "        clean_string = re.sub(r'\\blicense\\b', '',clean_string )#remove the word license\n",
    "        clean_string = re.sub(r'\\bwithout\\b', '',clean_string )#remove the word without\n",
    "        clean_string = re.sub(r'\\bfig\\b', '',clean_string )#remove the word fig\n",
    "        clean_string = re.sub(r'\\bfigure\\b', '', clean_string)#remove the word figure\n",
    "        clean_string = re.sub(r'\\b[a-zA-Z]\\b', '', clean_string)#remove all the single letters      \n",
    "\n",
    "        clean_string = re.sub(r'^\\d+\\s|\\s\\d+\\s|\\s\\d+$', '', clean_string) #remove any digit that is not part of a word\n",
    "        clean_text.append(clean_string) #add the string to the list \n",
    "stemmed_text=[] # a list which will contain the stemmed bodies of all docs      \n",
    "for val in clean_text: #for the body (which was stripped out of stop words) of each paper\n",
    "    stemmed_val=\"\" #create an empty string\n",
    "    words = val.split(' ') #split body of paper on spaces\n",
    "    for word in words: #for each word in the body\n",
    "        if(word!=\"\"): #if the word is not an empty string\n",
    "            if (word==\"coronaviruses\"):#since the stem and lemmatize functions do not recognize the word coronaviruses as the plural of coronavirus, a conidition was created\n",
    "                word =\"coronavirus\"\n",
    "            stem_lemma =wnl.lemmatize(word) if wnl.lemmatize(word).endswith('e') else porter.stem(word)\n",
    "            stemmed_val+=stem_lemma+\" \" #stem/lemmatize the word and concat it to the string stemmed_val\n",
    "    stemmed_text.append(stemmed_val) #add the string stemmed_val to the list which will contain the stemmed bodies of all docs  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "# this cell is no longer needed and is kept for reference\n",
    "\n",
    "\n",
    "text = ' '.join(stemmed_text) #convert all the stemmed bodies into one string containing all the stemmed bodies of all papers \n",
    "s=text.split(' ')#split the combination of papers body on space\n",
    "vocab_count = Counter(s)# count the number of vocab (unique words)\n",
    "most_occur = vocab_count.most_common(200) #get the most common 200 words\n",
    "#print(most_occur) #print the most common 200 words"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[('02', 0), ('03', 1), ('08', 2), ('10', 3), ('100', 4), ('11', 5), ('12', 6), ('13', 7), ('14', 8), ('15', 9), ('16', 10), ('17', 11), ('18', 12), ('19', 13), ('1a', 14), ('1b', 15), ('1c', 16), ('1d', 17), ('1e', 18), ('1f', 19), ('20', 20), ('2019', 21), ('2020', 22), ('21', 23), ('22', 24), ('229e', 25), ('23', 26), ('24', 27), ('25', 28), ('26', 29), ('28', 30), ('29', 31), ('2a', 32), ('2b', 33), ('2c', 34), ('2d', 35), ('2e', 36), ('30', 37), ('34', 38), ('35', 39), ('36', 40), ('37', 41), ('38', 42), ('3a', 43), ('3b', 44), ('3c', 45), ('3d', 46), ('3e', 47), ('40', 48), ('48', 49), ('4a', 50), ('4b', 51), ('4c', 52), ('4d', 53), ('4e', 54), ('50', 55), ('56', 56), ('5a', 57), ('5b', 58), ('5c', 59), ('5d', 60), ('5e', 61), ('60', 62), ('64', 63), ('6b', 64), ('72', 65), ('80', 66), ('aa', 67), ('abil', 68), ('able', 69), ('abnorm', 70), ('above', 71), ('absence', 72), ('absent', 73), ('absolute', 74), ('abstract', 75), ('abund', 76), ('abundance', 77), ('academ', 78), ('acceler', 79), ('accept', 80), ('acceptable', 81), ('access', 82), ('accessible', 83), ('accord', 84), ('accordingli', 85), ('account', 86), ('accumul', 87), ('accur', 88), ('accuraci', 89), ('accurate', 90), ('ace2', 91), ('achiev', 92), ('achieve', 93), ('acid', 94), ('acknowledge', 95), ('acquir', 96), ('across', 97), ('act', 98), ('action', 99), ('activ', 100), ('active', 101), ('actual', 102), ('acute', 103), ('ad', 104), ('adapt', 105), ('adaptive', 106), ('add', 107), ('addit', 108), ('address', 109), ('adjac', 110), ('adjust', 111), ('administr', 112), ('admiss', 113), ('admit', 114), ('adopt', 115), ('adult', 116), ('advanc', 117), ('advance', 118), ('advantage', 119), ('advice', 120), ('affect', 121), ('affin', 122), ('aforement', 123), ('african', 124), ('agarose', 125), ('age', 126), ('agent', 127), ('aggreg', 128), ('aggregate', 129), ('agreement', 130), ('aid', 131), ('aim', 132), ('air', 133), ('alanine', 134), ('albeit', 135), ('aldrich', 136), ('algorithm', 137), ('align', 138), ('aliquot', 139), ('allow', 140), ('almost', 141), ('alone', 142), ('along', 143), ('alreadi', 144), ('alter', 145), ('altern', 146), ('alternative', 147), ('alveolar', 148), ('alway', 149), ('ambigu', 150), ('american', 151), ('amino', 152), ('amongst', 153), ('amount', 154), ('amplicon', 155), ('amplif', 156), ('amplifi', 157), ('analog', 158), ('analys', 159), ('analyse', 160), ('analyt', 161), ('analyz', 162), ('analyze', 163), ('ancestor', 164), ('anchor', 165), ('and', 166), ('angiotensin', 167), ('anim', 168), ('annot', 169), ('announc', 170), ('anonym', 171), ('anoth', 172), ('anova', 173), ('answer', 174), ('anti', 175), ('antibodi', 176), ('anticip', 177), ('antigen', 178), ('antivir', 179), ('appar', 180), ('appear', 181), ('appendix', 182), ('appli', 183), ('applic', 184), ('applicable', 185), ('approach', 186), ('appropriate', 187), ('approv', 188), ('approxim', 189), ('approximate', 190), ('april', 191), ('architecture', 192), ('ard', 193), ('area', 194), ('argu', 195), ('argue', 196), ('aris', 197), ('around', 198), ('array', 199), ('arriv', 200), ('arrow', 201), ('article', 202), ('artifici', 203), ('asia', 204), ('asian', 205), ('ask', 206), ('aspect', 207), ('assay', 208), ('assembl', 209), ('assess', 210), ('assign', 211), ('associ', 212), ('associate', 213), ('assum', 214), ('assume', 215), ('assumpt', 216), ('asymptomat', 217), ('atom', 218), ('attach', 219), ('attack', 220), ('attempt', 221), ('attent', 222), ('attenu', 223), ('attractive', 224), ('attribut', 225), ('autoimmune', 226), ('autom', 227), ('automat', 228), ('avail', 229), ('averag', 230), ('average', 231), ('avoid', 232), ('awar', 233), ('axi', 234), ('back', 235), ('background', 236), ('bacteri', 237), ('bacteria', 238), ('baidu', 239), ('balance', 240), ('ban', 241), ('band', 242), ('bar', 243), ('barcod', 244), ('bas', 245), ('baseline', 246), ('basi', 247), ('basic', 248), ('bat', 249), ('batch', 250), ('bayesian', 251), ('bead', 252), ('bear', 253), ('became', 254), ('becom', 255), ('become', 256), ('bed', 257), ('began', 258), ('begin', 259), ('behavior', 260), ('behaviour', 261), ('behind', 262), ('beij', 263), ('believ', 264), ('believe', 265), ('belong', 266), ('benchmark', 267), ('benefici', 268), ('benefit', 269), ('besid', 270), ('best', 271), ('beta', 272), ('better', 273), ('beyond', 274), ('bi', 275), ('bia', 276), ('bias', 277), ('binari', 278), ('bind', 279), ('binomi', 280), ('bio', 281), ('bioinformat', 282), ('biolog', 283), ('black', 284), ('blast', 285), ('block', 286), ('blood', 287), ('blot', 288), ('blue', 289), ('board', 290), ('bodi', 291), ('bond', 292), ('bootstrap', 293), ('border', 294), ('born', 295), ('bottom', 296), ('bound', 297), ('boundari', 298), ('bovine', 299), ('box', 300), ('bp', 301), ('branch', 302), ('break', 303), ('breath', 304), ('briefli', 305), ('bring', 306), ('broad', 307), ('broadli', 308), ('buffer', 309), ('build', 310), ('built', 311), ('busi', 312), ('by', 313), ('byintern', 314), ('ca', 315), ('calcul', 316), ('calculate', 317), ('calibr', 318), ('call', 319), ('came', 320), ('cancer', 321), ('candidate', 322), ('canon', 323), ('capabl', 324), ('capable', 325), ('capac', 326), ('captur', 327), ('capture', 328), ('care', 329), ('carri', 330), ('carrier', 331), ('case', 332), ('catalyt', 333), ('categor', 334), ('categori', 335), ('caus', 336), ('cause', 337), ('cc', 338), ('cdc', 339), ('cdna', 340), ('cell', 341), ('cellular', 342), ('center', 343), ('central', 344), ('centre', 345), ('certain', 346), ('chain', 347), ('challeng', 348), ('challenge', 349), ('chance', 350), ('chang', 351), ('change', 352), ('channel', 353), ('charact', 354), ('character', 355), ('characterist', 356), ('characterize', 357), ('charg', 358), ('charge', 359), ('check', 360), ('chemic', 361), ('children', 362), ('china', 363), ('chinese', 364), ('chip', 365), ('choice', 366), ('choose', 367), ('chose', 368), ('chosen', 369), ('chronic', 370), ('ci', 371), ('circle', 372), ('circul', 373), ('circumstance', 374), ('citi', 375), ('clade', 376), ('claim', 377), ('clarifi', 378), ('class', 379), ('classic', 380), ('classif', 381), ('classifi', 382), ('clean', 383), ('clear', 384), ('clearli', 385), ('cleav', 386), ('cleavage', 387), ('clinic', 388), ('clinician', 389), ('clone', 390), ('close', 391), ('closer', 392), ('closest', 393), ('closure', 394), ('cluster', 395), ('co', 396), ('coat', 397), ('cod', 398), ('code', 399), ('codon', 400), ('coeffici', 401), ('cohort', 402), ('coincid', 403), ('cold', 404), ('coli', 405), ('collect', 406), ('color', 407), ('column', 408), ('combat', 409), ('combin', 410), ('combine', 411), ('come', 412), ('commerci', 413), ('commiss', 414), ('committee', 415), ('common', 416), ('commonli', 417), ('commun', 418), ('comorbid', 419), ('compar', 420), ('comparable', 421), ('comparative', 422), ('compare', 423), ('comparison', 424), ('compart', 425), ('compartment', 426), ('compet', 427), ('compil', 428), ('complement', 429), ('complementari', 430), ('complet', 431), ('complete', 432), ('complex', 433), ('compliance', 434), ('complic', 435), ('compon', 436), ('compos', 437), ('composit', 438), ('compound', 439), ('comprehensive', 440), ('compris', 441), ('compromise', 442), ('comput', 443), ('compute', 444), ('concentr', 445), ('concept', 446), ('concern', 447), ('conclud', 448), ('conclude', 449), ('conclus', 450), ('condit', 451), ('conduct', 452), ('confer', 453), ('confidence', 454), ('confirm', 455), ('conflict', 456), ('conform', 457), ('confound', 458), ('conjug', 459), ('conjugate', 460), ('connect', 461), ('consecutive', 462), ('consensu', 463), ('consequ', 464), ('consequence', 465), ('conserv', 466), ('conservative', 467), ('consid', 468), ('consider', 469), ('considerable', 470), ('consist', 471), ('constant', 472), ('constantli', 473), ('constitut', 474), ('constrain', 475), ('constraint', 476), ('construct', 477), ('consult', 478), ('consumpt', 479), ('contact', 480), ('contagi', 481), ('contain', 482), ('contamin', 483), ('content', 484), ('context', 485), ('contin', 486), ('conting', 487), ('continu', 488), ('continue', 489), ('contract', 490), ('contrast', 491), ('contribut', 492), ('contribute', 493), ('control', 494), ('convent', 495), ('convers', 496), ('convert', 497), ('coordinate', 498), ('copi', 499), ('core', 500), ('corona', 501), ('coronaviru', 502), ('coronavirus', 503), ('correct', 504), ('correctli', 505), ('correl', 506), ('correlate', 507), ('correspond', 508), ('corrobor', 509), ('cost', 510), ('costli', 511), ('cough', 512), ('could', 513), ('count', 514), ('counter', 515), ('counti', 516), ('countri', 517), ('country', 518), ('coupl', 519), ('course', 520), ('cov', 521), ('cover', 522), ('coverage', 523), ('covid', 524), ('covid19', 525), ('creat', 526), ('create', 527), ('crisi', 528), ('criteria', 529), ('criterion', 530), ('critic', 531), ('cross', 532), ('crowd', 533), ('crucial', 534), ('crystal', 535), ('ct', 536), ('cultur', 537), ('culture', 538), ('cumulative', 539), ('curat', 540), ('cure', 541), ('current', 542), ('curve', 543), ('custom', 544), ('cut', 545), ('cycle', 546), ('cytokine', 547), ('cytometri', 548), ('cytoplasm', 549), ('cytotox', 550), ('daili', 551), ('damage', 552), ('danger', 553), ('dash', 554), ('database', 555), ('dataset', 556), ('date', 557), ('day', 558), ('de', 559), ('dead', 560), ('deal', 561), ('death', 562), ('decade', 563), ('decemb', 564), ('decid', 565), ('decide', 566), ('decis', 567), ('decision', 568), ('declar', 569), ('declare', 570), ('decline', 571), ('decreas', 572), ('decrease', 573), ('dedic', 574), ('deem', 575), ('deep', 576), ('default', 577), ('defense', 578), ('defin', 579), ('define', 580), ('definit', 581), ('degrad', 582), ('degree', 583), ('delay', 584), ('delet', 585), ('deliv', 586), ('deliveri', 587), ('demand', 588), ('demograph', 589), ('demonstr', 590), ('demonstrate', 591), ('denot', 592), ('denote', 593), ('densiti', 594), ('depart', 595), ('depend', 596), ('depict', 597), ('deplet', 598), ('deploy', 599), ('deposit', 600), ('depth', 601), ('deriv', 602), ('derivative', 603), ('derive', 604), ('describ', 605), ('describe', 606), ('descript', 607), ('design', 608), ('desirable', 609), ('despite', 610), ('destin', 611), ('detail', 612), ('detect', 613), ('detectable', 614), ('determin', 615), ('determine', 616), ('detriment', 617), ('develop', 618), ('development', 619), ('deviat', 620), ('device', 621), ('diagnos', 622), ('diagnosi', 623), ('diagnost', 624), ('diagram', 625), ('diamond', 626), ('die', 627), ('differ', 628), ('difference', 629), ('differenti', 630), ('differentiate', 631), ('difficult', 632), ('difficulti', 633), ('diffus', 634), ('diffuse', 635), ('digest', 636), ('dilut', 637), ('dimens', 638), ('dimension', 639), ('dimer', 640), ('diminish', 641), ('direct', 642), ('directli', 643), ('discard', 644), ('discharge', 645), ('discov', 646), ('discoveri', 647), ('discrep', 648), ('discrete', 649), ('discuss', 650), ('disease', 651), ('disord', 652), ('dispers', 653), ('display', 654), ('disrupt', 655), ('dissemin', 656), ('dissoci', 657), ('distanc', 658), ('distance', 659), ('distinct', 660), ('distinguish', 661), ('distress', 662), ('distribut', 663), ('district', 664), ('divers', 665), ('diverse', 666), ('divid', 667), ('dna', 668), ('dnase', 669), ('document', 670), ('domain', 671), ('domin', 672), ('done', 673), ('donor', 674), ('dose', 675), ('dot', 676), ('doubl', 677), ('double', 678), ('down', 679), ('download', 680), ('downstream', 681), ('dr', 682), ('draft', 683), ('dramat', 684), ('drastic', 685), ('drawn', 686), ('dri', 687), ('drive', 688), ('driven', 689), ('drop', 690), ('droplet', 691), ('drug', 692), ('dual', 693), ('due', 694), ('durat', 695), ('dye', 696), ('dynam', 697), ('earli', 698), ('earlier', 699), ('ease', 700), ('easi', 701), ('easili', 702), ('east', 703), ('ebola', 704), ('econom', 705), ('economi', 706), ('edge', 707), ('edit', 708), ('effect', 709), ('effective', 710), ('effector', 711), ('efficaci', 712), ('effici', 713), ('effort', 714), ('eg', 715), ('egg', 716), ('eight', 717), ('either', 718), ('elderli', 719), ('electron', 720), ('electrostat', 721), ('element', 722), ('elev', 723), ('elicit', 724), ('elimin', 725), ('else', 726), ('elsewhere', 727), ('elut', 728), ('embed', 729), ('emerg', 730), ('emerge', 731), ('emergence', 732), ('emphasize', 733), ('empir', 734), ('employ', 735), ('empti', 736), ('enabl', 737), ('enable', 738), ('encod', 739), ('encode', 740), ('encourag', 741), ('end', 742), ('endocytosi', 743), ('endogen', 744), ('energi', 745), ('engag', 746), ('engin', 747), ('enhanc', 748), ('enhance', 749), ('enough', 750), ('enrich', 751), ('ensemble', 752), ('ensur', 753), ('ensure', 754), ('enter', 755), ('entir', 756), ('entire', 757), ('entri', 758), ('envelop', 759), ('envelope', 760), ('environ', 761), ('environment', 762), ('envis', 763), ('enzymat', 764), ('enzyme', 765), ('epidem', 766), ('epidemiolog', 767), ('epidemiologist', 768), ('epitheli', 769), ('epitope', 770), ('eq', 771), ('equal', 772), ('equat', 773), ('equilibr', 774), ('equilibrium', 775), ('equip', 776), ('equival', 777), ('erad', 778), ('error', 779), ('especi', 780), ('essenti', 781), ('establish', 782), ('estim', 783), ('estimate', 784), ('etc', 785), ('ethic', 786), ('etiolog', 787), ('europe', 788), ('european', 789), ('evalu', 790), ('evaluate', 791), ('even', 792), ('event', 793), ('eventu', 794), ('ever', 795), ('everi', 796), ('evid', 797), ('evidence', 798), ('evolut', 799), ('evolutionari', 800), ('evolv', 801), ('exact', 802), ('exactli', 803), ('examin', 804), ('examine', 805), ('example', 806), ('exceed', 807), ('excel', 808), ('except', 809), ('excess', 810), ('excessive', 811), ('exclud', 812), ('exclude', 813), ('exclus', 814), ('exhibit', 815), ('exist', 816), ('existence', 817), ('exit', 818), ('exogen', 819), ('exon', 820), ('expand', 821), ('expans', 822), ('expect', 823), ('experi', 824), ('experienc', 825), ('experience', 826), ('experiment', 827), ('expert', 828), ('explain', 829), ('explan', 830), ('explicitli', 831), ('exploit', 832), ('explor', 833), ('explore', 834), ('exponenti', 835), ('expos', 836), ('exposed', 837), ('exposure', 838), ('express', 839), ('extend', 840), ('extens', 841), ('extensive', 842), ('extent', 843), ('extern', 844), ('extra', 845), ('extracellular', 846), ('extract', 847), ('extrem', 848), ('extreme', 849), ('f1', 850), ('face', 851), ('facil', 852), ('facilit', 853), ('facilitate', 854), ('fact', 855), ('factor', 856), ('fail', 857), ('failure', 858), ('fair', 859), ('fairli', 860), ('fall', 861), ('false', 862), ('famili', 863), ('far', 864), ('fashion', 865), ('fast', 866), ('faster', 867), ('fatal', 868), ('favor', 869), ('feasible', 870), ('feature', 871), ('feb', 872), ('februari', 873), ('feder', 874), ('feed', 875), ('female', 876), ('fetal', 877), ('fever', 878), ('fewer', 879), ('field', 880), ('fifth', 881), ('fig', 882), ('fig1', 883), ('figure', 884), ('file', 885), ('fill', 886), ('filter', 887), ('final', 888), ('financi', 889), ('find', 890), ('finish', 891), ('firstli', 892), ('fisher', 893), ('fit', 894), ('five', 895), ('fix', 896), ('flag', 897), ('flatten', 898), ('flexibl', 899), ('flow', 900), ('flu', 901), ('fluctuat', 902), ('fluid', 903), ('fluorescence', 904), ('focu', 905), ('focus', 906), ('fold', 907), ('follow', 908), ('food', 909), ('force', 910), ('forecast', 911), ('form', 912), ('format', 913), ('former', 914), ('formul', 915), ('formula', 916), ('forward', 917), ('found', 918), ('foundat', 919), ('four', 920), ('fourth', 921), ('fraction', 922), ('fragment', 923), ('frame', 924), ('framework', 925), ('france', 926), ('free', 927), ('freeli', 928), ('frequenc', 929), ('frequent', 930), ('fresh', 931), ('frozen', 932), ('fulfil', 933), ('full', 934), ('fulli', 935), ('function', 936), ('fundament', 937), ('furthermore', 938), ('fuse', 939), ('fusion', 940), ('future', 941), ('gain', 942), ('gamma', 943), ('gap', 944), ('gather', 945), ('gaussian', 946), ('gc', 947), ('ge', 948), ('gel', 949), ('genbank', 950), ('gender', 951), ('gene', 952), ('gener', 953), ('genera', 954), ('generate', 955), ('genet', 956), ('genom', 957), ('genome', 958), ('genotype', 959), ('genu', 960), ('geograph', 961), ('germani', 962), ('get', 963), ('gfp', 964), ('gisaid', 965), ('github', 966), ('githubcom', 967), ('give', 968), ('given', 969), ('global', 970), ('glycoprotein', 971), ('go', 972), ('goal', 973), ('good', 974), ('google', 975), ('govern', 976), ('grade', 977), ('gradual', 978), ('grant', 979), ('granular', 980), ('graph', 981), ('graphpad', 982), ('gray', 983), ('great', 984), ('greater', 985), ('greatli', 986), ('green', 987), ('ground', 988), ('group', 989), ('grow', 990), ('growth', 991), ('guid', 992), ('guidance', 993), ('guide', 994), ('guideline', 995), ('h1n1', 996), ('half', 997), ('hand', 998), ('happen', 999), ('harbor', 1000), ('hard', 1001), ('harvest', 1002), ('hcov', 1003), ('head', 1004), ('health', 1005), ('healthcare', 1006), ('healthi', 1007), ('heat', 1008), ('heavi', 1009), ('heavili', 1010), ('held', 1011), ('helic', 1012), ('helix', 1013), ('help', 1014), ('hence', 1015), ('hepat', 1016), ('herd', 1017), ('heterogen', 1018), ('higher', 1019), ('highest', 1020), ('highli', 1021), ('highlight', 1022), ('histogram', 1023), ('histor', 1024), ('histori', 1025), ('hit', 1026), ('hiv', 1027), ('hku1', 1028), ('hold', 1029), ('holiday', 1030), ('home', 1031), ('homogen', 1032), ('homolog', 1033), ('hong', 1034), ('hope', 1035), ('hopkin', 1036), ('hospit', 1037), ('host', 1038), ('hour', 1039), ('household', 1040), ('http', 1041), ('hub', 1042), ('hubei', 1043), ('huge', 1044), ('human', 1045), ('humid', 1046), ('hundr', 1047), ('hydrogen', 1048), ('hydrophob', 1049), ('hypothes', 1050), ('hypothesi', 1051), ('hypothesize', 1052), ('icu', 1053), ('id', 1054), ('idea', 1055), ('ideal', 1056), ('ident', 1057), ('identif', 1058), ('identifi', 1059), ('ie', 1060), ('ignor', 1061), ('ii', 1062), ('iii', 1063), ('il', 1064), ('ill', 1065), ('illumina', 1066), ('illustr', 1067), ('imag', 1068), ('image', 1069), ('immedi', 1070), ('immediate', 1071), ('immun', 1072), ('immune', 1073), ('immunodefici', 1074), ('impact', 1075), ('implement', 1076), ('impli', 1077), ('implic', 1078), ('implicitli', 1079), ('import', 1080), ('importance', 1081), ('importantli', 1082), ('impos', 1083), ('impossible', 1084), ('improv', 1085), ('improve', 1086), ('incidence', 1087), ('includ', 1088), ('include', 1089), ('inclus', 1090), ('incomplete', 1091), ('incorpor', 1092), ('increas', 1093), ('increase', 1094), ('increasingli', 1095), ('increment', 1096), ('incub', 1097), ('inde', 1098), ('independ', 1099), ('index', 1100), ('indic', 1101), ('indicate', 1102), ('indicative', 1103), ('indirect', 1104), ('individu', 1105), ('induc', 1106), ('induce', 1107), ('induct', 1108), ('industri', 1109), ('infecti', 1110), ('infectious', 1111), ('infer', 1112), ('inference', 1113), ('inflamm', 1114), ('inflammatori', 1115), ('influenc', 1116), ('influence', 1117), ('influenza', 1118), ('inform', 1119), ('informative', 1120), ('inher', 1121), ('inhibit', 1122), ('inhibitor', 1123), ('initi', 1124), ('initiative', 1125), ('inject', 1126), ('injuri', 1127), ('input', 1128), ('insert', 1129), ('inside', 1130), ('insight', 1131), ('inspect', 1132), ('instance', 1133), ('instead', 1134), ('institut', 1135), ('institute', 1136), ('instruct', 1137), ('instrument', 1138), ('insuffici', 1139), ('integr', 1140), ('intend', 1141), ('intens', 1142), ('intensive', 1143), ('inter', 1144), ('interact', 1145), ('interactive', 1146), ('interest', 1147), ('interestingli', 1148), ('interface', 1149), ('interferon', 1150), ('intern', 1151), ('interpret', 1152), ('interquartile', 1153), ('interrupt', 1154), ('interv', 1155), ('intervent', 1156), ('intracellular', 1157), ('intrins', 1158), ('introduc', 1159), ('introduce', 1160), ('introduct', 1161), ('invas', 1162), ('investig', 1163), ('investigate', 1164), ('invitrogen', 1165), ('involv', 1166), ('involve', 1167), ('ion', 1168), ('irrespective', 1169), ('isol', 1170), ('isolate', 1171), ('issue', 1172), ('itali', 1173), ('italian', 1174), ('iter', 1175), ('iv', 1176), ('jan', 1177), ('januari', 1178), ('japan', 1179), ('john', 1180), ('journal', 1181), ('june', 1182), ('justifi', 1183), ('keep', 1184), ('kept', 1185), ('key', 1186), ('keyword', 1187), ('kidney', 1188), ('kinase', 1189), ('kind', 1190), ('kinet', 1191), ('kingdom', 1192), ('kit', 1193), ('know', 1194), ('knowledge', 1195), ('known', 1196), ('kong', 1197), ('korea', 1198), ('lab', 1199), ('label', 1200), ('labor', 1201), ('laboratori', 1202), ('laboratory', 1203), ('lack', 1204), ('language', 1205), ('larg', 1206), ('large', 1207), ('larger', 1208), ('largest', 1209), ('last', 1210), ('lastli', 1211), ('late', 1212), ('latenc', 1213), ('latent', 1214), ('later', 1215), ('latest', 1216), ('latter', 1217), ('layer', 1218), ('le', 1219), ('lead', 1220), ('learn', 1221), ('least', 1222), ('leav', 1223), ('led', 1224), ('left', 1225), ('length', 1226), ('let', 1227), ('level', 1228), ('li', 1229), ('librari', 1230), ('lie', 1231), ('life', 1232), ('ligand', 1233), ('ligat', 1234), ('light', 1235), ('like', 1236), ('likelihood', 1237), ('limit', 1238), ('line', 1239), ('lineage', 1240), ('linear', 1241), ('link', 1242), ('linkage', 1243), ('lipid', 1244), ('liquid', 1245), ('list', 1246), ('literature', 1247), ('little', 1248), ('liu', 1249), ('live', 1250), ('liver', 1251), ('load', 1252), ('loc', 1253), ('local', 1254), ('locat', 1255), ('lock', 1256), ('lockdown', 1257), ('locu', 1258), ('log', 1259), ('logarithm', 1260), ('logist', 1261), ('long', 1262), ('longer', 1263), ('look', 1264), ('loop', 1265), ('loss', 1266), ('lot', 1267), ('low', 1268), ('lower', 1269), ('lowest', 1270), ('luciferase', 1271), ('lung', 1272), ('lymphocyte', 1273), ('lymphopenia', 1274), ('lysi', 1275), ('machine', 1276), ('made', 1277), ('mafft', 1278), ('magnitude', 1279), ('main', 1280), ('mainland', 1281), ('mainli', 1282), ('maintain', 1283), ('maintenance', 1284), ('major', 1285), ('mak', 1286), ('make', 1287), ('maker', 1288), ('male', 1289), ('mammal', 1290), ('mammalian', 1291), ('manag', 1292), ('manage', 1293), ('mani', 1294), ('manifest', 1295), ('mann', 1296), ('manner', 1297), ('manual', 1298), ('manufactur', 1299), ('manuscript', 1300), ('map', 1301), ('mar', 1302), ('march', 1303), ('mark', 1304), ('marker', 1305), ('market', 1306), ('mask', 1307), ('mass', 1308), ('massive', 1309), ('match', 1310), ('materi', 1311), ('mathemat', 1312), ('matric', 1313), ('matrix', 1314), ('mature', 1315), ('max', 1316), ('maxim', 1317), ('maximum', 1318), ('mean', 1319), ('meanwhile', 1320), ('measur', 1321), ('measure', 1322), ('mechan', 1323), ('mechanist', 1324), ('medi', 1325), ('media', 1326), ('median', 1327), ('medianday', 1328), ('mediat', 1329), ('mediate', 1330), ('medic', 1331), ('medicine', 1332), ('medium', 1333), ('meet', 1334), ('member', 1335), ('membrane', 1336), ('ment', 1337), ('mention', 1338), ('mer', 1339), ('merg', 1340), ('mers', 1341), ('metapopul', 1342), ('method', 1343), ('methodolog', 1344), ('methyl', 1345), ('metric', 1346), ('mg', 1347), ('microscope', 1348), ('microscopi', 1349), ('middle', 1350), ('might', 1351), ('migrat', 1352), ('mild', 1353), ('million', 1354), ('mimic', 1355), ('min', 1356), ('mind', 1357), ('mini', 1358), ('minim', 1359), ('minimize', 1360), ('minimum', 1361), ('ministri', 1362), ('minor', 1363), ('minute', 1364), ('mismatch', 1365), ('miss', 1366), ('mitig', 1367), ('mitigate', 1368), ('mix', 1369), ('mixture', 1370), ('ml', 1371), ('mm', 1372), ('mobil', 1373), ('mobile', 1374), ('mode', 1375), ('model', 1376), ('moderate', 1377), ('modif', 1378), ('modifi', 1379), ('module', 1380), ('mol', 1381), ('molecular', 1382), ('molecule', 1383), ('moment', 1384), ('monitor', 1385), ('month', 1386), ('morbid', 1387), ('moreov', 1388), ('mortal', 1389), ('mostli', 1390), ('motif', 1391), ('motiv', 1392), ('mouse', 1393), ('move', 1394), ('movement', 1395), ('mrna', 1396), ('much', 1397), ('multi', 1398), ('multiple', 1399), ('multiplex', 1400), ('multipli', 1401), ('municip', 1402), ('muscle', 1403), ('must', 1404), ('mutant', 1405), ('mutat', 1406), ('nacl', 1407), ('name', 1408), ('nasal', 1409), ('nation', 1410), ('native', 1411), ('natur', 1412), ('nature', 1413), ('nc', 1414), ('ncbi', 1415), ('ncintern', 1416), ('ncov', 1417), ('nd', 1418), ('ndintern', 1419), ('near', 1420), ('nearli', 1421), ('neb', 1422), ('necessari', 1423), ('necessarili', 1424), ('need', 1425), ('neg', 1426), ('negative', 1427), ('negligible', 1428), ('neighbor', 1429), ('neither', 1430), ('net', 1431), ('network', 1432), ('neutr', 1433), ('neutral', 1434), ('neutralize', 1435), ('never', 1436), ('nevertheless', 1437), ('new', 1438), ('newli', 1439), ('news', 1440), ('next', 1441), ('ng', 1442), ('nih', 1443), ('nine', 1444), ('nl63', 1445), ('nm', 1446), ('node', 1447), ('noise', 1448), ('non', 1449), ('none', 1450), ('nonetheless', 1451), ('norm', 1452), ('normal', 1453), ('northern', 1454), ('notabl', 1455), ('notable', 1456), ('note', 1457), ('notic', 1458), ('notice', 1459), ('noticeable', 1460), ('novel', 1461), ('novo', 1462), ('nt', 1463), ('nuclear', 1464), ('nuclease', 1465), ('nucleic', 1466), ('nucleotide', 1467), ('nucleu', 1468), ('numer', 1469), ('object', 1470), ('objective', 1471), ('observ', 1472), ('observe', 1473), ('obtain', 1474), ('obviou', 1475), ('oc43', 1476), ('occasion', 1477), ('occup', 1478), ('occupi', 1479), ('occur', 1480), ('occurrence', 1481), ('odd', 1482), ('of', 1483), ('off', 1484), ('offer', 1485), ('offici', 1486), ('often', 1487), ('old', 1488), ('omit', 1489), ('on', 1490), ('ongo', 1491), ('online', 1492), ('onset', 1493), ('onto', 1494), ('onward', 1495), ('open', 1496), ('oper', 1497), ('opinion', 1498), ('opportun', 1499), ('oppos', 1500), ('opposite', 1501), ('optim', 1502), ('optimist', 1503), ('option', 1504), ('or', 1505), ('order', 1506), ('ordinari', 1507), ('orf', 1508), ('organ', 1509), ('organis', 1510), ('orient', 1511), ('origin', 1512), ('originate', 1513), ('other', 1514), ('otherwise', 1515), ('out', 1516), ('outbreak', 1517), ('outcome', 1518), ('outlin', 1519), ('outperform', 1520), ('output', 1521), ('outside', 1522), ('over', 1523), ('overal', 1524), ('overcome', 1525), ('overexpress', 1526), ('overlap', 1527), ('overnight', 1528), ('overwhelm', 1529), ('oxford', 1530), ('oxygen', 1531), ('package', 1532), ('page', 1533), ('pair', 1534), ('pairwise', 1535), ('pandem', 1536), ('panel', 1537), ('panic', 1538), ('paper', 1539), ('parallel', 1540), ('paramet', 1541), ('parent', 1542), ('part', 1543), ('partial', 1544), ('particip', 1545), ('particle', 1546), ('particular', 1547), ('particularli', 1548), ('partit', 1549), ('partner', 1550), ('pass', 1551), ('passeng', 1552), ('past', 1553), ('path', 1554), ('pathogen', 1555), ('pathogenesi', 1556), ('patholog', 1557), ('pathway', 1558), ('patient', 1559), ('pattern', 1560), ('pb', 1561), ('pcr', 1562), ('pdb', 1563), ('peak', 1564), ('pearson', 1565), ('pellet', 1566), ('people', 1567), ('peptide', 1568), ('per', 1569), ('percent', 1570), ('percentage', 1571), ('perfectli', 1572), ('perform', 1573), ('performance', 1574), ('perhap', 1575), ('period', 1576), ('peripher', 1577), ('perman', 1578), ('permiss', 1579), ('perpetu', 1580), ('persist', 1581), ('person', 1582), ('perspective', 1583), ('ph', 1584), ('pharmaceut', 1585), ('phase', 1586), ('phenomenon', 1587), ('phenotype', 1588), ('phone', 1589), ('phosphate', 1590), ('phylogenet', 1591), ('phylogeni', 1592), ('physic', 1593), ('physiolog', 1594), ('picture', 1595), ('pipeline', 1596), ('place', 1597), ('plan', 1598), ('plasma', 1599), ('plasmid', 1600), ('plate', 1601), ('platform', 1602), ('plausible', 1603), ('play', 1604), ('plot', 1605), ('plu', 1606), ('pneumonia', 1607), ('point', 1608), ('polici', 1609), ('policymak', 1610), ('poly', 1611), ('polymerase', 1612), ('polyprotein', 1613), ('pool', 1614), ('poor', 1615), ('poorli', 1616), ('popul', 1617), ('population', 1618), ('portion', 1619), ('pose', 1620), ('posit', 1621), ('positive', 1622), ('possibl', 1623), ('possible', 1624), ('post', 1625), ('posterior', 1626), ('potent', 1627), ('potenti', 1628), ('power', 1629), ('practic', 1630), ('practice', 1631), ('pre', 1632), ('precaut', 1633), ('preced', 1634), ('precis', 1635), ('precise', 1636), ('predict', 1637), ('predictive', 1638), ('predictor', 1639), ('prefer', 1640), ('preference', 1641), ('preferenti', 1642), ('preliminari', 1643), ('preliminarili', 1644), ('prepar', 1645), ('prepare', 1646), ('presence', 1647), ('present', 1648), ('pressure', 1649), ('presum', 1650), ('preval', 1651), ('prevalence', 1652), ('prevent', 1653), ('previou', 1654), ('previous', 1655), ('primari', 1656), ('primarili', 1657), ('prime', 1658), ('primer', 1659), ('prior', 1660), ('prioriti', 1661), ('prism', 1662), ('pro', 1663), ('probabl', 1664), ('probable', 1665), ('probe', 1666), ('problem', 1667), ('procedure', 1668), ('proceed', 1669), ('process', 1670), ('produc', 1671), ('produce', 1672), ('product', 1673), ('profession', 1674), ('profil', 1675), ('profile', 1676), ('program', 1677), ('progress', 1678), ('project', 1679), ('prolifer', 1680), ('prolong', 1681), ('promis', 1682), ('promot', 1683), ('promote', 1684), ('prompt', 1685), ('prone', 1686), ('pronounc', 1687), ('proof', 1688), ('propag', 1689), ('proper', 1690), ('properli', 1691), ('properti', 1692), ('proport', 1693), ('propos', 1694), ('propose', 1695), ('protease', 1696), ('protect', 1697), ('protective', 1698), ('protein', 1699), ('protocol', 1700), ('prove', 1701), ('proven', 1702), ('provid', 1703), ('provide', 1704), ('province', 1705), ('proxi', 1706), ('proxim', 1707), ('public', 1708), ('publicli', 1709), ('publish', 1710), ('pulmonari', 1711), ('pure', 1712), ('purif', 1713), ('purifi', 1714), ('purpose', 1715), ('put', 1716), ('putative', 1717), ('qiagen', 1718), ('qpcr', 1719), ('qualiti', 1720), ('quantif', 1721), ('quantifi', 1722), ('quantit', 1723), ('quantitative', 1724), ('quantiti', 1725), ('quarantin', 1726), ('quarantine', 1727), ('queri', 1728), ('question', 1729), ('quick', 1730), ('quickli', 1731), ('quite', 1732), ('r0', 1733), ('rabbit', 1734), ('rais', 1735), ('raise', 1736), ('random', 1737), ('randomli', 1738), ('rang', 1739), ('range', 1740), ('rank', 1741), ('rapid', 1742), ('rapidli', 1743), ('rare', 1744), ('rate', 1745), ('rather', 1746), ('ratio', 1747), ('ration', 1748), ('raw', 1749), ('ray', 1750), ('rbd', 1751), ('rdrp', 1752), ('re', 1753), ('reach', 1754), ('reaction', 1755), ('reactive', 1756), ('read', 1757), ('readili', 1758), ('reagent', 1759), ('real', 1760), ('realist', 1761), ('reason', 1762), ('reasonable', 1763), ('receiv', 1764), ('recent', 1765), ('receptor', 1766), ('recogn', 1767), ('recognit', 1768), ('recognize', 1769), ('recombin', 1770), ('recommend', 1771), ('reconstruct', 1772), ('record', 1773), ('recov', 1774), ('recoveri', 1775), ('recruit', 1776), ('red', 1777), ('reduc', 1778), ('reduce', 1779), ('reduct', 1780), ('refer', 1781), ('reference', 1782), ('refin', 1783), ('reflect', 1784), ('regard', 1785), ('regardless', 1786), ('regime', 1787), ('region', 1788), ('regress', 1789), ('regul', 1790), ('regular', 1791), ('regulate', 1792), ('regulatori', 1793), ('rel', 1794), ('relat', 1795), ('relationship', 1796), ('relative', 1797), ('relax', 1798), ('releas', 1799), ('release', 1800), ('relev', 1801), ('reli', 1802), ('reliabl', 1803), ('reliable', 1804), ('remain', 1805), ('remark', 1806), ('remov', 1807), ('remove', 1808), ('repeat', 1809), ('replac', 1810), ('replic', 1811), ('replicate', 1812), ('repositori', 1813), ('repres', 1814), ('represent', 1815), ('representative', 1816), ('reproduc', 1817), ('reproduct', 1818), ('reproductive', 1819), ('repurpos', 1820), ('request', 1821), ('requir', 1822), ('require', 1823), ('research', 1824), ('resemble', 1825), ('reserv', 1826), ('resid', 1827), ('residu', 1828), ('residue', 1829), ('resistance', 1830), ('resolut', 1831), ('resolv', 1832), ('resource', 1833), ('respect', 1834), ('respective', 1835), ('respiratori', 1836), ('respond', 1837), ('respons', 1838), ('response', 1839), ('responsible', 1840), ('rest', 1841), ('restrict', 1842), ('retain', 1843), ('retriev', 1844), ('retrospective', 1845), ('return', 1846), ('reuse', 1847), ('reveal', 1848), ('revers', 1849), ('reverse', 1850), ('right', 1851), ('rigor', 1852), ('rise', 1853), ('risk', 1854), ('rna', 1855), ('rnai', 1856), ('rnase', 1857), ('robust', 1858), ('role', 1859), ('room', 1860), ('root', 1861), ('roughli', 1862), ('route', 1863), ('routine', 1864), ('rt', 1865), ('rule', 1866), ('run', 1867), ('s1', 1868), ('s10', 1869), ('s11', 1870), ('s2', 1871), ('s3', 1872), ('s4', 1873), ('s4a', 1874), ('s5', 1875), ('s5a', 1876), ('s6', 1877), ('s8', 1878), ('s9', 1879), ('safe', 1880), ('said', 1881), ('sampl', 1882), ('sample', 1883), ('sar', 1884), ('sars', 1885), ('satur', 1886), ('save', 1887), ('say', 1888), ('scaffold', 1889), ('scale', 1890), ('scan', 1891), ('scatter', 1892), ('scenario', 1893), ('schemat', 1894), ('scheme', 1895), ('school', 1896), ('science', 1897), ('scientif', 1898), ('scientist', 1899), ('scope', 1900), ('score', 1901), ('screen', 1902), ('script', 1903), ('sd', 1904), ('se', 1905), ('search', 1906), ('season', 1907), ('second', 1908), ('secondari', 1909), ('secondli', 1910), ('section', 1911), ('see', 1912), ('seed', 1913), ('seek', 1914), ('seem', 1915), ('seen', 1916), ('seir', 1917), ('select', 1918), ('self', 1919), ('sense', 1920), ('sensit', 1921), ('sensitive', 1922), ('separ', 1923), ('separate', 1924), ('sepsi', 1925), ('seq', 1926), ('sequenc', 1927), ('sequence', 1928), ('sequenti', 1929), ('seri', 1930), ('serial', 1931), ('seriou', 1932), ('serum', 1933), ('serv', 1934), ('serve', 1935), ('server', 1936), ('service', 1937), ('set', 1938), ('seven', 1939), ('seventh', 1940), ('sever', 1941), ('severe', 1942), ('sex', 1943), ('shanghai', 1944), ('shape', 1945), ('share', 1946), ('shed', 1947), ('sheet', 1948), ('shift', 1949), ('ship', 1950), ('short', 1951), ('shorter', 1952), ('shown', 1953), ('shut', 1954), ('shutdown', 1955), ('sick', 1956), ('side', 1957), ('sigma', 1958), ('sign', 1959), ('signal', 1960), ('signific', 1961), ('significance', 1962), ('significantli', 1963), ('silenc', 1964), ('similar', 1965), ('similarli', 1966), ('simple', 1967), ('simpli', 1968), ('simplic', 1969), ('simplifi', 1970), ('simul', 1971), ('simulate', 1972), ('simultan', 1973), ('since', 1974), ('singapore', 1975), ('single', 1976), ('sir', 1977), ('sirna', 1978), ('site', 1979), ('situat', 1980), ('six', 1981), ('size', 1982), ('skew', 1983), ('slightli', 1984), ('slow', 1985), ('slower', 1986), ('small', 1987), ('smaller', 1988), ('smallest', 1989), ('smith', 1990), ('smooth', 1991), ('social', 1992), ('societ', 1993), ('societi', 1994), ('sodium', 1995), ('software', 1996), ('sole', 1997), ('solid', 1998), ('solut', 1999), ('solv', 2000), ('solve', 2001), ('solvent', 2002), ('soon', 2003), ('sophist', 2004), ('sort', 2005), ('sought', 2006), ('source', 2007), ('south', 2008), ('southern', 2009), ('space', 2010), ('spain', 2011), ('span', 2012), ('spatial', 2013), ('special', 2014), ('specie', 2015), ('specif', 2016), ('specifi', 2017), ('specimen', 2018), ('spectrum', 2019), ('specul', 2020), ('speculate', 2021), ('speed', 2022), ('spent', 2023), ('spike', 2024), ('split', 2025), ('sporad', 2026), ('spread', 2027), ('squar', 2028), ('square', 2029), ('stabil', 2030), ('stable', 2031), ('staff', 2032), ('stage', 2033), ('stain', 2034), ('stand', 2035), ('standard', 2036), ('start', 2037), ('state', 2038), ('statist', 2039), ('statu', 2040), ('stay', 2041), ('stem', 2042), ('step', 2043), ('sterile', 2044), ('still', 2045), ('stimul', 2046), ('stochast', 2047), ('stock', 2048), ('stop', 2049), ('store', 2050), ('storm', 2051), ('straightforward', 2052), ('strain', 2053), ('strand', 2054), ('strategi', 2055), ('strength', 2056), ('stress', 2057), ('stretch', 2058), ('strict', 2059), ('strike', 2060), ('string', 2061), ('stringent', 2062), ('strong', 2063), ('stronger', 2064), ('strongest', 2065), ('strongli', 2066), ('structur', 2067), ('structure', 2068), ('student', 2069), ('sub', 2070), ('subgroup', 2071), ('subject', 2072), ('submit', 2073), ('subsequ', 2074), ('subset', 2075), ('substanti', 2076), ('substitut', 2077), ('substrate', 2078), ('subunit', 2079), ('success', 2080), ('suffer', 2081), ('suffici', 2082), ('suggest', 2083), ('suit', 2084), ('suitable', 2085), ('sum', 2086), ('summar', 2087), ('summari', 2088), ('summer', 2089), ('sun', 2090), ('superior', 2091), ('supernat', 2092), ('supplement', 2093), ('supplementari', 2094), ('suppli', 2095), ('support', 2096), ('suppress', 2097), ('sure', 2098), ('surface', 2099), ('surprisingli', 2100), ('surround', 2101), ('surveillance', 2102), ('survey', 2103), ('surviv', 2104), ('suscept', 2105), ('susceptible', 2106), ('suspect', 2107), ('sustain', 2108), ('sustainable', 2109), ('swab', 2110), ('switch', 2111), ('symptom', 2112), ('symptomat', 2113), ('syndrome', 2114), ('synonym', 2115), ('synthes', 2116), ('synthesi', 2117), ('synthet', 2118), ('system', 2119), ('systemat', 2120), ('t7', 2121), ('table', 2122), ('tag', 2123), ('tail', 2124), ('taiwan', 2125), ('take', 2126), ('taken', 2127), ('target', 2128), ('task', 2129), ('team', 2130), ('technic', 2131), ('technique', 2132), ('technolog', 2133), ('temperature', 2134), ('template', 2135), ('tempor', 2136), ('ten', 2137), ('tend', 2138), ('term', 2139), ('termin', 2140), ('terminu', 2141), ('test', 2142), ('text', 2143), ('th', 2144), ('thank', 2145), ('theoret', 2146), ('theori', 2147), ('therapeut', 2148), ('therapi', 2149), ('therebi', 2150), ('therefore', 2151), ('thermo', 2152), ('think', 2153), ('third', 2154), ('though', 2155), ('thought', 2156), ('thousand', 2157), ('threat', 2158), ('threaten', 2159), ('three', 2160), ('threshold', 2161), ('throat', 2162), ('throughout', 2163), ('throughput', 2164), ('thu', 2165), ('till', 2166), ('time', 2167), ('tissue', 2168), ('titer', 2169), ('title', 2170), ('to', 2171), ('togeth', 2172), ('toll', 2173), ('took', 2174), ('tool', 2175), ('toolkit', 2176), ('top', 2177), ('topic', 2178), ('topolog', 2179), ('total', 2180), ('totalcas', 2181), ('totalpati', 2182), ('touch', 2183), ('toward', 2184), ('town', 2185), ('toxic', 2186), ('trace', 2187), ('track', 2188), ('tract', 2189), ('trade', 2190), ('tradit', 2191), ('traffic', 2192), ('train', 2193), ('trajectori', 2194), ('transcrib', 2195), ('transcript', 2196), ('transcriptase', 2197), ('transduct', 2198), ('transfect', 2199), ('transfer', 2200), ('transform', 2201), ('transient', 2202), ('transit', 2203), ('transl', 2204), ('translat', 2205), ('translate', 2206), ('transmiss', 2207), ('transmit', 2208), ('transport', 2209), ('travel', 2210), ('treat', 2211), ('treatment', 2212), ('tree', 2213), ('trend', 2214), ('tri', 2215), ('trial', 2216), ('trigger', 2217), ('trim', 2218), ('trip', 2219), ('tropism', 2220), ('true', 2221), ('truncat', 2222), ('trypsin', 2223), ('tube', 2224), ('turn', 2225), ('twice', 2226), ('type', 2227), ('typic', 2228), ('ubiquit', 2229), ('uk', 2230), ('ultim', 2231), ('unable', 2232), ('uncertain', 2233), ('uncertainti', 2234), ('unchang', 2235), ('unclear', 2236), ('uncov', 2237), ('underestim', 2238), ('underli', 2239), ('understand', 2240), ('understood', 2241), ('undetect', 2242), ('unexpect', 2243), ('unfortun', 2244), ('uniform', 2245), ('unique', 2246), ('unit', 2247), ('univers', 2248), ('unknown', 2249), ('unless', 2250), ('unlik', 2251), ('unlike', 2252), ('unpreced', 2253), ('up', 2254), ('updat', 2255), ('update', 2256), ('upon', 2257), ('upper', 2258), ('upstream', 2259), ('urgent', 2260), ('us', 2261), ('usa', 2262), ('usage', 2263), ('user', 2264), ('usual', 2265), ('util', 2266), ('utilize', 2267), ('vaccin', 2268), ('vaccine', 2269), ('valid', 2270), ('validate', 2271), ('valu', 2272), ('valuable', 2273), ('value', 2274), ('vari', 2275), ('variabl', 2276), ('variable', 2277), ('variance', 2278), ('variant', 2279), ('variat', 2280), ('varieti', 2281), ('variou', 2282), ('vast', 2283), ('vector', 2284), ('ventil', 2285), ('verifi', 2286), ('version', 2287), ('versu', 2288), ('vertic', 2289), ('via', 2290), ('view', 2291), ('viral', 2292), ('virion', 2293), ('virolog', 2294), ('virtual', 2295), ('viru', 2296), ('virus', 2297), ('visible', 2298), ('visit', 2299), ('visual', 2300), ('visualize', 2301), ('vital', 2302), ('vitamin', 2303), ('vitro', 2304), ('vivo', 2305), ('volume', 2306), ('vs', 2307), ('vulnerable', 2308), ('wait', 2309), ('wang', 2310), ('want', 2311), ('ward', 2312), ('warn', 2313), ('wash', 2314), ('water', 2315), ('wave', 2316), ('way', 2317), ('weak', 2318), ('weather', 2319), ('web', 2320), ('website', 2321), ('week', 2322), ('weight', 2323), ('well', 2324), ('western', 2325), ('wherea', 2326), ('wherebi', 2327), ('whether', 2328), ('white', 2329), ('whitney', 2330), ('whole', 2331), ('whose', 2332), ('wide', 2333), ('widespread', 2334), ('wild', 2335), ('window', 2336), ('wish', 2337), ('within', 2338), ('withinhour', 2339), ('word', 2340), ('work', 2341), ('worker', 2342), ('workflow', 2343), ('workplace', 2344), ('world', 2345), ('worldwide', 2346), ('worse', 2347), ('worsen', 2348), ('worst', 2349), ('worth', 2350), ('would', 2351), ('write', 2352), ('written', 2353), ('wrong', 2354), ('wu', 2355), ('wuhan', 2356), ('wwwgisaidorg', 2357), ('wwwncbinlmnihgov', 2358), ('yang', 2359), ('year', 2360), ('yellow', 2361), ('yet', 2362), ('yield', 2363), ('york', 2364), ('young', 2365), ('younger', 2366), ('zero', 2367), ('zhang', 2368), ('zhou', 2369), ('zika', 2370), ('zone', 2371), ('zoonot', 2372), ('µg', 2373), ('µl', 2374)]\n",
      "[[0. 0. 0. ... 0. 0. 0.]\n",
      " [0. 0. 0. ... 0. 0. 0.]\n",
      " [0. 0. 0. ... 0. 0. 0.]\n",
      " ...\n",
      " [0. 0. 0. ... 0. 0. 0.]\n",
      " [0. 0. 0. ... 0. 0. 0.]\n",
      " [0. 0. 0. ... 0. 0. 0.]]\n"
     ]
    }
   ],
   "source": [
    "#create a tf-idf matrix for the words above\n",
    "v = TfidfVectorizer(sublinear_tf = True, min_df = 0.05,max_df = 0.8)\n",
    "#For the above vectorizer, the first paramater specifies that the 1+log(tf) is going to be used instead of tf.\n",
    "#The second parameter specifies that if a term appears in less than 5% of the docs, ignore it\n",
    "#The third parameter, considers word that do not occur in more than 80% of the corpus\n",
    "tfidf = v.fit_transform(stemmed_text) #fit and transform the stemmed texts which were previously cleaned.\n",
    "print(sorted(v.vocabulary_.items(), key=lambda x : x[1]))\n",
    "print(tfidf.toarray())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Enter your query: middle east coronavirus\n"
     ]
    }
   ],
   "source": [
    "#Next, the search engine will be implemented with the help of the tf-idf matrix created above and the cosine similarity\n",
    "pd.set_option('display.max_colwidth', -1)\n",
    "query = input(\"Enter your query: \") \n",
    "finalQuery=\"\" #create an empty string\n",
    "words = query.split(' ') #split the query on spaces\n",
    "for word in words: #for each word in the query\n",
    "    if(word!=\"\"): #if the word is not an empty string\n",
    "        if (word==\"coronaviruses\"):#since the stem and lemmatize functions do not recognize the word coronaviruses as the plural of coronavirus, a conidition was created\n",
    "            word =\"coronavirus\"\n",
    "        stem_lem_query = porter.stem(word)\n",
    "        finalQuery+=stem_lem_query+\" \" #stem/lemmatize the word and concat it to the string stemmed_val\n",
    "query = [finalQuery.strip()]# remove any trailing spaces from the query\n",
    "query_tfidf = v.transform(query)\n",
    "cosineSimilarities = cosine_similarity(query_tfidf, tfidf).flatten() #get the cosine similarity list \n",
    "answer_indicies = sorted(range(len(cosineSimilarities)), key=lambda i: cosineSimilarities[i])[-5:] #get the top 5 indices with the highest cosine similarity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Forecasting the Worldwide Spread of COVID-19 based on Logistic Model and SEIR Model</td>\n",
       "      <td>[{'text': 'Background: With the outbreak of coronavirus disease 2019 (COVID-19), a sudden case increase in late February 2020 led to deep concern globally. Italy, South Korea, Iran, France, Germany, Spain, the US and Japan are probably the countries with the most severe outbreaks.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Collecting epidemiological data and predicting epidemic trends are important for the development and measurement of public intervention strategies. Epidemic prediction results yielded by different mathematical models are inconsistent; therefore, we sought to compare different models and their prediction results to generate objective conclusions. : medRxiv preprint 2 logistic growth model, basic SEIR model and adjusted SEIR model were adopted for prediction.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Given that different model inputs may infer different model outputs, we implemented three model predictions with three scenarios of epidemic development.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': ': When comparing all 8 countries' short-term prediction results and peak predictions, the differences among the models were relatively large. The logistic growth model estimated a smaller epidemic size than the basic SERI model did; however, once we added parameters that considered the effects of public health interventions and control measures, the adjusted SERI model results demonstrated a considerably rapid deceleration of epidemic development. Our results demonstrated that contact rate, quarantine scale, and the initial quarantine time and length are important factors in controlling epidemic size and length.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'We demonstrated a comparative assessment of the predictions of the COVID-19 outbreak in eight high-risk countries using multiple methods. By forecasting epidemic size and peak time as well as simulating the effects of public health interventions, the intent of this paper is to help clarify the transmission dynamics of COVID-19 and recommend operation suggestions to slow down the epidemic. It is suggested that the quick detection of cases, sufficient implementation of quarantine and public self-protection behaviors are critical to slow down the epidemic.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>A novel coronavirus caused pneumonia cases in Wuhan, a city in Hubei Province in China, beginning at the end of 2019. Governments had implemented various measures to protect their cities or countries, such as traffic restrictions, quarantine requirements for travelers, and contact tracing, However, a large-scale global movement of the population still inevitably caused the rapid spread of the disease, resulting in an epidemic throughout China and worldwide. In February 2020, the World Health Organization named the disease COVID-19, which stands for coronavirus disease 2019 1 . The virus that causes COVID-19 is named severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2). The SARS-CoV-2 virus spreads mainly between people who are in close contact with one another and through respiratory droplets produced . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint .\\n\\nwhen an infected person coughs or sneezes 2 . The global spread of COVID-19 caused a surge in Asia, Europe, the Middle East and North America. As of March 11, 2020 , with the global risk continuously increasing, there were already more than 118,000 cases in 114 countries, and 4,291 people have lost their lives. The WHO has characterized COVID-19 as a pandemic 3 .\\n\\nWith the number of cases growing in more than 150 countries and regions, modeling the transmission dynamics and estimating the development of COVID-19 are crucial to providing decisional support for public health departments and healthcare policy makers. Mathematical models are widely used in evaluating epidemic transmissions, forecasting the trend of disease spread, and providing optimal intervention strategies and control measures. A considerable number of recent studies have contended to estimate the scale and severity of COVID-19, and several mathematical models and predicting approaches have attempted to estimate the transmission of COVID-19 [4] [5] [6] [7] [8] . The majority of the studies have estimated the basic reproductive number R 0 , a key parameter to evaluate the potential for COVID-19 transmission. However, different models often yield different conclusions in terms of differences in model structure and input parameters. It is imperative and critical to improve the early predictive and warning capabilities of potential models for the pandemic.\\n\\nIn the face of this new infectious disease and its complicated features with many unknown factors, single model estimations may infer biased results; therefore, we tried to make overall rigorous estimations by comparing different model results. To achieve an objective estimation, we investigated and implemented the two most common approaches and one extended approach: the logistic model, the susceptible-exposed-infected-removed (SEIR) model and the adjusted SEIR model. Countries had different initial times and levels of interventions and measures to reduce the risk of domestic secondary infections of COVID-19. We compared the models that have taken these effects into account or not, predicted the spread of the epidemic, and tried to compare different recommendations from three different models.\\n\\nWe collected data on the epidemic situation of COVID-19 in eight high-risk countries located on three continents and compared the results with those of the logistic model and the SEIR model with different parameter setting scenarios. Data are from the Coronavirus COVID-19 Global Cases published by the Center for Systems Science and Engineering (CSSE) of Johns Hopkins University 9 . We used the existing reported data from January 23 to March 20, 2020, for . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint observing, performing parameter estimation, and forecasting COVID-19 dynamics in different countries/regions.\\n\\nIn Scenario 1, we assume that the epidemic trend obeys a logistic growth curve. We used a logistic model to predict the disease trends. The logistic model's essence is that curve fitting and its prediction results highly depend on historical data. It has often been used in the prediction of epidemic dynamics in previous studies 4, 10,11 . Mathematically, the logistic model describes the dynamic evolution of infected individuals being controlled by the growth rate and population capacity. According to the following ordinary differential equation (a), we will obtain the logistic function (b). The model describes the dynamic evolution of the reported number of confirmed cases P being controlled by the growth rate r, and the initial value of P 0 is the confirmed number of cases when T=0. The maximum case volume in the environment is K, which is the limit that can be reached by increasing to the final value of P ሺ t ሻ , and r is the growth rate. We used the least squares method to fit the logistic growth function and then to predict the number of future confirmed cases. Since the case numbers reported at very early stages are usually inaccurate or missing, the initial date of the model was set as the day since the 100th confirmed case was reached.\\n\\nTo evaluate the logistic model's ability to predict infectious diseases such as COVID-19, we fit the logistic curve every 10 days since the day the 100th confirmed case was reached, and each time, we made 7-day predictions and used reported data for evaluation. In the experimental analysis, for each time of perdition, we obtained different result errors, as listed in Appendix A.\\n\\nAs the number of confirmed cases increases, the predicted following 7 days of infections and future peak size and peak time are constantly changing. In addition, the shape of the curve will . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint probably change due to exogenous effects, such as a new burst of infection, the implementation of control measures and public behaviors.\\n\\nBased on the epidemiological characteristics of COVID-19 infection, the SEIR model is more commonly adopted to study the dynamics of this disease. SEIR is a deterministic metapopulation transmission model that simulates each individual in the population as a separate compartment, with the assumption that each individual in the same compartment has the same characteristics.\\n\\nBy plugging in different settings of parameters, however, the models yield different results, and we compared their results to observe patterns of the COVID-19 spread under two different scenarios, namely, the basic SEIR (Scenario 2: without any interventions and measures) and the adjusted SEIR (Scenario 3: with strict interventions and measures).\\n\\nIn Scenario 2, we only used the basic SEIR model, and the population is divided into four classes: susceptible (S), exposed (E), infectious (I) and removed (R). The essence of the SEIR model is a system of ordinary differential equations over time. The disease trend it predicts only depends on parameters and the start time. The model is measured by the equation below, and the entire population was initially susceptible, with the assumption that all people have no immunity against COVID-19. The initial number of cases was collected from the reported data. Because reliable data are still scarce during the early days of a new outbreak, the initial date of the model was set as the day since the 100th confirmed case was reached for each country, which indicates different initial dates of the 8 observed countries.\\n\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity. is the rate at which exposed individuals develop into those with infections.\\n\\nIn Scenario 3, we further used an adjusted SEIR model for COVID-19 estimation. According to recent studies, some investigations was conducted to measure the contribution to epidemic dynamics by public health intervention factors, such as the government locking down cities, taking measures to track and quarantine people who have close contact with confirmed cases, advocating citizens to maintain social distance and wash their hands frequently, a conclusion was yield that it will limit the number of confirmed cases 96% fewer than expected in the absence of interventions by implementing multiple measures together and interactively 12 . Therefore, this adjusted SEIR model considered the contact rate and quarantined proportion of COVID-19\\n\\ntransmission and divided the population into seven classes: susceptible, exposed, infectious, removed, quarantined susceptible, quarantined exposed and quarantined infected. A fraction of the susceptible population was quarantined and identified as Sq, and a fraction of the exposed population was isolated and identified as Eq. We provide their detailed equation as follows 8 :\\n\\nCC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint All countries reduced social contact among its people, although by implementing multiple different control measures, we give a preliminary average estimation for the contact rate and quarantined proportions as model parameters. In addition, as the quarantine and control measures take effect after a period of increasing cases, it has typically surged into a considerable number; therefore, the initial dates of 8 observed countries used in the adjusted model were different, and they were defined as the dates when the countries' governments implemented strict interventions and control measures for a large-scale population. We assumed that the contact rate decreased since the government implemented these strict control measures. In terms of our previous study and related studies 8 , the contact rate was below 8 after a large-scale intervention was initiated.\\n\\nThe initial populations of each country were acquired from published data 13 , and the initial infected and recovered population was set based on the reported data 9 . Model parameters are estimated on the basis of fitting reported data from the initial date, and the probability of transmission per contact of each country was estimated using early-stage data from each country based on Monte Carlo simulation. We assumed that the median incubation period was 5-6 days (ranging from 0-14 days) based on the WHO report 14 , the quarantine proportion of uninfected susceptible individuals was 60%~80% of the population (under the assumption that a strict largescale quarantine policy is executed) with the period set as 28 days, and the mortality rate was derived from reported data of each country. Based on the above assumptions, we implemented the adjusted SEIR model to obtain the minimal estimation.\\n\\nTo disclose the epidemic growth potential of each country, we used two parameters for the assessment. First, the greater the growth rate of the logistic curves is, the faster the curve grows.\\n\\nSecond, the basic reproduction number R 0 15 . We estimated the early transmission R 0 of each country using the 10 days reported data since the 100th confirmed case was reached. We used the next-generation matrix to derive a formula for the R 0 16 , as follows:\\n\\nwhere β is the product of the population exposed to the infected population (k) and the probability of transmission (b).\\n\\nγ =1/D is the average rate of recovery or death in infected . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint populations, where D is the average duration of the infection. In terms of related studies for the COVID-19 infection assumption 7 , we set D=7 to obtain the R 0 estimation.\\n\\nWe summarized the epidemic curves of the 8 observed countries to help clarify the global trends and disclose the spread pattern of different countries. Cases increased rapidly in these countries, In Figure 2 , the death tolls in Italy due to COVID-19 reached 4032 as of March 20, and the mortality rate of Italy reached 8.57%. Iran's early negligence also caused a very high mortality rate of 40% at its early stage. Some countries also have high mortality rates, such as the US and Japan. However, South Korea and Germany controlled the mortality rate at approximately 1%, which is a positive indication.\\n\\nWhen contrasting the results from the three prediction models, we achieved quite different results for COVID-19 development, as shown in Table 1 and Table 2 . Because different models were built on different theories and assumptions, their output measurements were varied, the . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint cumulative number was used for the logistic model, and the active number was measured by SEIR models. The results disclosed the differences in the three mathematical models and further disclosed the prediction differences without/with the consideration of interventions.\\n\\nIn Scenario 1, based on the logistic model, we predicted the epidemic trends of the 8 countries. Table 1 and Appendix B list the detailed results of the prediction and growth trajectories of the 8 countries. For instance, the model placed the peak time of Italy as 50 days after its initial date of February 23, 2020, with a maximum number of infected individuals of approximately 97,442.\\n\\nThe infection of US was unable to be predicted based on a logistic model using reported data because it is exponentially growing, which makes it difficult to fit the logistic curve, which reflects that the US is in a stage of high spread of the disease.\\n\\nIn Scenario 2, the basic SEIR results showed that the confirmed cases will take 9-22 months (275-650 days) to reach their peak, and most of the population would eventually be infected over a long period of time if there are no control measures. Appendix C lists the detailed results of the basic SEIR model predictions of different countries. In addition, the active number of cases at the peak time will reach approximately 10-20% of these countries' populations, thus overloading the healthcare system, which is the worst possible scenario, as shown in Table 2 .\\n\\nIn Scenario 3, the adjusted SEIR model results show that under strict control measures, the number of active cases will reach a peak in 16-32 days (from early April to middle April 2020) after the initial date, when the number of cases reaches 100. Appendix D shows the detailed results of the adjusted SEIR model predictions of the 8 countries. Japan and South Korea took isolation measures when the disease was still in the early stage of transmission, its peak value was low, and the cumulative number of infected people was relatively small. Especially in Japan, our predicted peak value is less than 1560, which shows that the disease spread has been well controlled in the early stage. However, the transmission in European countries and the US is in the outbreak phase as of the date of our data collection. The adjusted SEIR model predicted that the peak values of Spain, Italy, Germany, France and other countries are between 10,420 and 85,750.\\n\\nAccording to our parameter estimation methods, we disclosed the dynamics of the number of cases. The Scenario 3 estimation is based on the strict quarantine assumption, and the results showed that implementing the control measures would decrease the epidemic peak significantly and bring it forward. This supports the conclusion that the contact rate is an important factor that reflects the effects of control measures; with the formulation and implementation of extreme epidemic prevention measures to reduce the rate of contact, the epidemic size and peak would . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint decrease. However, the epidemic still shows a long tail after the peak, and our study found that a longer quarantine period of susceptibility would reduce the long tail after the peak. Table 1 . Short-term epidemic predictions of the 8 countries Table 2 . Epidemic peak predictions of the 8 countries\\n\\nThe growth rates of the logistic curves are also listed as a spreading potential index for comparing the situations of the 8 countries in Table 3 . Our results suggest that these countries all have a high risk of rapid virus transmission except for Japan, where the spread is slowing down.\\n\\nOn the basis of evidence from previous transmission dynamics studies, the documented COVID- . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint 19 basic reproductive number (R 0 ) ranges from 2.0 to 4.9 17-20 . Our estimated R 0 range is between 1.687~3.864 (with the assumption of a 7-day mean infection period in terms of reported COVID-19 studies). \\n\\nAccording to the COVID-19 data tracking, these 8 countries have delayed the phase of preventing the epidemic and have indeed entered the outbreak phase with community-spread cases. Mortality rate analysis disclosed that undetected transmission events may have occurred in some countries. Seven of the eight countries have R0 levels above 3, which deserves our attention.\\n\\nThe high mortality was reported because the main reason for undetected transmission is probably that some of the cases are asymptomatic or lack testing kits. Our Scenario 2 and Scenario 3 are separately based on maximal assumption (without any interventions) and minimal assumption (with strict interventions). In Scenario 3, under the assumption that infected people are promptly tested and counted, the adjusted model suggested that the peak time will be reached after the strict implementation of two large-scale rounds of 14 days of quarantine. The decrease in the epidemic size in Japan and South Korea showed positive evidence, as shown in Table 1 .\\n\\nHowever, considering that each country has different cultures and healthcare situations and that the implementation of the policies and control measures are at different levels 21 , the actual situation of those countries should be between Scenario 2 and Scenario 3. It is worth pointing out that, compared with implementing no interventions, if governments take strict control measures to reduce the movement of the population and implement prompt diagnosis and isolation, the peak time will be reached, and the peak size would greatly decrease to a relatively low level in approximately 30 days; it is best triggered in the early stage of the epidemic; however, it will need to be maintained for several months until a vaccine becomes available. Conversely, as long . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint as the epidemic continues, there is still a possibility of future outbreaks if governments loosen interventions and allow people to return to close social distances.\\n\\nThe results of spreading potentials show that the R 0 of COVID-19 is high, which is similar to that of SARS viruses (R 0 values between 2.0 and 5.0 22 ), higher influenza viruses H1N1 (R 0 values between 1.2 and 3.7 23 ), and Ebola viruses (R 0 values between 1.34 and 3.65 24 ). COVID-19 is a highly contagious human-to-human transmission disease. The R 0 is expected to decrease substantially compared to values at the early stage after governments implement control measures; however, regardless of the kind of policy each country executes, each policy has its limitations in defending against COVID-19, and sustained transmission chains will occur until there is a vaccine or the until the virus disappears due to seasonal or population immunity 25 From the view of mathematical models, the SEIR model is designed for infectious disease estimation; however, the logistic growth model is designed to fit the development of the curves.\\n\\nThe logistic model may fit the existing data better when compared with the SEIR model, since it is trained from the existing data, but it cannot be accurately judged and incorporates infectious characteristics. Therefore, we believe that the logistic model is better for near-term predictions.\\n\\nOn the other hand, the SEIR model introduces more variables and factors by considering the interaction and association among multiple groups of people, and it is more reasonable than the logistic model as it follows the rules of infectious disease development, but the prediction results vary greatly with respect to different interventions and settings.\\n\\nThe study has some limitations. The mathematical models allow for the quick incorporation of multiple inputs to yield prediction results. However, this process involves making assumptions about uncertain factors; for example, it is difficult to determine the exact extent to which people follow the local government's quarantine policies or measures and engage in behaviors such as washing hands, using masks, and social distancing. This may affect the actual contact rate and the subsequent development of the epidemic. The models also lack enough data to estimate the quarantine proportions of a certain population. In fact, the evolution of the epidemic is quite . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint complicated, and our study has only taken into account a few factors. In addition, a lack of testing kits means many cases have not been tested in some countries, and without robust testing, the official number of cases is incomplete. When working with incomplete data, a small error in one factor can have an outsize effect.\\n\\nIn conclusion, our study demonstrated that reducing the contact rate is a key measure in controlling the spread of disease at an early stage, and spending enough time in quarantine would decrease the scale of cases after the peak. Therefore, implementing strong containment policies during the early spreading stages of COVID-19 and flattening the peak to avoid overloading the healthcare system should be listed as the main actions in these high-risk countries/regions. After the strict quarantine period, governments still need to raise public awareness of precautions and the importance of engaging in self-protective behaviors to bring the epidemic under control as a series of scattered events. . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint\\n\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                  title  \\\n",
       "11  Forecasting the Worldwide Spread of COVID-19 based on Logistic Model and SEIR Model   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  abstract  \\\n",
       "11  [{'text': 'Background: With the outbreak of coronavirus disease 2019 (COVID-19), a sudden case increase in late February 2020 led to deep concern globally. Italy, South Korea, Iran, France, Germany, Spain, the US and Japan are probably the countries with the most severe outbreaks.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Collecting epidemiological data and predicting epidemic trends are important for the development and measurement of public intervention strategies. Epidemic prediction results yielded by different mathematical models are inconsistent; therefore, we sought to compare different models and their prediction results to generate objective conclusions. : medRxiv preprint 2 logistic growth model, basic SEIR model and adjusted SEIR model were adopted for prediction.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Given that different model inputs may infer different model outputs, we implemented three model predictions with three scenarios of epidemic development.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': ': When comparing all 8 countries' short-term prediction results and peak predictions, the differences among the models were relatively large. The logistic growth model estimated a smaller epidemic size than the basic SERI model did; however, once we added parameters that considered the effects of public health interventions and control measures, the adjusted SERI model results demonstrated a considerably rapid deceleration of epidemic development. Our results demonstrated that contact rate, quarantine scale, and the initial quarantine time and length are important factors in controlling epidemic size and length.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'We demonstrated a comparative assessment of the predictions of the COVID-19 outbreak in eight high-risk countries using multiple methods. By forecasting epidemic size and peak time as well as simulating the effects of public health interventions, the intent of this paper is to help clarify the transmission dynamics of COVID-19 and recommend operation suggestions to slow down the epidemic. It is suggested that the quick detection of cases, sufficient implementation of quarantine and public self-protection behaviors are critical to slow down the epidemic.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       full_text  \n",
       "11  A novel coronavirus caused pneumonia cases in Wuhan, a city in Hubei Province in China, beginning at the end of 2019. Governments had implemented various measures to protect their cities or countries, such as traffic restrictions, quarantine requirements for travelers, and contact tracing, However, a large-scale global movement of the population still inevitably caused the rapid spread of the disease, resulting in an epidemic throughout China and worldwide. In February 2020, the World Health Organization named the disease COVID-19, which stands for coronavirus disease 2019 1 . The virus that causes COVID-19 is named severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2). The SARS-CoV-2 virus spreads mainly between people who are in close contact with one another and through respiratory droplets produced . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint .\\n\\nwhen an infected person coughs or sneezes 2 . The global spread of COVID-19 caused a surge in Asia, Europe, the Middle East and North America. As of March 11, 2020 , with the global risk continuously increasing, there were already more than 118,000 cases in 114 countries, and 4,291 people have lost their lives. The WHO has characterized COVID-19 as a pandemic 3 .\\n\\nWith the number of cases growing in more than 150 countries and regions, modeling the transmission dynamics and estimating the development of COVID-19 are crucial to providing decisional support for public health departments and healthcare policy makers. Mathematical models are widely used in evaluating epidemic transmissions, forecasting the trend of disease spread, and providing optimal intervention strategies and control measures. A considerable number of recent studies have contended to estimate the scale and severity of COVID-19, and several mathematical models and predicting approaches have attempted to estimate the transmission of COVID-19 [4] [5] [6] [7] [8] . The majority of the studies have estimated the basic reproductive number R 0 , a key parameter to evaluate the potential for COVID-19 transmission. However, different models often yield different conclusions in terms of differences in model structure and input parameters. It is imperative and critical to improve the early predictive and warning capabilities of potential models for the pandemic.\\n\\nIn the face of this new infectious disease and its complicated features with many unknown factors, single model estimations may infer biased results; therefore, we tried to make overall rigorous estimations by comparing different model results. To achieve an objective estimation, we investigated and implemented the two most common approaches and one extended approach: the logistic model, the susceptible-exposed-infected-removed (SEIR) model and the adjusted SEIR model. Countries had different initial times and levels of interventions and measures to reduce the risk of domestic secondary infections of COVID-19. We compared the models that have taken these effects into account or not, predicted the spread of the epidemic, and tried to compare different recommendations from three different models.\\n\\nWe collected data on the epidemic situation of COVID-19 in eight high-risk countries located on three continents and compared the results with those of the logistic model and the SEIR model with different parameter setting scenarios. Data are from the Coronavirus COVID-19 Global Cases published by the Center for Systems Science and Engineering (CSSE) of Johns Hopkins University 9 . We used the existing reported data from January 23 to March 20, 2020, for . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint observing, performing parameter estimation, and forecasting COVID-19 dynamics in different countries/regions.\\n\\nIn Scenario 1, we assume that the epidemic trend obeys a logistic growth curve. We used a logistic model to predict the disease trends. The logistic model's essence is that curve fitting and its prediction results highly depend on historical data. It has often been used in the prediction of epidemic dynamics in previous studies 4, 10,11 . Mathematically, the logistic model describes the dynamic evolution of infected individuals being controlled by the growth rate and population capacity. According to the following ordinary differential equation (a), we will obtain the logistic function (b). The model describes the dynamic evolution of the reported number of confirmed cases P being controlled by the growth rate r, and the initial value of P 0 is the confirmed number of cases when T=0. The maximum case volume in the environment is K, which is the limit that can be reached by increasing to the final value of P ሺ t ሻ , and r is the growth rate. We used the least squares method to fit the logistic growth function and then to predict the number of future confirmed cases. Since the case numbers reported at very early stages are usually inaccurate or missing, the initial date of the model was set as the day since the 100th confirmed case was reached.\\n\\nTo evaluate the logistic model's ability to predict infectious diseases such as COVID-19, we fit the logistic curve every 10 days since the day the 100th confirmed case was reached, and each time, we made 7-day predictions and used reported data for evaluation. In the experimental analysis, for each time of perdition, we obtained different result errors, as listed in Appendix A.\\n\\nAs the number of confirmed cases increases, the predicted following 7 days of infections and future peak size and peak time are constantly changing. In addition, the shape of the curve will . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint probably change due to exogenous effects, such as a new burst of infection, the implementation of control measures and public behaviors.\\n\\nBased on the epidemiological characteristics of COVID-19 infection, the SEIR model is more commonly adopted to study the dynamics of this disease. SEIR is a deterministic metapopulation transmission model that simulates each individual in the population as a separate compartment, with the assumption that each individual in the same compartment has the same characteristics.\\n\\nBy plugging in different settings of parameters, however, the models yield different results, and we compared their results to observe patterns of the COVID-19 spread under two different scenarios, namely, the basic SEIR (Scenario 2: without any interventions and measures) and the adjusted SEIR (Scenario 3: with strict interventions and measures).\\n\\nIn Scenario 2, we only used the basic SEIR model, and the population is divided into four classes: susceptible (S), exposed (E), infectious (I) and removed (R). The essence of the SEIR model is a system of ordinary differential equations over time. The disease trend it predicts only depends on parameters and the start time. The model is measured by the equation below, and the entire population was initially susceptible, with the assumption that all people have no immunity against COVID-19. The initial number of cases was collected from the reported data. Because reliable data are still scarce during the early days of a new outbreak, the initial date of the model was set as the day since the 100th confirmed case was reached for each country, which indicates different initial dates of the 8 observed countries.\\n\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity. is the rate at which exposed individuals develop into those with infections.\\n\\nIn Scenario 3, we further used an adjusted SEIR model for COVID-19 estimation. According to recent studies, some investigations was conducted to measure the contribution to epidemic dynamics by public health intervention factors, such as the government locking down cities, taking measures to track and quarantine people who have close contact with confirmed cases, advocating citizens to maintain social distance and wash their hands frequently, a conclusion was yield that it will limit the number of confirmed cases 96% fewer than expected in the absence of interventions by implementing multiple measures together and interactively 12 . Therefore, this adjusted SEIR model considered the contact rate and quarantined proportion of COVID-19\\n\\ntransmission and divided the population into seven classes: susceptible, exposed, infectious, removed, quarantined susceptible, quarantined exposed and quarantined infected. A fraction of the susceptible population was quarantined and identified as Sq, and a fraction of the exposed population was isolated and identified as Eq. We provide their detailed equation as follows 8 :\\n\\nCC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint All countries reduced social contact among its people, although by implementing multiple different control measures, we give a preliminary average estimation for the contact rate and quarantined proportions as model parameters. In addition, as the quarantine and control measures take effect after a period of increasing cases, it has typically surged into a considerable number; therefore, the initial dates of 8 observed countries used in the adjusted model were different, and they were defined as the dates when the countries' governments implemented strict interventions and control measures for a large-scale population. We assumed that the contact rate decreased since the government implemented these strict control measures. In terms of our previous study and related studies 8 , the contact rate was below 8 after a large-scale intervention was initiated.\\n\\nThe initial populations of each country were acquired from published data 13 , and the initial infected and recovered population was set based on the reported data 9 . Model parameters are estimated on the basis of fitting reported data from the initial date, and the probability of transmission per contact of each country was estimated using early-stage data from each country based on Monte Carlo simulation. We assumed that the median incubation period was 5-6 days (ranging from 0-14 days) based on the WHO report 14 , the quarantine proportion of uninfected susceptible individuals was 60%~80% of the population (under the assumption that a strict largescale quarantine policy is executed) with the period set as 28 days, and the mortality rate was derived from reported data of each country. Based on the above assumptions, we implemented the adjusted SEIR model to obtain the minimal estimation.\\n\\nTo disclose the epidemic growth potential of each country, we used two parameters for the assessment. First, the greater the growth rate of the logistic curves is, the faster the curve grows.\\n\\nSecond, the basic reproduction number R 0 15 . We estimated the early transmission R 0 of each country using the 10 days reported data since the 100th confirmed case was reached. We used the next-generation matrix to derive a formula for the R 0 16 , as follows:\\n\\nwhere β is the product of the population exposed to the infected population (k) and the probability of transmission (b).\\n\\nγ =1/D is the average rate of recovery or death in infected . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint populations, where D is the average duration of the infection. In terms of related studies for the COVID-19 infection assumption 7 , we set D=7 to obtain the R 0 estimation.\\n\\nWe summarized the epidemic curves of the 8 observed countries to help clarify the global trends and disclose the spread pattern of different countries. Cases increased rapidly in these countries, In Figure 2 , the death tolls in Italy due to COVID-19 reached 4032 as of March 20, and the mortality rate of Italy reached 8.57%. Iran's early negligence also caused a very high mortality rate of 40% at its early stage. Some countries also have high mortality rates, such as the US and Japan. However, South Korea and Germany controlled the mortality rate at approximately 1%, which is a positive indication.\\n\\nWhen contrasting the results from the three prediction models, we achieved quite different results for COVID-19 development, as shown in Table 1 and Table 2 . Because different models were built on different theories and assumptions, their output measurements were varied, the . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint cumulative number was used for the logistic model, and the active number was measured by SEIR models. The results disclosed the differences in the three mathematical models and further disclosed the prediction differences without/with the consideration of interventions.\\n\\nIn Scenario 1, based on the logistic model, we predicted the epidemic trends of the 8 countries. Table 1 and Appendix B list the detailed results of the prediction and growth trajectories of the 8 countries. For instance, the model placed the peak time of Italy as 50 days after its initial date of February 23, 2020, with a maximum number of infected individuals of approximately 97,442.\\n\\nThe infection of US was unable to be predicted based on a logistic model using reported data because it is exponentially growing, which makes it difficult to fit the logistic curve, which reflects that the US is in a stage of high spread of the disease.\\n\\nIn Scenario 2, the basic SEIR results showed that the confirmed cases will take 9-22 months (275-650 days) to reach their peak, and most of the population would eventually be infected over a long period of time if there are no control measures. Appendix C lists the detailed results of the basic SEIR model predictions of different countries. In addition, the active number of cases at the peak time will reach approximately 10-20% of these countries' populations, thus overloading the healthcare system, which is the worst possible scenario, as shown in Table 2 .\\n\\nIn Scenario 3, the adjusted SEIR model results show that under strict control measures, the number of active cases will reach a peak in 16-32 days (from early April to middle April 2020) after the initial date, when the number of cases reaches 100. Appendix D shows the detailed results of the adjusted SEIR model predictions of the 8 countries. Japan and South Korea took isolation measures when the disease was still in the early stage of transmission, its peak value was low, and the cumulative number of infected people was relatively small. Especially in Japan, our predicted peak value is less than 1560, which shows that the disease spread has been well controlled in the early stage. However, the transmission in European countries and the US is in the outbreak phase as of the date of our data collection. The adjusted SEIR model predicted that the peak values of Spain, Italy, Germany, France and other countries are between 10,420 and 85,750.\\n\\nAccording to our parameter estimation methods, we disclosed the dynamics of the number of cases. The Scenario 3 estimation is based on the strict quarantine assumption, and the results showed that implementing the control measures would decrease the epidemic peak significantly and bring it forward. This supports the conclusion that the contact rate is an important factor that reflects the effects of control measures; with the formulation and implementation of extreme epidemic prevention measures to reduce the rate of contact, the epidemic size and peak would . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint decrease. However, the epidemic still shows a long tail after the peak, and our study found that a longer quarantine period of susceptibility would reduce the long tail after the peak. Table 1 . Short-term epidemic predictions of the 8 countries Table 2 . Epidemic peak predictions of the 8 countries\\n\\nThe growth rates of the logistic curves are also listed as a spreading potential index for comparing the situations of the 8 countries in Table 3 . Our results suggest that these countries all have a high risk of rapid virus transmission except for Japan, where the spread is slowing down.\\n\\nOn the basis of evidence from previous transmission dynamics studies, the documented COVID- . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint 19 basic reproductive number (R 0 ) ranges from 2.0 to 4.9 17-20 . Our estimated R 0 range is between 1.687~3.864 (with the assumption of a 7-day mean infection period in terms of reported COVID-19 studies). \\n\\nAccording to the COVID-19 data tracking, these 8 countries have delayed the phase of preventing the epidemic and have indeed entered the outbreak phase with community-spread cases. Mortality rate analysis disclosed that undetected transmission events may have occurred in some countries. Seven of the eight countries have R0 levels above 3, which deserves our attention.\\n\\nThe high mortality was reported because the main reason for undetected transmission is probably that some of the cases are asymptomatic or lack testing kits. Our Scenario 2 and Scenario 3 are separately based on maximal assumption (without any interventions) and minimal assumption (with strict interventions). In Scenario 3, under the assumption that infected people are promptly tested and counted, the adjusted model suggested that the peak time will be reached after the strict implementation of two large-scale rounds of 14 days of quarantine. The decrease in the epidemic size in Japan and South Korea showed positive evidence, as shown in Table 1 .\\n\\nHowever, considering that each country has different cultures and healthcare situations and that the implementation of the policies and control measures are at different levels 21 , the actual situation of those countries should be between Scenario 2 and Scenario 3. It is worth pointing out that, compared with implementing no interventions, if governments take strict control measures to reduce the movement of the population and implement prompt diagnosis and isolation, the peak time will be reached, and the peak size would greatly decrease to a relatively low level in approximately 30 days; it is best triggered in the early stage of the epidemic; however, it will need to be maintained for several months until a vaccine becomes available. Conversely, as long . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint as the epidemic continues, there is still a possibility of future outbreaks if governments loosen interventions and allow people to return to close social distances.\\n\\nThe results of spreading potentials show that the R 0 of COVID-19 is high, which is similar to that of SARS viruses (R 0 values between 2.0 and 5.0 22 ), higher influenza viruses H1N1 (R 0 values between 1.2 and 3.7 23 ), and Ebola viruses (R 0 values between 1.34 and 3.65 24 ). COVID-19 is a highly contagious human-to-human transmission disease. The R 0 is expected to decrease substantially compared to values at the early stage after governments implement control measures; however, regardless of the kind of policy each country executes, each policy has its limitations in defending against COVID-19, and sustained transmission chains will occur until there is a vaccine or the until the virus disappears due to seasonal or population immunity 25 From the view of mathematical models, the SEIR model is designed for infectious disease estimation; however, the logistic growth model is designed to fit the development of the curves.\\n\\nThe logistic model may fit the existing data better when compared with the SEIR model, since it is trained from the existing data, but it cannot be accurately judged and incorporates infectious characteristics. Therefore, we believe that the logistic model is better for near-term predictions.\\n\\nOn the other hand, the SEIR model introduces more variables and factors by considering the interaction and association among multiple groups of people, and it is more reasonable than the logistic model as it follows the rules of infectious disease development, but the prediction results vary greatly with respect to different interventions and settings.\\n\\nThe study has some limitations. The mathematical models allow for the quick incorporation of multiple inputs to yield prediction results. However, this process involves making assumptions about uncertain factors; for example, it is difficult to determine the exact extent to which people follow the local government's quarantine policies or measures and engage in behaviors such as washing hands, using masks, and social distancing. This may affect the actual contact rate and the subsequent development of the epidemic. The models also lack enough data to estimate the quarantine proportions of a certain population. In fact, the evolution of the epidemic is quite . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint complicated, and our study has only taken into account a few factors. In addition, a lack of testing kits means many cases have not been tested in some countries, and without robust testing, the official number of cases is incomplete. When working with incomplete data, a small error in one factor can have an outsize effect.\\n\\nIn conclusion, our study demonstrated that reducing the contact rate is a key measure in controlling the spread of disease at an early stage, and spending enough time in quarantine would decrease the scale of cases after the peak. Therefore, implementing strong containment policies during the early spreading stages of COVID-19 and flattening the peak to avoid overloading the healthcare system should be listed as the main actions in these high-risk countries/regions. After the strict quarantine period, governments still need to raise public awareness of precautions and the importance of engaging in self-protective behaviors to bring the epidemic under control as a series of scattered events. . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03.26.20044289 doi: medRxiv preprint\\n\\n  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza</td>\n",
       "      <td>[]</td>\n",
       "      <td>The world is currently faced with a pandemic of novel coronavirus disease 2019 (COVID- 19) , which is caused by Severe Acute Respiratory Syndrome coronavirus 2 (SARS-CoV-2) and has no vaccine or cure. It is predicted the development of a safe and effective vaccine to prevent COVID-19 will take one year to 18 months, by which time it is likely that several hundreds of thousands to millions of people may have been infected. With a rapidly growing number of cases and deaths around the world, this emerging threat requires a nimble and targeted means of protection. Since coronaviruses causing COVID-19, Severe Acute Respiratory Syndrome (SARS), and Middle East Respiratory Syndrome (MERS) are able to suddenly transfer to humans from diverse animal hosts that act as viral reservoirs, there is a pressing need to develop methods to combat other potential coronaviruses that may emerge in the future [1] [2] [3] . A recent report further showed two strains (L and S) of SARS-CoV-2 with different genome sequences are circulating and likely evolving, further highlighting the need for a pan-coronavirus vaccination strategy 4 .\\n\\nThe novel coronavirus causing COVID-19 belongs to a family of positive-sense RNA viruses, which typically infect the upper and lower respiratory tracks and cause disease by direct cytotoxic effects and the induction of host cytokines disease 5 . The SARS-CoV-2 life cycle is likely similar to closely related coronaviruses such as the virus that causes SARS, in which the virus enters the cell, releases its RNA genome into the cytoplasm, and synthesizes negativesense genomic and subgenomic RNAs from which viral mRNAs and a new copy of the positive sense viral genome are synthesized (Fig. 1A) 6, 7 . Whereas traditional vaccines work by priming the human immune system to recognize viral proteins or weakened viruses and decrease viral entry into cells 8 , the alternative antiviral approach we propose here relies on a CRISPR-based system for recognizing and degrading the intracellular viral genome and its resulting viral mRNAs (Fig. 1B) . Targeting the positive-sense genome and viral mRNAs to simultaneously degrade viral genome templates for replication and viral gene expression would be expected to robustly limit viral replication.\\n\\nTo inhibit RNA viruses in human cells, we used the CRISPR-Cas13d system derived from Ruminococcus flavefaciens XPD3002, a recently discovered RNA-guided RNA endonuclease 9, 10 . Cas13d employs CRISPR-associated RNAs (crRNAs) that contain a customizable 22 nucleotide (nt) spacer sequence that can direct the Cas13d protein to specific RNA molecules for targeted RNA degradation. The high catalytic activity of Cas13d in mammalian cells provides a potential mechanism for targeting SARS-CoV-2 for specific viral RNA genome degradation and viral gene expression inhibition. Because of its small size (967 amino acids), high specificity, and strong catalytic activity, we chose Cas13d rather than other Cas13 proteins to target and destroy RNA viruses including SARS-CoV-2 and Influenza A virus (IAV) [10] [11] [12] .\\n\\nIn this work, we developed a Prophylactic Antiviral CRISPR in huMAN cells (PAC-MAN) strategy as a form of genetic intervention to target SARS-CoV-2, IAV, and potentially all sequenced coronaviruses. We created a bioinformatic pipeline to define conserved regions across the SARS-CoV-2 genomes and target these regions using CRISPR-Cas13d for viral genome degradation and viral gene inhibition. At the time of paper submission, there is no widely available laboratory strains of SARS-CoV-2. Therefore, we tested our approach using synthesized fragments of SARS-CoV-2, as well as with live H1N1 IAV. We designed and screened a panel of crRNA pools targeting conserved viral regions and defined the most effective crRNAs. We demonstrated the ability of our approach to cleave SARS-CoV-2 fragments and to reduce the replication of IAV in human lung epithelial cells. Our bioinformatics author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript analysis revealed a group of six crRNAs that can target 91% of sequenced coronaviruses, as well as a group of 22 crRNAs able to target all sequenced coronaviruses. The potential pancoronavirus protection using CRISPR-Cas13d offers an alternative and complementary approach over traditional pharmaceuticals or vaccines. Through the use of crRNA pools targeting different regions of the same virus or different strains of coronavirus, this system could possibly buffer against viral evolution and escape, as well as enable rapid development and deployment against emerging viruses. When combined with an effective delivery platform, PAC-MAN is a promising strategy to combat not only coronaviruses including that causing COVID-19, but also a broad range of other genera of viruses. author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript\\n\\nTo create effective and specific crRNA sequences to target and cleave SARS-CoV-2, we first performed a bioinformatic analysis by aligning published SARS-CoV-2 genomes from 47 patients with SARS-CoV and MERS-CoV genomes at the single-nucleotide resolution level. SARS-CoV-2 has a single-stranded RNA genome with ~30,000 nucleotides that encodes 12 putative, functional open reading frames [13] [14] [15] . Our analysis found regions with a high conservation between 47 SARS-CoV-2, SARS-CoV and MERS-CoV genomes ( Fig. 2A) . The two highly-conserved regions respectively contain the RNA-dependent RNA polymerase (RdRP) gene in the polypeptide ORF1ab region, which maintains the proliferation of all coronaviruses, and the Nucleocapsid (N) gene at the 3' end of the genome, which encodes the capsid protein for viral packaging.\\n\\nauthor/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript\\n\\nTo target highly conserved SARS-CoV-2 regions, we generated an in silico collection of all 3,802 possible crRNAs (Supplemental Fig. 1 ). After excluding crRNAs either predicted to have potential off-target binding (≤1 mismatch) in the human transcriptome or having poly-T (≥4 Ts) sequences that may prevent crRNA expression, we obtained a collection of 3,203 crRNAs (Supplemental Table 1 ). We designed and synthesized 40 crRNAs, with 20 crRNAs each targeting the conserved sequences of the RdRP and N genes (Supplemental Table 2 ). Our rationale for densely targeting these two regions was to significantly reduce both viral genomic RNA and mRNA templates for expressing essential viral proteins and to minimize the risk of loss of targeting activity by mutational escape.\\n\\nTo evaluate whether Cas13d is effective for targeting SARS-CoV-2 sequences, we created two reporters expressing synthesized fragments of SARS-CoV-2 fused to GFP (SARS-CoV-2-F1 and SARS-CoV-2-F2; Fig. 2B , Supplemental Table 3 ). As SARS-CoV-2 appears to mainly infect respiratory tract cells in patients 5 , we chose to use human lung epithelial A549 cells as a model cell line. We created a stable A549 cell line expressing Cas13d through lentiviral infection, followed by sorting for the mCherry marker that was co-expressed with Cas13d ( Fig.  2C) .\\n\\nWe first investigated which regions of the SARS-CoV-2 fragments were most susceptible to Cas13d-mediated targeting and cleavage. To do this, we transfected or transduced the SARS-CoV-2 reporters into Cas13d A549 cells that had been transduced with pools of four crRNAs targeting either RdRP or N gene regions (Fig. 2D , see Methods). We chose to evaluate the effectiveness of our system using a pool of crRNAs, since in the context of live viral infection, this would prevent the possibility of escape from inhibition due to a mutation involving a single crRNA target site. A total of ten groups (G1-G10) were tested for all 40 crRNAs (Supplemental Table 2 ). Twenty-four hours after SARS-CoV-2 reporter transfection, we performed flow cytometry to examine levels of GFP protein expression and collected RNA from cells to perform quantitative real-time PCR (qRT-PCR) to examine mRNA transcript abundance.\\n\\nWe observed that most RdRP-targeting crRNA pools were able to repress reporter expression to some extent compared to the control, and one pool targeting the middle of the SARS-CoV-2- Table 2 for the designed crRNA sequences and Supplemental Table 3 author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript F1 RdRP region (G4; SARS-CoV-2 genome coordinates 13,784-14,149) was able to repress GFP by 86% (p = 2x10 -6 ) (Fig. 3A, S2A -B, Supplemental Table 4 ). One pool of N genetargeting crRNAs (G8; SARS-CoV-2 genome coordinates 28,803-29,034) was able to substantially repress SARS-CoV-2-F2 GFP, causing a 71% decrease in expression (p = 2x10 -7 ) (Fig. 3B, S2A -B, Supplemental Table S4 ). Our qRT-PCR analysis showed consistent results at the RNA level, with G4 and G8 inhibiting their respective reporter mRNA abundance by 83% (p = 3x10 -11 ) and 79% (p = 2x10 -12 ) ( Fig. 3A -B, Supplemental Table S4 ). The variations in the ability of different crRNA pools to repress the SARS-CoV-2 reporters are likely due to RNA secondary structure inherent in the SARS-CoV-2 genome or differences in binding affinities for each crRNA sequence composition. Table 4 . P values for C-E are labeled for each group.\\n\\nauthor/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript\\n\\nWe next independently validated the performance of adjacent crRNA pools when SARS-CoV-2-F1 or -F2 was introduced via lentiviral transduction. We argue that this approach, in which the lentivirus-derived RNA serves as a PAC-MAN target prior to lentivirus integration into DNA, mimics the crRNA targeting conditions that would apply for natural COVID-19 infection. We tested on a few selected crRNA pools, including the 3 best groups G4, G5, and G8 and one less efficient group G1, with a lentiviral multiplicity of infection (MOI) of 0.5. Our results revealed similar patterns to our SARS-CoV-2 transfection experiments, with the best groups (G4, G5, G8) able to repress their reporters by 69%, 71%, and 60% (p = 7x10 -6 , 10 -3 , 3x10 -6 for G4, G5, G8, respectively) (Fig. 3C) . qRT-PCR analysis also showed consistent results, with G4, G5, and G8 repressing 70%, 68%, and 49% of RNA abundance (p = 8x10 -8 , 4x10 -6 , 10 -4 for G4, G5, G8 respectively) (Fig. 3D) .\\n\\nWe further validated the effectiveness of Cas13d-mediated repression of SARS-CoV-2 reporters by co-transfecting each reporter with a pool of 5 crRNAs tiled along the whole fragment, compared to a pool of 5 non-targeting control crRNAs (Table S2) . We found that the RdRPtargeting crRNAs were able to repress SARS-CoV-2-F1 GFP by 81% (p = 0.01), while Ntargeting crRNAs were able to repress SARS-CoV-2-F2 GFP by 90% (p = 5x10 -4 ) (Fig. 3E,  S3A) . We also did an analysis of the extent to which expression of Cas13d or crRNAs affected SARS-CoV-2 GFP reporter inhibition. Our analysis revealed that SARS-CoV-2 inhibition was more sensitive to crRNA concentration in our samples, while lower Cas13d expression only moderately decreased inhibitory activity (Fig. S3B) . Thus, maintaining a high level of crRNA concentration is likely a determining factor for more effective SARS-CoV-2 sequence targeting and cleavage.\\n\\nThese data together suggest: 1) Cas13d PAC-MAN can be an effective system to target and degrade SARS-CoV-2 sequences in human cells and 2) proper design of crRNAs are important for obtaining a high efficiency of SARS-CoV-2 inhibition.\\n\\nSince we do not yet have access to live SARS-CoV-2 virus, as a proof-of-concept we elected to test the CRISPR-Cas13d strategy on inhibiting H1N1 IAV, a RNA virus with a similar tropism as SARS-CoV-2 for respiratory tract epithelial cells. In contrast to SARS-CoV-2, which has only a single, continuous RNA genome, the IAV genome is contained in 8 negative-sense RNA segments, in which each plays an important role in packaging the viral RNA into budding virions 16, 17 .\\n\\nTo create crRNAs targeting a broad range of IAV strains, we aligned all complete IAV genomes retrieved from the Influenza Research Database and searched for the most conserved regions (Supplemental Table 5 ). Our analysis and the work of others showed that the segment ends are highly conserved for all 8 segments [16] [17] [18] [19] [20] [21] . The segment ends are potentially attractive regions for targeting with Cas13d because of the potential to inhibit a broad range of IAV strains and because of the previous evidence that interfering with the packaging of one segment encoded on the segment ends can decrease the packaging efficiency of other segments and overall virion packaging 18, [22] [23] [24] [25] . Using sequence representation from ~140 different strains of influenza, we designed crRNAs that could be robust across as many different IAV strains as possible. Our bioinformatics analysis resulted in 6 crRNAs targeting highly conserved genome regions for each of the 8 IAV segments, for a total of 48 crRNAs (Supplemental Table 6 ).\\n\\nauthor/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript\\n\\nTo test the efficiency of IAV-targeting crRNAs, we performed a screen of our crRNA pools in the stable Cas13d A549 cell line to determine which were most effective at inhibiting IAV infection (Fig. 4A) . To do this, we transfected a pool of 6 crRNAs targeting a given IAV genome segment. Two days after crRNA transfection, the Cas13d A549 cells were challenged with PR8-mNeon, a strain of H1N1 IAV (A/Puerto Rico/8/1934) engineered to express the mNeonGreen gene, a fluorescent reporter protein (hereafter referred to as PR8-mNeon) at an MOI of 2.5 or 5.0 [26] [27] [28] . At approximately 18 hours post-challenge, the cells were analyzed for IAV infection through flow cytometry and microscopy. We examined the percentage of cells that were mNeon positive for each segment-targeting crRNA pool and calculated the fold-change compared to cells transfected with a pool of non-targeting crRNAs). Out of all crRNAs, the crRNA pool targeting segment 6 (S6) delivered the most consistent and robust results across different MOIs (72% reduction for MOI = 2.5 and 52% reduction for MOI = 5; Fig. 4B, S4) . Additionally, the segment 4 (S4) pool also showed moderate but consistent inhibition. To verify Cas13d-mediated inhibitory effects on S4 and S6, we further tested at a lower MOI of 0.5, which showed better inhibition of IAV across samples (e.g., 79% reduction using crRNAs targeting S6; Fig. S5 ). This confirms consistent inhibitory activity of IAV by Cas13d when targeting these two segments. There is a possible trend of greater inhibition at lower viral titers, suggesting that Cas13d PAC-MAN could be a particularly effective strategy for prevention of new infections, which typically occurs from exposure to a low level of virus.\\n\\nWe further characterized the S6-targeting crRNA pool. S6 encodes the gene neuraminidase (NA), which releases budding virions from the host cell 16 . We performed microscopy quantification of mNeon+ cells and found that mNeon expression was reduced by 62% (p = 10 -5 ) and 73% (p = 5x10 -9 ) with an MOI of 2.5 and 5, respectively, with S6-targeting crRNAs (Fig. 4C) and this trend was corroborated by flow cytometry (Fig. 4D) . Together, our data show that Cas13d PAC-MAN is able to target highly conserved viral regions and robustly inhibit viral replication in human lung epithelial cells.\\n\\nIn the past two decades, multiple variants of coronavirus, including those causing COVID-19, SARS and MERS, emerged from animal reservoirs and infected humans, each time causing significant disease and fatalities 1-3 . Thus, the design of a strategy that could broadly prevent and target viral threats, including all coronavirus strains that currently reside in animals, would be a valuable resource. We asked whether it is possible to design a minimal number of crRNAs that could target a majority of known coronaviruses found in both humans and animals.\\n\\nTo do this, we analyzed all known coronavirus genomes (3,051) to identify all possible crRNAs that can target each of those genomes, ending up with approximately 6.48 million possible crRNAs (Fig. 5A, S6) . We then refined our list to define the smallest number of crRNAs that could target all known coronavirus genomes with zero mismatches. We found that just two crRNAs could target ~50% of coronavirus genomes, including those causing COVID-19, SARS, and MERS; 6 crRNAs were able to target ~91% of coronavirus genomes; and 22 crRNAs covered all sequenced coronaviruses with no mismatches (Fig. 5B, Supplemental Table 7 ). The ability to use a relatively small number of crRNAs to broadly target most or all coronavirus strains further highlights the uniqueness of our approach in contrast to traditional pharmaceutical or vaccination approaches.\\n\\nAs a resource, we provide a map to show how our top 6 crRNAs (PAC-MAN-T6) targeted the coronavirus phylogenetic tree (Fig. 5A, red box) . Beta coronaviruses, the genus of viruses causing COVID-19, SARS, and MERS, were mostly covered by 3 crRNAs (crRNA 1-3) , while the other genera were covered by these three plus the remaining 3 crRNAs. Altogether, the PAC-MAN-T6 can target all known human coronavirus with broad coverage against other animal coronaviruses.\\n\\nThe minimal set of six crRNAs also included one crRNA (designated crRNA-N18f) that we validated. A recent work analyzed 103 SARS-CoV-2 genomes and indicated that there are two major types (L and S) of SARS-CoV-2 existing with different viral signature sequences 4 . To verify whether crRNA-N18f can target all L and S strains, we retrieved a set of 202 recently sequenced SARS-CoV-2 sequences from Global Initiative on Sharing All Influenza Data author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript\\n\\n(GISAID). Notably, crRNA-N18f targets all 201 genomes, with only one SARS-CoV-2 sequence carrying a single mismatch mutation (Supplemental Table 8 ). This suggests that our approach can robustly target multiple SARS-CoV-2 strains, but also suggests the potential need to use a pool of crRNAs to avoid mutational escapes. author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript\\n\\nCRISPR-Cas13 emerged as a natural bacterial defense against bacteriophages, a powerful antiviral system that is able to use sequence-specific crRNAs to protect host bacterial cells from infections 29 . Here, we have repurposed the RNA-guided RNA endonuclease activity of Cas13d in mammalian cells against emergent viral targets, SARS-CoV-2 and IAV, in our PAC-MAN strategy. This expands the applications of CRISPR-Cas13 systems in addition to their uses for diagnostics such as SHERLOCK and live-cell imaging [30] [31] [32] [33] . Prior work showed Cas13a/b systems could inhibit ssRNA viruses including influenza, which is consistent with our observed high efficiency of SARS-CoV-2 reporter and IAV inhibition using the PAC-MAN approach 31 . We demonstrated that Cas13d-based genetic targeting can effectively target and cleave the RNA sequences of SARS-CoV-2 fragments and IAV in lung epithelial cell cultures. Our bioinformatics analysis also suggested a minimal pool of 6 crRNAs able to target over 90% of sequenced coronaviruses. While this strategy is a proof-of-concept and will require further testing using replication-competent SARS-CoV-2 viruses and validation in animal models before clinical tests in humans, it represents a unique approach to implement a rapid and broad antiviral defense in humans against emerging pathogens for which there are no effective vaccines.\\n\\nFor both SARS-CoV-2 and IAV, we found highly conserved regions of the viral genomes to target with Cas13d. In the case of SARS-CoV-2, these genome regions encode the RdRP and Nucleocapsid proteins, which are essential for coronavirus replication and function 13, 34 . RdRP is responsible for catalyzing the replication of all viral mRNAs and Nucleocapsid binds to genomic RNA, protecting it and serving as one of the two main structural proteins in virions. Targeted inhibition of these proteins could have an outsized effect on disabling virus production and function, in addition to reducing viral load through degradation of the viral genome itself. For IAV, the crRNAs that showed the best viral inhibition target the conserved ends of the segment S6 that encodes Neuraminidase (NA), a viral surface protein essential for mediating budding of new virions 35 . Prior mutation analysis has shown that mutating the packaging signals of one segment end leads to decreased packaging of other segments, indicating that there is a cooperative mechanism for packaging the IAV genome 18 . Thus, by inhibiting a single IAV genome segment, we could reduce the overall packaging of IAV in a synergistic fashion.\\n\\nNotably, the SARS-CoV-2 RdRP gene and IAV NA gene that we targeted in our study are also the targets of two antiviral drugs, Remdesivir and Oseltamivir. Remdesivir, created by Gilead Sciences (development code GS-5734), is an RdRP inhibitor that was developed to treat Ebola and Marburg virus infections and has been reported to have antiviral activity against MERS and SARS 36-38 . Remdesivir is already administered for COVID-19 through compassionate use requests, and clinical trials for its use against COVID-19 have already started in the U.S.\\n\\n(clinical trial ID NCT04280705) and China. Oseltamivir (also known as Tamiflu), is an antiviral medication for treating and preventing IAV that acts as an inhibitor of NA, thereby preventing the egress of new viral particles 39 . It will be interesting to see if these drugs could be used in combination with Cas13d PAC-MAN to synergistically cripple viral function and replication.\\n\\nWhile we demonstrate that the PAC-MAN system is able to exhibit strong repression of viral sequences in an in vitro setting, it must be coupled with an effective in vivo delivery system for therapeutic use. There are several potential strategies that could be employed for transient in vivo expression of CRISPR components. Cas13d and its cognate crRNAs could be delivered as RNAs within polymers or lipid nanoparticles, perhaps with chemical alterations to increase stability 40 . In addition, a DNA-based liposomal delivery strategy, such as the recently developed HEDGES platform is also attractive 41 . Yet another strategy would be to deliver a author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript 13 ribonucleoprotein complex containing purified Cas13d assembled with crRNAs 42, 43 . All of these could potentially be used both as a therapeutic and as a prophylactic measure.\\n\\nIf the delivery barriers are overcome and a strategy like this is able to be used therapeutically, there are many potential benefits over traditional vaccines. Traditional vaccines rely on priming the immune system through exposure to viral proteins or peptides often derived from surface proteins that exhibit a high rate of mutation, which increased the chances of viral evasion of the host immune response 44 . In contrast, here we have demonstrated a genetic strategy that is able to target highly conserved regions, which would be expected to make it much more unlikely for the virus to escape inhibition through mutation. In addition, the ability of Cas13d to process its own crRNAs from a crRNA array means that multiple crRNAs targeting different regions (e.g. Fig. 3E , and PAC-MAN-T6 in Fig. 5A ) could be delivered simultaneously 9 , further reducing the chances of viral escape. As a further advantage, we demonstrate a potential pan-coronavirus strategy to target not only viruses that circulate in humans, but also those that currently are found in animal reservoirs that might transfer to humans to cause disease. If the crRNAs targeting these viruses can be tested and validated before they ever infect humans, we can greatly accelerate the development of countermeasures for future emergent threats.\\n\\nIn summary, our PAC-MAN strategy represents a potentially powerful new approach for inhibiting viral function and replication, and we envision it could be useful for a diverse array of circulating and emergent viral threats.\\n\\n47 complete SARS-CoV-2 assemblies were downloaded from (https://www.ncbi.nlm.nih.gov/genbank/2019-ncov-seqs/) on February 21, 2020. All the other 3,108 Coronavirinae complete genome sequences were downloaded from ViPR (Virus Pathogen Resource) database on February 21, 2020. After excluding very short genomes, we collected in total 3,051 Coronavirinae virus genomes. 202 SARS-CoV-2 genome sequences were downloaded from GISAID (Global Initiative on Sharing All Influenza Data; https://www.gisaid.org/).\\n\\nWe developed a full computational workflow to design crRNAs for the Cas13d system. To design all possible crRNAs for the three pathogenic RNA viruses (SARS-CoV-2, SARS-CoV, and MERS-CoV), the reference genomes of SARS-CoV, MERS-CoV, along with SARS-CoV-2 genomes derived from 47 patients were first aligned by MAFFT using the --auto flag. crRNA candidates were identified by using a sliding window to extract all 22 nt sequences with perfect identity among the SARS-CoV-2 genomes. We annotated each crRNA candidate with the number of mismatches relative to the SARS-CoV and MERS-CoV genomes, as well as the GC content. 3,802 crRNA candidates were selected with perfect match against the 47 SARS-CoV-2 genomes and with ≤1 mismatch to SARS-CoV or MERS-CoV sequences. To characterize the specificity of 22nt crRNAs, we ensured that crRNA do not target any sequences in the human transcriptome. We used Bowtie 1.2.2 to align crRNAs to the human transcriptome (HG38; including non-coding RNA) and removed crRNAs that mapped to the human transcriptome with fewer than 2 mismatches.\\n\\nWe maximized the vertical coverage of pooled crRNAs to target a maximal set of the 3,051 Coronavirinae viruses with a minimal set of crRNAs. A sliding window was used to extract all unique 22 nt sequences in the Coronavirinae genomes, yielding 6,480,955 unique crRNA candidates. A hash table was used to map each crRNA candidate to the number of Coronavirinae genomes that it targets with a perfect match. A greedy algorithm was used to identify a minimal set of covering crRNAs. In each iteration, the crRNAs were sorted in descending order by the number of genomes targeted. The crRNA that targeted the most genomes was added to the minimal set of crRNAs, and all of the genomes it targeted were removed from the hash table values. If multiple crRNAs were tied for targeting the most genomes, the crRNAs were ranked by the length and quantity of stretches of repeated nucleotides, and the lowest ranking crRNA (i.e. the one with the fewest, shortest stretches of repeated nucleotides) was added to the set of minimal crRNAs. Using this approach, a set of 22 crRNAs was identified that was able to target all 3150 Coronavirinae virus genomes (including 47 SARS-CoV-2). The phylogenetic tree and the associated annotations were generated by the Interactive Tree Of Life (iTOL v5.5; https://itol.embl.de/).\\n\\nAll the complete SARS-CoV-2 genome sequences were aligned with the SARS-CoV and MERS-CoV NCBI Refseq sequences by MAFFT using the --auto flag. The alignment was visualized using the Geneious Prime software (Geneious 2020.0.4; https://www.geneious.com).\\n\\nauthor/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript\\n\\nA consensus nucleotide (which is shared by &gt;50% of non-gap characters at the given position) was assigned to each position in the alignment. Not every position has a consensus nucleotide.\\n\\nThe percent conservation at each position was calculated as the percentage of sequences matching the consensus at that position. To aid interpretation, the resulting plot of percentage conservation was smoothed using a moving average with a window size of 1,001 bp centered at each position. Positions corresponding to gap characters in the SARS-CoV-2 reference sequence were omitted.\\n\\nThe crRNA plasmids were cloned using standard restriction-ligation cloning. Forward and reverse oligos for each spacer (IDT) were annealed and inserted into backbone using T4 DNA Ligase (NEB). Assembled oligos were either inserted into a pHR or pUC19 backbone, obtained and modified from Addgene (#121514 and #109054, respectively). Spacer sequences for all crRNAs can be found in Supplemental Table 2 and 6.\\n\\nTo clone SARS-CoV-2 reporters, pHR-PGK-scFv GCN4-sfGFP (pSLQ1711) was digested with EcoR1 and Sbf1 and gel purified. The PGK promoter and sfGFP were PCR amplified from pSLQ1711 and gel purified. The SARS-CoV-2-F1 and SARS-CoV-2-F2 fragments were synthesize from Integrated DNA Technologies (IDT). At last, the PGK, sfGFP, and SARS-CoV-2-F1/F2 fragments were inserted into the linearized pSLQ1711 vector using In-Fusion cloning (Takara Bio).\\n\\nOn day 1, one confluent 100mm dish (Corning) of HEK293T cells were seeded into three 150mm dishes (Corning). On day 2, cells were ~50-70% confluent at the time of transfection. For A549 cells, lentivirus precipitation solution (Alstem) was added and precipitated as per the manufacturer's protocol. One well of a 6-well plate (Corning) of A549 cells at ~50% confluency were transduced with the precipitated lentivirus. After 2-3 days of growth, the cell supernatant containing virus was removed and the cells were expanded. Cells were then sorted for mCherry+ cells using a Sony SH800S cell sorter. Table S2 ). On day 3, cells were switched to fresh medium with Puromycin at 2 ug/mL. Twenty-four hours after Puromycin selection, cells with transfected crRNAs were challenged with SARS-CoV-2 reporter plasmids or SARS-CoV-2 reporter lentivirus. Twenty-four hours after transfection or 48 hours after transduction, cells were tested using flow cytometry on a Beckman-Coulter CytoFLEX S and qRT-PCR on a Biorad author/funder. All rights reserved. No reuse allowed without permission. \\n\\nThe PR8 mNeon IAV was generated by insertion of the mNeon fluorescent gene into segment 4 of the H1N1 A/Puerto Rico/8/1934 strain (Nicholas Heaton's Lab, Duke University) as previously described . The PR8-mNeon virus was propagated in chicken eggs, and harvested allantoic fluid containing the virus was titered using MDCK plaque assays, and frozen in aliquots for later use.\\n\\n3.1x10 6 Cas13d A549 cells were plated in two 100 mm dishes. The following day, the cells were transfected with 15 μg pooled crRNAs targeting the IAV genome or non-targeting controls (Supplemental Table 6 ) in 1.5 mL Opti-MEM reduced serum media and 30 μL Mirus LT1 reagent. After approximately 24 hours, the transfected Cas13d A549 cells were dissociated and re-plated in 24-well plates at a density of 15-20,000 cells/well and incubated overnight. Flow cytometry was performed on excess cells to quantify the transfection efficiency. Cells were challenged overnight with mNeonGreen expressing H1N1 A/Puerto Rico/8/1934 at an MOI of 2.5 and 5. Uninfected cells were included as a negative control. An independent two-tailed T test was used to calculate the p values and statistics in Fig. 4C-D. author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript\\n\\nReal-time qRT-PCR was performed to quantify RNA abundance. For each sample, total RNA was isolated by using the RNeasy Plus Mini Kit (Qiagen Cat# 74134), followed by cDNA synthesis using the iScript cDNA Synthesis Kit (BioRad, Cat# 1708890). qRT-PCR primers were ordered from Integrated DNA Technologies (IDT). Quantitative PCR was performed using the PrimePCR assay with the SYBR Green Master Mix (BioRad), and run on a Biorad CFX384 realtime system (C1000 Touch Thermal Cycler), according to manufacturers' instructions. Cq values were used to quantify RNA abundance. The relative abundance of the SARS-CoV-2 fragments was normalized to a GAPDH internal control.\\n\\nMicroscopy for SARS-CoV-2 experiments was performed on a Nikon TiE inverted confocal microscope equipped with an Andor iXon Ultra-897 EM-CCD camera and 405 nm, 488 nm, 561 nm and 642 nm lasers, using the 20x PLAN APO air objective (NA = 0.75). Image were taken using NIS Elements version 4.60 software. Microscopy for IAV experiments was performed on a Keyence BZ-X810 microscope equipped with a GFP and Texas Red filter was used to obtain images with a 20X objective lens.\\n\\nA549 cells that express Cas13d and are transfected with crRNAs and challenged with PR8 mNeon virus were washed with 1X PBS, treated with 1X Trypsin EDTA and fixed in 1.6% paraformaldehyde in PBS for 10 minutes. Cells were washed with FACS buffer (1X PBS, 0.3% BSA, 1mM EDTA). BD Accuri C6 plus flow cytometer was used to collect raw data. Raw data were processed with Cytobank (BD) software.\\n\\nauthor/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint\\n\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                         title  \\\n",
       "30  Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza   \n",
       "\n",
       "   abstract  \\\n",
       "30  []        \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           full_text  \n",
       "30  The world is currently faced with a pandemic of novel coronavirus disease 2019 (COVID- 19) , which is caused by Severe Acute Respiratory Syndrome coronavirus 2 (SARS-CoV-2) and has no vaccine or cure. It is predicted the development of a safe and effective vaccine to prevent COVID-19 will take one year to 18 months, by which time it is likely that several hundreds of thousands to millions of people may have been infected. With a rapidly growing number of cases and deaths around the world, this emerging threat requires a nimble and targeted means of protection. Since coronaviruses causing COVID-19, Severe Acute Respiratory Syndrome (SARS), and Middle East Respiratory Syndrome (MERS) are able to suddenly transfer to humans from diverse animal hosts that act as viral reservoirs, there is a pressing need to develop methods to combat other potential coronaviruses that may emerge in the future [1] [2] [3] . A recent report further showed two strains (L and S) of SARS-CoV-2 with different genome sequences are circulating and likely evolving, further highlighting the need for a pan-coronavirus vaccination strategy 4 .\\n\\nThe novel coronavirus causing COVID-19 belongs to a family of positive-sense RNA viruses, which typically infect the upper and lower respiratory tracks and cause disease by direct cytotoxic effects and the induction of host cytokines disease 5 . The SARS-CoV-2 life cycle is likely similar to closely related coronaviruses such as the virus that causes SARS, in which the virus enters the cell, releases its RNA genome into the cytoplasm, and synthesizes negativesense genomic and subgenomic RNAs from which viral mRNAs and a new copy of the positive sense viral genome are synthesized (Fig. 1A) 6, 7 . Whereas traditional vaccines work by priming the human immune system to recognize viral proteins or weakened viruses and decrease viral entry into cells 8 , the alternative antiviral approach we propose here relies on a CRISPR-based system for recognizing and degrading the intracellular viral genome and its resulting viral mRNAs (Fig. 1B) . Targeting the positive-sense genome and viral mRNAs to simultaneously degrade viral genome templates for replication and viral gene expression would be expected to robustly limit viral replication.\\n\\nTo inhibit RNA viruses in human cells, we used the CRISPR-Cas13d system derived from Ruminococcus flavefaciens XPD3002, a recently discovered RNA-guided RNA endonuclease 9, 10 . Cas13d employs CRISPR-associated RNAs (crRNAs) that contain a customizable 22 nucleotide (nt) spacer sequence that can direct the Cas13d protein to specific RNA molecules for targeted RNA degradation. The high catalytic activity of Cas13d in mammalian cells provides a potential mechanism for targeting SARS-CoV-2 for specific viral RNA genome degradation and viral gene expression inhibition. Because of its small size (967 amino acids), high specificity, and strong catalytic activity, we chose Cas13d rather than other Cas13 proteins to target and destroy RNA viruses including SARS-CoV-2 and Influenza A virus (IAV) [10] [11] [12] .\\n\\nIn this work, we developed a Prophylactic Antiviral CRISPR in huMAN cells (PAC-MAN) strategy as a form of genetic intervention to target SARS-CoV-2, IAV, and potentially all sequenced coronaviruses. We created a bioinformatic pipeline to define conserved regions across the SARS-CoV-2 genomes and target these regions using CRISPR-Cas13d for viral genome degradation and viral gene inhibition. At the time of paper submission, there is no widely available laboratory strains of SARS-CoV-2. Therefore, we tested our approach using synthesized fragments of SARS-CoV-2, as well as with live H1N1 IAV. We designed and screened a panel of crRNA pools targeting conserved viral regions and defined the most effective crRNAs. We demonstrated the ability of our approach to cleave SARS-CoV-2 fragments and to reduce the replication of IAV in human lung epithelial cells. Our bioinformatics author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript analysis revealed a group of six crRNAs that can target 91% of sequenced coronaviruses, as well as a group of 22 crRNAs able to target all sequenced coronaviruses. The potential pancoronavirus protection using CRISPR-Cas13d offers an alternative and complementary approach over traditional pharmaceuticals or vaccines. Through the use of crRNA pools targeting different regions of the same virus or different strains of coronavirus, this system could possibly buffer against viral evolution and escape, as well as enable rapid development and deployment against emerging viruses. When combined with an effective delivery platform, PAC-MAN is a promising strategy to combat not only coronaviruses including that causing COVID-19, but also a broad range of other genera of viruses. author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript\\n\\nTo create effective and specific crRNA sequences to target and cleave SARS-CoV-2, we first performed a bioinformatic analysis by aligning published SARS-CoV-2 genomes from 47 patients with SARS-CoV and MERS-CoV genomes at the single-nucleotide resolution level. SARS-CoV-2 has a single-stranded RNA genome with ~30,000 nucleotides that encodes 12 putative, functional open reading frames [13] [14] [15] . Our analysis found regions with a high conservation between 47 SARS-CoV-2, SARS-CoV and MERS-CoV genomes ( Fig. 2A) . The two highly-conserved regions respectively contain the RNA-dependent RNA polymerase (RdRP) gene in the polypeptide ORF1ab region, which maintains the proliferation of all coronaviruses, and the Nucleocapsid (N) gene at the 3' end of the genome, which encodes the capsid protein for viral packaging.\\n\\nauthor/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript\\n\\nTo target highly conserved SARS-CoV-2 regions, we generated an in silico collection of all 3,802 possible crRNAs (Supplemental Fig. 1 ). After excluding crRNAs either predicted to have potential off-target binding (≤1 mismatch) in the human transcriptome or having poly-T (≥4 Ts) sequences that may prevent crRNA expression, we obtained a collection of 3,203 crRNAs (Supplemental Table 1 ). We designed and synthesized 40 crRNAs, with 20 crRNAs each targeting the conserved sequences of the RdRP and N genes (Supplemental Table 2 ). Our rationale for densely targeting these two regions was to significantly reduce both viral genomic RNA and mRNA templates for expressing essential viral proteins and to minimize the risk of loss of targeting activity by mutational escape.\\n\\nTo evaluate whether Cas13d is effective for targeting SARS-CoV-2 sequences, we created two reporters expressing synthesized fragments of SARS-CoV-2 fused to GFP (SARS-CoV-2-F1 and SARS-CoV-2-F2; Fig. 2B , Supplemental Table 3 ). As SARS-CoV-2 appears to mainly infect respiratory tract cells in patients 5 , we chose to use human lung epithelial A549 cells as a model cell line. We created a stable A549 cell line expressing Cas13d through lentiviral infection, followed by sorting for the mCherry marker that was co-expressed with Cas13d ( Fig.  2C) .\\n\\nWe first investigated which regions of the SARS-CoV-2 fragments were most susceptible to Cas13d-mediated targeting and cleavage. To do this, we transfected or transduced the SARS-CoV-2 reporters into Cas13d A549 cells that had been transduced with pools of four crRNAs targeting either RdRP or N gene regions (Fig. 2D , see Methods). We chose to evaluate the effectiveness of our system using a pool of crRNAs, since in the context of live viral infection, this would prevent the possibility of escape from inhibition due to a mutation involving a single crRNA target site. A total of ten groups (G1-G10) were tested for all 40 crRNAs (Supplemental Table 2 ). Twenty-four hours after SARS-CoV-2 reporter transfection, we performed flow cytometry to examine levels of GFP protein expression and collected RNA from cells to perform quantitative real-time PCR (qRT-PCR) to examine mRNA transcript abundance.\\n\\nWe observed that most RdRP-targeting crRNA pools were able to repress reporter expression to some extent compared to the control, and one pool targeting the middle of the SARS-CoV-2- Table 2 for the designed crRNA sequences and Supplemental Table 3 author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript F1 RdRP region (G4; SARS-CoV-2 genome coordinates 13,784-14,149) was able to repress GFP by 86% (p = 2x10 -6 ) (Fig. 3A, S2A -B, Supplemental Table 4 ). One pool of N genetargeting crRNAs (G8; SARS-CoV-2 genome coordinates 28,803-29,034) was able to substantially repress SARS-CoV-2-F2 GFP, causing a 71% decrease in expression (p = 2x10 -7 ) (Fig. 3B, S2A -B, Supplemental Table S4 ). Our qRT-PCR analysis showed consistent results at the RNA level, with G4 and G8 inhibiting their respective reporter mRNA abundance by 83% (p = 3x10 -11 ) and 79% (p = 2x10 -12 ) ( Fig. 3A -B, Supplemental Table S4 ). The variations in the ability of different crRNA pools to repress the SARS-CoV-2 reporters are likely due to RNA secondary structure inherent in the SARS-CoV-2 genome or differences in binding affinities for each crRNA sequence composition. Table 4 . P values for C-E are labeled for each group.\\n\\nauthor/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript\\n\\nWe next independently validated the performance of adjacent crRNA pools when SARS-CoV-2-F1 or -F2 was introduced via lentiviral transduction. We argue that this approach, in which the lentivirus-derived RNA serves as a PAC-MAN target prior to lentivirus integration into DNA, mimics the crRNA targeting conditions that would apply for natural COVID-19 infection. We tested on a few selected crRNA pools, including the 3 best groups G4, G5, and G8 and one less efficient group G1, with a lentiviral multiplicity of infection (MOI) of 0.5. Our results revealed similar patterns to our SARS-CoV-2 transfection experiments, with the best groups (G4, G5, G8) able to repress their reporters by 69%, 71%, and 60% (p = 7x10 -6 , 10 -3 , 3x10 -6 for G4, G5, G8, respectively) (Fig. 3C) . qRT-PCR analysis also showed consistent results, with G4, G5, and G8 repressing 70%, 68%, and 49% of RNA abundance (p = 8x10 -8 , 4x10 -6 , 10 -4 for G4, G5, G8 respectively) (Fig. 3D) .\\n\\nWe further validated the effectiveness of Cas13d-mediated repression of SARS-CoV-2 reporters by co-transfecting each reporter with a pool of 5 crRNAs tiled along the whole fragment, compared to a pool of 5 non-targeting control crRNAs (Table S2) . We found that the RdRPtargeting crRNAs were able to repress SARS-CoV-2-F1 GFP by 81% (p = 0.01), while Ntargeting crRNAs were able to repress SARS-CoV-2-F2 GFP by 90% (p = 5x10 -4 ) (Fig. 3E,  S3A) . We also did an analysis of the extent to which expression of Cas13d or crRNAs affected SARS-CoV-2 GFP reporter inhibition. Our analysis revealed that SARS-CoV-2 inhibition was more sensitive to crRNA concentration in our samples, while lower Cas13d expression only moderately decreased inhibitory activity (Fig. S3B) . Thus, maintaining a high level of crRNA concentration is likely a determining factor for more effective SARS-CoV-2 sequence targeting and cleavage.\\n\\nThese data together suggest: 1) Cas13d PAC-MAN can be an effective system to target and degrade SARS-CoV-2 sequences in human cells and 2) proper design of crRNAs are important for obtaining a high efficiency of SARS-CoV-2 inhibition.\\n\\nSince we do not yet have access to live SARS-CoV-2 virus, as a proof-of-concept we elected to test the CRISPR-Cas13d strategy on inhibiting H1N1 IAV, a RNA virus with a similar tropism as SARS-CoV-2 for respiratory tract epithelial cells. In contrast to SARS-CoV-2, which has only a single, continuous RNA genome, the IAV genome is contained in 8 negative-sense RNA segments, in which each plays an important role in packaging the viral RNA into budding virions 16, 17 .\\n\\nTo create crRNAs targeting a broad range of IAV strains, we aligned all complete IAV genomes retrieved from the Influenza Research Database and searched for the most conserved regions (Supplemental Table 5 ). Our analysis and the work of others showed that the segment ends are highly conserved for all 8 segments [16] [17] [18] [19] [20] [21] . The segment ends are potentially attractive regions for targeting with Cas13d because of the potential to inhibit a broad range of IAV strains and because of the previous evidence that interfering with the packaging of one segment encoded on the segment ends can decrease the packaging efficiency of other segments and overall virion packaging 18, [22] [23] [24] [25] . Using sequence representation from ~140 different strains of influenza, we designed crRNAs that could be robust across as many different IAV strains as possible. Our bioinformatics analysis resulted in 6 crRNAs targeting highly conserved genome regions for each of the 8 IAV segments, for a total of 48 crRNAs (Supplemental Table 6 ).\\n\\nauthor/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript\\n\\nTo test the efficiency of IAV-targeting crRNAs, we performed a screen of our crRNA pools in the stable Cas13d A549 cell line to determine which were most effective at inhibiting IAV infection (Fig. 4A) . To do this, we transfected a pool of 6 crRNAs targeting a given IAV genome segment. Two days after crRNA transfection, the Cas13d A549 cells were challenged with PR8-mNeon, a strain of H1N1 IAV (A/Puerto Rico/8/1934) engineered to express the mNeonGreen gene, a fluorescent reporter protein (hereafter referred to as PR8-mNeon) at an MOI of 2.5 or 5.0 [26] [27] [28] . At approximately 18 hours post-challenge, the cells were analyzed for IAV infection through flow cytometry and microscopy. We examined the percentage of cells that were mNeon positive for each segment-targeting crRNA pool and calculated the fold-change compared to cells transfected with a pool of non-targeting crRNAs). Out of all crRNAs, the crRNA pool targeting segment 6 (S6) delivered the most consistent and robust results across different MOIs (72% reduction for MOI = 2.5 and 52% reduction for MOI = 5; Fig. 4B, S4) . Additionally, the segment 4 (S4) pool also showed moderate but consistent inhibition. To verify Cas13d-mediated inhibitory effects on S4 and S6, we further tested at a lower MOI of 0.5, which showed better inhibition of IAV across samples (e.g., 79% reduction using crRNAs targeting S6; Fig. S5 ). This confirms consistent inhibitory activity of IAV by Cas13d when targeting these two segments. There is a possible trend of greater inhibition at lower viral titers, suggesting that Cas13d PAC-MAN could be a particularly effective strategy for prevention of new infections, which typically occurs from exposure to a low level of virus.\\n\\nWe further characterized the S6-targeting crRNA pool. S6 encodes the gene neuraminidase (NA), which releases budding virions from the host cell 16 . We performed microscopy quantification of mNeon+ cells and found that mNeon expression was reduced by 62% (p = 10 -5 ) and 73% (p = 5x10 -9 ) with an MOI of 2.5 and 5, respectively, with S6-targeting crRNAs (Fig. 4C) and this trend was corroborated by flow cytometry (Fig. 4D) . Together, our data show that Cas13d PAC-MAN is able to target highly conserved viral regions and robustly inhibit viral replication in human lung epithelial cells.\\n\\nIn the past two decades, multiple variants of coronavirus, including those causing COVID-19, SARS and MERS, emerged from animal reservoirs and infected humans, each time causing significant disease and fatalities 1-3 . Thus, the design of a strategy that could broadly prevent and target viral threats, including all coronavirus strains that currently reside in animals, would be a valuable resource. We asked whether it is possible to design a minimal number of crRNAs that could target a majority of known coronaviruses found in both humans and animals.\\n\\nTo do this, we analyzed all known coronavirus genomes (3,051) to identify all possible crRNAs that can target each of those genomes, ending up with approximately 6.48 million possible crRNAs (Fig. 5A, S6) . We then refined our list to define the smallest number of crRNAs that could target all known coronavirus genomes with zero mismatches. We found that just two crRNAs could target ~50% of coronavirus genomes, including those causing COVID-19, SARS, and MERS; 6 crRNAs were able to target ~91% of coronavirus genomes; and 22 crRNAs covered all sequenced coronaviruses with no mismatches (Fig. 5B, Supplemental Table 7 ). The ability to use a relatively small number of crRNAs to broadly target most or all coronavirus strains further highlights the uniqueness of our approach in contrast to traditional pharmaceutical or vaccination approaches.\\n\\nAs a resource, we provide a map to show how our top 6 crRNAs (PAC-MAN-T6) targeted the coronavirus phylogenetic tree (Fig. 5A, red box) . Beta coronaviruses, the genus of viruses causing COVID-19, SARS, and MERS, were mostly covered by 3 crRNAs (crRNA 1-3) , while the other genera were covered by these three plus the remaining 3 crRNAs. Altogether, the PAC-MAN-T6 can target all known human coronavirus with broad coverage against other animal coronaviruses.\\n\\nThe minimal set of six crRNAs also included one crRNA (designated crRNA-N18f) that we validated. A recent work analyzed 103 SARS-CoV-2 genomes and indicated that there are two major types (L and S) of SARS-CoV-2 existing with different viral signature sequences 4 . To verify whether crRNA-N18f can target all L and S strains, we retrieved a set of 202 recently sequenced SARS-CoV-2 sequences from Global Initiative on Sharing All Influenza Data author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript\\n\\n(GISAID). Notably, crRNA-N18f targets all 201 genomes, with only one SARS-CoV-2 sequence carrying a single mismatch mutation (Supplemental Table 8 ). This suggests that our approach can robustly target multiple SARS-CoV-2 strains, but also suggests the potential need to use a pool of crRNAs to avoid mutational escapes. author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript\\n\\nCRISPR-Cas13 emerged as a natural bacterial defense against bacteriophages, a powerful antiviral system that is able to use sequence-specific crRNAs to protect host bacterial cells from infections 29 . Here, we have repurposed the RNA-guided RNA endonuclease activity of Cas13d in mammalian cells against emergent viral targets, SARS-CoV-2 and IAV, in our PAC-MAN strategy. This expands the applications of CRISPR-Cas13 systems in addition to their uses for diagnostics such as SHERLOCK and live-cell imaging [30] [31] [32] [33] . Prior work showed Cas13a/b systems could inhibit ssRNA viruses including influenza, which is consistent with our observed high efficiency of SARS-CoV-2 reporter and IAV inhibition using the PAC-MAN approach 31 . We demonstrated that Cas13d-based genetic targeting can effectively target and cleave the RNA sequences of SARS-CoV-2 fragments and IAV in lung epithelial cell cultures. Our bioinformatics analysis also suggested a minimal pool of 6 crRNAs able to target over 90% of sequenced coronaviruses. While this strategy is a proof-of-concept and will require further testing using replication-competent SARS-CoV-2 viruses and validation in animal models before clinical tests in humans, it represents a unique approach to implement a rapid and broad antiviral defense in humans against emerging pathogens for which there are no effective vaccines.\\n\\nFor both SARS-CoV-2 and IAV, we found highly conserved regions of the viral genomes to target with Cas13d. In the case of SARS-CoV-2, these genome regions encode the RdRP and Nucleocapsid proteins, which are essential for coronavirus replication and function 13, 34 . RdRP is responsible for catalyzing the replication of all viral mRNAs and Nucleocapsid binds to genomic RNA, protecting it and serving as one of the two main structural proteins in virions. Targeted inhibition of these proteins could have an outsized effect on disabling virus production and function, in addition to reducing viral load through degradation of the viral genome itself. For IAV, the crRNAs that showed the best viral inhibition target the conserved ends of the segment S6 that encodes Neuraminidase (NA), a viral surface protein essential for mediating budding of new virions 35 . Prior mutation analysis has shown that mutating the packaging signals of one segment end leads to decreased packaging of other segments, indicating that there is a cooperative mechanism for packaging the IAV genome 18 . Thus, by inhibiting a single IAV genome segment, we could reduce the overall packaging of IAV in a synergistic fashion.\\n\\nNotably, the SARS-CoV-2 RdRP gene and IAV NA gene that we targeted in our study are also the targets of two antiviral drugs, Remdesivir and Oseltamivir. Remdesivir, created by Gilead Sciences (development code GS-5734), is an RdRP inhibitor that was developed to treat Ebola and Marburg virus infections and has been reported to have antiviral activity against MERS and SARS 36-38 . Remdesivir is already administered for COVID-19 through compassionate use requests, and clinical trials for its use against COVID-19 have already started in the U.S.\\n\\n(clinical trial ID NCT04280705) and China. Oseltamivir (also known as Tamiflu), is an antiviral medication for treating and preventing IAV that acts as an inhibitor of NA, thereby preventing the egress of new viral particles 39 . It will be interesting to see if these drugs could be used in combination with Cas13d PAC-MAN to synergistically cripple viral function and replication.\\n\\nWhile we demonstrate that the PAC-MAN system is able to exhibit strong repression of viral sequences in an in vitro setting, it must be coupled with an effective in vivo delivery system for therapeutic use. There are several potential strategies that could be employed for transient in vivo expression of CRISPR components. Cas13d and its cognate crRNAs could be delivered as RNAs within polymers or lipid nanoparticles, perhaps with chemical alterations to increase stability 40 . In addition, a DNA-based liposomal delivery strategy, such as the recently developed HEDGES platform is also attractive 41 . Yet another strategy would be to deliver a author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript 13 ribonucleoprotein complex containing purified Cas13d assembled with crRNAs 42, 43 . All of these could potentially be used both as a therapeutic and as a prophylactic measure.\\n\\nIf the delivery barriers are overcome and a strategy like this is able to be used therapeutically, there are many potential benefits over traditional vaccines. Traditional vaccines rely on priming the immune system through exposure to viral proteins or peptides often derived from surface proteins that exhibit a high rate of mutation, which increased the chances of viral evasion of the host immune response 44 . In contrast, here we have demonstrated a genetic strategy that is able to target highly conserved regions, which would be expected to make it much more unlikely for the virus to escape inhibition through mutation. In addition, the ability of Cas13d to process its own crRNAs from a crRNA array means that multiple crRNAs targeting different regions (e.g. Fig. 3E , and PAC-MAN-T6 in Fig. 5A ) could be delivered simultaneously 9 , further reducing the chances of viral escape. As a further advantage, we demonstrate a potential pan-coronavirus strategy to target not only viruses that circulate in humans, but also those that currently are found in animal reservoirs that might transfer to humans to cause disease. If the crRNAs targeting these viruses can be tested and validated before they ever infect humans, we can greatly accelerate the development of countermeasures for future emergent threats.\\n\\nIn summary, our PAC-MAN strategy represents a potentially powerful new approach for inhibiting viral function and replication, and we envision it could be useful for a diverse array of circulating and emergent viral threats.\\n\\n47 complete SARS-CoV-2 assemblies were downloaded from (https://www.ncbi.nlm.nih.gov/genbank/2019-ncov-seqs/) on February 21, 2020. All the other 3,108 Coronavirinae complete genome sequences were downloaded from ViPR (Virus Pathogen Resource) database on February 21, 2020. After excluding very short genomes, we collected in total 3,051 Coronavirinae virus genomes. 202 SARS-CoV-2 genome sequences were downloaded from GISAID (Global Initiative on Sharing All Influenza Data; https://www.gisaid.org/).\\n\\nWe developed a full computational workflow to design crRNAs for the Cas13d system. To design all possible crRNAs for the three pathogenic RNA viruses (SARS-CoV-2, SARS-CoV, and MERS-CoV), the reference genomes of SARS-CoV, MERS-CoV, along with SARS-CoV-2 genomes derived from 47 patients were first aligned by MAFFT using the --auto flag. crRNA candidates were identified by using a sliding window to extract all 22 nt sequences with perfect identity among the SARS-CoV-2 genomes. We annotated each crRNA candidate with the number of mismatches relative to the SARS-CoV and MERS-CoV genomes, as well as the GC content. 3,802 crRNA candidates were selected with perfect match against the 47 SARS-CoV-2 genomes and with ≤1 mismatch to SARS-CoV or MERS-CoV sequences. To characterize the specificity of 22nt crRNAs, we ensured that crRNA do not target any sequences in the human transcriptome. We used Bowtie 1.2.2 to align crRNAs to the human transcriptome (HG38; including non-coding RNA) and removed crRNAs that mapped to the human transcriptome with fewer than 2 mismatches.\\n\\nWe maximized the vertical coverage of pooled crRNAs to target a maximal set of the 3,051 Coronavirinae viruses with a minimal set of crRNAs. A sliding window was used to extract all unique 22 nt sequences in the Coronavirinae genomes, yielding 6,480,955 unique crRNA candidates. A hash table was used to map each crRNA candidate to the number of Coronavirinae genomes that it targets with a perfect match. A greedy algorithm was used to identify a minimal set of covering crRNAs. In each iteration, the crRNAs were sorted in descending order by the number of genomes targeted. The crRNA that targeted the most genomes was added to the minimal set of crRNAs, and all of the genomes it targeted were removed from the hash table values. If multiple crRNAs were tied for targeting the most genomes, the crRNAs were ranked by the length and quantity of stretches of repeated nucleotides, and the lowest ranking crRNA (i.e. the one with the fewest, shortest stretches of repeated nucleotides) was added to the set of minimal crRNAs. Using this approach, a set of 22 crRNAs was identified that was able to target all 3150 Coronavirinae virus genomes (including 47 SARS-CoV-2). The phylogenetic tree and the associated annotations were generated by the Interactive Tree Of Life (iTOL v5.5; https://itol.embl.de/).\\n\\nAll the complete SARS-CoV-2 genome sequences were aligned with the SARS-CoV and MERS-CoV NCBI Refseq sequences by MAFFT using the --auto flag. The alignment was visualized using the Geneious Prime software (Geneious 2020.0.4; https://www.geneious.com).\\n\\nauthor/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript\\n\\nA consensus nucleotide (which is shared by >50% of non-gap characters at the given position) was assigned to each position in the alignment. Not every position has a consensus nucleotide.\\n\\nThe percent conservation at each position was calculated as the percentage of sequences matching the consensus at that position. To aid interpretation, the resulting plot of percentage conservation was smoothed using a moving average with a window size of 1,001 bp centered at each position. Positions corresponding to gap characters in the SARS-CoV-2 reference sequence were omitted.\\n\\nThe crRNA plasmids were cloned using standard restriction-ligation cloning. Forward and reverse oligos for each spacer (IDT) were annealed and inserted into backbone using T4 DNA Ligase (NEB). Assembled oligos were either inserted into a pHR or pUC19 backbone, obtained and modified from Addgene (#121514 and #109054, respectively). Spacer sequences for all crRNAs can be found in Supplemental Table 2 and 6.\\n\\nTo clone SARS-CoV-2 reporters, pHR-PGK-scFv GCN4-sfGFP (pSLQ1711) was digested with EcoR1 and Sbf1 and gel purified. The PGK promoter and sfGFP were PCR amplified from pSLQ1711 and gel purified. The SARS-CoV-2-F1 and SARS-CoV-2-F2 fragments were synthesize from Integrated DNA Technologies (IDT). At last, the PGK, sfGFP, and SARS-CoV-2-F1/F2 fragments were inserted into the linearized pSLQ1711 vector using In-Fusion cloning (Takara Bio).\\n\\nOn day 1, one confluent 100mm dish (Corning) of HEK293T cells were seeded into three 150mm dishes (Corning). On day 2, cells were ~50-70% confluent at the time of transfection. For A549 cells, lentivirus precipitation solution (Alstem) was added and precipitated as per the manufacturer's protocol. One well of a 6-well plate (Corning) of A549 cells at ~50% confluency were transduced with the precipitated lentivirus. After 2-3 days of growth, the cell supernatant containing virus was removed and the cells were expanded. Cells were then sorted for mCherry+ cells using a Sony SH800S cell sorter. Table S2 ). On day 3, cells were switched to fresh medium with Puromycin at 2 ug/mL. Twenty-four hours after Puromycin selection, cells with transfected crRNAs were challenged with SARS-CoV-2 reporter plasmids or SARS-CoV-2 reporter lentivirus. Twenty-four hours after transfection or 48 hours after transduction, cells were tested using flow cytometry on a Beckman-Coulter CytoFLEX S and qRT-PCR on a Biorad author/funder. All rights reserved. No reuse allowed without permission. \\n\\nThe PR8 mNeon IAV was generated by insertion of the mNeon fluorescent gene into segment 4 of the H1N1 A/Puerto Rico/8/1934 strain (Nicholas Heaton's Lab, Duke University) as previously described . The PR8-mNeon virus was propagated in chicken eggs, and harvested allantoic fluid containing the virus was titered using MDCK plaque assays, and frozen in aliquots for later use.\\n\\n3.1x10 6 Cas13d A549 cells were plated in two 100 mm dishes. The following day, the cells were transfected with 15 μg pooled crRNAs targeting the IAV genome or non-targeting controls (Supplemental Table 6 ) in 1.5 mL Opti-MEM reduced serum media and 30 μL Mirus LT1 reagent. After approximately 24 hours, the transfected Cas13d A549 cells were dissociated and re-plated in 24-well plates at a density of 15-20,000 cells/well and incubated overnight. Flow cytometry was performed on excess cells to quantify the transfection efficiency. Cells were challenged overnight with mNeonGreen expressing H1N1 A/Puerto Rico/8/1934 at an MOI of 2.5 and 5. Uninfected cells were included as a negative control. An independent two-tailed T test was used to calculate the p values and statistics in Fig. 4C-D. author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint Development of CRISPR as a prophylactic strategy to combat novel coronavirus and influenza Abbott et al. Manuscript\\n\\nReal-time qRT-PCR was performed to quantify RNA abundance. For each sample, total RNA was isolated by using the RNeasy Plus Mini Kit (Qiagen Cat# 74134), followed by cDNA synthesis using the iScript cDNA Synthesis Kit (BioRad, Cat# 1708890). qRT-PCR primers were ordered from Integrated DNA Technologies (IDT). Quantitative PCR was performed using the PrimePCR assay with the SYBR Green Master Mix (BioRad), and run on a Biorad CFX384 realtime system (C1000 Touch Thermal Cycler), according to manufacturers' instructions. Cq values were used to quantify RNA abundance. The relative abundance of the SARS-CoV-2 fragments was normalized to a GAPDH internal control.\\n\\nMicroscopy for SARS-CoV-2 experiments was performed on a Nikon TiE inverted confocal microscope equipped with an Andor iXon Ultra-897 EM-CCD camera and 405 nm, 488 nm, 561 nm and 642 nm lasers, using the 20x PLAN APO air objective (NA = 0.75). Image were taken using NIS Elements version 4.60 software. Microscopy for IAV experiments was performed on a Keyence BZ-X810 microscope equipped with a GFP and Texas Red filter was used to obtain images with a 20X objective lens.\\n\\nA549 cells that express Cas13d and are transfected with crRNAs and challenged with PR8 mNeon virus were washed with 1X PBS, treated with 1X Trypsin EDTA and fixed in 1.6% paraformaldehyde in PBS for 10 minutes. Cells were washed with FACS buffer (1X PBS, 0.3% BSA, 1mM EDTA). BD Accuri C6 plus flow cytometer was used to collect raw data. Raw data were processed with Cytobank (BD) software.\\n\\nauthor/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.13.991307 doi: bioRxiv preprint\\n\\n  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>83</td>\n",
       "      <td>In trans variant calling reveals enrichment for compound heterozygous variants in genes involved in neuronal development and growth</td>\n",
       "      <td>[{'text': 'Compound heterozygotes occur when different mutations at the same locus on both maternal and paternal chromosomes produce a recessive trait. Here we present the tool VarCount for the quantification of mutations at the individual level. We used VarCount to characterize compound heterozygous coding variants in patients with epileptic encephalopathy and in the 1000 genomes participants. The Epi4k data contains variants identified by whole exome sequencing in patients with either Lennox-Gastaut Syndrome (LGS) or Infantile Spasms (IS), as well as their parents. We queried the Epi4k dataset (264 trios) and the phased 1000 genomes data (2504 participants) for recessive variants. To assess enrichment, transcript counts were compared between the Epi4k and 1000 genomes participants using minor allele frequency (MAF) cutoffs of 0.5% and 1.0%, and including all ancestries or only probands of European ancestry. In the Epi4k participants, we found enrichment for rare, compound heterozygous mutations in six genes, including three involved in neuronal growth and development -PRTG (p=0.00086, 1% MAF, combined ancestries), TNC (p=0.0221% MAF, combined ancestries), and MACF1 (p=0.0245, 0.5% MAF, EU ancestry). Due the total number of transcripts considered in these analyses, the enrichment detected was not significant after correction for multiple testing and higher powered or prospective studies are necessary to validate the candidacy of these genes.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'However, PRTG, TNC, and MACF1 are potential novel recessive epilepsy genes and our results highlight that compound heterozygous mutations should be considered in sporadic epilepsy.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>Using the premise that effective variants are in linkage disequilibrium (LD) with common polymorphisms and haplotypes, linkage and association studies have identified genes involved in the development of traits and pathologies. Upon their identification, the regions flanking associated markers are sequenced to find the linked, penetrant mutation. However, rare variants are often not detectable using LD-based methods.\\n\\nThis problem has been alleviated by recent advances in next-generation sequencing (NGS) , and the detection of highly penetrant rare variants associated with disease has reduced the heritability gap for such diseases as autism, Crohn's disease, and osteoporosis (Bomba, Walter, &amp; Soranzo, 2017; Kosmicki, Churchhouse, Rivas, &amp; Neale, 2016) . Despite these advances, for most traits and complex disorders the underlying genes and mutations remain elusive.\\n\\nRecessive traits are caused by mutations in both copies of a gene. The mutations may be homozygous, i.e. identical, or compound heterozygous. Compound heterozygous (CH) mutations are two different mutations in a gene on opposite alleles of a chromosome and it is speculated that compound heterozygous mutations account for many recessive diseases (Li et al., 2010; Sanjak, Long, &amp; Thornton, 2017) . Lack of detection of CH may explain a significant portion of missing heritability for all phenotypes (Li et al., 2010; Sanjak et al., 2017; Zhong, Karssen, Kayser, &amp; Liu, 2016) .\\n\\nAssociation studies using polymorphisms are LD-based and recent association studies using rare variants compare total variant burden between cases and controls to account for the contributions of multiple alleles at a locus to phenotype. Importantly, because LD-based studies require recessive mutations to be on the same genetic background and total variant burden analyses are not allele-specific, neither discerns between dominant and recessive models of inheritance. 2010). The vcf files were then annotated with minor allele frequencies (MAFs) from EVS (Exome Variant Server, NHLBI GO Exome Sequencing Project (ESP), Seattle, WA (URL: http://evs.gs.washington.edu/EVS/)), 1000 genomes and ExAC (Monkol Lek et al., 2015) , and with information regarding the effect of each variant using SNPSift/SNPEff (Cingolani et al., 2012) . The databases used for annotation were dbNSFP2.9 (for MAF and CADD score) and GRCh37.75 for protein effect prediction.\\n\\nSnpSift was used to remove any variants not inducing a protein-changing event (not \"HIGH\" or \"MODERATE\" impact) based on SNPEff annotation -this includes missense, nonsense, splice-site and insertion/deletion variants. Variants with quality flags and multiallelic variants, i.e. those with more than two known nucleotide values, were also removed. Variants remaining after filtering were cross-referenced with the 1000 genomes variants from the same MAF threshold to ensure that any variants removed from one dataset were removed from the other. The annotated vcf was used as input for VarCount. Ancestry for each exome was determined using LASER (Wang et al.) and this information was input to VarCount via the SampleInfo.txt file. Ancestry and phenotype information for each proband are described in Supplementary Table 1. In addition to the annotated vcf file, the parameters.txt and subjectinfo.txt (containing sex and ancestry information) were used as input. Within the parameters file, the following qualifications were selected: (1) counting at the transcript (rather than gene) level, (2) protein-changing effects, (3) MAF threshold of either 0.005 or 0.01, (4) all within-dataset and annotated (1000 genomes, ExAC and EVS) MAFs, and (5) either compound heterozygous or homozygous mutations. Analyses were run separately for the two MAFs and using all Epi4k probands (264) and only those of Eurpoean ancestry (207) . Because the variants were not phased, VarCount was used to query the vcf file for individuals with two or more mutations in each transcript. The output, a list of author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint counts for each transcript was then used to query the parental vcf files for genotype information to determine which sets of variants composed in trans combinations of mutations. Final counts were determined using parental genotype information. Custom python scripts were used to query for parental genotypes and to count true compound heterozygotes or homozygotes. De novo mutations were excluded in the determination of true in trans mutations.\\n\\nVcf files for the 2504 participants in the 1000 genomes sequencing project (Genomes Project et al., 2015) were downloaded by chromosome from the 1000 genomes ftp site.\\n\\nTo reduce input file size, the genomic regions for the hg19 mRNA transcripts were downloaded via UCSC's Table Browser and used to remove non-coding regions from the vcf files. Including all exons from UCSC allowed for a more conservative analysis, given that the Epi4k data were sequenced using various exome captures, which are not inclusive of all possible exons. The variants were annotated and filtered via the same steps as the Epi4k vcf file. Multi-allelic variants were also removed prior to analysis by VarCount. A diagram showing the steps involved in processing and analyzing the variant files is shown in (Figure 1 ). VCF files were queried for homozygous and compound heterozygous mutations using VarCount. Because the variants in the 1000 genomes vcf files are phased, determining true compound heterozygotes is automatic using VarCount. In addition to the annotated vcf file, the parameters.txt and subjectinfo.txt (containing sex and ancestry information) were used as input. Within the parameters file, the same qualifications used in the Epi4k analysis were selected: Analyses were run for each of the two MAFs (0.5% and 1.0%) and for all 1000 genomes participants and using only author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint those of European (EUR) ancestry. The final output from analyses was for each MAF cutoff and for each population, counts for every transcript in which at least one individual harbored recessive variants.\\n\\nUsing R statistical software, a Fisher's exact test was used to detect transcripts with significant differences in the proportion of individuals with homozygous or compound heterozygous variants between the Epi4k dataset and the 1000 genomes dataset. Odds ratios and p-values were calculated using the number of individuals with and without qualifying mutations in each superpopulation. Analyses were performed using all ancestries, and for only individuals of European ancestry. Both Bonferroni and Benjamini-Hochberg adjustments were used to determine significance thresholds after correction for multiple testing. The number of tests was based on the number of transcripts with at least one individual in either the Epi4k or 1000 genomes dataset with in trans coding variants with minor allele frequencies below the set threshold.\\n\\nThe three-dimensional structure of Protogenin (PRTG) was modeled off the crystal structure of the human receptor protein tyrosine phosphatase sigma (PDB: 4PBX; 25.1% sequence identity) using MODELLER 9.14 (Webb &amp; Sali, 2016 ). The resultant model superimposed with the template had an RSMD of 4.94 Å over 442 Ca atoms.\\n\\nProtein and solvent dielectric constants were set to 2.0 and 78.0, respectively. All structural figures were generated by PyMOL(W).\\n\\nVarcount is a free, open source tool useful for the quantification of heterozygous, homozygous, or compound heterozygous mutations per sample. Input variants may be phased or unphased. All python scripts and supporting files may be downloaded from Github at https://github.com/GeneSleuth/VarCount. Supporting files include the \"parameters.txt\" file where the user may select variant filters for mutation effect, minor allele frequency, and inheritance pattern (homozygous, compound heterozygous, one mutation or two mutations), and Sample filters based on information entered into the \"SampleInfo file\". Input vcf files must be annotated with SNPSift/SNPEff using the dbNSFP and GRCh37/38 databases. A readme file with instructions is also provided.\\n\\nA flow diagram with the steps involved in processing of data is depicted in Figure 1 .\\n\\n[ Figure 1 near here]\\n\\nWe used VarCount to query the Epi4k dataset for rare homozygous and CH mutations.\\n\\nThe Epi4k data are whole exome data from 264 trios with a child affected by epileptic encephalopathy, either infantile spasms (IS) or Lennox-Gastaut syndrome (LGS) (6).\\n\\nCounts were performed using individuals of all ancestries or just those of European ancestry (207/264). Individuals from the 1000 genomes study were used as controls.\\n\\nThe individual counts and p-values for the analyses are listed in (Supplementary Tables   2-5 ). Including only rare variants (MAFs below 0.5% and 1.0%) determined enrichment for compound heterozygous mutations in six genes. For combined ancestries, the six genes are in order of significance: OSBP2, PRTG, ABCC11, MACF1, STAB1, and TNC.\\n\\nPRTG and TNC were also highly ranked in the 1% MAF analysis, with one additional count for each transcript. Variants for all six genes are listed in Table 1 . In our analysis of just individuals of European ancestry, MACF1 was the most significantly enriched gene using a 0.5% MAF. The p-values indicated in Table 1 are for individual tests; there were no p-values significant after correction for multiple testing.\\n\\n[ Table 1 near here]\\n\\nThe variants for the three individuals with CH PRTG mutations are depicted in [ Figure 2 near here]\\n\\nThe de novo variants identified by Epi4K Consortium and the Epilepsy Phenome/Genome Project (Epi et al., 2013) in the nine probands with either PRTG, TNC, or MACF1 recessive variants are described in Table 2 . For the three patients with compound heterozygous PRTG variants, one patient harbors a de novo missense variant in HSF2, the second has a nonsense variant in CELSR1, and the third patient has two de novo variants -a missense in Fam102A and a 3'UTR variant in USP42. De novo mutations were only reported in one of the probands with in trans TNC variants -a author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint missense variant in DIP2C and a splice donor change in IFT172. All three patients with compound heterozygous variants in MACF1 were reported to have de novo mutations.\\n\\nThe first patient has a 5' and 3'UTR de novo variant in FAM19A2 and GLRA2, respectively and the second patient also has a 3'UTR de novo change in the gene LRRC8D and a missense change in SNX30. One de novo variant was identified in the third proband in the gene FAM227A. Polyphen2 categories and CADD scores for each de novo variant as well as missense and loss-of-function constraint metric values for each gene (from ExAC) are also listed in Table 2 . The z-score is a ratio of expected to identified missense variants in a particular gene, and pLI is a gene's probability of being loss-of-function intolerant. These constraint metrics are calculated using genomic data from controls without severe genetic diseases in the ExAC database (Monkol Lek et al., 2015) .\\n\\n[ Table 2 near here]\\n\\nEpileptic encephalopathies are a group of severe, early-onset seizure disorders with consistent EEG abnormalities that over time interfere with development and cause cognitive decline (Covanis, 2012) . The Epi4k dataset contains exome sequence from 264 trios that include a proband with epileptic encephalopathy, either Lennox-Gastaut Syndrome (LGS) or Infantile Spasms (IS). LGS is characterized by frequent, mixed epileptic seizures that arise most frequently between the ages of 3 and 5 (Amrutkar and Riel-Romero, 2018) . IS occurs during the first year of life and is cryptic in its presentation, with mild head bobbing and is often not detected until the seizures have caused significant neurological damage (Kossoff, 2010) . IS often progress into LGS over time.\\n\\nauthor/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint\\n\\nWe developed a free and user-friendly tool, VarCount, to query vcf files for individuals harboring variants that qualify according to user specification. To test its function, we used VarCount to quantify rare, compound heterozygous mutations in probands from the Epi4k trio dataset and found enrichment for variants in six genes including PRTG, TNC and MACF1. PRTG codes for protogenin, a member of the immunoglobulin superfamily that is involved in axis elongation and neuronal growth during early vertebrate development (Toyoda, Nakamura, &amp; Watanabe, 2005; Vesque, Anselme, Couve, Charnay, &amp; Schneider-Maunoury, 2006) . TNC and MACF1 are also directly involved in neuronal development and/or growth. TNC (Tenascin-C) is an extracellular matrix glycoprotein involved in axonal growth and guidance (Jakovcevski, Miljkovic, Schachner, &amp; Andjus, 2013) . Seizures up-regulate TNC in the hippocampus, and in a pilocarpine epilepsy model up-regulation was shown to be mediated by TGF-b signalling (Mercado-Gomez, Landgrave-Gomez, Arriaga-Avila, Nebreda-Corona, &amp; Guevara-Guzman, 2014). MACF1 is a cytoskeletal crosslinking protein highly expressed in the brain and is crucial for neuron development and migration (Moffat, Ka, Jung, Smith, &amp; Kim, 2017) . MACF1 mutations are associated with the neurological pathologies Parkinson's disease, autism, and schizophrenia (Moffat et al., 2017) .\\n\\nRecently, highly penetrant de novo MACF1 mutations were identified in several patients with a newly characterized lissencephaly with a complex brain malformation (Dobyns et al., 2018) . This new phenotype highlights MACF1 mutations' variable impact on disease pathogenesis. Given both the enrichment in Epi4k probands for compound heterozygous mutations in these genes as well as their known involvement in neuronal processes, we suggest that PRTG, TNC, and MACF1 are candidate recessive epilepsy genes.\\n\\nauthor/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint\\n\\nThe primary publication reporting analysis of the Epi4k trio dataset was a description of de novo mutations in the probands (Epi et al., 2013 ). An analysis of compound heterozygous variants was also reported, using a minor allele frequency cutoff of 0.15%, which is lower than the cutoff used in the work presented here. In this analysis, the parents were used as internal controls, and compound heterozygous variants in 351 genes were identified, without genome-wide significance. The authors only listed five of the genes which are known to cause Mendelian disorders that include a seizure phenotype -ASPM, CNTNAP2, GPR98, PCNT, and POMGNT1. In our analysis using the 1000 genomes participants as controls, enrichment for compound heterozygous variants was not detected in any of these genes. Using the number of individuals with in trans variants in a gene (transcript) as an indicator function required at least two probands to have qualifying variants in order to detect single-test significance, with complete absence of qualifying variants in controls. It is clear from the analyses using either internal controls or the 1000 genomes as controls that a larger sample size is required to achieve genome-wide significance.\\n\\nThe de novo variants reported by the Epi4K Consortium and the Epilepsy Phenome/Genome Project (Epi et al., 2013) in the nine probands with compound heterozygous variants in PRTG, TNC, or MACF1 are described in Table 2 . Of the twelve genes with de novo variants identified in the nine patients, three are implicated in neurological disease. CELSR1 is a planar cell polarity gene in which mutations are known to cause neural tube defects including spina bifida (Robinson et al., 2012) . De novo deletions of DIP2C have been reported in two patients with cerebral palsy, one of whom also had ADHD, and the other had seizures in infancy (Zarrei et al., 2018) . In another report, deletions including DIP2C and/or ZMYND11 were identified in several patients with developmental delay including three patients with seizures (DeScipio et author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint al., 2012). GLRA2 is a glycine receptor involved in neurodevelopment in which mutations are implicated in autism, (Pilorge et al., 2016; Lin et al., 2017) including a patient with comorbid epilepsy (Zhang et al., 2017) .\\n\\nOf the de novo variants reported in these genes, the nonsense variant in CELSR1 identified in one of the probands with in trans PRTG variants is the most likely to be pathogenic. However, regarding their involvement in neural tube defects, mutations in CELSR1 are thought to contribute to pathogenesis but not in a Mendelian fashion, as variants have been found to be inherited from unaffected parents or to be ineffective in functional assays (Robinson et al., 2012; Allache et al., 2012) . The nonsense CELSR1 mutation in the patient reported here may contribute to epilepsy in the presence of a genetic modifier. The de novo missense mutation in DIP2C is predicted to be deleterious (CADD = 23.6) and has a low rate of benign missense variation based on constraint metrics (z = 5.82). The de novo variant in GLRA2 is in the 3'UTR so it is difficult to predict its impact on gene function and subsequent pathogenicity.\\n\\nThe compound heterozygous variants in PRTG, TNC, and MACF1 are similarly variable in predicted pathogenicity, with CADD scores ranging from between less than one to thirty-eight. PRTG and TNC both have constraint metrics indicative of a high tolerance to both missense and loss-of-function variants, while MACF1 is moderately intolerant of missense variants (z = 2.63) and extremely intolerant of loss-of-function variants (pLI = 1.0). Interestingly, aside from the 3'UTR variant in GLRA2, none of the de novo variants in the Epi4k participants with MACF1 compound heterozygous variants are in genes associated with neurological disease or predicted with confidence to have a negative impact on gene function. This, in addition to MACF1's intolerance to missense or nonsense variants, is supportive of the pathogenicity of the biallelic variants in the gene. author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint\\n\\nIn summary, we present a free tool VarCount for the quantification of qualifying mutations as an indicator function per individual in the analysis of variant lists (vcf files). We used VarCount to assess enrichment of rare, coding, compound heterozygous variants in a cohort of 264 epilepsy probands and found enrichment in three genes involved in neurodevelopmental processes -PRTG, TNC and MACF1. A missense change at the E104 residue of PRTG was identified three times in two different probands. Significance was not maintained after correction for multiple testing, and larger cohorts or candidate gene studies using a different sample set are necessary to validate this enrichment. In the context of the de novo mutations also present in these patients, experimentation is necessary in order to delineate if the compound heterozygous or de novo mutations, or both, are pathogenic in the development of epileptic encephalopathy. PRTG, TNC, and MACF1 are candidate recessive epilepsy genes and our work highlights that inheritance of compound heterozygous variants should not be excluded from gene discovery or diagnostic analyses of patients with epilepsy.\\n\\na Phen=phenotype, b Anc=ancestry, c AF=allele frequency, d y/n corresponds to yes/no counts of individuals with qualifying mutations *bp (base pair position) in hg19/Build37 Ancestries: EU=European, E Asia = East Asia, C/S Asia = Central/South Asia, ME = Middle East p-values in bold are the most significant for the specific analysis EU(y/n)d: All(y/n) Ancestries: EU=European, C/S Asia = Central/South Asia; Phenotypes: IS = Infantile spasms, LGS = Lennox-Gastaut syndrome a z-score is a measure of tolerance to missense variants, based on ratio of expected to identified; b pLI is the probability that a gene is intolerant to loss-of-function variants; c Polyphen2 -prediction of a missense mutation's impact on protein structure and function: B=benign, P=possibly damaging, D=damaging (Adzhubei et al., 2010); d CADD = phred-scaled score of Combined Annotation Dependent Depletion, a measure of the deleteriousness of a SNP or INDEL (Kircher et al., 2014) . author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint The mutations in red and blue were found using a 0.5% MAF threshold and the mutations in black were found using a 1% MAF threshold. (B) Bottom: electrostatic potential surface of PRTG calculated in APBS.\\n\\nTop: Close-up of the PRTG electrostatic potential surface at the site of mutation. The p.Glu104Gly mutation leads to a loss of negative charge, which may disrupt interactions with putative PRTG binding partners. The p.Glu104Asp mutation does not lead to a change in charge or electrostatic potential. author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint LGS F a Ancestry determined by LASER/TRACE software, b IS = Infantile Spasms, LGS = Lennox Gastaut Syndrome *Ancestries: C/S Asia = Central/South Asia, E Asia = East Asia, EU = European, ME = ME\\n\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                  title  \\\n",
       "83  In trans variant calling reveals enrichment for compound heterozygous variants in genes involved in neuronal development and growth   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               abstract  \\\n",
       "83  [{'text': 'Compound heterozygotes occur when different mutations at the same locus on both maternal and paternal chromosomes produce a recessive trait. Here we present the tool VarCount for the quantification of mutations at the individual level. We used VarCount to characterize compound heterozygous coding variants in patients with epileptic encephalopathy and in the 1000 genomes participants. The Epi4k data contains variants identified by whole exome sequencing in patients with either Lennox-Gastaut Syndrome (LGS) or Infantile Spasms (IS), as well as their parents. We queried the Epi4k dataset (264 trios) and the phased 1000 genomes data (2504 participants) for recessive variants. To assess enrichment, transcript counts were compared between the Epi4k and 1000 genomes participants using minor allele frequency (MAF) cutoffs of 0.5% and 1.0%, and including all ancestries or only probands of European ancestry. In the Epi4k participants, we found enrichment for rare, compound heterozygous mutations in six genes, including three involved in neuronal growth and development -PRTG (p=0.00086, 1% MAF, combined ancestries), TNC (p=0.0221% MAF, combined ancestries), and MACF1 (p=0.0245, 0.5% MAF, EU ancestry). Due the total number of transcripts considered in these analyses, the enrichment detected was not significant after correction for multiple testing and higher powered or prospective studies are necessary to validate the candidacy of these genes.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'However, PRTG, TNC, and MACF1 are potential novel recessive epilepsy genes and our results highlight that compound heterozygous mutations should be considered in sporadic epilepsy.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            full_text  \n",
       "83  Using the premise that effective variants are in linkage disequilibrium (LD) with common polymorphisms and haplotypes, linkage and association studies have identified genes involved in the development of traits and pathologies. Upon their identification, the regions flanking associated markers are sequenced to find the linked, penetrant mutation. However, rare variants are often not detectable using LD-based methods.\\n\\nThis problem has been alleviated by recent advances in next-generation sequencing (NGS) , and the detection of highly penetrant rare variants associated with disease has reduced the heritability gap for such diseases as autism, Crohn's disease, and osteoporosis (Bomba, Walter, & Soranzo, 2017; Kosmicki, Churchhouse, Rivas, & Neale, 2016) . Despite these advances, for most traits and complex disorders the underlying genes and mutations remain elusive.\\n\\nRecessive traits are caused by mutations in both copies of a gene. The mutations may be homozygous, i.e. identical, or compound heterozygous. Compound heterozygous (CH) mutations are two different mutations in a gene on opposite alleles of a chromosome and it is speculated that compound heterozygous mutations account for many recessive diseases (Li et al., 2010; Sanjak, Long, & Thornton, 2017) . Lack of detection of CH may explain a significant portion of missing heritability for all phenotypes (Li et al., 2010; Sanjak et al., 2017; Zhong, Karssen, Kayser, & Liu, 2016) .\\n\\nAssociation studies using polymorphisms are LD-based and recent association studies using rare variants compare total variant burden between cases and controls to account for the contributions of multiple alleles at a locus to phenotype. Importantly, because LD-based studies require recessive mutations to be on the same genetic background and total variant burden analyses are not allele-specific, neither discerns between dominant and recessive models of inheritance. 2010). The vcf files were then annotated with minor allele frequencies (MAFs) from EVS (Exome Variant Server, NHLBI GO Exome Sequencing Project (ESP), Seattle, WA (URL: http://evs.gs.washington.edu/EVS/)), 1000 genomes and ExAC (Monkol Lek et al., 2015) , and with information regarding the effect of each variant using SNPSift/SNPEff (Cingolani et al., 2012) . The databases used for annotation were dbNSFP2.9 (for MAF and CADD score) and GRCh37.75 for protein effect prediction.\\n\\nSnpSift was used to remove any variants not inducing a protein-changing event (not \"HIGH\" or \"MODERATE\" impact) based on SNPEff annotation -this includes missense, nonsense, splice-site and insertion/deletion variants. Variants with quality flags and multiallelic variants, i.e. those with more than two known nucleotide values, were also removed. Variants remaining after filtering were cross-referenced with the 1000 genomes variants from the same MAF threshold to ensure that any variants removed from one dataset were removed from the other. The annotated vcf was used as input for VarCount. Ancestry for each exome was determined using LASER (Wang et al.) and this information was input to VarCount via the SampleInfo.txt file. Ancestry and phenotype information for each proband are described in Supplementary Table 1. In addition to the annotated vcf file, the parameters.txt and subjectinfo.txt (containing sex and ancestry information) were used as input. Within the parameters file, the following qualifications were selected: (1) counting at the transcript (rather than gene) level, (2) protein-changing effects, (3) MAF threshold of either 0.005 or 0.01, (4) all within-dataset and annotated (1000 genomes, ExAC and EVS) MAFs, and (5) either compound heterozygous or homozygous mutations. Analyses were run separately for the two MAFs and using all Epi4k probands (264) and only those of Eurpoean ancestry (207) . Because the variants were not phased, VarCount was used to query the vcf file for individuals with two or more mutations in each transcript. The output, a list of author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint counts for each transcript was then used to query the parental vcf files for genotype information to determine which sets of variants composed in trans combinations of mutations. Final counts were determined using parental genotype information. Custom python scripts were used to query for parental genotypes and to count true compound heterozygotes or homozygotes. De novo mutations were excluded in the determination of true in trans mutations.\\n\\nVcf files for the 2504 participants in the 1000 genomes sequencing project (Genomes Project et al., 2015) were downloaded by chromosome from the 1000 genomes ftp site.\\n\\nTo reduce input file size, the genomic regions for the hg19 mRNA transcripts were downloaded via UCSC's Table Browser and used to remove non-coding regions from the vcf files. Including all exons from UCSC allowed for a more conservative analysis, given that the Epi4k data were sequenced using various exome captures, which are not inclusive of all possible exons. The variants were annotated and filtered via the same steps as the Epi4k vcf file. Multi-allelic variants were also removed prior to analysis by VarCount. A diagram showing the steps involved in processing and analyzing the variant files is shown in (Figure 1 ). VCF files were queried for homozygous and compound heterozygous mutations using VarCount. Because the variants in the 1000 genomes vcf files are phased, determining true compound heterozygotes is automatic using VarCount. In addition to the annotated vcf file, the parameters.txt and subjectinfo.txt (containing sex and ancestry information) were used as input. Within the parameters file, the same qualifications used in the Epi4k analysis were selected: Analyses were run for each of the two MAFs (0.5% and 1.0%) and for all 1000 genomes participants and using only author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint those of European (EUR) ancestry. The final output from analyses was for each MAF cutoff and for each population, counts for every transcript in which at least one individual harbored recessive variants.\\n\\nUsing R statistical software, a Fisher's exact test was used to detect transcripts with significant differences in the proportion of individuals with homozygous or compound heterozygous variants between the Epi4k dataset and the 1000 genomes dataset. Odds ratios and p-values were calculated using the number of individuals with and without qualifying mutations in each superpopulation. Analyses were performed using all ancestries, and for only individuals of European ancestry. Both Bonferroni and Benjamini-Hochberg adjustments were used to determine significance thresholds after correction for multiple testing. The number of tests was based on the number of transcripts with at least one individual in either the Epi4k or 1000 genomes dataset with in trans coding variants with minor allele frequencies below the set threshold.\\n\\nThe three-dimensional structure of Protogenin (PRTG) was modeled off the crystal structure of the human receptor protein tyrosine phosphatase sigma (PDB: 4PBX; 25.1% sequence identity) using MODELLER 9.14 (Webb & Sali, 2016 ). The resultant model superimposed with the template had an RSMD of 4.94 Å over 442 Ca atoms.\\n\\nProtein and solvent dielectric constants were set to 2.0 and 78.0, respectively. All structural figures were generated by PyMOL(W).\\n\\nVarcount is a free, open source tool useful for the quantification of heterozygous, homozygous, or compound heterozygous mutations per sample. Input variants may be phased or unphased. All python scripts and supporting files may be downloaded from Github at https://github.com/GeneSleuth/VarCount. Supporting files include the \"parameters.txt\" file where the user may select variant filters for mutation effect, minor allele frequency, and inheritance pattern (homozygous, compound heterozygous, one mutation or two mutations), and Sample filters based on information entered into the \"SampleInfo file\". Input vcf files must be annotated with SNPSift/SNPEff using the dbNSFP and GRCh37/38 databases. A readme file with instructions is also provided.\\n\\nA flow diagram with the steps involved in processing of data is depicted in Figure 1 .\\n\\n[ Figure 1 near here]\\n\\nWe used VarCount to query the Epi4k dataset for rare homozygous and CH mutations.\\n\\nThe Epi4k data are whole exome data from 264 trios with a child affected by epileptic encephalopathy, either infantile spasms (IS) or Lennox-Gastaut syndrome (LGS) (6).\\n\\nCounts were performed using individuals of all ancestries or just those of European ancestry (207/264). Individuals from the 1000 genomes study were used as controls.\\n\\nThe individual counts and p-values for the analyses are listed in (Supplementary Tables   2-5 ). Including only rare variants (MAFs below 0.5% and 1.0%) determined enrichment for compound heterozygous mutations in six genes. For combined ancestries, the six genes are in order of significance: OSBP2, PRTG, ABCC11, MACF1, STAB1, and TNC.\\n\\nPRTG and TNC were also highly ranked in the 1% MAF analysis, with one additional count for each transcript. Variants for all six genes are listed in Table 1 . In our analysis of just individuals of European ancestry, MACF1 was the most significantly enriched gene using a 0.5% MAF. The p-values indicated in Table 1 are for individual tests; there were no p-values significant after correction for multiple testing.\\n\\n[ Table 1 near here]\\n\\nThe variants for the three individuals with CH PRTG mutations are depicted in [ Figure 2 near here]\\n\\nThe de novo variants identified by Epi4K Consortium and the Epilepsy Phenome/Genome Project (Epi et al., 2013) in the nine probands with either PRTG, TNC, or MACF1 recessive variants are described in Table 2 . For the three patients with compound heterozygous PRTG variants, one patient harbors a de novo missense variant in HSF2, the second has a nonsense variant in CELSR1, and the third patient has two de novo variants -a missense in Fam102A and a 3'UTR variant in USP42. De novo mutations were only reported in one of the probands with in trans TNC variants -a author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint missense variant in DIP2C and a splice donor change in IFT172. All three patients with compound heterozygous variants in MACF1 were reported to have de novo mutations.\\n\\nThe first patient has a 5' and 3'UTR de novo variant in FAM19A2 and GLRA2, respectively and the second patient also has a 3'UTR de novo change in the gene LRRC8D and a missense change in SNX30. One de novo variant was identified in the third proband in the gene FAM227A. Polyphen2 categories and CADD scores for each de novo variant as well as missense and loss-of-function constraint metric values for each gene (from ExAC) are also listed in Table 2 . The z-score is a ratio of expected to identified missense variants in a particular gene, and pLI is a gene's probability of being loss-of-function intolerant. These constraint metrics are calculated using genomic data from controls without severe genetic diseases in the ExAC database (Monkol Lek et al., 2015) .\\n\\n[ Table 2 near here]\\n\\nEpileptic encephalopathies are a group of severe, early-onset seizure disorders with consistent EEG abnormalities that over time interfere with development and cause cognitive decline (Covanis, 2012) . The Epi4k dataset contains exome sequence from 264 trios that include a proband with epileptic encephalopathy, either Lennox-Gastaut Syndrome (LGS) or Infantile Spasms (IS). LGS is characterized by frequent, mixed epileptic seizures that arise most frequently between the ages of 3 and 5 (Amrutkar and Riel-Romero, 2018) . IS occurs during the first year of life and is cryptic in its presentation, with mild head bobbing and is often not detected until the seizures have caused significant neurological damage (Kossoff, 2010) . IS often progress into LGS over time.\\n\\nauthor/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint\\n\\nWe developed a free and user-friendly tool, VarCount, to query vcf files for individuals harboring variants that qualify according to user specification. To test its function, we used VarCount to quantify rare, compound heterozygous mutations in probands from the Epi4k trio dataset and found enrichment for variants in six genes including PRTG, TNC and MACF1. PRTG codes for protogenin, a member of the immunoglobulin superfamily that is involved in axis elongation and neuronal growth during early vertebrate development (Toyoda, Nakamura, & Watanabe, 2005; Vesque, Anselme, Couve, Charnay, & Schneider-Maunoury, 2006) . TNC and MACF1 are also directly involved in neuronal development and/or growth. TNC (Tenascin-C) is an extracellular matrix glycoprotein involved in axonal growth and guidance (Jakovcevski, Miljkovic, Schachner, & Andjus, 2013) . Seizures up-regulate TNC in the hippocampus, and in a pilocarpine epilepsy model up-regulation was shown to be mediated by TGF-b signalling (Mercado-Gomez, Landgrave-Gomez, Arriaga-Avila, Nebreda-Corona, & Guevara-Guzman, 2014). MACF1 is a cytoskeletal crosslinking protein highly expressed in the brain and is crucial for neuron development and migration (Moffat, Ka, Jung, Smith, & Kim, 2017) . MACF1 mutations are associated with the neurological pathologies Parkinson's disease, autism, and schizophrenia (Moffat et al., 2017) .\\n\\nRecently, highly penetrant de novo MACF1 mutations were identified in several patients with a newly characterized lissencephaly with a complex brain malformation (Dobyns et al., 2018) . This new phenotype highlights MACF1 mutations' variable impact on disease pathogenesis. Given both the enrichment in Epi4k probands for compound heterozygous mutations in these genes as well as their known involvement in neuronal processes, we suggest that PRTG, TNC, and MACF1 are candidate recessive epilepsy genes.\\n\\nauthor/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint\\n\\nThe primary publication reporting analysis of the Epi4k trio dataset was a description of de novo mutations in the probands (Epi et al., 2013 ). An analysis of compound heterozygous variants was also reported, using a minor allele frequency cutoff of 0.15%, which is lower than the cutoff used in the work presented here. In this analysis, the parents were used as internal controls, and compound heterozygous variants in 351 genes were identified, without genome-wide significance. The authors only listed five of the genes which are known to cause Mendelian disorders that include a seizure phenotype -ASPM, CNTNAP2, GPR98, PCNT, and POMGNT1. In our analysis using the 1000 genomes participants as controls, enrichment for compound heterozygous variants was not detected in any of these genes. Using the number of individuals with in trans variants in a gene (transcript) as an indicator function required at least two probands to have qualifying variants in order to detect single-test significance, with complete absence of qualifying variants in controls. It is clear from the analyses using either internal controls or the 1000 genomes as controls that a larger sample size is required to achieve genome-wide significance.\\n\\nThe de novo variants reported by the Epi4K Consortium and the Epilepsy Phenome/Genome Project (Epi et al., 2013) in the nine probands with compound heterozygous variants in PRTG, TNC, or MACF1 are described in Table 2 . Of the twelve genes with de novo variants identified in the nine patients, three are implicated in neurological disease. CELSR1 is a planar cell polarity gene in which mutations are known to cause neural tube defects including spina bifida (Robinson et al., 2012) . De novo deletions of DIP2C have been reported in two patients with cerebral palsy, one of whom also had ADHD, and the other had seizures in infancy (Zarrei et al., 2018) . In another report, deletions including DIP2C and/or ZMYND11 were identified in several patients with developmental delay including three patients with seizures (DeScipio et author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint al., 2012). GLRA2 is a glycine receptor involved in neurodevelopment in which mutations are implicated in autism, (Pilorge et al., 2016; Lin et al., 2017) including a patient with comorbid epilepsy (Zhang et al., 2017) .\\n\\nOf the de novo variants reported in these genes, the nonsense variant in CELSR1 identified in one of the probands with in trans PRTG variants is the most likely to be pathogenic. However, regarding their involvement in neural tube defects, mutations in CELSR1 are thought to contribute to pathogenesis but not in a Mendelian fashion, as variants have been found to be inherited from unaffected parents or to be ineffective in functional assays (Robinson et al., 2012; Allache et al., 2012) . The nonsense CELSR1 mutation in the patient reported here may contribute to epilepsy in the presence of a genetic modifier. The de novo missense mutation in DIP2C is predicted to be deleterious (CADD = 23.6) and has a low rate of benign missense variation based on constraint metrics (z = 5.82). The de novo variant in GLRA2 is in the 3'UTR so it is difficult to predict its impact on gene function and subsequent pathogenicity.\\n\\nThe compound heterozygous variants in PRTG, TNC, and MACF1 are similarly variable in predicted pathogenicity, with CADD scores ranging from between less than one to thirty-eight. PRTG and TNC both have constraint metrics indicative of a high tolerance to both missense and loss-of-function variants, while MACF1 is moderately intolerant of missense variants (z = 2.63) and extremely intolerant of loss-of-function variants (pLI = 1.0). Interestingly, aside from the 3'UTR variant in GLRA2, none of the de novo variants in the Epi4k participants with MACF1 compound heterozygous variants are in genes associated with neurological disease or predicted with confidence to have a negative impact on gene function. This, in addition to MACF1's intolerance to missense or nonsense variants, is supportive of the pathogenicity of the biallelic variants in the gene. author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint\\n\\nIn summary, we present a free tool VarCount for the quantification of qualifying mutations as an indicator function per individual in the analysis of variant lists (vcf files). We used VarCount to assess enrichment of rare, coding, compound heterozygous variants in a cohort of 264 epilepsy probands and found enrichment in three genes involved in neurodevelopmental processes -PRTG, TNC and MACF1. A missense change at the E104 residue of PRTG was identified three times in two different probands. Significance was not maintained after correction for multiple testing, and larger cohorts or candidate gene studies using a different sample set are necessary to validate this enrichment. In the context of the de novo mutations also present in these patients, experimentation is necessary in order to delineate if the compound heterozygous or de novo mutations, or both, are pathogenic in the development of epileptic encephalopathy. PRTG, TNC, and MACF1 are candidate recessive epilepsy genes and our work highlights that inheritance of compound heterozygous variants should not be excluded from gene discovery or diagnostic analyses of patients with epilepsy.\\n\\na Phen=phenotype, b Anc=ancestry, c AF=allele frequency, d y/n corresponds to yes/no counts of individuals with qualifying mutations *bp (base pair position) in hg19/Build37 Ancestries: EU=European, E Asia = East Asia, C/S Asia = Central/South Asia, ME = Middle East p-values in bold are the most significant for the specific analysis EU(y/n)d: All(y/n) Ancestries: EU=European, C/S Asia = Central/South Asia; Phenotypes: IS = Infantile spasms, LGS = Lennox-Gastaut syndrome a z-score is a measure of tolerance to missense variants, based on ratio of expected to identified; b pLI is the probability that a gene is intolerant to loss-of-function variants; c Polyphen2 -prediction of a missense mutation's impact on protein structure and function: B=benign, P=possibly damaging, D=damaging (Adzhubei et al., 2010); d CADD = phred-scaled score of Combined Annotation Dependent Depletion, a measure of the deleteriousness of a SNP or INDEL (Kircher et al., 2014) . author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint The mutations in red and blue were found using a 0.5% MAF threshold and the mutations in black were found using a 1% MAF threshold. (B) Bottom: electrostatic potential surface of PRTG calculated in APBS.\\n\\nTop: Close-up of the PRTG electrostatic potential surface at the site of mutation. The p.Glu104Gly mutation leads to a loss of negative charge, which may disrupt interactions with putative PRTG binding partners. The p.Glu104Asp mutation does not lead to a change in charge or electrostatic potential. author/funder. All rights reserved. No reuse allowed without permission.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/496133 doi: bioRxiv preprint LGS F a Ancestry determined by LASER/TRACE software, b IS = Infantile Spasms, LGS = Lennox Gastaut Syndrome *Ancestries: C/S Asia = Central/South Asia, E Asia = East Asia, EU = European, ME = ME\\n\\n  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>TITLE: The early scientific literature response to the novel Coronavirus outbreak: who published what? CORRESPONDING AUTHOR</td>\n",
       "      <td>[{'text': 'All rights reserved. No reuse allowed without permission. author/funder, who has granted medRxiv a license to display the preprint in perpetuity.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>Recent events highlight how emerging and re-emerging pathogens are actually becoming global challenges for public health. [1] Coronaviruses are enveloped RNA viruses which are broadly distributed either in humans or in a vast majority of other mammals and birds. These viruses have the possibility to cause respiratory, enteric, hepatic, and neurologic diseases. [2, 3] Coronaviruses are highly prevalent in many species. They have a large genetic diversity. Given their RNA, they are susceptible to frequent recombination and mutations of their genomes. In context in which there are increasing human-animal interactions, novel coronaviruses are very likely to emerge periodically. If the newly created cross-species pathogens acquires the ability to infect humans or to be transmitted human to human it can lead to occasional spillover events and epidemics. [4, 5, 6] In the past years two other strains of Coronavirus -severe acute respiratory syndrome coronavirus (SARS-CoV) and Middle East respiratory syndrome coronavirus (MERS-CoV) -have emerged as potential public health worldwide threats. [5] SARS-CoV was the causal agent of the severe acute respiratory syndrome outbreaks in 2002 and 2003 in Guangdong Province, China; [7, 8, 9] MERS-CoV was the pathogen responsible for severe respiratory disease outbreaks in 2012 in the Middle East. [10] On 31 December 2019, the World Health Organization (WHO) China Country Office was informed of cases of pneumonia of unknown etiology detected in Wuhan City, located in the Hubei Province, China, associated with exposures in a seafood and wet wholesale market in the same city. A new type of coronavirus was isolated on 7 January 2020. [11] On 30 January 2020 WHO declared the outbreak to be a Public Health Emergency of International Concern (PHEIC). [12] This novel coronavirus suddenly turned out to be a global health concern for a disease, called Coronavirus disease 2019 (COVID-19), [13] which was characterized as a pandemic by WHO on 11 March 2020. [14] Starting from the second-half of January 2020, scientific literature has been particularly focused on the description of this new viral outbreak; the main topics addressed were the epidemiological, clinical and virological aspects as well as the possible public health choices necessary to contain the spread of the disease. Nevertheless, several aspects are still unclear and have not been thoroughly explored, leaving grey areas in our knowledge of the disease and of the outbreak. The aim of this paper is to perform a bibliometric analysis on the first papers published in the early stages of the SARS-CoV-2 outbreak, in order to give a glimpse to the researchers of \"who published what\" at the very beginning of this Public Health Emergency of International Concern.\\n\\nWe searched MEDLINE (PubMed) electronic database in order to find all relevant studies. After a preliminary search starting from 1 December 2019, we found out that the first article meeting our inclusion criteria was published on 14 January 2020. [15] [Supplementary materials, Table 1 , ID: 1] We subsequently extended our research to the first 30 days after this milestone article, up to 13 January 2020. We decided to scan the reference lists of the reviews retrieved in order to find potential additional pertinent articles, and to test string sensitivity. The search string used was: coronavirus* OR Pneumonia of Unknown Etiology OR COVID-19 OR nCoV. The virus name was updated from \"2019-nCoV\" to \"SARS-CoV-2\" by the International Committee on Taxonomy of Viruses on 11 February 2020, [16] yet we performed the search using the term \"nCoV\" since we presumed that no paper published between 11 February 2020 and 13 February 2020 would have used the term \"SARS-COV-2\". To achieve the highest sensitivity, we decided to use only a combination of keywords avoiding Mesh terms. Asterisks are used to truncate words, so that every ending after the asterisks was searched. We placed a language restriction for English, without other limits to the search.\\n\\nThe papers were than classified on the basis of a standardized list of key information including: Digital Object Identifier number (DOI), title, journal name, day of first publication (online or paper form), country of the first author affiliation, article type, source of the data and topics addressed. The topics addressed were identified according to 8 categories: generic discussion, epidemiology, virology, pathology and clinical presentation, diagnosis, therapy, transmission, preparedness. Articles that addressed more than a topic in an extensive way were accounted for more than one of the 8 topic categories; the papers that could not be classified in any of the 8 chosen topic categories were classified in the \"Other\" topic category. Eight of us (FS, EA, EB, EB, FE, FM, FS, SS) independently screened all identified articles. Procedure was carried out through title and abstract scanning in order to verify the inclusion criteria. Any disagreements were resolved through discussion and consensus between the reviewers. If disagreement persisted, another reviewer (DG) was called as tie-breaker.\\n\\nThe \"Number Needed to Read\" (NNR) is defined as the ratio of the number of retrieved abstracts to the number of the ones which are pertinent for the research purpose. [17] We calculated the NNR to identify the proportion of papers which were not editorials and commentaries, as well as the proportion of papers with directly collected data. Chi2 test was used in order to test for significant differences in the comparison of percentages, using the statistical software STATA 15 (StataCorp. 2017. Stata Statistical Software: Release 15. College Station, TX: StataCorp LLC).\\n\\nA total of 442 papers were retrieved; 234 articles were found to be pertinent (see Figure 1 ) and read in extenso in order to define their main characteristics. The affiliation of the first authors was distributed among 36 countries: 36.3% of the papers (85 out of 234) had a first author with a Chinese affiliation, 18.4% and 13.7% (43 out of 234 and 32 out of 234) had a first author with an affiliation from UK and USA respectively, while first authors affiliation from Canada accounted for 3.8% of the total (9 out of 234), from Italy for 3% (7 out of 234), and from Australia and Korea both accounted for 2.1% (5 out of 234 each). All the other countries accounted for less than 2% of the first author affiliations (less than 5 out of 234). 63.7% of the papers (149 out of 234) were editorials, commentaries or letters (mainly reported data). 10.7% of the papers (25 out of 234) were secondary papers, mainly narrative reviews, which collected the knowledge available up to that point on some specific topics (i.e. genomics of the virus, transmissibility, etc.). The remaining 25.6% (60 out of 234) were original primary studies: among these, case reports accounted for 43.3% of the total, while in vitro or in vivo studies or genomic studies accounted for 21.7% of the total. The remaining primary studies were cohort studies, case control studies and surveys. Chinese first authors published more original data and primary studies than authors from the UK and the USA (32.9% vs 4.7% of the UK and 6.3% of the USA; p&lt;0.001), who published mostly editorials and commentaries. Of note, 61.5% of the analysed papers used reported/non original data and 15.8% used official data. Conversely, only 17.5% of the papers used data which were directly collected on the field. In 5.2% of the papers the source of the data was not clearly specified. The supplementary materials (Table 1) , available online, report all the findings of this review with the reference and description of the 234 papers selected for the bibliometric analysis.\\n\\nAll rights reserved. No reuse allowed without permission.\\n\\nauthor/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03. 25.20043315 doi: medRxiv preprint Topics and content of the papers published \"Generic discussion\" papers were the most frequent ones (29.5% of the total), and consisted mostly of editorials and commentaries, with no original data. The second most frequent topic was \"Preparedness\" (23.1%): 83.3% of these papers consisted of commentaries, and only 1 paper was a primary study. The \"Epidemiology\" topic was addressed in 15.4% of the papers, and only 2 (5.6%) were primary studies. The lack of primary studies could be observed also for the \"Virology\" topic (14.1%) and the \"Transmission\" topic (12.8%), with only 12 (36.4%) and 6 (20%) papers reporting original data, respectively. On the other hand, the \"Pathology and clinical characteristics\" topic (12.8%) consisted mostly of primary studies (22, 73.3%). \"Therapy\" topic was addressed in 12% of the papers, with 9 papers (33.3%) being primary studies. The least addressed topic was \"Diagnosis\" (9.4%), with 12 papers (54.5%) reporting original data. Finally, 3% of the papers were classified in the \"Other\" topic category. Table 1 , ID: 218] assessed the required expertise and capacity for molecular detection in specialised laboratories into the European Union/European Economic Area countries. This paper emphasized the need for countries to put in place strong measures in order to detect and laboratory-confirm cases early. It was also highlighted the need to perform molecular testing (RT-PCR, as also indicated by the WHO [19] ) on different specimens including: nasopharyngeal swabs, bronchoalveolar lavage, oropharyngeal swab, nasopharyngeal aspirate, sputum, (endo) tracheal aspirate and nasal wash. The communication report also underlined the need for a clinical validation of the test specificity and sensitivity. If nasopharyngeal and oropharyngeal swabs are the recommended types of specimen for diagnostic testing an interesting brief report published by Chinese authors [20] [Supplementary materials 165, 167, 197, 222] were published in the second half of the time frame selected, mostly by authors with a Chinese affiliation. These papers addressed several aspects of COVID-19 treatment, such as interferon inhalation, use of and efficacy of antiviral drugs, potential repurposing treatments with angiotensin receptor blockers, use of antibiotics for bacterial co-infections, and respiratory support therapy (e.g. oxygen saturation, CPAP, invasive mechanical ventilation). Interestingly, one paper published on 4 February 2020 [21] [Supplementary materials, Table 1 , ID: 91] investigated the possibility that the receptor that SARS-CoV-2 uses to infect lung cells might be ACE2, a cell-surface protein on lung AT2 alveolar epithelial cells; one of the known regulators of endocytosis for the AT2 cells is the AP2associated protein kinase 1 (AAK1). The authors suggested the possibility to use high-affinity AAK1-binding drugs to inhibit endocytosis of AT2 cells, such as baricitinib (an oral, targeted All rights reserved. No reuse allowed without permission. author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03. 25.20043315 doi: medRxiv preprint synthetic disease-modifying antirheumatic drug used for the treatment of rheumatoid arthritis), in order to reduce both the viral entry and the inflammation in patients.\\n\\nFrom the second half of January to the first half of February the proportion of editorials and commentaries changed the NNR from 3.6 to 2.6, meaning that we passed from a mean 3.6 papers to read in order to find an non-editorial/commentary to 2.6 papers. The mean chance to read papers with directly collected data increased, with the NNR passing from 6.5 in January to 5.4 in February.\\n\\nThe novel coronavirus infection that emerged two months ago (as already discussed, the new coronavirus was isolated on 7 January 2020) places emphasis on the importance of ensuring that health professionals, researchers and the public have the best possible scientific information.\\n\\nBibliometrics is a scientific method widely used in many fields for quantifying and analysing published information. [22] This kind of literature analysis in the first month after the first publication in Pubmed shows a great number of papers published as expected, mainly by authors with a Chinese affiliation, followed by authors with an UK or USA affiliation. The reason for this distribution is ascribed to the fact that China is the country where the virus was originally isolated and the first one with a COVID-19 outbreak. Moreover, we have to point out that, as shown in a commentary published in Nature, in recent years China produced more science PhD than any other country in the world and the Chinese government has been assuming a leading role in supporting its own scientific research.\\n\\n[23] Therefore the research and publication response to the outbreak has been prompt and lively, giving to the scientific community many elements for understanding the new infection. The second country with the highest number of publications was the UK, followed by the USA, probably due to the fact they are the most prolific countries for scientific research. Authors from the UK and the USA wrote mostly editorials and commentaries.\\n\\nThe publication policies of many of the most important scientific journals were adapted to quickly review and publish scientific papers on SARS-CoV-2 and COVID-19. This strategy facilitated the spread of new knowledge on the new infection and on the outbreak. Hence, it is important to have scientific shared papers in order to reduce misinformation which is itself a public health threat. [24] In the early scientific literature we analysed there is a lack of primary studies with original data, as expected, given the very short period of time considered after the first pertinent publication on the new infection. Nevertheless, editorials, news and commentaries are important to share opinions and the viewpoint of an expert or a panel of experts, rather than on producing new scientific evidence.\\n\\nthe variety and severity of the presentation of COVID-19. Further publication and reviews on this topic are needed to summarize the knowledge arising from all the case reports. Diagnostic tools, except for CT scan, were poorly represented and discussed, especially in terms of molecular diagnosis and standardization of the techniques. The \"Therapy\" topic was also poorly represented, since more time is needed for clinical intervention studies.\\n\\nAnalyzing the selected time frame, the percentage of the editorials, news and commentaries has decreased. This was highlighted particularly by our analysis, which shows that from the second half of January to the first half of February the proportion of editorials and commentaries lowered the NNR, thus indicating an increase in the level of evidence produced as well as the increase of primary data availability.\\n\\nThere are several limitations in our study. First, we based our research only on PubMed, that does not index all of the scientific impacted journals. It is strongly recommended to consult more than one relevant database when performing systematic reviews of the literature, [25] even though it has been reported that most of the high-quality articles, like those included in Cochrane Reviews, are indexed in PubMed. [26] In addition, we excluded all the publications not in English. We acknowledge that, given the outbreak started in China, this may have led to a partial selection bias. Nevertheless, papers excluded which were written in Chinese and had an English abstract, were for the greatest part reports of the Chinese Government (data not shown). We considered only the articles published until 13 February 2020, and taking into account that literature is now changing day-by-day for this topic, extending the bibliometric research for the next months would be of paramount importance.\\n\\nIn conclusion, as far as we know, our very early review is the first bibliometric study analyzing the early scientific research output about COVID-19 from a quantitative and descriptive standpoint. Due to the huge amount of interest and concern related to this outbreak, our analysis shows that the scientific publication has been very reactive but still preliminar, with an expected deficit of original data and an excess of editorials and commentaries. Further research is needed to review and synthesize the growing knowledge on SARS-CoV-2 and COVID-19, with a special aim on the most important knowledge gaps in terms of diagnosis and effective preventive and therapeutic measures.\\n\\nAll authors declare no support from any organisation for the submitted work; no financial relationships with any organisations that might have an interest in the submitted work in the previous three years; no other relationships or activities that could appear to have influenced the submitted work.\"\\n\\nThe author(s) received no financial support for the research, authorship, and/or publication of this article.\\n\\nThe authors declare that all the data relevant to the study are included in the article or uploaded as supplementary information in Supplementary author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03. 25.20043315 doi: medRxiv preprint   -R  e  g  a  r  d  i  n  g  \"  J  o  u  r  n  a  l  P  u  b  l  i  c  a  t  i  o  n  D  a  t  e  \"  ,  s  o  m  e  d  a  t  e  s  m  a  y  b  e  c  o  n  s  e  c  u  t  i  v  e  t  o  1  3  F  e  b  r  u  a  r  y   2  0  2  0  ;  t  h  i  s  i  s  d  u  e  t  o  t  h  e  f  a  c  t  t  h  a  t  s  o  m  e  p  a  p  e  r  w  e  r  e  a  l  r  e  a  d  y  i  n  d  e  x  e  d  i  n  M  E  D  L  I  N  E  d  a  t  a  b  a  s  e   (  P  u  b  M  e  d  )  a  t  t  h  e  m  o  m  e  n  t  w  e  p  e  r  f  o  r  m  e  d  o  u  r  s  e  a  r  c author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03. 25.20043315 doi: medRxiv preprint \\n\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                          title  \\\n",
       "12  TITLE: The early scientific literature response to the novel Coronavirus outbreak: who published what? CORRESPONDING AUTHOR   \n",
       "\n",
       "                                                                                                                                                                                                                     abstract  \\\n",
       "12  [{'text': 'All rights reserved. No reuse allowed without permission. author/funder, who has granted medRxiv a license to display the preprint in perpetuity.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         full_text  \n",
       "12  Recent events highlight how emerging and re-emerging pathogens are actually becoming global challenges for public health. [1] Coronaviruses are enveloped RNA viruses which are broadly distributed either in humans or in a vast majority of other mammals and birds. These viruses have the possibility to cause respiratory, enteric, hepatic, and neurologic diseases. [2, 3] Coronaviruses are highly prevalent in many species. They have a large genetic diversity. Given their RNA, they are susceptible to frequent recombination and mutations of their genomes. In context in which there are increasing human-animal interactions, novel coronaviruses are very likely to emerge periodically. If the newly created cross-species pathogens acquires the ability to infect humans or to be transmitted human to human it can lead to occasional spillover events and epidemics. [4, 5, 6] In the past years two other strains of Coronavirus -severe acute respiratory syndrome coronavirus (SARS-CoV) and Middle East respiratory syndrome coronavirus (MERS-CoV) -have emerged as potential public health worldwide threats. [5] SARS-CoV was the causal agent of the severe acute respiratory syndrome outbreaks in 2002 and 2003 in Guangdong Province, China; [7, 8, 9] MERS-CoV was the pathogen responsible for severe respiratory disease outbreaks in 2012 in the Middle East. [10] On 31 December 2019, the World Health Organization (WHO) China Country Office was informed of cases of pneumonia of unknown etiology detected in Wuhan City, located in the Hubei Province, China, associated with exposures in a seafood and wet wholesale market in the same city. A new type of coronavirus was isolated on 7 January 2020. [11] On 30 January 2020 WHO declared the outbreak to be a Public Health Emergency of International Concern (PHEIC). [12] This novel coronavirus suddenly turned out to be a global health concern for a disease, called Coronavirus disease 2019 (COVID-19), [13] which was characterized as a pandemic by WHO on 11 March 2020. [14] Starting from the second-half of January 2020, scientific literature has been particularly focused on the description of this new viral outbreak; the main topics addressed were the epidemiological, clinical and virological aspects as well as the possible public health choices necessary to contain the spread of the disease. Nevertheless, several aspects are still unclear and have not been thoroughly explored, leaving grey areas in our knowledge of the disease and of the outbreak. The aim of this paper is to perform a bibliometric analysis on the first papers published in the early stages of the SARS-CoV-2 outbreak, in order to give a glimpse to the researchers of \"who published what\" at the very beginning of this Public Health Emergency of International Concern.\\n\\nWe searched MEDLINE (PubMed) electronic database in order to find all relevant studies. After a preliminary search starting from 1 December 2019, we found out that the first article meeting our inclusion criteria was published on 14 January 2020. [15] [Supplementary materials, Table 1 , ID: 1] We subsequently extended our research to the first 30 days after this milestone article, up to 13 January 2020. We decided to scan the reference lists of the reviews retrieved in order to find potential additional pertinent articles, and to test string sensitivity. The search string used was: coronavirus* OR Pneumonia of Unknown Etiology OR COVID-19 OR nCoV. The virus name was updated from \"2019-nCoV\" to \"SARS-CoV-2\" by the International Committee on Taxonomy of Viruses on 11 February 2020, [16] yet we performed the search using the term \"nCoV\" since we presumed that no paper published between 11 February 2020 and 13 February 2020 would have used the term \"SARS-COV-2\". To achieve the highest sensitivity, we decided to use only a combination of keywords avoiding Mesh terms. Asterisks are used to truncate words, so that every ending after the asterisks was searched. We placed a language restriction for English, without other limits to the search.\\n\\nThe papers were than classified on the basis of a standardized list of key information including: Digital Object Identifier number (DOI), title, journal name, day of first publication (online or paper form), country of the first author affiliation, article type, source of the data and topics addressed. The topics addressed were identified according to 8 categories: generic discussion, epidemiology, virology, pathology and clinical presentation, diagnosis, therapy, transmission, preparedness. Articles that addressed more than a topic in an extensive way were accounted for more than one of the 8 topic categories; the papers that could not be classified in any of the 8 chosen topic categories were classified in the \"Other\" topic category. Eight of us (FS, EA, EB, EB, FE, FM, FS, SS) independently screened all identified articles. Procedure was carried out through title and abstract scanning in order to verify the inclusion criteria. Any disagreements were resolved through discussion and consensus between the reviewers. If disagreement persisted, another reviewer (DG) was called as tie-breaker.\\n\\nThe \"Number Needed to Read\" (NNR) is defined as the ratio of the number of retrieved abstracts to the number of the ones which are pertinent for the research purpose. [17] We calculated the NNR to identify the proportion of papers which were not editorials and commentaries, as well as the proportion of papers with directly collected data. Chi2 test was used in order to test for significant differences in the comparison of percentages, using the statistical software STATA 15 (StataCorp. 2017. Stata Statistical Software: Release 15. College Station, TX: StataCorp LLC).\\n\\nA total of 442 papers were retrieved; 234 articles were found to be pertinent (see Figure 1 ) and read in extenso in order to define their main characteristics. The affiliation of the first authors was distributed among 36 countries: 36.3% of the papers (85 out of 234) had a first author with a Chinese affiliation, 18.4% and 13.7% (43 out of 234 and 32 out of 234) had a first author with an affiliation from UK and USA respectively, while first authors affiliation from Canada accounted for 3.8% of the total (9 out of 234), from Italy for 3% (7 out of 234), and from Australia and Korea both accounted for 2.1% (5 out of 234 each). All the other countries accounted for less than 2% of the first author affiliations (less than 5 out of 234). 63.7% of the papers (149 out of 234) were editorials, commentaries or letters (mainly reported data). 10.7% of the papers (25 out of 234) were secondary papers, mainly narrative reviews, which collected the knowledge available up to that point on some specific topics (i.e. genomics of the virus, transmissibility, etc.). The remaining 25.6% (60 out of 234) were original primary studies: among these, case reports accounted for 43.3% of the total, while in vitro or in vivo studies or genomic studies accounted for 21.7% of the total. The remaining primary studies were cohort studies, case control studies and surveys. Chinese first authors published more original data and primary studies than authors from the UK and the USA (32.9% vs 4.7% of the UK and 6.3% of the USA; p<0.001), who published mostly editorials and commentaries. Of note, 61.5% of the analysed papers used reported/non original data and 15.8% used official data. Conversely, only 17.5% of the papers used data which were directly collected on the field. In 5.2% of the papers the source of the data was not clearly specified. The supplementary materials (Table 1) , available online, report all the findings of this review with the reference and description of the 234 papers selected for the bibliometric analysis.\\n\\nAll rights reserved. No reuse allowed without permission.\\n\\nauthor/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03. 25.20043315 doi: medRxiv preprint Topics and content of the papers published \"Generic discussion\" papers were the most frequent ones (29.5% of the total), and consisted mostly of editorials and commentaries, with no original data. The second most frequent topic was \"Preparedness\" (23.1%): 83.3% of these papers consisted of commentaries, and only 1 paper was a primary study. The \"Epidemiology\" topic was addressed in 15.4% of the papers, and only 2 (5.6%) were primary studies. The lack of primary studies could be observed also for the \"Virology\" topic (14.1%) and the \"Transmission\" topic (12.8%), with only 12 (36.4%) and 6 (20%) papers reporting original data, respectively. On the other hand, the \"Pathology and clinical characteristics\" topic (12.8%) consisted mostly of primary studies (22, 73.3%). \"Therapy\" topic was addressed in 12% of the papers, with 9 papers (33.3%) being primary studies. The least addressed topic was \"Diagnosis\" (9.4%), with 12 papers (54.5%) reporting original data. Finally, 3% of the papers were classified in the \"Other\" topic category. Table 1 , ID: 218] assessed the required expertise and capacity for molecular detection in specialised laboratories into the European Union/European Economic Area countries. This paper emphasized the need for countries to put in place strong measures in order to detect and laboratory-confirm cases early. It was also highlighted the need to perform molecular testing (RT-PCR, as also indicated by the WHO [19] ) on different specimens including: nasopharyngeal swabs, bronchoalveolar lavage, oropharyngeal swab, nasopharyngeal aspirate, sputum, (endo) tracheal aspirate and nasal wash. The communication report also underlined the need for a clinical validation of the test specificity and sensitivity. If nasopharyngeal and oropharyngeal swabs are the recommended types of specimen for diagnostic testing an interesting brief report published by Chinese authors [20] [Supplementary materials 165, 167, 197, 222] were published in the second half of the time frame selected, mostly by authors with a Chinese affiliation. These papers addressed several aspects of COVID-19 treatment, such as interferon inhalation, use of and efficacy of antiviral drugs, potential repurposing treatments with angiotensin receptor blockers, use of antibiotics for bacterial co-infections, and respiratory support therapy (e.g. oxygen saturation, CPAP, invasive mechanical ventilation). Interestingly, one paper published on 4 February 2020 [21] [Supplementary materials, Table 1 , ID: 91] investigated the possibility that the receptor that SARS-CoV-2 uses to infect lung cells might be ACE2, a cell-surface protein on lung AT2 alveolar epithelial cells; one of the known regulators of endocytosis for the AT2 cells is the AP2associated protein kinase 1 (AAK1). The authors suggested the possibility to use high-affinity AAK1-binding drugs to inhibit endocytosis of AT2 cells, such as baricitinib (an oral, targeted All rights reserved. No reuse allowed without permission. author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03. 25.20043315 doi: medRxiv preprint synthetic disease-modifying antirheumatic drug used for the treatment of rheumatoid arthritis), in order to reduce both the viral entry and the inflammation in patients.\\n\\nFrom the second half of January to the first half of February the proportion of editorials and commentaries changed the NNR from 3.6 to 2.6, meaning that we passed from a mean 3.6 papers to read in order to find an non-editorial/commentary to 2.6 papers. The mean chance to read papers with directly collected data increased, with the NNR passing from 6.5 in January to 5.4 in February.\\n\\nThe novel coronavirus infection that emerged two months ago (as already discussed, the new coronavirus was isolated on 7 January 2020) places emphasis on the importance of ensuring that health professionals, researchers and the public have the best possible scientific information.\\n\\nBibliometrics is a scientific method widely used in many fields for quantifying and analysing published information. [22] This kind of literature analysis in the first month after the first publication in Pubmed shows a great number of papers published as expected, mainly by authors with a Chinese affiliation, followed by authors with an UK or USA affiliation. The reason for this distribution is ascribed to the fact that China is the country where the virus was originally isolated and the first one with a COVID-19 outbreak. Moreover, we have to point out that, as shown in a commentary published in Nature, in recent years China produced more science PhD than any other country in the world and the Chinese government has been assuming a leading role in supporting its own scientific research.\\n\\n[23] Therefore the research and publication response to the outbreak has been prompt and lively, giving to the scientific community many elements for understanding the new infection. The second country with the highest number of publications was the UK, followed by the USA, probably due to the fact they are the most prolific countries for scientific research. Authors from the UK and the USA wrote mostly editorials and commentaries.\\n\\nThe publication policies of many of the most important scientific journals were adapted to quickly review and publish scientific papers on SARS-CoV-2 and COVID-19. This strategy facilitated the spread of new knowledge on the new infection and on the outbreak. Hence, it is important to have scientific shared papers in order to reduce misinformation which is itself a public health threat. [24] In the early scientific literature we analysed there is a lack of primary studies with original data, as expected, given the very short period of time considered after the first pertinent publication on the new infection. Nevertheless, editorials, news and commentaries are important to share opinions and the viewpoint of an expert or a panel of experts, rather than on producing new scientific evidence.\\n\\nthe variety and severity of the presentation of COVID-19. Further publication and reviews on this topic are needed to summarize the knowledge arising from all the case reports. Diagnostic tools, except for CT scan, were poorly represented and discussed, especially in terms of molecular diagnosis and standardization of the techniques. The \"Therapy\" topic was also poorly represented, since more time is needed for clinical intervention studies.\\n\\nAnalyzing the selected time frame, the percentage of the editorials, news and commentaries has decreased. This was highlighted particularly by our analysis, which shows that from the second half of January to the first half of February the proportion of editorials and commentaries lowered the NNR, thus indicating an increase in the level of evidence produced as well as the increase of primary data availability.\\n\\nThere are several limitations in our study. First, we based our research only on PubMed, that does not index all of the scientific impacted journals. It is strongly recommended to consult more than one relevant database when performing systematic reviews of the literature, [25] even though it has been reported that most of the high-quality articles, like those included in Cochrane Reviews, are indexed in PubMed. [26] In addition, we excluded all the publications not in English. We acknowledge that, given the outbreak started in China, this may have led to a partial selection bias. Nevertheless, papers excluded which were written in Chinese and had an English abstract, were for the greatest part reports of the Chinese Government (data not shown). We considered only the articles published until 13 February 2020, and taking into account that literature is now changing day-by-day for this topic, extending the bibliometric research for the next months would be of paramount importance.\\n\\nIn conclusion, as far as we know, our very early review is the first bibliometric study analyzing the early scientific research output about COVID-19 from a quantitative and descriptive standpoint. Due to the huge amount of interest and concern related to this outbreak, our analysis shows that the scientific publication has been very reactive but still preliminar, with an expected deficit of original data and an excess of editorials and commentaries. Further research is needed to review and synthesize the growing knowledge on SARS-CoV-2 and COVID-19, with a special aim on the most important knowledge gaps in terms of diagnosis and effective preventive and therapeutic measures.\\n\\nAll authors declare no support from any organisation for the submitted work; no financial relationships with any organisations that might have an interest in the submitted work in the previous three years; no other relationships or activities that could appear to have influenced the submitted work.\"\\n\\nThe author(s) received no financial support for the research, authorship, and/or publication of this article.\\n\\nThe authors declare that all the data relevant to the study are included in the article or uploaded as supplementary information in Supplementary author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03. 25.20043315 doi: medRxiv preprint   -R  e  g  a  r  d  i  n  g  \"  J  o  u  r  n  a  l  P  u  b  l  i  c  a  t  i  o  n  D  a  t  e  \"  ,  s  o  m  e  d  a  t  e  s  m  a  y  b  e  c  o  n  s  e  c  u  t  i  v  e  t  o  1  3  F  e  b  r  u  a  r  y   2  0  2  0  ;  t  h  i  s  i  s  d  u  e  t  o  t  h  e  f  a  c  t  t  h  a  t  s  o  m  e  p  a  p  e  r  w  e  r  e  a  l  r  e  a  d  y  i  n  d  e  x  e  d  i  n  M  E  D  L  I  N  E  d  a  t  a  b  a  s  e   (  P  u  b  M  e  d  )  a  t  t  h  e  m  o  m  e  n  t  w  e  p  e  r  f  o  r  m  e  d  o  u  r  s  e  a  r  c author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03. 25.20043315 doi: medRxiv preprint \\n\\n  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>A spatial model of CoVID-19 transmission in England and Wales: early spread and peak timing</td>\n",
       "      <td>[{'text': 'An outbreak of a novel coronavirus, named CoVID-19, was first reported in China on 31 December 2019. As of 9 February 2020, cases have been reported in 25 countries, including probable cases of human-to-human transmission in England.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'We adapted an existing national-scale metapopulation model to capture the spread of CoVID-19 in England and Wales. We used 2011 census data to capture population sizes and population movement, together with parameter estimates from the current outbreak in China.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'We predict that a CoVID-19 outbreak will peak 126 to 147 days (~4 months) after the start of person-to-person transmission in England and Wales in the absence of controls, assuming biological parameters remain unchanged. Therefore, if person-to-person transmission persists from February, we predict the epidemic peak would occur in June. The starting location has minimal impact on peak timing, and model stochasticity varies peak timing by 10 days. Incorporating realistic parameter uncertainty leads to estimates of peak time ranging from 78 days to 241 days after person-to-person transmission has been established. Seasonal changes in transmission rate substantially impact the timing and size of the epidemic peak, as well as the total attack rate.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'We provide initial estimates of the potential course of CoVID-19 in England and Wales in the absence of control measures. These results can be refined with improved estimates of epidemiological parameters, and permit investigation of control measures and cost effectiveness analyses. Seasonal changes in transmission rate could shift the timing of the peak into winter months, which will have important implications for health-care capacity planning.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>An outbreak of a novel coronavirus, recently renamed CoVID-19, was first reported from Wuhan, China on 31 December 2019. During January 2020, the outbreak spread to multiple cities in China, and the first cases started appearing outside China. By the end of January 2020, 9,720 cases had been confirmed in China, with 106 confirmed cases outside China across 19 different countries (1) .\\n\\nEpidemiological analysis of the outbreak was quickly used to start estimating epidemiologicallyrelevant parameters, such as the basic reproduction number, the serial interval, the incubation period and the case fatality rate (2) (3) (4) (5) (6) (7) . Initial estimates suggested that the reproduction number was between 2 and 3 and the case fatality rate was less than 4% (8) . Control of spread by contact tracing and isolation appears to be challenging, given what is currently known about the virus (9) .\\n\\nMathematical models are useful tools for understanding and predicting the possible course of an outbreak, given a set of underlying assumptions. Here, we adapt a metapopulation model of disease transmission in England and Wales to capture the spread of CoVID-19 (10) . The aim is to provide predictions about the likely timing of the peak of the epidemic in England and Wales and spatial features of spread.\\n\\nWe use an existing national-scale stochastic metapopulation model of disease transmission in England and Wales. The model structure is based on the metapopulation model described in detail in Danon et al (2009) (10) . In this model, the population is divided into electoral wards. Because of the changes in data availability, we restricted the model to England and Wales, whereas the original model covered Great Britain.\\n\\nTransmission between wards occurs via the daily movement of individuals. For each ward, we assume that individuals contribute to the force of infection in their \"home\" ward during the night and their \"work\" ward during the day. See Danon et al 2009 for further details (10) .\\n\\nData for population and movement of individuals come from the 2011 census of the United Kingdom. The population size of each of the 8,570 electoral wards is available directly from the Office of National Statistics (ONS) website. The number of individuals moving between locations is also available from the ONS website, but at the level of census output areas (OAs). We aggregated the data from OA level to electoral wards level. Spatial location of electoral ward centres are extracted from maps available from the ONS websites.\\n\\nWe use a Susceptible-Exposed-Infectious-Infectious-Recovered (SEIIR) model within each ward to capture the progression of disease within an individual (figure 1). Initial analyses used SARS-like parameters for the incubation period and infectious period, which now appear to differ from CoVID-19(4). Li et al (2020) analysed data on 425 cases reported in Wuhan in China and fitted a log-normal distribution to the incubation period, and a gamma distribution for the serial interval (2) . The infectious period for SARS was estimated as the serial interval minus the . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint incubation period, but as Li et al did not report the correlation between incubation period and serial interval, we were not able to estimate the infectious period distribution from the data, but used a uniform distribution between 2 and 3 days, to give a mean serial interval of approximately 7-8, in line with current estimates. We used two infectious states to represent a mildly symptomatic or prodromal period and a period with more pronounced symptoms. In the absence of data on the relative magnitude of these two infections states, we assumed the same length of time in each infectious state and assumed that each state was equally infectious. We sampled from each of the distributions 100 times independently (Table 1 ). \\n\\nThe census data are used to initialise the population sizes within each of the 8,570 wards. At the start of the model, all individuals are assumed to be susceptible to infection with no 3 e . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint underlying immunity in the population. We investigated a range of starting scenarios by seeding the infection in example wards London, Birmingham, Brighton, Sheffield and Cardiff. To seed infection in a ward, we move five individuals (non-commuters) from the susceptible compartment to the first infectious state.\\n\\nWe investigated the impact of a seasonally affected transmission rate, to capture potential decreased transmission during the summer months. We captured seasonal transmission by replacing the constant transmission rate with a time-varying transmission rate given by:\\n\\nis the magnitude of the seasonal difference in transmission, ranging from\\n\\n(maximum seasonality with no transmission at the peak of the summer).\\n\\nFrom the model, we extracted the total number of infections per day, as the number of individuals in both of the Infectious states, and the number of infected wards per day as the total number of wards with at least one individual in one of the two Infectious states. The spatial growth of the epidemic in England and Wales was visualised using interactive maps. We estimated the timing of the epidemic peak from the aggregated epidemic curve and calculated 95% prediction intervals from the model simulations.\\n\\nThe model is coded in C and is available on github (http://github.com/ldanon/MetaWards). The data are freely available from the Office for National Statistics website, or can be downloaded with the code at the github repository.\\n\\nWe predict that, in the absence of any interventions, a disease with \"best-guess\" CoVID-19-like parameters will peak a median of 133 days (range 126 -147 days) following the start of personto-person transmission in England and Wales. Intrinsic model stochasticity is responsible for variation between model runs. Using exactly the same parameters and seeding the infection in the same initial wards resulted in a difference in peak timing of +/-10 days (figure 2). The attack rate for best-guess parameters had a median of 45799874 (81.67% range 81.64-81.69), with a peak incidence median 1,116,692.\\n\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint\\n\\nThe initial location of cases had some, but limited impact on the timing of the epidemic in England and Wales. Epidemics seeded in Brighton, London, Birmingham and Sheffield resulted in synchronised epidemics in England, reaching urban areas first followed by rural areas. Epidemics started in Cardiff had a slower time to peak but still resulted in a generalised outbreak.\\n\\nSpatially, some disaggregation between England and Wales regions is observed. An outbreak starting in Brighton, (South East England) peaks in London and the South East first, and North East England, Yorkshire and Humber and Wales last, with a ten-day lag between regions ( Figure 3 ).\\n\\nModel predictions are highly sensitive to parameter values and incorporating parameter uncertainty increases the variability of model predictions. In the absence of any control measures, all predictions resulted in epidemics that peaked within a year from the start of person-to-person transmission in England and Wales. Estimates of peak time ranged from 78 days to 241 days (Figure 4 ).\\n\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint However, seasonality in transmission has a large impact on epidemic timing, peak incidence, and final attack rates. Assuming no difference in transmission rate during the year leads to a single large epidemic after approximately 4 months (June time if transmission starts in February), as above. With a 25% reduction in transmission the epidemic is smaller and peaks later, reducing the overall attack rate by 20%. A 50% reduction in transmission results in a smaller epidemic before the summer, followed by a resurgence in cases in the following winter. The attack rate is 10% less than a non-seasonal epidemic. A 75% reduction in transmission over the summer resulted in a delayed large outbreak, but with a similar attack rate. If transmission decreases to zero over the summer then the resulting outbreak is much reduced, with an attack rate of less than 1% ( Figure 5 , Table 2 ). . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint Table 2 : Effect of seasonal variation on the timing (shown in days following initial seeding), the height of the peak, and the final attack rate.\\n\\nWe predict that, in the absence of control measures and with no seasonality in transmission, the introduction of CoVID-19 in England and Wales has the potential to result in a synchronised outbreak that peaks at around 4 months following the start of person-to-person transmission. Our findings suggest that the height of the epidemic and the attack rate is highly dependent on seasonality of transmission and that even small changes in transmission risk can lead to large changes in attack rate due to the spatial disaggregation of the population at risk.\\n\\nA combination of control measures and seasonal changes in transmission rate could shift the peak of the outbreak to the winter of 2020/21, with little effect on the final attack rate. If contact tracing and isolation efforts succeed in reducing transmission, but are unable to control the epidemic (9), an additional influx of severe CoVID-19 cases may exacerbate existing challenges with winter healthcare demand. A careful analysis of the impact of control measures on the timing of incidence of severe cases is warranted.\\n\\nThe strength of this model lies in the spatial heterogeneity which tempers transmission. As a comparison, an equivalent non-spatial model results in the epidemic peaking after 34 days, nearly four times faster than this spatial model, and would be unable to capture the interaction between spatial transmission and seasonality. The estimated total number of people infected in the spatial model is marginally smaller than for a non-spatial model, as the infection has the opportunity to die-out in local parts of the country. As the model framework was developed and published in 2009, it was possible to re-deploy the model for these new circumstances; developing such a model from scratch during an outbreak would be a significant challenge.\\n\\nA key element missing from our model is morbidity, mortality and the treatment of cases. The model in its current form predicts the total number of infections in the community rather than diagnosed cases. Observations from China suggest that many cases have mild symptoms and that only around 5% of cases have been reported and diagnosed (3) . The parameter estimates we used from China appear to be substantially different to previous coronaviruses (6) . Should CoVID-19 continue spreading the UK it will become possible to get UK-specific parameter estimates and improve prediction accuracy.\\n\\nAs with all modelling, it is impossible to capture the full complexity of an epidemic. In this model, the major assumptions are that we have assumed that there is no change in behaviour during the course of the epidemic. In practice, as the epidemic starts spreading in England and Wales there may well be a systematic change in behaviour as was seen during the H1N1 influenza pandemic in 2009. We have not included any age-effects, such as differential mixing, susceptibility or infectiousness. That means that we are not able to investigate the impact of school closures or the impact of the summer holidays, which had a large impact on the H1N1 influenza pandemic in 2009.\\n\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                          title  \\\n",
       "31  A spatial model of CoVID-19 transmission in England and Wales: early spread and peak timing   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               abstract  \\\n",
       "31  [{'text': 'An outbreak of a novel coronavirus, named CoVID-19, was first reported in China on 31 December 2019. As of 9 February 2020, cases have been reported in 25 countries, including probable cases of human-to-human transmission in England.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'We adapted an existing national-scale metapopulation model to capture the spread of CoVID-19 in England and Wales. We used 2011 census data to capture population sizes and population movement, together with parameter estimates from the current outbreak in China.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'We predict that a CoVID-19 outbreak will peak 126 to 147 days (~4 months) after the start of person-to-person transmission in England and Wales in the absence of controls, assuming biological parameters remain unchanged. Therefore, if person-to-person transmission persists from February, we predict the epidemic peak would occur in June. The starting location has minimal impact on peak timing, and model stochasticity varies peak timing by 10 days. Incorporating realistic parameter uncertainty leads to estimates of peak time ranging from 78 days to 241 days after person-to-person transmission has been established. Seasonal changes in transmission rate substantially impact the timing and size of the epidemic peak, as well as the total attack rate.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'We provide initial estimates of the potential course of CoVID-19 in England and Wales in the absence of control measures. These results can be refined with improved estimates of epidemiological parameters, and permit investigation of control measures and cost effectiveness analyses. Seasonal changes in transmission rate could shift the timing of the peak into winter months, which will have important implications for health-care capacity planning.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 full_text  \n",
       "31  An outbreak of a novel coronavirus, recently renamed CoVID-19, was first reported from Wuhan, China on 31 December 2019. During January 2020, the outbreak spread to multiple cities in China, and the first cases started appearing outside China. By the end of January 2020, 9,720 cases had been confirmed in China, with 106 confirmed cases outside China across 19 different countries (1) .\\n\\nEpidemiological analysis of the outbreak was quickly used to start estimating epidemiologicallyrelevant parameters, such as the basic reproduction number, the serial interval, the incubation period and the case fatality rate (2) (3) (4) (5) (6) (7) . Initial estimates suggested that the reproduction number was between 2 and 3 and the case fatality rate was less than 4% (8) . Control of spread by contact tracing and isolation appears to be challenging, given what is currently known about the virus (9) .\\n\\nMathematical models are useful tools for understanding and predicting the possible course of an outbreak, given a set of underlying assumptions. Here, we adapt a metapopulation model of disease transmission in England and Wales to capture the spread of CoVID-19 (10) . The aim is to provide predictions about the likely timing of the peak of the epidemic in England and Wales and spatial features of spread.\\n\\nWe use an existing national-scale stochastic metapopulation model of disease transmission in England and Wales. The model structure is based on the metapopulation model described in detail in Danon et al (2009) (10) . In this model, the population is divided into electoral wards. Because of the changes in data availability, we restricted the model to England and Wales, whereas the original model covered Great Britain.\\n\\nTransmission between wards occurs via the daily movement of individuals. For each ward, we assume that individuals contribute to the force of infection in their \"home\" ward during the night and their \"work\" ward during the day. See Danon et al 2009 for further details (10) .\\n\\nData for population and movement of individuals come from the 2011 census of the United Kingdom. The population size of each of the 8,570 electoral wards is available directly from the Office of National Statistics (ONS) website. The number of individuals moving between locations is also available from the ONS website, but at the level of census output areas (OAs). We aggregated the data from OA level to electoral wards level. Spatial location of electoral ward centres are extracted from maps available from the ONS websites.\\n\\nWe use a Susceptible-Exposed-Infectious-Infectious-Recovered (SEIIR) model within each ward to capture the progression of disease within an individual (figure 1). Initial analyses used SARS-like parameters for the incubation period and infectious period, which now appear to differ from CoVID-19(4). Li et al (2020) analysed data on 425 cases reported in Wuhan in China and fitted a log-normal distribution to the incubation period, and a gamma distribution for the serial interval (2) . The infectious period for SARS was estimated as the serial interval minus the . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint incubation period, but as Li et al did not report the correlation between incubation period and serial interval, we were not able to estimate the infectious period distribution from the data, but used a uniform distribution between 2 and 3 days, to give a mean serial interval of approximately 7-8, in line with current estimates. We used two infectious states to represent a mildly symptomatic or prodromal period and a period with more pronounced symptoms. In the absence of data on the relative magnitude of these two infections states, we assumed the same length of time in each infectious state and assumed that each state was equally infectious. We sampled from each of the distributions 100 times independently (Table 1 ). \\n\\nThe census data are used to initialise the population sizes within each of the 8,570 wards. At the start of the model, all individuals are assumed to be susceptible to infection with no 3 e . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint underlying immunity in the population. We investigated a range of starting scenarios by seeding the infection in example wards London, Birmingham, Brighton, Sheffield and Cardiff. To seed infection in a ward, we move five individuals (non-commuters) from the susceptible compartment to the first infectious state.\\n\\nWe investigated the impact of a seasonally affected transmission rate, to capture potential decreased transmission during the summer months. We captured seasonal transmission by replacing the constant transmission rate with a time-varying transmission rate given by:\\n\\nis the magnitude of the seasonal difference in transmission, ranging from\\n\\n(maximum seasonality with no transmission at the peak of the summer).\\n\\nFrom the model, we extracted the total number of infections per day, as the number of individuals in both of the Infectious states, and the number of infected wards per day as the total number of wards with at least one individual in one of the two Infectious states. The spatial growth of the epidemic in England and Wales was visualised using interactive maps. We estimated the timing of the epidemic peak from the aggregated epidemic curve and calculated 95% prediction intervals from the model simulations.\\n\\nThe model is coded in C and is available on github (http://github.com/ldanon/MetaWards). The data are freely available from the Office for National Statistics website, or can be downloaded with the code at the github repository.\\n\\nWe predict that, in the absence of any interventions, a disease with \"best-guess\" CoVID-19-like parameters will peak a median of 133 days (range 126 -147 days) following the start of personto-person transmission in England and Wales. Intrinsic model stochasticity is responsible for variation between model runs. Using exactly the same parameters and seeding the infection in the same initial wards resulted in a difference in peak timing of +/-10 days (figure 2). The attack rate for best-guess parameters had a median of 45799874 (81.67% range 81.64-81.69), with a peak incidence median 1,116,692.\\n\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint\\n\\nThe initial location of cases had some, but limited impact on the timing of the epidemic in England and Wales. Epidemics seeded in Brighton, London, Birmingham and Sheffield resulted in synchronised epidemics in England, reaching urban areas first followed by rural areas. Epidemics started in Cardiff had a slower time to peak but still resulted in a generalised outbreak.\\n\\nSpatially, some disaggregation between England and Wales regions is observed. An outbreak starting in Brighton, (South East England) peaks in London and the South East first, and North East England, Yorkshire and Humber and Wales last, with a ten-day lag between regions ( Figure 3 ).\\n\\nModel predictions are highly sensitive to parameter values and incorporating parameter uncertainty increases the variability of model predictions. In the absence of any control measures, all predictions resulted in epidemics that peaked within a year from the start of person-to-person transmission in England and Wales. Estimates of peak time ranged from 78 days to 241 days (Figure 4 ).\\n\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint However, seasonality in transmission has a large impact on epidemic timing, peak incidence, and final attack rates. Assuming no difference in transmission rate during the year leads to a single large epidemic after approximately 4 months (June time if transmission starts in February), as above. With a 25% reduction in transmission the epidemic is smaller and peaks later, reducing the overall attack rate by 20%. A 50% reduction in transmission results in a smaller epidemic before the summer, followed by a resurgence in cases in the following winter. The attack rate is 10% less than a non-seasonal epidemic. A 75% reduction in transmission over the summer resulted in a delayed large outbreak, but with a similar attack rate. If transmission decreases to zero over the summer then the resulting outbreak is much reduced, with an attack rate of less than 1% ( Figure 5 , Table 2 ). . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.12.20022566 doi: medRxiv preprint Table 2 : Effect of seasonal variation on the timing (shown in days following initial seeding), the height of the peak, and the final attack rate.\\n\\nWe predict that, in the absence of control measures and with no seasonality in transmission, the introduction of CoVID-19 in England and Wales has the potential to result in a synchronised outbreak that peaks at around 4 months following the start of person-to-person transmission. Our findings suggest that the height of the epidemic and the attack rate is highly dependent on seasonality of transmission and that even small changes in transmission risk can lead to large changes in attack rate due to the spatial disaggregation of the population at risk.\\n\\nA combination of control measures and seasonal changes in transmission rate could shift the peak of the outbreak to the winter of 2020/21, with little effect on the final attack rate. If contact tracing and isolation efforts succeed in reducing transmission, but are unable to control the epidemic (9), an additional influx of severe CoVID-19 cases may exacerbate existing challenges with winter healthcare demand. A careful analysis of the impact of control measures on the timing of incidence of severe cases is warranted.\\n\\nThe strength of this model lies in the spatial heterogeneity which tempers transmission. As a comparison, an equivalent non-spatial model results in the epidemic peaking after 34 days, nearly four times faster than this spatial model, and would be unable to capture the interaction between spatial transmission and seasonality. The estimated total number of people infected in the spatial model is marginally smaller than for a non-spatial model, as the infection has the opportunity to die-out in local parts of the country. As the model framework was developed and published in 2009, it was possible to re-deploy the model for these new circumstances; developing such a model from scratch during an outbreak would be a significant challenge.\\n\\nA key element missing from our model is morbidity, mortality and the treatment of cases. The model in its current form predicts the total number of infections in the community rather than diagnosed cases. Observations from China suggest that many cases have mild symptoms and that only around 5% of cases have been reported and diagnosed (3) . The parameter estimates we used from China appear to be substantially different to previous coronaviruses (6) . Should CoVID-19 continue spreading the UK it will become possible to get UK-specific parameter estimates and improve prediction accuracy.\\n\\nAs with all modelling, it is impossible to capture the full complexity of an epidemic. In this model, the major assumptions are that we have assumed that there is no change in behaviour during the course of the epidemic. In practice, as the epidemic starts spreading in England and Wales there may well be a systematic change in behaviour as was seen during the H1N1 influenza pandemic in 2009. We have not included any age-effects, such as differential mixing, susceptibility or infectiousness. That means that we are not able to investigate the impact of school closures or the impact of the summer holidays, which had a large impact on the H1N1 influenza pandemic in 2009.\\n\\n  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#loop on the top 5 indicies and display the dataframe\n",
    "for index in answer_indicies:\n",
    "    display(FullPaperDataframe.iloc[[index]] )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.decomposition import PCA\n",
    "\n",
    "pca = PCA(n_components=0.90,random_state=30) #keep 90% of the variance\n",
    "tf_idf_norm_kmeans= pca.fit_transform(tfidf.toarray())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 29/29 [00:04<00:00,  6.01it/s]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 1440x1440 with 0 Axes>"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYEAAAEaCAYAAAD3+OukAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAgAElEQVR4nOzdd1gUV/cH8O8sS+9VQlOxt+hrBRtE0WiMUWOLFVuMYjRR3zeoKfYI9hIsMUYiyU/RRNEUNWJDjUZFjdjFqEGxoICNJjvn9wdhdWFmGRFYYc/nefLEnTk79+zOsmdn5s69AhERGGOMGSWVoRNgjDFmOFwEGGPMiHERYIwxI8ZFgDHGjBgXAcYYM2JcBBhjzIhxEaiAEhIS0Lx5c1hYWKBKlSqGTscgIiMjoVarS2x7Q4YMQVBQUIltj5WMffv2QRAE3Lhxw9CplFtcBPTIzMzE559/jho1asDS0hLOzs5o1qwZli5daujU9Prkk09gZ2eHCxcu4NixY7Jxa9euRZMmTWBnZwdbW1vUqVMH77//fhlmaliBgYEQBAGCIMDU1BRubm4ICAjAwoULkZmZqRO7ZMkSbNq0SfG21Wo1IiMjSzhjw1i0aBEsLS2Rmpoqub5r165o3bo1AEAURcyfPx/169eHtbU1HBwc0LBhQ3z22WfFavv7779H27ZtYW9vD2tra9SvXx+ffPIJbt68WezXU5RZs2YZ1Y8nLgJ6jB49GuvWrcO8efNw7tw57NmzB2PGjEF6erqhU9Pr8uXLCAgIQJUqVeDq6ioZExkZiVGjRmHo0KE4fvw44uPjERYWBo1GU8bZSiMiPH36tNTb6d+/P27duoVr165h165d6NWrFxYsWIDGjRvj7t272jh7e3s4OjqWej6vouDgYABAVFRUoXU3b97E9u3bMXLkSADAjBkzMHv2bEyaNAmnT5/GoUOHMGXKFDx58uSF2x0+fDiGDx+Otm3bYvv27Th37hyWLl2K27dvY8GCBS/3ospITk6OoVMoGjFZ9vb2tGzZMr0xwcHB1L59e51lUVFR9PxbO3XqVKpWrRpFR0dT9erVydLSkrp160YPHjygn376iWrWrEk2NjbUs2dPSk9P19tecnIy9e3bl+zt7cnCwoICAgLo2LFjRER09epVAqDz39SpUyW3061bN+rZs2eR70F0dDRVq1aNzM3Nyd/fn7Zu3UoA6MCBA0REtHfvXgJASUlJOs8zMTGhtWvXah9PmTKFateuTZaWluTl5UUffPCBzmtdu3YtmZiY0J49e6hRo0ZkampKP//8MxER/f7779SyZUuysLAgDw8PGjJkCN27d0/7XFEU6bPPPiNXV1eytramvn370sKFC8nExETvawsICKDhw4cXWp6UlEQODg40dOhQ7bKC+/nMmTPUsWNHsre3JysrK6pduzatW7eOiIgqV65caD8QEaWmptKAAQPI29ubLCwsqGbNmjR//nwSRbFQO6tWrSIfHx+ytbWld955h+7evauT465du6h169ZkaWlJdnZ21LZtW0pMTNSuX79+PTVs2JDMzc2pcuXKNH78eHr8+LF2/YEDB6hly5ZkY2NDNjY29Prrr9OOHTtk36uBAwdSvXr1Ci2fMWMGOTg4UEZGBhERNWzYkCZOnCi7HaV+/PFHAkDr16+XXJ+amkpEhT9/Sj+Ps2fPpqpVq5KZmRm5uLhQx44dKSMjg9auXSv7N/T06VOaOnUqValShczNzalu3bq0cuVKnXYA0JIlS6hfv35kZ2dHvXr10tveq4CLgB61a9emLl260P3792VjlBYBKysreuutt+ivv/6iffv2kYuLC3Xo0IE6d+5Mp06dori4OHJzc6NPPvlEti1RFKl58+bUsGFDOnDgAJ0+fZr69OlDDg4OlJKSQrm5uXTr1i3y8vKi0NBQunXrFj169EhyW6NGjaLKlSvTxYsXZds7ceIECYJAkyZNogsXLtBPP/1EVapUKVYRmDlzJsXFxdHVq1cpNjaWatWqRYMHD9auX7t2LQmCQE2bNqXdu3fTlStX6O7du7R7926ytLSkpUuX0qVLl+jo0aMUGBhIbdq00X55Ll68mKysrCgyMpIuXrxI4eHhZG9vX+wiQEQ0ZswYsrOzI41GQ0SF93ODBg2oX79+dPbsWbpy5Qr99ttv2qJ19+5dMjExocWLF9OtW7fo1q1bRER069YtCgsLo/j4ePr7778pKiqKrK2t6dtvv9VuNzg4mOzs7Oi9996jhIQEOnToEPn4+Oi8V7t27SKVSkUfffQRnTp1is6fP0/ffPMNnT9/XvteOjg40Lp16+jKlSu0f/9+atCgAQ0cOJCIiHJzc8nR0ZHGjx9Ply5dokuXLtHmzZspLi5O9r06cOAAAaA//vhDu0yj0VDlypVp7Nix2mWdOnWipk2b0o0bN/S+90Xp1q0bVa9evci44hSBn376iWxtbWnbtm10/fp1OnnyJC1atIgyMjIoIyODQkNDycvLS7vv8v+GgoODqUGDBrRz5076+++/acOGDWRvb0/ffPONth0A5OTkREuXLqXExES6ePGi3vZeBVwE9Dh48CD5+PiQSqWiBg0a0Pvvv08xMTGSv9yeJ1UETExMKCUlRbssJCSEVCqVzi+8cePGUZMmTWTziY2NJQB09uxZ7bKsrCxyd3en6dOna5dVrlyZZs6cqfe13bp1i1q1akUAqHLlytSnTx9atWoVPXnyRBszYMAA8vf313nesmXLilUECtq8eTOZmZlpv2Tzf4EV/CIKCAig0NBQnWXXr18nAHTy5EkiIvL09KQpU6boxPTs2fOlisCKFSsIAN25c4eICu9nOzs7va+vqNefb9y4cRQUFKR9HBwcTC4uLpSVlaVdNmfOHHJ3d9c+bt26NXXp0kV2m5UrV6YVK1boLNu/fz8BoNTUVEpNTSUAtHfv3iLze17dunV1jo527NhBACghIUG77Pz581SvXj0SBIFq1qxJgwcPpu+//56ePn36Qm3VqVOHunbtWmRccYrAwoULqUaNGpSTkyO5zZkzZ1LlypV1lv39998kCIK20OabPn06NWzYUPsYAA0bNkwnpqj2DI2vCejRqlUrXLlyBQcOHEBwcDDu3LmDnj174p133gG94Lh7np6ecHFx0T52d3eHu7u7zjl7d3d3nfPQBZ09exbOzs6oW7eudpm5uTlatGiBs2fPvlA+7u7uOHjwIM6dO4fJkyfD2toan3zyCerVq6fN4dy5c2jVqpXO8/IvAL6ozZs3o23btvDw8ICNjQ0GDBiAnJwc3L59WyeuWbNmOo+PHTuGxYsXw8bGRvtf/uu/fPkyHj58iJs3b6Jly5Ylkme+/P0rCILk+v/+978YMWIEAgMDMW3aNJw4caLIbYqiiLCwMDRq1AguLi6wsbHBypUrcf36dZ24OnXqwNzcXPvY09MTd+7c0T6Oj49Hx44dJdtISUnB9evXMWHCBJ33rHPnzgCAxMREODo6YsSIEXjzzTfRuXNnhIWF4eLFi0XmP3LkSERHR+Phw4cAgNWrV8Pf3x/169fXxtSuXRsJCQmIj4/Hhx9+iJycHIwYMQJ+fn6FLrbrQ0Sy7/3L6tOnD54+fYrKlStjyJAhiIqKwqNHj/Q+5/jx4yAiNG3aVOd9/fLLL3H58mWd2ObNm790e2WJi0AR1Go1WrZsiYkTJ2Lr1q2IjIzEL7/8gri4OACASqUqVBCkLmiamprqPM7vkVJwmSiKevOR+sN4mT+YOnXq4IMPPsC3336LkydP4saNG1ixYoXi7apUKm1sPo1Go/M6/vzzT/Tu3Rtt27bFli1bcOLECaxcuRKA7oUzExMTWFhY6GxfFEWEhobi1KlTOv9dvnwZnTt3LvLLurjOnDkDBwcHODs7S67//PPPcenSJfTp0wdnzpyBn59fkT1gFixYgDlz5mDs2LHYtWsXTp06hREjRhS6eGhmZqbzWBCEQp8xudeb/74vWbJE5/3666+/cPnyZTRo0ABA3hd4fHw8OnTogP3796N+/fpYtWqV3vwHDx4MIsIPP/yAO3fuYNu2bdoLwgVz+89//oOxY8di/fr12LVrF+Lj47Fx40a9239erVq1cO7cOcXx+ZR8Hj09PXHhwgV8++23cHNzw8yZM1GrVi0kJSXJbjf/+X/88YfO+3rmzBmcPn1aJ9ba2lrncXHaK0tcBF5QnTp1AED7a9nNzQ3Jyck6MUp+FRZHvXr1cO/ePZ0/juzsbBw9ehT16tV76e1XqVIFVlZW2tdWr149HDp0SCem4GM3NzcA0HkPTp06pfNHePDgQbi4uGDWrFlo0aIFatasqbhfd9OmTXH27FlUr1690H82Njawt7eHp6dnkXm+iBs3buCHH35Ar169tF8qUnx9fRESEoIff/wRM2bM0BZPIO+LvGBPq7i4OHTq1AnDhw/Hf/7zH1SvXr3Qr0glmjRpgp07d0quq1SpEry9vXHx4kXJ9+z5Ilu/fn1MmDAB27dvx/Dhw/H111/rbdfR0RG9e/fG6tWrERkZCSsrK/Tp06fIfAv+zSgxcOBAJCYmYsOGDZLr09LSJJcr+TwCeUfQnTp1wty5c5GQkICMjAzExMQAkN53TZo0AQD8888/hd7TatWqFfl69LVnaCV3N00FFBAQgH79+qFp06ZwdXVFYmIipkyZAgcHB7zxxhsAgKCgIISHh+Orr75C586dsWfPnhf6xfMi2rVrh+bNm6N///6IiIiAvb09Zs6ciaysLIwePfqFtjV69Gi4u7ujXbt28PHxwb1797BkyRI8fPgQ3bt3BwCMHz8ezZo1w6efforg4GCcPXu2UNe86tWro3Llypg2bRoWLVqEe/fuYcqUKTq/VGvVqoWUlBSsWbMGb7zxBg4ePIjly5crynPGjBno2LEjxo8fj+DgYNja2uLy5cvYtGkTvvrqK1haWmLixIn4/PPPUbt2bfj5+WHbtm2IjY1VtP3MzEzcvn0boiji3r17iIuLw5w5c+Dp6Yk5c+ZIPufx48cIDQ1Fz549UbVqVaSnp2PHjh06p+mqVq2KvXv3onPnzjAzM4OLiwtq1aqFqKgo7N27F56enli3bh3+/PPPF+56+vnnn6Nz5874+OOPMWzYMJibm+Pw4cPw9/dHrVq1MHv2bAwfPhwODg7o3r07TE1Ncf78eWzfvh2rVq1CYmIiVq9eja5du8Lb2xvJyck4cOAAGjduXGTbH3zwAVq1aoVr165h4MCBsLKy0lnfs2dPtGzZEi1btoSHhwdu3ryJWbNmwdTUFF26dAEAHD16FIMHD8a6desKnTrJ16tXLwwePFj7uXvrrbfg6emJq1evIjIyEo6Ojli4cGGh5yn5PK5ZswaiKKJ58+ZwcHDA7t278ejRI+3+q1q1Km7fvo3Dhw+jRo0asLKyQvXq1TFs2DC8//77mDt3Lvz9/fHkyRPEx8cjJSUFoaGhsu9ZUe0ZnCEuRJQXc+bModatW5OrqyuZm5uTt7c3DRgwQOfCLBHRrFmzyMPDg6ytrem9996jr776SrKL6POkLj7NmTOHPD099eZUsIto27ZttV1E8ym5MPzTTz9R165dydPTk8zMzMjNzY2CgoLot99+04lbv349+fr6kpmZGTVv3pxiYmJ0LgwTER05coQaN25MFhYW9Prrr1NcXFyhC6OfffYZubm5kZWVFXXu3Jn+7//+jwDQ1atXiehZF1EpcXFx1L59e7KxsdF2x/zoo4+0Fxs1Gg1NnjyZnJ2dycrKinr27Km4iyj+7QaoVqvJ2dmZ2rZtSwsWLCjUc+P5C8OZmZnUr18/bVdBV1dX6tOnD/3zzz/a+O3bt1Pt2rXJzMxM+1lIT0+n3r17k62tLTk5OVFISAh99tlnOp8DJR0NiPIuyvr5+ZGFhQXZ2dlRYGAgXblyRbt+y5Yt5OfnR5aWlmRra0sNGzbUdh5ITk6mHj16aPf9a6+9RiNGjCiye3K++vXrEwD666+/Cq37+uuvKSgoiNzd3cnMzIw8PDyoW7duOr2K8i/eKrkwHRkZSa1atSJbW1uysrKievXqUWhoKCUnJ+ts6/kLwUV9Hn/66Sfy9/cnBwcHsrS0pHr16un08MnJyaF+/fqRo6OjThfR3NxcCg8Pp1q1apGpqan287Jx40btcwFQVFSUzmsoqj1DE4h4ZjGm3LVr11C1alUcOHDgpS++MsYMj68JMMaYEeMiwBhjRoxPBzHGmBHjIwHGGDNiXAQYY8yIlcv7BArenKWUi4sL7t27Z9Sxhm6/vMUauv2KHGvo9it6bEEeHh6Sy/lIgDHGjBgXAcYYM2JcBBhjzIhxEWCMMSPGRYAxxoxYuewd9KLEI/tAW6JwJ+0e4OgCoccgqPwCDZ0WY4wZXIUvAuKRfaCoCCAnO29BagooKgIiwIWAMWb0KvzpINoS9awA5MvJzlvOGGNGrsIXAaTK3Fght5wxxoxIxS8CTi4vtpwxxoxIhS8CQo9BgJl54RW+tco+GcYYe8VU+AvDKr9AiPj32sC/vYPg4AwcPwjRpxpUnXsaOkXGGDOYCl8EgH97AfkFagdfIo0G9O0i0ObvIKoEqN5819ApMsaYQRhFEShIMDEBho0HiEA/RkKEANWbPQydFmOMlTmjLALAv4Vg+AQAAP24FqIgQNWxu4GzYoyxsmW0RQB4rhAQgTZ9m1cIOnQzdFqMMVZmjLoIAP8WghET8wrBxjXQ/JMIXDrHQ0wwxoyC0RcB4FkhoHt3gSP7n63gISYYYxVchb9PQClBrQYepRdewUNMMMYqMC4Cz+MhJhhjRoaLwPN4iAnGmJHhIvAc2SEmmrct+2QYY6wMcBF4jsovEMKgMYCTKyAIz4aY2Psr6MoFQ6fHGGMljnsHFVBoiIkHaRDDQyEunQHV/76E4FXF0CkyxliJ4SOBIgj2jlBNmAmYmUFcPBV095ahU2KMsRLDRUABwaUSVONnAJpciIu+AKXdN3RKjDFWIrgIKCR4+EA1bhrw6GFeIXj80NApMcbYSyuzawJPnjzBypUrkZSUBEEQMHr0aJw6dQq7d++GnZ0dAKBfv35o3LhxWaX0woSqNaAa+xnExdMgLpkO1cSZECysDJ0WY4wVW5kVgbVr16JRo0aYOHEicnNzkZ2djVOnTqFLly545513yiqNlybUagDVqFCIy7+EOPu/QE4W7qTd53GGGGPlUpmcDsrIyMD58+fRrl07AIBarYa1tXVZNF0qhIbNgTYdgds38u4mJno2ztCRfYZOjzHGFFN0JHDmzBm4ubnBzc0NaWlp+OGHH6BSqdC/f384ODgU+fy7d+/Czs4Oy5cvx/Xr1+Hr64shQ4YAAHbu3Im4uDj4+vpi8ODBsLGxeakXVGYS4gsvyx9niI8GGGPlhEBEVFTQ+PHj8emnn8LFxQVLliwBAJiZmeHhw4cIDQ0tspErV67g008/xcyZM1GjRg2sXbsWlpaW6NSpk/Z6QHR0NNLS0hASElLo+bGxsYiNjQUAhIWFIScn54VeZD61Wo3c3NwSib3zbqu8I4CCBAGVNh8qkxyKE2vo9stbrKHbr8ixhm6/oscWZGZmJr1NJU9OTU2Fi4sLNBoN/vrrLyxfvhxqtRoffPCBosadnZ3h7OyMGjVqAAD8/PwQExOjcxTRvn17hIeHSz4/KCgIQUFB2sf37hVvQLf8G8BKJNbRBUhNKbzc1l7v80o0h2LEGrr98hZr6PYrcqyh26/osQV5eHhILld0TcDS0hLp6ek4d+4cvLy8YGFhAQCKK5KDgwOcnZ2RnJwMAEhISICXlxfS0tK0MUePHoW3t7ei7b0KZMcZevwI4tG4sk+IMcaKQdGRQKdOnTB58mTk5uZqz+VfuHABnp6eihsaNmwYli5ditzcXLi5uSEkJARr167FtWvXIAgCXF1dMXLkyGK9CENQ+QVCBPKuAfw7Cxk69wSOxoFWz4d4KwlC134QVHwrBmPs1aWoCHTv3h3NmzeHSqWCu7s7AMDJyQmjRo1S3FCVKlUQFhams2zs2LEvkOqrp+A4QwBArTuAvl8B+iUalJwE1bCPIZhbGDZRxhiTofhnan7PoD/++ANAXhFwc3MrtcTKK0FtCiF4LITew4CThyHOnQziSWkYY68oRUcC//zzD8LDw2Fqaor79++jZcuWOHfuHPbv34/x48eXdo7ljiAIEDp2B73mBfHreRC/nAgEdAIOxvIE9oyxV4qiI4HVq1ejb9++WLx4MdTqvLpRt25dXLjAY+zrIzRoCtWkeYBGA2xbn9ebiG8sY4y9QhQVgRs3bqBNmzY6yywsLIrdX9+YCJ4+gKlp4RU8gT1j7BWgqAi4urri77//1lmWmJiovUjMipCWKr2crxUwxgxM0TWBvn37IiwsDB06dEBubi62bNmCXbt2Kb5ZzOg5ydxYZu9Y9rkwxthzFB0JNGnSBJMnT8bDhw9Rt25dpKSk4L///S8aNmxY2vlVCLI3lmU8Av11rOwTYoyxfykeStrX1xe+vr6lmUuFJXljWYduwOG9ECNmQeg2AMJbvSEIgqFTZYwZGUVFYP78+ejSpQvq1KmjXXb+/Hn89ttvmDhxYqklV5FI3ljW5k3Quq9AMd+Dkv6GashHECwsDZsoY8yoKDoddO7cOdSqVUtnWc2aNXH27NlSScpYCObmEEZMgNBrKHDiCMSwT0Aptw2dFmPMiCg6EjA1NUVWVhasrJ5NpZiVlQUTE5NSS8xYCIIA4c0eIM/KEFfPgzh7Yt6ENUfj+MYyxlipU3Qk0LBhQ3z99dfIyMgAkDdT2Jo1a9CoUaNSTc6YCPUbQ/Xpgrx7Cnb8xDeWMcbKhKIiMHjwYGRmZmLYsGEYMWIEhg0bhoyMDO2IoqxkCG4egCCxS/jGMsZYKVF0OsjGxgaTJ09Geno67t27BxcXF0XTSrJiSLsvvZxvLGOMlYIXGuxeEATY2toiOzsbd+7cwZ07d0orL+Pl5CKz3Lls82CMGQVFRwKnTp3CihUrkJ6eXmhddHR0iSdlzIQeg0BREUBOtu4KS2tQdjYEc4mbzhhjrJgUFYE1a9agZ8+eCAwMlJ2smJUMyRvL6jYCDsVCXDwVqrGfQ7CyNnSajLEKQlERePz4MTp06MB3tJYRqRvLxLqNQGsWQlz4OVQfTYNga2fYJBljFYKiawLt2rXD3r17SzsXpoeqWRuoQqYAN69DnDcZlC5zAZkxxl6AoiOBy5cvY/v27di6dWuhXkHTp08vlcRYYcLrzaD6aCrEr2ZDnDsZqgkzIbhUMnRajLFyTFERaNeuHdq1a1fauTAFhNqvQzVhBsQl0yCGTwLadwX2/sp3FzPGikVREQgMDCzlNNiLEHxrQfW/LyGGTQJ+iny2Iv/uYoALAWNMEcVDSaenpyMxMRGPHj0CEWmX8xGCYQheVQELSyA7U3dF/t3FXAQYYwooKgJHjx7FsmXL8NprryEpKQne3t5ISkpC7dq1uQgY0oM06eWpKaDcXAhqxTWeMWakFH1LREdHIyQkBP7+/hg6dCjmzp2LvXv3IikpqbTzY/rITVsJQJw0HELL9hBad4Dg9hrEI/tAW6L42gFjTIeiLqL37t2Dv7+/zrKAgADExcWVSlJMGclpK83M82Ytq1IDtGMzxE8/gOaLMaDvlvLIpIyxQhQdCdjZ2SE9PR0ODg5wdXXFpUuXYGtrC1EUSzs/pofU3cXP/8KntPugQ7Ggn9cDBfcVXztgjEFhEWjfvj0uXLgAPz8/dOnSBdOnT4cgCHj77bdLOz9WBKm7i/MJjs4Q3u4Lzdb/k34yj0zKmNFTVAS6d++u/XdAQADq1auHrKwseHl5lVpirATJXTuwsOBB6RgzcoquCcydO1fnsYuLC7y8vDB//vxSSYqVLMlrByoTICsT4qyPQVcvGSYxxpjBKSoCchPK80Tz5YPKLxDCoDGAkysgCICTK4ShH0E1YSaQnQ0x7BOIW38A5eYaOlXGWBnTezoof66A3NzcQvMG3LlzB66uroobevLkCVauXImkpCQIgoDRo0fDw8MDixYtQkpKClxdXTF+/HjY2NgU42WwoshdO1BNWwrasBr0SzTo9HGoho+H4OFjuEQZY2VKbxG4fz9vpEpRFLX/zufi4oI+ffoobmjt2rVo1KgRJk6ciNzcXGRnZ2PLli1o0KABunfvjpiYGMTExGDgwIHFeBmsuAQrGwjDxoMa+UGMioA4czzQuCWQeI7vKWDMCOgtAiEhIQCAmjVrIigoqNiNZGRk4Pz58xgzZkxeo2o11Go1jh07hmnTpgHIu+A8bdo0LgIGIjT2h6p6bYhLZgBH9z9bweMRMVahCfT8QEAybty4ARsbGzg4OCArKwvbtm2DSqVC165dYa6gZ8m1a9ewatUqeHl54fr16/D19cWQIUMwatQoREZGauOGDh2KtWvXFnp+bGwsYmNjAQBhYWHIycl5gZf4jFqtRq7C894VNbaouJSRPSCmFJ47WuVaCa5fbynTXF+FWEO3X5FjDd1+RY8tSG5WSEVdRJcsWYLx48fDwcEB69atw61bt2Bqaoqvv/4aY8eOLfL5Go0GV69exbBhw1CjRg2sXbsWMTExipMPCgrSORIp2B9eKam+9MYWW1ScmHJXZvkd3N0WDaFZGwimhT9Mhn5dpRVr6PYrcqyh26/osQV5eHhILldUBFJSUuDh4QEiwrFjx7BgwQKYmZnhww8/VNS4s7MznJ2dUaNGDQCAn58fYmJiYG9vj7S0NDg6OiItLQ12djxlosHJ3VOgMgGtXQLa9C2EVh0gBHaG4FKJxyRirJxTVARMTU2RmZmJGzduwNnZGXZ2dtBoNHj69KmiRhwcHODs7Izk5GR4eHggISEBXl5e8PLywv79+9G9e3fs378fzZo1e6kXw16e0GMQKCoCyMl+ttDMHBg0Bip7R4h7fwX9HgP6fQvgVQW4dQPI/fdzwNcPGCt3FBWBVq1aYcaMGcjMzESnTp0AAFevXoWbm5vihoYNG4alS5ciNzcXbm5uCAkJARFh0aJF2LNnD1xcXDBhwoTivQpWYooaj8ikTkNQagpo/07Q9k15A9I9j8ckYqxcUVQEhgwZgr/++gsmJiaoX78+AEAQBAQHBytuqEqVKggLCyu0/DJ5aQAAACAASURBVIsvvlC8DVY29I1HBACCkyuEHgOh+W2T9AZ4TCLGyg3Fs440bNhQ53G1atVKPBlWzshdP3ByKftcGGPFIlsEZs+ejU8//RRA3q91QRAk46ZPn146mbFXnuT1AwDwqQYikv3MMMZeHbJFICAgQPtvnkKSSZG6fgBnN+DUEdDWH4BuA7gQMPaKky0CrVu31v47MDCwLHJh5VDB6wckiqDvl4N+3Zg3kU2PQVwIGHuFFXlNICcnB3v37sX58+fx5MkTWFtbo27duggMDJS9A40ZL0GlAgaGACoVaPuPeYWgZzAXAsZeUXqHks7IyMDkyZOxefNmqNVqVK1aFWq1Gj/99BMmT56MjIyMssqTlSOCSgVhwGgIgW+Bdm4G/RgJBaOTMMYMQO+RQExMDOzs7DB79mxYWFhol2dlZWHevHmIiYlB//79Sz1JVv4IggD0/wAQhLwby0QR6DOMjwgYe8XoLQInTpxASEiITgEAAAsLCwwYMAARERFcBJgsQRCAfiPzTg3FbgUl/wPcvsFDTDD2CtFbBFJSUuDjIz3BiI+PD1JSJPqIM/YcQRCAviNAt28CZ088W8FDTDD2Sihyekm1WrpOqNVqPrRnigiCANxKKrwif4gJxpjB6D0SePr0aaFpJZ9X3HGtmRGSG0qCh5hgzKD0FoHWrVsXmlbyea1atSrxhFgFJTfEhIkJ6MoFCNVql31OjDFl00sy9rIkh5hQqwFTc4hhn0DwbwehVzAEO0fDJcmYEVI8gBxjL0NuiGqhUQvQbxtBv28FnfoTQrf+efcXHDvAk9UwVga4CLAyIzdEtfBuMKhle4jrV4M2rAbt2Aw8fsiT1TBWBorsHcRYWRDcvaD6eBpUoycDD9OeFYB83JOIsVLBRwLslSEIAtDYHxBlhpjgnkSMlTjZInDmzBlFG8ifaYyxEiPXk8jWruxzYayCky0CK1as0HmcmpoKQRBga2uLR48egYjg7OyMr776qtSTZMZFdrKaRw8grpoLofdQCE6uhkmOsQpGtghERERo/71582Y8fvwYffv2hbm5ObKzsxEdHQ1bW9sySZIZF8nJat7pByH1Hmj7j6DTxyC83RdCUDcIpqaGTpexck3RNYFff/0Vq1at0g4hYW5ujv79++ODDz5Ajx49SjVBZpzkehKR/xsQo9eANq8DHdoN1Xvvgx4/5O6kjBWToiJgYWGBxMRE1K797K7OK1euwNzcvNQSY0yK4FIJJmOmgM7EQ1y/GuKSaYCgAkjMC+DupIy9EEVFoG/fvvjyyy/RpEkTODs74/79+zhx4gSGDx9e2vkxJkmo3wSqaa9DnDgYyHyiuzK/OykXAcaKpKgItG3bFr6+vjhy5AjS0tLg6emJnj17wsvLq7TzY0yWYGoKZMrMbifVu4gxVoji+wS8vLzw7rvv4sGDB3B05PFd2CtCrjspAPH75RA69YTgUqmMk2Ks/FBUBJ48eYJvvvkGR44cgVqtRlRUFI4fP47ExES89957pZ0jY7Iku5OamgG+tUEHY0EHd0HwC4TQuTeESh4Qj+zji8iMPUdREVi9ejWsra2xfPlyTJgwAQBQs2ZNrFu3josAMyi5gelUfoGg1Hug37eA4naC/tgLVK0JJF0BnvKYRIzlU1QEEhISdLqIAoCdnR0ePHhQaokxppTswHROLhDeex/0Vi/Q7zGgnVsKP5kvIjMjp2gAOSsrKzx69Ehn2b179/jaACsXBDtHqHoNlQ/gMYmYEVNUBNq3b48FCxbgzJkzICJcunQJERER6NChQ2nnx1jJkRtqwtwclCY/gx5jFZmi00HdunWDqakp1qxZA41GgxUrViAoKAhvvfWW4obGjBkDCwsLqFQqmJiYICwsDBs3bsTu3bthZ5c3MFi/fv3QuHHj4r0SxoogeRFZpQJysiF++gGEN7rk9SbigeqYESmyCIiiiH379qFjx47o0qXLSzU2depU7Rd+vi5duuCdd955qe0ypoTs7GbVaoN+Xg/aFQOK2wGhQ3eQgzPwazT3ImIVXpFFQKVSYd26dWjXrl1Z5MNYqZK9iDxsPOjNnhC3/QD6eb3uk7gXEavABCKSmcHjmWXLlsHf3x9NmzYtdkNjxoyBjY0NAKBDhw4ICgrCxo0bsX//flhaWsLX1xeDBw/WxjwvNjYWsbGxAICwsDDk5OQUKwe1Wo3c3FyjjjV0++UhNmXo2xDTUwstV7lWguvXEj2MDJhreY41dPsVPbYgMzMzyeWKisDChQtx/Phx1KxZE87OznkzQP3rww8/VJRAamoqnJyc8ODBA8yaNQtDhw6Fh4eH9vRQdHQ00tLSEBISUuS2kpOTFbVZUMFff8YYa+j2y0Os5v1uAKT+LASYrN5a6u0bS6yh26/osQV5eHhILld0Ydjb2xve3t7Fajifk5MTAMDe3h7NmjVDYmIi6tatq13fvn17hIeHv1QbjJUIuaEoTE1BTx5BsOZ5NFjFoagI9O7d+6UaycrKAhHB0tISWVlZOH36NHr16oW0tDTtvQZHjx596ULDWEmQ7EVkogZyn0KcOR6qUaEQqtQwWH6MlSTFA8jl5uYiOTkZDx8+1FmuZI7hBw8eYP78+QAAjUaD1q1bo1GjRli2bBmuXbsGQRDg6uqKkSNHvmD6jJU82V5ElTwgrgyHGB4Koe8ICAGddU6NMlYeKSoCFy5cwMKFC/H06VNkZmZqf9ErnWO4UqVKmDdvXqHlY8eOffGMGSsDcr2IVJ8vgrhmEeiHlUDieWDQGAjmFoZLlLGXpKgIfPfdd3jnnXfw9ttvY+jQoVi7di1+/PFH2avNjFVUgo0dVGM/B/22CbTt/0D//A20bA/s/ZXvKWDlkqJhI5KTkwvdHdy9e3f8+uuvpZIUY68yQaWC6u2+UH08Pe8C8k+Ref8nenZPwZF9hk6TMUUUDyCXmZkJAHBwcMCNGzfw+PFjZGVllWpyjL3KhLqNAEvrwivyRyZlrBxQdDqoRYsWOHnyJFq3bo127dph+vTpMDExgb+/f2nnx9irTeKmMgB5RwQZTyBYSRQJxl4hiorAkCFDtP/u2rUrqlevjqysLDRs2LC08mKsfNA3veV/gyE0agHB/w2g7n8gmJjwzGbslaO4i+jz6tSpU9J5MFYuSd5TYGYOdHoXwsMHoGMHQMcOAHYOgLcvcCmBZzZjrxRFReCLL76Q7Q89ffr0Ek2IsfJE3/SWAEB9hwMJ8RCP7AVOHC68AZ7ZjBmYoiJQcATR9PR07N27F23atCmVpBgrT+TuKQAAQW0K/McPJv/xg+Z9mSHTeWYzZkCKikBgYGChZX5+fli+fDl69epV0jkxVjE5uUpfP3ByKftcGPuXoi6iUpycnHD9+vWSzIWxCk3oMSjvekFBbyifoY+xkqboSGDPnj06j3NycvDnn3+iZs2apZIUYxVRoesH9o5AZiYQtxPUqgNPa8kMQlEROHDggM5jc3Nz1KpV66Wnm2TM2BS8fkBXLkCc/ynElXOgGj8j7xoCY2VIURGYOnVqaefBmFESqtWGMGQc6JsFoO+XA8HjeGRSVqYUFYE7d+4o2lilSpVeKhnGjJGqRQDE2zdBv2wAXvOG8Oa7hk6JGRFFRWDcuHGKNhYdHf1SyTBmrISu7wG3b4B++g5UyRNCoxaGTokZCUVFYNSoUUhISEDv3r3h6uqKlJQU/Pjjj2jQoIFk91HG2IsRVCpg6Eege3cgfrMAqk/CIPj4GjotZgQUdRGNjo7GqFGj8Nprr0GtVuO1117DyJEjsWHDhtLOjzGjIZiZQzXmU8DKBmLELNCDNEOnxIyAoiMBIsLdu3fh5eWlXZaSkgJRFEstMcaMkeDgBNWHn0IMnwQxLBTQ5OJO+n0ebI6VGkVFoEuXLpgxYwYCA591bdu/fz93EWWsFAg+1YC2bwKx254t5MHmWClRVATeeecd+Pj44PDhw7h27RocHBwwevRoNGrUqLTzY8w48WBzrIwoHkq6UaNG/KXPWFmRG1QuNQV0NxmCm0fZ5sMqLL1F4NSpU7C0tEStWrUAALdv30ZERAT++ecf1KxZEyEhIXB0dCyTRBkzKvomq/l0FOBTDULT1hCatoLg6s6T1bBi09s7KDo6WufuxRUrVsDKygofffQRzM3NERXF86gyVhokB5szMwf6joDQeyhgYgLa/B3EKSOhmTQCFLmEJ7tnxaL3SOD27duoVq0aAODBgwe4ePEili9fDicnJ1SvXh3/+9//yiRJxoxNUZPVoGMP0L07oPg/8mI0Gt0N8PUDppDiawKXLl2Cm5sbnJycAAC2trbIysoqtcQYM3b6JqsBAMGlEoQ3e0DzY6T0BlJTQLlPeVA6ppfe00HVq1fH9u3bkZGRgd27d+tcGL5z5w5sbW1LPUHGWBH0TEojTvkA4q6toKyMMkyIlSd6i0BwcDB27tyJoUOH4tatW+jevbt2XVxcHE84z9grQPb6wZs9AFd30MY1EEOHQ9zyPehhGsQj+6AJHY4777aCJnQ4XzswcnpPB3l5eWHZsmV49OhRoV/9Xbp0gVqt+GwSY6yUFDnZ/d8XIe7cDNq+CbT9R0AAkH+3P9+EZvQUfYtLnfaxtrYu8WQYY8Wjd7J731owGT0ZdPsGxFkTgOwC1/L4IrJRK/Ycw4yx8kVw9wKys6VXpqaAiMo2IfZKKLPzOWPGjIGFhQVUKhVMTEwQFhaGx48fY9GiRUhJSYGrqyvGjx8PGxubskqJMeOj7ya0sE+g6twTeL153tDWzCjIFoEdO3agU6dOAPLuF3B3d3/pxqZOnQo7u2eTacfExKBBgwbo3r07YmJiEBMTg4EDB750O4wxaUKPQaCoCCDnuSMCUzOgaRvgUgLEiC8BDx8InXpCaNYGdPwg34lcwcmW+/Xr12v/HRoaWiqNHzt2DAEBAQCAgIAAHDt2rFTaYYzlUfkFQhg0BnByBQQBcHKFMPhDmAz7CKrZqyAMnwAIAujbRRD/GwyKXMp3IldwAsmcCAwNDUW9evXg5eWFNWvWYPjw4ZIbaNeunaKGxowZoz3V06FDBwQFBWHIkCGIjIzUxgwdOhRr164t9NzY2FjExsYCAMLCwpCTk6OozYLUajVyc3ONOtbQ7Ze3WEO3b4hYIkLO8T+QPncKkPu00HqVayW4fr3lpXN4ld+DihBbkJmZmfQ25Z7w0UcfYdu2bTh06BA0Gg0OHDggGae0CMycORNOTk548OABZs2aBQ8P5aMgBgUFISgoSPtY6u5JJeTuvDSmWEO3X95iDd2+wWKr1gJkvmzElLt62+D39tWILUjuO1e2CHh4eGDUqFEAgBkzZuCLL74oVsP58oebsLe3R7NmzZCYmAh7e3ukpaXB0dERaWlpOtcLGGMGJncR2YFHDq5IFHUB+OKLL6DRaHDu3DkcPHgQ58+fh6bggFV6ZGVlITMzU/vv06dPw8fHB02bNsX+/fsBAPv370ezZs2K8RIYY6VB8k5kAHjyGHTqSNknxEqFoi6iycnJ2nPxzs7OuH//PkxNTREaGqoz77CcBw8eYP78+QAAjUaD1q1bo1GjRqhWrRoWLVqEPXv2wMXFBRMmTHi5V8MYKzFSdyKj/TvA0f0QI76E8GYPCN0HQeCRA8o1RXtv9erVCAoKQteuXbXzC2zbtg1r1qzB1KlTi3x+pUqVMG/evELLbW1tX/o0E2Os9EjdiUxvdAZtXAPauQV05SJUH/wPgoOzYRNlxabodNC1a9fw9ttv60ww06VLF1y7dq208mKMvaIEUzOoBoyGMGIi8M8ViDM+Bp3/iwemK6cUHQk4OTnh3LlzqF+/vnbZ+fPneWpJxoyYqkUAyLsqxBVhEBd+DpiYPJvchgemKzcUFYF+/fohPDwcTZo00R4WnjhxAmPHji3t/BhjrzDBwweqTxdAnDhY9y5kgAemKycUFYGmTZsiPDwchw8fRlpaGry9vdGnT58X6uvPGKuYBAtLQO4GztTi9WlnZUfxZX0PDw/07NmzNHNhjJVXsgPTEcS1SyC07gBUrwNBECAe2cfjEb1CuG8XY+ylSQ5MpzYFqtUCxf8B+mM34O4JePsCp/4Env575MDXDgyOiwBj7KXpm92MsjLzCsHBXcAxieFn+NqBQXERYIyVCLnZzQQLSwit2gOt2kPz/jvST+ZrBwbDM0cwxsqOk6v0cke+2cxQFB0J3Lt3D5s2bcK1a9eQlaU7P+mSJUtKJTHGWMUjee0AADS5oLvJENy4x2FZU1QEFi5cCA8PD/Tp00d2TGrGGCuK5HhE/oHAvh0Q5/wPqtFTINSsZ+g0jYqiInDz5k3MmjULKp53lDH2kiTHI2rZHuLSmRAXfQ5h8Fio/N8wbJJGRNG3epMmTXDu3LnSzoUxZqQENw+oJs8FqtXJm9oy5nuQKBo6LaOg6Ehg2LBh+Oyzz1CpUiXY29vrrAsJCSmVxBhjxkWwtoXq42mgH1aCft0I3EkGhn4EQWpOA1ZiFBWB5cuXQ6VSwdPTk68JMMZKjaA2BQZ/CFTyAG1eB7qWCGie4k56Kt9dXEoUFYEzZ85g1apVsLS0LO18GGNGThAECJ16QpOeBuze9mwF311cKhRdE6hcuTIePXpU2rkwxtgzJw8XXpZ/dzErMYqOBOrVq4fZs2cjMDCw0DWBdu3alUpijDEjJ3cXcWoK6MwJoN5/dCa6YsWjqAhcvHgRTk5OOH36dKF1XAQYY6VCbmRSQQVxyTTA3QtC+64Q/N+AYG7Bo5MWk6IioGQeYcYYK0mSdxebmQMDRkFQmYBit4F+WJF3eqh6HeD8Xzw6aTEoKgKinv66fAMZY6w06BuZFACoRQBw5TzE2G1A/B+FN8CjkyqieHpJOdHR0SWWDGOMPU9uZFIgrxcRqteFSfW6PDrpS1BUBL766iudx2lpaYiJiUHTpk1LJSnGGHshTq7S1w9MTUFXL0OoWqPscyonFJ3LcXV11fmvZs2a+PDDD7F169bSzo8xxook9BiUd73geSYmgCBA/HIiNEtngK5dNkxyr7hiTyqTkZGBhw8flmQujDFWLHLXD4RGzUG7fwH9HgNx9kSgYXOouvYD3UrinkT/UlQEli1bptMfNzs7G+fPn0ebNm1KLTHGGHsRsjObdekDavc2aPfPoF0xEGeNBwQVQP92eDHynkSKioC7u7vOY3Nzc3To0AGvv/56qSTFGGMlSbC0gvB2X1C7tyFOGg5kZugGGHFPIkVFoHfv3qWdB2OMlTrByhrIzJReaaQ9ifQWgb///htqtRo+Pj4AgIcPHyIyMhJJSUmoUaMGBg8eDAsLizJJlDHGSoTcnchW1iAioxuKQm/voMjISKSnp2sfr1y5Erdu3UL79u2RlJSE77//vtQTZIyxkiTZk0hQARmPIa4MA2U8NkxiBqL3SODmzZuoU6cOAODJkyc4efIkFixYAA8PDzRt2hSff/45RowYobgxURQxadIkODk5YdKkSYiIiMC5c+dgZWUFABgzZgyqVKlS/FfDGGNFkJznuPtACA/TQVvWQZw5HqqRnxjNvQV6i4BGo4FanRdy+fJlODg4wMPDAwDg4uKCJ0+evFBjv/32Gzw9PZH53Dm5QYMGwc/P70XzZoyxYpPrSUTV60D8eh7E8FAIvYbkDVBXwU8P6T0d5O3tjcOH88b0PnToEBo0aKBdl5qaqv0Fr8T9+/dx4sQJtG/fvpipMsZY6RKq1Ybqi8VA/cag6G8gLp8Dzf4d0IQOx513W0ETOhzikX2GTrNECUREcisvXLiA8PBwAHkDxc2cOVN7JPDLL7/g8uXLGD9+vKKGFixYgB49eiAzMxM///yz9nTQpUuXYGpqivr162PAgAEwNTUt9NzY2FjExsYCAMLCwpCTk/PCLxQA1Go1cnNzjTrW0O2Xt1hDt1+RYw3dvr5YIkLGz9F4HLkMKPgVaW4Ou9GTYBnw5iuTrxJyUwPrLQIAkJmZiVu3buG1117TmV4yOTkZFhYWcHJyKrLx+Ph4nDx5EiNGjMDZs2e1RSAtLQ0ODg7Izc3FqlWr4O7ujl69ehW5veTk5CJjpEgNQmVssYZuv7zFGrr9ihxr6PaVxGomDgYephde4eQKk/A1ZZJDcWMLyv8BX1CR9wlYWlrC19dX8QalXLx4EcePH8fJkyeRk5ODzMxMLF26FOPGjQMAmJqa4o033sDPP/+seJuMMVbqHj6QXl6B7iko9thBL6J///7o378/AGiPBMaNG4e0tDQ4OjqCiHDs2DF4e3uXRTqMMaaM3D0FZmagRw8g2NoXXlfOlEkRkLN06VLtIHSVK1fGyJEjDZkOY4zpkJzdTGUCPM2B+NloCD0HQ2jdEUI5nlyrzItAvXr1UK9ePQA8bSVj7NUmOzqpjy/EH1aCopaDDu2GasBoCD6FT5uXBwY9EmCMsVed3D0Fqv/OBh3ZB9r0LcRZEyC0fxvk4QP8El2uhqjmIsAYY8UgCAIE/zdArzcDbV4Hit2mG1BOhqguvyeyGGPsFSBY20A1KASwcyi8Mn+I6lcYFwHGGCsJ5bQ7KRcBxhgrCU4u0ssdir6h1pC4CDDGWAmQHKIayDsldPOfsk9IIS4CjDFWAlR+gRAGjQGcXAFByPt/94GAqRnE8FDQxQRDpyiJewcxxlgJkepOSn6BEJdMh7h4KoShH0PVvK1hkyyAjwQYY6wUCc5uUIWGAVVrglbPh/j7FhQxbmeZ4iMBxhgrZYK1LVTjZ4DWLAJtWguk3oNYuRoQ84PBbyzjIsAYY2VAMDUDRv4P2OScd2OZoAJIzFtZxI1l4pF9oC1RpVIw+HQQY4yVEUGlgqrvCMDK+lkByJeTDdq0FnT7Big9FZSVCSLKKwBREXmjmRI9KxglNMMZHwkwxlhZy8iQXv4wDeLnIc8e589vXPAaQv6dyCVwNMBFgDHGyprcPAW29hD6DAeyMoHsTCArE/RLtPQ2SuhOZC4CjDFWxiTnKTAzh9BneKFz/Zo/9kgXDLk7lF8QXxNgjLEyJnVjmTBojOTFXsk7kc3M85aXAD4SYIwxA5Cbp0AqTmpim5LqHcRFgDHGXnFKC0axtl2iW2OMMVaucBFgjDEjxkWAMcaMGBcBxhgzYlwEGGPMiAn0Ko1pyhhjrEwZ1ZHApEmTjD7W0O2Xt1hDt1+RYw3dfkWPVcqoigBjjDFdXAQYY8yImUybNm2aoZMoS76+vkYfa+j2y1usoduvyLGGbr+ixyrBF4YZY8yI8ekgxhgzYlwEGGPMiBnFKKL37t1DREQE0tPTIQgCgoKC8NZbb0nG5uTkYOrUqcjNzYVGo4Gfnx/69Okju21RFDFp0iQ4OTnp7b41ZswYWFhYQKVSwcTEBGFhYbKxT548wcqVK5GUlARBEDB69GjUrFmzUFxycjIWLVqkfXz37l306dMHXbp0kdzuL7/8gj179kAQBHh7eyMkJARmZmaSsb/99ht2794NIkL79u11trl8+XKcOHEC9vb2WLBgAQDg8ePHWLRoEVJSUuDq6orx48fDxsZGMvbw4cPYtGkTbt68iS+//BLVqlWT3W5UVBTi4+OhVqtRqVIlhISEwNraWjJ2w4YNOH78OARBgL29PUJCQrBhw4ZCcfm2bduG77//Ht988w3s7Owkt7lx40bs3r0bdnZ2AIB+/fqhcePGkrEAsH37duzYsQMmJiZo3LgxBg4cKBm7aNEiJCcnAwAyMjJgZWWFefPmScZeu3YNq1evRk5ODkxMTDBixAhUr15db2xWVhZcXV0xbtw4ZGRkSH7+pfZZVlaWZKzUPpP7u5LaZ5mZmZKxUvtMFEW9f6/P77ecnBzJ2IL7rUuXLti/f7/kNgvus06dOkluU2qfhYaGSsZK7TMHBwe9sc/vM7VaLfk9dPfuXSxevBiPHz9G1apVMXbsWKjVL/k1TkYgNTWVrly5QkREGRkZNG7cOEpKSpKMFUWRMjMziYjo6dOnNHnyZLp48aLstn/++WdavHgxzZkzR28OISEh9ODBA0X5Llu2jGJjY7U5PH78uMjnaDQaGjFiBN29e1dy/f379ykkJISys7OJiGjBggW0d+9eydjr16/ThAkTKCsri3Jzc2nGjBmUnJysXX/27Fm6cuUKTZgwQbssKiqKtmzZQkREW7ZsoaioKNnYpKQkunnzJk2dOpUSExP1bvfUqVOUm5urbUPfdp88eaL996+//kqrVq2SjCMiSklJoVmzZtHo0aO1+0UqNjo6mrZu3VroPZKKTUhIoBkzZlBOTg4REaWnp8vGPu+7776jTZs2ycbOnDmTTpw4QURE8fHxNHXqVNnYSZMm0dmzZ4mIaPfu3bR+/XrZz7/UPpOLldpncrFS+0wuVmqf6ft7Lbjf5GIL7je5OKl9puT7In+fycVK7TO5WKl9Jvc9tGDBAjp48CAREa1atYp27twp+Zl6EUZxOsjR0VF7Rd3S0hKenp5ITU2VjBUEARYWFgAAjUYDjUYDIX+y5wLu37+PEydOoH379iWWa0ZGBs6fP4927doBANRqNaytrYt8XkJCAtzd3eHq6iobI4oicnJyoNFokJOTA0dHR8m4mzdvokaNGjA3N4eJiQnq1KmDo0ePatfXrVsXNjY2Os85duwYAgICAAABAQE4duyYbKyXlxc8PDwKtSsV27BhQ5iYmAAAatasqd1vUrFWVlbaf2dnZ0MQBMk4APjuu+8wYMAAnX0rFytFKvb3339Ht27dYGpqCgCwt7cvcrtEhMOHD6NVq1aysYIgIDMzE0De5yN/v0nFJicno06dOgCA119/HX/++afs519qn8nFSu0zuVipfSYXK7XP9P29FtxvSv+25eKk9llR23x+n8nFSu0zuVipfSb3PXT27Fn4+fkBAAIDA7V/Zy/DKE4HPe/u3bu4evUqqlevLhsjiiJCQ0Nx+/ZtvPnmm6hRo4ZkXGRkJAYOHKjd2UWZPXs2AKBDhw4ICgqSzS//1MT169fhL07RRQAADYJJREFU6+uLIUOGaD8Qcg4dOqT9IpHi5OSErl27YvTo0TAzM0PDhg3RsGFDyVhvb29s2LABjx49gpmZGU6ePKk9ZSPnwYMH2i8nR0dHPHz4UG98cezZswctW7bUG7N+/XrExcXBysoKU6dOlYw5fvw4nJycUKVKFUXt7ty5E3FxcfD19cXgwYNlv9Bv3bqFCxcuYMOGDTA1NcWgQYP0fs4A4Pz587C3t8drr70mGxMcHIzZs2cjKioKoihi1qxZsrHe3t44fvw4mjVrhiNHjuD+/fs665///Be1z5T8rRQVK7XPCsbq22fPxxa1356PvXDhgux+ez4uKipK7z6Tel1y++z52KL22fOxcvus4PdQpUqVYGVlpS2wTk5Osj9mX4RRHAnky8rKwoIFCzBkyBCdXyAFqVQqzJs3DytXrsSVK1fwzz//FIqJj4+Hvb294j67M2fORHh4OKZMmYKdO3fi3LlzknEajQZXr15Fx44dMXfuXJibmyMmJkbvtnNzcxEfH6/9hSDl8ePHOHbsGCIiIrBq1SpkZWUhLi5OMtbLywvdunXDrFmz8OWXX6Jy5cpQqQz7Udm8eTNMTEzQpk0bvXH9+vXDihUr0Lp1a+zYsaPQ+uzsbGzevBl9+/ZV1G7Hjh2xbNkyzJ07F46Ojli3bp1srCiKePz4MWbPno1BgwZh0aJFoCJ6YBdVvIG8I4zg4GCsWLECwcHBWLlypWzs6NGjsXPnToSGhiIzM1PnfLHSz39JxUrtM6lYuX32fKyJiYne/VZwu3L7rWCcvn0m97qk9lnBWH37rGCs3D4r+D108+ZNvfuhuIymCOTm5mLBggVo06YNWrRooeg51tbWqFu3Lk6dOlVo3cWLF3H8+HGMGTMGixcvxpkzZ7B06VLZbTk5OQHIO9xs1qwZEhMTJeOcnZ3h7OysPfrw8/PD1atX9eZ58uRJVK1aFQ4ODrIxCQkJcHNzg52dHdRqNVq0aIFLly7Jxrdr1w7h4eGYPn06bGxs9P5SzX9daWlpAIC0tDTtBbmSsG/fPsTHx2PcuHGyp+YKat26Nf78889Cy+/cuYO7d+/if//7H8aMGYP79+8jNDQU6enpkttxcHCASqWCSqVC+/btceXKFdk2nZyc0KJFCwiCgOrVq0OlUuHRo0ey8RqNBkePHi3y6Gb//v3az6y/v7/sZwcAPD098dlnnyE8PBytWrVCpUqVAEh//uX22Yv8rcjFSu2zorb7/D4rGKtvv0ltV2q/ScXJ7TO5XKX2mVSs3D6TipXbZ/nyv4cuX76MjIwMaDQaAEBqaqr2e+VlGEURICKsXLkSnp6eePvtt/XGPnz4EE+ePAGQ11MoISEBnp6eheL69++PlStXIiIiAh9//DHq16+PcePGSW4zKytLe8ooKysLp0+fho+Pj2Ssg4MDnJ2dtb0QEhIS4OXlpTdnJb8mXVxccPnyZWRnZ4OIZF9XvgcPHgDI61l19OjRIrfftGlT7N+/H0DeH0CzZs30xit16tQpbN26FaGhoTA3N9cbe+vWLe2/jx8/LnndwcfHB9988w0iIiIQEREBZ2dnhIeHyxbQ/C9JADh69Ci8vb1l22/WrBnOnDkDIO/cfG5uLmxtbWXjExIS4OHhAWdnZ72vy8nJSXvkeObMGbi7u8vG5u83URSxefNmdOjQQfbzL7XPXuRvRS5Wap/JxUrtM6lYuf1mb28vud2C+83Ly0syTmqf2djYyL4HBfeZ3OuS2mdysVL7TO57qF69ejhy5AiAvELbtGlTvftICaO4Y/jChQv44osv4OPjo/1Vkt/Vr6Dr168jIiICoiiCiODv749evXrp3f7Zs2fx888/y3YRvXPnDubPnw8g75dE69at8e6778pu79q1a1i5ciVyc3Ph5uaGkJAQ2fPQ2dnZGD16NL766qsiD9s3btyIP/74AyYmJqhSpQpGjRqlvSBW0BdffIFHjx5BrVZj8ODBaNCggXbd4sWLce7c/7d3tyFNvW8cwL+bs3RtLnOajl6YOAvKmVSCpLMsI+xRE6MgMrMVJT0YkRhCvTGjB1MThFqzYYhBKiRGZWZiCkpTLH9hSUjZnDofcKJbbuv3Ijy4djT5JX/8e67PO921e/e83a6d++xc1z8wmUyQSCRITEzExo0bkZOTA6PRCKlUirS0NIhEItZYkUiEBw8eYGRkBEuWLIG/vz8uX77MGlteXs68MAFALpdDpVKxxup0OvT09IDH40EqlUKlUkGr1TrFTZ50B359dffatWvw8PBgHbO9vR1dXV3g8Xjw9vaGSqWCp6cna6xSqWTO5QgEAhw+fBhr165ljY2OjkZBQQHkcjm2b98+499WJpNBo9HAbrfD1dUVKSkpCAgIYI01m814/vw5ACAsLAyHDh1CR0cH6/+/XC53WrPu7m7WWKvV6rRm+/fvZ43VaDROa6ZUKllja2pqnNasr6/vj6/XyXXT6/WssW/fvnVYt+joaNy4ccMpTqFQOK2ZQCCY9vF/X7Pp3luEQqHTmv348YM11mAwOK3Z169fWd+Hent7nb4iOt1reLY4kQQIIYSw48R2ECGEEHaUBAghhMMoCRBCCIdREiCEEA6jJEAIIRxGSYD833j8+PGMF+SR6WVlZaG2tpb1tsnqs5MXIRFu4VztIDK/1dfXo7KyEt+/f4e7uzv8/f0RHx+P1atXz8n4fX19SE1NRUlJCVODZS5YrVaUlZWhvr6eufp2zZo1SEhIgI+Pz1+NfeXKFURGRv5VocKMjIy/mgNZuCgJkHmjsrISFRUVOH78OEJCQiAQCNDa2orm5uY5SwJ/y2azsSaPW7duYXBwEGfOnMHKlSthsVhQV1eHDx8+OFyc9r+cEyGzQUmAzAtjY2MoLS3FqVOnHGq1bNiwgfXS+Pb2duTn5zsU5jp9+jROnDgBhUKBzs5O3L9/Hz09PVi0aBEiIiJw5MgRpkplUlISACAzMxNBQUGoqanB06dPMTw8jMDAQKhUKqYsd2JiIpKTk1FVVQWbzYaCggKHubS1taGtrQ25ubmQSqUAfpW13rFjh8Pze/jwIVpaWsDj8bBlyxYkJiaCz+ejtrYWr169glwux+vXryEUCpGSkoLQ0FCUlJTg48eP+Pz5M4qKirB582YcO3aMdU4dHR0oKiqCXq+HTCZDUlISVq1aBcDxaMJut6O4uBhv3ryBu7v7H8tDkIWNkgCZFz59+oSJiQmEhYXNyXgajQaxsbFQKpUwm81MJdirV68iNTUVRUVFzKfnpqYmlJeX49KlS/Dz80NFRQVyc3Mdyv82NzcjKyuLtRPb+/fvERgYyCQANnfv3sXSpUuRl5cHi8WC7OxseHl5ISYmBgDQ2dmJqKgoqNVqVFdXo7CwEIWFhTh48CA6OjpYt4Omzml0dBTZ2dk4evQoNm3ahMbGRmRnZyMvL8+pflF1dTV0Oh2uX78ONzc3p45rhFvoxDCZF0wmE8Ri8ZxtawgEAhgMBoyMjMDNzY21Peek6upqxMXFYcWKFXBxcUFcXBy6urrQ39/PxMTFxUEkErEmAZPJNG2DHgAYHh5Ga2sr0xdCIpFg586daGhoYGKkUim2bdsGPp+PqKgoDA0NMYXFpjN1TjqdDr6+vlAqlXBxcUFERARkMhnevXvndL/GxkbExsZCKpVCJBJh3759Mz4OWdjoSIDMC2KxGCaTac72t0+ePInS0lKcP38ePj4+SEhIwPr161lj+/v7odFoHHoF/Pz5E4ODg8yW0EyVPsVisUM1zN8ZjUbYbDaoVCqH8aeOObWK6WTlTbPZPONznHr/qXOd5O3tzdp0ZGhoyOGoZaZudGThoyRA5oWgoCC4urqiubl5xuY4kxYvXgyLxcL8bLfbHTpj+fn54dy5c7Db7WhqasLt27ehVqtZ+xFIpVLEx8fP2LBmpj4GwcHBqKqqwsDAAGuy8PLygkAggFqt/k8JbrrHnvr7ZcuWOfVPMBqNWLdundP9PD09YTQaHeIId9F2EJkXhEIhDhw4ALVajaamJlgsFlitVrS0tKC4uNgpXiaTYWJiAjqdDlarFU+ePMHExARze11dHUZGRsDn85kS23w+Hx4eHuDxeOjt7WViY2JiUFFRgW/fvgH4dRK3sbFx1nNXKBRQKBS4efMmvnz5ApvNhvHxcbx48QI1NTXw9PRESEgItFotxsbGYLfbYTAYpu0u9zuJROIwXzahoaHo6elBfX09bDYbGhoa0N3dzVouPTw8HM+ePcPAwABGR0f/2LmOLGx0JEDmjV27dkEikaCsrAz5+flwc3NDQEAAa++FyW/QFBYWwm63Y8+ePQ6fwltbW6HVamGxWODt7Y2zZ88y+/nx8fHIzMyEzWZDRkYGwsLCYDabcefOHRiNRgiFQgQHByM8PHzWc79w4QLKysqQk5OD4eFhiMViKBQKphdFamoqHj16hLS0NIyPj2P58uXYu3fvrMaOjY1FQUEBXr58icjISCQnJzvFiMVipKenQ6PR4N69e/D19UV6ejprh7etW7dCr9fj4sWLcHd3x+7du5nGKoR7qJ8AIYRwGG0HEUIIh1ESIIQQDqMkQAghHEZJgBBCOIySACGEcBglAUII4TBKAoQQwmGUBAghhMP+BepdmrZrFx6fAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 1440x1440 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from sklearn.cluster import KMeans\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "plt.style.use('ggplot')\n",
    "max_number_of_clusters = 30\n",
    "max_number_of_clusters = max_number_of_clusters+1 # since the range method is exclusive, a one should be added.\n",
    "Ks = range(2, max_number_of_clusters, 1) #Hold the range of clusters k (from 2-max_number_of_clusters)\n",
    "    \n",
    "sum_of_squared_distances = [] #a list which will carry the sum of squared distances at each k\n",
    "for k in tqdm(Ks):\n",
    "    sum_of_squared_distances.append(KMeans(n_clusters=k,random_state=30).fit(tf_idf_norm_kmeans).inertia_)#calculate the sum of squared distance.\n",
    "        \n",
    "        \n",
    "f, ax = plt.subplots(1, 1)\n",
    "ax.plot(Ks, sum_of_squared_distances, marker='o')\n",
    "ax.set_xlabel('Cluster Centroid')\n",
    "ax.set_xticks(Ks)\n",
    "ax.set_xticklabels(Ks)\n",
    "ax.set_ylabel('Sum of Squared Distances')\n",
    "ax.set_title('Sum of Squared Distances VS. Clusters')\n",
    "plt.figure(figsize=(20,20))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "clusters = KMeans(n_clusters=16).fit_predict(tf_idf_norm_kmeans)\n",
    "df =FullPaperDataframe.head(100).copy(deep=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "      <th>clusters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>The RNA pseudoknots in foot-and-mouth disease ...</td>\n",
       "      <td>[{'text': 'word count: 194 22 Text word count:...</td>\n",
       "      <td>VP3, and VP0 (which is further processed to VP...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Analysis Title: Regaining perspective on SARS-...</td>\n",
       "      <td>[{'text': 'During the past three months, a new...</td>\n",
       "      <td>In December 2019, a novel coronavirus, SARS-Co...</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Healthcare-resource-adjusted vulnerabilities t...</td>\n",
       "      <td>[]</td>\n",
       "      <td>The 2019-nCoV epidemic has spread across China...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Relationship between Average Daily Temperature...</td>\n",
       "      <td>[{'text': 'The rapid outbreak of the new Coron...</td>\n",
       "      <td>The outbreak of infectious diseases has always...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>CHEER: hierarCHical taxonomic classification f...</td>\n",
       "      <td>[{'text': 'The fast accumulation of viral meta...</td>\n",
       "      <td>Metagenomic sequencing, which allows us to dir...</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               title  \\\n",
       "0  The RNA pseudoknots in foot-and-mouth disease ...   \n",
       "1  Analysis Title: Regaining perspective on SARS-...   \n",
       "2  Healthcare-resource-adjusted vulnerabilities t...   \n",
       "3  Relationship between Average Daily Temperature...   \n",
       "4  CHEER: hierarCHical taxonomic classification f...   \n",
       "\n",
       "                                            abstract  \\\n",
       "0  [{'text': 'word count: 194 22 Text word count:...   \n",
       "1  [{'text': 'During the past three months, a new...   \n",
       "2                                                 []   \n",
       "3  [{'text': 'The rapid outbreak of the new Coron...   \n",
       "4  [{'text': 'The fast accumulation of viral meta...   \n",
       "\n",
       "                                           full_text  clusters  \n",
       "0  VP3, and VP0 (which is further processed to VP...         1  \n",
       "1  In December 2019, a novel coronavirus, SARS-Co...        10  \n",
       "2  The 2019-nCoV epidemic has spread across China...         7  \n",
       "3  The outbreak of infectious diseases has always...         4  \n",
       "4  Metagenomic sequencing, which allows us to dir...        10  "
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['clusters'] = clusters\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "#tf- idf\n",
    "# # finding the tf-idf matrix for all the abstracts\n",
    "# # vec = TfidfVectorizer()\n",
    "\n",
    "# # ve = vec.fit_transform(clean_abs)\n",
    "\n",
    "# # # displaying the tf-idf of the word in the abstact  \n",
    "# # pd.DataFrame(ve.toarray(), columns=sorted(vec.vocabulary_.keys()))\n",
    "\n",
    "# # #applying tf-idf cosine similarity and printing the result(testing)\n",
    "# # query = [\"prophylaxis\"]\n",
    "# # query_tfidf = vec.transform(query)\n",
    "# # cosineSimilarities = cosine_similarity(query_tfidf, ve).flatten()\n",
    "# # print(cosineSimilarities)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# taged_abs=[]\n",
    "# nouns=[]\n",
    "# # tagging the clean data\n",
    "# for val in clean_abs:\n",
    "#     taged_abs.append(pos_tag(val.split()))\n",
    "# # extracting the nouns\n",
    "# for i in range(len(taged_abs)):\n",
    "#     doc=[]\n",
    "#     for j in range(len(taged_abs[i])):\n",
    "#         if(taged_abs[i][j][1]=='NN' or taged_abs[i][j][1]=='NNS'):\n",
    "#             doc.append(taged_abs[i][j][0])\n",
    "#     nouns.append(doc)\n",
    "        \n",
    "# print(nouns)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Above we dcided to calculate the tf-idf so that we can represent every word that is present in the abstract quatitavily. By doing so we can further use the results in order model the topic according to the abstract that we just quatified.Furthermore we will use (Non-negative Matrix Factorization) NMF in order to come up with topic's that carry most weight in the abstract. To accomplish this we are going to filter all the nouns that are avaliable in the abstract and use them to represent the different topis that are avaliable."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
