{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h1> NLP Project </h1>\n",
    "<h2 style =\"color:blue\"> Search Engine and Literature Clustering</h2>\n",
    "<p>This notebook utilizes two major topics in natural language processing to tackle and answer questions related to COVID-19. The first one is: a <i>search engine</i> which allows the user to enter a query and fetches for them the most relevant papers related to that query (max 5). The other topic is <i>literature clustering</i> where PCA algorithm was used to reduce the dimension of the data, followed by K-means which was used to cluster the papers.</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package stopwords to\n",
      "[nltk_data]     C:\\Users\\HP\\AppData\\Roaming\\nltk_data...\n",
      "[nltk_data]   Package stopwords is already up-to-date!\n",
      "[nltk_data] Downloading package punkt to\n",
      "[nltk_data]     C:\\Users\\HP\\AppData\\Roaming\\nltk_data...\n",
      "[nltk_data]   Package punkt is already up-to-date!\n",
      "[nltk_data] Downloading package wordnet to\n",
      "[nltk_data]     C:\\Users\\HP\\AppData\\Roaming\\nltk_data...\n",
      "[nltk_data]   Package wordnet is already up-to-date!\n"
     ]
    }
   ],
   "source": [
    "# basic libraries\n",
    "import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\n",
    "import numpy as np\n",
    "from tqdm import tqdm\n",
    "import re #import the regular expression library\n",
    "import string\n",
    "import os\n",
    "import json\n",
    "import matplotlib.pyplot as plt \n",
    "\n",
    "# sklearn libraries used for tfidf and cosine similarity\n",
    "from sklearn.feature_extraction.text import CountVectorizer, TfidfVectorizer\n",
    "from sklearn.metrics.pairwise import cosine_similarity\n",
    "from sklearn.datasets import load_files\n",
    "\n",
    "# nltk libraries for stop word and puncuation removal.\n",
    "# nltk libraries for word lemmitization\n",
    "import nltk\n",
    "from nltk import pos_tag\n",
    "from nltk.tokenize import RegexpTokenizer\n",
    "from nltk.corpus import stopwords #import the stopwords from the ntlk.corpus library\n",
    "nltk.download('stopwords')\n",
    "from nltk.tokenize import word_tokenize #import the word_tokenize method, which is used to turn sentences into words\"\n",
    "nltk.download('punkt')\n",
    "from collections import Counter\n",
    "from nltk.stem import WordNetLemmatizer, PorterStemmer\n",
    "nltk.download('wordnet')\n",
    "\n",
    "import spacy\n",
    "from spacy.lang.en.stop_words import STOP_WORDS\n",
    "# !pip install scispacy scipy\n",
    "# !pip install https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.2.4/en_core_sci_lg-0.2.4.tar.gz\n",
    "import en_core_sci_lg  \n",
    "plt.style.use('ggplot')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h4>STEP 1: DATA PREPROCESSING</h4>\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<p> Before applying the NLP techniques on the data, data must be preprocessed, cleaned, structured,etc. In the following cell, the folder(s) containing the data in the json format are accessed. Each json file has, then, three majors information extracted from it:<br/>\n",
    " <ol>\n",
    "     <li>The title of the paper</li>\n",
    "     <li>The abstract of the paper</li>\n",
    "     <li>The body/full text of the paper</li>\n",
    "</ol>\n",
    "These are then added to the list which will contain all the papers with those 3 features or attributes. The list is then converted to a dataframe which will be used later in this notebook."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "_cell_guid": "b1076dfc-b9ad-4769-8c92-a6c4dae69d19",
    "_uuid": "8f2839f25d086af736a60e9eeb907d3b93b6e0e5"
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 1934/1934 [00:37<00:00, 51.12it/s]\n"
     ]
    }
   ],
   "source": [
    "FullPaper=[]#a list which will save all the papers \n",
    "directories = [\"biorxiv_medrxiv\"] #the folder name which contains the papers\n",
    "for directory in directories: #for each of the folders carrying the json format of different research papers\n",
    "    for file in tqdm(os.listdir(f\"{directory}/{directory}/pdf_json\")): #for every json file\n",
    "        file_path=f\"{directory}/{directory}/pdf_json/{file}\" #set the file path to the file_path variable \n",
    "        paper = json.load(open(file_path,\"rb\")) #load the json version of the file\n",
    "        title = paper['metadata']['title'] #set the title variable to the file's title\n",
    "        try:\n",
    "            abstract = paper['abstract']#set the abstract variable to the file's abstract\n",
    "        except:\n",
    "            abstarct=\"\"   #set the abstract variable to the an empty string if the file does not have an abstract             \n",
    "        full_text=\"\" #create an empty string which will hold the body of the paper (file)    \n",
    "        \n",
    "        for text in paper['body_text']: #for every word in the body of the file\n",
    "            full_text += (text['text'] +'\\n') #concatenate the word to the variable full_text\n",
    "        FullPaper.append([title,abstract,full_text]) #since the 3 variable: title, abstract, full_text, now have a value, a paper can be created and added to the FullPaper list\n",
    "        \n",
    "FullPaperDataframe=pd.DataFrame(FullPaper,columns=['title','abstract','full_text'])#create a dataframe which will hold all the papers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h5>Data Cleaning</h5>\n",
    "<p>In the below cell, each body text for each paper is stripped off stop words, punctuation, and is lemmatized. The <i>en_core_sci_lg</i> model was used as it works well on biomedical and scientific papers </p>\n",
    "<h6 style = \"color:green\"> PROS Data Cleaning </h6>\n",
    "<p>Data Cleaning is an essential step in producing high quality data to be further used. By removing stopwords, unnessary puncuations and lemmitization we are able to refine the data avaliable in the dataset to produce high quality results in our project.</p>\n",
    "<p>We choose to use lemmitization instead of stemming as stemming removes only the suffixes of the word rather than search for its root in a dictionary which is done using lemmtization.</p>\n",
    "<h6 style = \"color:red\"> CONS associated with data cleaning </h6>\n",
    "<p> Some semantic meaning of the data may be lost during the cleaning process.</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 1934/1934 [1:20:45<00:00,  2.51s/it]\n",
      "100%|██████████| 1934/1934 [2:47:47<00:00,  5.21s/it]  \n"
     ]
    }
   ],
   "source": [
    "full_text = FullPaperDataframe['full_text']\n",
    "\n",
    "clean_text = [] #a list which will hold the clean paper text (after data processing)\n",
    "dirty_text = []# a list which will hold the dirty paper text (before data processing)\n",
    "# Intializing parser lemmtizer\n",
    "lemmatizer = WordNetLemmatizer()\n",
    "\n",
    "# Using the scispacy library to get the biomedical terms \n",
    "parser=en_core_sci_lg.load(disable=[\"tagger\", \"ner\"])\n",
    "parser.max_length = 7000000\n",
    "\n",
    "# extending and customizing the stopwords list (Since we found these words in the text before adding the our_stopwords list)\n",
    "all_stopwords = stopwords.words('english')\n",
    "# 'the','we','it','they',\n",
    "our_stopwords=['copyright','https','et','al','preprint',\n",
    "               'this','these','also','however','although','among','in','medrxiv'\n",
    "                'biorxiv','license','without','fig','figure','doi']\n",
    "all_stopwords.extend(our_stopwords)\n",
    "\n",
    "pattern = \"\"\"!\"#$%&'()*+,.:;<=>?@[\\]^`{|}~\"\"\"\n",
    "\n",
    "#  removing all stop words and extracting puctuations\n",
    "for val in tqdm(full_text):\n",
    "    val = val.lower()\n",
    "    body_tokens = word_tokenize(val)\n",
    "    paper_body_without_stopwords =  [token for token in body_tokens if not token in all_stopwords and nltk.pos_tag([token.lower()])!='PRP']#a list which will hold the tokens stripped off stop words and words that aren't pronouns\n",
    "    dirty_paper_tokens = [token for token in body_tokens] #will hold the tokens as is\n",
    "    clean_string = ' '.join(paper_body_without_stopwords) #convert the list into string\n",
    "    clean_string = re.sub(rf\"[{pattern}]\", '', clean_string) #Strip punctuation from the text\n",
    "    clean_string = re.sub(r'\\b[a-zA-Z]\\b', '', clean_string) # from single letters from the text\n",
    "    clean_string = re.sub(r'^\\d+\\s|\\s\\d+\\s|\\s\\d+$', '', clean_string) #remove digits that aren't associated with a word from the text\n",
    "    clean_text.append(clean_string)\n",
    "    dirty_string=' '.join(dirty_paper_tokens)#convert the list of the dirty tokens into a string\n",
    "    dirty_text.append(dirty_string) #this list will be used later for analysis\n",
    "lemmatized_text=[]    # a list which will carry the lemmatized text\n",
    "\n",
    "# applying the parserer on the dataset\n",
    "for val in tqdm(clean_text):\n",
    "    lemmatized_string=[]\n",
    "    words=val.split(' ')\n",
    "    for word in words:\n",
    "        if(len(word)>1):\n",
    "            token = parser(word)\n",
    "            lemm_token = lemmatizer.lemmatize(token.text)\n",
    "            lemmatized_string.append(lemm_token)\n",
    "    if(len(lemmatized_string)>0):\n",
    "        lemmatized_text.append(lemmatized_string)\n",
    "        \n",
    "for i in range(len(lemmatized_text)):\n",
    "    lemmatized_text[i]=\" \".join(lemmatized_text[i])\n",
    "    \n",
    "#print(lemmatized_text)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h5> Corpus Word Count Graph Before Vs. After Data Cleaning </h5>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<p>To demonstrate how effective the data cleaning is, below is a bar chart comparing the corpus word count before and after the data cleaning.</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "38131765\n",
      "24901716\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAccAAAFRCAYAAAABw2bsAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAgAElEQVR4nO3de7wXVb3/8dfiInjJ69Z0i0oqamppmqblz/AaeTpQJh9vqXRMstJMrY6pKVqanUozNQ3Rg5fSPnkpNI+eOoqXMgtJLfMSJgaiIKIkXlBgfn+s9ZVh8d37+917f/f3u8H38/HYD5g1a2bWzHdmPrPWrJkJRVEgIiIiS/VrdQFERET6GgVHERGRjIKjiIhIRsFRREQko+AoIiKSUXAUERHJKDi+g4UQJoYQftvqcvQF1bZFCOH4EMLMEMKSEMK4lDY6hPBUCGFxCGFiK8q6IlqZt1sIYWgIoQgh7NGg+Z0bQpid5jmmEfNslBDCuBDCtBYtu6nnqz4dHEMI64UQ/iuE8EQI4Y0QwpwQwj0hhCNDCANaXb7uCCEcHUJ4K4Twriz9kU7Sr2puKZcVQtg9hHBTOmDfSCe5a0MIO7WgLEPSSWN4HXknp7xF2rZzQgh3hxBOCiGsmmU/ARhdmrYd+CHwHWBj4PshhP7AlYADm6ZpVkqlE37lb1EIYUYI4cf5PlrHvN4x262WEMKgEMLcEMLrIYS2KuM/BHwDGAtsBPw8hDAhhDC5CWVbLYRwejrnvBZCmBdCeCBdJK7W28uvwzLHaG/rs8ExhDAEmAp8Gjgb2An4CHAF8FVg+x7Me5VGlLGbfgsMAD5aSUgHyXbAc1XStwf+r7sL6+m6hhA+C9wLvAUcDrwXOBiYDlzYk3k3yc+IJ5mhwH7ADcDJwNQQwgaVTEVRzC+K4qXSdJsTj49JRVE8VxTFgjSfNYDbiqJ4tiiK+d0pUAihXwoYK4JRLN1+x6ThH3ZxHo3abgNDCKE70/YhnwaeAe4CxlQZPwxYUhTFr4qieL4oitcbufCOzgchhDWB3wHHA5cAHwZ2Br4PGLB/I8vRHVWO0V5fYJ/8A24BngfWqjJuILB66f/nAc8CbwJ/Aw7L8hfAl4knyvnAL4gHewEcQQw+rwNPA4eXpqvk2SOb3zRgXGn4c8BjwBvAi8A9wJBO1m0acEFp2ICHgYuqpBfAxqW0o9I6LgRmAt8GBpTGTyZeQHyLGGxfSOnrAD8HXgVmp+muAn7bSTnb0zpd1sH4dUr/3xr4NbAg/d0CbFkaPwZYlE0/JK3f8DQ8PA3vl7bha2ldP5b9luW/6Z2UfzIwoUr6EOAl4MpS2sTKtgDGVVnOmCpplXLvDPxvWu8XgJuAzUrzHpd+84OBx4FFwPZp3CHAQ2k7TwfOJ+3b5XUAvkk8Hualsq6erdPBwIMs3Qf/J/t9jk/LfgP4O3Baeb+pso2GUn3f/wHwcJbW4frX2G4HpDIvBOYAP87WfSLxYvL4tG2WAGt0c30CcDnwFPFY/wdwLjCoyu80Ks37VWIQ2yKbl6V8bwC/B0ZW21YdlOPuVHYDnsjGTayyrcZVSRuT8q9BvEB9lnis/Bk4sMpveDhwW1qf73dQrovSdnlPB9tu7fI2ysbvRwysr6ey/DewXmn8TsT9cU7aR/4EjMjmMZ1YCbqQuI/PJgbm/tWO0Wz/GEu84PgX8Ctg/WzeXyGeK18D7iCe8ws6OUcXRdE3gyOwLrAYOL2OvN8jngxGA1sBpxIPon1KeYqU53hgi5SvsuPMSjvP1sSAsQT4YI0TxNvBkXhiWAQcCWwGvI8YLDsLjj8BHikNXwZcAHyqSvpjpeF/S9vlG2kdDiae5L9VyjMZeCVNuy3wvpR+cyr33sRa6rVpZ+osOH6lrp0IVk075/+l7bEz8aQyDVgl5RlD/cHxYWAE8Sr6auBllh6cH0h5DgQ2zA+EbP6TqRIc07iLiRdK/fIDj3jSOTAt5wNpOasDu6S0kSltlbSNFwBnAduk3/8XwJPA4NIJ5TXiiXG39Nu9K22Tl4gH6+bAnsAjwDXZOryc9o9t0nZ5GTirlOezxJr9N1N53k9sgmorLf8Z4v71HmJQ+md5v6myfYaS7fupjI8Cl5bSOl3/tG9U227vJx43FxBbIz6eylRe94nEffRmYMc07wHdXJ9+xOP7Q2ndRhIvHsvbcRwxgNxO3Id3IAacu0t5PkA8R3yHeM44kHhRXTM4pu2zEGgDBhGDwPDS+LXS77YobacNifviT4lBuJK2KjFg3ZX2jz3SbzOWWEHYJ/sNZwKfSXmqBb9+xPNj1WMlyzuOUnAknk9eI55bh6Xf+i7ixW0oHddHpX1lq/Q7vAlsVZrPdOKxcEqaz8FpO3w22x/y4DgfuI7YwvbhtF9cVcpzYJrPCWm+Y4jn/BU2OO6aCn9gjXyrpZ3ti1n6zcCdpeECuKKDg/9bWfrvgWs7OkGk9HJw/FT6gdbswvoZ8QDbIA0/STxY100/ZDn9otJ09wKezesE4hVbJQhNTtP1K+XZMq3HfqW0VYhXeZ0Fxx8D8+tYn6PTAdJWSnt3KteRaXgM9QfH8tXvhintY9WmqVGuyXQcHI9N86ls64kse+BVyjKklLbc/pCmuz6b96C0PT6Zhsel33vTLN904Ngsbc+0jHVK6/BIlucy4P7S8D+Bizs5Rl5j+Sv1I4GXO9l2lXV9jRj83kjDd5Jqb11Y/2rb7Rrgj9l0o9J22qw075ez5XVrfTpYxxOBv5eGxxGPv/VLaYekMlUudK4Ffp/N57h8/TpY3gXAzdnx9bMszxiWP04mAJOztOHpN1krS78S+GW23b9Zo1wbpHwn1bHNxrFscJwMnJfl2TTNb8dO5vMwcFp2LEzK8twOXJfta3lwfIFla/+nAM+Vhn9H6YIrpZ1HHcGxr3ZqqdxXKGrk2xJYZb/99vuWmX3R3Sv3Ie8m1q7K/mhmFwB7AYwYMWLN22+/HeD+LN/vgH26UNbfEJtong4h/IZ48ripKIq5nUxzZ/p3nxDCvcQruruLopgfQniklD4M+Fppuu2ITaNldxOv0LcgNu0CPFgUxZJSnm3Tv7+vJBRF8WYI4U/EK9OO1Ht/Zzvgb+V1LopidgjhiTSuqx4qzef5EMJiYrBtpHr3sVp2AbYMISzI0gcTf7+K2UVR/PPthYewPrGl4fwQwverlGtLYvMTlLZH8izpHlC6b7oJsVmzmu2INY0bQwjlde0PDA4hrF8UxQudrN9niU2f/Ygn228Dk0II+6Z9rN71r1auO7O0u4nrvy2xBgCx5WRBNl231ieEcAyxVWcosSVgAMv3u5iVTf9sKtMGxIuQbVm+D8B9HaxjedmDiQH86FLyROCeEMJ6RVG8WGsemV1IF7jZbdhViM3MZX+sVbz0b3eOhV2A3UIIx1UZNwx4KO3rZxFrmRsSt/tg4v5fVm0/f0+N5T9WFMXCbJryuWJb4u20svycX1VfDY5/J16tbUesBXZq5syZn1977bXHlZICy//Qr7r7iZWB4cOHn0m8CsqV97QlVdIg3ucEoCiKBSGEDxI7C+1LrJH8Vwhhn6IoHqxW3qIo5oYQHiYG4VWAqcXSTgp3ldIXE6/Mlpm8g/KW01/tZJ264glgzRDCkKIoZtbIW+3AKv8OS6qMH1glDWKTS67Rnce2J9ZKunpSyvUj1oLOqzKuPO/8N6mszwnE3zxX3t759ihYfnt0dGKr5BtNbFHIzetguopni6KodN1/MoTwCvEiazgxuNW7/tV0VObO9uVurU8IYTSxo8kpxCD8rzSPc7Ks1bZ1ebnVzi31OIjYMnRDFsz6E5scz+/i/PoRW6x2qTIuX4d8G+ZeIDZpdudCth/wXeI+kHs+/TuRWJv8OrEJ+nXgeuI5rqye/TxXbZr8fNetC+A+GRyLopgXQvgf4LgQwkVF1rsthDCQuGGnAQtnzJixzfbbL+28OnDgwI+vscYaA83sQWITzHIGDx48Mv13N+LN6ordWVoDq1xBtpeWvQGxa3+5vIuJbez3hBDOJHUKIl5xd+T/iD3XBrLsFfRdxJvjA4Ep2bo/SuzNekkpbU+WdjDoyKPp3w8Ta7qVXmu7sHRdq/kF8aR3OjHoLyOEsE4Re489ChwbQmir1B5DCO8m3l+o1IrmAP1DCO8uimJ2SuvOoyCVg6HbvT1TT+jDgV9kNezumEK8f/ZUkdps6pFq1jOArYuiuLy7Cy+KYk4IYSbwMWInqNyjxOa3zYuiuK3K+K5alP6tdO3v1vqzdF8u+yjxRPa3GtN1Z332BP5cFMXbQSiEMLQL05eX/5EsLR+u5vPEIPGDLP0IYi/gzoLjmyy/v08B1iY29/61juV3qCiKJSGEnwFHhxDOKYri6fL41EN4zfw8XCrHdqULqGr2BL5eFMWkNL/Via1lPSp3nf5GPKf/uJS2Wz0T9tlHOYAvEjsZPBhCOCyEsG0IYcsQwmeIP8iwoiheA360YMGCk6dPn75mCGFYCOHUt956a79Bgwad6O47Ex/7WIaZbdavX79N0uDRaf5bhRDOJm7IHwIUsRv174CvhxB2CCHsTOwg8nY1PoQwKoRwYghh5xDCpsAnic1cnR3gEIPjUOI9y3JwvCdN/ymWb775DvDpEMIpqbxGrP3+oCiKarUt0npMAyYBl4QQ9gohbEu8j9Hp82pFUTxLvJ9yTAjh+hDCPun5t51CCGcRe4ZBbLZ4gfhM1k5pO11PbOKoNAP/kdhR6Lz0O40Azuhs+R2YS7wHtn8IYcMQwjo18q+a8rWHEN6fmn8eSGXLm96741xih5JrQwi7hhDek7bxhSGEzWtMexrw5fRs2fYhhK1DCJ8MIfyki2U4C/h8COGbIYT3hhC2CyEcly5WFqQynpvStk7jDwkhfLeOea9b2n67E0/ic1jaRN/d9f8esFMI4fwQwjZpf7gI+Gm5+TnXg/V5AnhfOl63CCGcQOys0VUXALuHEM5Jx+CniI8GdSgdb3sQe0f/tfxH7Jy3TQhhz05m8XTKs10IoS2EMIh4zvgtcFMI4VMhhM3TOej41HzcVacRW+z+EEIYm85370nrdzfpdlQVZwCjQggXhBB2TNt2RAjhirD0WeIngMNDCO8LIexI7EDTrEeZfgAckrbLliGEI4nN21CrRlnrBmwr/4D108o9SbxanEP8oT5D6rYNDBw8ePClgwYNeot4hfXYrrvu+ubo0aMfqvyljfCZynxHjx79n/vuu+9Elj7KMZmlXemPyMqwVVrmq8Sd50CW7ZCzJ3FHfYGl3cpPIfXU6mTdVk/lfRNYLRv3QCrbXlWmO4pY23uTeII/h+Uf5aj2+MJ6xIewX01l/Q41HuUoTbsH8Mu0/RcSa6lXU7rhTuy5dxtLH+W4ldKjHCnPv6WyVy46Pkb1DjlDsukWkbqvp+EjiSeMt6j9KEeR/t4iBta7gZOAVbO8E+lGh5yU/j7ihcJLad2mAeOBddP4cWTd30vTfpJ4D+Q1YlPfQ8AZnf2exJr89CztcGInh4XE5sxfk3r4pvFHs/SRkZfSPvaFTrZdZV0rf0uIzWQ3kx5D6cL6d7Tdyo9yvABcSpVHOTooX1fXZyAxEM1L2/lnpI40pTzL/U7Efb8AhpbSDiE+ErIwLXdUtfUr5a88blH1nEC8t1zpBDiG5TvkrEs8tuaz7KMcqxJbdp4mng+eJ3Zi2buz7V7jnHQmsUb3emm7fol0vHSwjf4fMVC/Qjy/PEasYFTO0e8jXky9TjzHfjHln1iax3SypxPIOiLl+0O1/YMYG4os7cS0/V8nPsoxNm2X9TrbHpWutis0MxsK3Oru25vZmsAT7r5RJ/n//PTTT39rypQpNwL/ryiKmjfURURkxRdCOAM4oSiK9TrL15ebVbvF3f8FPG1mowHMLJjZDpXxZrY1sM7UqVOntqqMIiLS+0J8q9Ip6ZbKFiGEzxGfAKh5n7+pHXLMrD/xfuGz7v6JbNwgYlPdzsRmoYPdfXod87yO2ATWZmYzic0ChwOXmtnpxOaU64lNTgCHAtcvWdLTfhgiItLHFcT4cDKxj8XTxHvW36s1YbN7q55AbI9es8q4o4GX3H1LMzuE2D344FozdPdDOxg1ooP840qDK/p7GkVEpANFUSyig1hQS9OaVc1sCLFDxoQOsowidhCB+HLofcxMwUtERJqumfccf0h8CLSj9syNgRkA7r6I2DOr0xumIiIivaEpzapm9glgjrs/aGbDO8hWrZa4XFdaMxtL7IpLeo5RREQk16OWx2bdc/wIMNLMDiC+U29NM7vW3T9TyjOT+PD7TDMbQHxD/XKvgnL38cRnqACKWbNm9W7JpVNtbW3MndvZa2RF3ll0TLRee3t77Uw1NCU4uvs3SG8jSTXHr2aBEeIbXI4iPhB9EHCnu6/4D2GKiMgKp6XvVjWzs4Ep7j6J+IHea8xsGrHGeEgryyYiIu9cK/obctSs2mJqQhJZlo6J1kvNqj2657jSvSFHRESkpxQcRUREMgqOIiIiGQVHERGRjIKjiIhIRsFRREQko+AoIiKSUXAUERHJtPQNOX3F4mNGtroIK6zZrS7ACqz/5ZNaXQQR6YBqjiIiIhkFRxERkYyCo4iISEbBUUREJKPgKCIiklFwFBERySg4ioiIZBQcRUREMgqOIiIiGQVHERGRjIKjiIhIRsFRREQko+AoIiKSUXAUERHJKDiKiIhkFBxFREQyTfnYsZkNBu4BBqVl3uDuZ2Z5xgDfA55NSRe7+4RmlE9ERKSsKcERWAjs7e4LzGwgcJ+Z/Y+7/yHL93N3P65JZRIREamqKcHR3QtgQRocmP6KZixbRESkq5pVc8TM+gMPAlsCl7j7A1WyfdrM9gSeBE509xnNKp+IiEhF04Kjuy8GdjSztYGbzWx7d/9rKcstwHXuvtDMjgWuAvbO52NmY4GxaZ60tbX1uGyzezwHka5rxL4rfc+AAQP0264EQlE0v3XTzM4EXnX373cwvj8wz93XqjGrYtasWT0uz+JjRvZ4HiJd1f/ySa0ugvSCtrY25s6d2+pivKO1t7cDhJ7MoymPcpjZ+qnGiJmtCuwLPJ7l2ag0OBJ4rBllExERyTWrWXUj4KpUI+wHuLvfamZnA1PcfRLwZTMbCSwC5gFjmlQ2ERGRZbSkWbWB1KwqKyw1q66c1KzaeitMs6qIiMiKRMFRREQko+AoIiKSUXAUERHJKDiKiIhkFBxFREQyCo4iIiIZBUcREZGMgqOIiEhGwVFERCSj4CgiIpJRcBQREckoOIqIiGQUHEVERDIKjiIiIhkFRxERkYyCo4iISEbBUUREJKPgKCIiklFwFBERySg4ioiIZBQcRUREMgqOIiIiGQVHERGRjIKjiIhIZkAzFmJmg4F7gEFpmTe4+5lZnkHA1cDOwIvAwe4+vRnlExERKWtWzXEhsLe77wDsCIwws92yPEcDL7n7lsAFwHebVDYREZFlNCU4unvh7gvS4MD0V2TZRgFXpf/fAOxjZqEZ5RMRESlr2j1HM+tvZg8Bc4DfuPsDWZaNgRkA7r4ImA+s16zyiYiIVDTlniOAuy8GdjSztYGbzWx7d/9rKUu1WmJeu8TMxgJj0zxpa2vrcdlm93gOIl3XiH1X+p4BAwbot10JNC04Vrj7y2Y2GRgBlIPjTGATYKaZDQDWAuZVmX48MD4NFnPnzu3dAov0Eu27K6e2tjb9ti3W3t7e43k0pVnVzNZPNUbMbFVgX+DxLNsk4Kj0/4OAO919uZqjiIhIb2tWzXEj4Coz608MyO7ut5rZ2cAUd58EXAFcY2bTiDXGQ5pUNhERkWWEolihK2fFrFmzejyTxceMbEBRRLqm/+WTWl0E6QVqVm291Kzao6cd9IYcERGRjIKjiIhIRsFRREQko+AoIiKSUXAUERHJKDiKiIhkFBxFREQyCo4iIiIZBUcREZGMgqOIiEhGwVFERCSj4CgiIpJRcBQREckoOIqIiGQUHEVERDIKjiIiIhkFRxERkYyCo4iISEbBUUREJKPgKCIiklFwFBERySg4ioiIZBQcRUREMgqOIiIiGQVHERGRzIBmLMTMNgGuBjYElgDj3f3CLM9w4FfA0ynpJnc/uxnlExERKWtKcAQWASe7+1QzexfwoJn9xt3/luW7190/0aQyiYiIVNWUZlV3f87dp6b/vwI8BmzcjGWLiIh0VbNqjm8zs6HAB4AHqoze3cweBmYBX3X3R5tZNhEREagzOJrZPHdft0r6HHffoN6FmdkawI3AV9z9X9noqcBm7r7AzA4AfgkMqzKPscBYAHenra2t3sV3aHaP5yDSdY3Yd6XvGTBggH7blUAoiqJmJjN7xd3flaUNBJ539/XqWVDKfytwh7ufX0f+6cAH3X1uJ9mKWbNm1bP4Ti0+ZmSP5yHSVf0vn9TqInRIx4S0QqOOifb2doDQk3l0WnM0s3uBAhhsZvdko4cAv69nIWYWgCuAxzoKjGa2ITDb3Qsz25V4P/TFeuYvIiLSSLWaVScQo+8uxOBWURBbI++sczkfAY4A/mJmD6W0U4FNAdz9MuAg4Atmtgh4HTjE3WtXa0VERBqs3mbVbdz98SaUp6vUrCorLDWriixrhWlWrXD3x81sf2BHYI1s3Bk9KYCIiEhfU29v1YsBA+4CXiuNUrOniIisdOp9zvFQYEd3n9GbhREREekL6n1DzovAy71ZEBERkb6i3prjD4Cfmtl3yJ6Zd/d/NLxUIiIiLVRvcLw0/Zu/FLwA+jeuOCIiIq1Xb29VffdRRETeMRT0REREMvU+ylF5jdxy3H3PhpZIRESkxeq95zghG94QOBq4trHFERERab167zlelaeZ2Y3AfwNnN7pQIiIirdSTe47PAu9vVEFERET6inrvOf5HlrQacCDwh4aXSEREpMXqved4RDb8KvFbjhc0tjgiIiKtV+89x716uyAiIiJ9Rb01R8xsGPEF5BsT7zde5+5/762CiYiItEpdHXLM7N+BB4FtgHnA1sAUM9MXUUVEZKVTb83xXGCUu99VSTCz4cDFQN/9nLmIiEg31PsoxxDg3iztvpQuIiKyUqk3OD4EnJylnZTSRUREVir1Nqt+AbjFzE4AZgCbEB/n0D1HERFZ6dRVc3T3x4H3Akb88LEB27r7Y71YNhERkZbotOZoZusCu7r77e6+iHifsTJuhJk94O4v9XYhRUREmqlWzfF0YOcOxn0AOK2xxREREWm9WsHxE8BPOhg3HhjV2OKIiIi0Xq0OORu6+9wOxs0D3l3PQsxsE+Bq4ncglwDj3f3CLE8ALgQOAF4Dxrj71HrmLyIi0ki1ao4vmdnWHYzbCni5zuUsAk529/cCuwFfMrNtszwfB4alv7HApXXOW0REpKFqBcebgR+Z2arlxDR8AXBDPQtx9+cqtUB3fwV4jPiO1rJRwNXuXrj7H4C1zWyjeuYvIiLSSLWaVb8J3An8w8xuB54DNgI+Rnze8cyuLtDMhhI78zyQjdo4zbNiZkp7rqvLEBER6YlOg6O7v2JmHwaOAvYBPgi8SAya17j7m11ZmJmtAdwIfMXd/5WNDlUmKarMYyyx2RV3p62trStFqGp2j+cg0nWN2Hd7i44JaYW+dEzUfEOOu78FTEh/3WZmA4mB8afuflOVLDOJb96pGALMqlKe8cSesgDF3Lkd9RcS6du074osq1HHRHt7e4/nUff3HHsi9US9AnjM3c/vINsk4Dgzux74EDDf3dWkKiIiTdeU4Ah8BDgC+IuZVV5WfiqwKYC7XwbcRnyMYxrxUY7PNqlsIiIiy2hKcHT3+6h+T7GcpwC+1IzyiIiIdKbeT1aJiIi8Y3RYczSza6jSWzTn7kc2tEQiIiIt1lnNcRrwVPqbD3wS6E/sVdqP+NB+vW/IERERWWF0WHN097Mq/zezO4B/c/d7S2l7EJ93FBERWanUe89xN+APWdoDwO6NLY6IiEjr1Rsc/wycW3nHavr3HOChTqcSERFZAdUbHMcQn1Wcb2azifcg9wDUGUdERFY6NZ9zTG+36QfsSXzpeDvwnLv/s5fLJiIi0hL1vFu1MLO/AO9y9xks++UMERGRlU5X7jlu1ZsFERER6SvqfX3cZOB2M5tIrDm+/XIAd7+y8cUSERFpnXqD40eAp4GPZukFoOAoIiIrlbqCo7vv1dsFERER6Svq/iqHma0D/DuwMfAscIu7v9RbBRMREWmVujrkmNnuxHesHgu8H/g88FRKFxERWanUW3P8IfBFd7++kmBmBwM/AnbpjYKJiIi0Sr2PcmwFeJZ2A7BlY4sjIiLSevUGx78Dh2Rpo4lNrSIiIiuVeptVvwLcamZfBp4BhgLDgE/0UrlERERapq6ao7v/HtgCuBh4ELgI2DKli4iIrFQ6rTmamQH3uPvz6bGNa5tTLBERkdap1az6bWALM3sKuAe4mxgsn+n1komIiLRIp82q7r4V8RNVpwGvAycTn298xsyuMbPPNaGMIiIiTVXPJ6tmA79If5jZ2sBY4CTgMGBCbxZQRESk2er92PGOxI8d7wl8GJhFfO7x3l4tnYiISAvU6pBzK7AT8ARwHzAeGOPur3RlIWZ2JfGxjznuvn2V8cOBXxG//AFwk7uf3ZVliIiINEqtmuPWwEJi0HoKmNbVwJhMJD4GcnUnee51dz03KSIiLddpcHT3YWb2bpY2qX7FzNqA3xGbVO9z94dqLcTd7zGzoQ0or4iISK/rSYec04H1gf4NKsvuZvYw8X7mV9390QbNV0REpEu60yFnD2BtYApwZYPKMRXYzN0XmNkBwC+Jr6erVp6xxOCMu9PW1tbjhc/u8RxEuq4R+25v0TEhrdCXjolQFEWHI83s18TeqasAD5BeAgDc7+5vdGVBqVn11modcqrknQ580N3n1shazJo1qyvFqGrxMSN7PA+Rrup/+aRWF6FDOiakFRp1TLS3twOEnsyjVs3xXuAc4E/u/lZPFtQZM9sQmO3uhZntSnw5wYu9tTwREZHO1OqQc14jFmJm1wHDgTYzmwmcCQxMy7gMOAj4gpktIr6J5xB377hKKyIi0ovq/WRVj7j7oTXGX0x81ENERKTl6v3YsYiIyDuGgqOIiEhGwVFERCSj4CgiIpJRcBQREckoOCUylpsAAAvlSURBVIqIiGQUHEVERDIKjiIiIhkFRxERkYyCo4iISEbBUUREJKPgKCIiklFwFBERySg4ioiIZBQcRUREMgqOIiIiGQVHERGRjIKjiIhIRsFRREQko+AoIiKSUXAUERHJKDiKiIhkFBxFREQyCo4iIiIZBUcREZHMgGYsxMyuBD4BzHH37auMD8CFwAHAa8AYd5/ajLKJiIjkmlVznAiM6GT8x4Fh6W8scGkTyiQiIlJVU4Kju98DzOskyyjgancv3P0PwNpmtlEzyiYiIpLrK/ccNwZmlIZnpjQREZGma8o9xzqEKmlFtYxmNpbY9Iq709bW1uOFz+7xHES6rhH7bm/RMSGt0JeOib4SHGcCm5SGhwCzqmV09/HA+DRYzJ07t5eLJtI7tO+KLKtRx0R7e3uP59FXguMk4Dgzux74EDDf3Z9rcZlEROQdqlmPclwHDAfazGwmcCYwEMDdLwNuIz7GMY34KMdnm1EuERGRapoSHN390BrjC+BLzSiLiIhILX2lt6qIiEifoeAoIiKSUXAUERHJKDiKiIhkFBxFREQyCo4iIiIZBUcREZGMgqOIiEhGwVFERCSj4CgiIpJRcBQREckoOIqIiGQUHEVERDIKjiIiIhkFRxERkYyCo4iISEbBUUREJKPgKCIiklFwFBERySg4ioiIZBQcRUREMgqOIiIiGQVHERGRjIKjiIhIRsFRREQkM6BZCzKzEcCFQH9ggrufl40fA3wPeDYlXezuE5pVPhERkYqmBEcz6w9cAuwHzAT+ZGaT3P1vWdafu/txzSiTiIhIR5rVrLorMM3d/+HubwLXA6OatGwREZEuaVaz6sbAjNLwTOBDVfJ92sz2BJ4ETnT3GVXyiIiI9KpmBcdQJa3Ihm8BrnP3hWZ2LHAVsHc+kZmNBcYCuDttbW09LtzsHs9BpOsase/2Fh0T0gp96ZhoVnCcCWxSGh4CzCpncPcXS4OXA9+tNiN3Hw+MT4PF3LlzG1hMkebRviuyrEYdE+3t7T2eR7PuOf4JGGZm7zGzVYBDgEnlDGa2UWlwJPBYk8omIiKyjKbUHN19kZkdB9xBfJTjSnd/1MzOBqa4+yTgy2Y2ElgEzAPGNKNsIiIiuVAU+a2/FUoxa9as2rlqWHzMyAYURaRr+l8+qXamFtExIa3QqGMiNatW6+tSN70hR0REJKPgKCIiklFwFBERySg4ioiIZBQcRUREMgqOIiIiGQVHERGRjIKjiIhIRsFRREQko+AoIiKSUXAUERHJKDiKiIhkFBxFREQyCo4iIiIZBUcREZGMgqOIiEhGwVFERCSj4CgiIpJRcBQREckoOIqIiGQUHEVERDIKjiIiIhkFRxERkYyCo4iISEbBUUREJDOgWQsysxHAhUB/YIK7n5eNHwRcDewMvAgc7O7Tm1U+ERGRiqbUHM2sP3AJ8HFgW+BQM9s2y3Y08JK7bwlcAHy3GWUTERHJNatZdVdgmrv/w93fBK4HRmV5RgFXpf/fAOxjZqFJ5RMREXlbs4LjxsCM0vDMlFY1j7svAuYD6zWldCIiIiXNuudYrQZYdCMPZjYWGAvg7rS3t/e8dL+e0vN5iKxMdEzIO1yzguNMYJPS8BBgVgd5ZprZAGAtYF4+I3cfD4zvpXJKF5nZFHf/YKvLIdJX6JhYOTQrOP4JGGZm7wGeBQ4BDsvyTAKOAu4HDgLudPflao4iIiK9rSn3HNM9xOOAO4DHYpI/amZnm9nIlO0KYD0zmwacBJzSjLKJiIjkmvaco7vfBtyWpZ1R+v8bwOhmlUcaRk3cIsvSMbESCEWhlksREZEyvT5OREQk07RmVembzGwx8BfiozSLgePc/fc1pvky8AVgqrsf3vulFGk9M/sUcBPwXnd/PKV9DziAeMvod8CT7v631pVSGkU1R3nd3Xd09x2AbwDfqWOaLwIH1BsY06M5Iiu6Q4H7iL3tKz4P7OTuXwM+SXw9Zt10bPRd+mGkbE3gpcqAmX0NMGAQcLO7n2lmlwGbA5PM7EriK/+uTGmvAWPd/REzGwe0A0OBuWZ2BHAeMDzN7xJ3/0mT1kukR8xsDeAjwF7Ex87GmdkkYHXgATO7GRgJfNTMTgc+nSa9BFifeGwc4+6Pm9lE4jPcHwCmAic3c12kPqo5yqpm9pCZPQ5MAL4FYGb7A8OI78XdEdjZzPZ092OJL3DYy90vAM4C/uzu7wdOJX5ZpWJnYJS7H0Z8sfx8d98F2AU4Jj33KrIi+CRwu7s/Ccwzs53cfSRLW17OIgbNr6Xhp4i9Vo93952BrwI/Ls1vK2Bfd1dg7KMUHKVycG8DjACuTi983z/9/Zl4dbsNMVjm9gCuAXD3O4nPqq6Vxk1y99fT//cHjjSzh4AHiO/NrTY/kb7oUOIHE0j/HtpZ5lTT/DDwi7TP/wTYqJTlF+6+uDcKKo2hZlV5m7vfb2ZtxGagAHynjqbPzt6J+2qW73h3v6PnJRVpHjNbD9gb2N7MCuI3aQsz+3onk/UDXnb3HTsY/2oH6dJHqOYobzOzbYgH/ovEtxn9R7oCxsw2NrMNqkx2D3B4yjMcmOvu/6qS7w7gC2Y2MOXdysxWb/xaiDTcQcDV7r6Zuw91902Ap4mtJmWvAO8CSMfA02Y2GsDMgpnt0MxCS8+o5iirpmYfiLW7o1Jzz/+a2XuB+80MYAHwGWBONv044L/N7BFip4OjOljOBGLnnKmp2fYF4n0ckb7uUGJnsrIbWf790NcDl6dHnQ4iXjRemjroDEzjH+7lskqD6A05IiIiGTWrioiIZBQcRUREMgqOIiIiGQVHERGRjIKjiIhIRsFRREQko+ccRXrAzA4DTiK+Xu8V4CHgHHe/r6UF64CZPQpslgZXBd4CFqXhc9393B7M+3ngoL667iJdoeAo0k1mdhJwCnAs8Q1AbxLfTzuK+GmjrsxrgLsvqp2zZ9x9u9IyJwPXuvuE3l6uyIpGwVGkG9LL1c8GPuvuN5VG3ZL+MLNBwHeJn/0CcOA/3X1hetXetcBFwInAb8zsipT2Y2JtdAFwmrv/NM1vMqVgZmZjgM+5+x7prUPnE9/KMgh4BjjM3f/ajXX7fFr++sD9xM+QPWtmewE/B3Zw9+fMbBfgf4lfX/kusAHxzUpLgFPd/UddXbZIX6F7jiLdszswGLi5kzynAbsRP/m1A/HzX6eXxm8IrEts5hxbSmsDNia+im+8mW1dR3n2B/YkfgppbeBg4jtyu8TMDgG+Avw78G7iV1muBXD3u4hfYLnSzFZL//+au//D3UcTXy24v7uvocAoKzoFR5HuWY/4kvXOmkIPB8529znu/gLx25dHlMYvAc5094WlT3sBfDOl3Q38mqU1z868RXzp9TZAcPfH3P25rqxQ8nng2+7+pLu/lcq8h5m9O40/FRhC/OzYE2qSlZWVgqNI97wItJlZZ7cm2onNmxXPpLSKF9z9jWyal9z91U6mqSp9S/Ni4pfnZ5vZeDNbs9Z0VWwGXGZmL5vZy8QXxC8iBkTcfSHxg9bbA9/vxvxFVgi65yjSPfcDbxC/LHJDB3lmEYPNo2l405RWUe2t/+uY2eqlALkpULlv+CqwWinvhuUJU1Pmj9KnxRz4GvDNutZmqRnEptIbq400s6HAN4CrgAvMbLdS7VlfMZCVhoKjSDe4+3wzOwO4xMwWETumvAXsC+zl7l8HrgNON7M/EQPHGaT7dzWcZWanAh8CPgGcmdIfAg40swnE2uTRwGyA1DmmHzCVGETfALrzpfnLUpn/6u5PmNk6wN7ufqOZ9SPWGi8iNrfemdbpjDTtbGBzuthTV6QvUrOqSDe5+/nEXp2nE5sfZwDHAb9MWb4NTAEeAf5CDFzfrjHb54GXiDXMnwLHuvvjadwFxMdFZhNrbj8tTbcmcHma9hlis2+Xmz3d/Tpi8+xNZvYvYkDeL43+GvHZyG+5+xJih6EvmdmH0vhzgHNSk+xxXV22SF+i7zmK9BGVxzvcfUiryyLyTqeao4iISEbBUUREJKNmVRERkYxqjiIiIhkFRxERkYyCo4iISEbBUUREJKPgKCIiklFwFBERyfx/fxrzX627x8UAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 504x360 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "count1=0\n",
    "count2 =0\n",
    "\n",
    "for d in dirty_text: #get the word count in all the corpus before preprocessing\n",
    "    count1 +=len(d)\n",
    "\n",
    "for st in lemmatized_text:#get the word count in all the corpus after preprocessing\n",
    "    count2+=len(st)\n",
    "labels_x=[\"Before\", \"After\"]\n",
    "values = [count1, count2]\n",
    "print(count1)\n",
    "print(count2)\n",
    "plt.figure(figsize=(7, 5))\n",
    "plt.bar(labels_x, values)\n",
    "plt.xlabel('Corpus Text')\n",
    "plt.ylabel('Word Count')\n",
    "plt.title('Corpus Word Count Difference Before and After Cleaning')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h4>STEP 2: SEARCH ENGINE IMPLEMENTATION</h4>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h5>TF-IDF</h5>\n",
    "<p> After the data has been preprocessed, a TF-IDF matrix is to be created. This is a matrix which will hold the values of each word (bag of words of all the corpus) with respect to the document (paper).</p> \n",
    "<h6 style = \"color:green\"> PROS of using tf-idf</h6>\n",
    "<p>tf-idf matrix considers the unique descriptives words in a corpus and gives them high values. On the contrary, it gives low values to reptitive words in the corpus. Hence, it helps to achieve accurate results for a search engine. </p>\n",
    "<h6 style = \"color:red\"> CONS of using tf-idf</h6>\n",
    "<p>tf-idf matrix does not account for semantics at all, unlike word embeddings, for example.</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[('05', 0), ('10', 1), ('100', 2), ('11', 3), ('12', 4), ('13', 5), ('14', 6), ('15', 7), ('16', 8), ('17', 9), ('18', 10), ('19', 11), ('1a', 12), ('1b', 13), ('1c', 14), ('1d', 15), ('1e', 16), ('20', 17), ('2019', 18), ('2020', 19), ('202003', 20), ('21', 21), ('22', 22), ('23', 23), ('24', 24), ('25', 25), ('27', 26), ('28', 27), ('29', 28), ('2a', 29), ('2b', 30), ('2c', 31), ('2d', 32), ('2e', 33), ('30', 34), ('31', 35), ('35', 36), ('37', 37), ('3a', 38), ('3b', 39), ('3c', 40), ('3d', 41), ('3e', 42), ('40', 43), ('4a', 44), ('4b', 45), ('4c', 46), ('4d', 47), ('50', 48), ('5a', 49), ('5b', 50), ('5c', 51), ('60', 52), ('6a', 53), ('6b', 54), ('70', 55), ('80', 56), ('96', 57), ('ability', 58), ('able', 59), ('abnormal', 60), ('abnormality', 61), ('absence', 62), ('absent', 63), ('absolute', 64), ('abundance', 65), ('abundant', 66), ('access', 67), ('accessible', 68), ('accession', 69), ('accordance', 70), ('according', 71), ('accordingly', 72), ('account', 73), ('accounted', 74), ('accounting', 75), ('accumulation', 76), ('accuracy', 77), ('accurate', 78), ('accurately', 79), ('ace2', 80), ('achieve', 81), ('achieved', 82), ('acid', 83), ('acquired', 84), ('across', 85), ('act', 86), ('action', 87), ('activated', 88), ('activation', 89), ('active', 90), ('activity', 91), ('actual', 92), ('actually', 93), ('acute', 94), ('adapted', 95), ('adaptive', 96), ('add', 97), ('added', 98), ('adding', 99), ('addition', 100), ('additional', 101), ('additionally', 102), ('address', 103), ('addressed', 104), ('adequate', 105), ('adjacent', 106), ('adjust', 107), ('adjusted', 108), ('adjustment', 109), ('administered', 110), ('administration', 111), ('admission', 112), ('admitted', 113), ('adopt', 114), ('adopted', 115), ('adult', 116), ('advance', 117), ('advanced', 118), ('advantage', 119), ('adverse', 120), ('affect', 121), ('affected', 122), ('affecting', 123), ('affinity', 124), ('age', 125), ('aged', 126), ('agency', 127), ('agent', 128), ('aggregate', 129), ('agreement', 130), ('aid', 131), ('aim', 132), ('aimed', 133), ('air', 134), ('al', 135), ('algorithm', 136), ('aligned', 137), ('alignment', 138), ('allow', 139), ('allowed', 140), ('allowing', 141), ('allows', 142), ('almost', 143), ('alone', 144), ('along', 145), ('alpha', 146), ('already', 147), ('alter', 148), ('altered', 149), ('alternative', 150), ('alternatively', 151), ('alveolar', 152), ('always', 153), ('amino', 154), ('amount', 155), ('amplification', 156), ('amplified', 157), ('analysed', 158), ('analysis', 159), ('analytical', 160), ('analyze', 161), ('analyzed', 162), ('analyzing', 163), ('and', 164), ('angiotensin', 165), ('animal', 166), ('annotated', 167), ('annotation', 168), ('announced', 169), ('another', 170), ('anova', 171), ('answer', 172), ('anti', 173), ('antibiotic', 174), ('antibody', 175), ('antigen', 176), ('antiviral', 177), ('apart', 178), ('apparent', 179), ('appear', 180), ('appeared', 181), ('appears', 182), ('appendix', 183), ('applicable', 184), ('application', 185), ('applied', 186), ('apply', 187), ('applying', 188), ('approach', 189), ('appropriate', 190), ('approval', 191), ('approved', 192), ('approximate', 193), ('approximately', 194), ('approximation', 195), ('april', 196), ('ards', 197), ('area', 198), ('around', 199), ('article', 200), ('asked', 201), ('aspect', 202), ('ass', 203), ('assay', 204), ('assembled', 205), ('assembly', 206), ('assessed', 207), ('assessing', 208), ('assessment', 209), ('assigned', 210), ('associated', 211), ('association', 212), ('assume', 213), ('assumed', 214), ('assumes', 215), ('assuming', 216), ('assumption', 217), ('asymptomatic', 218), ('attack', 219), ('attempt', 220), ('attention', 221), ('attributed', 222), ('authority', 223), ('automated', 224), ('availability', 225), ('average', 226), ('avoid', 227), ('awareness', 228), ('away', 229), ('axis', 230), ('back', 231), ('background', 232), ('bacteria', 233), ('bacterial', 234), ('balance', 235), ('band', 236), ('bar', 237), ('barrier', 238), ('base', 239), ('based', 240), ('baseline', 241), ('basic', 242), ('basis', 243), ('bat', 244), ('bayesian', 245), ('became', 246), ('become', 247), ('becomes', 248), ('becoming', 249), ('bed', 250), ('began', 251), ('begin', 252), ('beginning', 253), ('behavior', 254), ('behaviour', 255), ('behind', 256), ('believe', 257), ('believed', 258), ('beneficial', 259), ('benefit', 260), ('besides', 261), ('best', 262), ('beta', 263), ('better', 264), ('beyond', 265), ('bias', 266), ('biased', 267), ('bilateral', 268), ('bind', 269), ('binding', 270), ('bio', 271), ('biological', 272), ('biology', 273), ('biorxiv', 274), ('black', 275), ('block', 276), ('blocking', 277), ('blood', 278), ('blue', 279), ('board', 280), ('body', 281), ('border', 282), ('bottom', 283), ('bound', 284), ('boundary', 285), ('box', 286), ('breath', 287), ('brief', 288), ('briefly', 289), ('bring', 290), ('broad', 291), ('broadly', 292), ('buffer', 293), ('build', 294), ('building', 295), ('built', 296), ('burden', 297), ('by', 298), ('byinternational', 299), ('ca', 300), ('calculate', 301), ('calculated', 302), ('calculating', 303), ('calculation', 304), ('call', 305), ('called', 306), ('cancer', 307), ('candidate', 308), ('capability', 309), ('capable', 310), ('capacity', 311), ('capture', 312), ('captured', 313), ('cardiac', 314), ('cardiovascular', 315), ('care', 316), ('carefully', 317), ('carried', 318), ('carrier', 319), ('carry', 320), ('carrying', 321), ('case', 322), ('categorical', 323), ('category', 324), ('cause', 325), ('caused', 326), ('causing', 327), ('caution', 328), ('cc', 329), ('cdc', 330), ('cdna', 331), ('cell', 332), ('cellular', 333), ('center', 334), ('central', 335), ('centre', 336), ('certain', 337), ('chain', 338), ('challenge', 339), ('challenging', 340), ('chance', 341), ('change', 342), ('changed', 343), ('changing', 344), ('channel', 345), ('characteristic', 346), ('characterization', 347), ('characterize', 348), ('characterized', 349), ('check', 350), ('checked', 351), ('chemical', 352), ('chest', 353), ('chi', 354), ('child', 355), ('china', 356), ('chinese', 357), ('choice', 358), ('choose', 359), ('chose', 360), ('chosen', 361), ('chronic', 362), ('ci', 363), ('circle', 364), ('circulating', 365), ('circumstance', 366), ('city', 367), ('class', 368), ('classical', 369), ('classification', 370), ('classified', 371), ('clear', 372), ('clearly', 373), ('cleavage', 374), ('clinic', 375), ('clinical', 376), ('clinically', 377), ('clinician', 378), ('close', 379), ('closed', 380), ('closely', 381), ('closer', 382), ('closure', 383), ('cluster', 384), ('clustered', 385), ('clustering', 386), ('co', 387), ('code', 388), ('coding', 389), ('codon', 390), ('coefficient', 391), ('cohort', 392), ('cold', 393), ('collect', 394), ('collected', 395), ('collection', 396), ('color', 397), ('column', 398), ('combination', 399), ('combine', 400), ('combined', 401), ('combining', 402), ('come', 403), ('coming', 404), ('commercial', 405), ('commission', 406), ('committee', 407), ('common', 408), ('commonly', 409), ('communication', 410), ('community', 411), ('comorbidities', 412), ('comparable', 413), ('comparative', 414), ('compare', 415), ('compared', 416), ('comparing', 417), ('comparison', 418), ('compartment', 419), ('competing', 420), ('complement', 421), ('complementary', 422), ('complete', 423), ('completed', 424), ('completely', 425), ('complex', 426), ('complexity', 427), ('complicated', 428), ('complication', 429), ('component', 430), ('composed', 431), ('composition', 432), ('compound', 433), ('comprehensive', 434), ('comprised', 435), ('computational', 436), ('compute', 437), ('computed', 438), ('computer', 439), ('computing', 440), ('concentrated', 441), ('concentration', 442), ('concept', 443), ('concern', 444), ('conclude', 445), ('conclusion', 446), ('condition', 447), ('conduct', 448), ('conducted', 449), ('confidence', 450), ('confirm', 451), ('confirmation', 452), ('confirmed', 453), ('conflict', 454), ('conformation', 455), ('connected', 456), ('connection', 457), ('consecutive', 458), ('consensus', 459), ('consent', 460), ('consequence', 461), ('consequently', 462), ('conservation', 463), ('conserved', 464), ('consider', 465), ('considerable', 466), ('considerably', 467), ('consideration', 468), ('considered', 469), ('considering', 470), ('consistent', 471), ('consistently', 472), ('consisting', 473), ('consists', 474), ('constant', 475), ('constraint', 476), ('construct', 477), ('constructed', 478), ('construction', 479), ('contact', 480), ('contagious', 481), ('contain', 482), ('contained', 483), ('containing', 484), ('containment', 485), ('contains', 486), ('content', 487), ('context', 488), ('continue', 489), ('continued', 490), ('continues', 491), ('continuous', 492), ('continuously', 493), ('contrast', 494), ('contribute', 495), ('contributed', 496), ('contributes', 497), ('contributing', 498), ('contribution', 499), ('control', 500), ('controlled', 501), ('controlling', 502), ('conventional', 503), ('conversely', 504), ('converted', 505), ('converting', 506), ('coordinate', 507), ('copy', 508), ('core', 509), ('corona', 510), ('coronavirus', 511), ('coronaviruses', 512), ('correct', 513), ('correction', 514), ('correctly', 515), ('correlate', 516), ('correlated', 517), ('correlation', 518), ('correspond', 519), ('corresponding', 520), ('corresponds', 521), ('cost', 522), ('cough', 523), ('could', 524), ('count', 525), ('counted', 526), ('country', 527), ('coupled', 528), ('course', 529), ('cov', 530), ('cover', 531), ('coverage', 532), ('covered', 533), ('covid', 534), ('create', 535), ('created', 536), ('crisis', 537), ('criterion', 538), ('critical', 539), ('critically', 540), ('cross', 541), ('crucial', 542), ('crystal', 543), ('ct', 544), ('culture', 545), ('cumulative', 546), ('current', 547), ('currently', 548), ('curve', 549), ('custom', 550), ('cut', 551), ('cutoff', 552), ('cycle', 553), ('cytokine', 554), ('daily', 555), ('damage', 556), ('dark', 557), ('dashed', 558), ('database', 559), ('dataset', 560), ('datasets', 561), ('date', 562), ('day', 563), ('de', 564), ('dead', 565), ('death', 566), ('decade', 567), ('december', 568), ('decision', 569), ('declare', 570), ('declared', 571), ('decline', 572), ('decrease', 573), ('decreased', 574), ('decreasing', 575), ('deep', 576), ('default', 577), ('define', 578), ('defined', 579), ('defining', 580), ('definition', 581), ('degree', 582), ('delay', 583), ('delayed', 584), ('deletion', 585), ('delivery', 586), ('demand', 587), ('demographic', 588), ('demonstrate', 589), ('demonstrated', 590), ('demonstrates', 591), ('demonstrating', 592), ('denote', 593), ('denoted', 594), ('denotes', 595), ('density', 596), ('department', 597), ('depend', 598), ('dependence', 599), ('dependent', 600), ('depending', 601), ('depends', 602), ('depth', 603), ('derive', 604), ('derived', 605), ('describe', 606), ('described', 607), ('describes', 608), ('describing', 609), ('description', 610), ('descriptive', 611), ('design', 612), ('designated', 613), ('designed', 614), ('despite', 615), ('detail', 616), ('detailed', 617), ('detect', 618), ('detectable', 619), ('detected', 620), ('detecting', 621), ('detection', 622), ('determinant', 623), ('determination', 624), ('determine', 625), ('determined', 626), ('determining', 627), ('develop', 628), ('developed', 629), ('developing', 630), ('development', 631), ('deviation', 632), ('device', 633), ('diabetes', 634), ('diagnosed', 635), ('diagnosis', 636), ('diagnostic', 637), ('diagram', 638), ('diarrhea', 639), ('die', 640), ('died', 641), ('differ', 642), ('difference', 643), ('different', 644), ('differential', 645), ('difficult', 646), ('difficulty', 647), ('diluted', 648), ('dilution', 649), ('dimensional', 650), ('direct', 651), ('directed', 652), ('direction', 653), ('directly', 654), ('discharge', 655), ('discharged', 656), ('discovered', 657), ('discovery', 658), ('discrepancy', 659), ('discrete', 660), ('discus', 661), ('discussed', 662), ('discussion', 663), ('disease', 664), ('disorder', 665), ('display', 666), ('displayed', 667), ('distance', 668), ('distancing', 669), ('distinct', 670), ('distinguish', 671), ('distress', 672), ('distributed', 673), ('distribution', 674), ('diverse', 675), ('diversity', 676), ('divided', 677), ('dna', 678), ('doctor', 679), ('documented', 680), ('domain', 681), ('dominant', 682), ('done', 683), ('dose', 684), ('dot', 685), ('double', 686), ('doubling', 687), ('down', 688), ('downloaded', 689), ('downstream', 690), ('dramatic', 691), ('dramatically', 692), ('draw', 693), ('drawn', 694), ('drive', 695), ('driven', 696), ('drop', 697), ('droplet', 698), ('drug', 699), ('dry', 700), ('due', 701), ('duration', 702), ('dynamic', 703), ('dysfunction', 704), ('dyspnea', 705), ('earlier', 706), ('early', 707), ('easily', 708), ('east', 709), ('easy', 710), ('ebola', 711), ('economic', 712), ('economy', 713), ('effect', 714), ('effective', 715), ('effectively', 716), ('effectiveness', 717), ('efficacy', 718), ('efficiency', 719), ('efficient', 720), ('efficiently', 721), ('effort', 722), ('eg', 723), ('eight', 724), ('either', 725), ('elderly', 726), ('electronic', 727), ('element', 728), ('elevated', 729), ('elsewhere', 730), ('emerged', 731), ('emergence', 732), ('emergency', 733), ('emerging', 734), ('empirical', 735), ('employ', 736), ('employed', 737), ('enable', 738), ('enables', 739), ('encoded', 740), ('encoding', 741), ('end', 742), ('energy', 743), ('enhance', 744), ('enhanced', 745), ('enough', 746), ('enriched', 747), ('enrichment', 748), ('enrolled', 749), ('ensure', 750), ('enter', 751), ('entered', 752), ('entire', 753), ('entirely', 754), ('entry', 755), ('envelope', 756), ('enveloped', 757), ('environment', 758), ('environmental', 759), ('enzyme', 760), ('epidemic', 761), ('epidemiological', 762), ('epidemiology', 763), ('epithelial', 764), ('eq', 765), ('equal', 766), ('equally', 767), ('equation', 768), ('equipment', 769), ('equivalent', 770), ('error', 771), ('especially', 772), ('essential', 773), ('essentially', 774), ('establish', 775), ('established', 776), ('estimate', 777), ('estimated', 778), ('estimating', 779), ('estimation', 780), ('etc', 781), ('ethic', 782), ('ethical', 783), ('europe', 784), ('european', 785), ('evaluate', 786), ('evaluated', 787), ('evaluating', 788), ('evaluation', 789), ('even', 790), ('event', 791), ('eventually', 792), ('every', 793), ('evidence', 794), ('evident', 795), ('evolution', 796), ('evolutionary', 797), ('evolved', 798), ('evolving', 799), ('exact', 800), ('exactly', 801), ('examination', 802), ('examine', 803), ('examined', 804), ('examining', 805), ('example', 806), ('except', 807), ('exception', 808), ('exclude', 809), ('excluded', 810), ('excluding', 811), ('exclusion', 812), ('exclusively', 813), ('exhibit', 814), ('exhibited', 815), ('exist', 816), ('existence', 817), ('existing', 818), ('exists', 819), ('expanded', 820), ('expansion', 821), ('expect', 822), ('expected', 823), ('experience', 824), ('experienced', 825), ('experiment', 826), ('experimental', 827), ('experimentally', 828), ('expert', 829), ('explain', 830), ('explained', 831), ('explanation', 832), ('explicitly', 833), ('explore', 834), ('explored', 835), ('exponential', 836), ('exponentially', 837), ('exposed', 838), ('exposure', 839), ('express', 840), ('expressed', 841), ('expressing', 842), ('expression', 843), ('extend', 844), ('extended', 845), ('extension', 846), ('extensive', 847), ('extensively', 848), ('extent', 849), ('external', 850), ('extra', 851), ('extract', 852), ('extracted', 853), ('extraction', 854), ('extreme', 855), ('extremely', 856), ('face', 857), ('facilitate', 858), ('facility', 859), ('fact', 860), ('factor', 861), ('failed', 862), ('failure', 863), ('fall', 864), ('false', 865), ('family', 866), ('far', 867), ('fast', 868), ('faster', 869), ('fatality', 870), ('fatigue', 871), ('feasible', 872), ('feature', 873), ('feb', 874), ('february', 875), ('female', 876), ('fever', 877), ('fewer', 878), ('field', 879), ('fig', 880), ('figure', 881), ('file', 882), ('filter', 883), ('filtered', 884), ('final', 885), ('finally', 886), ('find', 887), ('finding', 888), ('first', 889), ('firstly', 890), ('fisher', 891), ('fit', 892), ('fitted', 893), ('fitting', 894), ('five', 895), ('fixed', 896), ('flow', 897), ('flu', 898), ('fluid', 899), ('fluorescence', 900), ('focus', 901), ('focused', 902), ('focusing', 903), ('fold', 904), ('follow', 905), ('followed', 906), ('following', 907), ('follows', 908), ('food', 909), ('force', 910), ('forecast', 911), ('forecasting', 912), ('form', 913), ('format', 914), ('formation', 915), ('formed', 916), ('former', 917), ('forming', 918), ('formula', 919), ('forward', 920), ('found', 921), ('four', 922), ('fourth', 923), ('fraction', 924), ('fragment', 925), ('frame', 926), ('framework', 927), ('france', 928), ('free', 929), ('frequency', 930), ('frequent', 931), ('frequently', 932), ('full', 933), ('fully', 934), ('function', 935), ('functional', 936), ('fundamental', 937), ('funding', 938), ('furthermore', 939), ('fusion', 940), ('future', 941), ('gain', 942), ('gamma', 943), ('gap', 944), ('gathering', 945), ('gave', 946), ('gel', 947), ('genbank', 948), ('gender', 949), ('gene', 950), ('general', 951), ('generalized', 952), ('generally', 953), ('generate', 954), ('generated', 955), ('generating', 956), ('generation', 957), ('genetic', 958), ('genome', 959), ('genomic', 960), ('genus', 961), ('geographic', 962), ('geographical', 963), ('germany', 964), ('get', 965), ('githubcom', 966), ('give', 967), ('given', 968), ('giving', 969), ('glass', 970), ('global', 971), ('globally', 972), ('globe', 973), ('glycoprotein', 974), ('go', 975), ('goal', 976), ('going', 977), ('good', 978), ('government', 979), ('gradient', 980), ('gradually', 981), ('granted', 982), ('graph', 983), ('great', 984), ('greater', 985), ('greatest', 986), ('greatly', 987), ('green', 988), ('grey', 989), ('ground', 990), ('group', 991), ('grow', 992), ('growing', 993), ('grown', 994), ('growth', 995), ('guidance', 996), ('guide', 997), ('guideline', 998), ('h1n1', 999), ('half', 1000), ('hand', 1001), ('hard', 1002), ('harvested', 1003), ('health', 1004), ('healthcare', 1005), ('healthy', 1006), ('heart', 1007), ('heat', 1008), ('heavily', 1009), ('help', 1010), ('helpful', 1011), ('hence', 1012), ('hepatitis', 1013), ('heterogeneity', 1014), ('high', 1015), ('higher', 1016), ('highest', 1017), ('highlight', 1018), ('highlighted', 1019), ('highly', 1020), ('history', 1021), ('hit', 1022), ('hiv', 1023), ('hold', 1024), ('home', 1025), ('homology', 1026), ('hong', 1027), ('hope', 1028), ('hospital', 1029), ('hospitalization', 1030), ('hospitalized', 1031), ('host', 1032), ('hour', 1033), ('house', 1034), ('household', 1035), ('http', 1036), ('hubei', 1037), ('huge', 1038), ('human', 1039), ('hundred', 1040), ('hypertension', 1041), ('hypothesis', 1042), ('hypothesized', 1043), ('icu', 1044), ('id', 1045), ('idea', 1046), ('ideal', 1047), ('identical', 1048), ('identification', 1049), ('identified', 1050), ('identify', 1051), ('identifying', 1052), ('identity', 1053), ('ie', 1054), ('ifn', 1055), ('igg', 1056), ('ii', 1057), ('iii', 1058), ('il', 1059), ('ill', 1060), ('illness', 1061), ('illustrate', 1062), ('illustrated', 1063), ('illustrates', 1064), ('image', 1065), ('imaging', 1066), ('immediate', 1067), ('immediately', 1068), ('immune', 1069), ('immunity', 1070), ('impact', 1071), ('implement', 1072), ('implementation', 1073), ('implemented', 1074), ('implementing', 1075), ('implication', 1076), ('implies', 1077), ('importance', 1078), ('important', 1079), ('importantly', 1080), ('imported', 1081), ('imposed', 1082), ('impossible', 1083), ('improve', 1084), ('improved', 1085), ('improvement', 1086), ('improving', 1087), ('in', 1088), ('inc', 1089), ('incidence', 1090), ('include', 1091), ('included', 1092), ('includes', 1093), ('including', 1094), ('inclusion', 1095), ('incomplete', 1096), ('incorporate', 1097), ('incorporated', 1098), ('increase', 1099), ('increased', 1100), ('increasing', 1101), ('increasingly', 1102), ('incubated', 1103), ('incubation', 1104), ('indeed', 1105), ('independent', 1106), ('independently', 1107), ('index', 1108), ('indicate', 1109), ('indicated', 1110), ('indicates', 1111), ('indicating', 1112), ('indication', 1113), ('indicator', 1114), ('indirect', 1115), ('individual', 1116), ('individually', 1117), ('induce', 1118), ('induced', 1119), ('induction', 1120), ('infect', 1121), ('infected', 1122), ('infecting', 1123), ('infection', 1124), ('infectious', 1125), ('infectiousness', 1126), ('infectivity', 1127), ('infer', 1128), ('inference', 1129), ('inferred', 1130), ('inflammation', 1131), ('inflammatory', 1132), ('influence', 1133), ('influenced', 1134), ('influenza', 1135), ('inform', 1136), ('information', 1137), ('informative', 1138), ('informed', 1139), ('inhibit', 1140), ('inhibited', 1141), ('inhibiting', 1142), ('inhibition', 1143), ('inhibitor', 1144), ('initial', 1145), ('initially', 1146), ('initiated', 1147), ('initiation', 1148), ('injury', 1149), ('innate', 1150), ('input', 1151), ('inside', 1152), ('insight', 1153), ('instance', 1154), ('instead', 1155), ('institute', 1156), ('institution', 1157), ('institutional', 1158), ('instruction', 1159), ('instrument', 1160), ('insufficient', 1161), ('integrated', 1162), ('integration', 1163), ('intensity', 1164), ('intensive', 1165), ('inter', 1166), ('interact', 1167), ('interacting', 1168), ('interaction', 1169), ('interest', 1170), ('interesting', 1171), ('interestingly', 1172), ('interface', 1173), ('interferon', 1174), ('intermediate', 1175), ('internal', 1176), ('international', 1177), ('interpretation', 1178), ('interpreted', 1179), ('interquartile', 1180), ('interval', 1181), ('intervention', 1182), ('intracellular', 1183), ('intrinsic', 1184), ('introduce', 1185), ('introduced', 1186), ('introduction', 1187), ('inverse', 1188), ('investigate', 1189), ('investigated', 1190), ('investigation', 1191), ('involve', 1192), ('involved', 1193), ('involvement', 1194), ('involves', 1195), ('involving', 1196), ('iqr', 1197), ('iran', 1198), ('isolate', 1199), ('isolated', 1200), ('isolation', 1201), ('issue', 1202), ('italian', 1203), ('italy', 1204), ('iv', 1205), ('january', 1206), ('japan', 1207), ('keep', 1208), ('keeping', 1209), ('kept', 1210), ('key', 1211), ('kidney', 1212), ('kinase', 1213), ('kind', 1214), ('kit', 1215), ('know', 1216), ('knowledge', 1217), ('known', 1218), ('kong', 1219), ('korea', 1220), ('lab', 1221), ('label', 1222), ('labeled', 1223), ('laboratory', 1224), ('lack', 1225), ('lacking', 1226), ('lag', 1227), ('language', 1228), ('large', 1229), ('largely', 1230), ('larger', 1231), ('largest', 1232), ('last', 1233), ('late', 1234), ('latent', 1235), ('later', 1236), ('latest', 1237), ('latter', 1238), ('law', 1239), ('layer', 1240), ('le', 1241), ('lead', 1242), ('leading', 1243), ('learning', 1244), ('least', 1245), ('leaving', 1246), ('led', 1247), ('left', 1248), ('length', 1249), ('lesion', 1250), ('let', 1251), ('level', 1252), ('li', 1253), ('library', 1254), ('life', 1255), ('ligand', 1256), ('light', 1257), ('like', 1258), ('likelihood', 1259), ('likely', 1260), ('limit', 1261), ('limitation', 1262), ('limited', 1263), ('limiting', 1264), ('line', 1265), ('lineage', 1266), ('linear', 1267), ('link', 1268), ('linked', 1269), ('liquid', 1270), ('list', 1271), ('listed', 1272), ('literature', 1273), ('little', 1274), ('live', 1275), ('liver', 1276), ('living', 1277), ('load', 1278), ('local', 1279), ('locally', 1280), ('located', 1281), ('location', 1282), ('lockdown', 1283), ('log', 1284), ('logistic', 1285), ('long', 1286), ('longer', 1287), ('look', 1288), ('looking', 1289), ('loop', 1290), ('loss', 1291), ('lost', 1292), ('lot', 1293), ('low', 1294), ('lower', 1295), ('lowest', 1296), ('lung', 1297), ('lymphocyte', 1298), ('machine', 1299), ('made', 1300), ('magnitude', 1301), ('main', 1302), ('mainland', 1303), ('mainly', 1304), ('maintain', 1305), ('maintained', 1306), ('maintaining', 1307), ('major', 1308), ('majority', 1309), ('make', 1310), ('making', 1311), ('male', 1312), ('mammalian', 1313), ('management', 1314), ('manifestation', 1315), ('mann', 1316), ('manner', 1317), ('manually', 1318), ('manufacturer', 1319), ('manuscript', 1320), ('many', 1321), ('map', 1322), ('mapped', 1323), ('mapping', 1324), ('march', 1325), ('marked', 1326), ('marker', 1327), ('market', 1328), ('mask', 1329), ('mass', 1330), ('massive', 1331), ('match', 1332), ('matched', 1333), ('matching', 1334), ('material', 1335), ('mathematical', 1336), ('matrix', 1337), ('matter', 1338), ('maximum', 1339), ('may', 1340), ('mean', 1341), ('meaning', 1342), ('meanwhile', 1343), ('measure', 1344), ('measured', 1345), ('measurement', 1346), ('measuring', 1347), ('mechanical', 1348), ('mechanism', 1349), ('median', 1350), ('mediated', 1351), ('medical', 1352), ('medication', 1353), ('medicine', 1354), ('medium', 1355), ('medrxiv', 1356), ('meet', 1357), ('member', 1358), ('membrane', 1359), ('mentioned', 1360), ('mers', 1361), ('met', 1362), ('meta', 1363), ('method', 1364), ('methodology', 1365), ('metric', 1366), ('mg', 1367), ('microscopy', 1368), ('mid', 1369), ('middle', 1370), ('might', 1371), ('migration', 1372), ('mild', 1373), ('million', 1374), ('min', 1375), ('minimal', 1376), ('minimize', 1377), ('minimum', 1378), ('minor', 1379), ('minute', 1380), ('missed', 1381), ('missing', 1382), ('mitigate', 1383), ('mitigation', 1384), ('mix', 1385), ('mixed', 1386), ('mixing', 1387), ('mixture', 1388), ('ml', 1389), ('mm', 1390), ('mobility', 1391), ('mode', 1392), ('model', 1393), ('modeled', 1394), ('modeling', 1395), ('modelled', 1396), ('modelling', 1397), ('moderate', 1398), ('modification', 1399), ('modified', 1400), ('molecular', 1401), ('molecule', 1402), ('moment', 1403), ('monitor', 1404), ('monitored', 1405), ('monitoring', 1406), ('month', 1407), ('morbidity', 1408), ('moreover', 1409), ('mortality', 1410), ('mostly', 1411), ('motif', 1412), ('mouse', 1413), ('move', 1414), ('movement', 1415), ('moving', 1416), ('mrna', 1417), ('much', 1418), ('multi', 1419), ('multiple', 1420), ('muscle', 1421), ('must', 1422), ('mutant', 1423), ('mutation', 1424), ('name', 1425), ('named', 1426), ('namely', 1427), ('nasal', 1428), ('nasopharyngeal', 1429), ('nation', 1430), ('national', 1431), ('natural', 1432), ('naturally', 1433), ('nature', 1434), ('nc', 1435), ('ncbi', 1436), ('ncinternational', 1437), ('ncov', 1438), ('ndinternational', 1439), ('near', 1440), ('nearly', 1441), ('necessarily', 1442), ('necessary', 1443), ('need', 1444), ('needed', 1445), ('negative', 1446), ('negatively', 1447), ('negligible', 1448), ('neither', 1449), ('network', 1450), ('neutrophil', 1451), ('never', 1452), ('nevertheless', 1453), ('new', 1454), ('newly', 1455), ('news', 1456), ('next', 1457), ('ng', 1458), ('nine', 1459), ('nm', 1460), ('node', 1461), ('noise', 1462), ('non', 1463), ('none', 1464), ('nonetheless', 1465), ('normal', 1466), ('normalized', 1467), ('normally', 1468), ('notable', 1469), ('notably', 1470), ('note', 1471), ('noted', 1472), ('noting', 1473), ('novel', 1474), ('nt', 1475), ('nucleic', 1476), ('nucleotide', 1477), ('numerical', 1478), ('numerous', 1479), ('objective', 1480), ('observation', 1481), ('observe', 1482), ('observed', 1483), ('obtain', 1484), ('obtained', 1485), ('obvious', 1486), ('obviously', 1487), ('occur', 1488), ('occurred', 1489), ('occurrence', 1490), ('occurring', 1491), ('occurs', 1492), ('of', 1493), ('off', 1494), ('offer', 1495), ('official', 1496), ('often', 1497), ('old', 1498), ('older', 1499), ('on', 1500), ('ongoing', 1501), ('online', 1502), ('onset', 1503), ('onto', 1504), ('open', 1505), ('opportunity', 1506), ('opposite', 1507), ('optimal', 1508), ('optimization', 1509), ('optimized', 1510), ('option', 1511), ('or', 1512), ('oral', 1513), ('orange', 1514), ('order', 1515), ('ordinary', 1516), ('organ', 1517), ('organism', 1518), ('organization', 1519), ('origin', 1520), ('original', 1521), ('originally', 1522), ('originated', 1523), ('others', 1524), ('otherwise', 1525), ('out', 1526), ('outbreak', 1527), ('outcome', 1528), ('output', 1529), ('outside', 1530), ('over', 1531), ('overall', 1532), ('overcome', 1533), ('overlap', 1534), ('overlapping', 1535), ('overnight', 1536), ('oxygen', 1537), ('package', 1538), ('page', 1539), ('pair', 1540), ('paired', 1541), ('pandemic', 1542), ('panel', 1543), ('paper', 1544), ('parallel', 1545), ('parameter', 1546), ('part', 1547), ('partial', 1548), ('partially', 1549), ('participant', 1550), ('particle', 1551), ('particular', 1552), ('particularly', 1553), ('passed', 1554), ('past', 1555), ('path', 1556), ('pathogen', 1557), ('pathogenesis', 1558), ('pathogenic', 1559), ('pathway', 1560), ('patient', 1561), ('pattern', 1562), ('pb', 1563), ('pcr', 1564), ('pdb', 1565), ('peak', 1566), ('pearson', 1567), ('people', 1568), ('peptide', 1569), ('per', 1570), ('percent', 1571), ('percentage', 1572), ('perform', 1573), ('performance', 1574), ('performed', 1575), ('performing', 1576), ('perhaps', 1577), ('period', 1578), ('perioddays', 1579), ('peripheral', 1580), ('permission', 1581), ('perpetuity', 1582), ('person', 1583), ('personal', 1584), ('perspective', 1585), ('ph', 1586), ('pharmaceutical', 1587), ('phase', 1588), ('phenomenon', 1589), ('phenotype', 1590), ('phosphate', 1591), ('phylogenetic', 1592), ('physical', 1593), ('physician', 1594), ('physiological', 1595), ('picture', 1596), ('pipeline', 1597), ('place', 1598), ('placed', 1599), ('plan', 1600), ('planning', 1601), ('plasma', 1602), ('plasmid', 1603), ('plate', 1604), ('platform', 1605), ('plausible', 1606), ('play', 1607), ('plot', 1608), ('plotted', 1609), ('plus', 1610), ('pneumonia', 1611), ('point', 1612), ('poisson', 1613), ('policy', 1614), ('polymerase', 1615), ('pool', 1616), ('pooled', 1617), ('poor', 1618), ('poorly', 1619), ('population', 1620), ('portion', 1621), ('pose', 1622), ('position', 1623), ('positive', 1624), ('positively', 1625), ('possibility', 1626), ('possible', 1627), ('possibly', 1628), ('post', 1629), ('posterior', 1630), ('potent', 1631), ('potential', 1632), ('potentially', 1633), ('power', 1634), ('powerful', 1635), ('practical', 1636), ('practice', 1637), ('pre', 1638), ('precise', 1639), ('precisely', 1640), ('precision', 1641), ('predict', 1642), ('predicted', 1643), ('predicting', 1644), ('prediction', 1645), ('predictive', 1646), ('predictor', 1647), ('predicts', 1648), ('predominantly', 1649), ('preliminary', 1650), ('preparation', 1651), ('prepared', 1652), ('presence', 1653), ('present', 1654), ('presentation', 1655), ('presented', 1656), ('presenting', 1657), ('pressure', 1658), ('prevalence', 1659), ('prevalent', 1660), ('prevent', 1661), ('preventing', 1662), ('prevention', 1663), ('preventive', 1664), ('previous', 1665), ('previously', 1666), ('primarily', 1667), ('primary', 1668), ('primer', 1669), ('principle', 1670), ('prior', 1671), ('pro', 1672), ('probability', 1673), ('probably', 1674), ('probe', 1675), ('problem', 1676), ('procedure', 1677), ('process', 1678), ('processed', 1679), ('processing', 1680), ('produce', 1681), ('produced', 1682), ('producing', 1683), ('product', 1684), ('production', 1685), ('professional', 1686), ('profile', 1687), ('prognosis', 1688), ('program', 1689), ('progress', 1690), ('progression', 1691), ('project', 1692), ('projected', 1693), ('projection', 1694), ('prolonged', 1695), ('promising', 1696), ('promote', 1697), ('prone', 1698), ('proper', 1699), ('property', 1700), ('proportion', 1701), ('proportional', 1702), ('propose', 1703), ('proposed', 1704), ('protease', 1705), ('protect', 1706), ('protection', 1707), ('protective', 1708), ('protein', 1709), ('protocol', 1710), ('proved', 1711), ('proven', 1712), ('provide', 1713), ('provided', 1714), ('provides', 1715), ('providing', 1716), ('province', 1717), ('proximity', 1718), ('proxy', 1719), ('public', 1720), ('publication', 1721), ('publicly', 1722), ('published', 1723), ('pulmonary', 1724), ('purification', 1725), ('purified', 1726), ('purpose', 1727), ('put', 1728), ('putative', 1729), ('qpcr', 1730), ('quality', 1731), ('quantification', 1732), ('quantified', 1733), ('quantify', 1734), ('quantitative', 1735), ('quantity', 1736), ('quarantine', 1737), ('quarantined', 1738), ('question', 1739), ('quickly', 1740), ('quite', 1741), ('r0', 1742), ('raise', 1743), ('raised', 1744), ('random', 1745), ('randomly', 1746), ('range', 1747), ('ranging', 1748), ('rank', 1749), ('rapid', 1750), ('rapidly', 1751), ('rare', 1752), ('rarely', 1753), ('rate', 1754), ('rather', 1755), ('ratio', 1756), ('raw', 1757), ('ray', 1758), ('rbd', 1759), ('re', 1760), ('reach', 1761), ('reached', 1762), ('reaching', 1763), ('reaction', 1764), ('reactive', 1765), ('read', 1766), ('readily', 1767), ('reading', 1768), ('reagent', 1769), ('real', 1770), ('realistic', 1771), ('reality', 1772), ('reason', 1773), ('reasonable', 1774), ('receive', 1775), ('received', 1776), ('recent', 1777), ('recently', 1778), ('receptor', 1779), ('recognition', 1780), ('recognized', 1781), ('recombinant', 1782), ('recommend', 1783), ('recommendation', 1784), ('recommended', 1785), ('record', 1786), ('recorded', 1787), ('recover', 1788), ('recovered', 1789), ('recovery', 1790), ('red', 1791), ('reduce', 1792), ('reduced', 1793), ('reduces', 1794), ('reducing', 1795), ('reduction', 1796), ('refer', 1797), ('reference', 1798), ('referred', 1799), ('reflect', 1800), ('reflected', 1801), ('reflects', 1802), ('regard', 1803), ('regarding', 1804), ('regardless', 1805), ('region', 1806), ('regional', 1807), ('regression', 1808), ('regular', 1809), ('regulated', 1810), ('regulation', 1811), ('regulatory', 1812), ('related', 1813), ('relation', 1814), ('relationship', 1815), ('relative', 1816), ('relatively', 1817), ('release', 1818), ('released', 1819), ('relevance', 1820), ('relevant', 1821), ('reliable', 1822), ('relies', 1823), ('rely', 1824), ('remain', 1825), ('remained', 1826), ('remaining', 1827), ('remains', 1828), ('remarkably', 1829), ('removal', 1830), ('remove', 1831), ('removed', 1832), ('removing', 1833), ('renal', 1834), ('repeat', 1835), ('repeated', 1836), ('replaced', 1837), ('replicate', 1838), ('replicates', 1839), ('replication', 1840), ('report', 1841), ('reported', 1842), ('reporting', 1843), ('repository', 1844), ('represent', 1845), ('representation', 1846), ('representative', 1847), ('represented', 1848), ('representing', 1849), ('represents', 1850), ('reproduction', 1851), ('reproductive', 1852), ('republic', 1853), ('request', 1854), ('require', 1855), ('required', 1856), ('requirement', 1857), ('requires', 1858), ('requiring', 1859), ('research', 1860), ('researcher', 1861), ('reserved', 1862), ('reservoir', 1863), ('resident', 1864), ('residual', 1865), ('residue', 1866), ('resistance', 1867), ('resistant', 1868), ('resolution', 1869), ('resolved', 1870), ('resource', 1871), ('respect', 1872), ('respective', 1873), ('respectively', 1874), ('respiratory', 1875), ('respond', 1876), ('response', 1877), ('responsible', 1878), ('rest', 1879), ('restricted', 1880), ('restriction', 1881), ('resulted', 1882), ('resulting', 1883), ('retrieved', 1884), ('retrospective', 1885), ('return', 1886), ('returned', 1887), ('reuse', 1888), ('reveal', 1889), ('revealed', 1890), ('reveals', 1891), ('reverse', 1892), ('review', 1893), ('right', 1894), ('rise', 1895), ('risk', 1896), ('rna', 1897), ('robust', 1898), ('robustness', 1899), ('role', 1900), ('room', 1901), ('root', 1902), ('roughly', 1903), ('route', 1904), ('routine', 1905), ('rt', 1906), ('rule', 1907), ('run', 1908), ('running', 1909), ('s1', 1910), ('s2', 1911), ('s3', 1912), ('s4', 1913), ('s5', 1914), ('s6', 1915), ('safe', 1916), ('safety', 1917), ('sample', 1918), ('sampled', 1919), ('sampling', 1920), ('sars', 1921), ('saturation', 1922), ('scale', 1923), ('scan', 1924), ('scenario', 1925), ('scheme', 1926), ('school', 1927), ('science', 1928), ('scientific', 1929), ('scope', 1930), ('score', 1931), ('screen', 1932), ('screened', 1933), ('screening', 1934), ('sd', 1935), ('seafood', 1936), ('search', 1937), ('searched', 1938), ('seasonal', 1939), ('second', 1940), ('secondary', 1941), ('secondly', 1942), ('section', 1943), ('see', 1944), ('seem', 1945), ('seems', 1946), ('seen', 1947), ('segment', 1948), ('seir', 1949), ('select', 1950), ('selected', 1951), ('selection', 1952), ('selective', 1953), ('self', 1954), ('sense', 1955), ('sensitive', 1956), ('sensitivity', 1957), ('separate', 1958), ('separated', 1959), ('separately', 1960), ('seq', 1961), ('sequence', 1962), ('sequenced', 1963), ('sequencing', 1964), ('serial', 1965), ('series', 1966), ('serious', 1967), ('serum', 1968), ('serve', 1969), ('service', 1970), ('set', 1971), ('setting', 1972), ('seven', 1973), ('several', 1974), ('severe', 1975), ('severely', 1976), ('severity', 1977), ('sex', 1978), ('shanghai', 1979), ('shape', 1980), ('share', 1981), ('shared', 1982), ('sharing', 1983), ('shift', 1984), ('shock', 1985), ('short', 1986), ('shortage', 1987), ('shorter', 1988), ('shortness', 1989), ('show', 1990), ('showed', 1991), ('showing', 1992), ('shown', 1993), ('side', 1994), ('sigma', 1995), ('sign', 1996), ('signal', 1997), ('signaling', 1998), ('signature', 1999), ('significance', 2000), ('significant', 2001), ('significantly', 2002), ('similar', 2003), ('similarity', 2004), ('similarly', 2005), ('simple', 2006), ('simply', 2007), ('simulate', 2008), ('simulated', 2009), ('simulation', 2010), ('simultaneously', 2011), ('since', 2012), ('singapore', 2013), ('single', 2014), ('sir', 2015), ('site', 2016), ('situation', 2017), ('six', 2018), ('size', 2019), ('slight', 2020), ('slightly', 2021), ('slow', 2022), ('slower', 2023), ('small', 2024), ('smaller', 2025), ('social', 2026), ('society', 2027), ('software', 2028), ('solid', 2029), ('solution', 2030), ('solve', 2031), ('solved', 2032), ('somewhat', 2033), ('soon', 2034), ('sought', 2035), ('source', 2036), ('south', 2037), ('space', 2038), ('spain', 2039), ('spatial', 2040), ('special', 2041), ('specie', 2042), ('specific', 2043), ('specifically', 2044), ('specificity', 2045), ('specified', 2046), ('specimen', 2047), ('spectrum', 2048), ('speed', 2049), ('spike', 2050), ('split', 2051), ('spread', 2052), ('spreading', 2053), ('spss', 2054), ('square', 2055), ('squared', 2056), ('stability', 2057), ('stable', 2058), ('staff', 2059), ('stage', 2060), ('stained', 2061), ('staining', 2062), ('standard', 2063), ('start', 2064), ('started', 2065), ('starting', 2066), ('state', 2067), ('statistic', 2068), ('statistical', 2069), ('statistically', 2070), ('status', 2071), ('stay', 2072), ('stem', 2073), ('step', 2074), ('still', 2075), ('stochastic', 2076), ('stop', 2077), ('stored', 2078), ('storm', 2079), ('strain', 2080), ('strand', 2081), ('stranded', 2082), ('strategy', 2083), ('stratified', 2084), ('strength', 2085), ('stress', 2086), ('strict', 2087), ('stringent', 2088), ('strong', 2089), ('stronger', 2090), ('strongly', 2091), ('structural', 2092), ('structure', 2093), ('structured', 2094), ('student', 2095), ('studied', 2096), ('studying', 2097), ('sub', 2098), ('subject', 2099), ('subjected', 2100), ('subsequent', 2101), ('subsequently', 2102), ('subset', 2103), ('substantial', 2104), ('substantially', 2105), ('substitution', 2106), ('substrate', 2107), ('subunit', 2108), ('success', 2109), ('successful', 2110), ('successfully', 2111), ('sufficient', 2112), ('sufficiently', 2113), ('suggest', 2114), ('suggested', 2115), ('suggesting', 2116), ('suggests', 2117), ('suitable', 2118), ('sum', 2119), ('summarized', 2120), ('summary', 2121), ('supernatant', 2122), ('supplemental', 2123), ('supplementary', 2124), ('supplemented', 2125), ('supply', 2126), ('support', 2127), ('supported', 2128), ('supporting', 2129), ('suppression', 2130), ('surface', 2131), ('surprisingly', 2132), ('surrounding', 2133), ('surveillance', 2134), ('survey', 2135), ('survival', 2136), ('susceptibility', 2137), ('susceptible', 2138), ('suspected', 2139), ('sustained', 2140), ('swab', 2141), ('symptom', 2142), ('symptomatic', 2143), ('syndrome', 2144), ('synthesis', 2145), ('synthesized', 2146), ('synthetic', 2147), ('system', 2148), ('systematic', 2149), ('systematically', 2150), ('systemic', 2151), ('table', 2152), ('tag', 2153), ('tail', 2154), ('take', 2155), ('taken', 2156), ('taking', 2157), ('target', 2158), ('targeted', 2159), ('targeting', 2160), ('task', 2161), ('team', 2162), ('technical', 2163), ('technique', 2164), ('technology', 2165), ('temperature', 2166), ('template', 2167), ('temporal', 2168), ('ten', 2169), ('tend', 2170), ('term', 2171), ('terminal', 2172), ('test', 2173), ('tested', 2174), ('testing', 2175), ('text', 2176), ('th', 2177), ('thank', 2178), ('the', 2179), ('theoretical', 2180), ('theory', 2181), ('therapeutic', 2182), ('therapy', 2183), ('thereby', 2184), ('therefore', 2185), ('thermo', 2186), ('third', 2187), ('though', 2188), ('thought', 2189), ('thousand', 2190), ('threat', 2191), ('three', 2192), ('threshold', 2193), ('throat', 2194), ('throughout', 2195), ('throughput', 2196), ('thus', 2197), ('timely', 2198), ('timing', 2199), ('tissue', 2200), ('titer', 2201), ('to', 2202), ('together', 2203), ('took', 2204), ('tool', 2205), ('top', 2206), ('total', 2207), ('toward', 2208), ('towards', 2209), ('tracing', 2210), ('track', 2211), ('tracking', 2212), ('tract', 2213), ('traditional', 2214), ('traffic', 2215), ('trained', 2216), ('training', 2217), ('trajectory', 2218), ('transcript', 2219), ('transcriptase', 2220), ('transcription', 2221), ('transfer', 2222), ('transferred', 2223), ('transformed', 2224), ('transition', 2225), ('translation', 2226), ('transmissibility', 2227), ('transmission', 2228), ('transmit', 2229), ('transmitted', 2230), ('transport', 2231), ('transportation', 2232), ('travel', 2233), ('treat', 2234), ('treated', 2235), ('treating', 2236), ('treatment', 2237), ('tree', 2238), ('trend', 2239), ('trial', 2240), ('trigger', 2241), ('true', 2242), ('tube', 2243), ('turn', 2244), ('twice', 2245), ('type', 2246), ('typical', 2247), ('typically', 2248), ('uk', 2249), ('ultimately', 2250), ('unable', 2251), ('uncertain', 2252), ('uncertainty', 2253), ('unclear', 2254), ('under', 2255), ('underlying', 2256), ('understand', 2257), ('understanding', 2258), ('understood', 2259), ('undetected', 2260), ('unexpected', 2261), ('unfortunately', 2262), ('uniform', 2263), ('unique', 2264), ('unit', 2265), ('united', 2266), ('universal', 2267), ('university', 2268), ('unknown', 2269), ('unless', 2270), ('unlike', 2271), ('unlikely', 2272), ('unprecedented', 2273), ('up', 2274), ('update', 2275), ('updated', 2276), ('upon', 2277), ('upper', 2278), ('urgent', 2279), ('us', 2280), ('usa', 2281), ('usage', 2282), ('use', 2283), ('useful', 2284), ('user', 2285), ('usually', 2286), ('utilized', 2287), ('vaccination', 2288), ('vaccine', 2289), ('valid', 2290), ('validate', 2291), ('validated', 2292), ('validation', 2293), ('valuable', 2294), ('value', 2295), ('values', 2296), ('variability', 2297), ('variable', 2298), ('variance', 2299), ('variant', 2300), ('variation', 2301), ('varied', 2302), ('varies', 2303), ('variety', 2304), ('various', 2305), ('vary', 2306), ('varying', 2307), ('vast', 2308), ('vector', 2309), ('ventilation', 2310), ('verified', 2311), ('verify', 2312), ('version', 2313), ('versus', 2314), ('via', 2315), ('viable', 2316), ('view', 2317), ('viral', 2318), ('virion', 2319), ('virus', 2320), ('visible', 2321), ('visit', 2322), ('visualization', 2323), ('visualized', 2324), ('vital', 2325), ('vitro', 2326), ('vivo', 2327), ('volume', 2328), ('vulnerable', 2329), ('wang', 2330), ('washed', 2331), ('washing', 2332), ('water', 2333), ('way', 2334), ('weak', 2335), ('web', 2336), ('website', 2337), ('week', 2338), ('weight', 2339), ('weighted', 2340), ('well', 2341), ('western', 2342), ('whereas', 2343), ('whether', 2344), ('white', 2345), ('whitney', 2346), ('whole', 2347), ('whose', 2348), ('wide', 2349), ('widely', 2350), ('wider', 2351), ('widespread', 2352), ('wild', 2353), ('window', 2354), ('within', 2355), ('woman', 2356), ('word', 2357), ('work', 2358), ('worker', 2359), ('working', 2360), ('world', 2361), ('worldwide', 2362), ('worse', 2363), ('worst', 2364), ('worth', 2365), ('would', 2366), ('writing', 2367), ('written', 2368), ('wuhan', 2369), ('year', 2370), ('yellow', 2371), ('yet', 2372), ('yield', 2373), ('young', 2374), ('younger', 2375), ('zero', 2376), ('zhang', 2377), ('zoonotic', 2378), ('µl', 2379)]\n",
      "[[0.         0.         0.         ... 0.         0.         0.        ]\n",
      " [0.02237029 0.         0.         ... 0.         0.         0.        ]\n",
      " [0.         0.         0.         ... 0.         0.         0.        ]\n",
      " ...\n",
      " [0.         0.04004387 0.         ... 0.         0.         0.        ]\n",
      " [0.         0.0441761  0.         ... 0.         0.         0.        ]\n",
      " [0.01191907 0.         0.         ... 0.01171261 0.         0.        ]]\n"
     ]
    }
   ],
   "source": [
    "#create a tf-idf matrix for the words above\n",
    "v = TfidfVectorizer(min_df = 0.05,max_df = 0.8)\n",
    "#For the above tf-idf vectorizer:\n",
    "#The first parameter specifies that if a term appears in less than 5% of the docs, ignore it\n",
    "#The second parameter, considers word that do not occur in more than 80% of the corpus\n",
    "tfidf = v.fit_transform(lemmatized_text) #fit and transform the lemmatized texts which were previously cleaned.\n",
    "print(sorted(v.vocabulary_.items(), key=lambda x : x[1]))\n",
    "print(tfidf.toarray())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h5>Cosine Similarity</h5>\n",
    "<p> Cosine similarity is then utilized to get the documents which are closest to the query. The closer the cosine similairty value is to 1, the closer the document is to the query (The more likely it is going to be output). However, for this notebook, we choose to display the top 5 papers with cosine similarity greater than or equal <i>0.1</i>. This number was chosen after several trials to fetch the most relevant papers for different queries.</p>\n",
    "<h6 style = \"color:green\"> PROS of using cosine similarity</h6>\n",
    "<p>Cosine similarity is of a good use, as it can match related documents and queries quite well compared to euclidean distance. Euclidean distance can classify documents and queries as non-similar even if the distribution of terms is significantly alike</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Enter your query: Disease models, including animal models for infection, disease and transmission\n",
      "[0.         0.02082769 0.01224657 ... 0.035525   0.03111784 0.02483629]\n",
      "1897\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#Next, the search engine will be implemented with the help of the tf-idf matrix created above and the cosine similarity\n",
    "flag = False # a flag to indicate whether or not a result is found\n",
    "pd.set_option('display.max_colwidth', -1)#to avoid dataframe column's data truncation\n",
    "query = input(\"Enter your query: \") #get the query from the user\n",
    "finalQuery=\"\" #create an empty string\n",
    "words = query.split(' ') #split the query on spaces\n",
    "for word in words: #for each word in the query\n",
    "    word = word.lower()\n",
    "    if(word!=\"\"): #if the word is not an empty string\n",
    "        if (word==\"coronaviruses\"):#since lemmatize functions do not recognize the word coronaviruses as the plural of coronavirus, a conidition was created\n",
    "            word =\"coronavirus\"\n",
    "        lem_query = lemmatizer.lemmatize(word)#lemmatize the word\n",
    "        finalQuery+=lem_query+\" \" #lemmatize the word and concat it to the string finalQuery\n",
    "query = [finalQuery.strip()]# remove any trailing spaces from the query\n",
    "query_tfidf = v.transform(query)\n",
    "cosineSimilarities = cosine_similarity(query_tfidf, tfidf).flatten() #get the cosine similarity list \n",
    "#IF NO RESULT\n",
    "print(cosineSimilarities)\n",
    "countzero_in2 = np.count_nonzero(cosineSimilarities) #get the number of non zero values in cosine similarity\n",
    "print(countzero_in2)\n",
    "if (countzero_in2==0):#if all cosine similarities are 0 (no results)\n",
    "    flag = True \n",
    "\n",
    "else:\n",
    "    if (countzero_in2>=5):#if there are 5 or more cosine similarities that are greater than 0\n",
    "        answer_indicies = sorted(range(len(cosineSimilarities)), key=lambda i: cosineSimilarities[i])[-5:] #get the top 5 indices with the highest cosine similarity\n",
    "    else: #else if they are less than 5, get all of them\n",
    "        answer_indicies = sorted(range(len(cosineSimilarities)), key=lambda i: cosineSimilarities[i])[-countzero_in2:] #get the top countzero_in2 indices with the highest cosine \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0.25595905845773503, 0.2618753791816746, 0.26811109153799906, 0.2736237709532856, 0.2849303971063369]\n",
      "[766, 1417, 1361, 1761, 178]\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAcYAAAFRCAYAAADuAQ3SAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAgAElEQVR4nO3debwcRbn/8U+RgCg7HBEStoCgsgmyiwIiYEQWNx4WQfb8QAERvArCBUHAAF6Re0EhF1AQJD6AQLgEEGV1ARNExLBo2JNAJISdkBjo3x9Vk1QmZ2b6JLOdk+/79TqvM91d3f1Mz/JMVVdXh6IoEBERkWixTgcgIiLSTZQYRUREMkqMIiIiGSVGERGRjBKjiIhIRolRREQko8Qo0g+EEO4LIVzQhv18OIRQhBA2b/W++lMsACGE4SmenmZup5nPM4RwRAjhjYXdzqJOibEfSh+ien9Pt3j/R1Tt74UQwo0hhI+0cr+tFEJYsuo5vRlC+HsI4egF2NZOaRurNDHEXYETm7i9hRJCWDqE8L0QwsMhhLdCCC+FEP4UQjgyhPDeJu3mn8CqwF+btL26QggjQggPhhBeDyG8FkKYEEK4MCtyR4rnpYXcVbO205vLgbUrEyGEw0IIb7dgPwPa4E4HIAtk1ezxlsCN6f9zad47bYjhLWAdIBA/iBcAt4YQ1i+K4s027L+mEMISRVHMWsDVDwf+D3gfMRmdH0J4tSiKK5oWYBJCWLwoin+XKVsUxfRm739BhRBWAO4FVgROAf4MvAFsARwLPAXcurD7KYriHeCFhd1OGSGEI4AfAccRE1cA1gc+m8UzqxnxNGs7uRBCAAYXRTEDmNHMbS+SiqLQXz/+Az4BFMBavSxbHrgUmAa8DdwPfCpb/uG07j7AXanMRGCvBvs8Anijat6n07a2T9MHAuOA14AXgTHAOn3dNzAEuDI9h9eIX8gfz5YPT9v5DPAnYCZwGLAC8Atgatr2M8AP6jynJdN2vlw1fwJwVdmYsueV/92alo0mJt3jUzzvEn+cBuCbwD9SrI8D3wYGZfu8D7ggPT467XuJqrhOJSalkMVyI/AqMJ2YrNavWucrwJNpv/cCX0wxb17nWP0vMRGu1suyxYDl0uNArOU+DcxKr+/Xq8p/GXiI+EPr5fQablh1LDevmv4icEtaZyKwd9U2lwUuBJ4H3gTGA7s3eE/fClzZoEzlvdZTNb0L8cfBjPT/Q8BH03N5M/1fr8525nmead65wGPpOT4L/A+wdPVnkPi+fwj4N7AT2Wcz20/+dxFwJPEzWf3++QHwWKe/07rhr+MB6G8hX8D6ifEm4In0gVkf+Gn6Alw7La98ICcBe6cP9DnEGudGdfbZW2L8eNrWTmn6cGKNax1gs/RFNoH4q7bUvoGlic1pVwMfA9YFTktfQOukMpUP/4S0v2HExDUqfSFuAayZjtMhdZ7TPImR+AU/nJhof5aVqxsTMAjYK21rI2AVYIW07mjgdcCBjYlfngEYSUxOe6T4dwemACdl+80TYw8x0Xyh6jn8EzgtPR5KTJ7nAxum430x8YdCJZ5tiMn5tHT8jfglXDMxEhP565VYGrw3jycmhoPTcToqxf2VtHwNYDbwjfS81wcOAD5S9R6pToz/JCbHDwLnpddozex1+wPwW+J7ch3g68TE8Yk6sf6M+MNknTplaiXGccD26TiPB/4C3APsAGxATJZ319lOb4nxFGBbYC1i4p0IXFz1GZxN/LG7fXqeKzFvYlyCWAN+m/g+XIX4o2GZ9BrunW1vMPE9d3ynv9O64a/jAehvIV/AGokxfSALYMdsXiAmkJ+k6coH8qSqdR8ALqmzz3kSI/AB4DZirWSFGuusmva1Wdl9p/08CSxWVeaPwMj0uPIlU13TvA24qA/HsZIYZxB/if87Tb8GbFr13BvFtFNad5WqMqOJyeq92bzliF/sO1SVHQG8kE3PSYxp+gbg+my68sPkg2l6JHBX1TYXI/4QOSJNXwv8rqrMt6ifGNdIy79W4pi+CJxeNe+nwCPpcSUxr1pj/VqJ8WtZmSXS8Tswez+8CSxVta1fAqPrxLpaOsZFen2vBg6teq1qJcbhWZkD0rzPZfP2Tc9ziRrbmS8x9hLfvsDrVe/DAtiiwWfzMODtXrY3CvhtNr1nOo49ZT8zA/lP5xgHrg2IH8bfV2YURVGEEO5Ny3J/qpr+I7GWV89SqfdbIJ6PmwB8sSiKlwFCCJsRf/V+lPhLNqT11iQmvzL73oL4RfxaPIUyx3uIzWS5P1dNXwD8KoSwDfA7YlPZ7UX6FqjjP1LZIcTmrJ8XRfFgtrwvMfXm4SKeB6rYmPjlfnMIIY9tELBkCGGZoihe72U7VwBXhxBWKoriJeIX8h+KopiYxbltLz0U30usvUGsoV1Xtfz31Fd50nWPYwhhZWLN9p6qRXcDh4UQFifWtO4GHg8h3E5sUv91URSTG8QwpzNOURSzQgjTiD/OID7v9wJTq16fJYCHa22wKIpJwNYhhA2ATwJbE2uj3wkhbF3UP8f7UPa4cu7wb1XzAvF4TKmznTlCCHsTm8zXIdbwKu+HFbNY3iHWThfERcD4EMLaRVE8SWzhub4oimkLuL0BRYlx0RNo8KXG3C+/et4CNknbmloUxZwv4BDCcsDt6e9AYvPdEsQvkCX6sO/FiF+C+/RSrrqDzzzTRVHcFEJYg3gO5lPAr4BxIYThRVG8W2f/L6TkMjGE8AXgsRDCQ0VR/HEBYupNdZlKz/A9iOcdy27z/4jNYfuEEP6X2Aya91pdDBhLbM6s9kr6X+a9UG0ysUZd/eOqlurtz3l9i6KYHULYEdiKWMveBzg7hLBnURS319lmdceqgrnHcTHgX8SWlGozGwZbFBOIP/IuCiGcRTzPdzhwdp3V8g5URZ15pa4CCCFsR6zhnk5sCn2F2Fw6ink/P28XsYNSnxVF8ZcQwgPEHykXEmuxuyzItgYiJcaBawLxg/gJYi+7Ss+1bYm/0nNbV8ok29C4i3yR1U6qbUjs/HJCURRPpX3vWKNsvX2PJ55Lmt7gF3utAKcBVwFXhRCuBO4k/gL/Z8n1p4QQLgF+TOz1Wzamyhf3oBK7+RvxS3RYURS/KxNXim1WCOFXwFeJNdX3Ec9dVlTifLao3UN3AvH9kKuert7v7LTfg0III1NNa44QwmLAMkVR/CuE8CLxCz1/XtsB/yhSb9xUg78v/Z0RQrgLOIj4o2pBjAdWJnZAKvU611HplLTyQm6nrz4JTCqK4rTKjBDC/gu4rVnUfh9eDHyf+P57ivj5EHQd44CVfvneBIxK19V9BPgJscPCf1UVPzKEYCGE9UIII4k1wfMXYvdPET9sx4QQ1g4h7EJsluxNvX1fTmyGujk9h7VCCFuHEE4OIXyuXgAhhLNDCJ9P2/0Q8RzNa8QaT1+cB2waQvhSH2J6Ov3/XAhh5RDCsrU2npqezwV+mK4PXS+EsGEIYb8QwpkNYrucmLD/ExhTFMUr2bIfEzsK/TqEsG2K85MhhJFh7oXkPwI+FUI4Ne13L+CYEsfk28ROOn8OIRwaQtgohDAshPBlYlPsNqncSOD4EMLBIYR1QwhHEc/bnQUQQtghhPDdEMKWIYQ10vtkfeCREjHUckuKYUwIYY8U1+YhhGNDCAfVWimEcGl6DT8RQlgzhLAFsVfzksTzue30ODA0hHBA+vwcQjxXuCCeAgaHEHYNIfSEEJbKll1N/EF1AvG8fl9bDwauTp/k1N/C/VH+co2Z1L5cY19iV/23ib+S926wz/l6pfZSZl9ij9i3ib/it0v72qcv+wbeT7w84Hnir99JxE4jlZ6r83RkyNb7PrFG9CaxKeoOYOs68fZ6uUZa9gvgUdLlE41iSmVOJp5PepeqyzXqHNO/pddpOvHc62HZ8nk632TzH6Wqs0e2bG3mdviZSUzYVwCrZ2W+SvzynEk8v/slGnQESestQ2zqm0DssFSJ+f8BS6YyAfguNS7XIJ5/vpXY1F6J7wfM33O5uvPN5lWxTCK2TlSmlwJ+SGyarlwzOBbYrs7z2Ru4mfjDaWZa5zZg56xMrc43PVmZ+TpeEXunFqTLW3rZTvXzDMSm2xeJ798xzO3Us0q9z2Bv84k/iF9M619Uteyn6Rh9oB3fV/3lr3K9kyyCQggfJn6xblEUxfhFZd8iEoUQxgAzi6LYq9OxdBOdYxQRWcSEEFYkdkr7HA3OKy+KlBhFRBY9jxDPL55WFMV9nQ6m26gpVUREJKNeqSIiIhklRhERkcxAPceo9mEREelNw5G9BmpiZMqUUkMSLpCenh6mTes/Qwr2t3hBMbdLf4u5v8ULirkdysY7ZMiQUttTU6qIiEhGiVFERCSjxCgiIpJRYhQREckoMYqIiGSUGEVERDJKjCIiIhklRhERkYwSo4iISEaJUUREJKPEKCIiklFirOOdw/fgncP36HQYIiLSRkqMIiIiGSVGERGRjBKjiIhIRolRREQko8QoIiKSUWIUERHJKDGKiIhklBhFREQySowiIiIZJUYREelanRh9TIlRREQko8QoIiKSUWIUERHJKDGKiIhklBhFRBaQbks3MCkxioiIZJQYRUREMkqMIiIiGSVGERGRjBKjiIhIRolRRKRN1Iu1f1BiFBERySgxioiIZJQYRUREMkqMIiIiGSVGkZLUcUJk0aDEKCIikhncrh2Z2XDgfGAQcIm7j6xafhxwGDAbeBE4xN2fScveAR5ORZ91d/10FxGRlmhLYjSzQcCFwM7AJGCcmY1x90eyYg8Cm7v7W2Z2JHAOsHdaNsPdN2lHrCIismhrV41xS2Ciuz8JYGajgT2BOYnR3e/Myt8H7N+m2EREROZoV2IcCjyXTU8CtqpT/lDglmx6STMbT2xmHenuNzQ/RBERkfYlxtDLvKK3gma2P7A5sH02ew13n2JmawN3mNnD7v5E1XojgBEA7k5PT89CBz01/a/e1uDBg5uy/Xbpb/FCd8Y8lfnfC7lujLmR/hZzt8Xb6D0B88Zcpnw36Kbj3Ndj3AztSoyTgNWz6dWAKdWFzGwn4CRge3efWZnv7lPS/yfN7C5gU2CexOjuo4BRabKYNm1a04Kv3lZPT89887pZf4sXujfmejF1a8z19LeYuzHeRvFUx9xt8fem245zX49xLUOGDCm1v3YlxnHAumY2DJgM7APslxcws02Bi4Hh7v6vbP4KwFvuPtPMeoBtiR1zREREmq4t1zG6+2zgKOA24NE4yyeY2elmVrn04lxgaeAaM/urmY1J8z8CjDezh4A7iecYH0FERKQF2nYdo7uPBcZWzTsle7xTjfX+CGzU2uhEREQijXwjIiKSUWIUERHJKDGKiIhklBhFREQySowiIiIZJUYREZGMEqOIiEhGiVFERCSjxCgiIpJRYhQREckoMYqIiGSUGEVERDJKjCIiIhklRhERkYwSo4iISEaJUUREJKPEKCIiklFiFBERyQwuW9DMdgb2AVZ2993NbHNgWXe/o2XRiYiItFmpGqOZHQ38FPgnsF2aPQM4o0VxiYiIdETZptRjgZ3cfSTwbpr3GPChlkQlIiLSIWUT4zLAc+lxkf4vDsxqekQiIiIdVDYx3gOcUDXvGODO5oYjIiLSWWU73xwN3GRmhwPLmNnjwGvA7i2LTEREpANK1Rjd/XlgC2BvYD/gQGArd3+hhbGJiIi0XenLNdy9AO5PfyIiIgNSqcRoZs8xt9PNPNx9jaZGJCIi0kFla4z7V02vCnwDGN3ccERERDqrVGJ097ur55nZXcCtwPlNjklERKRjFmas1JnAsGYFIiIi0g3KnmM8vWrW+4BdgVuaHpGIiEgHlT3HuHrV9JvAj4BfNDccERGRzip7jvHgVgciIiLSDWomRjPbscwGdNspEREZSOrVGC8tsX4BrN2kWERERDquZmJ0d/U4FRGRRU7pIeEWlpkNJ17zOAi4JN3bMV9+HHAYMBt4ETjE3Z9Jyw4ETk5Fz3D3y9sVt4iILFrKXq6xLPA9YHugBwiVZWWGhDOzQcCFwM7AJGCcmY1x90eyYg8Cm7v7W2Z2JHAOsLeZrQicCmxObLp9IK37cpnYRURE+qJsjfEnwGrA6cCVxCHi/gO4ruT6WwIT3f1JADMbDewJzEmM7p7f2/E+5g5D9xngdnefnta9HRgOXF1y3yIiIqWVHflmF+BL7n4j8E76vzdwQMn1hwLPZdOT0rxaDmXu4AF9XVdERGSBla0xLga8mh6/YWbLA88DHyy5fuhlXq936zCz/YnNptv3ZV0zGwGMAHB3enp6SoZW29T0v3pbgwcPbsr226W/xQvdGfNU5n8v5Lox5kb6W8zdFm+j9wTMG3OZ8t2gm45zX49xM5RNjA8RE9XvgHuJ5wvfAP5Rcv1JzDt6zmrAlOpCZrYTcBKwvbvPzNbdoWrdu6rXdfdRwKg0WUybNq1kaI1Vb6unp2e+ed2sv8UL3RtzvZi6NeZ6+lvM3Rhvo3iqY+62+HvTbce5r8e4liFDhpTaX9nEeDhza27HAD8Alge+WnL9ccC6ZjYMmAzsA+yXFzCzTYGLgeHu/q9s0W3AWWa2QpreBTix5H5FRET6pG5iNLP13f2RSqcZAHd/kXhZRWnuPtvMjiImuUHAZe4+IQ1OPt7dxwDnAksD15gZwLPuvoe7Tzez7xOTK8DplY44IiIizdaoxvigmf0N+Dkw2t1fWtAduftYYGzVvFOyxzvVWfcy4LIF3beIiEhZjRLjEGBfYpPpf5nZLcQkebO7z25xbCIiIm1XNzGmGuIFwAVm9iHi5Rk/Bi5J1yJe4e7j6m1DRESkPyk9JJy7P04clu1kM9uZ2AP0a8RzhiIiIgNCn8ZKNbOtic2qRryu8fRWBCUiItIpDROjma1JbEI9APgAcC3wBXe/t8WxiYiItF2jyzXuBrYB7gROA6539xntCExERKQTGtUYbwH2c/fJ7QhGRESk0xr1Sh1Zb7mIiMhAU/buGiIiIosEJUYREZGMEqOIiEim1HWMZvYgcDlwtbtPbVReRESkvyp7gf8ZwFeAM83sHuAX6NINEREZgEo1pbr7de7+ReLNhm8kDgX3vJldZmY7tjJAERGRdurTOcZ0H8QrgIuAZ4EvAaPM7B9mVvO2USIiIv1F2XOMiwE7E4eF2w34EzCS1JxqZl8CrgRWaVWgIiIi7VD2HOMUYBqxtvhtd5+SL3T368zsqGYHJyIi0m5lE+Nu7j6+XgF3/1QT4hEREemosucYf9PbTDP7VxNjERER6biyiXHx6hlmtji6SbGIiAwwjW47dS9QAEum6xdzqwF/bFVgIiIindDoHOMlQAC2AC7N5hfAVOCOFsUlIiLSEY1uO3U5gJnd5+6PtSckERGRzqmZGM3sAHf/RZr8uJl9vLdy7n5ZSyITERHpgHo1xn2JY6JCvLC/NwWgxCgiIgNGzcTo7rsCmFkADgWedffZ7QpMRESkExperuHuBfAw8G7rwxEREemsstcxPgis18pAREREukHZIeHuAm41s58DzxHPLQLqfCMiIgNL2cS4LfAUsH3VfHW+ERGRAaVUYtQA4SIisqgoW2OcI/VSDZVpd1enHBERGTDK3qh4KHABsB2wfNViDSQuIiIDRtleqRcBs4BPA28AHwPGAEe0KC4REZGOKJsYPw4c4u5/BQp3f4h40f/xLYtMRESkA8omxneAyqg3r5jZ+4E3gaEtiUpERKRDyibG+4Fd0+PbgF8BvwbGtyIoERGRTinbK/UA5ibRY4FvAUsDPy67IzMbDpxP7KxzibuPrFq+XdrexsA+7n5ttuwd4rB0EMds3aPsfkVERPqi7HWMr2SPZwDf78tOzGwQcCGwMzAJGGdmY9z9kazYs8BBxKRbbYa7b9KXfYqIiCyIevdjPL3MBtz9lBLFtgQmuvuTadujgT2BOYnR3Z9Oy3RdpIiIdEy9GuPqTdzPUOIYqxWTgK36sP6SZjae2AFopLvf0MTYRERE5qh3P8aDm7if0Mu8opd5tazh7lPMbG3gDjN72N2fyAuY2QhgBIC709PTs+DRJlPT/+ptDR48uCnbb5f+Fi90Z8xTmf+9kOvGmBvpbzF3W7yN3hMwb8xlyneDbjrOfT3GzVCvKXWtrHlz7VrlKs2jDUxi3hroasCUkjHi7lMq+zKzu4BNgSeqyowCRqXJYtq0aWU331D1tnp6euab1836W7zQvTHXi6lbY66nv8XcjfE2iqc65m6Lvzfddpz7eoxrGTJkSKn91WtKfRhYJj2eSKzhVdf8CsoNCTcOWNfMhgGTgX2A/coEaGYrAG+5+0wz6yHe6eOcMuuKiIj0Vb2m1GWyx2Wvd6y1rdlmdhTxGshBwGXuPiF18Bnv7mPMbAvgemAFYHczO83dNwA+AlycOuUsRjzH+EiNXYmIiCyUPt9dY0G5+1hgbNW8U7LH44hNrNXr/RHYqOUBioiIUP7uGmsApxLP7S2dL3P39VoQl4iISEeUrTFeAzwGnALMaF04IiIinVU2MX4Y2EY3JRYRkYGubKeam4DtWxmIiIhINyhbYzwG+KOZPcHc694BcPdDmh6ViIhIh5RNjD8j3pPxUXSOUUREBrCyiXFHYIi7v97KYERERDqt7DnGvwErtTIQERGRblC2xngH8Bsz+xnzn2O8rOlRiYiIdEjZxPgJ4hinu1TNLwAlRhERGTBKJUZ3/1SrAxEREekG9W47Fdy9SI9rnovURf8iIjKQ1Ksxvgosmx7PZv4bCwfK33ZKRESkX6iXGDfIHg9rdSAiIiLdoN79GJ/LHj+TLzOzFdz95VYGJiIi0gl1O9+Y2VeBqe5+W5renHgz4SFmNhHYw90fb32YIiIi7dHoAv/jgRey6VHAb4GN0/9zWxSXiIhIRzRKjGsADwOY2erARsDx7j4BOAHYqrXhiYiItFejxDgbWCI9/jjwmLtPT9NvAe9tVWAiIiKd0Cgx3g2caWYbA0cT78tY8WHmbWYVERHp9xolxm8AmwJ/INYQz86WHQDc2qK4REREOqJur1R3n0y85VRvy05oSUQiIiIdVPa2UyIiIosEJUYREZGMEqOIiEhGiVFERCRT9kbFmNnOwD7Ayu6+exoebll3v6Nl0YmIiLRZqRqjmR0N/BT4J7Bdmj0DOKNFcYmIiHRE2abUY4Gd3H0kULkx8WPAh1oSlYiISIeUTYzLAJXbUFVuWLw4MKvpEYmIiHRQ2cR4D3HQ8NwxwJ3NDUdERKSzyna+ORq4ycwOB5Yxs8eB14DdWxaZiIhIB5SqMbr788AWwN7AfsCBwFburkHERURkQCl9HaO7F+5+P3Ad8GcAM9N1kCIiMqCUako1s48BFwIbA0um2YHYEWdQa0ITERFpv7LnGC8n3ovxEOLtp0RERAaksolxTeAkdy8alqzBzIYD5xNrmJekayLz5dsBPybWSvdx92uzZQcCJ6fJM9z98gWNQ0REpJ6y5wivB3ZZ0J2Y2SBiU+xngfWBfc1s/apizwIHAb+sWndF4FRgK2BL4FQzW2FBYxEREamnbI1xSeB6M/s9ME9PVHf/aon1twQmuvuTAGY2GtgTeCTbztNp2btV634GuN3dp6fltwPDgatLxi4iIlJa2cT4CFkSWwBDmTtyDsAkYg1wQdcduhCxiIiI1FQqMbr7aQu5n9DLvLLnK0uta2YjgBEA7k5PT0/56GqYmv5Xb2vw4MFN2X679Ld4oTtjnsr874VcN8bcSH+LudvibfSegHljLlO+G3TTce7rMW6GmonRzLZz93vS4x1rlSt526lJwOrZ9GrAlJIxTgJ2qFr3rl7iGAWMSpPFtGnTSm6+sept9fT0zDevm/W3eKF7Y64XU7fGXE9/i7kb420UT3XM3RZ/b7rtOPf1GNcyZMiQUvurV2P8CbBhenxpjTIFsHaJ/YwD1jWzYcBk4n0d9ysVIdwGnJV1uNkFOLHkuiIiIn1SMzG6+4bZ42ELsxN3n21mRxGT3CDgMnefYGanA+PdfYyZbUHs/boCsLuZnebuG7j7dDP7PjG5Apxe6YgjIiLSbGU738zDzD4FzHb3e8uu4+5jgbFV807JHo8jNpP2tu5lwGULEquIiEhflLqO0czuNrNt0+PvAKOB0Wb23VYGJyIi0m5lL/DfELgvPT6c2Blma+CIFsQkIiLSMWWbUhcDCjNbBwju/iiARqAREZGBpmxi/D1wAbAqsYMMKUl2T39eERGRJijblHoQ8ArwN+B7ad6HiYOCi4iIDBhlR755Cfhu1bybWxKRiIhIB5W9UfHixNs+HQAMIY5a8wvgTHef1brwRERE2qvsOcZziHfIOAJ4hnh/xv8ElgW+2ZrQRERE2q9sYtwL+GhqUgV43Mz+AjyEEqOIiAwgZTvf9HaHi3rzRURE+qWyNcZrgJvM7DTgWWJT6smAtyowERGRTiibGL9NTIQXEjvfTCYOC3dGi+ISERHpiLKXa8wCTkl/IiIiA1bdc4xmtq2ZnV1j2Ugz27o1YYmIiHRGo8433wXuqbHsbuCk5oYjIiLSWY0S4ybArTWW3Q5s1txwREREOqtRYlwWWKLGssWBZZobjoiISGc1SoyPAbvUWLZLWi4iIjJgNOqVeh5wsZkNAm5w93fNbDHg88RLN45rdYAiIiLtVLfG6O6/JI6TejnwtplNAd4Gfg6c4+5XtzxCERGRNmo4JJy7/wgYCuwOfCv9X83dz2txbCIiIm1X9gL/14DbWhyLiIhIx5UdRFxERGSRoMQoIiKSUWIUERHJKDGKiIhklBhFREQySowiIiIZJUYREZGMEqOIiEhGiVFERCSjxCgiIpJRYhQREckoMYqIiGSUGEVERDJKjCIiIhklRhERkUyp+zE2g5kNB84HBgGXuPvIquXvAa4ANgNeAvZ296fNbC3gUeDxVPQ+dz+iXXGLiMiipS2J0cwGARcCOwOTgHFmNsbdH8mKHQq87O4fNLN9gLOBvdOyJ9x9k3bEKiIii7Z2NaVuCUx09yfdfRYwGtizqsyewOXp8bXAp80stCk+ERERoH1NqUOB57LpScBWtcq4+2wzexVYKS0bZmYPAq8BJ7v7vdU7MLMRwIi0Pj09PQsd9NT0v3pbgwcPbsr226W/xQvdGfNU5srF1IoAAA8ASURBVH8v5Lox5kb6W8zdFm+j9wTMG3OZ8t2gm45zX49xM7QrMfZW8ytKlnkeWMPdXzKzzYAbzGwDd38tL+juo4BRlfWmTZu2sDHPUb2tnp6e+eZ1s/4WL3RvzPVi6taY6+lvMXdjvI3iqY652+LvTbcd574e41qGDBlSan/takqdBKyeTa8GTKlVxswGA8sB0919pru/BODuDwBPAOu1PGIREVkktavGOA5Y18yGAZOBfYD9qsqMAQ4E/gR8GbjD3Qszez8xQb5jZmsD6wJPtiluERFZxLSlxujus4GjgNuIl164u08ws9PNbI9U7FJgJTObCBwHnJDmbwf8zcweInbKOcLdp7cjbhERWfS07TpGdx8LjK2ad0r2+G1gr17Wuw64ruUBioiIoJFvRERE5qHEKCIiklFiFBERySgxioiIZJQYRUREMkqMIiIiGSVGERGRjBKjiIhIRolRREQko8QoIiKSUWIUERHJKDGKiIhklBhFREQySowiIiIZJUYREZGMEqOIiEhGiVFERCSjxCgiIpJRYhQREckoMYqIiGSUGEVERDJKjCIiIhklRhERkYwSo4iISEaJUUREJKPEKCIiklFiFBERySgxioiIZJQYRUREMkqMIiIiGSVGERGRjBKjiIhIRolRREQko8QoIiKSUWIUERHJKDGKiIhkBrdrR2Y2HDgfGARc4u4jq5a/B7gC2Ax4Cdjb3Z9Oy04EDgXeAY5x99vaFbeIiCxa2lJjNLNBwIXAZ4H1gX3NbP2qYocCL7v7B4HzgLPTuusD+wAbAMOBn6TtiYiINF27mlK3BCa6+5PuPgsYDexZVWZP4PL0+Frg02YW0vzR7j7T3Z8CJqbtiYiINF27EuNQ4LlselKa12sZd58NvAqsVHJdERGRpmjXOcbQy7yiZJky62JmI4ARAO7OkCFD+hrj/G4eX3NRU7bfRv0tXujCmOu8Hyq6LuYS+lvMXRVvifcEZDGXLN8NuuY49/UYN0G7aoyTgNWz6dWAKbXKmNlgYDlgesl1cfdR7r65u29OTKYt+zOzB1q9j0U5XsWsmAdKvIq5K+NtqF01xnHAumY2DJhM7EyzX1WZMcCBwJ+ALwN3uHthZmOAX5rZj4AhwLrAn9sUt4iILGLaUmNM5wyPAm4DHo2zfIKZnW5me6RilwIrmdlE4DjghLTuBMCBR4Bbga+7+zvtiFtERBY9bbuO0d3HAmOr5p2SPX4b2KvGumcCZ7Y0wL4Z1ekA+qi/xQuKuV36W8z9LV5QzO3Q1HhDUczXj0VERGSRpSHhREREMm1rSu0vzGx14tB0qwDvAqPc/XwzWxH4FbAW8DRg7v5yGoTgfGBX4C3gIHf/S4diHwSMBya7+26ps9NoYEXgL8AB7j6r3vB7bYx1eeASYEPi5TeHAI/TxcfYzL4JHJbifRg4GFiVLjrGZnYZsBvwL3ffMM3r83vXzA4ETk6bPcPdL29zzOcCuwOzgCeAg939lbSs1yEiGw072eqYs2XfAs4F3u/u07rhONeK18yOJvb/mA3c7O7fTvO78hib2SbARcCSKeavufufm32MVWOc32zgeHf/CLA18PU0LN0JwO/cfV3gd2ka4jB366a/EcBP2x/yHN8gdm6qOBs4L8X8MvGNDjWG32uz84Fb3f3DwEeJcXftMTazocAxwObpQzqI2Lu6247xz4lDJ+b6dFxTIj0V2Io4ytSpZrZCm2O+HdjQ3TcG/gGcmGLrdYjIksNOtjrmyg/rnYFns9ndcJzni9fMPkUcWWxjd98A+GGa383H+BzgNHffBDglTUOTj7ESYxV3f77yS8PdXyd+YQ9l3iHrLgc+nx7vCVzh7oW73wcsb2artjlszGw14HPEWhjpF9SOxOH1YP6YK88lH36vXbEuC2xH7ImMu89KtYGuPsbEFpb3puts3wc8T5cdY3e/h3j9b66vx/UzwO3uPt3dXyYmqfmSQCtjdvffpN7sAPcRr1+uxNzbEJFlhp1saczJecC3mXcQko4f5xrxHgmMdPeZqcy/sni79RgXwLLp8XLMvaa9qcdYibEOM1sL2BS4H/iAuz8PMXkCK6di3TJk3Y+JH8h30/RKwCvZl0seV63h99plbeBF4Gdm9qCZXWJmS9HFx9jdJxN/UT9LTIivAg/Qvcc419fj2vHjXeUQ4Jb0uGtjTpeeTXb3h6oWdWvM6wGfNLP7zexuM9size/WeAGOBc41s+eIn8cT0/ymxqzEWIOZLQ1cBxzr7q/VKdpbLaCtXX3NrNIO/0A2u15cnY55MPAx4KfuvinwJnOb93rT6XhJzS97AsOIA00sRWy+qRVXx2MuoVaMXRO7mZ1EPL1xVZrVlTGb2fuAk4jNe9W6Mmbi53AF4imj/wA8tWp0a7wQa7nfdPfVgW+SWp1ocsxKjL0ws8WJSfEqd/91mj210nyX/leaHUoNWddi2wJ7mNnTxOaNHYk1yOVTs191XLWG32uXScAkd78/TV9LTJTdfIx3Ap5y9xfd/d/Ar4GP073HONfX49oNx7vSaWI34CvuXvky69aY1yH+aHoofQ5XA/5iZqvUia3TMU8Cfp2aH/9MbG3qqRNXp+OFODpa5Tv5GubeaampMatXapX0i+lS4FF3/1G2qDJk3cj0/8Zs/lFmNpp4gvfVSrNVu7j7icztnLAD8C13/4qZXUMcXm90LzHPN/xeG+N9wcyeM7MPufvjwKeJIxs9QpceY2IT6tapZjAjxTweuJMuPMZV+vTeNbPbgLOyTgq7MLfJqi1S78fvANu7+1vZolpDRAYaDzvZMu7+MHObqEnJcfPUK7Vbj/MNxB/Rd5nZesASwDS69BgnU4DtgbtS7P9M85t6jJUY57ctcADwsJn9Nc37LvFLxc3sUOKXZGWUnrHELsITid2ED25vuHV9BxhtZmcADzK32eFS4Bdp+L3pxDd4ux0NXGVmSwBPEo/bYnTpMXb3+83sWuIlGbOJx3MUcDNddIzN7GpgB6DHzCYRe+T16b3r7tPN7PvEMY4BTnf3ltV2a8R8IvAe4HYzA7jP3Y/wOJRkZYjI2WRDRJpZZdjJQcBlHoeTbFvM7n5pjeIdP841jvFlwGVm9nfiZTEHph9vXXuMgcOB81MrzNukOyrR5GOskW9EREQyOscoIiKSUWIUERHJKDGKiIhklBhFREQySowiIiIZJUaRDjGzwsw+2ORtftLMHm/mNkUWNbqOUbpeulj6A8Rb4LxJvGbpaHd/o8NxHQQc5u6f6GQcOXe/F/jQgqybns+lxAEMII5nexfwA3f/R8lt/Jw4qtHJjcrWWP97xKHVZmazT3f3c3pfQ6T5VGOU/mJ3d1+aOHTcFsy9v1pbZMO+DXR/Ssd5OeIweDOAB8xsw/qrNdWv3H3p7G++pGhmwcz0/SUtsah82GWAcPfJZnYL8QbHmNnBxLuKrEas4Zzt7henZTsAVwI/AY4D3gBOcver0vL3AGcCRhxl5XriAMUzsnX/hzhY8e3EEZFqSjXbC4CvAmsCtxJHE3k7Lf+PFEdBVWJvEMt3gC8An3D32WZ2JPHmsptVtp1tZwfgSndfrUxMtaSRTp4AvmZmawDfIw5tRxpq8JPAe4GHgCPTiDQjgK8AhZkdC9zp7rub2QnEEUtWJt7p4CR3v77e/ntjZncBfyCOhvIxYCMz+ySNX///Br5FbHE4kjjKy4+J44L+0N3PSuUXS9s6HFieeO/KI1o56o90J/3ikn7F4o1gdyUOvwZxQOzdiPdoOxg4z8w+lq2yCvELcChxnNBRZlZpajybeOudTYAPpjKnVK27IjGhjKAcI97vbRiwMXBQins48ct5Z+LYkztVrVcvlnOJX+Ynm9m6wFnA/o2SW6OY+uDXxERYcUt6DisTh8i7CsDdR6XH56Sa3u6p/BNp/eWA04ArbcHvp3kA8bVYBniGcq//ksw9nv8L7A9slmI6xczWTmWPId6rcnviGKEvE2/MK4sY1Rilv7jBzCr3NbyZmBxw95uzMneb2W+IX3h/yeb/Z7oZ691mdjNgaWzTw4l3L59OnHkW8EvmDjL8LnEMzPx8VyP/7e5T0vZuIiY6iMnpZ+7+97Tse8C+6XGoF4u7v2tmX03PaW9i4nmQ8mrFVNYU4g8EANz9ssrj9DxeNrPl3P3V3lZ292uyyV+Z2YnEuyLc2Fv5uFnbLZtevxI/8POq8Tkbvf7/Bs5093fSANOjgPM93oR8gplNIP5YeBL4f8BR7j4pe27PmtkB2T03ZRGgxCj9xefd/bfVM83ss8TBhdcjtoC8D3g4K/Kyu7+ZTT9DrA28P5V9IA1SDfHuAYOysi/2oVZW8UL2+K20L9L//H6Zz2SPG8bi7k+b2Z3E2nJfazG1YiprKOmWWWY2iNjku1eKu3Jj7B7ij5b5pKR+HLBWmrV0Kl+Lu/v+NZblN50t8/q/VBkAm7mdiqZmy2ekeCC2DFxvZu9my98hdvyaXCdeGWCUGKXfSuflriOeP7vR3f9tZjcw781JVzCzpbLkuAbwd+LtdWYAG7h7rS+9Zo6w/zzz3hdujexxw1jMbFdgG+J5r3OJtZt2+QJwb3q8H/GGzTsBTxObR19m7jGf55iZ2ZrE5stPEzv2vJPuWtPbDWTLmLP9kq9/XzwHHOLuf1jA9WWA0DlG6c+WIHZUeRGYnWoPu/RS7jQzWyJ11NgNuMbd3yV+YZ9nZisDmNlQM/tMi2J14CAzW9/iPR1PnbOgQSxm1kO8jOIw4nnS3VOibBkzG2Rmw8zsf4idXU5Li5YhXkrxErF2dlbVqlOBtbPppYjJ7MW03YNJHaeaoOzrX9ZFwJkpmWNm7zezPRc+TOlvlBil30rniY4hJp2XibWZMVXFXkjLphA7hhzh7o+lZd8h3r/tPjN7DfgtC3gNYIlYbyH2hLwj7fOOqiL1YhlFrBGNdfeXgEOBS8xspRaEuo2ZvQG8RryGcVlgi3QjXoAriM3Ak4n367uvav1LgfXN7BUzu8HdHwH+i3jD5qnARsSepQut5OvfF+en9X9jZq8Tn9tWCxun9D+6H6MMWNWXLoiIlKEao4iISEaJUUREJKOmVBERkYxqjCIiIhklRhERkYwSo4iISEaJUUREJKPEKCIiklFiFBERyfx/+I1egy2kQ/IAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 504x360 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "if (flag == False):\n",
    "    values = [] # a list which will hold the cosine value of each paper outputted\n",
    "    for i in answer_indicies:\n",
    "        values.append(cosineSimilarities[i])\n",
    "    print(values)\n",
    "    print(answer_indicies)\n",
    "    plt.figure(figsize=(7, 5))\n",
    "    plt.bar(answer_indicies, values,width=6)\n",
    "    plt.xlabel('Paper Index in Data Frame')\n",
    "    plt.ylabel('Cosine Similarity Value')\n",
    "    plt.title('Top Papers Retreived Cosine Similarity')\n",
    "    plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<p>In the below cell, the code verifies whether or not the cosine similarity is >=0.1.  </p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>766</td>\n",
       "      <td>Comparison of alternative models of human movement and the spread of disease</td>\n",
       "      <td>[{'text': 'Predictive models for the spatial spread of infectious diseases has received much attention in recent years as tools for the management of infectious diseas outbreaks. Prominently, various versions of the so-called gravity model, borrowed from transportation theory, have been used. However, the original literature suggests that the model has some potential misspecifications inasmuch as it fails to capture higher-order interactions among population centers. The fields of economics, geography and network sciences holds alternative formulations for the spatial coupling within and among conurbations. These includes Stouffer's rank model, Fotheringham's competing destinations model and the radiation model of Simini et al. Since the spread of infectious disease reflects mobility through the filter of age-specific susceptibility and infectivity and since, moreover, disease may alter spatial behavior, it is essential to confront with epidemiological data on spread. To study their relative merit we, accordingly, fit variants of these models to the uniquely detailed dataset of prevaccination measles in the 954 cities and towns of England and Wales over the years 1944-65 and compare them using a consistent likelihood framework. We find that while the gravity model is a reasonable first approximation, both Stouffer's rank model, an extended version of the radiation model and the Fotheringham competing destinations model provide significantly better fits, Stouffer's model being the best. Through a new method of spatially disaggregated likelihoods we identify areas of relatively poorer fit, and show that it is indeed in densely-populated conurbations that higher order spatial interactions are most important. Our main conclusion is that it is premature to narrow in on a single class of models for predicting spatial spread of infectious disease. The supplemental materials contain all code for reproducing the results and applying the methods to other data sets.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'The ability to predict how infectious disease will spread is of great importance in the 1 face of the numerous emergent and re-emergent pathogens that currently threatening 2 : bioRxiv preprint human well-being. We identified a variety of alternative models that predict human 3 mobility as as a function of population distribution across a landscape. These consider 4 some models that account for pair-wise interactions between population centers, as well 5 as some that allow for higher-order interactions. We trained the models using a 6 uniquely rich spatiotemporal data set on pre-vaccination measles in England and wales 7 (1944-65), which comprises more than a million records from 954 cities and towns.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Likelihood rankings of the different models reveal strong evidence for higher-order 9 interactions in the form of competition among cities as destinations for travelers and, 10 thus, dilution of spatial transmission. The currently most commonly used so-called 11 'gravity' models were far from the best in capturing spatial disease dynamics.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': '12 42 movement patterns [13] including a recent incarnation as the 'radiation' model [14].', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'While these models have proved useful in their original applications, their use in 44 describing infectious disease spread is complicated by two considerations. First, spatial 45 interaction models predict bulk migration fluxes between population centers, but the 46 impact of these movements on the dynamics of infectious disease can depend strongly 47 not only on the magnitude, but also on the composition, of the migrant pool. In 48 particular, the extent to which migrants are more or less likely to be susceptible to 49 infection, or actually infected, than the general population can be critically important, 50 as can the age profile of the migrant pool, due to pronounced age-specific patterns of 51 behavior, susceptibility, and infectiousness [5, 7, 15] . Second, infectious disease dynamics 52 and Wales using a spatial hazards approach [18] . We considered four main classes of 268 models, including the gravity, competing destinations, Stouffer rank, and radiation 269 models, in addition to pure diffusion and mean field models which are interesting special 270', 'cite_spans': [{'start': 754, 'end': 757, 'text': '[5,', 'ref_id': 'BIBREF4'}, {'start': 758, 'end': 760, 'text': '7,', 'ref_id': 'BIBREF6'}, {'start': 761, 'end': 764, 'text': '15]', 'ref_id': 'BIBREF13'}, {'start': 849, 'end': 853, 'text': '[18]', 'ref_id': 'BIBREF16'}], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>Accurately predicting the geographical spread of emerging, re-emerging, and recurrent 14 epidemics in an increasingly globalized world is a matter of international urgency in the 15 wake of outbreaks of emerging pathogens such as severe acute respiratory syndrome 16 (SARS), Middle East respiratory syndrome (MERS), and ebola virus disease (EBVD) as 17 well as re-emerging and resurgent pathogens such as influenza subtype A-H1N1, 18 whooping cough, and measles. In response to this challenge, various 'spatial interaction' 19 models describing human movement as a function of population distribution have been 20 proposed. Some of these borrow from economics and human geography while others 21 adapt models from movement ecology and the physics of reaction and diffusion on 22 heterogenous landscapes. 23 In recent years, the field has seen the widespread adoption of a family of so-called 24 'gravity' models from transportation theory and human geography (see, e.g., [1, 2] ). In 25 its most common form, the gravity model posits that the migration flux between a pair 26 of cities is log-linearly dependent on their respective sizes and on their separating 27 distance. The application of this simple model to disease spread was originally proposed 28 by Murray and Cliff [3] , but over the last decade, many studies have used it to explain 29 historical, or predict future, disease spread in a range of infections including measles [4] , 30 influenza [5] [6] [7] , cholera [8] , and yellow fever [9] . While its application has yielded 31 insights, the geography literature has highlighted a prominent shortcoming of the 32 gravity model. In particular, these models ignore the potential for competitive or 33 synergistic interactions among population centers ( Fig. 1 ; [10, 11] ). Thus, for example, 34 movement between Boston and Washington DC is assumed to be unaffected by the 35 presence of the intervening city of New York. 36 As it happens, there are distinct families of models, arising from economics and 37 geography, that predict human movement patterns while allowing for higher-order 38 interactions among cities. In particular, Stouffer's [12] 'law of intervening opportunities' 39 posits that \"the number of persons going a given distance is directly proportional to the 40 number of opportunities at that distance and inversely proportional to the number of 41 intervening opportunities\". This idea has given rise to alternative models for human can feed back onto migration. Most obviously, disease symptoms can influence 53 movement behavior, as seen, for example, in the fact that a mild cold may induce 54 minimal changes in movement behavior but severe hemorrhagic fever or acute paralytic 55 disease will typically slow the movement of the infected hosts. These two considerations 56 do not preclude the utility of spatial interaction models in the infectious disease context; 57 they do, however, complicate their use and the importance of understanding 58 transmission relevant migration fluxes, which will be some sort of 'effective average' of 59 movement as filtered through such aforementioned complications.\\nSeveral spatial interaction models have been parameterized and tested using various 61 mobility data such as commuter flows (e.g., [5, 14] ), mobile phone geolocations 62 (e.g., [16] ), social media (e.g., [13] ), and microsimulations (e.g., [17] ). However, in view 63 of the challenges just noted, the ultimate test of the models is against data on the actual 64 spread of infection rather than bulk movement of people or cell phones. Bjørnstad and 65 Grenfell [18] proposed that for acute immunizing infections, the spatio-temporal 66 patterns of fade-outs (i.e., local disease extinction) across metapopulations provide 67 valuable information on disease spread because these patterns reflect spatial 68 transmission unclouded by local transmission. In this paper, we extract information 69 from fade-out patterns and a consistent likelihood framework to compare and contrast 70 to a suite of models including (i) the gravity model [1] , (ii) Fotheringham's competing 71 destinations model [10] , (iii) Stouffer's rank model [12] , and (iv) the radiation model [14] . 72 . We confront these models weekly data on measles incidence from all 954 cities and 73 towns in England and Wales from 1944 to 1965 [4, 19] . Comparing fits and predictions, 74 we show that while the gravity model is a reasonable first approximation, Stouffer's 75 rank model, an extended version of the radiation model, and the competing destinations 76 model all provide significantly better fits, Stouffer's model performing the best.\\nData 79 Historical incidence of measles in England and Wales has been an influential testbed for 80 models and methods in disease dynamics since Bartlett's [20, 21] seminal work on its 81 recurrent epidemics. We use the spatially resolved weekly measles data across all 954 82 cities and towns of England and Wales from 1944, when notification was made 83 mandatory by the UK Registrar General (OPCS), until 1965, which saw subtle shifts in 84 political boundaries around London. Vaccination was not introduced in the UK until 85 1967, so that these data span a period where measles dynamics were unaffected by mass 86 immunization. The data set is complete except for a region-wide underreporting rate of 87 around 50% [22, 23] . Grenfell et al. [19] give a detailed description of the data; the 88 entire data set has been made available by Lau et al. [24] . 89 An important feature of the system is that, between 29% and 38% of the population 90 (c. 47M during this period) resided in a small number (13-28 depending on exact 91 definition) of communities above a critical community size (CCS) of c. 250-300k. Cities 92 larger than the CCS tend to sustain local chains of transmission. The remaining 60-70% 93 were distributed among the more than 900 communities smaller than the CCS where 94 local extinctions are more or less frequent (depending on population size and degree of 95 isolation) and, consequently, the rate of reintroduction of the pathogen via spatial 96 transmission is an important determinant of measles incidence. In ecological terms, the 97 prevaccination measles system represented an epidemic mainland-island metapopulation 98 (e.g., [25, 26] ). Our analysis exploits this fact, using the timing and spatial pattern and 99 timing of reintroductions to inform the parameters of each of the spatial interaction 100 models.\\nUsing a spatially-extended time series susceptible-infected-recovered (TSIR) framework 103 (e.g., [4, 27] ), we can calculate, for each population center, the probability that a spatial 104 interaction happens (i.e., a contact between a resident susceptible host and a 105 non-resident infectious host) and that the contact results in a new local chain of 106 transmission [18] . In previous analyses, we showed that the signature of such events will 107 be drowned out in the presence of endemic circulation (see [23] ). Following a local 108 extinction, however, the re-colonization rate contains critical information on the spatial 109 interactions.\\nThe spatially-extended TSIR model predicts that the expected number of new 111 infected hosts in community i in epidemic generation t + 1 will be\\nHere, β s(t) is the seasonal transmission rate as molded by opening and closing of schools 113 and the proportion of transmission that occurs within the school setting as the school 114 year progresses [23] ; s(t) = t mod 26 is the seasonality function; S i,t /N is the 115 probability that a local individual is susceptible; I i,t is the local number of infections; α 116 is a correction for the discrete-time approximation of the underlying continuous-time 117 process [28] ; and ι i,t is the sum of transmission-relevant spatial interactions i has with 118 the other communities in the epidemic metapopulation. Following a local extinction,\\nAssuming demographic stochasticity in transmission due to the underlying epidemic 121 birth-and-death process, the realized number of cases in the next generation will be [23] : 122\\nAs a consequence, the number of susceptibles in the next generation will be\\nwhere B i,t is the recruitment of local susceptibles through births during the generation 124 interval. The 'serial interval' for measles is 10-14 days [29] , so we follow previous TSIR 125 analyses in aggregating the weekly data in 2-week increments (though [23] investigate 126 shorter intervals). Using susceptible reconstruction methods [30] , we estimate all local 127 parameters using the method of Finkenstadt et al. [31] . The online supplement contains 128 full documentation of these analyses and code for exact replication of all results 129 presented in this paper.\\nConditioning on the local parameters and restricting analyses at each location to 131 biweeks in which measles incidence was zero, we use Eqs. 2 and 4 to form a negative 132 binomial likelihood to estimate all parameters of each of the candidate spatial 133 interaction models.\\nSpatial interaction models 135 When including special cases, such as pure diffusion and variants, we consider a total of 136 10 spatial interaction models (Fig. 1) . Each of these amounts to a specification of the ι 137 term in Eqs. 1 and 2.\\nGravity model. Under the gravity model, the spatial interaction between locations i 139 and j takes the form θN τ1 i N τ2 j d −ρ ij , where θ, τ 1 , τ 2 , and ρ are non-negative parameters. 140 This leads to the following formulation for disease-relevant spatial interactions:\\nwhere the sum is across all non-self potential donor communities, and I j /N j is the 142 fraction of infected individuals in donor community j. The gravity model has two 143 important special cases: ρ = 0, τ 1 = τ 2 = 1, which is a mean-field model, and 144 τ 1 = τ 2 = 0, which is simple spatial diffusion. The φ parameter represents background 145 spatial transmission that is not predictable on the basis of distance and size [24] . 146 Xia's model. In the original analysis of the spatiotemporal dynamics of measles, Xia et al. [4] used a formulation rooted in the gravity-model literature but with a slightly different formulation:\\nwhich is the same as Eq. 5 if τ 2 = 1, but not otherwise.\\nCompeting destinations model. Fotheringham [11] noted that gravity models may be misspecified because they only consider pairwise interactions among locations. He argued that nearby or intervening destinations may, in fact, make spatial interactions either more or less likely. A synergistic effect occurs, for example, if individuals are disproportionately inclined to do all their shopping in districts with many shops. Antagonistic effects may arise where travel between two cities, say Boston and Washington, is made less likely by the presence of an intervening city such as New York because individuals from either Boston or Washington can get their out-of-town needs fulfilled by visiting New York. Under Fotheringham's [11] competing destinations formulation the flux between i and j takes the form θ N τ1\\nAs in the gravity model, τ 1 and τ 2 control how 'eagerness to travel' and city 'attractiveness' scales with population size; and ρ measures how the likelihood of travel decays with distance. The parameter δ quantifies how destinations k, of various sizes, at distances d jk from the donor j, modulate the spatial interaction between recipient i and and donor j. In particular, δ &gt; 0 indicates a synergistic effect; δ &lt; 0, an antagonistic effect. The resultant formulation for disease-relevant spatial interactions between community i and everywhere else is:\\nwhere φ again represents background, spatially unpredictable spread. 148 Stouffer's rank model. Noulas et al. [13] recently proposed that human mobility 149 patterns should be studied using Stouffer's [12] rank (\"law of intervening author/funder. This article is a US Government work. It is not subject to copyright under 17 USC 105 and is also made available for use under\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2019.12.19.882175 doi: bioRxiv preprint population size as a proxy for \"opportunities\". Accordingly, we let Ω(i, j) be the 155 collection of towns located closer to town i than is town j:\\n156 The Stouffer model for disease-relevant spatial   157 interactions is, then\\nStouffer's original model was framed in the context of continuous distribution of 159 population across a landscape. In the present metapopulation context, we therefore 164\\nExtended radiation model. The radiation model proposed by Simini et al. [14] was derived independently but is in spirit related to the original ideas of Stouffer [12] . Our metapopulation version of this model with respect to disease-relevant spatial interactions is\\nAs with the Stouffer model, we consider the set Ω(i, j) to either exclude ('radiation\") or 166 include ('radiation variant\") community size i itself in the spatial interaction \\nOf the models considered, the gravity variants (including mean-field and pure 174 diffusion) include purely pairwise interactions whilst the other models allow for 175 higher-order interactions either explicitly (as in the competing destination model) or 176 implicitly (as in the Stouffer and radiation models).\\nStatistical inference 178 We use maximum likelihood to estimate the parameters of each spatial interaction 179 model and profile likelihoods to study correlation among parameters and possible 180 identifiability issues [27] . author/funder. This article is a US Government work. It is not subject to copyright under 17 USC 105 and is also made available for use under\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . Table 1 . MLE estimates. Models: SV -Stouffer variant, XR -extended radiation, CD -competing destinations, S -Stouffer, G -gravity, X -Xia, RV -radiation variant, R -radiation, MF -mean field, D -diffusion. Here, denotes the log likelihood,ĉ is the computed variance inflation factor, ∆QAIC is the QAIC relative to that of the best-fitting model, and the other parameters are as in the model equations. biweek of local disease absence [18] ) All calculations are documented in the online 189 supplement, which includes code for the detailed replication of all results.\\n190 Table 1 shows estimated parameters and likelihoods for each of the spatial interaction networks. To explore the drivers among model differences in fit, we therefore contrast 214 model predictions of disease import and export rates. Fig. 2 shows the mean predicted 215 export and import rates vs. population size for each of the models. The dependence of 216 mean export rate on population size contrasts somewhat among models ( Fig. 2A) , but 217 the importation rates follow similar patterns (Fig. 2B) . Despite their differences in 218 underlying mathematical equations, there must be more nuanced differences among the 219 models that contribute their relative fit/lack-of-fit. This begs the question: \"What are 220\\nDecember 16, 2019 7/15 a CC0 license.\\nauthor/funder. This article is a US Government work. It is not subject to copyright under 17 USC 105 and is also made available for use under\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2019.12.19.882175 doi: bioRxiv preprint these differences and are they biologically meaningful or spatially random?\" 221 A cursory review of the recent literature suggests that the gravity model is the most 222 prominent in the infectious disease context, followed by the radiation model. Stouffer's 223 model and the competing destinations model have rarely been applied in this field. It is 224 therefore useful to use the gravity model fit as a baseline and explore how the 225 better-fitting models diverge from this baseline. Comparing matrices with half a million 226 entries is very difficult, so we amploy a new 'spatial likelihood contrast' (SLiC) method 227 with which to study the relative merit of the different spatial interaction models. The 228 idea is to disaggregate the overall hazard likelihoods by individual cities to study how 229 particular communities contribute to improved or diminished relative fit of each model. 230 To do so we normalize each location's contribution to the overall likelihood by the 231 number of data points each conurbation contributes to the likelihood (the accumulated 232 stretches of measles absence) and then map the model-model differences onto the 233 landscape (Fig. 3) . Inspection of Fig. 3 suggests that the poorer fit of the gravity model 234 vs. the Stouffer or competing destinations models is primarily in the urban northwest. 235 The supplement provides the full set of pairwise SLiC contrasts among all models. To 236 test whether the apparent patterns are statistically significant, we compute local 237 indicators of spatial association (LISA; [34] ) statistics for all population centers with 238 fewer than 50k inhabitants (Fig. 3A, B) . The main failure of the gravity model is in (e.g. [35, 36] ). In the case of epidemics playing out within a population center, models 254 closer to the mean field model (i.e., with uniform connectivity) often do remarkably well 255 (see [37] for a social network perspective on this). However, such models generally do 256 not capture regional spread of human infections because of the complicated patterns of 257 human movement (the third wave of the 2009 influenza pandemic across the USA, 258 perhaps, being an unusually diffusive counterexample [7] ) . To elucidate these patterns, 259 many empirical studies have been performed and a variety of abstractions have been 260 proposed. The task of choosing among these abstractions for help in help predicting 261 spread of infectious disease is complicated by the fact that (i) model formulations may 262 differ with respect to how well they actually reproduce aggregate mobility patterns, and 263 (ii) model fit may be shaped by the manner in which mobility is filtered by transmission 264 and behavior.\\nTo advance the discussion of these issues, we considered a suite of candidate models 266 of spatial interactions and fit them to the pre-vaccination measles data from England cases. The gravity model-which has gained recent prominence in infectious disease As the gravity model has had a very rapid adoption in spatial disease epidemiology 284 during the last decade (though Noulas et al. [13] have noted the importance of 285 considering alternatives), we used this as a baseline. Despite its common usage it 286 significantly under-performs relative to the several other models. Combining SLiC and 287 LISA statistics we found that the greatest tension between the gravity model and the 288 preferred alternatives is in the Manchester-Liverpool-Leeds conurbation of northwestern 289 England, where many towns and villages commingle amongst several major cities. formulation) that spatial interaction, once discounted for higher-order interference, is 295 roughly proportional to the geometric mean of the community sizes. This curious 296 suggestion warrants future investigation.\\nIn this study we have used actual disease incidence data, as opposed to raw 298 measurments of human movement, to further our understanding of the principles 299 governing how human hosts spread ifectious disease across a populated landscape. We 300 have found both strong evidence of higher-order interactions among population centers, 301 and that candidate models differ significantly in their ability to capture the empirical 302 patterns. We hope our findings will help stimulate a systematic, data-driven discussion 303 of the relative merit of alternative predictive models for the probable path for spatial 304 spread of infectious disease. Clearly, the prospects are good for good for further 305 refinements and improved parameterizations. Spatial interaction models predict the flux of human movements between population centers (cities, towns, villages) as a function of the distribution of the population on the landscape. In this diagram, the relative magnitude of the fluxes from a focal town to other population centers are represented by the widths of the arrows. In the widely-employed gravity models (A), interactions among cities is strictly pairwise. Thus the addition of a new town (B) has no effect on the movement to other towns. In Fotheringham's competing destinations model (C), however, competition or synergy among nearby communities can reduce or augment fluxes. Stouffer's model of intervening opportunities and the radiation model (D) posit that movement from one city to another is diminished by the presence of opportunities in communities more proximal to the source city. author/funder. This article is a US Government work. It is not subject to copyright under 17 USC 105 and is also made available for use under\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2019.12.19.882175 doi: bioRxiv preprint\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                            title  \\\n",
       "766  Comparison of alternative models of human movement and the spread of disease   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        abstract  \\\n",
       "766  [{'text': 'Predictive models for the spatial spread of infectious diseases has received much attention in recent years as tools for the management of infectious diseas outbreaks. Prominently, various versions of the so-called gravity model, borrowed from transportation theory, have been used. However, the original literature suggests that the model has some potential misspecifications inasmuch as it fails to capture higher-order interactions among population centers. The fields of economics, geography and network sciences holds alternative formulations for the spatial coupling within and among conurbations. These includes Stouffer's rank model, Fotheringham's competing destinations model and the radiation model of Simini et al. Since the spread of infectious disease reflects mobility through the filter of age-specific susceptibility and infectivity and since, moreover, disease may alter spatial behavior, it is essential to confront with epidemiological data on spread. To study their relative merit we, accordingly, fit variants of these models to the uniquely detailed dataset of prevaccination measles in the 954 cities and towns of England and Wales over the years 1944-65 and compare them using a consistent likelihood framework. We find that while the gravity model is a reasonable first approximation, both Stouffer's rank model, an extended version of the radiation model and the Fotheringham competing destinations model provide significantly better fits, Stouffer's model being the best. Through a new method of spatially disaggregated likelihoods we identify areas of relatively poorer fit, and show that it is indeed in densely-populated conurbations that higher order spatial interactions are most important. Our main conclusion is that it is premature to narrow in on a single class of models for predicting spatial spread of infectious disease. The supplemental materials contain all code for reproducing the results and applying the methods to other data sets.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'The ability to predict how infectious disease will spread is of great importance in the 1 face of the numerous emergent and re-emergent pathogens that currently threatening 2 : bioRxiv preprint human well-being. We identified a variety of alternative models that predict human 3 mobility as as a function of population distribution across a landscape. These consider 4 some models that account for pair-wise interactions between population centers, as well 5 as some that allow for higher-order interactions. We trained the models using a 6 uniquely rich spatiotemporal data set on pre-vaccination measles in England and wales 7 (1944-65), which comprises more than a million records from 954 cities and towns.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Likelihood rankings of the different models reveal strong evidence for higher-order 9 interactions in the form of competition among cities as destinations for travelers and, 10 thus, dilution of spatial transmission. The currently most commonly used so-called 11 'gravity' models were far from the best in capturing spatial disease dynamics.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': '12 42 movement patterns [13] including a recent incarnation as the 'radiation' model [14].', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'While these models have proved useful in their original applications, their use in 44 describing infectious disease spread is complicated by two considerations. First, spatial 45 interaction models predict bulk migration fluxes between population centers, but the 46 impact of these movements on the dynamics of infectious disease can depend strongly 47 not only on the magnitude, but also on the composition, of the migrant pool. In 48 particular, the extent to which migrants are more or less likely to be susceptible to 49 infection, or actually infected, than the general population can be critically important, 50 as can the age profile of the migrant pool, due to pronounced age-specific patterns of 51 behavior, susceptibility, and infectiousness [5, 7, 15] . Second, infectious disease dynamics 52 and Wales using a spatial hazards approach [18] . We considered four main classes of 268 models, including the gravity, competing destinations, Stouffer rank, and radiation 269 models, in addition to pure diffusion and mean field models which are interesting special 270', 'cite_spans': [{'start': 754, 'end': 757, 'text': '[5,', 'ref_id': 'BIBREF4'}, {'start': 758, 'end': 760, 'text': '7,', 'ref_id': 'BIBREF6'}, {'start': 761, 'end': 764, 'text': '15]', 'ref_id': 'BIBREF13'}, {'start': 849, 'end': 853, 'text': '[18]', 'ref_id': 'BIBREF16'}], 'ref_spans': [], 'section': 'Abstract'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       full_text  \n",
       "766  Accurately predicting the geographical spread of emerging, re-emerging, and recurrent 14 epidemics in an increasingly globalized world is a matter of international urgency in the 15 wake of outbreaks of emerging pathogens such as severe acute respiratory syndrome 16 (SARS), Middle East respiratory syndrome (MERS), and ebola virus disease (EBVD) as 17 well as re-emerging and resurgent pathogens such as influenza subtype A-H1N1, 18 whooping cough, and measles. In response to this challenge, various 'spatial interaction' 19 models describing human movement as a function of population distribution have been 20 proposed. Some of these borrow from economics and human geography while others 21 adapt models from movement ecology and the physics of reaction and diffusion on 22 heterogenous landscapes. 23 In recent years, the field has seen the widespread adoption of a family of so-called 24 'gravity' models from transportation theory and human geography (see, e.g., [1, 2] ). In 25 its most common form, the gravity model posits that the migration flux between a pair 26 of cities is log-linearly dependent on their respective sizes and on their separating 27 distance. The application of this simple model to disease spread was originally proposed 28 by Murray and Cliff [3] , but over the last decade, many studies have used it to explain 29 historical, or predict future, disease spread in a range of infections including measles [4] , 30 influenza [5] [6] [7] , cholera [8] , and yellow fever [9] . While its application has yielded 31 insights, the geography literature has highlighted a prominent shortcoming of the 32 gravity model. In particular, these models ignore the potential for competitive or 33 synergistic interactions among population centers ( Fig. 1 ; [10, 11] ). Thus, for example, 34 movement between Boston and Washington DC is assumed to be unaffected by the 35 presence of the intervening city of New York. 36 As it happens, there are distinct families of models, arising from economics and 37 geography, that predict human movement patterns while allowing for higher-order 38 interactions among cities. In particular, Stouffer's [12] 'law of intervening opportunities' 39 posits that \"the number of persons going a given distance is directly proportional to the 40 number of opportunities at that distance and inversely proportional to the number of 41 intervening opportunities\". This idea has given rise to alternative models for human can feed back onto migration. Most obviously, disease symptoms can influence 53 movement behavior, as seen, for example, in the fact that a mild cold may induce 54 minimal changes in movement behavior but severe hemorrhagic fever or acute paralytic 55 disease will typically slow the movement of the infected hosts. These two considerations 56 do not preclude the utility of spatial interaction models in the infectious disease context; 57 they do, however, complicate their use and the importance of understanding 58 transmission relevant migration fluxes, which will be some sort of 'effective average' of 59 movement as filtered through such aforementioned complications.\\nSeveral spatial interaction models have been parameterized and tested using various 61 mobility data such as commuter flows (e.g., [5, 14] ), mobile phone geolocations 62 (e.g., [16] ), social media (e.g., [13] ), and microsimulations (e.g., [17] ). However, in view 63 of the challenges just noted, the ultimate test of the models is against data on the actual 64 spread of infection rather than bulk movement of people or cell phones. Bjørnstad and 65 Grenfell [18] proposed that for acute immunizing infections, the spatio-temporal 66 patterns of fade-outs (i.e., local disease extinction) across metapopulations provide 67 valuable information on disease spread because these patterns reflect spatial 68 transmission unclouded by local transmission. In this paper, we extract information 69 from fade-out patterns and a consistent likelihood framework to compare and contrast 70 to a suite of models including (i) the gravity model [1] , (ii) Fotheringham's competing 71 destinations model [10] , (iii) Stouffer's rank model [12] , and (iv) the radiation model [14] . 72 . We confront these models weekly data on measles incidence from all 954 cities and 73 towns in England and Wales from 1944 to 1965 [4, 19] . Comparing fits and predictions, 74 we show that while the gravity model is a reasonable first approximation, Stouffer's 75 rank model, an extended version of the radiation model, and the competing destinations 76 model all provide significantly better fits, Stouffer's model performing the best.\\nData 79 Historical incidence of measles in England and Wales has been an influential testbed for 80 models and methods in disease dynamics since Bartlett's [20, 21] seminal work on its 81 recurrent epidemics. We use the spatially resolved weekly measles data across all 954 82 cities and towns of England and Wales from 1944, when notification was made 83 mandatory by the UK Registrar General (OPCS), until 1965, which saw subtle shifts in 84 political boundaries around London. Vaccination was not introduced in the UK until 85 1967, so that these data span a period where measles dynamics were unaffected by mass 86 immunization. The data set is complete except for a region-wide underreporting rate of 87 around 50% [22, 23] . Grenfell et al. [19] give a detailed description of the data; the 88 entire data set has been made available by Lau et al. [24] . 89 An important feature of the system is that, between 29% and 38% of the population 90 (c. 47M during this period) resided in a small number (13-28 depending on exact 91 definition) of communities above a critical community size (CCS) of c. 250-300k. Cities 92 larger than the CCS tend to sustain local chains of transmission. The remaining 60-70% 93 were distributed among the more than 900 communities smaller than the CCS where 94 local extinctions are more or less frequent (depending on population size and degree of 95 isolation) and, consequently, the rate of reintroduction of the pathogen via spatial 96 transmission is an important determinant of measles incidence. In ecological terms, the 97 prevaccination measles system represented an epidemic mainland-island metapopulation 98 (e.g., [25, 26] ). Our analysis exploits this fact, using the timing and spatial pattern and 99 timing of reintroductions to inform the parameters of each of the spatial interaction 100 models.\\nUsing a spatially-extended time series susceptible-infected-recovered (TSIR) framework 103 (e.g., [4, 27] ), we can calculate, for each population center, the probability that a spatial 104 interaction happens (i.e., a contact between a resident susceptible host and a 105 non-resident infectious host) and that the contact results in a new local chain of 106 transmission [18] . In previous analyses, we showed that the signature of such events will 107 be drowned out in the presence of endemic circulation (see [23] ). Following a local 108 extinction, however, the re-colonization rate contains critical information on the spatial 109 interactions.\\nThe spatially-extended TSIR model predicts that the expected number of new 111 infected hosts in community i in epidemic generation t + 1 will be\\nHere, β s(t) is the seasonal transmission rate as molded by opening and closing of schools 113 and the proportion of transmission that occurs within the school setting as the school 114 year progresses [23] ; s(t) = t mod 26 is the seasonality function; S i,t /N is the 115 probability that a local individual is susceptible; I i,t is the local number of infections; α 116 is a correction for the discrete-time approximation of the underlying continuous-time 117 process [28] ; and ι i,t is the sum of transmission-relevant spatial interactions i has with 118 the other communities in the epidemic metapopulation. Following a local extinction,\\nAssuming demographic stochasticity in transmission due to the underlying epidemic 121 birth-and-death process, the realized number of cases in the next generation will be [23] : 122\\nAs a consequence, the number of susceptibles in the next generation will be\\nwhere B i,t is the recruitment of local susceptibles through births during the generation 124 interval. The 'serial interval' for measles is 10-14 days [29] , so we follow previous TSIR 125 analyses in aggregating the weekly data in 2-week increments (though [23] investigate 126 shorter intervals). Using susceptible reconstruction methods [30] , we estimate all local 127 parameters using the method of Finkenstadt et al. [31] . The online supplement contains 128 full documentation of these analyses and code for exact replication of all results 129 presented in this paper.\\nConditioning on the local parameters and restricting analyses at each location to 131 biweeks in which measles incidence was zero, we use Eqs. 2 and 4 to form a negative 132 binomial likelihood to estimate all parameters of each of the candidate spatial 133 interaction models.\\nSpatial interaction models 135 When including special cases, such as pure diffusion and variants, we consider a total of 136 10 spatial interaction models (Fig. 1) . Each of these amounts to a specification of the ι 137 term in Eqs. 1 and 2.\\nGravity model. Under the gravity model, the spatial interaction between locations i 139 and j takes the form θN τ1 i N τ2 j d −ρ ij , where θ, τ 1 , τ 2 , and ρ are non-negative parameters. 140 This leads to the following formulation for disease-relevant spatial interactions:\\nwhere the sum is across all non-self potential donor communities, and I j /N j is the 142 fraction of infected individuals in donor community j. The gravity model has two 143 important special cases: ρ = 0, τ 1 = τ 2 = 1, which is a mean-field model, and 144 τ 1 = τ 2 = 0, which is simple spatial diffusion. The φ parameter represents background 145 spatial transmission that is not predictable on the basis of distance and size [24] . 146 Xia's model. In the original analysis of the spatiotemporal dynamics of measles, Xia et al. [4] used a formulation rooted in the gravity-model literature but with a slightly different formulation:\\nwhich is the same as Eq. 5 if τ 2 = 1, but not otherwise.\\nCompeting destinations model. Fotheringham [11] noted that gravity models may be misspecified because they only consider pairwise interactions among locations. He argued that nearby or intervening destinations may, in fact, make spatial interactions either more or less likely. A synergistic effect occurs, for example, if individuals are disproportionately inclined to do all their shopping in districts with many shops. Antagonistic effects may arise where travel between two cities, say Boston and Washington, is made less likely by the presence of an intervening city such as New York because individuals from either Boston or Washington can get their out-of-town needs fulfilled by visiting New York. Under Fotheringham's [11] competing destinations formulation the flux between i and j takes the form θ N τ1\\nAs in the gravity model, τ 1 and τ 2 control how 'eagerness to travel' and city 'attractiveness' scales with population size; and ρ measures how the likelihood of travel decays with distance. The parameter δ quantifies how destinations k, of various sizes, at distances d jk from the donor j, modulate the spatial interaction between recipient i and and donor j. In particular, δ > 0 indicates a synergistic effect; δ < 0, an antagonistic effect. The resultant formulation for disease-relevant spatial interactions between community i and everywhere else is:\\nwhere φ again represents background, spatially unpredictable spread. 148 Stouffer's rank model. Noulas et al. [13] recently proposed that human mobility 149 patterns should be studied using Stouffer's [12] rank (\"law of intervening author/funder. This article is a US Government work. It is not subject to copyright under 17 USC 105 and is also made available for use under\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2019.12.19.882175 doi: bioRxiv preprint population size as a proxy for \"opportunities\". Accordingly, we let Ω(i, j) be the 155 collection of towns located closer to town i than is town j:\\n156 The Stouffer model for disease-relevant spatial   157 interactions is, then\\nStouffer's original model was framed in the context of continuous distribution of 159 population across a landscape. In the present metapopulation context, we therefore 164\\nExtended radiation model. The radiation model proposed by Simini et al. [14] was derived independently but is in spirit related to the original ideas of Stouffer [12] . Our metapopulation version of this model with respect to disease-relevant spatial interactions is\\nAs with the Stouffer model, we consider the set Ω(i, j) to either exclude ('radiation\") or 166 include ('radiation variant\") community size i itself in the spatial interaction \\nOf the models considered, the gravity variants (including mean-field and pure 174 diffusion) include purely pairwise interactions whilst the other models allow for 175 higher-order interactions either explicitly (as in the competing destination model) or 176 implicitly (as in the Stouffer and radiation models).\\nStatistical inference 178 We use maximum likelihood to estimate the parameters of each spatial interaction 179 model and profile likelihoods to study correlation among parameters and possible 180 identifiability issues [27] . author/funder. This article is a US Government work. It is not subject to copyright under 17 USC 105 and is also made available for use under\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . Table 1 . MLE estimates. Models: SV -Stouffer variant, XR -extended radiation, CD -competing destinations, S -Stouffer, G -gravity, X -Xia, RV -radiation variant, R -radiation, MF -mean field, D -diffusion. Here, denotes the log likelihood,ĉ is the computed variance inflation factor, ∆QAIC is the QAIC relative to that of the best-fitting model, and the other parameters are as in the model equations. biweek of local disease absence [18] ) All calculations are documented in the online 189 supplement, which includes code for the detailed replication of all results.\\n190 Table 1 shows estimated parameters and likelihoods for each of the spatial interaction networks. To explore the drivers among model differences in fit, we therefore contrast 214 model predictions of disease import and export rates. Fig. 2 shows the mean predicted 215 export and import rates vs. population size for each of the models. The dependence of 216 mean export rate on population size contrasts somewhat among models ( Fig. 2A) , but 217 the importation rates follow similar patterns (Fig. 2B) . Despite their differences in 218 underlying mathematical equations, there must be more nuanced differences among the 219 models that contribute their relative fit/lack-of-fit. This begs the question: \"What are 220\\nDecember 16, 2019 7/15 a CC0 license.\\nauthor/funder. This article is a US Government work. It is not subject to copyright under 17 USC 105 and is also made available for use under\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2019.12.19.882175 doi: bioRxiv preprint these differences and are they biologically meaningful or spatially random?\" 221 A cursory review of the recent literature suggests that the gravity model is the most 222 prominent in the infectious disease context, followed by the radiation model. Stouffer's 223 model and the competing destinations model have rarely been applied in this field. It is 224 therefore useful to use the gravity model fit as a baseline and explore how the 225 better-fitting models diverge from this baseline. Comparing matrices with half a million 226 entries is very difficult, so we amploy a new 'spatial likelihood contrast' (SLiC) method 227 with which to study the relative merit of the different spatial interaction models. The 228 idea is to disaggregate the overall hazard likelihoods by individual cities to study how 229 particular communities contribute to improved or diminished relative fit of each model. 230 To do so we normalize each location's contribution to the overall likelihood by the 231 number of data points each conurbation contributes to the likelihood (the accumulated 232 stretches of measles absence) and then map the model-model differences onto the 233 landscape (Fig. 3) . Inspection of Fig. 3 suggests that the poorer fit of the gravity model 234 vs. the Stouffer or competing destinations models is primarily in the urban northwest. 235 The supplement provides the full set of pairwise SLiC contrasts among all models. To 236 test whether the apparent patterns are statistically significant, we compute local 237 indicators of spatial association (LISA; [34] ) statistics for all population centers with 238 fewer than 50k inhabitants (Fig. 3A, B) . The main failure of the gravity model is in (e.g. [35, 36] ). In the case of epidemics playing out within a population center, models 254 closer to the mean field model (i.e., with uniform connectivity) often do remarkably well 255 (see [37] for a social network perspective on this). However, such models generally do 256 not capture regional spread of human infections because of the complicated patterns of 257 human movement (the third wave of the 2009 influenza pandemic across the USA, 258 perhaps, being an unusually diffusive counterexample [7] ) . To elucidate these patterns, 259 many empirical studies have been performed and a variety of abstractions have been 260 proposed. The task of choosing among these abstractions for help in help predicting 261 spread of infectious disease is complicated by the fact that (i) model formulations may 262 differ with respect to how well they actually reproduce aggregate mobility patterns, and 263 (ii) model fit may be shaped by the manner in which mobility is filtered by transmission 264 and behavior.\\nTo advance the discussion of these issues, we considered a suite of candidate models 266 of spatial interactions and fit them to the pre-vaccination measles data from England cases. The gravity model-which has gained recent prominence in infectious disease As the gravity model has had a very rapid adoption in spatial disease epidemiology 284 during the last decade (though Noulas et al. [13] have noted the importance of 285 considering alternatives), we used this as a baseline. Despite its common usage it 286 significantly under-performs relative to the several other models. Combining SLiC and 287 LISA statistics we found that the greatest tension between the gravity model and the 288 preferred alternatives is in the Manchester-Liverpool-Leeds conurbation of northwestern 289 England, where many towns and villages commingle amongst several major cities. formulation) that spatial interaction, once discounted for higher-order interference, is 295 roughly proportional to the geometric mean of the community sizes. This curious 296 suggestion warrants future investigation.\\nIn this study we have used actual disease incidence data, as opposed to raw 298 measurments of human movement, to further our understanding of the principles 299 governing how human hosts spread ifectious disease across a populated landscape. We 300 have found both strong evidence of higher-order interactions among population centers, 301 and that candidate models differ significantly in their ability to capture the empirical 302 patterns. We hope our findings will help stimulate a systematic, data-driven discussion 303 of the relative merit of alternative predictive models for the probable path for spatial 304 spread of infectious disease. Clearly, the prospects are good for good for further 305 refinements and improved parameterizations. Spatial interaction models predict the flux of human movements between population centers (cities, towns, villages) as a function of the distribution of the population on the landscape. In this diagram, the relative magnitude of the fluxes from a focal town to other population centers are represented by the widths of the arrows. In the widely-employed gravity models (A), interactions among cities is strictly pairwise. Thus the addition of a new town (B) has no effect on the movement to other towns. In Fotheringham's competing destinations model (C), however, competition or synergy among nearby communities can reduce or augment fluxes. Stouffer's model of intervening opportunities and the radiation model (D) posit that movement from one city to another is diminished by the presence of opportunities in communities more proximal to the source city. author/funder. This article is a US Government work. It is not subject to copyright under 17 USC 105 and is also made available for use under\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2019.12.19.882175 doi: bioRxiv preprint\\n  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1417</td>\n",
       "      <td>Comparing Metapopulation Dynamics of Infectious Diseases under Different Models of Human Movement</td>\n",
       "      <td>[{'text': 'Newly available data sets present an exciting opportunity to investigate how human population movement contributes to the spread of infectious diseases across large geographical distances. It is now possible to construct realistic models of infectious disease dynamics for the purposes of understanding global-scale epidemics. Nevertheless, a remaining unanswered question is how best to leverage the new data to parameterize models of movement, and whether one's choice of movement model impacts modeled disease outcomes. We adapt three well-studied models of infectious disease dynamics, the SIR model; the SIS model; and the Ross-Macdonald model, to incorporate either of two candidate movement models. We describe the effect that the choice of movement model has on each disease model's results, finding that in all cases there are parameter regimes where choosing one movement model instead of another has a profound impact on epidemiological outcomes. We further demonstrate the importance of choosing an appropriate movement model using the applied case of malaria transmission and importation on Bioko Island, Equatorial Guinea, finding that one model produces intelligible predictions of R 0 while the other produces nonsensical results.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Mathematical modeling | Infectious disease dynamics | Human population movement | Malaria DT Citron developed the project and carried out the analysis. DL Smith provided guidance and context for the project. CA Guerra provided the data for building the Bioko Island model. SL Wu, JM Henry, HM Sánchez C, and D Dolgert discussed and reviewed the analysis.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>Mathematical models are important tools for understanding disease transmission, making it possible to estimate the size, timing, and impact of epidemics and the effectiveness of interventions. There have been many such mathematical models adapted and applied to study epidemics that spread across large geographical distances (1, 2) . Many such models are often motivated by recent pandemic illness events, such as SARS (3); Ebola virus disease (4); Zika fever (5) ; and the 2019 novel coronavirus (6) , which spread over large geographical ranges and across international borders. In each of these cases, long-distance travel of infected human hosts proved to be an important driver of the spread of infectious disease between geographically separated populations.\\nParameterizing realistic models of disease transmission occurring across wide geographical ranges is now possible because of the recent availability of large, highly detailed data sets describing human movement patterns (7) . Census data attempts to describe population migration that occurs between census years (8) . Traditional surveys of commuters (9, 10) , patients (11, 12) , or residents in disease-affected areas (13, 14) provide another description of recent travel activity. More recently, mobile phone service providers have shared privacy-protected data sets, showing how large numbers of users tend to move between cell phone towers (12, (15) (16) (17) (18) . Studies involving GPS trip loggers also track participants' movement activities (19) .\\nThe new abundance of movement data provide an exciting new opportunity to use models to quantify how much human host movement affects epidemiological outcomes. Nevertheless, each method of data collected only provides a partial picture of true movement patterns. It remains an unanswered question how best to use the data to parameterize models representing host movement. One consideration is finding a statistical model that fits adequately well to the data, predicting the frequency of travel or flow volume between two locations. There are already many candidate movement kernel models, such as gravity and radiation models, which have been evaluated against data in a variety of different settings (13, 18, (20) (21) (22) . A fitted movement kernel, once it has been evaluated as sufficiently accurate, may then be used to calibrate the movement-related parameters of a mechanistic disease transmission model.\\nFor the purposes of modeling movement together with infectious disease dynamics, however, an additional point of consideration is whether or not the details of the movement model-the rules which govern how hosts' movement patterns are represented-affect the model's quantitative behavior and subsequent predicted epidemiological outcomes. Two example classes of movement models which provide simple mechanistic representations of travel behavior are Eulerian movement models, which specify the rates at which hosts at one location travel to any other location, or Lagrangian movement models, which specify how frequently hosts travel away from home before returning (23) . Depending on the level of detail contained in the available data, one might be able to fit any number of candidate mechanistic movement models. There is precedent in calibrating different movement models using the same travel data, only to find that the disease dynamics have changed as a result (9) . Moving beyond the question of how to best fit a statistical model to that data set, the question becomes whether choosing one mechanistic movement model over another can affect epidemiological outcomes, and whether there are certain settings or parameter regimes where one type of movement model may be more accurate and appropriate.\\nIn the present analysis we explore these questions using compartmental metapopulation models which integrate together host movement and infectious disease dynamics. We use three infectious disease transmission models-the Susceptible-Infected-Recovered model; the Susceptible-Infected-Susceptible model; and the Ross-Macdonald model-which represent a suite of tools for modeling the transmission dynamics of a wide variety of pathogens. We adapt each of the disease transmission models to incorporate two mechanistic representations of movement: the first is an Eulerian movement model which we call the Flux model, and the second is a Lagrangian movement model which we call the Simple Trip model. We directly compare the Flux and Simple Trip models by setting parameters such that the total flux of travelers between each sub-population remains constant, thus emulating a case where one might have a single data set from which one could calibrate either movement model. We examine how disease model outcomes can change with different movement models for each transmission model. For all three transmission models, we find that the modeled quantities of interest relating to disease dynamics can differ dramatically depending on one's choice of movement model. We conclude by applying this analysis to a model of malaria transmission and importation on Bioko Island in Equatorial Guinea, and use intuition based on our prior analysis to frame our understanding for why the Flux model surprisingly fails to produce meaningful predictions of transmission intensity in that context.\\nHost movement models. We use metapopulation models to represent a network of geographically isolated sub-populations of hosts. Each sub-population occupies a site where local conditions may affect transmission intensity. Disease transmission is assumed to be completely local, such that hosts from different sub-populations come into contact with one another only if they travel to occupy the same location. We describe and compare two simple models of host movement, each of which represents a different set of rules governing how hosts move from one location to another. The \"Flux model\" is an Eulerian movement model which describes hosts as diffusing from one metapopulation to another (23):\\nwhere Ni counts the number of hosts currently located at site i. The total number of hosts remains constant over time\\nNi). The constant fi,j represents the rate at which hosts located at i travel to j, where fi,i = 0 for all i. Making an equilibrium assumption that the total number of residents at each site remains constant also constrains fi,j = fj,i, such that the fully specified Flux Model requires K(K − 1)/2 parameters. At equilibrium, when the derivatives on the left hand side of Eq. 1 equal zero the set of populations at each location may be found by solving for the nontrivial null vector of the matrix F where Fi,j = fi,j + fj,i.\\nThe \"Simple Trip model\" is a Lagrangian movement model which assigns home locations to each host and describes how hosts travel temporarily to other locations before returning home (23) . Unlike the Flux model, the Simple Trip model differentiates between the residents and the visitors currently located at the same site. Visitors are able to interact with residents at a given site, but they return home at fixed rates:\\nThe notation in Eq. 2 has changed to account for hosts retaining memory of their home sites: a host from i who is currently located at j will be counted as belonging to the Ni,j population, and the number of hosts whose home is i remains constant over time, even if members visit other sites (Ni = K j=1 Ni,j). The constant φi,j represents the rate at which hosts whose home is i travel to j, while the constant τi,j is the rate at which hosts visiting j from i return home to i. Both φi,i = 0 and τi,i = 0 for all i. The fully specified Simple Trip model requires 2K(K − 1) parameters, meaning that more information is required to parameterize this model than the Flux model.\\nAt equilibrium, when the derivatives on the left hand side of Eq. 2 equal zero, the population Ni is distributed across the K metapopulation sites as follows:\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04.05.20054304 doi: medRxiv preprint\\nWe can use Eqs. 3 to define a \"Time at Risk Matrix\" ψ where ψi,j ≡ N * i,j /Ni, the average number of individuals from i found at site j at equilibrium, or the fraction of time on average that hosts who live in site i spend in site j.\\nTo directly compare the quantitative behavior of both models, we will assume that at equilibrium the total number of hosts moving from one site to another is the same across both models. That is to say, the number of people traveling i → j in the Flux model is the same as the number of i residents traveling to j + the number of j resident travelers returning home from i in the Simple Trip model:\\nIt can be shown under these circumstances that the equilibrium populations in the Flux model N * i equal the equilibrium populations in the Simple Trip model. SIR Models. The next step is to incorporate both movement models into three compartmental models of disease transmission. The first such model is the SIR model, which describes a single outbreak of disease. For a single population, the deterministic ordinary differential equation SIR model is as follows:\\nCombining the basic SIR model with the Flux movement model 1, we obtain an analogous set of 3K equations:\\nThe parameters β and γ in Eq. 6 represent the transmission rate and recovery rate, respectively.\\nCombining the basic SIR model with the Simple Trip model 2, we obtain an analogous set of 3K 2 equations describing transmission between hosts while they are at their home location or while they are traveling:\\nNote that for both movement models we allow the parameter representing transmission intensity βi to vary across locations, not across individuals, a choice that reflects how transmission intensity reflects the contact network structure or other local conditions affecting transmission. γ, by contrast, is held constant for all individuals regardless of location. . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04.05.20054304 doi: medRxiv preprint SIS Models. Our second disease transmission model is the SIS model, which can be used to describe persistent endemic disease brought about by cycles of repeated infection. We adapt the SIS model to include either the Flux or Simple Trip models to describe movement between sub-populations. For a single population, the SIS model equation is as follows:\\nCombining the basic SIS model with the Flux model (Eq. 1), we obtain a set of K equations for transmission in the ith site:\\nfj,iIj [9] Combining the basic model with the Time at Risk model (Eq. 2), we obtain a set of K 2 equations among both residents at home and travelers who are away:\\nAgain, in both movement models the transmission intensity βi varies across locations, not across individuals, and γ is held constant for all individuals in all locations.\\nRoss-Macdonald Models. Our third disease model is the Ross-Macdonald model, which was developed to describe malaria transmission (24, 25) . The Ross-Macdonald model is similar to the SIS model, except it includes a vector population which serves as a mechanism of transmission from infected to susceptible human hosts. The Ross-Macdonald model tracks infection dynamics among human hosts (X) and vectors (Z). The basic Ross-Macdonald model is as follows:\\nThe first of Eqs. 11 describes how human hosts become infected through contact with infectious mosquitoes, while the second describes how mosquitoes become infected through contact with infectious humans. Analyzing Eq. 11 under equilibrium conditions, it is possible to solve for an expression for R0 as a function of prevalence (24, 25) :\\nAs before, we adapt the basic Ross-Macdonald model by allowing human hosts to move from one metapopulation to another. At present we will hold the Ross-Macdonald model's parameters (r, a, b, c, g, n) to be constant across all sites, while the stable mosquito population supported by the local environment (Mi) is allowed to vary across all sites. Incorporating the Flux model into the metapopulation Ross-Macdonald model (23):\\nWe again perform a standard analysis of Eqs. 13 under equilibrium conditions and obtain an expression for R {i} 0\\nin each site as a function of prevalence in each site Xi:\\nWe also extend the basic Ross-Macdonald model by allowing human hosts to move from one metapopulation to another as described by the Simple Trip model. Following the analysis of (23) and (17, 26) , we simplify the analysis using the Time\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04.05.20054304 doi: medRxiv preprint at Risk matrix ψ to rewrite the force of infection experienced by human hosts as a weighted average which combines local transmission and the average fraction time spent at risk in different locations:\\nThe expression κi represents the fraction of infectious humans located at i, accounting for both local residents as well as visitors from other locations. From Eq. 15, and following the analysis of (17), we obtain an expression for R0 which reflects how the movement model alters transmission at each metapopulation site:\\nShort Travel Duration\\nWe numerically integrate Eqs. 6 and 7 to find the size of the residual population of susceptibles following the outbreak. We plot the residual population fraction in location 1 (S ∞ 1 /N1) as a function of transmission intensity in location 1 (β1/γ) while holding transmission intensity in location 2 (β2/γ) constant.\\nWe explore four different parameter regimes defined by the transmission intensity of location 2 and the duration of travel. In all four parameter regimes there is a dramatic separation between the predicted outbreak sizes that depends on whether one uses the Flux (blue) or the Simple Trip (red) model. As long as residents of location 1 travel from a low to a high transmission environment (β2 &gt; β1) the Simple Trip model's outbreak will be smaller and the residual population will be larger; the opposite is true when residents of location 1 travel from a high to a low transmission environment (β2 &lt; β1). A: Short travel duration regime (φi = 1, τi,j = 10); high transmission in location 2 (β2 = 2.5) B: Short travel duration regime (φi = 1, τi,j = 10); low transmission in location 2 (β2 = 0.5) C: Long travel duration regime (φi = 0.01, τi,j = 0.1); high transmission in location 2 (β2 = 2.5) D: Long travel duration regime (φi = 0.01, τi,j = 0.1); low transmission in location 2 (β2 = 0.5) SIR models. We begin by directly comparing the predicted outcomes of the two versions of the SIR model with Flux and Simple Trip movement. We simulate epidemics in two coupled metapopulations of Ni = 500 human hosts by numerically integrating Eq. 6 and Eq. 7 and measuring the size of the outbreak. The population of susceptibles remaining after the outbreak has ended S∞ quantifies the size of the outbreak and allows one to infer R0 (27) . Furthermore, S∞ quantifies the population at risk in subsequent future epidemics, which is relevant for modeling seasonally recurring pathogens such as influenza (28) .\\nWe set γ = 1 in all locations, hold β2 constant, and measure the residual population S∞ in location 1 as β1 varies. We identify two transmission parameter regimes: β2 &gt; γ, which is high enough to cause an outbreak in location 2, and β2 &lt; γ, . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04.05.20054304 doi: medRxiv preprint which in isolation does not lead to an outbreak in location 2. We also define two travel parameter regimes, one where travel occurs frequently and lasts short periods of time (τi,j &gt; γ), and one where travel occurs infrequently and lasts long periods of time (τi,j &lt; γ). We specify the parameter values for φi,j and τi,j in each of these regimes for the Simple Trip model and use Eq. 4 to also parameterize the Flux model, such that we emulate using the same data inputs to parameterize both movement models. Figure 1 compares how the two SIR models with metapopulation movement behave in different parameter regimes. In all cases the outcomes are dramatically different, showing that SIR model predictions depend strongly on how host movement is represented in the metapopulation model. The two movement models do agree when transmission parameters are equal in both locations (β1 = β2), but there are large quantitative differences whenever there is some heterogeneity in the local transmission conditions across the different locations.\\nThe major difference between the Flux and Simple Trip models is that the Simple Trip model constrains the amount of time that a traveler spends away. For example, the larger residual population for the Simple Trip model, seen in the left hand column of Figure 1 , follows from a limit on the amount of time that residents of population one spend in the high-transmission location 2. Another difference between the Flux and the Simple Trip models is that hosts in the Flux model have no home residence and continue to move freely between populations after the outbreak ends. As a result, as the system reaches equilibrium after the epidemic the fractions of susceptibles become constant across all locations. In contrast with the Simple Trip model, the Flux model effectively erases any evidence that the transmission intensity differed between the two locations.\\nSIS models. We next compare the respective behavior of the SIS Flux metapopulation model (Eq. 9) and the SIS Simple Trip metapopulation model (Eq. 10) for K = 2 metapopulations of Ni = 500 coupled together. Again, we set γ = 1 at all locations. The outcome of interest for the SIS model is now prevalence, which we find by numerically solving for the endemic equilbria in Eq. 9 and Eq. 10. We hold β2 constant and solve for the prevalence in location 1 (I * 1 /N1 for the Flux model and (I * 1,1 + I * 1,2 )/N1 for the Simple Trip model) as a function of the transmission intensity in location 1 (β1). We define two transmission parameter regimes: β2 &gt; γ, which is high enough to sustain endemic transmission locally in site 2, and β2 &lt; γ, which cannot locally sustain endemic transmission. We again define two travel parameter regimes, one where travel occurs frequently and over short periods of time (τi,j &gt; γ) and one where travel occurs infrequently and over long periods of time (τi,j &lt; γ). In each regime we specify travel rate parameter values for the Simple Trip model and use Eq. 4 to set parameters for the Flux model.\\n! i,j &gt; \" Long Travel Duration ! i,j &lt; \" # 2 /\" &gt; 1 # 2 /\" &lt; 1 D C\\nWe numerically solve Eqs. 9 and 10 for the endemic equilibrium prevalence. We plot the endemic prevalence in location 1 I * 1 /N1 as a function of transmission intensity in location 1 (β1/γ) while holding transmission intensity in location 2 (β2/γ) constant. We explore four different parameter regimes defined by the transmission intensity of location 2 and the duration of travel. When travel duration is short (upper row), the Flux (red) and Simple Trip (blue) models produce dramatically different relationships between transmission intensity and prevalence. When travel duration is long (lower row), the two movement models' results converge. Figure 2 shows the results from both the Simple Trip and Flux models in each of the four parameter regimes. In the short travel duration regime (2 A, B), as long as the transmission intensity differs between the two locations, there is a noticeable separation between the two movement models' predicted relationships between prevalence and local transmission intensity. In 2 B it is even the case that the apparent critical point, above which location 1 has sustained endemic disease, changes depending on which movement model is used.\\nThe discrepancy in the short travel duration regime arises because the Simple Trip model imposes an extra constraint on the amount of time a traveler spends away from home. The flux model does not specify the amount of time spent away, and its movement rates (from Eq. 4 tend to overestimate the time spent away. The Simple Trip model endemic prevalence in Figure 2 A is suppressed compared to the Flux model endemic prevalence because the amount of time that travelers from location 1 spend at risk in the high transmission location 2 is constrained to be small. Similarly, the Simple Trip model endemic prevalence in 2 B is elevated because travelers spend less of their time at risk in the low transmission location 2. In the long travel duration regime (Figure 2, lower row) , however, if the trip time is longer than the disease infectious period there is more time for prevalence among travelers to location 2 to equilibrate with the prevalence among residents of location 2: in this parameter regime, the Flux model and Simple trip models converge, because the constraint on time spent away from home makes less of a difference.\\nWe examine the behavior of the Ross-Macdonald models with Flux and Simple Trip movement. We use the model to represent a malaria-causing pathogen with a mean infectious period of r −1 = 200 days. The outcome of interest in the Ross-Macdonald model is R0, which quantifies the local transmission intensity required to produce the prevalence. We compare the two movement models using K = 2 connected sub-populations of Ni = 500 human hosts by specifying prevalence and using Eqs. 14 and 16 to calculate R0 for the Flux and Simple Trip models, respectively. Again, we define two parameter regimes based on high (X = 0.25) versus low (X = 0.025) prevalence in location 2, and another pair of parameter regimes based on high frequency, short duration trips (φi = 1/360, τi,j = 1/10) versus low frequency, long duration trips (φi = 1/36000, τi,j = 1/1000). As before, Flux model parameters are defined from the Simple Trip model parameters using Eq. 4.\\n! i,j &gt; \" Long Travel Duration ! i,j &lt; \" # 2 /$ 2 = 0.25 # 2 /$ 2 = 0.025 D C\\nWe numerically solve Eqs. 14 and 16 to find the relationship between prevalence in location 1 (X1/N1) and transmission intensity (R0). We plot the relationship between location 1 prevalence (X1/N1) as a function of transmission intensity in location 1 while holding prevalence constant (X2/N2) in location 2. Note that we plot prevalence on the y-axis and transmission intensity on the x-axis, to emphasize how the Ross-Macdonald model behaves similarly to the SIS model ( Figure 2) . We explore four different parameter regimes defined by the prevalence of location 2 and the duration of travel in relation to the average duration of infection (r −1 = 200 days). In the short travel duration regime, there is a large quantitative difference in the relationship between prevalence and R0 for the two different movement models. In the regime with short travel duration and high location 2 prevalence (upper left), using the Flux model would suggest low transmission (R . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04.05.20054304 doi: medRxiv preprint location are different, there is a dramatic separation between the Flux and Simple Trip model results, particularly in the short travel duration regime (Figure 3 A, B) . If travel duration is long, however, the two models' predicted relationships between prevalence and R0 do converge towards one another. The relationships between prevalence and transmission intensity for the two Ross-Macdonald models are qualitatively similar to the one shown previously for the SIS model in Figure 2 .\\nIn some parameter regimes R0 does appear to run negative, such as with the Flux model predictions in Figure 3 A. This occurs when incidence and, by consequence, prevalence among travelers to a higher transmission environment is too high for the model to reconcile the prevalence specified for each area. In this context, R0 &lt; 0 has no meaningful real-world interpretation beyond indicating that the data are inconsistent with the chosen mechanistic model. (14) . Each pixel shows the mean estimated prevalence within a 1 × 1 km 2 area. B Travel \"prevalence\" map, reproduced with permission from (14) , where each pixel shows the probability that a resident leaves home during an eight week survey period. C Modeled rates of travel {φ i,k } for each of four example pixels, estimated from MIS data reporting recent travel. Arrows pointing off-island represent travel to Mainland EG. Arrow weights are proportional to rate of travel from origin pixel to each of the 7 destination regions, or the sum total rate of travel to all of the destination pixels within each region.\\nWe further illustrate the quantitative disagreements between the Flux and Simple Trip models by applying the Ross-Macdonald models with host movement to the problem of estimating R0 in a real-world setting. Bioko Island is located in the Atlantic Ocean about 225km to the northwest of mainland Equatorial Guinea (EG) in Western Africa. From population data collected during a bednet mass-distribution campaign in 2018, human population count was 239,159 people, about 85% of whom lived in EG's capital city Malabo in the north (29) . During this sanitary census, however, there was a 12% absenteeism that needs to be accounted for. Following the recent efforts of an intensive malaria control and elimination program (Bioko Island Malaria Elimination Program, BIMEP), the average malaria parasite rate (PR, a measure of prevalence) in chhildren 2-14 years old has fallen from an average of 0.43 to 0.11 between 2004 and 2016, with further progress appearing to have stagnated in recent years since 2015 (30) . A recent analysis of data collected through malaria indicator surveys (MIS) (31-33) has used geostatistical techniques (34) to map the estimated prevalence of malaria (Figure 4 A) and the estimated probability of a resident leaving (Figure 4 B) across different locations on Bioko Island (14) . The analysis showed an elevated risk of infection among people who reported traveling recently (14) , a pattern corroborated by other studies (30, 35) . In particular, this suggests that travelers who had spent time in mainland EG, which still has an estimated mean prevalence of around 0.43 (36, 37) , are very likely to bring back infections with them that would contribute to the sustained endemic prevalence measured on Bioko Island. We can use the Ross-Macdonald model to infer local transmission intensity (R0) based on the known prevalence and travel behavior. We treat each of the pixels shown in the maps in Figure 4 as its own sub-population with its own local transmission environment, which allows the model to contain the spatial heterogeneity known from the data. We the population data to set the population Ni and the geospatial estimates to set prevalence Xi/Ni at each location. For the travel parameters we use trip destinations reported in the MIS (38) to fit a multinomial probability model of destination choice. Together with the travel frequency data (Figure 4 B) , we find a set of travel frequency parameters {φ i,k }, illustrated in Figure 4 C. We use MIS data on trip duration (38) to fit an exponential model of trip duration and find that travel within Bioko Island and to mainland equatorial had a mean duration of τ −1 i,BI = 21.2 and τ −1 i,EG = 10.3 days, respectively. (Further details may be found in the Supplementary Information.) The rate parameters fi,j for the Flux model were calculated using Eq. 4.\\nIn this way, we construct a fully parameterized pair of Ross-Macdonald models with movement between sub-populations, from which we can infer the transmission intensity in each location on Bioko Island. Figure 5 A maps estimates of R 0,Flux calculated using the Flux model (Eq. 14). The dark pixels of the map represent areas where the R 0,Flux estimates are calculated to be less than 0. In these locations, the rate at which people become infected while traveling and return home with malaria infections is too high for the transmission model to reconcile local transmission intensity with local prevalence. We frame our understanding of these results in terms of the K = 2 sub-population example discussed previously for the Ross-Macdonald Model. Each sub-population on Bioko Island exists in a parameter regime where travel duration is short (τi,j &gt; r) and the prevalence in the off-island travel destination is higher than the local prevalence. We recall the results shown in Figure 3 A, where we see how for low prevalence values the Flux model tends to underestimate R0 compared to the Simple Trip model. The Simple Trip model constrains the amount of time that travelers spend on mainland EG, and therefore reduces the number of people who return home from traveling with malaria infections. The Flux model does not include such a constraint, and in this parameter regime we see how in some locations the rate of imported infections is too high and forces the transmission intensity to be negative.\\nWe have shown that models of infectious disease dynamics are sensitive to the choices that modelers make when including movement-mediated interactions between sub-populations. To do so, we have adapted each of three transmission models to include two types of movement between sub-populations. For all three transmission models we have identified parameter regimes where choosing to incorporate one movement model instead of the other will result in dramatically different results. Specifically, when there is a difference in transmission intensity between two interacting locations, and when mean travel duration is short compared to the mean duration of infection.\\nWe emphasize that in all instances we set movement parameters to match the total flux of individuals moving between each pair of sub-populations. In this way, we have emulated a situation where the modeler has a single movement data set and must choose which movement model to implement. The full impact of the choice of movement model became clear from the example of modeling malaria transmission and importation on Bioko Island: We calibrated each of the two movement models using the same data set and found that each movement model produced dramatically different results. The Flux model predicted R0 &lt; 0, which has no meaningful interpretation beyond the Flux model being inconsistent with the data used to calibrate it. The Simple Trip model, by specifying the rate at which travelers return home, adds a constraint on the amount of time that travelers spend while away from home. As a consequence, when combined with a the Simple Trip model constrains the amount of risk experienced by travelers. The Flux model has no such constraint: as long as there are variations in the transmission intensity across different locations there will be significant differences between the two movement models' results.\\nThe way that each of the movement models contributes to the infectious disease model results can lead to real confusion over the true relationship between transmission intensity and epidemiological outcomes. In some applications, as with the Bioko Island example, the purpose of the model is to infer transmission conditions which may be difficult to measure empirically based on more readily available prevalence or incidence measures. Returning to Figure 2 A, we can imagine trying to determine the transmission intensity β1 based on a measured prevalence of 0.25. From the Flux model we infer β1 &lt; 1, which suggests that local transmission risk is low. From the Simple Trip model, however, we infer β1 &gt; 1, which suggests the opposite. A similar inconsistency is illustrated for the Ross-Macdonald model in Figure 3 , and appears again in the Bioko Island example in Figure 5 .\\nIn the case of modeling malaria transmission and importation on Bioko Island we have shown how the Flux model is not suitable for producing accurate results. Despite this drawback, the Flux model remains an attractive option for creating a basic quantitative description of host movement in other epidemiological contexts. First, the model requires only K(K − 1)/2 . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04.05.20054304 doi: medRxiv preprint parameters to specify completely. For this reason, parameterization is possible with relatively simple data sets which count the number of individuals moving from one each location to each other location over time. Unfortunately, data sets which record flux volumes do not necessarily lend themselves to parameterizing Simple Trip models, as they do not distinguish between the rates of leaving and returning home.\\nFor example, mobile phone call data records (CDRs) are a source of high-volume data sets for measuring the number of individuals who travel from one location to another. Often, for the sake of preserving subscriber privacy, CDRs are aggregated across individual users, time, and geographical area before becoming public. That is to say, publicly available CDR data sets count the number of mobile phones traveling from one service tower catchment area to another on a daily or weekly basis: these data sets do not record an individual's rate of leaving home or time spent while traveling away from home. The loss of individual identity from data sets has already been shown to impact the predicted outcome of disease models in other contexts (9) . While the limits and biases inherent to CDR data sets have been well-documented (39), the present study suggests an additional drawback when it comes to using CDR data sets for epidemiological modeling: because it is not clear that a Flux model is sufficient to produce accurate modeling results in some contexts, it is likewise not clear that flux data are sufficient to quantify travel behavior when building a movement model. This is not to say that CDR data or Flux models are useless, only that one needs to carefully consider the epidemiological context before incorporating a Flux model into a disease transmission model.\\nThe present work is restricted to discussing only two very basic deterministic models of movement. In theory, one may imagine similarly comparing more detailed and complicated movement models, such as incorporating travel behavior with multiple stops; adding stochastic effects; or using agent-based models to incorporate movement behavior heterogeneity among the travelers. It remains to be seen the extent to which such additional model features might further affect the epidemiological outcomes of interest. In practice, however, the data required for calibrating such models may not be readily available, and using a Flux or Simple Trip description of travel behavior may be more practical. Even for such simple movement models there is a risk of misusing the travel data that are available to calibrate a movement model which cannot be reconciled with the data and produces outputs that are difficult or impossible to interpret.\\nData and code supporting the analysis presented in the Malaria Modeling Example section, including Figures 3 and 4 , may be found at https://doi.org/10.6084/m9.figshare.12084831.v1.\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                  title  \\\n",
       "1417  Comparing Metapopulation Dynamics of Infectious Diseases under Different Models of Human Movement   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              abstract  \\\n",
       "1417  [{'text': 'Newly available data sets present an exciting opportunity to investigate how human population movement contributes to the spread of infectious diseases across large geographical distances. It is now possible to construct realistic models of infectious disease dynamics for the purposes of understanding global-scale epidemics. Nevertheless, a remaining unanswered question is how best to leverage the new data to parameterize models of movement, and whether one's choice of movement model impacts modeled disease outcomes. We adapt three well-studied models of infectious disease dynamics, the SIR model; the SIS model; and the Ross-Macdonald model, to incorporate either of two candidate movement models. We describe the effect that the choice of movement model has on each disease model's results, finding that in all cases there are parameter regimes where choosing one movement model instead of another has a profound impact on epidemiological outcomes. We further demonstrate the importance of choosing an appropriate movement model using the applied case of malaria transmission and importation on Bioko Island, Equatorial Guinea, finding that one model produces intelligible predictions of R 0 while the other produces nonsensical results.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'Mathematical modeling | Infectious disease dynamics | Human population movement | Malaria DT Citron developed the project and carried out the analysis. DL Smith provided guidance and context for the project. CA Guerra provided the data for building the Bioko Island model. SL Wu, JM Henry, HM Sánchez C, and D Dolgert discussed and reviewed the analysis.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      full_text  \n",
       "1417  Mathematical models are important tools for understanding disease transmission, making it possible to estimate the size, timing, and impact of epidemics and the effectiveness of interventions. There have been many such mathematical models adapted and applied to study epidemics that spread across large geographical distances (1, 2) . Many such models are often motivated by recent pandemic illness events, such as SARS (3); Ebola virus disease (4); Zika fever (5) ; and the 2019 novel coronavirus (6) , which spread over large geographical ranges and across international borders. In each of these cases, long-distance travel of infected human hosts proved to be an important driver of the spread of infectious disease between geographically separated populations.\\nParameterizing realistic models of disease transmission occurring across wide geographical ranges is now possible because of the recent availability of large, highly detailed data sets describing human movement patterns (7) . Census data attempts to describe population migration that occurs between census years (8) . Traditional surveys of commuters (9, 10) , patients (11, 12) , or residents in disease-affected areas (13, 14) provide another description of recent travel activity. More recently, mobile phone service providers have shared privacy-protected data sets, showing how large numbers of users tend to move between cell phone towers (12, (15) (16) (17) (18) . Studies involving GPS trip loggers also track participants' movement activities (19) .\\nThe new abundance of movement data provide an exciting new opportunity to use models to quantify how much human host movement affects epidemiological outcomes. Nevertheless, each method of data collected only provides a partial picture of true movement patterns. It remains an unanswered question how best to use the data to parameterize models representing host movement. One consideration is finding a statistical model that fits adequately well to the data, predicting the frequency of travel or flow volume between two locations. There are already many candidate movement kernel models, such as gravity and radiation models, which have been evaluated against data in a variety of different settings (13, 18, (20) (21) (22) . A fitted movement kernel, once it has been evaluated as sufficiently accurate, may then be used to calibrate the movement-related parameters of a mechanistic disease transmission model.\\nFor the purposes of modeling movement together with infectious disease dynamics, however, an additional point of consideration is whether or not the details of the movement model-the rules which govern how hosts' movement patterns are represented-affect the model's quantitative behavior and subsequent predicted epidemiological outcomes. Two example classes of movement models which provide simple mechanistic representations of travel behavior are Eulerian movement models, which specify the rates at which hosts at one location travel to any other location, or Lagrangian movement models, which specify how frequently hosts travel away from home before returning (23) . Depending on the level of detail contained in the available data, one might be able to fit any number of candidate mechanistic movement models. There is precedent in calibrating different movement models using the same travel data, only to find that the disease dynamics have changed as a result (9) . Moving beyond the question of how to best fit a statistical model to that data set, the question becomes whether choosing one mechanistic movement model over another can affect epidemiological outcomes, and whether there are certain settings or parameter regimes where one type of movement model may be more accurate and appropriate.\\nIn the present analysis we explore these questions using compartmental metapopulation models which integrate together host movement and infectious disease dynamics. We use three infectious disease transmission models-the Susceptible-Infected-Recovered model; the Susceptible-Infected-Susceptible model; and the Ross-Macdonald model-which represent a suite of tools for modeling the transmission dynamics of a wide variety of pathogens. We adapt each of the disease transmission models to incorporate two mechanistic representations of movement: the first is an Eulerian movement model which we call the Flux model, and the second is a Lagrangian movement model which we call the Simple Trip model. We directly compare the Flux and Simple Trip models by setting parameters such that the total flux of travelers between each sub-population remains constant, thus emulating a case where one might have a single data set from which one could calibrate either movement model. We examine how disease model outcomes can change with different movement models for each transmission model. For all three transmission models, we find that the modeled quantities of interest relating to disease dynamics can differ dramatically depending on one's choice of movement model. We conclude by applying this analysis to a model of malaria transmission and importation on Bioko Island in Equatorial Guinea, and use intuition based on our prior analysis to frame our understanding for why the Flux model surprisingly fails to produce meaningful predictions of transmission intensity in that context.\\nHost movement models. We use metapopulation models to represent a network of geographically isolated sub-populations of hosts. Each sub-population occupies a site where local conditions may affect transmission intensity. Disease transmission is assumed to be completely local, such that hosts from different sub-populations come into contact with one another only if they travel to occupy the same location. We describe and compare two simple models of host movement, each of which represents a different set of rules governing how hosts move from one location to another. The \"Flux model\" is an Eulerian movement model which describes hosts as diffusing from one metapopulation to another (23):\\nwhere Ni counts the number of hosts currently located at site i. The total number of hosts remains constant over time\\nNi). The constant fi,j represents the rate at which hosts located at i travel to j, where fi,i = 0 for all i. Making an equilibrium assumption that the total number of residents at each site remains constant also constrains fi,j = fj,i, such that the fully specified Flux Model requires K(K − 1)/2 parameters. At equilibrium, when the derivatives on the left hand side of Eq. 1 equal zero the set of populations at each location may be found by solving for the nontrivial null vector of the matrix F where Fi,j = fi,j + fj,i.\\nThe \"Simple Trip model\" is a Lagrangian movement model which assigns home locations to each host and describes how hosts travel temporarily to other locations before returning home (23) . Unlike the Flux model, the Simple Trip model differentiates between the residents and the visitors currently located at the same site. Visitors are able to interact with residents at a given site, but they return home at fixed rates:\\nThe notation in Eq. 2 has changed to account for hosts retaining memory of their home sites: a host from i who is currently located at j will be counted as belonging to the Ni,j population, and the number of hosts whose home is i remains constant over time, even if members visit other sites (Ni = K j=1 Ni,j). The constant φi,j represents the rate at which hosts whose home is i travel to j, while the constant τi,j is the rate at which hosts visiting j from i return home to i. Both φi,i = 0 and τi,i = 0 for all i. The fully specified Simple Trip model requires 2K(K − 1) parameters, meaning that more information is required to parameterize this model than the Flux model.\\nAt equilibrium, when the derivatives on the left hand side of Eq. 2 equal zero, the population Ni is distributed across the K metapopulation sites as follows:\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04.05.20054304 doi: medRxiv preprint\\nWe can use Eqs. 3 to define a \"Time at Risk Matrix\" ψ where ψi,j ≡ N * i,j /Ni, the average number of individuals from i found at site j at equilibrium, or the fraction of time on average that hosts who live in site i spend in site j.\\nTo directly compare the quantitative behavior of both models, we will assume that at equilibrium the total number of hosts moving from one site to another is the same across both models. That is to say, the number of people traveling i → j in the Flux model is the same as the number of i residents traveling to j + the number of j resident travelers returning home from i in the Simple Trip model:\\nIt can be shown under these circumstances that the equilibrium populations in the Flux model N * i equal the equilibrium populations in the Simple Trip model. SIR Models. The next step is to incorporate both movement models into three compartmental models of disease transmission. The first such model is the SIR model, which describes a single outbreak of disease. For a single population, the deterministic ordinary differential equation SIR model is as follows:\\nCombining the basic SIR model with the Flux movement model 1, we obtain an analogous set of 3K equations:\\nThe parameters β and γ in Eq. 6 represent the transmission rate and recovery rate, respectively.\\nCombining the basic SIR model with the Simple Trip model 2, we obtain an analogous set of 3K 2 equations describing transmission between hosts while they are at their home location or while they are traveling:\\nNote that for both movement models we allow the parameter representing transmission intensity βi to vary across locations, not across individuals, a choice that reflects how transmission intensity reflects the contact network structure or other local conditions affecting transmission. γ, by contrast, is held constant for all individuals regardless of location. . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04.05.20054304 doi: medRxiv preprint SIS Models. Our second disease transmission model is the SIS model, which can be used to describe persistent endemic disease brought about by cycles of repeated infection. We adapt the SIS model to include either the Flux or Simple Trip models to describe movement between sub-populations. For a single population, the SIS model equation is as follows:\\nCombining the basic SIS model with the Flux model (Eq. 1), we obtain a set of K equations for transmission in the ith site:\\nfj,iIj [9] Combining the basic model with the Time at Risk model (Eq. 2), we obtain a set of K 2 equations among both residents at home and travelers who are away:\\nAgain, in both movement models the transmission intensity βi varies across locations, not across individuals, and γ is held constant for all individuals in all locations.\\nRoss-Macdonald Models. Our third disease model is the Ross-Macdonald model, which was developed to describe malaria transmission (24, 25) . The Ross-Macdonald model is similar to the SIS model, except it includes a vector population which serves as a mechanism of transmission from infected to susceptible human hosts. The Ross-Macdonald model tracks infection dynamics among human hosts (X) and vectors (Z). The basic Ross-Macdonald model is as follows:\\nThe first of Eqs. 11 describes how human hosts become infected through contact with infectious mosquitoes, while the second describes how mosquitoes become infected through contact with infectious humans. Analyzing Eq. 11 under equilibrium conditions, it is possible to solve for an expression for R0 as a function of prevalence (24, 25) :\\nAs before, we adapt the basic Ross-Macdonald model by allowing human hosts to move from one metapopulation to another. At present we will hold the Ross-Macdonald model's parameters (r, a, b, c, g, n) to be constant across all sites, while the stable mosquito population supported by the local environment (Mi) is allowed to vary across all sites. Incorporating the Flux model into the metapopulation Ross-Macdonald model (23):\\nWe again perform a standard analysis of Eqs. 13 under equilibrium conditions and obtain an expression for R {i} 0\\nin each site as a function of prevalence in each site Xi:\\nWe also extend the basic Ross-Macdonald model by allowing human hosts to move from one metapopulation to another as described by the Simple Trip model. Following the analysis of (23) and (17, 26) , we simplify the analysis using the Time\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04.05.20054304 doi: medRxiv preprint at Risk matrix ψ to rewrite the force of infection experienced by human hosts as a weighted average which combines local transmission and the average fraction time spent at risk in different locations:\\nThe expression κi represents the fraction of infectious humans located at i, accounting for both local residents as well as visitors from other locations. From Eq. 15, and following the analysis of (17), we obtain an expression for R0 which reflects how the movement model alters transmission at each metapopulation site:\\nShort Travel Duration\\nWe numerically integrate Eqs. 6 and 7 to find the size of the residual population of susceptibles following the outbreak. We plot the residual population fraction in location 1 (S ∞ 1 /N1) as a function of transmission intensity in location 1 (β1/γ) while holding transmission intensity in location 2 (β2/γ) constant.\\nWe explore four different parameter regimes defined by the transmission intensity of location 2 and the duration of travel. In all four parameter regimes there is a dramatic separation between the predicted outbreak sizes that depends on whether one uses the Flux (blue) or the Simple Trip (red) model. As long as residents of location 1 travel from a low to a high transmission environment (β2 > β1) the Simple Trip model's outbreak will be smaller and the residual population will be larger; the opposite is true when residents of location 1 travel from a high to a low transmission environment (β2 < β1). A: Short travel duration regime (φi = 1, τi,j = 10); high transmission in location 2 (β2 = 2.5) B: Short travel duration regime (φi = 1, τi,j = 10); low transmission in location 2 (β2 = 0.5) C: Long travel duration regime (φi = 0.01, τi,j = 0.1); high transmission in location 2 (β2 = 2.5) D: Long travel duration regime (φi = 0.01, τi,j = 0.1); low transmission in location 2 (β2 = 0.5) SIR models. We begin by directly comparing the predicted outcomes of the two versions of the SIR model with Flux and Simple Trip movement. We simulate epidemics in two coupled metapopulations of Ni = 500 human hosts by numerically integrating Eq. 6 and Eq. 7 and measuring the size of the outbreak. The population of susceptibles remaining after the outbreak has ended S∞ quantifies the size of the outbreak and allows one to infer R0 (27) . Furthermore, S∞ quantifies the population at risk in subsequent future epidemics, which is relevant for modeling seasonally recurring pathogens such as influenza (28) .\\nWe set γ = 1 in all locations, hold β2 constant, and measure the residual population S∞ in location 1 as β1 varies. We identify two transmission parameter regimes: β2 > γ, which is high enough to cause an outbreak in location 2, and β2 < γ, . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04.05.20054304 doi: medRxiv preprint which in isolation does not lead to an outbreak in location 2. We also define two travel parameter regimes, one where travel occurs frequently and lasts short periods of time (τi,j > γ), and one where travel occurs infrequently and lasts long periods of time (τi,j < γ). We specify the parameter values for φi,j and τi,j in each of these regimes for the Simple Trip model and use Eq. 4 to also parameterize the Flux model, such that we emulate using the same data inputs to parameterize both movement models. Figure 1 compares how the two SIR models with metapopulation movement behave in different parameter regimes. In all cases the outcomes are dramatically different, showing that SIR model predictions depend strongly on how host movement is represented in the metapopulation model. The two movement models do agree when transmission parameters are equal in both locations (β1 = β2), but there are large quantitative differences whenever there is some heterogeneity in the local transmission conditions across the different locations.\\nThe major difference between the Flux and Simple Trip models is that the Simple Trip model constrains the amount of time that a traveler spends away. For example, the larger residual population for the Simple Trip model, seen in the left hand column of Figure 1 , follows from a limit on the amount of time that residents of population one spend in the high-transmission location 2. Another difference between the Flux and the Simple Trip models is that hosts in the Flux model have no home residence and continue to move freely between populations after the outbreak ends. As a result, as the system reaches equilibrium after the epidemic the fractions of susceptibles become constant across all locations. In contrast with the Simple Trip model, the Flux model effectively erases any evidence that the transmission intensity differed between the two locations.\\nSIS models. We next compare the respective behavior of the SIS Flux metapopulation model (Eq. 9) and the SIS Simple Trip metapopulation model (Eq. 10) for K = 2 metapopulations of Ni = 500 coupled together. Again, we set γ = 1 at all locations. The outcome of interest for the SIS model is now prevalence, which we find by numerically solving for the endemic equilbria in Eq. 9 and Eq. 10. We hold β2 constant and solve for the prevalence in location 1 (I * 1 /N1 for the Flux model and (I * 1,1 + I * 1,2 )/N1 for the Simple Trip model) as a function of the transmission intensity in location 1 (β1). We define two transmission parameter regimes: β2 > γ, which is high enough to sustain endemic transmission locally in site 2, and β2 < γ, which cannot locally sustain endemic transmission. We again define two travel parameter regimes, one where travel occurs frequently and over short periods of time (τi,j > γ) and one where travel occurs infrequently and over long periods of time (τi,j < γ). In each regime we specify travel rate parameter values for the Simple Trip model and use Eq. 4 to set parameters for the Flux model.\\n! i,j > \" Long Travel Duration ! i,j < \" # 2 /\" > 1 # 2 /\" < 1 D C\\nWe numerically solve Eqs. 9 and 10 for the endemic equilibrium prevalence. We plot the endemic prevalence in location 1 I * 1 /N1 as a function of transmission intensity in location 1 (β1/γ) while holding transmission intensity in location 2 (β2/γ) constant. We explore four different parameter regimes defined by the transmission intensity of location 2 and the duration of travel. When travel duration is short (upper row), the Flux (red) and Simple Trip (blue) models produce dramatically different relationships between transmission intensity and prevalence. When travel duration is long (lower row), the two movement models' results converge. Figure 2 shows the results from both the Simple Trip and Flux models in each of the four parameter regimes. In the short travel duration regime (2 A, B), as long as the transmission intensity differs between the two locations, there is a noticeable separation between the two movement models' predicted relationships between prevalence and local transmission intensity. In 2 B it is even the case that the apparent critical point, above which location 1 has sustained endemic disease, changes depending on which movement model is used.\\nThe discrepancy in the short travel duration regime arises because the Simple Trip model imposes an extra constraint on the amount of time a traveler spends away from home. The flux model does not specify the amount of time spent away, and its movement rates (from Eq. 4 tend to overestimate the time spent away. The Simple Trip model endemic prevalence in Figure 2 A is suppressed compared to the Flux model endemic prevalence because the amount of time that travelers from location 1 spend at risk in the high transmission location 2 is constrained to be small. Similarly, the Simple Trip model endemic prevalence in 2 B is elevated because travelers spend less of their time at risk in the low transmission location 2. In the long travel duration regime (Figure 2, lower row) , however, if the trip time is longer than the disease infectious period there is more time for prevalence among travelers to location 2 to equilibrate with the prevalence among residents of location 2: in this parameter regime, the Flux model and Simple trip models converge, because the constraint on time spent away from home makes less of a difference.\\nWe examine the behavior of the Ross-Macdonald models with Flux and Simple Trip movement. We use the model to represent a malaria-causing pathogen with a mean infectious period of r −1 = 200 days. The outcome of interest in the Ross-Macdonald model is R0, which quantifies the local transmission intensity required to produce the prevalence. We compare the two movement models using K = 2 connected sub-populations of Ni = 500 human hosts by specifying prevalence and using Eqs. 14 and 16 to calculate R0 for the Flux and Simple Trip models, respectively. Again, we define two parameter regimes based on high (X = 0.25) versus low (X = 0.025) prevalence in location 2, and another pair of parameter regimes based on high frequency, short duration trips (φi = 1/360, τi,j = 1/10) versus low frequency, long duration trips (φi = 1/36000, τi,j = 1/1000). As before, Flux model parameters are defined from the Simple Trip model parameters using Eq. 4.\\n! i,j > \" Long Travel Duration ! i,j < \" # 2 /$ 2 = 0.25 # 2 /$ 2 = 0.025 D C\\nWe numerically solve Eqs. 14 and 16 to find the relationship between prevalence in location 1 (X1/N1) and transmission intensity (R0). We plot the relationship between location 1 prevalence (X1/N1) as a function of transmission intensity in location 1 while holding prevalence constant (X2/N2) in location 2. Note that we plot prevalence on the y-axis and transmission intensity on the x-axis, to emphasize how the Ross-Macdonald model behaves similarly to the SIS model ( Figure 2) . We explore four different parameter regimes defined by the prevalence of location 2 and the duration of travel in relation to the average duration of infection (r −1 = 200 days). In the short travel duration regime, there is a large quantitative difference in the relationship between prevalence and R0 for the two different movement models. In the regime with short travel duration and high location 2 prevalence (upper left), using the Flux model would suggest low transmission (R . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04.05.20054304 doi: medRxiv preprint location are different, there is a dramatic separation between the Flux and Simple Trip model results, particularly in the short travel duration regime (Figure 3 A, B) . If travel duration is long, however, the two models' predicted relationships between prevalence and R0 do converge towards one another. The relationships between prevalence and transmission intensity for the two Ross-Macdonald models are qualitatively similar to the one shown previously for the SIS model in Figure 2 .\\nIn some parameter regimes R0 does appear to run negative, such as with the Flux model predictions in Figure 3 A. This occurs when incidence and, by consequence, prevalence among travelers to a higher transmission environment is too high for the model to reconcile the prevalence specified for each area. In this context, R0 < 0 has no meaningful real-world interpretation beyond indicating that the data are inconsistent with the chosen mechanistic model. (14) . Each pixel shows the mean estimated prevalence within a 1 × 1 km 2 area. B Travel \"prevalence\" map, reproduced with permission from (14) , where each pixel shows the probability that a resident leaves home during an eight week survey period. C Modeled rates of travel {φ i,k } for each of four example pixels, estimated from MIS data reporting recent travel. Arrows pointing off-island represent travel to Mainland EG. Arrow weights are proportional to rate of travel from origin pixel to each of the 7 destination regions, or the sum total rate of travel to all of the destination pixels within each region.\\nWe further illustrate the quantitative disagreements between the Flux and Simple Trip models by applying the Ross-Macdonald models with host movement to the problem of estimating R0 in a real-world setting. Bioko Island is located in the Atlantic Ocean about 225km to the northwest of mainland Equatorial Guinea (EG) in Western Africa. From population data collected during a bednet mass-distribution campaign in 2018, human population count was 239,159 people, about 85% of whom lived in EG's capital city Malabo in the north (29) . During this sanitary census, however, there was a 12% absenteeism that needs to be accounted for. Following the recent efforts of an intensive malaria control and elimination program (Bioko Island Malaria Elimination Program, BIMEP), the average malaria parasite rate (PR, a measure of prevalence) in chhildren 2-14 years old has fallen from an average of 0.43 to 0.11 between 2004 and 2016, with further progress appearing to have stagnated in recent years since 2015 (30) . A recent analysis of data collected through malaria indicator surveys (MIS) (31-33) has used geostatistical techniques (34) to map the estimated prevalence of malaria (Figure 4 A) and the estimated probability of a resident leaving (Figure 4 B) across different locations on Bioko Island (14) . The analysis showed an elevated risk of infection among people who reported traveling recently (14) , a pattern corroborated by other studies (30, 35) . In particular, this suggests that travelers who had spent time in mainland EG, which still has an estimated mean prevalence of around 0.43 (36, 37) , are very likely to bring back infections with them that would contribute to the sustained endemic prevalence measured on Bioko Island. We can use the Ross-Macdonald model to infer local transmission intensity (R0) based on the known prevalence and travel behavior. We treat each of the pixels shown in the maps in Figure 4 as its own sub-population with its own local transmission environment, which allows the model to contain the spatial heterogeneity known from the data. We the population data to set the population Ni and the geospatial estimates to set prevalence Xi/Ni at each location. For the travel parameters we use trip destinations reported in the MIS (38) to fit a multinomial probability model of destination choice. Together with the travel frequency data (Figure 4 B) , we find a set of travel frequency parameters {φ i,k }, illustrated in Figure 4 C. We use MIS data on trip duration (38) to fit an exponential model of trip duration and find that travel within Bioko Island and to mainland equatorial had a mean duration of τ −1 i,BI = 21.2 and τ −1 i,EG = 10.3 days, respectively. (Further details may be found in the Supplementary Information.) The rate parameters fi,j for the Flux model were calculated using Eq. 4.\\nIn this way, we construct a fully parameterized pair of Ross-Macdonald models with movement between sub-populations, from which we can infer the transmission intensity in each location on Bioko Island. Figure 5 A maps estimates of R 0,Flux calculated using the Flux model (Eq. 14). The dark pixels of the map represent areas where the R 0,Flux estimates are calculated to be less than 0. In these locations, the rate at which people become infected while traveling and return home with malaria infections is too high for the transmission model to reconcile local transmission intensity with local prevalence. We frame our understanding of these results in terms of the K = 2 sub-population example discussed previously for the Ross-Macdonald Model. Each sub-population on Bioko Island exists in a parameter regime where travel duration is short (τi,j > r) and the prevalence in the off-island travel destination is higher than the local prevalence. We recall the results shown in Figure 3 A, where we see how for low prevalence values the Flux model tends to underestimate R0 compared to the Simple Trip model. The Simple Trip model constrains the amount of time that travelers spend on mainland EG, and therefore reduces the number of people who return home from traveling with malaria infections. The Flux model does not include such a constraint, and in this parameter regime we see how in some locations the rate of imported infections is too high and forces the transmission intensity to be negative.\\nWe have shown that models of infectious disease dynamics are sensitive to the choices that modelers make when including movement-mediated interactions between sub-populations. To do so, we have adapted each of three transmission models to include two types of movement between sub-populations. For all three transmission models we have identified parameter regimes where choosing to incorporate one movement model instead of the other will result in dramatically different results. Specifically, when there is a difference in transmission intensity between two interacting locations, and when mean travel duration is short compared to the mean duration of infection.\\nWe emphasize that in all instances we set movement parameters to match the total flux of individuals moving between each pair of sub-populations. In this way, we have emulated a situation where the modeler has a single movement data set and must choose which movement model to implement. The full impact of the choice of movement model became clear from the example of modeling malaria transmission and importation on Bioko Island: We calibrated each of the two movement models using the same data set and found that each movement model produced dramatically different results. The Flux model predicted R0 < 0, which has no meaningful interpretation beyond the Flux model being inconsistent with the data used to calibrate it. The Simple Trip model, by specifying the rate at which travelers return home, adds a constraint on the amount of time that travelers spend while away from home. As a consequence, when combined with a the Simple Trip model constrains the amount of risk experienced by travelers. The Flux model has no such constraint: as long as there are variations in the transmission intensity across different locations there will be significant differences between the two movement models' results.\\nThe way that each of the movement models contributes to the infectious disease model results can lead to real confusion over the true relationship between transmission intensity and epidemiological outcomes. In some applications, as with the Bioko Island example, the purpose of the model is to infer transmission conditions which may be difficult to measure empirically based on more readily available prevalence or incidence measures. Returning to Figure 2 A, we can imagine trying to determine the transmission intensity β1 based on a measured prevalence of 0.25. From the Flux model we infer β1 < 1, which suggests that local transmission risk is low. From the Simple Trip model, however, we infer β1 > 1, which suggests the opposite. A similar inconsistency is illustrated for the Ross-Macdonald model in Figure 3 , and appears again in the Bioko Island example in Figure 5 .\\nIn the case of modeling malaria transmission and importation on Bioko Island we have shown how the Flux model is not suitable for producing accurate results. Despite this drawback, the Flux model remains an attractive option for creating a basic quantitative description of host movement in other epidemiological contexts. First, the model requires only K(K − 1)/2 . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04.05.20054304 doi: medRxiv preprint parameters to specify completely. For this reason, parameterization is possible with relatively simple data sets which count the number of individuals moving from one each location to each other location over time. Unfortunately, data sets which record flux volumes do not necessarily lend themselves to parameterizing Simple Trip models, as they do not distinguish between the rates of leaving and returning home.\\nFor example, mobile phone call data records (CDRs) are a source of high-volume data sets for measuring the number of individuals who travel from one location to another. Often, for the sake of preserving subscriber privacy, CDRs are aggregated across individual users, time, and geographical area before becoming public. That is to say, publicly available CDR data sets count the number of mobile phones traveling from one service tower catchment area to another on a daily or weekly basis: these data sets do not record an individual's rate of leaving home or time spent while traveling away from home. The loss of individual identity from data sets has already been shown to impact the predicted outcome of disease models in other contexts (9) . While the limits and biases inherent to CDR data sets have been well-documented (39), the present study suggests an additional drawback when it comes to using CDR data sets for epidemiological modeling: because it is not clear that a Flux model is sufficient to produce accurate modeling results in some contexts, it is likewise not clear that flux data are sufficient to quantify travel behavior when building a movement model. This is not to say that CDR data or Flux models are useless, only that one needs to carefully consider the epidemiological context before incorporating a Flux model into a disease transmission model.\\nThe present work is restricted to discussing only two very basic deterministic models of movement. In theory, one may imagine similarly comparing more detailed and complicated movement models, such as incorporating travel behavior with multiple stops; adding stochastic effects; or using agent-based models to incorporate movement behavior heterogeneity among the travelers. It remains to be seen the extent to which such additional model features might further affect the epidemiological outcomes of interest. In practice, however, the data required for calibrating such models may not be readily available, and using a Flux or Simple Trip description of travel behavior may be more practical. Even for such simple movement models there is a risk of misusing the travel data that are available to calibrate a movement model which cannot be reconciled with the data and produces outputs that are difficult or impossible to interpret.\\nData and code supporting the analysis presented in the Malaria Modeling Example section, including Figures 3 and 4 , may be found at https://doi.org/10.6084/m9.figshare.12084831.v1.\\n  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1361</td>\n",
       "      <td>A Note on COVID-19 Diagnosis Number Prediction Model in China</td>\n",
       "      <td>[{'text': 'Importance: To predict the diagnosed COVID-19 patients and the trend of the epidemic in China. It may give the public some scientific information to ease the fear of the epidemic. Objective: In December 2019, pneumonia infected with the novel coronavirus burst in Wuhan, China. We aimed to use a mathematical model to predict number of diagnosed patients in future to ease anxiety on the emergent situation. Design: According to all diagnosis number from WHO website and combining with the transmission mode of infectious diseases, the mathematical model was fitted to predict future trend of outbreak. Setting: Our model was based on the epidemic situation in China, which could provide referential significance for disease prediction in other countries, and provide clues for prevention and intervention of relevant health authorities.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>In December of 2019, the increasing number of patients with pneumonia of unknown cause emerged in Wuhan, the capital of Hubei province, Central China. Most of the early patients had directly or indirectly related to wild animal source in the seafood and wet animal wholesale market through epidemiological survey [1] [2] [3] [4] [5] . Then laboratory tests found that the infection was caused by a novel coronavirus (now named as COVID-19 6-7 ).\\nThe spread of COVID-19 is expeditious. According to WHO statistics, the total number of diagnosed people in China was 309 in Jan 21 8 , but rapidly grew to 4537 in Jan 27 9 . Considering a simple constant exponential growth rate model [10] [11] , the diagnosis number will be unimaginable soon. This triggered a major concern and motivated this study [12] [13] [14] . A mathematical model is required to predict the spread trend and emergent situation.\\nOver the last few decades, mathematical models of disease transmission have been useful to gain insights into the transmission dynamics of infectious diseases and the potential role of intervention strategies 15 . Epidemic transmission models focused on the spread of an infectious disease through a population. But this spread can be modeled in two fundamental ways: by looking at the entire population as a single group or set of linked subgroups (population-based) or by considering everyone that makes up the population and how interactions between unique individuals facilitate disease spread (individual-based). Common models of infectious disease included meta-population models, individual-based network models, and simple SIR-type models that incorporate the effects of reactive behavior changes or inhomogeneous mixing 16 .\\nConsidering the limited number of data points and the complexity of the real situation, a simple but robust model is expected to work better than sophisticated epidemiology models. In this paper, we propose a robust model for next day diagnosis number predict. This model had excellent performance on prediction of infected patients in past few days. The prediction is expected to provide practical significance on social and economic application. &amp; # and its median aggregates were calculated (Fig.1) . It was observed that &amp; # manifests a random fluctuation around a constant center. It is reasonable to assume that is a constant. The estimation of currently (Feb 10) is 0.904. A future diagnosis number formula is obtained based on this estimation:\\nPrevious numbers of diagnoses in China are used to predict the number for the next day. The prediction is started from Jan 29 as several data points are needed to build the model (Table 1) . We obtain nice concordance (&lt;2.5% error) with the real numbers next day. \\nWe predict the new diagnosis number in China according to our model as Fig.2 . The model predicts that new diagnosis will last to March.\\nOur model indicates that the COVID-19 disease infection rate decreases exponentially. This model converges to a maximum number as time increases, indicating a limited impact of COVID-19. The model also predict that the new diagnosis number will last to March, indicating that the COVID-19 disease will be well controlled.\\nWe only analyzed the total number of diagnoses in China. The province scale data is available but is not as complete as total number. Here, we did not show these details analyses.\\nOur model is attractive as it not only predicts data accurately but also is minimal.\\n. CC-BY-NC-ND 4.0 International license It is made available under a is the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint . https://doi.org/10.1101/2020.02.19.20025262 doi: medRxiv preprint\\nThe minimal number of parameters and robust statistics make our model robust in the real-world scenario. We develop a formula to predict the next day diagnosis number:\\n) \" ，and estimate that is 0.904( &lt;1) which shows that the diagnosis number of COVID-19 patients will decrease. It is very effective in controlling the spread of infectious diseases that the Chinese health authorities have taken appropriate health care actions. If the outbreak is no longer controlled, the value of will be around 1 indicating that the outbreak will wreak havoc. We obtain this elegant formula with both curve plot visualization and mathematical intuition. We plot ln (Ni/Ni-1) against time (Fig.3 ). First few observations showed a linear decay of spread rate. However, later data points realized us at Feb 4 that it is an exponential decay. Later data points further confirm our model. \\nDaily diagnosis number of China is download from WHO situation reports (https://www.who.int/emergencies/diseases/novel-coronavirus-2019/situation-reports). The data starts from Jan 21 and currently it ends at Feb 10.\\nThe following model is proposed based on our observations:\\nwhere Ni is the number of diagnosed patients in China according to WHO at the ith day, is a parameter modeling the decay of diagnosis growth rate. A simple estimation of at the ith day would be\\nAs we have multiple days of observations, historical &amp; # with the median statistic, which is well known for its robustness, can be aggregated.\\nThe source code of model estimation is at: https://github.com/wangyi-fudan/ncp_ model\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                              title  \\\n",
       "1361  A Note on COVID-19 Diagnosis Number Prediction Model in China   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           abstract  \\\n",
       "1361  [{'text': 'Importance: To predict the diagnosed COVID-19 patients and the trend of the epidemic in China. It may give the public some scientific information to ease the fear of the epidemic. Objective: In December 2019, pneumonia infected with the novel coronavirus burst in Wuhan, China. We aimed to use a mathematical model to predict number of diagnosed patients in future to ease anxiety on the emergent situation. Design: According to all diagnosis number from WHO website and combining with the transmission mode of infectious diseases, the mathematical model was fitted to predict future trend of outbreak. Setting: Our model was based on the epidemic situation in China, which could provide referential significance for disease prediction in other countries, and provide clues for prevention and intervention of relevant health authorities.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    full_text  \n",
       "1361  In December of 2019, the increasing number of patients with pneumonia of unknown cause emerged in Wuhan, the capital of Hubei province, Central China. Most of the early patients had directly or indirectly related to wild animal source in the seafood and wet animal wholesale market through epidemiological survey [1] [2] [3] [4] [5] . Then laboratory tests found that the infection was caused by a novel coronavirus (now named as COVID-19 6-7 ).\\nThe spread of COVID-19 is expeditious. According to WHO statistics, the total number of diagnosed people in China was 309 in Jan 21 8 , but rapidly grew to 4537 in Jan 27 9 . Considering a simple constant exponential growth rate model [10] [11] , the diagnosis number will be unimaginable soon. This triggered a major concern and motivated this study [12] [13] [14] . A mathematical model is required to predict the spread trend and emergent situation.\\nOver the last few decades, mathematical models of disease transmission have been useful to gain insights into the transmission dynamics of infectious diseases and the potential role of intervention strategies 15 . Epidemic transmission models focused on the spread of an infectious disease through a population. But this spread can be modeled in two fundamental ways: by looking at the entire population as a single group or set of linked subgroups (population-based) or by considering everyone that makes up the population and how interactions between unique individuals facilitate disease spread (individual-based). Common models of infectious disease included meta-population models, individual-based network models, and simple SIR-type models that incorporate the effects of reactive behavior changes or inhomogeneous mixing 16 .\\nConsidering the limited number of data points and the complexity of the real situation, a simple but robust model is expected to work better than sophisticated epidemiology models. In this paper, we propose a robust model for next day diagnosis number predict. This model had excellent performance on prediction of infected patients in past few days. The prediction is expected to provide practical significance on social and economic application. & # and its median aggregates were calculated (Fig.1) . It was observed that & # manifests a random fluctuation around a constant center. It is reasonable to assume that is a constant. The estimation of currently (Feb 10) is 0.904. A future diagnosis number formula is obtained based on this estimation:\\nPrevious numbers of diagnoses in China are used to predict the number for the next day. The prediction is started from Jan 29 as several data points are needed to build the model (Table 1) . We obtain nice concordance (<2.5% error) with the real numbers next day. \\nWe predict the new diagnosis number in China according to our model as Fig.2 . The model predicts that new diagnosis will last to March.\\nOur model indicates that the COVID-19 disease infection rate decreases exponentially. This model converges to a maximum number as time increases, indicating a limited impact of COVID-19. The model also predict that the new diagnosis number will last to March, indicating that the COVID-19 disease will be well controlled.\\nWe only analyzed the total number of diagnoses in China. The province scale data is available but is not as complete as total number. Here, we did not show these details analyses.\\nOur model is attractive as it not only predicts data accurately but also is minimal.\\n. CC-BY-NC-ND 4.0 International license It is made available under a is the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint . https://doi.org/10.1101/2020.02.19.20025262 doi: medRxiv preprint\\nThe minimal number of parameters and robust statistics make our model robust in the real-world scenario. We develop a formula to predict the next day diagnosis number:\\n) \" ，and estimate that is 0.904( <1) which shows that the diagnosis number of COVID-19 patients will decrease. It is very effective in controlling the spread of infectious diseases that the Chinese health authorities have taken appropriate health care actions. If the outbreak is no longer controlled, the value of will be around 1 indicating that the outbreak will wreak havoc. We obtain this elegant formula with both curve plot visualization and mathematical intuition. We plot ln (Ni/Ni-1) against time (Fig.3 ). First few observations showed a linear decay of spread rate. However, later data points realized us at Feb 4 that it is an exponential decay. Later data points further confirm our model. \\nDaily diagnosis number of China is download from WHO situation reports (https://www.who.int/emergencies/diseases/novel-coronavirus-2019/situation-reports). The data starts from Jan 21 and currently it ends at Feb 10.\\nThe following model is proposed based on our observations:\\nwhere Ni is the number of diagnosed patients in China according to WHO at the ith day, is a parameter modeling the decay of diagnosis growth rate. A simple estimation of at the ith day would be\\nAs we have multiple days of observations, historical & # with the median statistic, which is well known for its robustness, can be aggregated.\\nThe source code of model estimation is at: https://github.com/wangyi-fudan/ncp_ model\\n  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1761</td>\n",
       "      <td>Public Exposure to Live Animals, Behavioural Change, and Support in Containment Measures in response to COVID-19 Outbreak: a population-based cross sectional survey in China</td>\n",
       "      <td>[{'text': 'Word count (main text): 3489 : medRxiv preprint', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'The public responded quickly to the outbreak, and reduced exposure to live animals, especially in Wuhan. With high public support in containment measures, better regulation of wet markets and healthy diets should be promoted.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>In response to the COVID-19 outbreak, we aimed to investigate behavioural change on exposure to live animals before and during the outbreak, and public support and confidence for governmental containment measures.\\nA population-based cross-sectional telephone survey via random dialing was conducted in Wuhan (the epicentre) and Shanghai (an affected city with imported cases) between 1 and 10 February, 2020. 510 residents in Wuhan and 501 residents in Shanghai were randomly sampled. Differences of outcome measures were compared before and during the outbreak, and between two cities.\\nProportion of respondents visiting wet markets at usual was 23.3% (119/510) in Wuhan and 20.4%\\n(102/501) in Shanghai. During the outbreak, it decreased to 3.1% (16) in Wuhan (p&lt;0·001), and 4.4% (22) in Shanghai (p&lt;0·001). Proportion of those consuming wild animal products declined from 10.2%\\n(52) to 0.6% (3) in Wuhan (p&lt;0·001), and from 5.2% (26) to 0.8% (4) in Shanghai (p&lt;0·001). 79.0%\\n(403) of respondents in Wuhan and 66.9% (335) of respondents in Shanghai supported permanent closure of wet markets (P&lt;0.001). 95% and 92% of respondents supported banning wild animal trade and quarantining Wuhan, and 75% were confident towards containment measures. Females and the more educated were more supportive for the above containment measures.\\nAll rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint 4\\nEvidence before this study On February 19, 2020, we searched PubMed for papers published after January 1, 2020, containing the following terms: \"2019 nCoV\" or \"COVID-19\". We identified 179 studies, most of which are research on clinical and epidemiological characteristics of COVID-19. To date there is no primary research to quantify public behavioural response and support in containment measures in response to the outbreak.\\nOnly four commentaries mentioned the influence of the outbreak on mental health. One commentary introduced the habit of consuming wild animal products in China. Another commentary briefly introduced isolation, quarantine, social distancing and community containment as public health measures in the outbreak. The Chinese government has introduced a series of strict containment measures, and societal acceptability of these measure is important for effective and sustained response.\\nEvidence is urgently needed to help policy makers understand public response to the outbreak and support for the containment measures, but no evidence available to date.\\nWe conducted a population-based cross-sectional telephone survey via random digital dialing in Wuhan (the epicentre) and Shanghai (an affected city with imported cases) between 1 and 10 February, 2020.\\nTo date, this is the only few analyses on behavioural response to the outbreak and societal acceptability of governmental containment measures, which has been listed as the current priority of China CDC. We provide an assessment of behavioural change on exposure to live animals during the outbreak, by comparison before and during the outbreak, and between two cities with diverse exposure intensities to COVID-19. We also provide evidence on public support in governmental containment measures, including strict regulation on wet markets to reduce animal-to-human transmission and city quarantine All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint 5 to reduce human transmission.\\nWe found that wild animal consumption was more prevalent in Wuhan (10.2%) than in Shanghai (5.2%). The public responded quickly to the outbreak, and significantly reduced exposure to live animals and stopped wild animal consumption, especially in Wuhan. They were very supportive of governmental containment measures. With high public support, wet markets should be better regulated, and healthy diets, including changing the traditional habit of eating wild animal products, should be promoted. This can inform policy makers in China and other countries to implement and adjust containment strategies in response to the outbreak in the future.\\nEarly in December 2019, several pneumonia cases caused by the novel coronavirus (now known as were first reported in Wuhan city, China. [1] [2] [3] The majority of the earliest cases were epidemiologically traced to the local Huanan (Southern China) Seafood Wholesale Market, where some wild animals were sold. [1] [2] [3] [4] On Jan 26, novel coronavirus was detected in environmental samples from this market by China CDC, suggesting that the virus possibly originated in wild animals sold in the market. [3] [4] This pathogenic virus has been identified as a new strain of coronavirus, which is in the same family as severe acute respiratory syndrome coronavirus (SARS-CoV) and Middle East respiratory syndrome coronavirus (MERS-CoV). [4] [5] The COVID-19 outbreak was declared by World Health Organization as a Public Health Emergency of International Concern (PHEIC) on Jan 30, 2020. 6 As of Feb 16, 2020, a total of 70,548 cases have been reported in mainland China, with 1,770 resulting deaths. 7 The outbreak has now spread beyond China to twenty-five other countries. 8 The SARS-CoV and MERS-CoV were proven to have originated from wild animals. [9] [10] [11] [12] [13] [14] The COVID-19 is also likely to have a zoonotic origin. The practice of consuming wild animal products in China dates back to prehistoric times and persists into today. 9 Trade and consumption of wild animals were not well regulated even after SARS. Until the amendment of Wild Animal Protection Act in 2016, the prohibition was firstly added, but only for a short list of the nationally protected animals. Previous studies explored population exposure to live poultry in China, [15] [16] [17] but information on exposure to wild animals is limited.\\nIn the efforts to contain the spread of the virus, the Chinese government introduced a series of All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint containment measures. 18 Specifically, to lower the risk of zoonotic virus transmission, China has issued strict regulations on wet markets where live and/or wild animals are traded, including closing live poultry markets (LPMs) and bans on wild animal transactions. Furthermore, Wuhan city has been quarantined to stop human-to-human transmission: starting on Jan 23, 2020, public transportation in Wuhan was closed, and all flights and trains leaving from Wuhan were cancelled.\\nThe COVID-19 outbreak and subsequent containment measures to lower the zoonotic virus transmission offer a unique opportunity to reassess the exposure to live animals, demand for wild animal consumption, and public readiness to further regulate wet markets. In this study, we aim to investigate behaviour patterns of population interaction to live animals, with a particular focus on differences before and during the outbreak, between Wuhan and Shanghai representing diverse exposure intensities to COVID-19. We also seek to understand levels of public support for containment measures and public confidence in their effectiveness to quell the outbreak. This study will help to clarify the public's response to the outbreak and has important policy implications regarding the social acceptability of infection containment measures and their application in different contexts.\\nA population-based cross-sectional telephone survey was conducted in Wuhan and Shanghai between Feb 1 and 10, 2020, one month since the outbreak. These two cities were selected to represent diverse exposure intensities to COVID-19. Wuhan is the site of origin of the COVID-19 outbreak. Shanghai was selected due to its status as a city that has been significantly affected by imported COVID-19 cases All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint 8 from Wuhan. The first imported case was confirmed in Shanghai on Jan 20. During the period of our survey, the number of confirmed cases rose rapidly in Wuhan, from 4,109 to 18,454, but maintained low epidemic rates in Shanghai, with reported cases rising from 182 to 303. Figure 1 illustrates the timeline of the outbreak progression and containment measures. This provides an opportunity to assess and compare public behavioural response and support for containment measures during the outbreak between two cities with different exposure intensities.\\nThe sample size (n=500 for each city) is sufficient to estimate the change of outcomes (p=0.5) with a margin of error of 0.05 and a 95% confidence interval, according to previous studies. [15] [16] Our target population was persons aged 18 or over who had been living in the selected city for at least 6 months in the year prior to the survey and were current residents at the time of the survey. In each city, the telephone survey was conducted through a computer-assisted interviewing system, which enabled the random generation of mobile phone numbers and systematic data collection. We attempted to contact each generated number three times a day at different hours; if no contact was established after the third call, this number was classified as invalid or unreachable. Random dials with proportional quota sampling were used to ensure that respondents were demographically representative of the general population in each city, with quotas based on sex and age. 19\\nThe outcomes of interest we studied were exposure to live animals before and during the COVID-19 outbreak, public support for containment measures that aim to reduce population exposure to live animals and human transmission, and confidence in these measures to quell the outbreak. Exposure to All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint live animals includes the following behaviours: visiting the live poultry or wild animal markets, buying live poultry or wild animals (i.e., frequency of purchases, practice of picking up live animals before purchasing, location where the purchased animals were slaughtered), eating wild animal products, raising pets, and touching stray animals. To measure behavioural change during the outbreak, each respondent was asked to report exposing behaviours both (1) prior to the outbreak: before Dec 31, 2019 when COVID-19 cases were first released by Wuhan Health Commission and (2) since the outbreak, between Jan 1 to Feb 10, 2020. For live animal consumption habits, where behaviour change cannot be measured in a short period of time, we assessed people's intention to: 1) buy live poultry or wild animals, and eat wild animal products (continue to buy/eat as usual, buy/eat less, not buy/eat any more) and 2) continue raising pets and touching stray animals (continue to raise/touch as usual, temporarily not during the outbreak, try or resolutely not even after the outbreak).\\nWe measured public support for two types of governmental containment measures in response to the outbreak; one is regulation on wet markets to reduce population exposure to live animals, and the other is quarantining measures meant to reduce human transmission, such as the closure of public transportation in Wuhan and the cancellation of flights and trains leaving from Wuhan. Each respondent was asked to rate the extent, on a five-point Likert scale, to which they supported 1) the permanent closure of LPMs, 2) a ban on the wild animal trade, and 3) the quarantining of Wuhan; respondents were also asked to report the extent of their confidence in the effectiveness of these measures to quell the outbreak.\\nCharacteristics of respondents, including gender, age, marital status, education, income, and All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint employment status, were collected in our survey. We also collected the history of fever or cough symptoms for respondents and their relatives or friends in the two weeks prior to the survey, and whether there were confirmed or suspected COVID-19 cases in the neighbourhood in which they live.\\nEach respondent was also asked at what point they started paying attention to this outbreak.\\nThe survey questionnaire was adjusted according to an instrument used during the outbreaks of SARS in 2003 20-21 and influenza A(H7N9) in 2013. 15 The survey instrument was pretested for face and content validity, length, and understandability by the public before our survey was administered.\\nDescriptive and univariate analysis were used to investigate changes in exposure to live animals during the outbreak, public support for containment measures, and confidence in containment measures to quell the outbreak, between Wuhan (the epicentre) and Shanghai (an affected city with imported cases).\\nChi-square tests or fisher's exact tests (when applicable) were used to compare differences in these measures between the two cities, and before and during the outbreak. Multivariate logistic regression was used to estimate factors associated with public support for containment measures and confidence in intervention effectiveness. The proportion and Odds Ratio with 95% confidence intervals (CI) were reported. All statistical analyses were performed using Stata 14.0 (StataCorp LP, College Station, TX).\\nA total of 18,576 telephone numbers were randomly selected via digital dialing system in Wuhan and Shanghai. After calling, 6,836 respondents were eligible to participate, of whom 5,640 declined and All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint 11 185 terminated before completing the interview. In total, 510 respondents in Wuhan and 501 respondents in Shanghai completed the interview, and the overall response rate was 14.8%\\n(1,011/6,836) (Appendix Figure 1 ).\\nRespondents in Wuhan and Shanghai were generally similar in terms of sex, marital status and occupation, but in Shanghai respondents tended to be older and had higher education and income levels than in Wuhan (Table 1 ). Our sample was consistent with characteristics of the general population in the two cities, as Shanghai is more developed and has an older population than Wuhan. Expectedly, respondents in Wuhan were much more intensely exposed to the COVID-19 outbreak and paid attention to the outbreak earlier than those in Shanghai.\\nWe first investigated patterns of usual exposure to live animals before the outbreak ( Table 2 Panel of 55, 58.2% in Shanghai, p=0.339); more than 90% of respondents always arranged for slaughter of purchased animals in wet markets, with no noticeable differences between the two cities. However, residents in Wuhan reported wild animal consumption at a higher rate than those in Shanghai; 47 (9.2%) of 510 respondents in Wuhan and 23 (4.6%) of 501 respondents in Shanghai reported eating All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is To investigate effects of the outbreak, we further assessed behavioural changes in exposure to live animals before and during the outbreak (Figure 2) , and intention to change after the outbreak ( Table 2 Panel B). We found that respondents dramatically reduced exposure to live animals during the outbreak, with respondents in Wuhan changing their exposure much more significantly than in Shanghai. Specifically, the frequency of visiting LPMs decreased from 23.3% (n=119) to 3.1% (n=16)\\nin Wuhan (p&lt;0·001), and from 20.4% (n=102) to 4.4%(n=22) in Shanghai (p&lt;0·001; Figure 2 ). The proportion of respondents buying live poultry or wild animals also decreased from around 11.0% to 2.6% in both cities (p&lt;0·001). We also asked whether respondents intended to change their habit of buying live poultry or wild animals since the outbreak among those exposing to live animals at usual ( (4/55) of respondents in Wuhan and 25.5% (14/55) in Shanghai stated they would continue to buy these products as usual.\\nThe proportion of respondents consuming wild animals declined significantly, 10.2%(n=52) to 0.6%(n=3) in Wuhan (p&lt;0·001) and 5.2%(n=26) to 0.8%(n=4) in Shanghai (p&lt;0·001) (Figure 2 ). Most All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is Wuhan, respectively, and 75% of respondents were confident that governmental containment measures would quell the outbreak. Table 3 presents socio-demographic factors associated with public support for the governmental containment measures implemented (i.e. closure of LPMs, banning the wild animal trade, and city quarantine) and confidence in the effectiveness of governmental response. There was evidence that females were more supportive towards all three containment measures and that there was a positive association between education level and the likelihood of supporting regulations on wet markets and trade of wild animals. Unsurprisingly, those who self-reported visiting a wet market and eating wild All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint animal products prior to the outbreak are much less likely to support closure of the market or a ban on the trade as containment measures (adjusted OR=0.58, 95%CI [0.41-0.82] and adjusted OR=0.31, 95%CI [0.14-0.70], respectively). We did not find any difference in population profiles of those who support Wuhan quarantine or show confidence in the effectiveness of governmental response.\\nThis study is the first to investigate behavioural changes of exposure to live animals during the COVID-19 outbreak, and public support for governmental containment measures in response to the outbreak. We found that population exposures to live animals were similar in Wuhan and Shanghai before the outbreak, except that wild animal consumption was more prevalent in Wuhan than in Shanghai. During the outbreak, residents in both cities significantly changed their behaviours and reduced exposures to live animals, and such changes were much more significant in Wuhan. Most respondents supported governmental containment measures implemented, and three in four had confidence on the effectiveness of these measures. Male, the less educated and those exposing to live animals at usual were less likely to support containment measures.\\nIn our survey, more than 20% of respondents visited wet markets at usual, which was similar in Wuhan and Shanghai, despite Shanghai being much more developed. Compared with the 30% level reported in 2013 in the same study sites, 15 the share of respondents visiting wet markets declined by 10 percentage points, but was still high. 10% of respondents consumed products of wild animals in Wuhan, twice that reported in Shanghai. In China, there exists a long tradition of eating the meat and products of wild animals, which persists into modern times due to a false but commonly held belief that these products All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint have therapeutic effects. [22] [23] These beliefs and preferences for fresh food are especially popular in southern China, where Wuhan is located, but were greatly challenged and changed during the 2003 SARS outbreak in China 9,12 . However, these practices have become popular again in recent years, due in part to the popularity of social media influencers who broadcast themselves eating wild animal products on livestream platforms. 22 The revival of this tradition may have set the socio-biological context for the occurrence and spread of COVID-19 in China.\\nWe found that respondents' risks of exposure to live animals declined significantly, reaching a very low level, and most respondents stopped wild animal consumption after this outbreak, especially in Wuhan.\\nThis indicates that the public responded quickly and adequately to the outbreak. In the survey, all respondents cared about the outbreak, and about one in five respondents in Wuhan reported starting to pay attention before Dec 31, 2019. The outbreak gradually gained intense nationwide attention, 24 and a series of policies were implemented by the Chinese government to contain the outbreak. As of Jan 30, all provinces had initiated first-level responses to major public health emergencies. [25] [26] These actions raised public attention further and induced significant behavioural changes. A national online survey on Jan 26 showed that the vast majority of people knew of the disease and related preventive measures, 66.5% reported that they would wear masks when outside their home, and 80% cancelled travel plans during the Spring Festival due to the outbreak. 27 The findings from both this survey and ours demonstrate behavioural change in terms of exposure to live animals or daily protection.\\nIn addition, more strict regulations were implemented on wet markets to lower the risk of disease transmission. Wuhan instituted prohibited the sale of live poultry and wild animals starting on Jan 22. All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint Shanghai started its seasonal closure of LPMs on Jan 25 this year. The Chinese Government further instituted nationwide bans on all forms of wild animal transactions on Jan 26. 28 These regulations may accelerate the decrease in exposing behaviours to live animals. However, the COVID-19 epidemic occurred prior to the Chinese lunar new year, when the demand for live animals is generally high. 23 This may produce a conflict between the public demand for live animals and regulations. Fortunately, we found that the public strongly supports the regulations on wet markets to control the outbreak.\\nAlthough permanent closure of LPMs was not fully backed by the public, it was still much higher than reported during the 2013 influenza A outbreak (27.3% in Wuhan, 39.3% in Shanghai). 15, 29 The strong support for this policy offers a great opportunity to improve regulations on wet markets and change the prevalent habit of eating wild animal products in China.\\nWe believe that the keys to quelling outbreaks of zoonotic viral diseases such as COVID-19 and preventing future outbreaks are changing the false beliefs and dangerous habits regarding eating wild animal products and strengthening regulations on wet markets. Firstly, healthy diets need to be vigorously promoted in China. Health China 2030 promotes healthy diets, but does not discuss the risks of eating wild animals. We also believe the internet and social media can be an effective channel to promote healthy eating habits and reduce the inappropriate and outdated tradition of eating wild animal products. 25, 30 Since the outbreak, spontaneous initiatives have emerged on the internet to explain the risks of and resist consuming wild animals. Secondly, live animal transactions should be more strictly regulated, and wild animal transactions should be even prohibited in the long term. While different animals should be sold separately in specific wet markets to lower risks of virus transmission, in practice, various kinds of live animals are sold together within one market, as found in Huanan Seafood All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint Wholesale Market. Laws should be strengthened to regulate and implement the strict separation of live animals from general markets. With the currently high public support, bans on the wild animal trade should be promoted, but the government must be aware of the risk of creating an unregulated black market by instituting a ban. For live poultry and non-wild animals, better regulations may be a better option than bans without the full support of the public. 23 The feasible measures include seasonal closure of wet markets, regular rest days, and banning leaving live animals in markets overnight. 10, 15 The above measures should be given priority in China and other countries to reduce current levels of COVID-19, as well as future threats from zoonotic diseases.\\nOur study has several limitations. First, there may be recall bias. We aimed to identify behavioral changes in a cross-sectional survey, and outcomes before the outbreak were measured based on participant recall. Second, our results may be affected by selection bias from telephone survey. We did attempt multiple calls to unanswered numbers to mitigate this bias, and quota sampling was introduced to get a representative sample. The anonymous phone interview addressed the concern of response bias due to fear of being seen as criticizing the government. Third, a cross-sectional design prevents us from assessing how long the effect on behavioral/intention change demonstrated in this study will last, when the outbreak is no longer fresh in people's minds. Thus, we plan to conduct a second-round survey at the end of the outbreak to track the long-term changes.\\nIn conclusion, exposure to live animals was common before the COVID-19 outbreak. The public responded quickly to this outbreak, and significantly reduced their exposure to live animals and stopped wild animal consumption, especially in Wuhan. They were very supportive and confident for All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint current mitigating measures. With high public support, wet markets should be better regulated, and healthy diets, including changing the habit of wild animal consumption, should be promoted, especially for male, the less educated and those exposing to live animals at usual. \\nThe study was approved by the Institutional Review Board of the School of Public Health, Fudan University (IRB#2020-TYSQ-01-1). Verbal informed consent was obtained from all participants.\\nH.Y. has received research funding from Sanofi Pasteur, GlaxoSmithKline, Yichang HEC Changjiang Pharmaceutical Company and Shanghai Roche Pharmaceutical Company. None of that research funding is related to 2019-nCov. All data are n (%). P value are from Chi square test or fisher's exact tests (when applicable).\\n*Fever or cough symptoms in the past 2 weeks.\\nAll rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint Resolutely not after outbreak 6(14.0) 3(5.2) All rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity. All rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity. Odds ratios with 95% CI were obtained from multivariate binary logistic regressions. Significance level: ** p&lt;0.01, * p&lt;0.05.\\nAll rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint WHO stands for World Health Organization; PHEIC stands for public health emergency of international concern.\\nAll rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint Notes: Proportions are shown, and error bars represent 95% CIs.\\nAll rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint\\nNotes: p value is shown between Wuhan and Shanghai.\\nAll rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint \\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                              title  \\\n",
       "1761  Public Exposure to Live Animals, Behavioural Change, and Support in Containment Measures in response to COVID-19 Outbreak: a population-based cross sectional survey in China   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                              abstract  \\\n",
       "1761  [{'text': 'Word count (main text): 3489 : medRxiv preprint', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'The public responded quickly to the outbreak, and reduced exposure to live animals, especially in Wuhan. With high public support in containment measures, better regulation of wet markets and healthy diets should be promoted.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             full_text  \n",
       "1761  In response to the COVID-19 outbreak, we aimed to investigate behavioural change on exposure to live animals before and during the outbreak, and public support and confidence for governmental containment measures.\\nA population-based cross-sectional telephone survey via random dialing was conducted in Wuhan (the epicentre) and Shanghai (an affected city with imported cases) between 1 and 10 February, 2020. 510 residents in Wuhan and 501 residents in Shanghai were randomly sampled. Differences of outcome measures were compared before and during the outbreak, and between two cities.\\nProportion of respondents visiting wet markets at usual was 23.3% (119/510) in Wuhan and 20.4%\\n(102/501) in Shanghai. During the outbreak, it decreased to 3.1% (16) in Wuhan (p<0·001), and 4.4% (22) in Shanghai (p<0·001). Proportion of those consuming wild animal products declined from 10.2%\\n(52) to 0.6% (3) in Wuhan (p<0·001), and from 5.2% (26) to 0.8% (4) in Shanghai (p<0·001). 79.0%\\n(403) of respondents in Wuhan and 66.9% (335) of respondents in Shanghai supported permanent closure of wet markets (P<0.001). 95% and 92% of respondents supported banning wild animal trade and quarantining Wuhan, and 75% were confident towards containment measures. Females and the more educated were more supportive for the above containment measures.\\nAll rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint 4\\nEvidence before this study On February 19, 2020, we searched PubMed for papers published after January 1, 2020, containing the following terms: \"2019 nCoV\" or \"COVID-19\". We identified 179 studies, most of which are research on clinical and epidemiological characteristics of COVID-19. To date there is no primary research to quantify public behavioural response and support in containment measures in response to the outbreak.\\nOnly four commentaries mentioned the influence of the outbreak on mental health. One commentary introduced the habit of consuming wild animal products in China. Another commentary briefly introduced isolation, quarantine, social distancing and community containment as public health measures in the outbreak. The Chinese government has introduced a series of strict containment measures, and societal acceptability of these measure is important for effective and sustained response.\\nEvidence is urgently needed to help policy makers understand public response to the outbreak and support for the containment measures, but no evidence available to date.\\nWe conducted a population-based cross-sectional telephone survey via random digital dialing in Wuhan (the epicentre) and Shanghai (an affected city with imported cases) between 1 and 10 February, 2020.\\nTo date, this is the only few analyses on behavioural response to the outbreak and societal acceptability of governmental containment measures, which has been listed as the current priority of China CDC. We provide an assessment of behavioural change on exposure to live animals during the outbreak, by comparison before and during the outbreak, and between two cities with diverse exposure intensities to COVID-19. We also provide evidence on public support in governmental containment measures, including strict regulation on wet markets to reduce animal-to-human transmission and city quarantine All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint 5 to reduce human transmission.\\nWe found that wild animal consumption was more prevalent in Wuhan (10.2%) than in Shanghai (5.2%). The public responded quickly to the outbreak, and significantly reduced exposure to live animals and stopped wild animal consumption, especially in Wuhan. They were very supportive of governmental containment measures. With high public support, wet markets should be better regulated, and healthy diets, including changing the traditional habit of eating wild animal products, should be promoted. This can inform policy makers in China and other countries to implement and adjust containment strategies in response to the outbreak in the future.\\nEarly in December 2019, several pneumonia cases caused by the novel coronavirus (now known as were first reported in Wuhan city, China. [1] [2] [3] The majority of the earliest cases were epidemiologically traced to the local Huanan (Southern China) Seafood Wholesale Market, where some wild animals were sold. [1] [2] [3] [4] On Jan 26, novel coronavirus was detected in environmental samples from this market by China CDC, suggesting that the virus possibly originated in wild animals sold in the market. [3] [4] This pathogenic virus has been identified as a new strain of coronavirus, which is in the same family as severe acute respiratory syndrome coronavirus (SARS-CoV) and Middle East respiratory syndrome coronavirus (MERS-CoV). [4] [5] The COVID-19 outbreak was declared by World Health Organization as a Public Health Emergency of International Concern (PHEIC) on Jan 30, 2020. 6 As of Feb 16, 2020, a total of 70,548 cases have been reported in mainland China, with 1,770 resulting deaths. 7 The outbreak has now spread beyond China to twenty-five other countries. 8 The SARS-CoV and MERS-CoV were proven to have originated from wild animals. [9] [10] [11] [12] [13] [14] The COVID-19 is also likely to have a zoonotic origin. The practice of consuming wild animal products in China dates back to prehistoric times and persists into today. 9 Trade and consumption of wild animals were not well regulated even after SARS. Until the amendment of Wild Animal Protection Act in 2016, the prohibition was firstly added, but only for a short list of the nationally protected animals. Previous studies explored population exposure to live poultry in China, [15] [16] [17] but information on exposure to wild animals is limited.\\nIn the efforts to contain the spread of the virus, the Chinese government introduced a series of All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint containment measures. 18 Specifically, to lower the risk of zoonotic virus transmission, China has issued strict regulations on wet markets where live and/or wild animals are traded, including closing live poultry markets (LPMs) and bans on wild animal transactions. Furthermore, Wuhan city has been quarantined to stop human-to-human transmission: starting on Jan 23, 2020, public transportation in Wuhan was closed, and all flights and trains leaving from Wuhan were cancelled.\\nThe COVID-19 outbreak and subsequent containment measures to lower the zoonotic virus transmission offer a unique opportunity to reassess the exposure to live animals, demand for wild animal consumption, and public readiness to further regulate wet markets. In this study, we aim to investigate behaviour patterns of population interaction to live animals, with a particular focus on differences before and during the outbreak, between Wuhan and Shanghai representing diverse exposure intensities to COVID-19. We also seek to understand levels of public support for containment measures and public confidence in their effectiveness to quell the outbreak. This study will help to clarify the public's response to the outbreak and has important policy implications regarding the social acceptability of infection containment measures and their application in different contexts.\\nA population-based cross-sectional telephone survey was conducted in Wuhan and Shanghai between Feb 1 and 10, 2020, one month since the outbreak. These two cities were selected to represent diverse exposure intensities to COVID-19. Wuhan is the site of origin of the COVID-19 outbreak. Shanghai was selected due to its status as a city that has been significantly affected by imported COVID-19 cases All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint 8 from Wuhan. The first imported case was confirmed in Shanghai on Jan 20. During the period of our survey, the number of confirmed cases rose rapidly in Wuhan, from 4,109 to 18,454, but maintained low epidemic rates in Shanghai, with reported cases rising from 182 to 303. Figure 1 illustrates the timeline of the outbreak progression and containment measures. This provides an opportunity to assess and compare public behavioural response and support for containment measures during the outbreak between two cities with different exposure intensities.\\nThe sample size (n=500 for each city) is sufficient to estimate the change of outcomes (p=0.5) with a margin of error of 0.05 and a 95% confidence interval, according to previous studies. [15] [16] Our target population was persons aged 18 or over who had been living in the selected city for at least 6 months in the year prior to the survey and were current residents at the time of the survey. In each city, the telephone survey was conducted through a computer-assisted interviewing system, which enabled the random generation of mobile phone numbers and systematic data collection. We attempted to contact each generated number three times a day at different hours; if no contact was established after the third call, this number was classified as invalid or unreachable. Random dials with proportional quota sampling were used to ensure that respondents were demographically representative of the general population in each city, with quotas based on sex and age. 19\\nThe outcomes of interest we studied were exposure to live animals before and during the COVID-19 outbreak, public support for containment measures that aim to reduce population exposure to live animals and human transmission, and confidence in these measures to quell the outbreak. Exposure to All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint live animals includes the following behaviours: visiting the live poultry or wild animal markets, buying live poultry or wild animals (i.e., frequency of purchases, practice of picking up live animals before purchasing, location where the purchased animals were slaughtered), eating wild animal products, raising pets, and touching stray animals. To measure behavioural change during the outbreak, each respondent was asked to report exposing behaviours both (1) prior to the outbreak: before Dec 31, 2019 when COVID-19 cases were first released by Wuhan Health Commission and (2) since the outbreak, between Jan 1 to Feb 10, 2020. For live animal consumption habits, where behaviour change cannot be measured in a short period of time, we assessed people's intention to: 1) buy live poultry or wild animals, and eat wild animal products (continue to buy/eat as usual, buy/eat less, not buy/eat any more) and 2) continue raising pets and touching stray animals (continue to raise/touch as usual, temporarily not during the outbreak, try or resolutely not even after the outbreak).\\nWe measured public support for two types of governmental containment measures in response to the outbreak; one is regulation on wet markets to reduce population exposure to live animals, and the other is quarantining measures meant to reduce human transmission, such as the closure of public transportation in Wuhan and the cancellation of flights and trains leaving from Wuhan. Each respondent was asked to rate the extent, on a five-point Likert scale, to which they supported 1) the permanent closure of LPMs, 2) a ban on the wild animal trade, and 3) the quarantining of Wuhan; respondents were also asked to report the extent of their confidence in the effectiveness of these measures to quell the outbreak.\\nCharacteristics of respondents, including gender, age, marital status, education, income, and All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint employment status, were collected in our survey. We also collected the history of fever or cough symptoms for respondents and their relatives or friends in the two weeks prior to the survey, and whether there were confirmed or suspected COVID-19 cases in the neighbourhood in which they live.\\nEach respondent was also asked at what point they started paying attention to this outbreak.\\nThe survey questionnaire was adjusted according to an instrument used during the outbreaks of SARS in 2003 20-21 and influenza A(H7N9) in 2013. 15 The survey instrument was pretested for face and content validity, length, and understandability by the public before our survey was administered.\\nDescriptive and univariate analysis were used to investigate changes in exposure to live animals during the outbreak, public support for containment measures, and confidence in containment measures to quell the outbreak, between Wuhan (the epicentre) and Shanghai (an affected city with imported cases).\\nChi-square tests or fisher's exact tests (when applicable) were used to compare differences in these measures between the two cities, and before and during the outbreak. Multivariate logistic regression was used to estimate factors associated with public support for containment measures and confidence in intervention effectiveness. The proportion and Odds Ratio with 95% confidence intervals (CI) were reported. All statistical analyses were performed using Stata 14.0 (StataCorp LP, College Station, TX).\\nA total of 18,576 telephone numbers were randomly selected via digital dialing system in Wuhan and Shanghai. After calling, 6,836 respondents were eligible to participate, of whom 5,640 declined and All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint 11 185 terminated before completing the interview. In total, 510 respondents in Wuhan and 501 respondents in Shanghai completed the interview, and the overall response rate was 14.8%\\n(1,011/6,836) (Appendix Figure 1 ).\\nRespondents in Wuhan and Shanghai were generally similar in terms of sex, marital status and occupation, but in Shanghai respondents tended to be older and had higher education and income levels than in Wuhan (Table 1 ). Our sample was consistent with characteristics of the general population in the two cities, as Shanghai is more developed and has an older population than Wuhan. Expectedly, respondents in Wuhan were much more intensely exposed to the COVID-19 outbreak and paid attention to the outbreak earlier than those in Shanghai.\\nWe first investigated patterns of usual exposure to live animals before the outbreak ( Table 2 Panel of 55, 58.2% in Shanghai, p=0.339); more than 90% of respondents always arranged for slaughter of purchased animals in wet markets, with no noticeable differences between the two cities. However, residents in Wuhan reported wild animal consumption at a higher rate than those in Shanghai; 47 (9.2%) of 510 respondents in Wuhan and 23 (4.6%) of 501 respondents in Shanghai reported eating All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is To investigate effects of the outbreak, we further assessed behavioural changes in exposure to live animals before and during the outbreak (Figure 2) , and intention to change after the outbreak ( Table 2 Panel B). We found that respondents dramatically reduced exposure to live animals during the outbreak, with respondents in Wuhan changing their exposure much more significantly than in Shanghai. Specifically, the frequency of visiting LPMs decreased from 23.3% (n=119) to 3.1% (n=16)\\nin Wuhan (p<0·001), and from 20.4% (n=102) to 4.4%(n=22) in Shanghai (p<0·001; Figure 2 ). The proportion of respondents buying live poultry or wild animals also decreased from around 11.0% to 2.6% in both cities (p<0·001). We also asked whether respondents intended to change their habit of buying live poultry or wild animals since the outbreak among those exposing to live animals at usual ( (4/55) of respondents in Wuhan and 25.5% (14/55) in Shanghai stated they would continue to buy these products as usual.\\nThe proportion of respondents consuming wild animals declined significantly, 10.2%(n=52) to 0.6%(n=3) in Wuhan (p<0·001) and 5.2%(n=26) to 0.8%(n=4) in Shanghai (p<0·001) (Figure 2 ). Most All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is Wuhan, respectively, and 75% of respondents were confident that governmental containment measures would quell the outbreak. Table 3 presents socio-demographic factors associated with public support for the governmental containment measures implemented (i.e. closure of LPMs, banning the wild animal trade, and city quarantine) and confidence in the effectiveness of governmental response. There was evidence that females were more supportive towards all three containment measures and that there was a positive association between education level and the likelihood of supporting regulations on wet markets and trade of wild animals. Unsurprisingly, those who self-reported visiting a wet market and eating wild All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint animal products prior to the outbreak are much less likely to support closure of the market or a ban on the trade as containment measures (adjusted OR=0.58, 95%CI [0.41-0.82] and adjusted OR=0.31, 95%CI [0.14-0.70], respectively). We did not find any difference in population profiles of those who support Wuhan quarantine or show confidence in the effectiveness of governmental response.\\nThis study is the first to investigate behavioural changes of exposure to live animals during the COVID-19 outbreak, and public support for governmental containment measures in response to the outbreak. We found that population exposures to live animals were similar in Wuhan and Shanghai before the outbreak, except that wild animal consumption was more prevalent in Wuhan than in Shanghai. During the outbreak, residents in both cities significantly changed their behaviours and reduced exposures to live animals, and such changes were much more significant in Wuhan. Most respondents supported governmental containment measures implemented, and three in four had confidence on the effectiveness of these measures. Male, the less educated and those exposing to live animals at usual were less likely to support containment measures.\\nIn our survey, more than 20% of respondents visited wet markets at usual, which was similar in Wuhan and Shanghai, despite Shanghai being much more developed. Compared with the 30% level reported in 2013 in the same study sites, 15 the share of respondents visiting wet markets declined by 10 percentage points, but was still high. 10% of respondents consumed products of wild animals in Wuhan, twice that reported in Shanghai. In China, there exists a long tradition of eating the meat and products of wild animals, which persists into modern times due to a false but commonly held belief that these products All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint have therapeutic effects. [22] [23] These beliefs and preferences for fresh food are especially popular in southern China, where Wuhan is located, but were greatly challenged and changed during the 2003 SARS outbreak in China 9,12 . However, these practices have become popular again in recent years, due in part to the popularity of social media influencers who broadcast themselves eating wild animal products on livestream platforms. 22 The revival of this tradition may have set the socio-biological context for the occurrence and spread of COVID-19 in China.\\nWe found that respondents' risks of exposure to live animals declined significantly, reaching a very low level, and most respondents stopped wild animal consumption after this outbreak, especially in Wuhan.\\nThis indicates that the public responded quickly and adequately to the outbreak. In the survey, all respondents cared about the outbreak, and about one in five respondents in Wuhan reported starting to pay attention before Dec 31, 2019. The outbreak gradually gained intense nationwide attention, 24 and a series of policies were implemented by the Chinese government to contain the outbreak. As of Jan 30, all provinces had initiated first-level responses to major public health emergencies. [25] [26] These actions raised public attention further and induced significant behavioural changes. A national online survey on Jan 26 showed that the vast majority of people knew of the disease and related preventive measures, 66.5% reported that they would wear masks when outside their home, and 80% cancelled travel plans during the Spring Festival due to the outbreak. 27 The findings from both this survey and ours demonstrate behavioural change in terms of exposure to live animals or daily protection.\\nIn addition, more strict regulations were implemented on wet markets to lower the risk of disease transmission. Wuhan instituted prohibited the sale of live poultry and wild animals starting on Jan 22. All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint Shanghai started its seasonal closure of LPMs on Jan 25 this year. The Chinese Government further instituted nationwide bans on all forms of wild animal transactions on Jan 26. 28 These regulations may accelerate the decrease in exposing behaviours to live animals. However, the COVID-19 epidemic occurred prior to the Chinese lunar new year, when the demand for live animals is generally high. 23 This may produce a conflict between the public demand for live animals and regulations. Fortunately, we found that the public strongly supports the regulations on wet markets to control the outbreak.\\nAlthough permanent closure of LPMs was not fully backed by the public, it was still much higher than reported during the 2013 influenza A outbreak (27.3% in Wuhan, 39.3% in Shanghai). 15, 29 The strong support for this policy offers a great opportunity to improve regulations on wet markets and change the prevalent habit of eating wild animal products in China.\\nWe believe that the keys to quelling outbreaks of zoonotic viral diseases such as COVID-19 and preventing future outbreaks are changing the false beliefs and dangerous habits regarding eating wild animal products and strengthening regulations on wet markets. Firstly, healthy diets need to be vigorously promoted in China. Health China 2030 promotes healthy diets, but does not discuss the risks of eating wild animals. We also believe the internet and social media can be an effective channel to promote healthy eating habits and reduce the inappropriate and outdated tradition of eating wild animal products. 25, 30 Since the outbreak, spontaneous initiatives have emerged on the internet to explain the risks of and resist consuming wild animals. Secondly, live animal transactions should be more strictly regulated, and wild animal transactions should be even prohibited in the long term. While different animals should be sold separately in specific wet markets to lower risks of virus transmission, in practice, various kinds of live animals are sold together within one market, as found in Huanan Seafood All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint Wholesale Market. Laws should be strengthened to regulate and implement the strict separation of live animals from general markets. With the currently high public support, bans on the wild animal trade should be promoted, but the government must be aware of the risk of creating an unregulated black market by instituting a ban. For live poultry and non-wild animals, better regulations may be a better option than bans without the full support of the public. 23 The feasible measures include seasonal closure of wet markets, regular rest days, and banning leaving live animals in markets overnight. 10, 15 The above measures should be given priority in China and other countries to reduce current levels of COVID-19, as well as future threats from zoonotic diseases.\\nOur study has several limitations. First, there may be recall bias. We aimed to identify behavioral changes in a cross-sectional survey, and outcomes before the outbreak were measured based on participant recall. Second, our results may be affected by selection bias from telephone survey. We did attempt multiple calls to unanswered numbers to mitigate this bias, and quota sampling was introduced to get a representative sample. The anonymous phone interview addressed the concern of response bias due to fear of being seen as criticizing the government. Third, a cross-sectional design prevents us from assessing how long the effect on behavioral/intention change demonstrated in this study will last, when the outbreak is no longer fresh in people's minds. Thus, we plan to conduct a second-round survey at the end of the outbreak to track the long-term changes.\\nIn conclusion, exposure to live animals was common before the COVID-19 outbreak. The public responded quickly to this outbreak, and significantly reduced their exposure to live animals and stopped wild animal consumption, especially in Wuhan. They were very supportive and confident for All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint current mitigating measures. With high public support, wet markets should be better regulated, and healthy diets, including changing the habit of wild animal consumption, should be promoted, especially for male, the less educated and those exposing to live animals at usual. \\nThe study was approved by the Institutional Review Board of the School of Public Health, Fudan University (IRB#2020-TYSQ-01-1). Verbal informed consent was obtained from all participants.\\nH.Y. has received research funding from Sanofi Pasteur, GlaxoSmithKline, Yichang HEC Changjiang Pharmaceutical Company and Shanghai Roche Pharmaceutical Company. None of that research funding is related to 2019-nCov. All data are n (%). P value are from Chi square test or fisher's exact tests (when applicable).\\n*Fever or cough symptoms in the past 2 weeks.\\nAll rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint Resolutely not after outbreak 6(14.0) 3(5.2) All rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity. All rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity. Odds ratios with 95% CI were obtained from multivariate binary logistic regressions. Significance level: ** p<0.01, * p<0.05.\\nAll rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02.21.20026146 doi: medRxiv preprint WHO stands for World Health Organization; PHEIC stands for public health emergency of international concern.\\nAll rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint Notes: Proportions are shown, and error bars represent 95% CIs.\\nAll rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint\\nNotes: p value is shown between Wuhan and Shanghai.\\nAll rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.02. 21.20026146 doi: medRxiv preprint \\n  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>178</td>\n",
       "      <td>Python nidoviruses, more than respiratory pathogens 2 3 4 Running title: Systemic nidovirus infection Institute of Veterinary Pathology</td>\n",
       "      <td>[{'text': 'In recent years nidoviruses have emerged as an important respiratory pathogen of reptiles, 29 affecting especially captive python populations. In pythons, nidovirus infection induces an 30 inflammation of the upper respiratory and alimentary tract which can develop into a severe and 31 often fatal proliferative pneumonia. We observed pyogranulomatous and fibrinonecrotic lesions 32 in organ systems other than the respiratory tract during full post mortem examinations on 30 33 nidovirus RT-PCR positive pythons of varying species originating from Switzerland and Spain.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': '34', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'The observations prompted us to study whether the atypical tissue tropism associates with 35 previously unknown nidoviruses or changes in the nidovirus genome. RT-PCR and inoculation 36 of Morelia viridis cell cultures served to recruit the cases and to obtain virus isolates. 37 Immunohistochemistry and immunofluorescence staining against nidovirus nucleoprotein 38 demonstrated that the virus not only infects a broad spectrum of epithelial (respiratory and 39 alimentary epithelium, hepatocytes, renal tubules, pancreatic ducts etc.), but also intravascular 40 monocytes, intralesional macrophages and endothelial cells. By next-generation sequencing we 41 obtained full length genome for a novel nidovirus species circulating in Switzerland. Analysis of 42 viral genomes recovered from pythons showing nidovirus infection-associated respiratory or 43 systemic disease did not explain the observed phenotypes. The results indicate that python 44 nidoviruses have a broad cell and tissue tropism, further suggesting that the course of infection 45 could vary and involve lesions in a broad spectrum of tissues and organ systems as a 46 consequence of monocyte-mediated systemic spread of the virus. 47 48 author/funder. All rights reserved. No reuse allowed without permission. IMPORTANCE 49 During the last years, python nidoviruses have become a primary cause of fatal disease in 50 pythons. Nidoviruses represent a threat to captive snake collections, as they spread rapidly and 51 can be associated with high morbidity and mortality. Our study indicates that, different from 52 previously evidence, the viruses do not only affect the respiratory tract, but can spread in the 53 entire body with blood monocytes, have a broad spectrum of target cells, and can induce a 54 variety of lesions. Nidovirales is an order of animal and human viruses that compromise 55 important zoonotic pathogens such as MERS-CoV and SARS-CoV, as well as the recently 56 emerged SARS-CoV-2. Python nidoviruses belong to the same subfamily as the mentioned 57 human viruses and show similar characteristics (rapid spread, respiratory and gastrointestinal 58 tropism, etc.). The present study confirms the relevance of natural animal diseases to better 59 understand the complexity of viruses of the order nidovirales.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': '60 61 62 author/funder. All rights reserved. No reuse allowed without permission.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>INTRODUCTION 63 respiratory epithelium. These findings and the fact that nidovirus infections affect several python 88 species led us to the hypothesis that python nidoviruses are not restricted to a certain species and 89 have a broad disease potential. Therefore, we undertook a larger study, making use of diagnostic (Table 1 ) as described (6). 168 Library preparation, data analysis, and genome assembly was done as described (6). 169 To study the frequencies of single nucleotide polymorphisms (SNPs) within samples, the NGS 170 reads were quality filtered using Trimmomatic and the reads with Q-score over 30 were 171 assembled against the consensus sequence of a given sample using the BWA-MEM algorithm 172 (15) followed by removal of potential PCR duplicates using SAMTools version 1.8 (16). The The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.04.10.036640 doi: bioRxiv preprint\\nGross presentation of nidovirus-associated disease in pythons. 197 Full post mortem examinations were performed on all cases. Most snakes that had died 198 spontaneously exhibited the typical, previously described respiratory changes, represented by a 199 varying amount of mucoid material in the airways and particularly in the lungs (Fig. 1A) . In 200 some animals, the oral cavity, the trachea, the lung and the air sacs were filled with mucoid 201 material and the lung parenchyma appeared thickened (nidovirus-associated proliferative 202 disease; Fig.1A) ; also, the mucoid material was occasionally mixed with purulent exudate. Two 203 euthanized carpet pythons with clinically reported respiratory distress (CH-B1 and -B2, Table 1) 204 did not exhibit any gross pulmonary changes, but showed mild mucus accumulation in the oral 205 cavity. In others, granulomatous and/or fibrinonecrotic lesions were observed in the oral, Table 1 . Oral cavity and respiratory tract. The respiratory tract was found to be affected in all animals and degenerated oral and nasal epithelial cells (Fig. 1D ). In two euthanized carpet pythons (CH-221 B1 and -B2, Table 1 ), these were the only pathological changes, suggesting that they represented 222 early (initial) lesions. The latter is further supported by the fact that in one of the two individuals disease. In all snakes with the above-described intestinal changes (CH-A7 and -A8, CH-B6, CH-244 F1), the mucosal lesions were found to overlie focal inflammatory processes in the intestinal 245 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.04.10.036640 doi: bioRxiv preprint wall. These were represented by multifocal (pyo)granulomatous perivascular infiltrates of 246 macrophages and fewer heterophils and lymphocytes which were predominantly seen in the 247 submucosa in close proximity to the gut associated lymphatic tissue (GALT) but occasionally 248 extended into the tunica muscularis (Fig. 2C, E) . These lesions contained abundant nidovirus NP The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.04.10.036640 doi: bioRxiv preprint to the inflammatory foci (e.g. pancreatic ducts, renal tubules etc.) without any evidence of (Fig. 3D ). In two cases (CH-B6 and -F1) ependymal cells were also 283 found to be infected, without evidence of correlating brain lesions (Fig 5D) . Table 1 ) and three positive liver specimens (CH-A7, -A8 and CH-B6, viruses had less than 5 % nucleotide and less than 2.4 % amino acid differences with recently 299 described Morelia viridis nidovirus isolates BH128 14-12 and BH171 14-7 and Serpentovirinae 300 sp isolates L3, L4 and H0-1, but more than 13 % nucleotide and 8.5 % amino acid differences to 301 all other nidovirus strains. Phylogenetic analysis based on ORF1b suggested that these virus 302 forms a sister clade to the previously known python-associated nidovirus species; BPNV and 303 MVNV (Fig 6) . CDSs of the same virus species were obtained from 11 cell culture supernatants, 304 including the isolation attempts from the liver samples included in the NGS analysis. One of the 305 cell culture supernatants analyzed by NGS yielded CDS of MVNV (animal E-C1, Table 1 ), and 306 for four cell culture supernatants NGS failed due to severe bacterial contamination.\\nFor three snakes (CH-A7-8, CH-C3, Table 1 ) the analysis included cell culture isolated virus 308 (lung homogenate as inoculum) and virus sequenced directly from tissue (liver). The consensus 309 sequences obtained from liver tissue and the respective cell culture isolate were identical for 310 each of these three snakes indicating that isolation did not induce a marked bias in the sequence The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.04.10.036640 doi: bioRxiv preprint\\nIn addition to the above snakes, we studied choanal and/or cloacal swabs from seven additional 369 animals from different collections for the presence of nidovirus RNA by RT-PCR. All of the 370 tested animals were found positive for nidovirus infection (Table 1) Table 1 ), and a necrotzing splenitis (CH-A1 and CH-C4, Table 1 ) and fibrinonecrotic enteritis 380 with intralesional coccoid bacterial colonies (E-B1, Table 1 ). Both conditions are likely a 381 consequence of debilation/dehydration due to chronic pulmonary disease. 382 383 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.04.10.036640 doi: bioRxiv preprint DISCUSSION 384 Nidoviruses have recently been described as the cause of respiratory tract disease in several 385 python species in the USA and Europe (5, 6, 10, 11), representing an emerging threat to python 386 traders and breeders in particular. The present study aimed to further elucidate the characteristics 387 and relevance of python nidoviruses. Particular emphasis was laid on the potential species 388 specificity of and/or susceptibility to the viruses, their phylogeny and geographic divergence as 389 well as viral shedding and transmission, host cell tropism and type of associated disease. 390 Experimental infection has been shown to reproduce the naturally occurring respiratory disease 391 (7), so our study focused on natural cases from different geographic regions, from breeding 392 colonies that often kept several python species, and from colonies that at least partly worked 393 closely together, exchanging animals. In our study, no differences were noted between the python species in the degree and distribution The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.04.10.036640 doi: bioRxiv preprint suggest increased susceptibility of snakes from this genus to nidovirus infection and/or disease, a 409 finding also supported by recent epidemiologic studies (14). In support, the analysis of the viral 410 genomes by NGS did not show differences in the viral genome that would explain the 411 differences in the disease manifestation. One could speculate that each python species has its 412 own nidovirus, and that cross-species transmission of the viruses results in morbidity. Should The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.04.10.036640 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.04.10.036640 doi: bioRxiv preprint Table 1 . Animals, clinical signs, lesions and viral target cells. All animals were from breeding collections in Switzerland (CH) or Spain (E). Breeders CH A-C and CH-F were working closely together, trading/exchanging snakes.\\nMorelia \\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                       title  \\\n",
       "178  Python nidoviruses, more than respiratory pathogens 2 3 4 Running title: Systemic nidovirus infection Institute of Veterinary Pathology   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         abstract  \\\n",
       "178  [{'text': 'In recent years nidoviruses have emerged as an important respiratory pathogen of reptiles, 29 affecting especially captive python populations. In pythons, nidovirus infection induces an 30 inflammation of the upper respiratory and alimentary tract which can develop into a severe and 31 often fatal proliferative pneumonia. We observed pyogranulomatous and fibrinonecrotic lesions 32 in organ systems other than the respiratory tract during full post mortem examinations on 30 33 nidovirus RT-PCR positive pythons of varying species originating from Switzerland and Spain.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': '34', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'The observations prompted us to study whether the atypical tissue tropism associates with 35 previously unknown nidoviruses or changes in the nidovirus genome. RT-PCR and inoculation 36 of Morelia viridis cell cultures served to recruit the cases and to obtain virus isolates. 37 Immunohistochemistry and immunofluorescence staining against nidovirus nucleoprotein 38 demonstrated that the virus not only infects a broad spectrum of epithelial (respiratory and 39 alimentary epithelium, hepatocytes, renal tubules, pancreatic ducts etc.), but also intravascular 40 monocytes, intralesional macrophages and endothelial cells. By next-generation sequencing we 41 obtained full length genome for a novel nidovirus species circulating in Switzerland. Analysis of 42 viral genomes recovered from pythons showing nidovirus infection-associated respiratory or 43 systemic disease did not explain the observed phenotypes. The results indicate that python 44 nidoviruses have a broad cell and tissue tropism, further suggesting that the course of infection 45 could vary and involve lesions in a broad spectrum of tissues and organ systems as a 46 consequence of monocyte-mediated systemic spread of the virus. 47 48 author/funder. All rights reserved. No reuse allowed without permission. IMPORTANCE 49 During the last years, python nidoviruses have become a primary cause of fatal disease in 50 pythons. Nidoviruses represent a threat to captive snake collections, as they spread rapidly and 51 can be associated with high morbidity and mortality. Our study indicates that, different from 52 previously evidence, the viruses do not only affect the respiratory tract, but can spread in the 53 entire body with blood monocytes, have a broad spectrum of target cells, and can induce a 54 variety of lesions. Nidovirales is an order of animal and human viruses that compromise 55 important zoonotic pathogens such as MERS-CoV and SARS-CoV, as well as the recently 56 emerged SARS-CoV-2. Python nidoviruses belong to the same subfamily as the mentioned 57 human viruses and show similar characteristics (rapid spread, respiratory and gastrointestinal 58 tropism, etc.). The present study confirms the relevance of natural animal diseases to better 59 understand the complexity of viruses of the order nidovirales.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': '60 61 62 author/funder. All rights reserved. No reuse allowed without permission.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          full_text  \n",
       "178  INTRODUCTION 63 respiratory epithelium. These findings and the fact that nidovirus infections affect several python 88 species led us to the hypothesis that python nidoviruses are not restricted to a certain species and 89 have a broad disease potential. Therefore, we undertook a larger study, making use of diagnostic (Table 1 ) as described (6). 168 Library preparation, data analysis, and genome assembly was done as described (6). 169 To study the frequencies of single nucleotide polymorphisms (SNPs) within samples, the NGS 170 reads were quality filtered using Trimmomatic and the reads with Q-score over 30 were 171 assembled against the consensus sequence of a given sample using the BWA-MEM algorithm 172 (15) followed by removal of potential PCR duplicates using SAMTools version 1.8 (16). The The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.04.10.036640 doi: bioRxiv preprint\\nGross presentation of nidovirus-associated disease in pythons. 197 Full post mortem examinations were performed on all cases. Most snakes that had died 198 spontaneously exhibited the typical, previously described respiratory changes, represented by a 199 varying amount of mucoid material in the airways and particularly in the lungs (Fig. 1A) . In 200 some animals, the oral cavity, the trachea, the lung and the air sacs were filled with mucoid 201 material and the lung parenchyma appeared thickened (nidovirus-associated proliferative 202 disease; Fig.1A) ; also, the mucoid material was occasionally mixed with purulent exudate. Two 203 euthanized carpet pythons with clinically reported respiratory distress (CH-B1 and -B2, Table 1) 204 did not exhibit any gross pulmonary changes, but showed mild mucus accumulation in the oral 205 cavity. In others, granulomatous and/or fibrinonecrotic lesions were observed in the oral, Table 1 . Oral cavity and respiratory tract. The respiratory tract was found to be affected in all animals and degenerated oral and nasal epithelial cells (Fig. 1D ). In two euthanized carpet pythons (CH-221 B1 and -B2, Table 1 ), these were the only pathological changes, suggesting that they represented 222 early (initial) lesions. The latter is further supported by the fact that in one of the two individuals disease. In all snakes with the above-described intestinal changes (CH-A7 and -A8, CH-B6, CH-244 F1), the mucosal lesions were found to overlie focal inflammatory processes in the intestinal 245 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.04.10.036640 doi: bioRxiv preprint wall. These were represented by multifocal (pyo)granulomatous perivascular infiltrates of 246 macrophages and fewer heterophils and lymphocytes which were predominantly seen in the 247 submucosa in close proximity to the gut associated lymphatic tissue (GALT) but occasionally 248 extended into the tunica muscularis (Fig. 2C, E) . These lesions contained abundant nidovirus NP The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.04.10.036640 doi: bioRxiv preprint to the inflammatory foci (e.g. pancreatic ducts, renal tubules etc.) without any evidence of (Fig. 3D ). In two cases (CH-B6 and -F1) ependymal cells were also 283 found to be infected, without evidence of correlating brain lesions (Fig 5D) . Table 1 ) and three positive liver specimens (CH-A7, -A8 and CH-B6, viruses had less than 5 % nucleotide and less than 2.4 % amino acid differences with recently 299 described Morelia viridis nidovirus isolates BH128 14-12 and BH171 14-7 and Serpentovirinae 300 sp isolates L3, L4 and H0-1, but more than 13 % nucleotide and 8.5 % amino acid differences to 301 all other nidovirus strains. Phylogenetic analysis based on ORF1b suggested that these virus 302 forms a sister clade to the previously known python-associated nidovirus species; BPNV and 303 MVNV (Fig 6) . CDSs of the same virus species were obtained from 11 cell culture supernatants, 304 including the isolation attempts from the liver samples included in the NGS analysis. One of the 305 cell culture supernatants analyzed by NGS yielded CDS of MVNV (animal E-C1, Table 1 ), and 306 for four cell culture supernatants NGS failed due to severe bacterial contamination.\\nFor three snakes (CH-A7-8, CH-C3, Table 1 ) the analysis included cell culture isolated virus 308 (lung homogenate as inoculum) and virus sequenced directly from tissue (liver). The consensus 309 sequences obtained from liver tissue and the respective cell culture isolate were identical for 310 each of these three snakes indicating that isolation did not induce a marked bias in the sequence The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.04.10.036640 doi: bioRxiv preprint\\nIn addition to the above snakes, we studied choanal and/or cloacal swabs from seven additional 369 animals from different collections for the presence of nidovirus RNA by RT-PCR. All of the 370 tested animals were found positive for nidovirus infection (Table 1) Table 1 ), and a necrotzing splenitis (CH-A1 and CH-C4, Table 1 ) and fibrinonecrotic enteritis 380 with intralesional coccoid bacterial colonies (E-B1, Table 1 ). Both conditions are likely a 381 consequence of debilation/dehydration due to chronic pulmonary disease. 382 383 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.04.10.036640 doi: bioRxiv preprint DISCUSSION 384 Nidoviruses have recently been described as the cause of respiratory tract disease in several 385 python species in the USA and Europe (5, 6, 10, 11), representing an emerging threat to python 386 traders and breeders in particular. The present study aimed to further elucidate the characteristics 387 and relevance of python nidoviruses. Particular emphasis was laid on the potential species 388 specificity of and/or susceptibility to the viruses, their phylogeny and geographic divergence as 389 well as viral shedding and transmission, host cell tropism and type of associated disease. 390 Experimental infection has been shown to reproduce the naturally occurring respiratory disease 391 (7), so our study focused on natural cases from different geographic regions, from breeding 392 colonies that often kept several python species, and from colonies that at least partly worked 393 closely together, exchanging animals. In our study, no differences were noted between the python species in the degree and distribution The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.04.10.036640 doi: bioRxiv preprint suggest increased susceptibility of snakes from this genus to nidovirus infection and/or disease, a 409 finding also supported by recent epidemiologic studies (14). In support, the analysis of the viral 410 genomes by NGS did not show differences in the viral genome that would explain the 411 differences in the disease manifestation. One could speculate that each python species has its 412 own nidovirus, and that cross-species transmission of the viruses results in morbidity. Should The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.04.10.036640 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.04.10.036640 doi: bioRxiv preprint Table 1 . Animals, clinical signs, lesions and viral target cells. All animals were from breeding collections in Switzerland (CH) or Spain (E). Breeders CH A-C and CH-F were working closely together, trading/exchanging snakes.\\nMorelia \\n  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#loop on the top 5 indicies and display the title, abstract, and fulltext of the paper that was fetched from the search\n",
    "if (flag == True):\n",
    "    print(\"Sorry, no results\")\n",
    "else:\n",
    "    counter =0\n",
    "    for index in answer_indicies:\n",
    "        if (cosineSimilarities[index]>=0.1): #get papers with cosine similarities greater than or equal 0.1\n",
    "            display(FullPaperDataframe.iloc[[index]])\n",
    "        else:\n",
    "            counter = counter+1\n",
    "    if (counter == len(cosineSimilarities)):\n",
    "        print(\"Sorry, no results\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h4>STEP 4: LITERATURE CLUSTERING</h4>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<p>The second part of this notebook attempts to before K-Means clustering on the papers, such that each group of papers related together via certain keywords are grouped together.</p> \n",
    "<h6 style = \"color:green\"> PROS of using K-means</h6>\n",
    "<p> The pros of using K-means is that many packages provide its functionality and it is easy to implement.</p>\n",
    "<h6 style = \"color:red\"> CONS of using K-means</h6>\n",
    "<p> When clustering a new paper, the distance between that paper and every other paper should be computed, which is time consuming. Besides, the number of clusters which are to be chosen from the optimum cluster number, must be chosen.</p>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h5>Dimensionality Reduction</h5>\n",
    "<p>At this stage, we have lots of features/words in the TIF-IDF matrix. However, to allow the K-means to perform better, dimensionality reduction is needed. Therefore, the PCA algorithm is chosen, as it is known that it performs well with K-Means </p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.decomposition import PCA #import PCA which will be used to reduce the dimensions of the dataset\n",
    "\n",
    "pca = PCA(n_components=0.90,random_state=30)#keep 90% of the variance, and start random centroid @ 30\n",
    "tf_idf_norm_kmeans= pca.fit_transform(tfidf.toarray())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h5>Elbow testing </h5>\n",
    "<p>In order to determine how many clusters are needed for the K-means. A graph is plotted, the point at which the elbow forms is then searched. That value should be used as the number of clusters. </p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 12/12 [00:59<00:00,  4.98s/it]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 1440x1440 with 0 Axes>"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAY4AAAEaCAYAAAAG87ApAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAgAElEQVR4nO3deZgUxfnA8W/NcsghCi6KgAoooIKKghyKiheiIorRNx7BMzFeP4/ERI3GW+MVj0S8Iijiga83oiiioijgAdGIongreAAit4Iw9fujemUYZ3Z7d2d2dmffz/PsszvVPV1vz/bMO91VXeW89xhjjDFxJQodgDHGmLrFEocxxphKscRhjDGmUixxGGOMqRRLHMYYYyrFEocxxphKscRhAHDObeece8M595Nz7vNCx1MIzrnjnHOrc7i9e5xzE3O1PZMbzrkBzjnvnGtf6FjqKkscOeaca+Kcu9w595Fz7kfn3PfOuTedc2cUOrYKXAssAbYGds62knPueOfcdOfcEufcUufcLOfcf2osygJzzk2KPnS8c+5n59w859zLzrk/OeeapK1+JnB4Jba92jl3XE4DLhDn3NnR8d8qy/KnnHOvRn8nnHPnOOdmOueWO+cWOefecc5dUcW6f+ece8U5tzja3kzn3LXOuXbV2acK6rywPn3hssSRe7cBxwB/AbYF9gKGAxsWMqgYOgMve+8/997Pz7RC9KF2O3A30AvoCZwHlNRUkOVxQcMaqOoBYFOgA7Av8AjwZ2CGc27jspW894u99z/UQDy10ajo97D0BdEH+P7AnVHRRcAFwNXA9sCuwFVAs8pW6pwbAYwAXonq2BY4A2hD+B/Ves65RoWOoULee/vJ4Q+wCDi9gnXuASamlf0u/Dt+eXwJ8DEgwEfACuAJoAVwKPAhsJTwobVBBfVtCoyJYvsRmAT0ipZ1AHzazyVZtvME8EiM10Ci2H8CpgBDou32j5YPiB63T3veauC4lMdXArOiff+KkLQ2SFl+XPScPYH/AquAwdGyfYHXov2dS0h2G6U81wGXA/OAZdHrczawuoJ9mwTclaG8PfADMDLb/xnoBjwX/R+WR/s2LFr2efr/ISpvCdwHfBnty4eED0CXXg9wEvAF4czxSaB1Woz7AJOj13Mx8DKwZcryI4C3o//b58ANQLOU5f2j13Rp9PMOsF85r9VoYGaG8r9Hr1WT6PHbwPU5eO/9JnrtjsiyvGWm468Sx+PfgE+BlcD86H/ZJDoOM76HgAaE9/Jn0ev6HvDHtHo8Ibk9EP1fHi6vvuq+Trn4KXgAxfYTfRiMA1qVs846HyhRWabEsRx4mvAtbI/o4JkAPAPsAOwGfAdcU05dDng9enP2B7YDHoreuKWEs4U2hA/mq6O/m2fZ1m3RB0qXcurbEUgC/wC6EpLcZ1QtcVwY7WMHYG/gA2BUyvLjorreJJzZdQJaR3+vAP6PcCa1M/AS4Vuoi557ZvT6Hgt0Af5K+ECvUuKIlt0SvfETmf7PwP+iD4dto1j3Z22iax3t/5nR/6BNVN4GOBfYCegYHSfLgOPTjqfFwINAd2AXQgJJfa32AdYAN0XHztbAicDWKa/lD4QzhE7A7lG8o6PlJcBCQjLpHP0MBXYr57XqH/2f+6WUJaJj6F8pZeOj/2G7ar73ngA+irHeOsdfnOORcBwvAQ4CNgd6AGcREkcTwnvnq7L/HdF7KPrf/A8YGP3/fhsdZyem1OOB7wnH65bR8Zi1vpr6LCv3NSx0AMX2QzjN/iJ6k/6PcDp+MBm+IaY9L1PiWA2UppQNj7bbOqXsZuCtcuLZOzowt00pawx8A1yUUvY5cGEF+9YGeDXa3ueEBHQS0DRlnfuAKWnPO50qJI4M9Q8lfPsq+2A+LtrObmnrTQKuTivbPFq3R/R4DnBl2jqPUL3EcXJUx8aZ/s+ED/fy9q/c/U/7nz+fdjzNBxqnlJ0HfJPyeDIwrpxtfg6cnFa2e7Q/LaMfDwyo5PvhPdY9C9sv2k73lLKtgZmELwEfEi5zHQ00qGRd7wNjY6y3zvEX53gknI3OBhpm2eaFwOdpZR2jfdo6rfwi4O2Uxx4YkbZOufUV+sfaOHLMe/8a4VvDboQ3wCbAo8BY55yr5Obmeu8XpDz+FvjWr9sG8S2wMdl1A7733r+fEuNKwllIt8oE473/1nvfn/CN+R+Eb+zXAu+lXNvflnA5I9WrlamnjHPu0KiR82vn3DLgfqARIYGlejPt8c7AWc65ZWU/hA8VgM7OuRZAO8JltGrHmRpy9NtnWX49cFfUwH6Jc26nCjcYGo7Pc8697ZxbEO3LycAWaavOiv6vZeYSjr0yPQlnq5nqaB1t74a012x8tMpWPrTV3AU855wbH8XUtaL4CV+cfhu95gB/AKZ672eWreC9/4BwJtyTcNbWKKprWoYOB+VxZH/tq0uBhsAXUW+5Yc659St4Tq8oprfSXte/Ec7YUr2Rg/pqjCWOPPDer/beT/He/9N7fzDhm/Fgwjc4CN9C0pNIpkbdn9M3naWsov9jpjdTld9k3vtZ3vs7vPcnEC5NtQdOqcR2kynrhj+cKyFlP5xzfYCHCZeXhhIu1ZwcLU5tPFzjvf8pbfsJ4BrC6X3qT2fCh2FFH/BV1Z1wGeL7TAu995cTLkNotO60GD2H/gycD/yb0G7Tg/Chmt6Auiq9On59jGXb37LX/UzWfb12ILxm70bx/4Hw4f484dLpTOfcHyuI/94ojqOdc5sQ2rvuTF/JB//13v/be38kYV97EtrL4vqQ8MWlsio8Hr33cwlnRicQ2sX+DnzonNusnO2WPX8X1n1duxMuP6danvqgivXVGEscNWNW9LvsW/k8oG3aOhV++6yi94BS59wvbyjnXGOgd7Ssuj4ntCeU7dt7hMt1qdIfz4t+p74GPVj3g64/sMB7f6H3/nXv/WxCgorjLaCb9/7jDD/LvPeLCd/IK4oztuiegKMJnQeS2dbz3n/qvb/Ve38Y4ZLFKSmLV/HrHmq7A89670dEH6wf8+tvq3FMJ1wmyhTTd4Tr812zvGY/paw703t/g/d+f0LvpZPKqzQ6U3mYcKZxHOFY0Rjxpr9n4rgP2Mo5d0Smhc65llmeF+d4xHu/0nv/rPf+r4QzpKbAIdHiTP+76dHvzTO8pp9UtDMV1FdQDQodQLFxzr1MaKR8i3DdeStC18JFhAZaCD1gznXOnU74BrwXlftmVRkvEk6DH3DOnUa4zv53YD1CY3dszrnbCJfGXiT08iklfEttQWiYBLgReNM5dyXhUl03ft0N8mNCO9Alzrmzo+1cxbrfiD8EWjvnTiS8bv2BU2OGehEwwTl3YxTDUsKH7eGEHm8/Av8ELnfOfQBMI3wT3ifm9ps459oQvniVEj7czycko/MzPcE515xwFvQoobPAhsAg1l5CIyrf0zk3HlgVXab8EBjmnNsz2v4xQB9CQ3ZlXA6Md87dBIwktBX1I1w2+pDQHXaEc24R4X/5M7ANsL/3/o/Oua0IH/5PEZJMW8Ll2Bkx6r6DcPmyA3Cf935F6kLn3KOEy4ZTgK8JlxEvjGJ4OlqnN+Hs5RjvffplHQC894845+4FRjnnuhE6kcwltDUcR3jN/pThqRUej9FxmCC8lxYR2g7XZ+3/7zOgjXOuH1EvSO/9x865kcB/nHN/BaYSuhj3JLRTXpPtBYtRX2EVupGl2H4IjZKTCd9ifiJ8wN5HSuN0tN4FhIN6GSHRnEaG7rhpz8nUAHceMKeCmNK7475M1B03ZZ3Pqbhx/FBgLKFheSWhR9fzhA+X1PWOAD6J1nmd0Dngl8bxaJ0+hG9kPxK6de7Gr3tVXR7VsZzwIXBktJ0O0fLjyNKYHW1vIiFplHV9vYmowZXwprwKWBAtf4T43XE9ay8bLohezz+R1uOFlMZxQqJ+gLXdMucROhdslrL+oCjOlWXHArAB4Rv6EsIlsOHR6/J5pnpSytbpbBGV7Uf48PqR8AXiJaBTyvJDouUrovreJupAER1Dj6X8778G/kMFXcFTtv1u9Jptn2HZH6Lj6Jto23MJySu1N9YAYjbOE3rKvRrtw3JCw/vVwKZp22qf8pxyj0fCsT+FkHxWRNtM7RnVMPr/LmTd7rglhB57HxDOSsqOl8NTnuuB32V4r2Wtr9A/ZV0Tjckb51wHwgfmbt776jZAG2MKzNo4jDHGVIolDmOMMZVil6qMMcZUip1xGGOMqZT60h3XTquMMabyMo52UV8SB19//XWVnldaWsqCBQsqXjEHarKuYq+vmPetpusr5n2r6frq0r61bZt+j/JadqnKGGNMpVjiMMYYUymWOIwxxlSKJQ5jjDGVYonDGGNMpdSbXlWVlZw2Cf/4aL77YQG0LMUNHUai74BCh2WMMQVniSOD5LRJ+NHDYVU0odrC+fjRw0mCJQ9jTL1nl6oy8I+PXps0yqxaGcqNMaaes8SRycIsN8xkKzfGmHrEEkcmrUorV26MMfWIJY4M3NBh0KjxuoWNGodyY4yp56xxPINE3wEkido6Fs4PhQccZg3jxhiDJY6sEn0HQN8BtGqyHvP/MBS++KTQIRljTK1gl6oqkGjWHLf3YPjvNPzcLwsdjjHGFJwljhjc3gdB4/Xw4x8udCjGGFNwljhicM1b4PbYH//GZPy8qs3rYYwxxcISR0xu34OhpAT/7GOFDsUYYwrKEkdMbsNWuP774qe8iC/raWWMMfWQJY5KcIMOBTz+uccLHYoxxhSMJY5KcBttjOs7AD95An7JD4UOxxhjCsISRyW5QYfB6tX4CU8WOhRjjCkISxyV5Nq0w/XaFT9pPH750kKHY4wxNc4SRxW4Aw6HlT/iX3iq0KEYY0yNs8RRBa59B+jRB//COPxPKwodjjHG1ChLHFWUOEBgxTL8pPGFDsUYY2qUJY4qch07w7Y74ic8gU+fLdAYY4qYJY5qSBx4OCxdjJ/8fKFDMcaYGmOJoxpcl+6w1bb45x7Dr/650OEYY0yNsMRRTYkDBX5YgJ/6UqFDMcaYGlFjEzmJyEhgMDBPVbunLTsHuA5oraoLRMQBNwMHACuA41R1RrTuscCF0VOvUNVRNbUPGXXbEbbYCj/+Efwue+NKSgoajjHG5FusMw4R2VNEOkZ/byoio0RkpIi0qURd9wCDMmx7M2BfIHWWpP2BztHPScBt0bqtgIuBPkBv4GIRaVmJGHLOORfOOuZ/i39zciFDMcaYGhH3UtWtwJro738CDQEP3Bm3IlV9BViYYdGNwF+j7ZU5GLhXVb2qTgM2FJFNgf2A51V1oar+ADxPhmRU43boDW03xz/zMD6ZLHQ0xhiTV3EvVbVT1S9FpAHhw3sLYBVQrVmNRGQIMFdV3xGRdeoDvkp5PCcqy1aeadsnEc5WUFVKS0urFGODBg1iPffH357AkhsvYf1P3me9fgPyWleuFHN9xbxvNV1fMe9bTddXLPsWN3EsEZFNgO7A+6q6TEQaEc48qkREmgIXAAMzLHYZynw55b+iqney9ozIL1iwoCphUlpaSpzn+q47wMabsnjMCJZu1Q3nMoWam7pypZjrK+Z9q+n6innfarq+urRvbdu2zbos7qWqfwNvAvcDw6OyXYEPqhRRsCXQEXhHRD4H2gMzonaTOcBmKeu2J5zdZCsvOFdSghv0G/jyE3hvRqHDMcaYvIl1xqGq14jI48AaVf0kKp4L/L6qFavqu8DGZY+j5NEr6lU1FjhdRMYQGsIXq+o3IvIccFVKg/hA4PyqxpBrrt+e+HFjSD6tJLrtVKWzDmOMqe0qcx/HZ0BbEflt9Hgu8GncJ4vIg8BUoKuIzBGRE8tZ/Zlo2x8D/wFOBVDVhcDlhLOfN4HLorJawTVoiNvvUPh4Fsx+r9DhGGNMXsQ64xCR7YCxwErC5aGHgD2AY4HflvPUX6jqkRUs75DytwdOy7LeSGBknDoLwfXfF/+0knz6IUq6dq/4CcYYU8fEPeO4DbhIVbcGysbWeBnon5eo6jDXqDFu4CEw6x38px8WOhxjjMm5uImjG3Bf9LcHUNXlQJN8BFXXuT0GQdPmJJ95uNChGGNMzsVNHJ8DPVMLRKQ3oQ3CpHHrNcXtMwTeeQM/57NCh2OMMTkVN3H8HXhaRC4FGonI+cDDrB0zyqRxew2G9Zrgn3mk0KEYY0xOxUocqjqOMH5Ua0LbxhbAoao6IY+x1WmuWXPcgAPwb72K/3ZOocMxxpiciT06bjQ67al5jKXouH0Pxr/4FH78o7jjzyx0OMYYkxNxR8d9TER2SyvbTUTsOkw5XIsNcbvth5/2En7Bd4UOxxhjciJuG8cewJS0sqnAnrkNp/i4gUPBJfDPPVboUIwxJifiJo6fgGZpZc1Ze0+HycK1KsXtshf+1Yn4Rd8XOhxjjKm2uInjOeAOEWkBEP2+BXg2X4EVEzfoN7BmDX7CE4UOxRhjqi1u4vgz0AJYKCLzCBMybQCcla/AionbeFNcn93xLz+LX7qk0OEYY0y1xB0d9wfgwGjI882Ar1T127xGVmTc/ofhp03CTxyLG/q7QodjjDFVVpnRcQGSwPdAUxHpJCKd8hBTUXJtN4ed+uFfehq/YnmhwzHGmCqLOzruIGAEsGnaIg+U5DqoYpU4QEjOmIp/6WncgVLxE4wxphaKewPgcMI8GKNU9cc8xlPU3BZbQvee+IlP4vcZgmu8XqFDMsaYSot7qaolcIcljepLHCiwbCn+lecKHYoxxlRJ3MQxAjg+n4HUF26rbaDrdvjnHsf/vKrQ4RhjTKXFvVTVFzhDRM4D1ulNpaq75zyqIpc44HCSN16Ef+0F3ID9Cx2OMcZUStzEcVf0Y3Jhmx2gYxf8s4/i+++LaxB7rEljjCm4uPdxjMp3IPWJc47EgULylivwb7yM22XvQodkjDGxxf6qKyKbAL2BUsCVlavqyDzEVfy23xnad8SPfwTfdwAuYb2ajTF1Q9xh1Q8BPgEuA+4A/i/6PSx/oRU35xzugMPh27n46VMLHY4xxsQWt1fVFcDxqrojsDz6fRIwPW+R1QOuZz9o0w7/zMN47wsdjjHGxBI3cWyuqg+nlY0CjslxPPWKS5Tg9j8M5nwG/3ur0OEYY0wscRPHvKiNA+BzEekHbIkNN1JtrvcesNHGJJ9+yM46jDF1QtzE8R+gf/T3jcBLwDvAbfkIqj5xDRpA1+3gs9nMO3RX1px7IslpkwodljHGZBW3O+41KX/fKyKTgGaqOitfgdUXyWmT4K3JawsWzsePHk4SSPQdUKCojDEmu7i9qp5MfayqX6rqLBGxibSryT8+GlalDT2yamUoN8aYWijupao9s5QPyFEc9dfCBZUrN8aYAiv3UpWIXBb92Sjl7zKdgC/yElV90qoUFs7/dXmLDWo+FmOMiaGiM47Nop9Eyt+bAe2Br4DD8xpdPeCGDoNGjdNL4cfl+C8/LUhMxhhTnnLPOFT1eAARmaKq/6mZkOqXRN8BJInaOn5YAC1LYd+DYcITJG++hMS5V+M2blvoMI0x5hdxx6p6TUQ2UdXvRKQ58BdgDXC9qq6o6MkiMhIYDMxT1e5R2eXAwYR5zOcBx6nq1yIyAHgS+Cx6+mOqeln0nEHAzYT7R+5S1atjxl+rJfoOgL4DKC0tZcGC0Lbhu+1I8trzSN5wEYnzrsFtuFFhgzTGmEjcxvEHgA2jv68Hdgf6EcariuMeYFBa2XWqur2q9gDGARelLJusqj2in7KkUUKYwnZ/YFvgSBHZNmb9dY7bdDMSZ1wCy5aQvOkS/PKlhQ7JGGOA+Imjg6p+KCIOGEpo2zgM2C/Ok1X1FWBhWtmSlIfNgIpum+4NfKyqn6rqKmAM4YylaLmOnUmcdgF8N5fkvy/Hr/yp0CEZY0zsxLFSRNYnfHh/paoLgJXAetWpXESuFJGvgKNZ94yjn4i8IyLjRaRbVNaO0CBfZk5UVtTcNjuQ+MM58OlskrdfjV/9c6FDMsbUc3HbOB4AXgTWB26JynZibTtElajqBcAFInI+cDpwMTAD2EJVl4nIAcATQGdS5gBJkfUsRUROIozgi6pSWlpapRgbNGhQ5efmrK6BQ1jhYOmtV9Pw/tvY4KyLcSXVHyasJvetpusr5n2r6fqKed9qur5i2be4Q46cLSIDgZ9V9aWoOAmcnaM4HgCeBi5OvYSlqs+IyK0iUko4w9gs5Tntga/LiflO4M7ooS9rdK6s1AbrfCu3rh13wR16LCsfG8X8Bo1wR/0R5zLl0hzVlwe15rW0+mptXcVeX13at7Zts/fmjD0DoKpOSHtcrXHARaSzqn4UPRwCfBCVtwG+U1UvIr0Jl9O+BxYBnUWkIzAXOAI4qjox1DWJ/X9DcvkS/HOPQ/MWuIPr1e4bY2qJrIlDRJ5V1UHR35PJcllIVXevqBIReZAwPEmpiMwhXJI6QES6Es5cvgBOjlY/DDhFRFYDPwJHqKoHVovI6cBzhO64I1X1vVh7WUTcb46DZUvx48aQbL4+ib0PKnRIxph6prwzjntT/r6rOpWo6pEZikdkWfcW1rajpC97BnimOrHUdc45GHYafvky/Jj/kGzWnETfbEOJGWNM7mVNHKr6QMrfo2omHBOHKykhcdI5JG++FH/3zfimzXHb71zosIwx9USFbRwi0gQ4HtgNaEW4H+MV4B5V/TG/4ZlsXMNGJE67gOQ/LyR5+zUkzroU16VbxU80xphqKvc+DhFpAbwBXAisInSV/Rn4O/BGtNwUiGvSlMSZF8NGrUnecgX+q2r1jjbGmFgqugHwPGA+0EVVj1XV81X1GKAL8G203BSQW38DEmddBus1IXnTxfh5WXsoG2NMTlSUOAYDf1HVZamF0ePzAOvSUwu4jVqTOPtSSK4hecNF+EXfFzokY0wRqyhxbAG8m2XZu0CHnEZjqmztoIhLbVBEY0xeVThWVTSgYLbyZM4jMlUWBkX8mw2KaIzJq4p6Va2XYcrYVOlT15kCKxsUMXn7tSRvv5rEaRfgGjQsdFjGmCJS0RnHA6w7ZWz6z4N5jc5UidtpF9ywU2HmDPzIm/DJNYUOyRhTRGJNHWvqnsRuA0kuW4p/bBQ0Wx9yMCiiMcZAJQY5NHWPDYpojMkHSxxFzgZFNMbkmiWOImeDIhpjci3u1LGmDisbFJGu24VBEf/3ZqFDMsbUYeXNx7FXnA2o6ou5C8fkyzqDIg6/Cpqvz3dLF0PLUtzQYST6Dih0iMaYOqK8S1Xp82W0I0zm9D2wEWEO8DlAp/yEZnLNNWkK/feB+++AJYtC4cL5+NHDSYIlD2NMLFkvValqx7If4D/Av4GWqtoWaAn8Kyo3dcn4R/nVZI6rVuIfH12QcIwxdU/cNo6zgfNUdQVA9Pt84E/5CszkycIsE9dnKzfGmDRxE8dyoHda2c7AityGY/KuVWnm8pIS/Hc2JLsxpmJxu+P+HXhWRJ4CviIMNzIYOC1fgZn8cEOH4UcPh1Ur1xY2aACuhOTlZ+OGnUqizx6FC9AYU+vFOuNQ1dFAH2AW0AL4AOgblZs6JNF3AG7YadCqNTgHrVrjjj2DxBW3wmYd8Hf9k+Q9/7KRdY0xWcW+AVBV3xeRD4BNVPWbPMZk8izRdwD0HUBpaSkLFqxt20iccxV+7IP48Q/jP/2QxEl/wbXvULA4jTG1U6wzDhHZUEQeAH4CPo7KhojIFfkMztQsV1JCYujvSJx1KSxfSvKqc0i+8ize+4qfbIypN+I2jt8OLCbMCFg2sdNU4Lf5CMoUltu2B4mLb4bO2+JH34q/8zr8iuWFDssYU0vETRx7A2dEl6g8gKrOBzbOV2CmsFyLliTOvAR36DH4GVNIXnE2/rOPCh2WMaYWiJs4FgPr9OMUkc0Ba+soYi6RILH/YST+8g9Ys4bkNX8lOeFxfNJmDDamPoubOO4CHhWRPYGEiPQDRhEuYZki57bahsRFN8F2O+MfvpvkLVfgly4pdFjGmAKJmziuARQYDjQERgJPAjfnKS5Ty7hm65M49XzckSfBrLdJXnYmfvbMQodljCmACrvjikgJcCxwm6relP+QTG3lnMPtNRi/1TYk77iO5PUX4g46Anfg4bhESaHDM8bUkArPOFR1DXCDqq6saF1TP7jNtyTx9xtwvXfDj32A5A0X4Rd9X+iwjDE1JO6lqqdExOYcNb9w6zXFnfgn3HFnwGezSV56Jn7m9EKHZYypAXHvHF8PeEREphLGqvrljjBVPSYfgZnazzmH23UffKeuJO+4luTNl+L2G4o7ZBiugc1KbEyxivvunhn9GPMrbtPNSPzteryOwD/3OH72eyT+cA6udZtCh2aMyYNYiUNVL61uRSIykjCi7jxV7R6VXQ4cDCSBecBxqvq1iDhCj60DCEO3H6eqM6LnHAtcGG32ClUdVd3YTPW5Ro1xvzsVv/X2JO+9heTlZ5M49nRcz10LHZoxJsdiX08QkUZAV8KNgK6svBJzjt8D3ALcm1J2nar+Pdr+GcBFwMnA/kDn6KcPcBvQR0RaARcDvQiXy6aLyFhV/SHufpj8cr36k9hiK5J3Xkfy9mtwA/bHd+gMYx/kux8W2BznxhSBWIlDRPoDDwONCcOqLwHWJ7R3xJpzXFVfEZEOaWWpd5E1Y23bycHAvarqgWnRIIubAgOA51V1YRTX88Ag4ME4MZia4Vq3IXHu1fjH78NPeBzcs1A2UKLNcW5MnRf3jONG4FpVvVFEflDVViJyETmYAVBErgSOIQxrsmdU3I6QlMrMicqylWfa7knASQCqSmlplpnvKtCgQYMqP7c211Uj9Z3yF+a9/hJ+8aJ1y1etxI29n9LBh+Wt6qJ7LQtYXzHvW03XVyz7FjdxdOHXd4lfDXwGXF+dAFT1AuACETkfOJ1wKcplWNWXU55pu3cCd5atkzrvRGWkz1mRTzVZV03V5xcvzlienD8vr3UX42tZqPqKed9qur66tG9t27bNuqwygxy2iP7+RkS2BVoCzasUUWYPAL+J/p5DmJ62THvg63LKTW2VbY7zxo3xSzMnFWNM7RY3cTxG6OEEMAJ4CZhOaPeoMhHpnPJwCGFKWoCxwDEi4kSkL7A4GtL9OWCgiLQUkZbAwKjM1FJu6DBo1HjdwkQJrFpJ8sJTSE56BsyvkicAAB0pSURBVJ9cU5jgjDFVErc77lkpf/9TRN4gnG3E/tAWkQcJjdulIjKHcEnqABHpSuiO+wWhRxXAM4RE9TGhHeX4qO6FURfeN6P1LitrKDe1U6LvAJKAf3w0pPSqcpt1IvngHfj7b8dPfp7E0SfjOnUtdLjGmBhcPZkW1H/9ddWuaNn1z/zV573HvzkZryNh8ULcbgNxQ4/Brd+inK1Ura58K+b6innfarq+urRvURtHpnbl2N1xJ5O9EXr3KkVl6j3nHK737vjte+GfGoN/4Sn89CnhjGT3gTbirjG1VNxeVXelPW4DnAjcl9twTH3k1muKO/wE/C77RJevbsO/Gl2+6til0OEZY9LEbeP41bAeIvIocDdwWa6DMvWTa7c5iT9fgX/jlTDT4D/+guu/b84uXxljcqM6Q5jOBbbPVSDGQHT5qs8e+O13xo8bg584Fj9jarh8tdu+dvnKmFogbhvHCWlFTYFDgWk5j8gYwDUpu3y1N8kH7sDfdyt+8gS7fGVMLRD3jGNY2uPlwBTCUCTG5I1rtwWJc660y1fG1CJx2zj2rHgtY/JjnctXTz0Yel/NmIo7dFhIInb5ypgaFfdSVdwRcD+tXjjGZOeaNMXJifhd9wmXr0bfin9lAomjT8F17FzxBowxORH3UtXHrL2Pw7HuPR1lN4h4wL76mbxb9/LVSJL/OCe6eXAYrrldvjIm3+ImjhOBfYBLCEODbEGYdOkFVb0nL5EZU46Ml6+mT4EefWDWOzZplDF5FDdxXA50VtUfo8cficgfgdmEmf2MKYh1Ll/dfg28NnHtQps0ypi8iDs6bgLokFa2BXZpytQSrt0WsGrlrxesWhkGWDTG5ExlZgB8UUTuJszAtxlwHNYd19QmC7MM5rZwPv6dN2C7XrhE3O9KxphsYr2LVPU6wtDmmxDmzWgDnKCq1+YxNmMqJ9ukUYkEyVuuIHnJ/5F87QX86p9rNi5jikzsIUdU9Vng2TzGYky1uKHD8KOHr3vJqlFjOPoUXEkJ/tlH8ffcjH/iPty+Q3C77Ydr0rRwARtTR5WbOERkELBEVadEj7cE7gW6A1OB46OZ+YwpuGyTRpU1jPveu8N7M0g++xj+4bvx4xQ3YH/c3gfhNmhZ0NiNqUsqOuO4HDgj5fFIwvzjRwEnANcDR+cnNGMqL9F3APQdkHECG+ccdO9JSfee+M8+Ivnco+Es5PkncbvshRs4FLdJ28IEbkwdUlHi2JJomlYR2RjYFdhCVeeKyOvA//IcnzF54Tp2puTk8/DffY2f8AR+ygv4yRNgx34kBh1qAykaU47KDKveD/hMVedGj78nzDtuTJ3lNmmLG3Yq/uAj8S+Mw096huSMKdB1OxL7HQrddwpnKsaYX1TUq+pN4AwRaQH8HhifsqwTUHOT5xqTR65FSxJDh5G4ZgTu8BNg3jck/3UpyUvPIDntJfzq1YUO0Zhao6LEcTZwGvAD0AW4OmXZMOCVPMVlTEG49ZqSGHgIiavuwB1/JiST+BE3krzgjyQnjsWv/KnQIRpTcOVeqlLV94EtRWQjVf0+bfFNwKq8RWZMAbkGDXG77I3vuye8Oz00pD90F37cQ7g9D8DtNRi3/gYkp03CPz7axsYy9Urc+TjSkwaquij34RhTu7hEAnbYmZIddsZ/8kHoyjvuIfxzj8NW28DHs+Dn6PuTjY1l6gkbf8GYmNyWW1Ny2t9IXHYrrs8eMOudtUmjjI2NZeoBSxzGVJLbtD2JY/8v+wrZxswypkhkTRwicnrK31vVTDjG1CGtWmcub9gQ/+UnNRuLMTWovDOOK1P+npHvQIypa9zQYWEsrFQlYaaB5OVns2b4VfgvbTZlU3zKaxz/RET+CbwHNBSREzKtpKoj8xKZMbVctrGx3PY74198Cv/8kyQvnwY9+pI46Ajc5p0KHbIxOVFe4jgC+CtwJNCQcN9GOk8Yv8qYeinb2Fhu8BH4vQaHu9Gff5Lk25ZATPHImjhUdTbhbnFE5AVV3bvGojKmCLimzXEHHYHfOy2B7NiXxGBLIKbuinsfx94i0gDYBWgHzAGmqqqNw2BMBdZJIBOfwk8cS/K/UQI56EjcZh0LHaIxlRIrcYhIV2Ac0IS1U8f+JCIHqeqsPMZnTNFwTZvjhhyJ3+egdRPITv3CGYglEFNHxB0d9zbgTuB6VfUAInIOcCuwZ55iM6YorZtAxoYEMmNqSCAHHYFrbwnE1G5xE0cPYN+ypBG5CbggzpNFZCQwGJinqt2jsuuAgwjjXX1CmE1wkYh0AGYBH0ZPn6aqJ0fP6QncQzjzeQY4My0mY+qMkECOwu89BP9CagLZhcRBv7UEYmqtuHeOfw3skVa2W1Qexz3AoLSy54Huqro9MBs4P2XZJ6raI/o5OaX8NuAkoHP0k75NY+oc16w5iSFHkfjHXbjBR8Cst0leeiZrbrsaP+ezQodnzK/EPeP4GzBWRMYBXwBbAAcCv4vzZFV9JTqTSC2bkPJwGnBYedsQkU2BFqo6NXp8L3AI684RYkyd5Zo1xx18FH6fIeES1gtjw6RSPXcJbSDtO9hovKZWiNuraqyI7AQI0BaYCVwUddnNhROAh1IedxSR/wJLgAtVdTJre3OVmROVZSQiJxHOTlBVSktLqxRYgwYNqvzc2lxXsddXp/ettBROPIOkHMeKpx5ixbiHSE6fQslW2+C/+GTd0XjvG06z9denyR775abuDOr0a1nL6iuWfYs9dWyUJK7IdQAicgGwGrg/KvoG2FxVv4/aNJ4QkW5Apvk7s7ZvqOqdhAZ9AJ96c1ZlpN/YlU81WVex11c0+zZwKG7XfWDiWNaMU351yK9cyZJ7b2V5t565rztSNK9lLaivLu1b27Ztsy4r6Oi4InIsodH86LJGblVdWTb/h6pOJzScdyGcYbRPeXp74rexGFNnuWbrkzj4aLJ+T1o4v0bjMaZgiUNEBgHnAkNUdUVKeWsRKYn+7kRoBP9UVb8BlopIXxFxwDHAkwUI3ZjCyDYaL7Dmur+RfPV5/I8rsq5jTK7EvlRVHSLyIDAAKBWROcDFhF5UjYHnRQTWdrvdHbhMRFYDa4CTVXVhtKlTWNsddzzWMG7qETd0GH70cFi1cm1hw0awXS+Y8zl+1L/xD9yB69EH128v2LYHLhqt15hcqpHEoapHZigekWXdR4FHsyx7C+iew9CMqTOyjcab6DsA7z18Nhs/9SX8G6/g35wMLTbE9dkD128vuyvd5FTcIUc2J5wl7Ag0T12mql3yEJcxJoOso/E6B5264jp1xcuJ8O5bJKe+hH/xafzzT0L7Drh+e+J674HbsFXhdsAUhbhnHA8DHwAXAT/mLxxjTHW5hg1hp36U7NQPv2wJ/s1X8VNfxD98N/6RUbDtDuEspEdfXOPGFW/QmDRxE8fWQD9VTeYzGGNMbrnmLXB7HgB7HoD/dg5+6iT8tJfwd/0T37gJrtcuoT2kczdcoqCdLE0dEjdxPEUYcuSlPMZijMkj16Y9bujv8AcfBR+9F85C3pqCf+0FaNUa13dAuJzVJvR6t7vUTTZxE8cZwBQR+QT4LnWBqmacUtYYUzu5RAK6bofruh3+yJPxb08LZyHjH8U/8zB07AJt2sNbr657l/ro4STBkoeJnTjuJnSNnYW1cRhTNFzjxrg+e0CfPfCLFuLfeBk/5UWY+uKvV161MvTossRR78VNHHsBbVV1aT6DMcYUjtuwFW7gUBg4lDV/GJJ5pYXz8ck1uITdH1KfxW0N+x+wUT4DMcbUIuXcpZ4890SSOgL/xcfh/hFT78Q943gRmCAid/PrNo6ROY/KGFNQWe9S7z8Qfpi/9v6QTdrheu8ebjTcJPugeKa4xE0c/YG5wMC0cg9Y4jCmyJR3lzqAX74UP31KuEt93Bj8Uw9Ch864Prvjeu1mNxkWubjzcdi84sbUM9nuUocwYq/bfT/YfT/8wgX4tybjX38Z/9AIvN4NW28XzkJ27Idr2qwwO2DyJu6QI1nbQuymQGPqN9eq9JdGdf/NV+Es5PWX8ff8C3/fbbB9LxK994Dte+EaNip0uCYH4l6qWk32SZOse4UxBgC36Wa4g4/GDzkKPv8oJJA3J5OcMRWaNMXt1A/Xe49wRmI9s+qsuIkjfWjNTYHzCHeUG2PMOpxz0LELrmMX/OEnwIf/w7/+SmgXee0F2KAlrlf/cA9Jh8445+xO9TokbhvHF2lFX0Sz971JluHRjTEGCHOCbLsjbtsd8UefHEbuff1l/Mvj8S88BRtvCm23gPdm2J3qdUR15uNoAWTv7G2MMWlco8bQc1dKeu6KX7EMP2Mq/o1X4O1pv17Z7lSvteI2jo9m3TaOpoSZ+u7LR1DGmOLnmjbH9d8X+u9b7p3qpvaJe8bxcdrj5cDtqjoxx/EYY+qjVq2zJok1119AYq/BsENvmwq3lojbxnFpvgMxxtRfWe9U36E3fPoBydv+EYZ+H3AAbrd9cc1bFC5YU37iEJGewEpVnRk9bg3cRJj3eypwjqouy3uUxpiiVu586mvWwDuvk3zxafxjo/BPPRiGOdlrMG7zToUOvV6q6IzjJuBSYGb0+C6gLXAncCRwLXBq3qIzxtQbWedTLymBnXahZKdd8HO/CONkTXsJ/9pE2GrbkEB27ItrUJ2+PqYyKhoddxtgMoCIbAjsDxytqsMJieOg/IZnjDFruXZbkBh2Kolr78YdfgIsXoi/81qS5/+e5Lgx+CU/FDrEeqGixNEAiDpW0xf4VlVnA6jqV8CGeYzNGGMycs2akxh4CIkrbiNx+t+h7Rb4Jx8IQ76PuAH/2exCh1jUKjq3ew84HFDgCOCXXlQi0g5YnL/QjDGmfC5RAjvsTMkOO+O/nRMuY015ET9tUrhzfa8DcT374xo2LHSoRaWiM45zgTtEZCFwIHBNyrLfAq/lKzBjjKkM16Y9iaP+SOK6u3FHnAQrluNH3Ejy3BNIPnk/ftH3hQ6xaJR7xqGqr4rI5kAXYHba1LFPA2PyGZwxxlSWa9IUt/dg/J4HwPtvk3zpafzTih//CG6nXXB7HghbbRMGYLSxsaqkwm4IUbKYnqH8w7xEZIwxOeASCei+EyXdd8LP+wY/6Rn8qxPxb06GVqWweBGsWR1WtrGxKiXunOPGGFNnuY03JSEnhstYvzsVFv+wNmmUKRsby1TIEocxpt5wjdcjsccgWJNl/rmF80m+MA4/75uaDayOsTtmjDH1T6vSzGNjJUrwY+7EjwHatMN174Xbrid06YZrYD2zyljiMMbUOxnHxmrUGDfsNFynLvh3p+PffSu0i0x8Eho3gW12wG3fC9e9J67lRoULvhawxGGMqXfKGxsLwO3dFvY+CL/yJ/jgfyGJvPsW/u1pYX6J9h1DEtmuJ3TqWu+mwbXEYYypl7KNjZXKNV4vDOe+Q2+89/D1l/j/vYWf+Rb+2UfxzzwMzdbHddsRtuuJ69YTt37xj9xbI4lDREYCg4F5qto9KruOMNbVKuAT4HhVXRQtOx84EVgDnKGqz0Xlg4CbgRLgLlW9uibiN8YY5xy02wLXbgvY/zdhBsP33oZ338LPnA5vvIIvm2t9u1647XrBZh1xiUTRzadeU2cc9wC3APemlD0PnK+qq0XkGuB84FwR2ZYwvEk3wki8E0WkS/Sc4cC+wBzgTREZq6rv19A+GGPML1zT5rid+8PO/fHJJHzxydpLWk/ej3/yftigJWzcFj77EFYXzz0jNZI4VPUVEemQVjYh5eE04LDo74OBMaq6EvhMRD4GekfLPlbVTwFEZEy0riUOY0xBuUQCOnbGdewMQ47EL/kBP/O/4Wxk+mvg/bpPWLUS//i9dXY+9drSxnEC8FD0dztCIikzJyoD+CqtvE+2DYrIScBJAKpKaWlplQJr0KBBlZ9bm+sq9vqKed9qur5i3re81VdaCp06wxDhu0N3zbzOwgUkbr6ERt13olH3HWnYeVtco8Y5DSNfr2XBE4eIXACsBu6PilyG1TyZb1b0GcoAUNU7CRNOAfhsjV8VKa/hLNdqsq5ir6+Y962m6yvmfauR+lpmuWdkvSb8vGQRPz80guVjPDRoCFtujevSHde1e+it1bBRtaquzr61bds267KCJg4ROZbQaL63qpYlgTnAZimrtQe+jv7OVm6MMbVS1ntGjj4lTI27fBl89B5+9kz8hzPx48bgn4oSSaeu6yaSHJ+RVFXBEkfUQ+pcYA9VXZGyaCzwgIjcQGgc7wy8QTgT6SwiHYG5hAb0o2o2amOMqZwK7xlp1hx69MH1CFfe/Ypl8NEs/Ox3QyJ5WvHjxkCDBqHHVtftcF26Q6etcY0Lk0icT2+0yQMReRAYAJQC3wEXE3pRNQbKBsmfpqonR+tfQGj3WA2cparjo/IDCPOglwAjVfXKmCH4r7+u2smJXRKom/UV877VdH3FvG81XV9V6vIrlsPH74ckMnsmfPEJ+CSUlCWS7iGRbLnNL4mkrPtvpkQVV3SpKlPTQc0kjlrAEkc9q6+Y962m6yvmfavp+nJRl/9xRVoi+RiSZYmkMzRbH977L6z+ee2TouFUKpM8ykscBW8cN8YYE59r0hTKbjAE/E8r4ONZaxPJO2/8+kllQ8bnqPuvJQ5jjKnD3HpNoXtPXPeeAKz5w8Fk7HC6MHdnVTYfhzHGFJNWWe7byFZeBZY4jDGmiLihwyC9226jxqE8R+xSlTHGFJGKuv/mgiUOY4wpMnGGjK/W9nO+RWOMMUXNEocxxphKscRhjDGmUixxGGOMqRRLHMYYYyql3oxVVegAjDGmDso4VlV9OeNwVf0RkenVeX5travY6yvmfbPXsu7WVwf3LaP6kjiMMcbkiCUOY4wxlWKJo2J3VrxKnayr2Osr5n2r6fqKed9qur6i2Lf60jhujDEmR+yMwxhjTKVY4jDGGFMpNjpuBiKyGXAv0AZIAneq6s01UG8J8BYwV1UH57muDYG7gO6E+1xOUNWpearrbOD3UT3vAser6k853P5IYDAwT1W7R2WtgIeADsDngKjqD3ms7zrgIGAV8AlhHxflq76UZecA1wGtVbXaw6Bmq0tE/g84HVgNPK2qf61uXdnqE5EewO3AelF9p6pqhvlQK11Xxvd1vo6VcurLy7FS0edWLo8VO+PIbDXwZ1XdBugLnCYi29ZAvWcCs2qgHoCbgWdVdWtgh3zVKyLtgDOAXtEHQwlwRI6ruQcYlFZ2HvCCqnYGXoge57O+54Huqro9MBs4P8/1lX1Q7At8mc+6RGRP4GBge1XtBlyfz/qAa4FLVbUHcFH0OBeyva/zdaxkqy9fx0rWz61cHyuWODJQ1W9UdUb091LCh2q7fNYpIu2BAwlnAXklIi2A3YERAKq6KlffjrNoADQRkQZAU+DrXG5cVV8BFqYVHwyMiv4eBRySz/pUdYKqro4eTgPa57O+yI3AX8nhyAhZ6joFuFpVV0brzMtzfR5oEf29ATk6Xsp5X+flWMlWX76OlQo+t3J6rFjiqICIdAB2BF7Pc1U3Ef6xyTzXA9AJmA/cLSL/FZG7RKRZPipS1bmEb6hfAt8Ai1V1Qj7qSrOJqn4TxfANsHEN1FnmBGB8PisQkSGES5rv5LOeSBdgNxF5XUReFpGd81zfWcB1IvIV4djJ5dkb8Kv3dd6PlXI+R/JyrKTWl49jxRJHOUSkOfAocJaqLsljPWXXeKfnq440DYCdgNtUdUdgObm9lPMLEWlJ+EbXEWgLNBOR3+WjrtpARC4gXDK4P491NAUuIFzGqQkNgJaEyx9/AVREsg5HkQOnAGer6mbA2URnxrlSU+/riurL17GSWl+0/ZwfK5Y4shCRhoQX/35VfSzP1e0KDBGRz4ExwF4icl8e65sDzFHVsm8/jxASST7sA3ymqvNV9WfgMWCXPNWV6jsR2RQg+p2zyyvZiMixhIbeo1U1nzdIbUlIxO9Ex0x7YIaItMlTfXOAx1TVR43USaA0T3UBHEs4TgAeBnrnasNZ3td5O1ayfY7k61jJUF9ejhXrVZVB9G1qBDBLVW/Id32qej7R6biIDADOUdW8fStX1W9F5CsR6aqqHwJ7A+/nqbovgb7Rt+Qfo7reylNdqcYSPoCujn4/mc/KRGQQcC6wh6quyGddqvouKZdTog+EXrnoVZXFE8BewCQR6QI0AvJVF4Q2jT2ASVG9H+Vio+W8r/NyrGSrL1/HSqb68nWs2J3jGYhIf2AyoetoWZvD31T1mRqoewAhceS7O24PQkN8I+BTQpfAnHRXzVDXpcBvCafN/wV+X9bQmqPtPwgMIHwL/g64mPBhp8DmhOR1uKpmamDOVX3nA42B76PVpqnqyfmqT1VHpCz/nBwljiz7NhoYCfQgdCE9R1VfrG5d5dT3IaHXXwPgJ0J33Gpfxs32via0O+T8WCmnvn+Rh2MlzueWJQ5jjDEFYW0cxhhjKsUShzHGmEqxxGGMMaZSLHEYY4ypFEscxhhjKsXu4zBFTUQuAbbK530xxUpExgNjVHVUhmUdgM+AhinjLpl6whKHqfNE5CjgT8DWwFLgbeBKVX01R9vvQB4+JEWkEaFf/9GE4VjmAy8Cl6nq59Xc9iTgPlWt8qCZqrp/dWIwxcsuVZk6TUT+RBgg8ipgE8JNXLcSxseqFaJRgTN5BBgCHEUYBXYHYDrh7vpCxWRMhezgMXWWiGwAXEa46z11PLGnop/09QcQvoW3Tyn7nHAn+0QR6U1IOl0Iw6Pcr6p/Al6JVl8kIgD7qupUETmBMOhfG+AN4CRV/SLaridMfHQW4X3WMS2WfQjzI3RR1a+i4sXA8LT9uwE4gHAn8N2Eu8bXiMhxhMmxpgEnAosId1iPF5Ergd0IQ73cBNyjqqdniklEdiHcpd2FMDfEmao6Jap/UvR63RVNMnYNcBywBPjnr/4hpt6wMw5Tl/UjzBL3eI62dzNws6q2IAwOp1H57tHvDVW1eZQ0DiFcZjoUaE0Y6uHBtO0dAvQBMk0Ctg/wRkrSyGQUYZiWrQhDZA8kJIsyfQjDc5QSJjsaISJOVS+I4jk9ivf0TDFFM989TRgCYyNCknpaRDbKEMsfCIPy7Qj0Ag4rJ25T5OyMw9RlGwELctju8DOwlYiURmP5TCtn3T8C/1DVWQAichXwNxHZouysI1qebcyjjQjzk2QkIpsA+xOS1Y/AchG5ETgJuCNa7QtV/U+0/ijC2dImwLflxP1LTCJyGPCRqo6Olj0oImcQpjW9Jz0k4KayRCci/yCMMWXqITvjMHXZ90BpDq/Xn0i4ZPOBiLwZzZOSzRbAzSKySEQWEWaxc6w7U2R5ZxPfA5tWsP2GwDcpddzBupMM/ZIgUkZZbV7ONtNjagt8kbb8CzLPdtk27bnpzzP1iJ1xmLpsKmH01EMIDc0VWU6YuhaA6Lp967LHqvoRcKSIJAiXoB6JLttkGgn0K0LPrfIm4SlvBNGJwJki0l5V52TZ/kqgtIpnVNnqTi3/mpCgUm0OPJvhed8Am6WtZ+opSxymzlLVxSJyETBcRFYDEwiXm/YB9lTVv6Y9ZTawnogcGK37N8Lw1gBEMxM+p6rzo2/4AGsI3WSThCl3Z0fltwOXi8jbqvpe1JA9UFUfjhn7RBF5HnhcRE4G3gGaELrmrlLVkSIyAfiniPwdWEZoYG+vqi/HqOK7KN7yPAP8O+rOrMBvCO0x4zKFDJwhIuPI44yRpm6wS1WmTosmrPkTcCHhA/4rQs+hJzKsuxg4lTAPyVzCB2Dqt/1BwHsisozQUH6Eqv4UXQa6EngtumzUV1UfJ/QyGiMiS4CZhDaJyjiM8OH9EKFH1UxCw/PEaPkxhPlS3gd+IJxVlXd5K9XNwGEi8oOI/CvTCqr6PaHB+8+ES2d/BQZnmavhP8BzhAQ3g7Uz9Jl6yObjMMYYUyl2xmGMMaZSLHEYY4ypFEscxhhjKsUShzHGmEqxxGGMMaZSLHEYY4ypFEscxhhjKsUShzHGmEr5f6phpZcjG7SsAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 1440x1440 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from sklearn.cluster import KMeans #import kmeans which will be used to cluster the data\n",
    "\n",
    "\n",
    "max_number_of_clusters = 25 #clusters can range from 2-max_number_of_clusters\n",
    "max_number_of_clusters = max_number_of_clusters+1 # since the range method is exclusive, a 1 should be added.\n",
    "Ks = range(2, max_number_of_clusters, 2) #Hold the range of clusters k (from 2-max_number_of_clusters)\n",
    "    \n",
    "sum_of_squared_distances = [] #a list which will carry the sum of squared distances at each k\n",
    "for k in tqdm(Ks):\n",
    "    sum_of_squared_distances.append(KMeans(n_clusters=k,random_state=30).fit(tf_idf_norm_kmeans).inertia_)#calculate the sum of squared distance for each k.\n",
    "        \n",
    "  #plot the elbow graph      \n",
    "f, ax = plt.subplots(1, 1)\n",
    "ax.plot(Ks, sum_of_squared_distances, marker='o')\n",
    "ax.set_xlabel('Cluster Centroid')\n",
    "ax.set_xticks(Ks)\n",
    "ax.set_xticklabels(Ks)\n",
    "ax.set_ylabel('Sum of Squared Distances')\n",
    "ax.set_title('Sum of Squared Distances VS. Clusters')\n",
    "plt.figure(figsize=(20,20))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h5>Elbow Graph Analysis </h5>\n",
    "<p>As exhibited in the graph above the elbow forms at k = 15, as the graph starts to level off</p>\n",
    "<h1 style =\"color:red\"> CHANGE K WHEN YOU RUN ALL THE DATASET. The above is for 1934 papers</h1>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h5> K-Means clustering</h5>\n",
    "<p>Now that the number of clusters is known, k-means can be performed as shown in the cell below </p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "clusters = KMeans(n_clusters=15,random_state=30).fit_predict(tf_idf_norm_kmeans) #from the elbow graph we get the n_clusters\n",
    "df =FullPaperDataframe.copy(deep=True)  #create a new df which will hold the papers attribute along with the cluster number\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<p>In the below cell, the cluster to which the paper belongs is added to the dataframe</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "      <th>clusters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>The RNA pseudoknots in foot-and-mouth disease virus are dispensable for genome replication but essential for the production of infectious virus. 2 3</td>\n",
       "      <td>[{'text': 'word count: 194 22 Text word count: 5168 23 24 25 author/funder. All rights reserved. No reuse allowed without permission. Abstract 27 The positive stranded RNA genomes of picornaviruses comprise a single large open reading 28 frame flanked by 5′ and 3′ untranslated regions (UTRs). Foot-and-mouth disease virus (FMDV) 29 has an unusually large 5′ UTR (1.3 kb) containing five structural domains. These include the 30 internal ribosome entry site (IRES), which facilitates initiation of translation, and the cis-acting 31 replication element (cre). Less well characterised structures are a 5′ terminal 360 nucleotide 32 stem-loop, a variable length poly-C-tract of approximately 100-200 nucleotides and a series of 33 two to four tandemly repeated pseudoknots (PKs). We investigated the structures of the PKs 34 by selective 2′ hydroxyl acetylation analysed by primer extension (SHAPE) analysis and 35 determined their contribution to genome replication by mutation and deletion experiments. 36 SHAPE and mutation experiments confirmed the importance of the previously predicted PK 37 structures for their function. Deletion experiments showed that although PKs are not essential 38', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'for replication, they provide genomes with a competitive advantage. However, although 39 replicons and full-length genomes lacking all PKs were replication competent, no infectious 40 virus was rescued from genomes containing less than one PK copy. This is consistent with our 41 earlier report describing the presence of putative packaging signals in the PK region. 42 43 author/funder. All rights reserved. No reuse allowed without permission.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>VP3, and VP0 (which is further processed to VP2 and VP4 during virus assembly) (6). The P2 64 and P3 regions encode the non-structural proteins 2B and 2C and 3A, 3B (1-3) (VPg), 3C pro and 4 structural protein-coding region is replaced by reporter genes, allow the study of genome 68 replication without the requirement for high containment (9, 10) ( figure 1A ).\\nThe FMDV 5′ UTR is the largest known picornavirus UTR, comprising approximately 1300 71 nucleotides and containing several highly structured regions. The first 360 nucleotides at the 5′ 72 end are predicted to fold into a single large stem loop termed the S-fragment, followed by a The PKs were originally predicted in 1987 and consist of two to four tandem repeats of a ~48 86 nucleotide region containing a small stem loop and downstream interaction site (figure 1B) 87 (12). Due to the sequence similarity between the PKs (figure 1C), it is speculated that they 88 were formed by duplication events during viral replication, probably involving recombination. 89 Between two and four PKs are present in different virus isolates but no strain has been 90 identified with less than two PKs, emphasising their potential importance in the viral life cycle 91 (19, 20) . The presence of PKs has been reported in the 5′ UTR of other picornaviruses such as 92 author/funder. All rights reserved. No reuse allowed without permission. can occur in the absence of PKs at least one is required for wild-type (wt) replication. 104 Furthermore, competition experiments showed that extra copies of PKs conferred a replicative 105 advantage to genomes. Although replicons and full-length genomes lacking PKs were 106 replication-competent, no infectious virus was rescued from genomes containing less than one 107 PK copy. This is consistent with our earlier report describing the presence of putative 108 packaging signals in the PK region (22). 109 110 author/funder. All rights reserved. No reuse allowed without permission. Plasmid construction. 117 The FMDV replicon plasmids, pRep-ptGFP, and the replication-defective polymerase mutant 118 control, 3D-GNN, have already been described (10).\\nTo introduce mutations into the PK region, the pRep-ptGFP replicon plasmid was digested 121 with SpeI and KpnI and the resulting fragment inserted into a sub-cloning vector (pBluescript) 122 to create the pBluescript PK. PKs 3 and 4 were removed by digestion with HindIII and AatII 123 before insertion of a synthetic DNA sequence with PK 3 and 4 deleted. PKs 2, 3 and 4 were 124 deleted by PCR amplification using ΔPK 234 Forward primer and FMDV 1331-1311 reverse 125 primer, the resultant product was digested with HindIII and AatII and ligated into the 126 pBluescript PK vector. Complete PK deletion was achieved by introduction of an AflII site at 127 the 3′ end of the poly-C tract by PCR mutagenesis to create the sub-cloning vector, pBluescript 128 C11, which was then used to remove all the PKs by PCR mutagenesis using ΔPK 1234 forward 129 primer and FMDV 1331-1311 reverse primer. The modified PK sequences were removed from 130 the sub-cloning vectors and inserted into the pRep-ptGFP plasmid using NheI-HF and KpnI-131 HF.\\n132 133 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 7 Mutations to disrupt and reform PK structure were introduced using synthetic DNA by 134 digestion with AflII and AatII and ligation into a similarly digested pBluescript PK vector.\\nMutations were then introduced into the replicon plasmid as described above.\\nTo assess the effects of truncation of the poly-C-tract on replication the entire sequence was 137 removed. This was performed by PCR mutagenesis using primers C0 SpeI, and FMDV 1331- In vitro transcription. 143 In vitro transcription reactions for replicon assays were performed as described previously (28).\\nTranscription reactions to produce large amounts of RNA for SHAPE analysis were performed 145 with purified linear DNA as described above, and 1 μg of linearised DNA was then used in a 146 HiScribe T7 synthesis kit (NEB), before DNase treatment and purification using a PureLink FastQ files were quality checked using FastQC with poor quality reads filtered using the 225 Sickle algorithm. Host cell reads were removed using FastQ Screen algorithm and FMDV 226 reads assembled de novo into contigs using IDBA-UD (35). Contigs that matched the FMDV 227 library (identified using Basic Local ALighnment Search Tool (BLAST)) were assembled 228 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint into consensus sequences using SeqMan Pro software in the DNA STAR Lasergene 13 229 package (DNA STAR) (36). The SHAPE data largely agreed with the predicted structures with the stems of PK 1, 2 and 3, interacting nucleotides showed little to no reactivity, suggesting NMIA could not interact with 300 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 14 these nucleotides either due to the predicted base pairing or steric hindrance (figure 2B). The\\nNMIA reactivity for the interacting nucleotides in the stem-loops with downstream residues of 302 PK 1, 2 and 3 again largely agreed with the predicted structure, although the SHAPE data 303 suggests that there might be fewer interactions than previously predicted. However, differences 304 here could be due to heterogeneity in the formation of PKs in this experiment. The evidence 305 for loop-downstream interaction was weaker for PK4. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint\\norientation. 351 Since removal of all four PKs resulted in a significant decrease in replication, the minimal 352 requirements to maintain wt levels of replication were investigated. As near wt level of 353 replication was observed when only one PK was present, all further mutagenesis was 354 performed in a C11 replicon plasmid containing only PK 1. In addition, the orientation of PK 1 was reversed by \"flipping\" the nucleotide sequence to 367 potentially facilitate hybridisation of the loop with upstream rather than downstream sequences.\\nChanging the orientation of the PK reduced replicon replication to a similar level seen in the replication decreased until at passage three there is a 2.5 fold reduction compared to that of 398 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint passage 0 (figure 5B). Therefore, it appears that replicons with a single PK are at a competitive 399 disadvantage compared to those with two or more. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 20 of infectious virus despite being able to replicate after transfection into cells, is consistent with 448 a requirement for RNA structure within the PK region being required for virus assembly. The 5′ UTR of FMDV is unique amongst picornaviruses due to its large size and the presence 454 of multiple RNA elements, some of which still have unknown function. One of these features 455 is a series of repeated PKs varying in number from 2-4, depending on virus strain. In this study, 456 we sequentially deleted or mutated the PKs to help understand their role in the viral life cycle. 457 We also confirmed the predicted PK structures by SHAPE mapping, although there may be Although all viruses isolated to date contain at least two PKs, replicons or viruses containing a 464 single PK were still replication competent. However, replicons with more than a single PK 465 were found to have a competitive advantage over replicons with a single PK when sequentially 466 passaged. Replicons lacking all PKs displayed poor passaging potential even when co-467 transfected with yeast tRNA, reinforcing the observation of a significant impact in replication.\\nMoreover, viruses recovered from genomes with reduced numbers of PKs were slower growing 469 and produced smaller plaques. In addition, these differences were more pronounced in more PKs is functionally competent as no differences was seen between replicons congaing a single 472 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 21 copy of PK1 or PK4. This observation is consistent with a previous report of deletion of PK1, 473 along with the Poly-C-tract, with no adverse effect in viral replication (37). This also supports 474 our findings that the truncation of the Poly-C-tract to create the C11 construct had no effect on 475 replicon replication in the cell lines tested. As has been described with Mengo virus, it is 476 possible that the role of the poly-C-tract is essential in other aspects of the viral lifecycle which 477 cannot be recapitulated in a standard tissue culture system (39).\\nThe presence of at least two PKs in all viral isolates sequenced so far suggests that multiple 480 PKs confer a competitive advantage in replication. Here we showed by sequential passage that 481 replicons containing at least two PKs were maintained at a level similar to wt, but replicons 482 containing only one PK showed a persistent decline. It is unclear why some viral isolates 483 contain two, three or four PKs is still unknown, but this may be stochastic variation or may 484 reflect subtle effects of host range or geographical localisation. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint Significance is shown comparing the replication of C11 PK disrupt and C11 PK restore (Aii). Significance shown is compared to wt replicon. Error bars are calculated by SEM, n = 3, * P 673 &lt; 0.05, **** P &lt; 0.0001. 674 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 33 675 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint \\n</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Analysis Title: Regaining perspective on SARS-CoV-2 molecular tracing and its implications</td>\n",
       "      <td>[{'text': 'During the past three months, a new coronavirus (SARS-CoV-2) epidemic has been growing exponentially, affecting over 100 thousand people worldwide, and causing enormous distress to economies and societies of affected countries. A plethora of analyses based on viral sequences has already been published, in scientific journals as well as through non-peer reviewed channels, to investigate SARS-CoV-2 genetic heterogeneity and spatiotemporal dissemination. We examined all full genome sequences currently available to assess the presence of sufficient information for reliable phylogenetic and phylogeographic studies. Our analysis clearly shows severe limitations in the present data, in light of which any finding should be considered, at the very best, preliminary and hypothesis-generating. Hence the need for avoiding stigmatization based on partial information, and for continuing concerted efforts to increase number and quality of the sequences required for robust tracing of the epidemic.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': '. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'is the (which was not peer-reviewed) The copyright holder for this preprint .', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>In December 2019, a novel coronavirus, SARS-CoV-2, was identified in Wuhan, China, as the etiologic agent of coronavirus disease 2019 , which by March 2020 has already spread across more than 80 countries 1 . Common symptoms of infection include fever, cough, and shortness of breath, while severe cases are characterized by advanced respiratory distress and pneumonia, often resulting in death 2 . Soon after the first epidemiological data together with SARS-CoV-2 genetic sequences were made available, a glut of phylogeny-based analyses began to circulate discussing, in scientific papers as well as (social) media, countries that might have been fueling the spread. The implications of misunderstanding the real dynamic of the COVID-19 pandemic are extremely dangerous. Ethnic or social discrimination resulting from unsupported assumptions on viral contagion -often amplified by irresponsible, uncontrollable communications -can be highly damaging for people and countries.\\nDespite (social) media are often vehicle for fake news and boast news hype, it is also worth noting the tremendous effort of the scientific community to provide free, up-to-date information on ongoing studies, as well as critical evaluations. In particular, the US-based NextStrain 3 team has been posting real-time updates on the epidemic tracing by molecular analyses. Several discussions and evidence-based debates on controversial hypotheses on the epidemic have ensued -such as the number of untraced infections in the US, the putative virus introduction in Italy through Germany, and the alleged lineage diversification in China 4 later criticized 5 -aided by a concerted and productive effort of internationally-renowned research groups in Edinburgh, Glasgow, and Oxford, UK, among others. Recently, an editorial published on Science 6 has also highlighted how unsupported or misleading claims circulating in forums, social media, and even peer-reviewed articles, have been led by a substantial over interpretation of the available data.\\nHence, the urgency to reframe the current debate in more rigorous scientific terms, and quantitatively evaluate whether sufficient information for reliable phylogenetic and phylogeographic studies currently exists, or which gaps need to be addressed.\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint Before carrying out any phylogeny-based analysis of virus evolution and spatiotemporal spread, it is crucial to test the quality of sequence data, since uneven sampling, presence of phylogenetic noise, and absence of temporal signal can affect reliability of the results (e.g. ancestral state reconstructions, molecular clock calibrations) 7 . We downloaded all SARS-CoV-2 full genome sequences deposited on GISAID (https://www.gisaid.org/) 8 to compare the number of high quality genomes sampled per country with the number of confirmed cases at the time of sampling, as well as the country's total population ( Figure 1 ). As March 10 th 2020, 331 genome sequences from 29 countries were available, almost twice as much than just one week earlier (March 3 rd 2020, 169 genome sequences from 22 countries). Yet, the sampling per country was inhomogeneous, with confirmed cases clearly uncorrelated to the number of genomes (Spearman correlation = 0.03). Moreover, correlation could only be investigated with confirmed cases (as proxy), since not all affected countries have made publicly available the total number of coronavirus testing performed. Even within the same country, sequenced genomes were usually sampled from few hotspots, not necessarily representative of the whole epidemic in that country.\\nItaly, for example, the second country per confirmed cases, has uploaded so far only two genomes from Lombardy and three from Lazio (Table S1) , i.e. only two of the twenty Italian regions, all with confirmed cases 9 . China and USA contributed with 123 and 63 genomes respectively (56% combined), while 13 countries contribute with only one or two samples (6.3% combined). Notably, three countries with less than 400 confirmed cases have provided a substantial number of genomes (Netherlands 28, Australia 18, Singapore 11; 17% combined) ( Figure 1 ).\\nThe effect of inhomogeneous sampling and missing data on phylogeography reconstructions, like the ones recently rushed through news and (social) media to claim specific dissemination routes of SARS-CoV-2 among countries, can be quite dramatic. An instructive example is the putative introduction of SARS-CoV-2 in Italy from Germany. The maximum likelihood (ML) tree, inferred from the full genome viral sequences available on March 3 rd , 2020 (Supplementary Figure S1 ), showed a well-supported cluster of European and Asian sequences (reported in Figure 2a ), which contained a subclade (subclade A, Figure 2a ) including a sequence isolated in Germany that . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint appears to be paraphyletic (with strong bootstrap support) to an Italian sequence clustering, in turn, with sequences from Finland, Mexico, Germany and Switzerland (Figure 2a ).\\nBased on this observation (which was available on NextStrain), a heated discussion circulated on social media about a transmission event from Germany to Italy followed by further spread from Italy the other countries. However, in a new tree inferred just one week later, when more than 135 new full genome sequences were made available on GISAID 8 ( Figure S2 ), the direct link between Germany and Italy has disappeared due to the additional clustering of previously unsampled sequences from Portugal, Brazil, Wales and Netherland ( Figure 2b ). Most importantly, there is no sufficient evidence of a higher likelihood (Shimodaira-Hasgawa test, Table 1 ) for tree topologies generated arbitrarily switching branches in the tree (see arrows in Figure 2b) compared to the ML tree inferred from the real data. In other words, it is not possible with the present data to decide which branching pattern (and, therefore, phylogeographic reconstruction)\\nis the most likely.\\nLack of resolution and uncertainty in the SARS-CoV-2 phylogenetic tree is to be expected, considering that relatively little genetic diversity can be accumulated during the first three months of an epidemic, even for an exponentially spreading and fast-evolving RNA virus. Plots of pairwise transitions/transversions vs. genetic distance 10 showed, indeed, low genetic diversity among SARS-CoV-2 genomes, considerable scattering, and overlap in transitions and transversions among closely related sequences (Figure 3a ). The 331 full genomes available contained 477 out of 29,564 variable sites (1.6%), of which 377 were singleton and only 100 (0.34%) parsimony informative, indicating a substantial lack of phylogenetic information, and raising the possibility of a number of sequencing errors. Further assessment of phylogenetic signal in the dataset was carried out using likelihood mapping analysis 11 , which estimates the likelihood of each of possible tree topology for any group of four sequences (quartet), randomly . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint chosen from an alignment, and reports them inside an equilateral triangle (the likelihood map)\\nwhere the corners represent distinct tree topologies and the center star-like trees. Quartets are considered \"resolved\" when the three likelihoods are significantly different (phylogenetic signal, most dots equally distributed in the corners, i.e. data are suitable for robust phylogeny inference), unresolved or partially resolved, when all three likelihood values or two of them are not significantly different (phylogenetic noise, most dots in sides or center areas, i.e. data may not be sufficient for robust phylogeny inference). Extensive simulation studies have shown that, for sequences to be considered robust in terms of phylogenetic signal, side/center areas of the likelihood mapping must include &lt;40% of the unresolved quartets 12 , way lower than the 52.9% unresolved quartets detected in the SARS-CoV-2 full genomes alignment ( Figure 3b ). It is important to keep in mind that such a lack of phylogenetic signal makes the overall topology of any SARS-CoV-2 tree unreliable, and even clades with high bootstrap values should be interpreted with extreme caution. Finally, the mean genetic distance of each sequence from the root of a ML tree indicated that the available genomic data lack temporal signal (no significant linear correlation between accumulated genetic distance and sampling time, Figure 3c ). This is, again, expected in genomes obtained over a very short period of time (~ three months) since the beginning of the outbreak. Bayesian analysis 13 , which infers phylogenetic and phylogeographic patterns from a posterior distribution of trees, might facilitate comparisons about different evolutionary scenarios, help in retrieving the correct topology, and estimate an accurate evolutionary rate. With more genome sequences sampled from different time points and geographic areas made available, literally, every day, Bayesian analysis may soon be a viable option. However, as long the new data do not increase phylogenetic and temporal signal, Bayesian phylogenetic and phylogeography results would remain equally questionable.\\nTaken together, our observations suggest that any conclusion drawn, at present, about existing lineages and direction of viral spread, based on phylogenetic analysis of SARS-CoV-2 sequence data, is potentially misleading. Nowadays, published scientific data and media are very easily . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint .\\naccessible to a worldwide audience, thus it is important more than ever to weigh the information being shared. We believe that current molecular epidemiology data are not solid enough to provide a scientifically sound analysis of SARS-CoV-2 spread. Any inference based on phylogenetic trees should be considered at the very best preliminary, and hypothesis-generating.\\nHence the need for 1) avoiding stigmatization based on partial information and 2) continuing the concerted scientific effort to increase the number and quality of available data. The evolutionary dynamics of SARS-CoV-2 spread will unveil an unprecedented amount of information, which is essential to make policy decisions. The whole of humanity is threatened by the current pandemic, and policy makers need to adjust their mitigation measures while the pandemic itself is developing. Some of the urgent answers required lie in the timely availability of abundant, high quality sequence data not only from countries experiencing a high number of reported cases, such as Italy and South Korea, but also from others that seem to be experiencing, at least for now, a lower number of infections.\\nGISAID was accessed on March 3 rd and 10 th , 2020 (final dataset information reported in Table   S1 ). After quality control of sequences that were not full genomes or contained extensive stretches of unknown nucleotides, the first dataset (March 3 rd 2020) consisted of 169 genomes from 22 countries, while the second (March 10 th , 2020) 331 from 29 countries. Confirmed cases were retrieved for the same day from interactive web-based dashboard developed by Johns Hopkins University 14 .\\nTransition/transversions vs. genetic distance plot were calculated using DAMBE6 10 . Evaluation of the presence of phylogenetic signal satisfying resolved phylogenetic relationships among sequences was carried out with IQ-TREE, allowing the software to search for all possible quartets . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint using the best-fitting nucleotide substitution model 11 . ML tree reconstruction was performed in IQ-TREE based on the best-fit model chosen according to Bayesian Information Criterion (BIC) 15, 16 . Topology was tested performing the Shimodaira-Hasegawa 17 on \"Subclade A\" by comparing the inferred tree to three trees where two branches: (Tree 1) Italy (EPI_ISL_412973) switched with Wales (EPI_ISL_413555), (Tree 2) Germany (EPI_ISL_406862 with) Brazil (EPI_ISL_412964), (Tree 3) and Portugal (EPI_ISL_413648) with Brazil (EPI_ISL_412964), (Tree 4) Germany (EPI_ISL_406862) with Portugal (EPI_ISL_413648). All tests performed 10,000 resamplings using the using the RELL approximation method 18 . Position of these genomes within \"Subclade A\" is indicated in Figure 1 with an arrow.\\nExploration of temporal structure, i.e. presence of molecular clock in the data, was assessed by regression of divergence -root-to-tip genetic distance-against sampling time using TempEst 19 . In this case, absence of a linear trend indicates that the data does not contain temporal signal and that the data is not appropriate for phylogenetic inference using molecular clock models.\\nAncestral state reconstruction of full datasets reported in Figures S1 and S2 were based on the ML tree obtained with IQ-TREE and performed using TreeTime 20 . We choose to show the \"European subclade\" and \"Subclade A\" as cladograms in Figure 1 as there are no phylogenetic and temporal signals in the datasets: scaling phylogenies in time or nucleotide substitution in this case is misleading. . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint Each bi-directional arrow, and corresponding number, connects two tip branches that were switched to generate an alternative tree topology to be tested (see Table 1 , Methods). CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity. EPI_ISL_412971_BetaCoV_Finland_FIN-25_2020 EPI_ISL_406533_BetaCoV_Guangzhou_20SF206_2020 EPI_ISL_407896_BetaCoV_Australia_QLD02_2020 EPI_ISL_408478_BetaCoV_Chongqing_YC01_2020\\nEPI_ISL_412979__BetaCoV_Wuhan_HBCDC-HB-03_2020\\nEPI_ISL_411220_BetaCoV_France_IDF0386-islP3_2020 EPI_ISL_410044_BetaCoV_USA_CA6_2020 EPI_ISL_413020_BetaCoV_Switzerland_1000477377_2020\\nEPI_ISL_406030_BetaCoV_Shenzhen_HKU-SZ-002_2020\\nEPI_ISL_408669_BetaCoV_Japan_KY-V-029_2020\\nEPI_ISL_408515_BetaCoV_Wuhan_IVDC-HB-envF13-21_2020 EPI_ISL_410713_BetaCoV_Singapore_7_2020 EPI_ISL_412873_BetaCoV_Korea_KCDC24_2020\\nEPI_ISL_402120_BetaCoV_Wuhan_IVDC-HB-04_2020\\nEPI_ISL_406970_BetaCoV_Hangzhou_HZ-1_2020 EPI_ISL_406973_BetaCoV_Singapore_1_2020 EPI_ISL_412029_BetaCoV_Hong_Kong_VM20001988_2020\\nEPI_ISL_410720_BetaCoV_France_IDF0372-isl_2020 EPI_ISL_411951_BetaCoV_Sweden_01_2020 EPI_ISL_408488_BetaCoV_Jiangsu_IVDC-JS-001_2020\\nEPI_ISL_402123_BetaCoV_Wuhan_IPBCAMS-WH-01_2019 EPI_ISL_407894_BetaCoV_Australia_QLD01_2020 EPI_ISL_411955_BetaCoV_USA_CA8_2020 EPI_ISL_403962_BetaCoV_Nonthaburi_61_2020 EPI_ISL_406801_BetaCov_Wuhan_WH04_2020 EPI_ISL_408978_BetaCoV_Wuhan_WH05_2020\\nEPI_ISL_407084_BetaCoV_Japan_AI_I-004_2020 EPI_ISL_407215_BetaCoV_USA_WA1-F6_2020 EPI_ISL_411060_BetaCoV_Fujian_8_2020 EPI_ISL_410984_BetaCoV_France_IDF0515-isl_2020\\nEPI_ISL_412980__BetaCoV_Wuhan_HBCDC-HB-04_2020 EPI_ISL_410546_BetaCoV_Italy_INMI1-cs_2020 EPI_ISL_408977_BetaCoV_Sydney_3_2020 EPI_ISL_412982__BetaCoV_Wuhan_HBCDC-HB-06_2020 EPI_ISL_407071_BetaCoV_England_01_2020 EPI_ISL_412912_BetaCoV_Germany_Baden-Wuerttemberg-1_2020 EPI_ISL_411927_BetaCoV_Taiwan_4_2020 EPI_ISL_410536_BetaCoV_Singapore_5_2020 EPI_ISL_406538_BetaCoV_Guangdong_20SF201_2020 EPI_ISL_407988_BetaCoV_Singapore_3_2020 EPI_ISL_406036_BetaCoV_USA_CA2_2020 EPI_ISL_411954_BetaCoV_USA_CA7_2020 EPI_ISL_412028_BetaCoV_Hong_Kong_VM20001061_2020 EPI_ISL_406862_BetaCoV_Germany_BavPat1_2020 EPI_ISL_406597_BetaCoV_France_IDF0373_2020 EPI_ISL_408009_BetaCoV_USA_CA4_2020 EPI_ISL_410716_BetaCoV_Singapore_10_2020 EPI_ISL_403933_BetaCoV_Guangdong_20SF013_2020 EPI_ISL_402129_BetaCoV_Wuhan_WIV06_2019\\nEPI_ISL_406594_BetaCoV_Shenzhen_SZTH-003_2020 EPI_ISL_407987_BetaCoV_Singapore_2_2020 EPI_ISL_413015_BetaCoV_Canada_ON_VIDO-01_2020 EPI_ISL_407976_BetaCoV_Belgium_GHB-03021_2020 EPI_ISL_406531_BetaCoV_Guangdong_20SF174_2020 EPI_ISL_410715_BetaCoV_Singapore_9_2020\\nEPI_ISL_404227_BetaCoV_Zhejiang_WZ-01_2020 EPI_ISL_408479_BetaCoV_Chongqing_ZX01_2020 EPI_ISL_406535_BetaCoV_Foshan_20SF210_2020 EPI_ISL_408976_BetaCoV_Sydney_2_2020 EPI_ISL_411066_BetaCoV_Fujian_13_2020 EPI_ISL_407073_BetaCoV_England_02_2020\\nEPI_ISL_403929_BetaCoV_Wuhan_IPBCAMS-WH-04_2019 EPI_ISL_410301_BetaCoV_Nepal_61_2020 EPI_ISL_403931_BetaCoV_Wuhan_IPBCAMS-WH-02_2019\\nEPI_ISL_410531_BetaCoV_Japan_NA-20-05-1_2020 EPI_ISL_403963_BetaCoV_Nonthaburi_74_2020 EPI_ISL_402125_BetaCoV_Wuhan-Hu-1_2019\\nEPI_ISL_408514_BetaCoV_Wuhan_IVDC-HB-envF13-20_2020\\nEPI_ISL_402119_BetaCoV_Wuhan_IVDC-HB-01_2019 EPI_ISL_411956_BetaCoV_USA_TX1_2020 EPI_ISL_406596_BetaCoV_France_IDF0372_2020 EPI_ISL_412967_BetaCoV_China_IQTC02_2020 EPI_ISL_408431_BetaCov_France_IDF0626_2020\\nEPI_ISL_410532_BetaCoV_Japan_OS-20-07-1_2020\\nEPI_ISL_411957_BetaCoV_China_WH-09_2020 EPI_ISL_412970_BetaCoV_USA_WA2_2020 EPI_ISL_412978__BetaCoV_Wuhan_HBCDC-HB-02_2020 EPI_ISL_412869_BetaCoV_Korea_KCDC05_2020 EPI_ISL_412965_BetaCoV_Canada_BC_37_0-2_2020 EPI_ISL_412871_BetaCoV_Korea_KCDC07_2020 EPI_ISL_408008_BetaCoV_USA_CA3_2020\\nEPI_ISL_406593_BetaCoV_Shenzhen_SZTH-002_2020 EPI_ISL_406536_BetaCoV_Foshan_20SF211_2020 EPI_ISL_412459_BetaCoV_Jingzhou_HBCDC-HB-01_2020 EPI_ISL_410718_BetaCoV_Australia_QLD04_2020 EPI_ISL_412862_BetaCoV_USA_CA9_2020 EPI_ISL_413019_BetaCoV_Switzerland_1000477102_2020\\nEPI_ISL_412981__BetaCoV_Wuhan_HBCDC-HB-05_2020\\nEPI_ISL_408668_BetaCoV_Vietnam_VR03-38142_2020 EPI_ISL_410535_BetaCoV_Singapore_4_2020 EPI_ISL_408665_BetaCoV_Japan_TY-WK-012_2020 EPI_ISL_406716_BetaCoV_China_WHU01_2020 EPI_ISL_402127_BetaCoV_Wuhan_WIV02_2019 EPI_ISL_411926_BetaCoV_Taiwan_3_2020 EPI_ISL_406223_BetaCoV_USA_AZ1_2020 EPI_ISL_407214_BetaCoV_USA_WA1-A12_2020 EPI_ISL_406717_BetaCoV_China_WHU02_2020 EPI_ISL_404228_BetaCoV_Zhejiang_WZ-02_2020\\nEPI_ISL_412898__BetaCoV_Wuhan_HBCDC-HB-02_2019 EPI_ISL_413017_BetaCoV_South_Korea_KUMC01_2020 EPI_ISL_408480_BetaCoV_Yunnan_IVDC-YN-003_2020 EPI_ISL_403936_BetaCoV_Guangdong_20SF028_2020 EPI_ISL_408667_BetaCoV_Japan_TY-WK-521_2020 EPI_ISL_411218_BetaCoV_France_IDF0571_2020 EPI_ISL_405839_BetaCoV_Shenzhen_HKU-SZ-005_2020 EPI_ISL_408010_BetaCoV_USA_CA5_2020 EPI_ISL_406031_BetaCoV_Taiwan_2_2020 EPI_ISL_406844_BetaCoV_Australia_VIC01_2020 EPI_ISL_411950_BetaCoV_Jiangsu_JS01_2020 EPI_ISL_412974_BetaCoV_Italy_SPL1_2020\\nEPI_ISL_403928_BetaCoV_Wuhan_IPBCAMS-WH-05_2020 EPI_ISL_412870_BetaCoV_Korea_KCDC06_2020 EPI_ISL_404253_BetaCoV_USA_IL1_2020 EPI_ISL_410045_BetaCoV_USA_IL2_2020 EPI_ISL_403932_BetaCoV_Guangdong_20SF012_2020\\nEPI_ISL_408484_BetaCoV_Sichuan_IVDC-SC-001_2020 EPI_ISL_410537_BetaCoV_Singapore_6_2020 EPI_ISL_412872_BetaCoV_Korea_KCDC12_2020 EPI_ISL_407079_BetaCoV_Finland_1_2020\\nEPI_ISL_410545_BetaCoV_Italy_INMI1-isl_2020 EPI_ISL_408670_BetaCoV_USA_WI1_2020 EPI_ISL_403937_BetaCoV_Guangdong_20SF040_2020 EPI_ISL_403934_BetaCoV_Guangdong_20SF014_2020\\nEPI_ISL_413014_BetaCoV_Canada_ON-PHL2445_2020\\nEPI_ISL_411915_BetaCoV_Taiwan_CGMH-CGU-01_2020 EPI_ISL_410717_BetaCoV_Australia_QLD03_2020 EPI_ISL_412026_BetaCoV_Hefei_2_2020 EPI_ISL_407313_BetaCoV_Hangzhou_HZCDC0001_2020 EPI_ISL_412972_BetaCoV_Mexico_CDMX_InDRE_01_2020 EPI_ISL_406034_BetaCoV_USA_CA1_2020 EPI_ISL_411952_BetaCoV_Jiangsu_JS02_2020 EPI_ISL_411902_BetaCoV_Cambodia_0012_2020\\nEPI_ISL_413016_BetaCov_Brazil_SPBR-02_2020 EPI_ISL_413018_BetaCoV_South_Korea_KUMC02_2020 EPI_ISL_403930_BetaCoV_Wuhan_IPBCAMS-WH-03_2019\\nEPI_ISL_412899__BetaCoV_Wuhan_HBCDC-HB-03_2019 EPI_ISL_408489_BetaCoV_Taiwan_NTU01_2020 EPI_ISL_409067_BetaCoV_USA_MA1_2020 EPI_ISL_404895_BetaCoV_USA_WA1_2020\\nEPI_ISL_408666_BetaCoV_Japan_TY-WK-501_2020 EPI_ISL_410218_BetaCov_Taiwan_NTU02_2020 EPI_ISL_407893_BetaCoV_Australia_NSW01_2020 EPI_ISL_407193_BetaCoV_South_Korea_KCDC03_2020\\nEPI_ISL_412983_BetaCoV_Tianmen_HBCDC-HB-07_2020 EPI_ISL_410714_BetaCoV_Singapore_8_2020 EPI_ISL_403935_BetaCoV_Guangdong_20SF025_2020 EPI_ISL_406798_BetaCov_Wuhan_WH01_2019 EPI_ISL_411953_BetaCoV_Jiangsu_JS03_2020 EPI_ISL_412030_BetaCoV_Hong_Kong_VB20026565_2020 EPI_ISL_412968_BetaCoV_Japan_Hu_DP_Kng_19-020_2020 EPI_ISL_412973_BetaCoV_Italy_CDG1_2020 EPI_ISL_402132_BetaCoV_Wuhan_HBCDC-HB-01_2019\\nEPI_ISL_408486_BetaCoV_Jiangxi_IVDC-JX-002_2020 EPI_ISL_408430_BetaCoV_France_IDF0515_2020 EPI_ISL_410486_BetaCoV_France_RA739_2020 EPI_ISL_412966_BetaCoV_China_IQTC01_2020\\nEPI_ISL_412969_BetaCoV_Japan_Hu_DP_Kng_19-027_2020\\nEPI_ISL_411219_BetaCoV_France_IDF0386-islP1_2020\\nEPI_ISL_408481_BetaCoV_Chongqing_IVDC-CQ-001_2020 EPI_ISL_412975_BetaCoV_Australia_NSW05_2020 EPI_ISL_406534_BetaCoV_Foshan_20SF207_2020 . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint \\n</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Healthcare-resource-adjusted vulnerabilities towards the 2019-nCoV epidemic across China</td>\n",
       "      <td>[]</td>\n",
       "      <td>The 2019-nCoV epidemic has spread across China and 24 other countries 1-3 as of February 8, 2020 . The mass quarantine measure implemented in the epidemic's epicenter, Wuhan, Hubei Province, China, on January 23, 2020, has stopped the human migration between Wuhan and other places. However, according to the local authorities, over five million people had already traveled from Wuhan to other parts of China in the weeks leading to January 23, 2020, with the majority (79.96%) to the other cities within Hubei.\\nHealthcare resources have become a major bottleneck in the response to this public health crisis. The sudden onset of the epidemic and the sheer size of the infection have caused significant shortage in healthcare resources across Hubei Province. So far, there have been over 11,000 non-local healthcare professionals, mobilized by the public health authorities from other parts of China, fighting against the epidemic in Wuhan. However, this is far from adequate for Wuhan given the severity of the epidemic. In addition to Wuhan, other cities in Hubei are in need of urgent external help as well with respect to health resources, as multiple localized outbreaks are taking place. On February 7, 2020, the National Health Commission of China (NHCC) initiated the \"One City Covered by One Province\" program, explicitly assigning a specific province to aid one city in Hubei. Two critical questions of operational nature remain, which cities, both in Hubei and other provinces, are most vulnerable to localized outbreaks, and how the public health authorities should balance the healthcare resources under stress to best fight the epidemic and minimize the risks? This study is aimed to shed light on these questions by developing a vulnerability analysis framework integrating the human movement and healthcare resource data.\\nThis study makes use of four public datasets: (a) the daily counts of confirmed 2019-nCoV cases, collected from NHCC 4 ; (b) the list of designated hospitals with the number of reserved beds, retrieved from the websites of the provincial and municipal health commissions; (c) the information of the hospitals designated for 2019-nCoV, retrieved from 99.com.cn, a public hospital database; and (d) the human movement data, provided by the Baidu Migration (http://qianxi.baidu.com/), a public site offering real-time migration information across mainland China.\\nRecent studies indicated that the confirmed case data for Hubei Province, in particular, Wuhan, was subject to the under-reporting bias, while the data from places outside of Hubei is more reliable for inferring the actual epidemic size 5 . Following Wu et al. 5 , we developed a susceptible-exposed-infectious-recovered (SEIR)-based metapopulation model, which has 100 sub-populations representing the 100 cities that have the greatest number of people traveling from Wuhan. Note that we adopted the same epidemiological parameters as Wu et al. 5 . This model was then used to estimate IDi, the number of infected individuals in each city. Lastly, healthcare-resource-adjusted vulnerability indexes to the 2019-nCoV outbreak for city i, Vi were obtained as follows: Vi = IDi / Bi, where Bi represents the total number of reserved beds in the designated hospitals.\\nThe Figure presents the estimated number of infected individuals, as well as the healthcareresource-adjusted vulnerability to the 2019-nCoV outbreak across mainland China, as of January 29, 2020. The epidemiological parameters may have changed after January 23, 2020, when the lockdown measure was implemented. Thus, we only use the data by January 23, 2020 and produce the estimation till January 29, 2020 to accommodate the six-day incubation period identified by Wu et al. 5 .\\nThe ten most vulnerable non-Hubei cities are also listed in the figure. Note that the cities in Hubei (which are being aided by other provinces) are not presented here because of their significantly higher vulnerability as compared to others. Among the ten most vulnerable non-Hubei cities, Changsha, Beijing, Chongqing and Shanghai are major metropolises with a large number of estimated infected individuals. Xinyang, Jiujiang, Nanchang and Yiyang are medium-sized cities neighboring Hubei. In particular, Xinyang appears to be in a particular vulnerable situation because of the large number of people traveling from Wuhan and its insufficient healthcare resources.\\nThe reported vulnerability analysis informs public health response to the 2019-nCoV epidemic in multiple ways. For the identified cities with high vulnerability, it is important to increase the reserve of healthcare resources or have contingent plans. In particular, some of these cities already contributed significant healthcare resources to Hubei, leaving themselves inadequately prepared for the localized outbreak. Our analysis revealed that several medium-sized cities, particularly those neighboring Hubei, are highly vulnerable due to the lack of healthcare resources. \\n</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Relationship between Average Daily Temperature and Average Cumulative Daily Rate of Confirmed Cases of COVID-19</td>\n",
       "      <td>[{'text': 'The rapid outbreak of the new Coronavirus pandemic and the spread of the virus worldwide, especially in the Northern Hemisphere, have prompted various investigations about the impact of environmental factors on the rate of development of this epidemic. Different studies have called the attention to various parameters that may have influenced the spread of the virus, and in particular, the impact of climatic parameters has been emphasized. The main purpose of this study is to investigate the correlation between the average daily temperature and the rate of coronavirus epidemic growth in the infected regions. The main hypothesis object of our research is that between regions exhibiting a significant difference in the mean daily temperature, a significant difference is also observed in the average cumulative daily rate of confirmed cases, and that this does not happen if there is no significant difference in mean daily temperature. To test this research hypothesis, we carried on the case study of three regions in each of five countries and analyzed the correlation through F-test, and Independent-Samples T-Test. In all five selected countries, we found that when there is a significant difference in the daily mean temperature between two regions of a country, a significant difference exists also in the average cumulative daily rate of confirmed cases. Conversely, if there are no significant differences in the mean daily temperature of two regions in the same country, no significant difference is observed in the average cumulative daily rate of confirmed cases for these regions.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>The outbreak of infectious diseases has always been one of the most important health problems in the world. With the rapid spread of the new coronavirus (COVID- 19) , much attention has been paid to subjects like rate of an epidemic, transition ways, prevention methods, and remaining time of the virus in the environment [1] [2] [3] [4] [5] . Therefore, the behavior of the virus in relation to environmental factors is critical. The coronavirus is one of the epidemic diseases that many countries have experienced in the new type of it, COVID-19, and can become a serious challenge and affect the previous efforts on sustainable development [6] [7] [8] [9] .\\nIt has been found that there are some delays between the detection of confirmed cases and the actual infection date due to the laboratory test, the time of the announcement of confirmed cases in media, and the time for the patients showing initial symptoms for a total usually of 3 to 5 days [10] [11] [12] . The analysis of Kampf et al. (2020) demonstrated that COVID-19 could remain up to 9 days on different materials based on temperature and humidity [13] . Cascella et al. (2020) conducted research on the new coronavirus structure. Their analysis shows that the virus belongs to the family of single-stranded RNA viruses (+ ssRNA), which has a length of about 30 kb and an envelope with spear structures and they tested the sensitivity of the virus to UV and temperature [14] . investigated the correlations between environmental factors and the number of positive cases of COVID-19 by the use of artificial intelligence techniques. The results of this study put evidence on the role of urban and weather conditions on the epidemic rate [15] . Chen et al. (2020) developed a time-dependent mathematical model for the prediction of the total number of positive cases [16] . Lytle and Sagripanti (2005) studied the impact of UV rays, and their results determined that it can affect the structure of coronaviruses because UV rays could damage the nucleic acids of viral RNA that is vital for the survival of the virus [17] . developed an assessment method for investigating the impact of climate and urban factors in confirmed cases of COVID-19 [18] . Bukhari et al., (2020) , studied the epidemic of COVID-19 on tropical zones and their results show that the highest number of infection happened in temperature among 3 to 17 ºC [19] .\\nThe analysis of Grant and Giovannucci (2009) shows that by increasing the UV, the level of vitamin D in the body increases and makes the infection rates decrease. Therefore, in areas on lower latitudes and closer to the equator line, the epidemic on COVID-19 might be different [20] . In the study of Chan et al. (2011) , the effect of temperature and humidity on SARS coronavirus was analyzed, and they found that increasing temperature decreases the epidemic rate by faster disabling virus structure on contaminated surfaces [21] . Pirouz and Violini (2020) , analyzed the outbreak patterns of COVID-19 by analysis of 117 countries. They suggested two mechanisms of triggering the development of the infection, namely the normal spread of an infection from one parent case and the consequence of a unique phenomenon of accumulation of contagious people [22] . Analysis of the previous studies demonstrates the necessity of further research about the effect of environmental factors on the COVID-19. Since some of the previous researches have shown the effect of weather conditions, especially temperature, on coronavirus viruses, the main purpose of this study is to investigate the better correlation of daily mean air temperature on the epidemic rate of the COVID-19.\\nIn this study, we analysed the number of positive contagions of coronavirus COVID-19, from February 29 th to March 19 th . The selection of February 29 th is due to existence on this date of at least one positive case of COVID-19 in all case studies studied. The analysis of the weather data started from February15 th and ended on March 18 th because the impact of the weather parameter would occur in the upcoming days. Thus, the considered average daily temperature is with an interval of 1 to 14 days before the definitive cases of COVID-19.\\n In order to minimize the impact of other parameters on the results, such as the policy of each government, cultural differences between countries, health care systems, or the protocols in getting coronavirus test, comparative analysis have been done for different regions of the same country, without comparing with each other the results of different countries;  The analysis period for average daily temperature ranges from February15 th , 2020 to March18 th , 2020 (33 days);  The analysis period for confirmed cases of coronavirus COVID-19 is from February 29 th to March19 th ;  For what concerns the cumulative daily rate of confirmed cases, the total number of confirmed cases in a region was divided into the population of that region in order to take into account the different size of the population in each region.\\nTo carry out the analysis, the datasets of five countries and in each country, of three regions/provinces have been considered. The selected case studies are presented in Table 1 . and the location of the three regions shown in Figures 1. \\nMathematical models have been successfully used for the analysis of many hazards [32, 33] . In complex subjects, different Mathematical models can be used to solve the problem [34] [35] [36] . Statistical tests are an appropriate method for the investigation of possible correlations between different datasets [37] . Many statistical tests can be used for the analysis of significant differences between the datasets, among which F-test and Independent-Samples T-Test [38] .\\nIn order to investigate the hypothesis about a correlation between significant differences (Sig.) of average daily temperatures and average cumulative daily rate of confirmed cases in two regions, the following assumptions, and statistical tests have been done:\\n The variances of the two groups are equal: H0  The variances of the two groups are not equal: H1\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10.20059337 doi: medRxiv preprint When using F-test, if the significance level (sig) is less than 0.05, hypothesis H0 is rejected, and then hypothesis H1 is accepted, i.e., the variances of the datasets in the two selected regions are not equal. Conversely, if the significance level (sig.) is larger than 0.05, the hypothesis H1 is rejected, and the hypothesis H0 is accepted, i.e., the variances of the datasets in the two selected regions are equal.\\nThrough F-test, one determines the presence or absence of differences in the variances of the two regions. Through another statistical test, the Independent-Samples T-Test, one can determine whether the difference is significant, has been used. When using Independent-Samples T-Test, a significance level (sig.) lower (higher) than 0.05, means that the difference is (is not) significant.\\nIn our analysis, we used SPSS software. The critical factors in the SPSS Model set up are as follow:\\n The datasets in different regions of the same countries were analyzed with respect to the average daily air temperature during the 33-day period from 15-Feb to 18-Mar and the cumulative daily rate of confirmed cases of COVID-19 during the 19 days from 29-Feb to 19-Mar.  In all statistical tests, the confidence level of 95% is considered.  First, the F-test was used to test whether there is a significant equality of variances for each pair of case studies, and then, depending on the F-test result, by the T-Test the existence of a significant difference between the data of the two cases was investigated.\\nThe average daily temperature [39, 40] , and cumulative daily rate of confirmed cases [41] , are shown in Figures 2 to 6.\\na. Average daily temperature b. Cumulative daily rate of confirmed cases Correlations between the average temperatures and the cumulative daily rate of confirmed cases were found especially for towns with significant differences, with respect to the other two towns of the same country, in the average daily temperature. This was the case for Bandar Abas in Iran, Milan in Italy, Santa Cruz in Spain, and Los Angeles in the US. The average daily temperature and cumulative daily rate of confirmed cases are listed in Table 2 . As a general feature of Table 2 , we notice that  In 11 out of the possible fifteen comparisons, the average cumulative rate of confirmed cases is higher when the average daily temperature is lower, and in the four remaining cases the differences in either variable are small.\\nA more specific analysis per country shows that  In Iran, the average daily temperature in Bandar Abbas was the highest (21.54 ºC), and the average cumulative rate of daily of confirmed cases was the lowest (0.000027). . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10.20059337 doi: medRxiv preprint  In Italy, the average daily temperature in Milan was the lowest (7.99 ºC), and the average cumulative rate of daily of confirmed cases was the highest (0.000690).  In Germany, the average daily air temperatures of the selected towns have similar values, as well as the spread of the COVID-19 infection is similar.  In Spain, the average daily air temperature in Barcelona and Valencia are similar, and the spread of the infection is also similar. However, the comparison of Santa Cruz with higher average daily temperatures shows a slower spread of COVID-19.  In the United States, the average daily air temperature in both New York and Seattle are lower than Los Angeles, and the average cumulative rate of confirmed cases shows the spread of the virus in both is faster than in Los Angeles.\\nThese qualitative results reflect in the statistical analysis of the data by F-test and Independent-Samples T-Test presented in Table 3 . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10.20059337 doi: medRxiv preprint In Table 3 . t values or p-values can be used to check the existence of a significant correlation between the data for two cities.\\nLet us discuss in detail one such comparison, that between Ramsar and Tehran. From the F-test, the F value is equal to 20.4. The criterion for variance equality in the F-test is that the F value must smaller than the critical F value. The statistical analysis has been done on a statistical confidence level of 95%, and therefore the error level is 0.05, and considering the degrees of freedom, the critical F value of the test that can be found through F distribution tables [42] , in this case, is equal to F = 1.57. Since the F value we found is 20.4, larger than the critical F value, the assumption of the equality of variances is rejected. It means that the average temperature variance of the Ramsar and Tehran are not equal.\\nAccording to the t-test, the t value for these two cities is equal to -1.929. Based on an error level of 0.05 and by considering the degrees of freedom, the critical values of the test indicated in the T distribution tables [43] , is in this case equal to t = ± 1.645. Since the t value is out of the critical boundary (-1.645 ≤ acceptable t ≤ +1.645), the assumption of equality in the averages is rejected, and the existence of difference in averages is confirmed. Thus from two tests, it is evident that there is a significant difference between the average temperature of the two cities of Ramsar and Tehran.\\nAn analogous result holds for the average cumulative rate of confirmed cases.\\nThe main results of Table 4 , with a confidence level of 95% are as follow:\\n In Iran, the city of Bandar Abbas is significantly different from Ramsar and Tehran in terms of both average daily temperature and average cumulative rate of confirmed cases.  In Italy, the city of Milan is significantly different from Rome and Naples in terms of both average daily temperature and average cumulative rate of confirmed cases. Instead, the two cities of Rome and Napoli are not significantly different in terms of these variables.  In Germany, there are no significant differences in average daily temperature in all three cities, and the average cumulative rate of confirmed cases is not significantly different.  In Spain, Barcelona and Valencia are not significantly different in terms of average daily temperature, and the average cumulative rate of confirmed cases is not significantly different. However, in Barcelona and Santa Cruz, the average daily temperature is significantly different, as well as the average cumulative rate of confirmed cases, and the same trend exists between Valencia and Santa Cruz.  In the United States, the two cities of New York and Seattle are not significantly different in terms of average daily temperature, and of the average cumulative rate of confirmed cases, the number of patients is not significantly different. Instead, the city of Los Angeles is significantly different in terms of average daily temperature with two cities of New York and Seattle, and the average cumulative rate of confirmed cases are different significantly.\\nThe results of our analysis of daily air temperature for three case studies from five countries showed the existence of correlations between the average temperatures and the epidemic rate, and this is especially evident when the differences in average daily temperature are larger. For example, in Iran and Italy, the regions with lower average daily temperature had a higher epidemic level of the new coronavirus (COVID- 19) , and the same trend exists in all other case studies.\\nThe statistical analysis showed that in all the case studies we considered, referring to five different countries, if there is a significant difference in daily mean temperature, there is also a significant difference in the average cumulative daily rate of confirmed cases, supporting the proposed research hypothesis about the correlation between air temperature and the epidemic rates of COVID-19. Due to the correlations among temperature and hours of sunshine for future research, it is recommended to study the effects of the sun's UV rays as well as the average hours of sunshine on the epidemic rates of COVID-19.\\nThis study analyzed the possible correlation of daily average air temperature with the epidemic rate of the coronavirus (COVID-19) using significant differences analysis between the datasets. The research hypothesis is that in regions with a significant difference in the average daily temperature, the average cumulative daily rate of confirmed cases would also is different significantly. Conversely, to the absence of a significant difference in daily temperature, would correspond no significant difference in the rate of confirmed cases. The hypothesis was investigated through statistical analysis. In order to minimize the impact on the results of factors like the policy of the government or cultural differences among countries (food, exercise, weight, etc.), three case studies within five countries, namely Iran, Italy, Germany, Spain, and United States were compared separately. This comparison using statistical analysis through F-test and Independent-Samples T-Test of the daily temperatures and daily rate of confirmed cases shows that there is a correlation between the average temperature and the epidemic rate, and this is especially evident when differences in average daily temperature are significantly larger, as it happens for Bandar Abbas in Iran, Milan in Italy, Santa Cruz in Spain, and Los Angeles in the US. Besides, the analysis of the average air temperatures shows that the epidemic rates of COVID-19 were higher in the case studies with a lower average temperature. Instead, when no significant differences exist in the average daily temperature of two cities in the same country, there is no significant difference in the average cumulative daily rate of confirmed cases. In conclusion, the results of this study support the research hypothesis and confirm the effectiveness of the proposed method for analysis of the epidemic rates.\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10 . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10.20059337 doi: medRxiv preprint \\n</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>CHEER: hierarCHical taxonomic classification for viral mEtagEnomic data via deep leaRning</td>\n",
       "      <td>[{'text': 'The fast accumulation of viral metagenomic data has contributed significantly to new RNA virus discovery. However, the short read size, complex composition, and large data size can all make taxonomic analysis difficult. In particular, commonly used alignment-based methods are not ideal choices for detecting new viral species. In this work, we present a novel hierarchical classification model named CHEER, which can conduct read-level taxonomic classification from order to genus for new species. By combining k-mer embedding-based encoding, hierarchically organized CNNs, and carefully trained rejection layer, CHEER is able to assign correct taxonomic labels for reads from new species. We tested CHEER on both simulated and real sequencing data. The results show that CHEER can achieve higher accuracy than popular alignmentbased and alignment-free taxonomic assignment tools. The source code, scripts, and pre-trained parameters for CHEER are available via GitHub: https://github.com/KennthShang/CHEER.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>Metagenomic sequencing, which allows us to directly obtain total genomic DNAs from host-associated and environmental samples, has led to important findings in many areas, such as digestive health [1] . While bacteria are the main focus of most metagenomic sequencing projects, there are fast accumulation of viral metagenomic data with sequencing viruses as the main purpose [2] . There are different types of viruses. We are mainly concerned with RNA viruses because many RNA viruses are notorious human pathogens, such as Influenza A, Human immunodeficiency virus (HIV), Ebola, SARS-CoV, and recently identified 2019-nCoV causing Wuhan pneumonia. Unlike DNA viruses, RNA viruses, which contain RNA genomes, lack faithful proofreading mechanisms during replication and thus can produce a group of related but different viral strains infecting the same host. This high genetic diversity within and across different hosts poses a great challenge for designing long-term protection strategies against these infectious diseases. For example, as the circulating strains can change every year, flu vaccine has to be administered every year.\\nAdvances in viral metagenomics have contributed significantly to new RNA virus discovery. According to a survey by Woolhouse et. al, the number of newly identified RNA viruses is changing from 1,899 in 2005 to 5,561 in 2018, which is 3 times of increase [3] . Largescale RNA virus sequencing projects using nextgeneration sequencing technologies have been conducted for different species. For example, Shi et al. have discovered a large number of new RNA viruses by sequencing samples from invertebrate and vertebrate animals [4, 5] . Claire et al. have discovered 25 new RNA viruses by sequencing samples associated with different drosophilid [6] . Bolduc et al. sequenced RNA viruses from Archaeal and bacterial samples [7] . Given fast accumulation of viral metagenomic data and expected discovery of new viruses, a key step is to conduct composition analysis for these data and assign taxonomic groups for possibly new species.\\nComposition analysis can be conducted at the read level or contig level. There is no doubt that contigs can provide more information than short reads in taxonomic classification. However, metagenomic assembly is still one of the most challenging computational problems in bioinformatics. Unlike single genome assembly, metagenomic assembly is more likely to produce chimeric contigs that can contain regions from different species. Thus, metagenomic assembly is often preceded by read binning that groups reads with the same or similar taxonomic labels [8] in order to achieve better performance.\\nIn this work, we design a read level taxonomic classification tool (named as \"CHEER\") for labeling new species in viral metagenomic data. To detect known species with available reference genomes, alignment-based methods can provide sufficient information for both classification and abundance estimation. This function has been incorporated in existing pipelines [9, 10, 11] and is thus not the focus of our tool. Instead, our tool is designed to handle the challenging cases of assigning taxonomic labels for reads of new species, which have not been observed before. This problem was clearly formulated in metagenomic phylogenetic classification tools such as Phymm and PhymmBL [8] . We first define the classification problem following the formulation in Phymm: the classification problem is the assignment of a phylogenetic group label to each read in input data sets. As we are only interested in the hard case of classifying reads of new species, which have not been sequenced and thus no label exists for these species, the expected labels for these reads are the higher-rank taxonomic groups such as genus, family, or order. All our data and experiments are designed so that the test species are masked in the reference database and the trained models.\\nPrevious experiments have shown that nucleotide-level homology search is not sensitive enough for assigning higher rank taxonomic groups to reads from new species [9, 10] . Protein-based homology search is thus preferred for phylogenetic classification [11, 12] . In particular, a recent tool, VirusSeeker [13] utilized BLASTx to classify reads into bacteria, phages, and other viruses. It is not clear whether the protein homology search is able to generate accurate classification at lower ranks. We will evaluate this in our experiments.\\nThere are also alignment-free methods for phylogenetic classification. Many of those tools implemented different machine learning models and were mainly designed for bacteria. For example, RDP [14] adopted a Naïve Bayes based classifier that can automatically learn features from the reads and assigned taxonomic labels to sequences from bacterial and archaeal 16S genes and fungal 28S genes. The NBC classifier [15, 16] encoded metagenomic reads using k-mers and implemented a bag-of-word model to achieve the readlevel taxonomic classification. Phymm [8] utilized interpolated Markov models (IMMs) for metagenomic read classification. These learning-based models enable users to label reads originating from new species.\\nThe previous works have shown that sequence composition is still an important and informative feature for phylogenetic classification. Recent applications of deep learning models have achieved better performance than conventional machine learning methods in different sequence classification problems [17] . In particular, the convolution filters in Convolutional Neural Network models (CNN) can represent degenerated sequence motifs and thus help CNN models to learn abstract composition-based features. For example, frequently activated convolution filters learned by DeepFam often represented the conserved motifs in different protein domain families [18] . The successful applications of deep learning in sequence classification motivated us to design a novel deep learning-based classification model for assigning taxonomic groups for new species in viral metagenomic data.\\nOur main contributions are summarized below. First, our deep learning model can assign higher rank phylogenetic group labels (such as genus) for reads from new species. To achieve this goal, our model organizes multiple CNN-based classifiers in a tree for hierarchical taxonomic group assignment. Second, to enable our model to capture as much information as possible from short reads, we implemented and compared two encoding methods: one hot vs. embedding. Third, as viral metagenomic data usually contains contaminations from either the host genomes or other microbes, we formulated the pre-processing step as an open set problem in targeted image classification and rejected non-viral reads by choosing appropriate negative training sets. We tested our model on both simulated metagenomic data and also real sequencing data. The results show that our model competes favorably with other popular methods.\\nIn this section, we will introduce our method for viral metagenomic taxonomic classification. First, we will show the architecture of CHEER, which is a hierarchical classification model from order to genus. Second, classifiers at each level will be described. Third, we will introduce DNA sequence encoding using skipgram based word embedding and compare it with onehot encoding. Forth, we will describe the rejection layer, following the idea of the open set problem, which is adopted to filter reads not belonging to RNA viruses. Finally, we will detail our dataset used in training and validation.\\nThe architecture of our model is sketched in Fig. 1. The key component is a tree model that consists of multiple classifiers from order to genus. In order to conduct phylogenetic classification for reads from new species, or even new genus, our classification is conducted using a top-down approach from the root to the leaf node. The top layer is a trained CNN that can reject reads not belonging to RNA viruses, which is an important preprocessing step for viral metagenomic data because of the contamination from the host genome or other microbes. Then, the filtered reads (i.e. mostly RNA viral reads) will be fed into the hierarchical classification model, which is referred to as the tree model hereafter. Specifically, the order-level classifier is responsible for classifying the reads into their originating orders. Then each order has a separately trained classifier for assigning input reads to families within that order (O1 to O3 in Fig. 1 ). For all the reads assigned into one family, the family classifier (F1-F7) will assign the reads to different genus within that family. As an example, a path for classifying a read from a species in genus 7 is highlighted in Fig. 1 . CHEER also implemented early stop functions at each level for stopping the classification path at higher ranks. This function can accommodate taxonomic classification for species from new genus or even higher ranks, which is possible for RNA viruses. It is convenient to add fine-grained classifiers within each genus so that we can assign species-level labels. However, as our focus is to conduct phylogenetic classification for reads from new species that do not have a species labels in the training data, we did not include the species-level assignment in this work.\\nAlternative designs of hierarchical classification work, like [19, 20] , either build a classifier for each taxonomic rank or build a binary classifier for each class. We implemented a structure that built just one classifier for each taxonomic rank and compared different structures' performance. According to the result shown in [Supplementary file 1], the structure shown in Fig.1 is better than one classifier per rank. Although one classifier per rank requires fewer classifiers, it needs to train many more parameters because the number of labels for each classifier increases by times. As a result, we observe overfitting more frequently.\\nEach classifier in the tree model is implemented using CNN, which has achieved superior performance in various sequence classification problems [18] . The convolution filters used in CNN resemble positionspecific weight matrix of motifs in genomic sequences. Frequently activated convolution filters can represent well-conserved motifs among sequences in the same class. Thus, a large number of convolution filters will be used in our CNN in order to learn well-conserved sequence features in different classes. Deep CNNs in fields such as computer vision often have multiple convolution layers in order to obtain more abstract features from training images. Here, we use a wide convolution design rather than a deep one, following the idea in [18] . Fig. 2 shows the classifier with the default hyper-parameters. First, reads will be encoded in matrices as input to the convolution layer using two methods. We will describe and compare these two encoding methods later in the Section 2.3. Then, multiple convolution filters of different sizes are utilized to learn the conserved sequence features. According to our empirical study, we choose 3, 7, 11, 15 as the filter sizes, with 256 filters for each size respectively (see Fig. 2 ). Max pooling is then applied to each convolution filter's output so that only the highest convolution value is kept. After the max-pooling layer, we have a 256-dimentional vector as the output of each convolution layer. We concatenate these four vectors into one 1024-dimentional vector and feed it to one fully connected layer with 512 hidden units. Dropout is applied to mitigate overfitting. The CrossEntropyLoss function supplied by Pytorch is adopted as an error estimation function. Due to the unbalanced size of each training class, we use the reciprocal of the proportion of each class as their error weight to help the model fairly learn from each class. And we utilize Adam optimizer with a 0.001 learning rate to update the parameters. The model is trained on HPCC with the 2080Ti GPU unit. \\nIn previous deep learning-based sequence classification models [21, 22] , one hot encoding was commonly adopted to convert a sequence into a matrix, which will then be used as input to convolution layers. There also exist some methods using k-mer composition [23] and frequency [15, 16] to encode the DNA sequence, similar to the bag-of-words model in the field of natural language processing. However, these k-mers representation cannot give complete sequential information about the original reads.\\nTo incorporate both the k-mer composition and also their ordering information, we build a Skip-Gram [24] based embedding layer that can learn which k-mers tend to occur close to each other. Embedding is widely used in natural language processing to learn the semantic and syntactic relationships from sentences. A neural network with one hidden layer is trained to map a word to an n-dimensional vector so that the words that usually occur together will be closer in the n-dimension space. For our DNA read classification task, k-mers are the words and the embedding layer will map proximate kmers into vectors of high similarity. The main hyper-parameter when training the skip-gram model is the size of the training context m, which defines the maximum context location at which the furthest k-mer is taken for training. Specifically, for a kmer at position i as the input to the skip-gram model, the output k-mers are its neighbors located at i+jk, where −m ≤ ≤ m. Larger m results in more training samples with a cost of training time. In our model, the default value of m is 1. As shown in Fig. 3 , we will sample 3 k-mers as our dataset at a time with the middle one as the input and the surrounding two as the output.\\nAs proven in [24] , Skip-Gram is able to automatically learn the relationship between proximate words and thus is expected to help learn the order between k-mers in genomic data. We implemented a hidden layer with 100 hidden units shown in Fig. 3 . After training the model, the co-occurrence features are embedded in the output of the hidden layer. Thus, a k-mer in a read will be represented as a 100-dimension vector, where each element in the vector is the output of the hidden unit in the hidden layer.\\nWhen we use this pre-trained embedding layer to encode the reads, we first split each read into sequential k-mers. Second, each k-mer is fed to the embedding layer and the output is a 100-dimension vector. Third, all these vectors are combined into a large matrix according to the original position of each k-mers. Finally, this matrix will be fed to the convolutional layer.\\nBased on our large-scale experiments, we found that classifiers are easy to overfit with increase of k. Thus, we choose 100-dimensional feature vectors with k being 3 as our default embedding layer parameter set based on our empirical analysis.\\nDue to the small genome size and limitations in library construction, it is not rare that many viral metagenomic datasets are still contaminated with the genomes of the host and other microbes. Viral read screening is thus a key pre-processing step before the downstream composition analysis. Specifically, our tree model should only be applied to viral reads. However, the close set nature of the classifier forces it to assign a known label to each input without being able to distinguish viral reads from others. In this work, we will reject non-viral reads by formulating it as an open set problem in targeted image classification [25] .\\nIn a majority of neural network models [26, 27] , the logit of the last layer will be fed into the SoftMax function, which takes a vector of n real numbers as input and output a normalized probability vector for all labels/classes. However, for an irrelevant input from an unknown clade, all classes in the model tend to have low probabilities and thus applying a threshold on uncertainty can be used to reject unknown classes [25] . Along with this idea, serval approaches [28, 29] have been proposed to solve the open set problem.\\nIn this work, we incorporate the pre-processing step in our hierarchical classification model by integrating a trained CNN and a SoftMax threshold. Contaminations such as bacteria will be rejected using empirically chosen SoftMax threshold τ. Our experiments show that SoftMax outputs a nearly uniform probability distribution of all labels if the input is not relevant. Thus, the SoftMax threshold τ is used to reject reads not belonging to either DNA or RNA viruses. However, using this thresholding method is not able to distinguish DNA viral reads from RNA viral reads due to their similar sequence composition. To accommodate this challenge, we designed a rejection classifier in the top layer using DNA viral reads as the negative class in the training (see Fig. 1 ).\\nAs the focus of our work is to assign labels for new species, the hierarchical classification ends at genuslevel assignment, which is conducted by the bottom layer in Fig. 1 . Given the fast accumulation of viral metagenomic data, there are cases where a new species does not belong to any existing genus and thus the label assignment should stop at the higher ranks. Thus, we allow our model to stop at a higher rank in the tree during the top-down classification. Towards this goal, we still apply SoftMax threshold in our early stop function, which stops reads from further classification by the child classifiers. This early stop function also serves as a quality control measure for preventing wrong classifications if the confidence is low. For example, when a read arrives, the model may be able to identify its order and family labels with high reliability but cannot decide which genus it belongs to. In this case, the model should only assign order and family label to this read rather than classifying it into a potentially wrong genus. To achieve this goal, SoftMax threshold is applied for every classifier in the tree. Only reads with a score higher than the threshold will be assigned to the child class.\\nAn example of early stop is highlighted by the yellow path in Fig. 1 . The model can classify the read into order 2 (O2) and then family 4 (F4) with high reliability but cannot decide which genus it belongs to.\\nAll the viruses used in the experiments were obtained from the RefSeq in the NCBI virus database. The The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.26.009001 doi: bioRxiv preprint viruses' taxonomic classification was downloaded from Virus Taxonomy 2018b Release [30] published by the International Committee on Taxonomic of Virus (ICVT). Because training deep learning classifiers required enough samples, we removed the orders with only one family, families with only one genus, and genera with less than 3 species from our dataset. Finally, there are 6 orders, 23 families, and 55 genera remained. The RefSeq genomes are available at the following URL:https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/fi nd-data/virus. The information of the taxa can be found in [Supplementary file 1]. As CHEER conducts read-level taxonomic classification, the training is also conducted using reads. In order to train our model on each complete virus genome, we generated simulated reads by uniformly extracting substrings of 250 bp with 200bp overlaps. All these reads formed the training set so that the model can learn the features from the whole genome. The validation set, however, is generated by read simulation tool WgSim [31] with specified error rate so that we can have a better estimation of the model's performance on real sequencing data. Some existing deep learning tools for taxonomic classification, like [21, 23] , generate reads first, then randomly choose some of them as training set and others as testing set. This strategy cannot guarantee that the training and testing data sets have no overlaps from the same species. In order to test whether our model can learn the label of new species, we use test species masking in the training set. Specifically, we split our RNA virus database into two independent sets. For any test species, their reads are not used in the training data. Thus, the model will only use the knowledge of known species for the new species classification.\\nIn this section, we will first give details of the hyperparameters of CHEER. Then we present the results of applying CHEER to both simulated and mock sequencing data. For the simulated data by WgSim, we evaluated how sequencing error rate affects the performance of CHEER and compared the taxonomic classification results with state-of-the-art. For the mock data, we tested CHEER's performance in both taxonomic classification and its ability in distinguishing viral reads from others.\\nIn our experiments, the minimum number of species within a genus is 3. We thus constructed three test sets, each containing reads from one species that is randomly chosen from each genus. The three test sets contain three different species, respectively. In total, 165 species from 55 genera are chosen to be \"new\" species for testing and the information of these species can be found in [Supplementary file 1].\\nWe used WgSim to simulate reads from testing species. In order to evaluate the impact of sequencing error on CHEER, we simulated reads with an error rate of 0, 0.01, 0.02, and 0.05, respectively. For each classifier (CNN), we computed the classification accuracy, which is defined in equation (1). To ensure that the computation of the accuracy is not affected by the number of reads from different species, we constructed a balanced test read set by simulating the same number of reads for each test species (1,000 reads for each). (1) For each component classifier in the tree model, we record the average accuracy for multiple test species. For each taxa level, we calculate the average accuracy of all the classifiers in the same level.\\nWe present the results of CHEER at each level in this section. Because CHEER is designed for new species, we benchmark it against alignment-free models. Based on the results shown by Wood et al. [7] , Naïve Bayes Classifier (NBC) outperformed other alignment-free models. Thus, we compared the performance of CHEER and NBC. NBC in [14] was designed for bacteria taxonomic classification and thus, we retrained NBC on our RNA virus dataset. We implemented the NBC model following [14] with the same parameters. Since each family has a separately trained classifier, which conducts genus classification within that family, there are totally 14 classifiers at this level. Fig. 5 shows the accuracy for labeling families. There are totally 6 classifiers that assign reads to families within each of the 6 orders. Fig. 6 shows the mean accuracy at each rank. The family and genus level accuracy are the average accuracy of all classifiers in Fig. 4 and Fig. 5 . There is only one classifier for order level classification.\\nAs shown in Fig. 4, Fig. 5, and Fig. 6 , CHEER has better performance than NBC across different ranks. We also compared the performance of using one-hot encoding and embedding layer in CHEER. The performance comparison at each rank shows that for most of the classifiers, the embedding layer improves the learning ability. Fig. 4, Fig. 5, and Fig. 6 also revealed the robustness of CHEER. Although the classification accuracy decreases as the error rate increases, the decrease is smaller than 2%. The accuracy is almost identical for many classifiers. Thus, CHEER is not very sensitive to sequencing errors. We also conducted experiments for reads with higher error rates (0.02 and 0.05). The results can be found in [Supplementary file 1].\\nThe next experiment is to evaluate our model on a real sequencing dataset and test CHEER's performance in both taxonomic classification and non-viral read rejection. We built a mock dataset by mixing Ebolavirus reads with gut bacterial reads from a gut amplicon sequencing dataset, which is chosen to ensure that we only use reads from bacteria. The taxonomic classification of Zaire Ebolavirus in the tree model is highlighted in Fig. 7 . We chose one of the Ebola species named Zaire Ebolavirus as our testing species because it is the most dangerous of the five Ebolavirus species and was responsible for the 2014 West African Ebola outbreak. We downloaded the dataset from NCBI with accession number SRR1930021. These reads were sequenced by Illumina using whole-genome amplification strategy. Since Zaire Ebolavirus usually causes fatal hemorrhagic fever and abdominal pain, it is possible that this Ebola virus occurs with other gut bacteria. Thus, we created a mock viral metagenomic data by mixing these Ebola reads with a gut bacteria amplicon sequencing data in order to test whether CHEER can reject the bacterial reads at the top rejection layer. The gut bacteria dataset is downloaded from NCBI with accession number: SRR10714026. There are totally 2,737 Ebola virus reads and 51,280 gut bacterial reads in the mock data. After the rejection layer, the Ebola reads will be used to evaluate our tree model.\\nTo treat Zaire Ebolavirus as a \"new\" species, we removed its genome from our training set and re-trained the whole pipeline. As shown in Fig. 7 , after removing this species, we still have four Ebola species in our training set. However, our read mapping experiment using Bowtie2 [32] showed that none of the reads from Zaire Ebolavirus can be mapped to other Ebola species with the default parameters, indicating that using usual alignment-based methods will have difficulty to conduct genus-level assignment.\\nIn our work, CHEER will first use rejection layer to remove the reads not belonging to the RNA virus. Second, the \"Order classifier\" will be applied to evaluate how many reads can be classified into Mononegavirales order. Third, \"Mononegavirales classifier\" takes all these Mononegavirales reads as input to predict how many reads will be assigned to the Filoviridae family. Finally, \"Filoviridae classifier\" is adopted to check how many reads belong to the Ebola genus. Fig. 8 show the ROC curve of the rejection layer. True positive rate reveals how many Ebolavirus reads pass the rejection layer. The false positive rate reveals how many gut bacterial reads pass the rejection layer. We generated thresholds from 0 to 1 with step size 0.01 and used each threshold for the rejection layer. Then for each threshold, we recorded the true positive rate and false positive rate for plotting the ROC curve. As shown in Fig. 8 , with the increase of the threshold, more bacterial reads are filtered by the rejection layer, at the expense of missing more real Ebola reads. The default threshold for the rejection layer is 0.6 in CHEER. User can adjust the threshold according to their needs.\\nAfter evaluating the rejection layer, we continued to test the hierarchical classification performance at each rank. We first compared our tree model with NBC. Then we also benchmarked CHEER with the method using protein-level alignment in Section 3.3.3. The performance of the complete hierarchical classification pipeline of NBC and CHEER is shown in Fig. 9 . From order to genus, each bar shows the accuracy of correctly classified reads by the parent classifier. For example, NBC correctly classified 1580 at the order level. Then, 1564 out of the 1580 reads are correctly labeled at the family level. While Fig. 4 , 5, and 6 have shown the performance of CHEER and NBC at each taxonomic rank using the same set of reads as input for each level of classifiers, this experiment focused on comparing the hierarchical classification pipeline using only correctly predicted reads by the parent classifier. The results demonstrated that CHEER can classify Ebola reads with better performance. NBC can only classify 1,564 (57.1%) reads into Ebolavirus. CHEER, however, had totally 2407 (87.9%) reads classified correctly.\\nAlthough nucleotide-level alignment models are not ideal choices for predicting new species, as mentioned in [8] , protein-level alignment such as BLASTx is still frequently used for homology search because of the conservation among homologous protein sequences. We followed VirusSeeker [10] , which is a BLASTx based model, to align translated reads to virus-only protein database (VIRUSDB NR) provided by NCBI. Then, we kept alignments with E-value cutoff 0.001, which is used in VirusSeeker. The output of BLASTx shows that a read can often align to multiple proteins, making the phylogenetic classification ambiguous (see Fig. 11 ). Also, existing tools such as VirusSeeker often use the best alignment only, which does not always yield the highest accuracy. To optimize the usage of the alignment-based method, we implemented k-Nearest Neighbor (kNN) algorithm to record the best k alignments rather than only using the best one. The label of the input read is determined by the majority vote in the labels of the top k alignments. According to Fig. 10 , most of the reads have hit numbers between 11 to 15 and 6 to10. Thus, we chose 1, 5, 10, 15 as our k when running kNN. There are also 512 reads that cannot be aligned with any VIRUSDB protein sequence. All these reads will be regarded as wrong prediction. Fig. 11 shows the classification results of kNN. When we use the best hit as the prediction the accuracy is only 64%. The best performance of kNN will be 74% with 2,026 correct predictions when k is 5. But the accuracy sharply decreases if k is larger than 5. Fig. 10 and Fig. 11 also revealed that it is hard for users to choose a proper k in real classification tasks. The most common value of the hit number is 11-15 shown in Fig. 11 . The best kNN result, however, has a k equal to 5. Thus, comparing to this alignment-based method, CHEER, with 2,407 (87.9%) reads classified correctly, is more practical and reliable.\\nThe two main purposes of the early stop function are: 1) to handle the cases of a new species from an unseen genus or higher ranks; 2) to serve as a quality control measure for preventing wrong classifications in child classifiers. As Fig. 6 shows, the higher rank classification tends to be more accurate than lower ranks. Because the genus of Ebolavirus exists in our classification tree, the evaluation of the early stop function focuses on the second purpose. Fig. 12 shows the percentage of how many reads are classified into Ebolavirus genus correctly for different classifiers. By allowing early stop, the number of wrong predictions decreased sharply. In short, although there are fewer reads being labelled at the genus level, the total number of reads with wrong classification is decreased. Thus, this is still a useful feature provided by CHEER.\\nAnother significant advantage of CHEER is the high classification speed. Although training the model uses heavy computing resources, the inference procedure of this algorithm is fast. And with the help of GPU acceleration, the training time also decreases a lot.\\nIn the experiments shown in Section 3.4, it takes around 46 minutes to run BLASTx against the virus protein database. Our method, however, only takes 7 second for the preprocessing step to convert the raw sequence into a vector, and then takes less than 35 second to make the prediction for all 2,737 reads.\\nIn this work, all training data are RNA virus genomes downloaded from the NCBI RefSeq database. The training reads are extracted uniformly from the genome to make sure that classifiers can learn the features from the whole genome. To evaluate the model on new species detection, all species in the test set are excluded from the training data. Error rates are also added to evaluate the robustness of the model. The results reveal that the model is tolerant to the sequencing error. In addition, a SoftMax threshold rejection layer is applied to filter reads from non-RNA viruses. The SoftMax threshold in each classifier will also help early stop classification with low confidence and thus decreases the wrong predication rate. We show a case study on real-world data to identify how this threshold will influence the rejection classifier and early stop function. The results show that CHEER competes favorably with state-of-the-art tools.\\n</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                  title  \\\n",
       "0  The RNA pseudoknots in foot-and-mouth disease virus are dispensable for genome replication but essential for the production of infectious virus. 2 3   \n",
       "1  Analysis Title: Regaining perspective on SARS-CoV-2 molecular tracing and its implications                                                             \n",
       "2  Healthcare-resource-adjusted vulnerabilities towards the 2019-nCoV epidemic across China                                                               \n",
       "3  Relationship between Average Daily Temperature and Average Cumulative Daily Rate of Confirmed Cases of COVID-19                                        \n",
       "4  CHEER: hierarCHical taxonomic classification for viral mEtagEnomic data via deep leaRning                                                              \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      abstract  \\\n",
       "0  [{'text': 'word count: 194 22 Text word count: 5168 23 24 25 author/funder. All rights reserved. No reuse allowed without permission. Abstract 27 The positive stranded RNA genomes of picornaviruses comprise a single large open reading 28 frame flanked by 5′ and 3′ untranslated regions (UTRs). Foot-and-mouth disease virus (FMDV) 29 has an unusually large 5′ UTR (1.3 kb) containing five structural domains. These include the 30 internal ribosome entry site (IRES), which facilitates initiation of translation, and the cis-acting 31 replication element (cre). Less well characterised structures are a 5′ terminal 360 nucleotide 32 stem-loop, a variable length poly-C-tract of approximately 100-200 nucleotides and a series of 33 two to four tandemly repeated pseudoknots (PKs). We investigated the structures of the PKs 34 by selective 2′ hydroxyl acetylation analysed by primer extension (SHAPE) analysis and 35 determined their contribution to genome replication by mutation and deletion experiments. 36 SHAPE and mutation experiments confirmed the importance of the previously predicted PK 37 structures for their function. Deletion experiments showed that although PKs are not essential 38', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'for replication, they provide genomes with a competitive advantage. However, although 39 replicons and full-length genomes lacking all PKs were replication competent, no infectious 40 virus was rescued from genomes containing less than one PK copy. This is consistent with our 41 earlier report describing the presence of putative packaging signals in the PK region. 42 43 author/funder. All rights reserved. No reuse allowed without permission.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]   \n",
       "1  [{'text': 'During the past three months, a new coronavirus (SARS-CoV-2) epidemic has been growing exponentially, affecting over 100 thousand people worldwide, and causing enormous distress to economies and societies of affected countries. A plethora of analyses based on viral sequences has already been published, in scientific journals as well as through non-peer reviewed channels, to investigate SARS-CoV-2 genetic heterogeneity and spatiotemporal dissemination. We examined all full genome sequences currently available to assess the presence of sufficient information for reliable phylogenetic and phylogeographic studies. Our analysis clearly shows severe limitations in the present data, in light of which any finding should be considered, at the very best, preliminary and hypothesis-generating. Hence the need for avoiding stigmatization based on partial information, and for continuing concerted efforts to increase number and quality of the sequences required for robust tracing of the epidemic.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': '. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'is the (which was not peer-reviewed) The copyright holder for this preprint .', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]                                                                                                                                                                                                                                                                                                                                         \n",
       "2  []                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
       "3  [{'text': 'The rapid outbreak of the new Coronavirus pandemic and the spread of the virus worldwide, especially in the Northern Hemisphere, have prompted various investigations about the impact of environmental factors on the rate of development of this epidemic. Different studies have called the attention to various parameters that may have influenced the spread of the virus, and in particular, the impact of climatic parameters has been emphasized. The main purpose of this study is to investigate the correlation between the average daily temperature and the rate of coronavirus epidemic growth in the infected regions. The main hypothesis object of our research is that between regions exhibiting a significant difference in the mean daily temperature, a significant difference is also observed in the average cumulative daily rate of confirmed cases, and that this does not happen if there is no significant difference in mean daily temperature. To test this research hypothesis, we carried on the case study of three regions in each of five countries and analyzed the correlation through F-test, and Independent-Samples T-Test. In all five selected countries, we found that when there is a significant difference in the daily mean temperature between two regions of a country, a significant difference exists also in the average cumulative daily rate of confirmed cases. Conversely, if there are no significant differences in the mean daily temperature of two regions in the same country, no significant difference is observed in the average cumulative daily rate of confirmed cases for these regions.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]                                                                                                       \n",
       "4  [{'text': 'The fast accumulation of viral metagenomic data has contributed significantly to new RNA virus discovery. However, the short read size, complex composition, and large data size can all make taxonomic analysis difficult. In particular, commonly used alignment-based methods are not ideal choices for detecting new viral species. In this work, we present a novel hierarchical classification model named CHEER, which can conduct read-level taxonomic classification from order to genus for new species. By combining k-mer embedding-based encoding, hierarchically organized CNNs, and carefully trained rejection layer, CHEER is able to assign correct taxonomic labels for reads from new species. We tested CHEER on both simulated and real sequencing data. The results show that CHEER can achieve higher accuracy than popular alignmentbased and alignment-free taxonomic assignment tools. The source code, scripts, and pre-trained parameters for CHEER are available via GitHub: https://github.com/KennthShang/CHEER.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               full_text  \\\n",
       "0  VP3, and VP0 (which is further processed to VP2 and VP4 during virus assembly) (6). The P2 64 and P3 regions encode the non-structural proteins 2B and 2C and 3A, 3B (1-3) (VPg), 3C pro and 4 structural protein-coding region is replaced by reporter genes, allow the study of genome 68 replication without the requirement for high containment (9, 10) ( figure 1A ).\\nThe FMDV 5′ UTR is the largest known picornavirus UTR, comprising approximately 1300 71 nucleotides and containing several highly structured regions. The first 360 nucleotides at the 5′ 72 end are predicted to fold into a single large stem loop termed the S-fragment, followed by a The PKs were originally predicted in 1987 and consist of two to four tandem repeats of a ~48 86 nucleotide region containing a small stem loop and downstream interaction site (figure 1B) 87 (12). Due to the sequence similarity between the PKs (figure 1C), it is speculated that they 88 were formed by duplication events during viral replication, probably involving recombination. 89 Between two and four PKs are present in different virus isolates but no strain has been 90 identified with less than two PKs, emphasising their potential importance in the viral life cycle 91 (19, 20) . The presence of PKs has been reported in the 5′ UTR of other picornaviruses such as 92 author/funder. All rights reserved. No reuse allowed without permission. can occur in the absence of PKs at least one is required for wild-type (wt) replication. 104 Furthermore, competition experiments showed that extra copies of PKs conferred a replicative 105 advantage to genomes. Although replicons and full-length genomes lacking PKs were 106 replication-competent, no infectious virus was rescued from genomes containing less than one 107 PK copy. This is consistent with our earlier report describing the presence of putative 108 packaging signals in the PK region (22). 109 110 author/funder. All rights reserved. No reuse allowed without permission. Plasmid construction. 117 The FMDV replicon plasmids, pRep-ptGFP, and the replication-defective polymerase mutant 118 control, 3D-GNN, have already been described (10).\\nTo introduce mutations into the PK region, the pRep-ptGFP replicon plasmid was digested 121 with SpeI and KpnI and the resulting fragment inserted into a sub-cloning vector (pBluescript) 122 to create the pBluescript PK. PKs 3 and 4 were removed by digestion with HindIII and AatII 123 before insertion of a synthetic DNA sequence with PK 3 and 4 deleted. PKs 2, 3 and 4 were 124 deleted by PCR amplification using ΔPK 234 Forward primer and FMDV 1331-1311 reverse 125 primer, the resultant product was digested with HindIII and AatII and ligated into the 126 pBluescript PK vector. Complete PK deletion was achieved by introduction of an AflII site at 127 the 3′ end of the poly-C tract by PCR mutagenesis to create the sub-cloning vector, pBluescript 128 C11, which was then used to remove all the PKs by PCR mutagenesis using ΔPK 1234 forward 129 primer and FMDV 1331-1311 reverse primer. The modified PK sequences were removed from 130 the sub-cloning vectors and inserted into the pRep-ptGFP plasmid using NheI-HF and KpnI-131 HF.\\n132 133 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 7 Mutations to disrupt and reform PK structure were introduced using synthetic DNA by 134 digestion with AflII and AatII and ligation into a similarly digested pBluescript PK vector.\\nMutations were then introduced into the replicon plasmid as described above.\\nTo assess the effects of truncation of the poly-C-tract on replication the entire sequence was 137 removed. This was performed by PCR mutagenesis using primers C0 SpeI, and FMDV 1331- In vitro transcription. 143 In vitro transcription reactions for replicon assays were performed as described previously (28).\\nTranscription reactions to produce large amounts of RNA for SHAPE analysis were performed 145 with purified linear DNA as described above, and 1 μg of linearised DNA was then used in a 146 HiScribe T7 synthesis kit (NEB), before DNase treatment and purification using a PureLink FastQ files were quality checked using FastQC with poor quality reads filtered using the 225 Sickle algorithm. Host cell reads were removed using FastQ Screen algorithm and FMDV 226 reads assembled de novo into contigs using IDBA-UD (35). Contigs that matched the FMDV 227 library (identified using Basic Local ALighnment Search Tool (BLAST)) were assembled 228 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint into consensus sequences using SeqMan Pro software in the DNA STAR Lasergene 13 229 package (DNA STAR) (36). The SHAPE data largely agreed with the predicted structures with the stems of PK 1, 2 and 3, interacting nucleotides showed little to no reactivity, suggesting NMIA could not interact with 300 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 14 these nucleotides either due to the predicted base pairing or steric hindrance (figure 2B). The\\nNMIA reactivity for the interacting nucleotides in the stem-loops with downstream residues of 302 PK 1, 2 and 3 again largely agreed with the predicted structure, although the SHAPE data 303 suggests that there might be fewer interactions than previously predicted. However, differences 304 here could be due to heterogeneity in the formation of PKs in this experiment. The evidence 305 for loop-downstream interaction was weaker for PK4. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint\\norientation. 351 Since removal of all four PKs resulted in a significant decrease in replication, the minimal 352 requirements to maintain wt levels of replication were investigated. As near wt level of 353 replication was observed when only one PK was present, all further mutagenesis was 354 performed in a C11 replicon plasmid containing only PK 1. In addition, the orientation of PK 1 was reversed by \"flipping\" the nucleotide sequence to 367 potentially facilitate hybridisation of the loop with upstream rather than downstream sequences.\\nChanging the orientation of the PK reduced replicon replication to a similar level seen in the replication decreased until at passage three there is a 2.5 fold reduction compared to that of 398 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint passage 0 (figure 5B). Therefore, it appears that replicons with a single PK are at a competitive 399 disadvantage compared to those with two or more. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 20 of infectious virus despite being able to replicate after transfection into cells, is consistent with 448 a requirement for RNA structure within the PK region being required for virus assembly. The 5′ UTR of FMDV is unique amongst picornaviruses due to its large size and the presence 454 of multiple RNA elements, some of which still have unknown function. One of these features 455 is a series of repeated PKs varying in number from 2-4, depending on virus strain. In this study, 456 we sequentially deleted or mutated the PKs to help understand their role in the viral life cycle. 457 We also confirmed the predicted PK structures by SHAPE mapping, although there may be Although all viruses isolated to date contain at least two PKs, replicons or viruses containing a 464 single PK were still replication competent. However, replicons with more than a single PK 465 were found to have a competitive advantage over replicons with a single PK when sequentially 466 passaged. Replicons lacking all PKs displayed poor passaging potential even when co-467 transfected with yeast tRNA, reinforcing the observation of a significant impact in replication.\\nMoreover, viruses recovered from genomes with reduced numbers of PKs were slower growing 469 and produced smaller plaques. In addition, these differences were more pronounced in more PKs is functionally competent as no differences was seen between replicons congaing a single 472 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 21 copy of PK1 or PK4. This observation is consistent with a previous report of deletion of PK1, 473 along with the Poly-C-tract, with no adverse effect in viral replication (37). This also supports 474 our findings that the truncation of the Poly-C-tract to create the C11 construct had no effect on 475 replicon replication in the cell lines tested. As has been described with Mengo virus, it is 476 possible that the role of the poly-C-tract is essential in other aspects of the viral lifecycle which 477 cannot be recapitulated in a standard tissue culture system (39).\\nThe presence of at least two PKs in all viral isolates sequenced so far suggests that multiple 480 PKs confer a competitive advantage in replication. Here we showed by sequential passage that 481 replicons containing at least two PKs were maintained at a level similar to wt, but replicons 482 containing only one PK showed a persistent decline. It is unclear why some viral isolates 483 contain two, three or four PKs is still unknown, but this may be stochastic variation or may 484 reflect subtle effects of host range or geographical localisation. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint Significance is shown comparing the replication of C11 PK disrupt and C11 PK restore (Aii). Significance shown is compared to wt replicon. Error bars are calculated by SEM, n = 3, * P 673 < 0.05, **** P < 0.0001. 674 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 33 675 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint \\n                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
       "1  In December 2019, a novel coronavirus, SARS-CoV-2, was identified in Wuhan, China, as the etiologic agent of coronavirus disease 2019 , which by March 2020 has already spread across more than 80 countries 1 . Common symptoms of infection include fever, cough, and shortness of breath, while severe cases are characterized by advanced respiratory distress and pneumonia, often resulting in death 2 . Soon after the first epidemiological data together with SARS-CoV-2 genetic sequences were made available, a glut of phylogeny-based analyses began to circulate discussing, in scientific papers as well as (social) media, countries that might have been fueling the spread. The implications of misunderstanding the real dynamic of the COVID-19 pandemic are extremely dangerous. Ethnic or social discrimination resulting from unsupported assumptions on viral contagion -often amplified by irresponsible, uncontrollable communications -can be highly damaging for people and countries.\\nDespite (social) media are often vehicle for fake news and boast news hype, it is also worth noting the tremendous effort of the scientific community to provide free, up-to-date information on ongoing studies, as well as critical evaluations. In particular, the US-based NextStrain 3 team has been posting real-time updates on the epidemic tracing by molecular analyses. Several discussions and evidence-based debates on controversial hypotheses on the epidemic have ensued -such as the number of untraced infections in the US, the putative virus introduction in Italy through Germany, and the alleged lineage diversification in China 4 later criticized 5 -aided by a concerted and productive effort of internationally-renowned research groups in Edinburgh, Glasgow, and Oxford, UK, among others. Recently, an editorial published on Science 6 has also highlighted how unsupported or misleading claims circulating in forums, social media, and even peer-reviewed articles, have been led by a substantial over interpretation of the available data.\\nHence, the urgency to reframe the current debate in more rigorous scientific terms, and quantitatively evaluate whether sufficient information for reliable phylogenetic and phylogeographic studies currently exists, or which gaps need to be addressed.\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint Before carrying out any phylogeny-based analysis of virus evolution and spatiotemporal spread, it is crucial to test the quality of sequence data, since uneven sampling, presence of phylogenetic noise, and absence of temporal signal can affect reliability of the results (e.g. ancestral state reconstructions, molecular clock calibrations) 7 . We downloaded all SARS-CoV-2 full genome sequences deposited on GISAID (https://www.gisaid.org/) 8 to compare the number of high quality genomes sampled per country with the number of confirmed cases at the time of sampling, as well as the country's total population ( Figure 1 ). As March 10 th 2020, 331 genome sequences from 29 countries were available, almost twice as much than just one week earlier (March 3 rd 2020, 169 genome sequences from 22 countries). Yet, the sampling per country was inhomogeneous, with confirmed cases clearly uncorrelated to the number of genomes (Spearman correlation = 0.03). Moreover, correlation could only be investigated with confirmed cases (as proxy), since not all affected countries have made publicly available the total number of coronavirus testing performed. Even within the same country, sequenced genomes were usually sampled from few hotspots, not necessarily representative of the whole epidemic in that country.\\nItaly, for example, the second country per confirmed cases, has uploaded so far only two genomes from Lombardy and three from Lazio (Table S1) , i.e. only two of the twenty Italian regions, all with confirmed cases 9 . China and USA contributed with 123 and 63 genomes respectively (56% combined), while 13 countries contribute with only one or two samples (6.3% combined). Notably, three countries with less than 400 confirmed cases have provided a substantial number of genomes (Netherlands 28, Australia 18, Singapore 11; 17% combined) ( Figure 1 ).\\nThe effect of inhomogeneous sampling and missing data on phylogeography reconstructions, like the ones recently rushed through news and (social) media to claim specific dissemination routes of SARS-CoV-2 among countries, can be quite dramatic. An instructive example is the putative introduction of SARS-CoV-2 in Italy from Germany. The maximum likelihood (ML) tree, inferred from the full genome viral sequences available on March 3 rd , 2020 (Supplementary Figure S1 ), showed a well-supported cluster of European and Asian sequences (reported in Figure 2a ), which contained a subclade (subclade A, Figure 2a ) including a sequence isolated in Germany that . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint appears to be paraphyletic (with strong bootstrap support) to an Italian sequence clustering, in turn, with sequences from Finland, Mexico, Germany and Switzerland (Figure 2a ).\\nBased on this observation (which was available on NextStrain), a heated discussion circulated on social media about a transmission event from Germany to Italy followed by further spread from Italy the other countries. However, in a new tree inferred just one week later, when more than 135 new full genome sequences were made available on GISAID 8 ( Figure S2 ), the direct link between Germany and Italy has disappeared due to the additional clustering of previously unsampled sequences from Portugal, Brazil, Wales and Netherland ( Figure 2b ). Most importantly, there is no sufficient evidence of a higher likelihood (Shimodaira-Hasgawa test, Table 1 ) for tree topologies generated arbitrarily switching branches in the tree (see arrows in Figure 2b) compared to the ML tree inferred from the real data. In other words, it is not possible with the present data to decide which branching pattern (and, therefore, phylogeographic reconstruction)\\nis the most likely.\\nLack of resolution and uncertainty in the SARS-CoV-2 phylogenetic tree is to be expected, considering that relatively little genetic diversity can be accumulated during the first three months of an epidemic, even for an exponentially spreading and fast-evolving RNA virus. Plots of pairwise transitions/transversions vs. genetic distance 10 showed, indeed, low genetic diversity among SARS-CoV-2 genomes, considerable scattering, and overlap in transitions and transversions among closely related sequences (Figure 3a ). The 331 full genomes available contained 477 out of 29,564 variable sites (1.6%), of which 377 were singleton and only 100 (0.34%) parsimony informative, indicating a substantial lack of phylogenetic information, and raising the possibility of a number of sequencing errors. Further assessment of phylogenetic signal in the dataset was carried out using likelihood mapping analysis 11 , which estimates the likelihood of each of possible tree topology for any group of four sequences (quartet), randomly . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint chosen from an alignment, and reports them inside an equilateral triangle (the likelihood map)\\nwhere the corners represent distinct tree topologies and the center star-like trees. Quartets are considered \"resolved\" when the three likelihoods are significantly different (phylogenetic signal, most dots equally distributed in the corners, i.e. data are suitable for robust phylogeny inference), unresolved or partially resolved, when all three likelihood values or two of them are not significantly different (phylogenetic noise, most dots in sides or center areas, i.e. data may not be sufficient for robust phylogeny inference). Extensive simulation studies have shown that, for sequences to be considered robust in terms of phylogenetic signal, side/center areas of the likelihood mapping must include <40% of the unresolved quartets 12 , way lower than the 52.9% unresolved quartets detected in the SARS-CoV-2 full genomes alignment ( Figure 3b ). It is important to keep in mind that such a lack of phylogenetic signal makes the overall topology of any SARS-CoV-2 tree unreliable, and even clades with high bootstrap values should be interpreted with extreme caution. Finally, the mean genetic distance of each sequence from the root of a ML tree indicated that the available genomic data lack temporal signal (no significant linear correlation between accumulated genetic distance and sampling time, Figure 3c ). This is, again, expected in genomes obtained over a very short period of time (~ three months) since the beginning of the outbreak. Bayesian analysis 13 , which infers phylogenetic and phylogeographic patterns from a posterior distribution of trees, might facilitate comparisons about different evolutionary scenarios, help in retrieving the correct topology, and estimate an accurate evolutionary rate. With more genome sequences sampled from different time points and geographic areas made available, literally, every day, Bayesian analysis may soon be a viable option. However, as long the new data do not increase phylogenetic and temporal signal, Bayesian phylogenetic and phylogeography results would remain equally questionable.\\nTaken together, our observations suggest that any conclusion drawn, at present, about existing lineages and direction of viral spread, based on phylogenetic analysis of SARS-CoV-2 sequence data, is potentially misleading. Nowadays, published scientific data and media are very easily . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint .\\naccessible to a worldwide audience, thus it is important more than ever to weigh the information being shared. We believe that current molecular epidemiology data are not solid enough to provide a scientifically sound analysis of SARS-CoV-2 spread. Any inference based on phylogenetic trees should be considered at the very best preliminary, and hypothesis-generating.\\nHence the need for 1) avoiding stigmatization based on partial information and 2) continuing the concerted scientific effort to increase the number and quality of available data. The evolutionary dynamics of SARS-CoV-2 spread will unveil an unprecedented amount of information, which is essential to make policy decisions. The whole of humanity is threatened by the current pandemic, and policy makers need to adjust their mitigation measures while the pandemic itself is developing. Some of the urgent answers required lie in the timely availability of abundant, high quality sequence data not only from countries experiencing a high number of reported cases, such as Italy and South Korea, but also from others that seem to be experiencing, at least for now, a lower number of infections.\\nGISAID was accessed on March 3 rd and 10 th , 2020 (final dataset information reported in Table   S1 ). After quality control of sequences that were not full genomes or contained extensive stretches of unknown nucleotides, the first dataset (March 3 rd 2020) consisted of 169 genomes from 22 countries, while the second (March 10 th , 2020) 331 from 29 countries. Confirmed cases were retrieved for the same day from interactive web-based dashboard developed by Johns Hopkins University 14 .\\nTransition/transversions vs. genetic distance plot were calculated using DAMBE6 10 . Evaluation of the presence of phylogenetic signal satisfying resolved phylogenetic relationships among sequences was carried out with IQ-TREE, allowing the software to search for all possible quartets . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint using the best-fitting nucleotide substitution model 11 . ML tree reconstruction was performed in IQ-TREE based on the best-fit model chosen according to Bayesian Information Criterion (BIC) 15, 16 . Topology was tested performing the Shimodaira-Hasegawa 17 on \"Subclade A\" by comparing the inferred tree to three trees where two branches: (Tree 1) Italy (EPI_ISL_412973) switched with Wales (EPI_ISL_413555), (Tree 2) Germany (EPI_ISL_406862 with) Brazil (EPI_ISL_412964), (Tree 3) and Portugal (EPI_ISL_413648) with Brazil (EPI_ISL_412964), (Tree 4) Germany (EPI_ISL_406862) with Portugal (EPI_ISL_413648). All tests performed 10,000 resamplings using the using the RELL approximation method 18 . Position of these genomes within \"Subclade A\" is indicated in Figure 1 with an arrow.\\nExploration of temporal structure, i.e. presence of molecular clock in the data, was assessed by regression of divergence -root-to-tip genetic distance-against sampling time using TempEst 19 . In this case, absence of a linear trend indicates that the data does not contain temporal signal and that the data is not appropriate for phylogenetic inference using molecular clock models.\\nAncestral state reconstruction of full datasets reported in Figures S1 and S2 were based on the ML tree obtained with IQ-TREE and performed using TreeTime 20 . We choose to show the \"European subclade\" and \"Subclade A\" as cladograms in Figure 1 as there are no phylogenetic and temporal signals in the datasets: scaling phylogenies in time or nucleotide substitution in this case is misleading. . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint Each bi-directional arrow, and corresponding number, connects two tip branches that were switched to generate an alternative tree topology to be tested (see Table 1 , Methods). CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity. EPI_ISL_412971_BetaCoV_Finland_FIN-25_2020 EPI_ISL_406533_BetaCoV_Guangzhou_20SF206_2020 EPI_ISL_407896_BetaCoV_Australia_QLD02_2020 EPI_ISL_408478_BetaCoV_Chongqing_YC01_2020\\nEPI_ISL_412979__BetaCoV_Wuhan_HBCDC-HB-03_2020\\nEPI_ISL_411220_BetaCoV_France_IDF0386-islP3_2020 EPI_ISL_410044_BetaCoV_USA_CA6_2020 EPI_ISL_413020_BetaCoV_Switzerland_1000477377_2020\\nEPI_ISL_406030_BetaCoV_Shenzhen_HKU-SZ-002_2020\\nEPI_ISL_408669_BetaCoV_Japan_KY-V-029_2020\\nEPI_ISL_408515_BetaCoV_Wuhan_IVDC-HB-envF13-21_2020 EPI_ISL_410713_BetaCoV_Singapore_7_2020 EPI_ISL_412873_BetaCoV_Korea_KCDC24_2020\\nEPI_ISL_402120_BetaCoV_Wuhan_IVDC-HB-04_2020\\nEPI_ISL_406970_BetaCoV_Hangzhou_HZ-1_2020 EPI_ISL_406973_BetaCoV_Singapore_1_2020 EPI_ISL_412029_BetaCoV_Hong_Kong_VM20001988_2020\\nEPI_ISL_410720_BetaCoV_France_IDF0372-isl_2020 EPI_ISL_411951_BetaCoV_Sweden_01_2020 EPI_ISL_408488_BetaCoV_Jiangsu_IVDC-JS-001_2020\\nEPI_ISL_402123_BetaCoV_Wuhan_IPBCAMS-WH-01_2019 EPI_ISL_407894_BetaCoV_Australia_QLD01_2020 EPI_ISL_411955_BetaCoV_USA_CA8_2020 EPI_ISL_403962_BetaCoV_Nonthaburi_61_2020 EPI_ISL_406801_BetaCov_Wuhan_WH04_2020 EPI_ISL_408978_BetaCoV_Wuhan_WH05_2020\\nEPI_ISL_407084_BetaCoV_Japan_AI_I-004_2020 EPI_ISL_407215_BetaCoV_USA_WA1-F6_2020 EPI_ISL_411060_BetaCoV_Fujian_8_2020 EPI_ISL_410984_BetaCoV_France_IDF0515-isl_2020\\nEPI_ISL_412980__BetaCoV_Wuhan_HBCDC-HB-04_2020 EPI_ISL_410546_BetaCoV_Italy_INMI1-cs_2020 EPI_ISL_408977_BetaCoV_Sydney_3_2020 EPI_ISL_412982__BetaCoV_Wuhan_HBCDC-HB-06_2020 EPI_ISL_407071_BetaCoV_England_01_2020 EPI_ISL_412912_BetaCoV_Germany_Baden-Wuerttemberg-1_2020 EPI_ISL_411927_BetaCoV_Taiwan_4_2020 EPI_ISL_410536_BetaCoV_Singapore_5_2020 EPI_ISL_406538_BetaCoV_Guangdong_20SF201_2020 EPI_ISL_407988_BetaCoV_Singapore_3_2020 EPI_ISL_406036_BetaCoV_USA_CA2_2020 EPI_ISL_411954_BetaCoV_USA_CA7_2020 EPI_ISL_412028_BetaCoV_Hong_Kong_VM20001061_2020 EPI_ISL_406862_BetaCoV_Germany_BavPat1_2020 EPI_ISL_406597_BetaCoV_France_IDF0373_2020 EPI_ISL_408009_BetaCoV_USA_CA4_2020 EPI_ISL_410716_BetaCoV_Singapore_10_2020 EPI_ISL_403933_BetaCoV_Guangdong_20SF013_2020 EPI_ISL_402129_BetaCoV_Wuhan_WIV06_2019\\nEPI_ISL_406594_BetaCoV_Shenzhen_SZTH-003_2020 EPI_ISL_407987_BetaCoV_Singapore_2_2020 EPI_ISL_413015_BetaCoV_Canada_ON_VIDO-01_2020 EPI_ISL_407976_BetaCoV_Belgium_GHB-03021_2020 EPI_ISL_406531_BetaCoV_Guangdong_20SF174_2020 EPI_ISL_410715_BetaCoV_Singapore_9_2020\\nEPI_ISL_404227_BetaCoV_Zhejiang_WZ-01_2020 EPI_ISL_408479_BetaCoV_Chongqing_ZX01_2020 EPI_ISL_406535_BetaCoV_Foshan_20SF210_2020 EPI_ISL_408976_BetaCoV_Sydney_2_2020 EPI_ISL_411066_BetaCoV_Fujian_13_2020 EPI_ISL_407073_BetaCoV_England_02_2020\\nEPI_ISL_403929_BetaCoV_Wuhan_IPBCAMS-WH-04_2019 EPI_ISL_410301_BetaCoV_Nepal_61_2020 EPI_ISL_403931_BetaCoV_Wuhan_IPBCAMS-WH-02_2019\\nEPI_ISL_410531_BetaCoV_Japan_NA-20-05-1_2020 EPI_ISL_403963_BetaCoV_Nonthaburi_74_2020 EPI_ISL_402125_BetaCoV_Wuhan-Hu-1_2019\\nEPI_ISL_408514_BetaCoV_Wuhan_IVDC-HB-envF13-20_2020\\nEPI_ISL_402119_BetaCoV_Wuhan_IVDC-HB-01_2019 EPI_ISL_411956_BetaCoV_USA_TX1_2020 EPI_ISL_406596_BetaCoV_France_IDF0372_2020 EPI_ISL_412967_BetaCoV_China_IQTC02_2020 EPI_ISL_408431_BetaCov_France_IDF0626_2020\\nEPI_ISL_410532_BetaCoV_Japan_OS-20-07-1_2020\\nEPI_ISL_411957_BetaCoV_China_WH-09_2020 EPI_ISL_412970_BetaCoV_USA_WA2_2020 EPI_ISL_412978__BetaCoV_Wuhan_HBCDC-HB-02_2020 EPI_ISL_412869_BetaCoV_Korea_KCDC05_2020 EPI_ISL_412965_BetaCoV_Canada_BC_37_0-2_2020 EPI_ISL_412871_BetaCoV_Korea_KCDC07_2020 EPI_ISL_408008_BetaCoV_USA_CA3_2020\\nEPI_ISL_406593_BetaCoV_Shenzhen_SZTH-002_2020 EPI_ISL_406536_BetaCoV_Foshan_20SF211_2020 EPI_ISL_412459_BetaCoV_Jingzhou_HBCDC-HB-01_2020 EPI_ISL_410718_BetaCoV_Australia_QLD04_2020 EPI_ISL_412862_BetaCoV_USA_CA9_2020 EPI_ISL_413019_BetaCoV_Switzerland_1000477102_2020\\nEPI_ISL_412981__BetaCoV_Wuhan_HBCDC-HB-05_2020\\nEPI_ISL_408668_BetaCoV_Vietnam_VR03-38142_2020 EPI_ISL_410535_BetaCoV_Singapore_4_2020 EPI_ISL_408665_BetaCoV_Japan_TY-WK-012_2020 EPI_ISL_406716_BetaCoV_China_WHU01_2020 EPI_ISL_402127_BetaCoV_Wuhan_WIV02_2019 EPI_ISL_411926_BetaCoV_Taiwan_3_2020 EPI_ISL_406223_BetaCoV_USA_AZ1_2020 EPI_ISL_407214_BetaCoV_USA_WA1-A12_2020 EPI_ISL_406717_BetaCoV_China_WHU02_2020 EPI_ISL_404228_BetaCoV_Zhejiang_WZ-02_2020\\nEPI_ISL_412898__BetaCoV_Wuhan_HBCDC-HB-02_2019 EPI_ISL_413017_BetaCoV_South_Korea_KUMC01_2020 EPI_ISL_408480_BetaCoV_Yunnan_IVDC-YN-003_2020 EPI_ISL_403936_BetaCoV_Guangdong_20SF028_2020 EPI_ISL_408667_BetaCoV_Japan_TY-WK-521_2020 EPI_ISL_411218_BetaCoV_France_IDF0571_2020 EPI_ISL_405839_BetaCoV_Shenzhen_HKU-SZ-005_2020 EPI_ISL_408010_BetaCoV_USA_CA5_2020 EPI_ISL_406031_BetaCoV_Taiwan_2_2020 EPI_ISL_406844_BetaCoV_Australia_VIC01_2020 EPI_ISL_411950_BetaCoV_Jiangsu_JS01_2020 EPI_ISL_412974_BetaCoV_Italy_SPL1_2020\\nEPI_ISL_403928_BetaCoV_Wuhan_IPBCAMS-WH-05_2020 EPI_ISL_412870_BetaCoV_Korea_KCDC06_2020 EPI_ISL_404253_BetaCoV_USA_IL1_2020 EPI_ISL_410045_BetaCoV_USA_IL2_2020 EPI_ISL_403932_BetaCoV_Guangdong_20SF012_2020\\nEPI_ISL_408484_BetaCoV_Sichuan_IVDC-SC-001_2020 EPI_ISL_410537_BetaCoV_Singapore_6_2020 EPI_ISL_412872_BetaCoV_Korea_KCDC12_2020 EPI_ISL_407079_BetaCoV_Finland_1_2020\\nEPI_ISL_410545_BetaCoV_Italy_INMI1-isl_2020 EPI_ISL_408670_BetaCoV_USA_WI1_2020 EPI_ISL_403937_BetaCoV_Guangdong_20SF040_2020 EPI_ISL_403934_BetaCoV_Guangdong_20SF014_2020\\nEPI_ISL_413014_BetaCoV_Canada_ON-PHL2445_2020\\nEPI_ISL_411915_BetaCoV_Taiwan_CGMH-CGU-01_2020 EPI_ISL_410717_BetaCoV_Australia_QLD03_2020 EPI_ISL_412026_BetaCoV_Hefei_2_2020 EPI_ISL_407313_BetaCoV_Hangzhou_HZCDC0001_2020 EPI_ISL_412972_BetaCoV_Mexico_CDMX_InDRE_01_2020 EPI_ISL_406034_BetaCoV_USA_CA1_2020 EPI_ISL_411952_BetaCoV_Jiangsu_JS02_2020 EPI_ISL_411902_BetaCoV_Cambodia_0012_2020\\nEPI_ISL_413016_BetaCov_Brazil_SPBR-02_2020 EPI_ISL_413018_BetaCoV_South_Korea_KUMC02_2020 EPI_ISL_403930_BetaCoV_Wuhan_IPBCAMS-WH-03_2019\\nEPI_ISL_412899__BetaCoV_Wuhan_HBCDC-HB-03_2019 EPI_ISL_408489_BetaCoV_Taiwan_NTU01_2020 EPI_ISL_409067_BetaCoV_USA_MA1_2020 EPI_ISL_404895_BetaCoV_USA_WA1_2020\\nEPI_ISL_408666_BetaCoV_Japan_TY-WK-501_2020 EPI_ISL_410218_BetaCov_Taiwan_NTU02_2020 EPI_ISL_407893_BetaCoV_Australia_NSW01_2020 EPI_ISL_407193_BetaCoV_South_Korea_KCDC03_2020\\nEPI_ISL_412983_BetaCoV_Tianmen_HBCDC-HB-07_2020 EPI_ISL_410714_BetaCoV_Singapore_8_2020 EPI_ISL_403935_BetaCoV_Guangdong_20SF025_2020 EPI_ISL_406798_BetaCov_Wuhan_WH01_2019 EPI_ISL_411953_BetaCoV_Jiangsu_JS03_2020 EPI_ISL_412030_BetaCoV_Hong_Kong_VB20026565_2020 EPI_ISL_412968_BetaCoV_Japan_Hu_DP_Kng_19-020_2020 EPI_ISL_412973_BetaCoV_Italy_CDG1_2020 EPI_ISL_402132_BetaCoV_Wuhan_HBCDC-HB-01_2019\\nEPI_ISL_408486_BetaCoV_Jiangxi_IVDC-JX-002_2020 EPI_ISL_408430_BetaCoV_France_IDF0515_2020 EPI_ISL_410486_BetaCoV_France_RA739_2020 EPI_ISL_412966_BetaCoV_China_IQTC01_2020\\nEPI_ISL_412969_BetaCoV_Japan_Hu_DP_Kng_19-027_2020\\nEPI_ISL_411219_BetaCoV_France_IDF0386-islP1_2020\\nEPI_ISL_408481_BetaCoV_Chongqing_IVDC-CQ-001_2020 EPI_ISL_412975_BetaCoV_Australia_NSW05_2020 EPI_ISL_406534_BetaCoV_Foshan_20SF207_2020 . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint \\n                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
       "2  The 2019-nCoV epidemic has spread across China and 24 other countries 1-3 as of February 8, 2020 . The mass quarantine measure implemented in the epidemic's epicenter, Wuhan, Hubei Province, China, on January 23, 2020, has stopped the human migration between Wuhan and other places. However, according to the local authorities, over five million people had already traveled from Wuhan to other parts of China in the weeks leading to January 23, 2020, with the majority (79.96%) to the other cities within Hubei.\\nHealthcare resources have become a major bottleneck in the response to this public health crisis. The sudden onset of the epidemic and the sheer size of the infection have caused significant shortage in healthcare resources across Hubei Province. So far, there have been over 11,000 non-local healthcare professionals, mobilized by the public health authorities from other parts of China, fighting against the epidemic in Wuhan. However, this is far from adequate for Wuhan given the severity of the epidemic. In addition to Wuhan, other cities in Hubei are in need of urgent external help as well with respect to health resources, as multiple localized outbreaks are taking place. On February 7, 2020, the National Health Commission of China (NHCC) initiated the \"One City Covered by One Province\" program, explicitly assigning a specific province to aid one city in Hubei. Two critical questions of operational nature remain, which cities, both in Hubei and other provinces, are most vulnerable to localized outbreaks, and how the public health authorities should balance the healthcare resources under stress to best fight the epidemic and minimize the risks? This study is aimed to shed light on these questions by developing a vulnerability analysis framework integrating the human movement and healthcare resource data.\\nThis study makes use of four public datasets: (a) the daily counts of confirmed 2019-nCoV cases, collected from NHCC 4 ; (b) the list of designated hospitals with the number of reserved beds, retrieved from the websites of the provincial and municipal health commissions; (c) the information of the hospitals designated for 2019-nCoV, retrieved from 99.com.cn, a public hospital database; and (d) the human movement data, provided by the Baidu Migration (http://qianxi.baidu.com/), a public site offering real-time migration information across mainland China.\\nRecent studies indicated that the confirmed case data for Hubei Province, in particular, Wuhan, was subject to the under-reporting bias, while the data from places outside of Hubei is more reliable for inferring the actual epidemic size 5 . Following Wu et al. 5 , we developed a susceptible-exposed-infectious-recovered (SEIR)-based metapopulation model, which has 100 sub-populations representing the 100 cities that have the greatest number of people traveling from Wuhan. Note that we adopted the same epidemiological parameters as Wu et al. 5 . This model was then used to estimate IDi, the number of infected individuals in each city. Lastly, healthcare-resource-adjusted vulnerability indexes to the 2019-nCoV outbreak for city i, Vi were obtained as follows: Vi = IDi / Bi, where Bi represents the total number of reserved beds in the designated hospitals.\\nThe Figure presents the estimated number of infected individuals, as well as the healthcareresource-adjusted vulnerability to the 2019-nCoV outbreak across mainland China, as of January 29, 2020. The epidemiological parameters may have changed after January 23, 2020, when the lockdown measure was implemented. Thus, we only use the data by January 23, 2020 and produce the estimation till January 29, 2020 to accommodate the six-day incubation period identified by Wu et al. 5 .\\nThe ten most vulnerable non-Hubei cities are also listed in the figure. Note that the cities in Hubei (which are being aided by other provinces) are not presented here because of their significantly higher vulnerability as compared to others. Among the ten most vulnerable non-Hubei cities, Changsha, Beijing, Chongqing and Shanghai are major metropolises with a large number of estimated infected individuals. Xinyang, Jiujiang, Nanchang and Yiyang are medium-sized cities neighboring Hubei. In particular, Xinyang appears to be in a particular vulnerable situation because of the large number of people traveling from Wuhan and its insufficient healthcare resources.\\nThe reported vulnerability analysis informs public health response to the 2019-nCoV epidemic in multiple ways. For the identified cities with high vulnerability, it is important to increase the reserve of healthcare resources or have contingent plans. In particular, some of these cities already contributed significant healthcare resources to Hubei, leaving themselves inadequately prepared for the localized outbreak. Our analysis revealed that several medium-sized cities, particularly those neighboring Hubei, are highly vulnerable due to the lack of healthcare resources. \\n                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
       "3  The outbreak of infectious diseases has always been one of the most important health problems in the world. With the rapid spread of the new coronavirus (COVID- 19) , much attention has been paid to subjects like rate of an epidemic, transition ways, prevention methods, and remaining time of the virus in the environment [1] [2] [3] [4] [5] . Therefore, the behavior of the virus in relation to environmental factors is critical. The coronavirus is one of the epidemic diseases that many countries have experienced in the new type of it, COVID-19, and can become a serious challenge and affect the previous efforts on sustainable development [6] [7] [8] [9] .\\nIt has been found that there are some delays between the detection of confirmed cases and the actual infection date due to the laboratory test, the time of the announcement of confirmed cases in media, and the time for the patients showing initial symptoms for a total usually of 3 to 5 days [10] [11] [12] . The analysis of Kampf et al. (2020) demonstrated that COVID-19 could remain up to 9 days on different materials based on temperature and humidity [13] . Cascella et al. (2020) conducted research on the new coronavirus structure. Their analysis shows that the virus belongs to the family of single-stranded RNA viruses (+ ssRNA), which has a length of about 30 kb and an envelope with spear structures and they tested the sensitivity of the virus to UV and temperature [14] . investigated the correlations between environmental factors and the number of positive cases of COVID-19 by the use of artificial intelligence techniques. The results of this study put evidence on the role of urban and weather conditions on the epidemic rate [15] . Chen et al. (2020) developed a time-dependent mathematical model for the prediction of the total number of positive cases [16] . Lytle and Sagripanti (2005) studied the impact of UV rays, and their results determined that it can affect the structure of coronaviruses because UV rays could damage the nucleic acids of viral RNA that is vital for the survival of the virus [17] . developed an assessment method for investigating the impact of climate and urban factors in confirmed cases of COVID-19 [18] . Bukhari et al., (2020) , studied the epidemic of COVID-19 on tropical zones and their results show that the highest number of infection happened in temperature among 3 to 17 ºC [19] .\\nThe analysis of Grant and Giovannucci (2009) shows that by increasing the UV, the level of vitamin D in the body increases and makes the infection rates decrease. Therefore, in areas on lower latitudes and closer to the equator line, the epidemic on COVID-19 might be different [20] . In the study of Chan et al. (2011) , the effect of temperature and humidity on SARS coronavirus was analyzed, and they found that increasing temperature decreases the epidemic rate by faster disabling virus structure on contaminated surfaces [21] . Pirouz and Violini (2020) , analyzed the outbreak patterns of COVID-19 by analysis of 117 countries. They suggested two mechanisms of triggering the development of the infection, namely the normal spread of an infection from one parent case and the consequence of a unique phenomenon of accumulation of contagious people [22] . Analysis of the previous studies demonstrates the necessity of further research about the effect of environmental factors on the COVID-19. Since some of the previous researches have shown the effect of weather conditions, especially temperature, on coronavirus viruses, the main purpose of this study is to investigate the better correlation of daily mean air temperature on the epidemic rate of the COVID-19.\\nIn this study, we analysed the number of positive contagions of coronavirus COVID-19, from February 29 th to March 19 th . The selection of February 29 th is due to existence on this date of at least one positive case of COVID-19 in all case studies studied. The analysis of the weather data started from February15 th and ended on March 18 th because the impact of the weather parameter would occur in the upcoming days. Thus, the considered average daily temperature is with an interval of 1 to 14 days before the definitive cases of COVID-19.\\n In order to minimize the impact of other parameters on the results, such as the policy of each government, cultural differences between countries, health care systems, or the protocols in getting coronavirus test, comparative analysis have been done for different regions of the same country, without comparing with each other the results of different countries;  The analysis period for average daily temperature ranges from February15 th , 2020 to March18 th , 2020 (33 days);  The analysis period for confirmed cases of coronavirus COVID-19 is from February 29 th to March19 th ;  For what concerns the cumulative daily rate of confirmed cases, the total number of confirmed cases in a region was divided into the population of that region in order to take into account the different size of the population in each region.\\nTo carry out the analysis, the datasets of five countries and in each country, of three regions/provinces have been considered. The selected case studies are presented in Table 1 . and the location of the three regions shown in Figures 1. \\nMathematical models have been successfully used for the analysis of many hazards [32, 33] . In complex subjects, different Mathematical models can be used to solve the problem [34] [35] [36] . Statistical tests are an appropriate method for the investigation of possible correlations between different datasets [37] . Many statistical tests can be used for the analysis of significant differences between the datasets, among which F-test and Independent-Samples T-Test [38] .\\nIn order to investigate the hypothesis about a correlation between significant differences (Sig.) of average daily temperatures and average cumulative daily rate of confirmed cases in two regions, the following assumptions, and statistical tests have been done:\\n The variances of the two groups are equal: H0  The variances of the two groups are not equal: H1\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10.20059337 doi: medRxiv preprint When using F-test, if the significance level (sig) is less than 0.05, hypothesis H0 is rejected, and then hypothesis H1 is accepted, i.e., the variances of the datasets in the two selected regions are not equal. Conversely, if the significance level (sig.) is larger than 0.05, the hypothesis H1 is rejected, and the hypothesis H0 is accepted, i.e., the variances of the datasets in the two selected regions are equal.\\nThrough F-test, one determines the presence or absence of differences in the variances of the two regions. Through another statistical test, the Independent-Samples T-Test, one can determine whether the difference is significant, has been used. When using Independent-Samples T-Test, a significance level (sig.) lower (higher) than 0.05, means that the difference is (is not) significant.\\nIn our analysis, we used SPSS software. The critical factors in the SPSS Model set up are as follow:\\n The datasets in different regions of the same countries were analyzed with respect to the average daily air temperature during the 33-day period from 15-Feb to 18-Mar and the cumulative daily rate of confirmed cases of COVID-19 during the 19 days from 29-Feb to 19-Mar.  In all statistical tests, the confidence level of 95% is considered.  First, the F-test was used to test whether there is a significant equality of variances for each pair of case studies, and then, depending on the F-test result, by the T-Test the existence of a significant difference between the data of the two cases was investigated.\\nThe average daily temperature [39, 40] , and cumulative daily rate of confirmed cases [41] , are shown in Figures 2 to 6.\\na. Average daily temperature b. Cumulative daily rate of confirmed cases Correlations between the average temperatures and the cumulative daily rate of confirmed cases were found especially for towns with significant differences, with respect to the other two towns of the same country, in the average daily temperature. This was the case for Bandar Abas in Iran, Milan in Italy, Santa Cruz in Spain, and Los Angeles in the US. The average daily temperature and cumulative daily rate of confirmed cases are listed in Table 2 . As a general feature of Table 2 , we notice that  In 11 out of the possible fifteen comparisons, the average cumulative rate of confirmed cases is higher when the average daily temperature is lower, and in the four remaining cases the differences in either variable are small.\\nA more specific analysis per country shows that  In Iran, the average daily temperature in Bandar Abbas was the highest (21.54 ºC), and the average cumulative rate of daily of confirmed cases was the lowest (0.000027). . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10.20059337 doi: medRxiv preprint  In Italy, the average daily temperature in Milan was the lowest (7.99 ºC), and the average cumulative rate of daily of confirmed cases was the highest (0.000690).  In Germany, the average daily air temperatures of the selected towns have similar values, as well as the spread of the COVID-19 infection is similar.  In Spain, the average daily air temperature in Barcelona and Valencia are similar, and the spread of the infection is also similar. However, the comparison of Santa Cruz with higher average daily temperatures shows a slower spread of COVID-19.  In the United States, the average daily air temperature in both New York and Seattle are lower than Los Angeles, and the average cumulative rate of confirmed cases shows the spread of the virus in both is faster than in Los Angeles.\\nThese qualitative results reflect in the statistical analysis of the data by F-test and Independent-Samples T-Test presented in Table 3 . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10.20059337 doi: medRxiv preprint In Table 3 . t values or p-values can be used to check the existence of a significant correlation between the data for two cities.\\nLet us discuss in detail one such comparison, that between Ramsar and Tehran. From the F-test, the F value is equal to 20.4. The criterion for variance equality in the F-test is that the F value must smaller than the critical F value. The statistical analysis has been done on a statistical confidence level of 95%, and therefore the error level is 0.05, and considering the degrees of freedom, the critical F value of the test that can be found through F distribution tables [42] , in this case, is equal to F = 1.57. Since the F value we found is 20.4, larger than the critical F value, the assumption of the equality of variances is rejected. It means that the average temperature variance of the Ramsar and Tehran are not equal.\\nAccording to the t-test, the t value for these two cities is equal to -1.929. Based on an error level of 0.05 and by considering the degrees of freedom, the critical values of the test indicated in the T distribution tables [43] , is in this case equal to t = ± 1.645. Since the t value is out of the critical boundary (-1.645 ≤ acceptable t ≤ +1.645), the assumption of equality in the averages is rejected, and the existence of difference in averages is confirmed. Thus from two tests, it is evident that there is a significant difference between the average temperature of the two cities of Ramsar and Tehran.\\nAn analogous result holds for the average cumulative rate of confirmed cases.\\nThe main results of Table 4 , with a confidence level of 95% are as follow:\\n In Iran, the city of Bandar Abbas is significantly different from Ramsar and Tehran in terms of both average daily temperature and average cumulative rate of confirmed cases.  In Italy, the city of Milan is significantly different from Rome and Naples in terms of both average daily temperature and average cumulative rate of confirmed cases. Instead, the two cities of Rome and Napoli are not significantly different in terms of these variables.  In Germany, there are no significant differences in average daily temperature in all three cities, and the average cumulative rate of confirmed cases is not significantly different.  In Spain, Barcelona and Valencia are not significantly different in terms of average daily temperature, and the average cumulative rate of confirmed cases is not significantly different. However, in Barcelona and Santa Cruz, the average daily temperature is significantly different, as well as the average cumulative rate of confirmed cases, and the same trend exists between Valencia and Santa Cruz.  In the United States, the two cities of New York and Seattle are not significantly different in terms of average daily temperature, and of the average cumulative rate of confirmed cases, the number of patients is not significantly different. Instead, the city of Los Angeles is significantly different in terms of average daily temperature with two cities of New York and Seattle, and the average cumulative rate of confirmed cases are different significantly.\\nThe results of our analysis of daily air temperature for three case studies from five countries showed the existence of correlations between the average temperatures and the epidemic rate, and this is especially evident when the differences in average daily temperature are larger. For example, in Iran and Italy, the regions with lower average daily temperature had a higher epidemic level of the new coronavirus (COVID- 19) , and the same trend exists in all other case studies.\\nThe statistical analysis showed that in all the case studies we considered, referring to five different countries, if there is a significant difference in daily mean temperature, there is also a significant difference in the average cumulative daily rate of confirmed cases, supporting the proposed research hypothesis about the correlation between air temperature and the epidemic rates of COVID-19. Due to the correlations among temperature and hours of sunshine for future research, it is recommended to study the effects of the sun's UV rays as well as the average hours of sunshine on the epidemic rates of COVID-19.\\nThis study analyzed the possible correlation of daily average air temperature with the epidemic rate of the coronavirus (COVID-19) using significant differences analysis between the datasets. The research hypothesis is that in regions with a significant difference in the average daily temperature, the average cumulative daily rate of confirmed cases would also is different significantly. Conversely, to the absence of a significant difference in daily temperature, would correspond no significant difference in the rate of confirmed cases. The hypothesis was investigated through statistical analysis. In order to minimize the impact on the results of factors like the policy of the government or cultural differences among countries (food, exercise, weight, etc.), three case studies within five countries, namely Iran, Italy, Germany, Spain, and United States were compared separately. This comparison using statistical analysis through F-test and Independent-Samples T-Test of the daily temperatures and daily rate of confirmed cases shows that there is a correlation between the average temperature and the epidemic rate, and this is especially evident when differences in average daily temperature are significantly larger, as it happens for Bandar Abbas in Iran, Milan in Italy, Santa Cruz in Spain, and Los Angeles in the US. Besides, the analysis of the average air temperatures shows that the epidemic rates of COVID-19 were higher in the case studies with a lower average temperature. Instead, when no significant differences exist in the average daily temperature of two cities in the same country, there is no significant difference in the average cumulative daily rate of confirmed cases. In conclusion, the results of this study support the research hypothesis and confirm the effectiveness of the proposed method for analysis of the epidemic rates.\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10 . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10.20059337 doi: medRxiv preprint \\n                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
       "4  Metagenomic sequencing, which allows us to directly obtain total genomic DNAs from host-associated and environmental samples, has led to important findings in many areas, such as digestive health [1] . While bacteria are the main focus of most metagenomic sequencing projects, there are fast accumulation of viral metagenomic data with sequencing viruses as the main purpose [2] . There are different types of viruses. We are mainly concerned with RNA viruses because many RNA viruses are notorious human pathogens, such as Influenza A, Human immunodeficiency virus (HIV), Ebola, SARS-CoV, and recently identified 2019-nCoV causing Wuhan pneumonia. Unlike DNA viruses, RNA viruses, which contain RNA genomes, lack faithful proofreading mechanisms during replication and thus can produce a group of related but different viral strains infecting the same host. This high genetic diversity within and across different hosts poses a great challenge for designing long-term protection strategies against these infectious diseases. For example, as the circulating strains can change every year, flu vaccine has to be administered every year.\\nAdvances in viral metagenomics have contributed significantly to new RNA virus discovery. According to a survey by Woolhouse et. al, the number of newly identified RNA viruses is changing from 1,899 in 2005 to 5,561 in 2018, which is 3 times of increase [3] . Largescale RNA virus sequencing projects using nextgeneration sequencing technologies have been conducted for different species. For example, Shi et al. have discovered a large number of new RNA viruses by sequencing samples from invertebrate and vertebrate animals [4, 5] . Claire et al. have discovered 25 new RNA viruses by sequencing samples associated with different drosophilid [6] . Bolduc et al. sequenced RNA viruses from Archaeal and bacterial samples [7] . Given fast accumulation of viral metagenomic data and expected discovery of new viruses, a key step is to conduct composition analysis for these data and assign taxonomic groups for possibly new species.\\nComposition analysis can be conducted at the read level or contig level. There is no doubt that contigs can provide more information than short reads in taxonomic classification. However, metagenomic assembly is still one of the most challenging computational problems in bioinformatics. Unlike single genome assembly, metagenomic assembly is more likely to produce chimeric contigs that can contain regions from different species. Thus, metagenomic assembly is often preceded by read binning that groups reads with the same or similar taxonomic labels [8] in order to achieve better performance.\\nIn this work, we design a read level taxonomic classification tool (named as \"CHEER\") for labeling new species in viral metagenomic data. To detect known species with available reference genomes, alignment-based methods can provide sufficient information for both classification and abundance estimation. This function has been incorporated in existing pipelines [9, 10, 11] and is thus not the focus of our tool. Instead, our tool is designed to handle the challenging cases of assigning taxonomic labels for reads of new species, which have not been observed before. This problem was clearly formulated in metagenomic phylogenetic classification tools such as Phymm and PhymmBL [8] . We first define the classification problem following the formulation in Phymm: the classification problem is the assignment of a phylogenetic group label to each read in input data sets. As we are only interested in the hard case of classifying reads of new species, which have not been sequenced and thus no label exists for these species, the expected labels for these reads are the higher-rank taxonomic groups such as genus, family, or order. All our data and experiments are designed so that the test species are masked in the reference database and the trained models.\\nPrevious experiments have shown that nucleotide-level homology search is not sensitive enough for assigning higher rank taxonomic groups to reads from new species [9, 10] . Protein-based homology search is thus preferred for phylogenetic classification [11, 12] . In particular, a recent tool, VirusSeeker [13] utilized BLASTx to classify reads into bacteria, phages, and other viruses. It is not clear whether the protein homology search is able to generate accurate classification at lower ranks. We will evaluate this in our experiments.\\nThere are also alignment-free methods for phylogenetic classification. Many of those tools implemented different machine learning models and were mainly designed for bacteria. For example, RDP [14] adopted a Naïve Bayes based classifier that can automatically learn features from the reads and assigned taxonomic labels to sequences from bacterial and archaeal 16S genes and fungal 28S genes. The NBC classifier [15, 16] encoded metagenomic reads using k-mers and implemented a bag-of-word model to achieve the readlevel taxonomic classification. Phymm [8] utilized interpolated Markov models (IMMs) for metagenomic read classification. These learning-based models enable users to label reads originating from new species.\\nThe previous works have shown that sequence composition is still an important and informative feature for phylogenetic classification. Recent applications of deep learning models have achieved better performance than conventional machine learning methods in different sequence classification problems [17] . In particular, the convolution filters in Convolutional Neural Network models (CNN) can represent degenerated sequence motifs and thus help CNN models to learn abstract composition-based features. For example, frequently activated convolution filters learned by DeepFam often represented the conserved motifs in different protein domain families [18] . The successful applications of deep learning in sequence classification motivated us to design a novel deep learning-based classification model for assigning taxonomic groups for new species in viral metagenomic data.\\nOur main contributions are summarized below. First, our deep learning model can assign higher rank phylogenetic group labels (such as genus) for reads from new species. To achieve this goal, our model organizes multiple CNN-based classifiers in a tree for hierarchical taxonomic group assignment. Second, to enable our model to capture as much information as possible from short reads, we implemented and compared two encoding methods: one hot vs. embedding. Third, as viral metagenomic data usually contains contaminations from either the host genomes or other microbes, we formulated the pre-processing step as an open set problem in targeted image classification and rejected non-viral reads by choosing appropriate negative training sets. We tested our model on both simulated metagenomic data and also real sequencing data. The results show that our model competes favorably with other popular methods.\\nIn this section, we will introduce our method for viral metagenomic taxonomic classification. First, we will show the architecture of CHEER, which is a hierarchical classification model from order to genus. Second, classifiers at each level will be described. Third, we will introduce DNA sequence encoding using skipgram based word embedding and compare it with onehot encoding. Forth, we will describe the rejection layer, following the idea of the open set problem, which is adopted to filter reads not belonging to RNA viruses. Finally, we will detail our dataset used in training and validation.\\nThe architecture of our model is sketched in Fig. 1. The key component is a tree model that consists of multiple classifiers from order to genus. In order to conduct phylogenetic classification for reads from new species, or even new genus, our classification is conducted using a top-down approach from the root to the leaf node. The top layer is a trained CNN that can reject reads not belonging to RNA viruses, which is an important preprocessing step for viral metagenomic data because of the contamination from the host genome or other microbes. Then, the filtered reads (i.e. mostly RNA viral reads) will be fed into the hierarchical classification model, which is referred to as the tree model hereafter. Specifically, the order-level classifier is responsible for classifying the reads into their originating orders. Then each order has a separately trained classifier for assigning input reads to families within that order (O1 to O3 in Fig. 1 ). For all the reads assigned into one family, the family classifier (F1-F7) will assign the reads to different genus within that family. As an example, a path for classifying a read from a species in genus 7 is highlighted in Fig. 1 . CHEER also implemented early stop functions at each level for stopping the classification path at higher ranks. This function can accommodate taxonomic classification for species from new genus or even higher ranks, which is possible for RNA viruses. It is convenient to add fine-grained classifiers within each genus so that we can assign species-level labels. However, as our focus is to conduct phylogenetic classification for reads from new species that do not have a species labels in the training data, we did not include the species-level assignment in this work.\\nAlternative designs of hierarchical classification work, like [19, 20] , either build a classifier for each taxonomic rank or build a binary classifier for each class. We implemented a structure that built just one classifier for each taxonomic rank and compared different structures' performance. According to the result shown in [Supplementary file 1], the structure shown in Fig.1 is better than one classifier per rank. Although one classifier per rank requires fewer classifiers, it needs to train many more parameters because the number of labels for each classifier increases by times. As a result, we observe overfitting more frequently.\\nEach classifier in the tree model is implemented using CNN, which has achieved superior performance in various sequence classification problems [18] . The convolution filters used in CNN resemble positionspecific weight matrix of motifs in genomic sequences. Frequently activated convolution filters can represent well-conserved motifs among sequences in the same class. Thus, a large number of convolution filters will be used in our CNN in order to learn well-conserved sequence features in different classes. Deep CNNs in fields such as computer vision often have multiple convolution layers in order to obtain more abstract features from training images. Here, we use a wide convolution design rather than a deep one, following the idea in [18] . Fig. 2 shows the classifier with the default hyper-parameters. First, reads will be encoded in matrices as input to the convolution layer using two methods. We will describe and compare these two encoding methods later in the Section 2.3. Then, multiple convolution filters of different sizes are utilized to learn the conserved sequence features. According to our empirical study, we choose 3, 7, 11, 15 as the filter sizes, with 256 filters for each size respectively (see Fig. 2 ). Max pooling is then applied to each convolution filter's output so that only the highest convolution value is kept. After the max-pooling layer, we have a 256-dimentional vector as the output of each convolution layer. We concatenate these four vectors into one 1024-dimentional vector and feed it to one fully connected layer with 512 hidden units. Dropout is applied to mitigate overfitting. The CrossEntropyLoss function supplied by Pytorch is adopted as an error estimation function. Due to the unbalanced size of each training class, we use the reciprocal of the proportion of each class as their error weight to help the model fairly learn from each class. And we utilize Adam optimizer with a 0.001 learning rate to update the parameters. The model is trained on HPCC with the 2080Ti GPU unit. \\nIn previous deep learning-based sequence classification models [21, 22] , one hot encoding was commonly adopted to convert a sequence into a matrix, which will then be used as input to convolution layers. There also exist some methods using k-mer composition [23] and frequency [15, 16] to encode the DNA sequence, similar to the bag-of-words model in the field of natural language processing. However, these k-mers representation cannot give complete sequential information about the original reads.\\nTo incorporate both the k-mer composition and also their ordering information, we build a Skip-Gram [24] based embedding layer that can learn which k-mers tend to occur close to each other. Embedding is widely used in natural language processing to learn the semantic and syntactic relationships from sentences. A neural network with one hidden layer is trained to map a word to an n-dimensional vector so that the words that usually occur together will be closer in the n-dimension space. For our DNA read classification task, k-mers are the words and the embedding layer will map proximate kmers into vectors of high similarity. The main hyper-parameter when training the skip-gram model is the size of the training context m, which defines the maximum context location at which the furthest k-mer is taken for training. Specifically, for a kmer at position i as the input to the skip-gram model, the output k-mers are its neighbors located at i+jk, where −m ≤ ≤ m. Larger m results in more training samples with a cost of training time. In our model, the default value of m is 1. As shown in Fig. 3 , we will sample 3 k-mers as our dataset at a time with the middle one as the input and the surrounding two as the output.\\nAs proven in [24] , Skip-Gram is able to automatically learn the relationship between proximate words and thus is expected to help learn the order between k-mers in genomic data. We implemented a hidden layer with 100 hidden units shown in Fig. 3 . After training the model, the co-occurrence features are embedded in the output of the hidden layer. Thus, a k-mer in a read will be represented as a 100-dimension vector, where each element in the vector is the output of the hidden unit in the hidden layer.\\nWhen we use this pre-trained embedding layer to encode the reads, we first split each read into sequential k-mers. Second, each k-mer is fed to the embedding layer and the output is a 100-dimension vector. Third, all these vectors are combined into a large matrix according to the original position of each k-mers. Finally, this matrix will be fed to the convolutional layer.\\nBased on our large-scale experiments, we found that classifiers are easy to overfit with increase of k. Thus, we choose 100-dimensional feature vectors with k being 3 as our default embedding layer parameter set based on our empirical analysis.\\nDue to the small genome size and limitations in library construction, it is not rare that many viral metagenomic datasets are still contaminated with the genomes of the host and other microbes. Viral read screening is thus a key pre-processing step before the downstream composition analysis. Specifically, our tree model should only be applied to viral reads. However, the close set nature of the classifier forces it to assign a known label to each input without being able to distinguish viral reads from others. In this work, we will reject non-viral reads by formulating it as an open set problem in targeted image classification [25] .\\nIn a majority of neural network models [26, 27] , the logit of the last layer will be fed into the SoftMax function, which takes a vector of n real numbers as input and output a normalized probability vector for all labels/classes. However, for an irrelevant input from an unknown clade, all classes in the model tend to have low probabilities and thus applying a threshold on uncertainty can be used to reject unknown classes [25] . Along with this idea, serval approaches [28, 29] have been proposed to solve the open set problem.\\nIn this work, we incorporate the pre-processing step in our hierarchical classification model by integrating a trained CNN and a SoftMax threshold. Contaminations such as bacteria will be rejected using empirically chosen SoftMax threshold τ. Our experiments show that SoftMax outputs a nearly uniform probability distribution of all labels if the input is not relevant. Thus, the SoftMax threshold τ is used to reject reads not belonging to either DNA or RNA viruses. However, using this thresholding method is not able to distinguish DNA viral reads from RNA viral reads due to their similar sequence composition. To accommodate this challenge, we designed a rejection classifier in the top layer using DNA viral reads as the negative class in the training (see Fig. 1 ).\\nAs the focus of our work is to assign labels for new species, the hierarchical classification ends at genuslevel assignment, which is conducted by the bottom layer in Fig. 1 . Given the fast accumulation of viral metagenomic data, there are cases where a new species does not belong to any existing genus and thus the label assignment should stop at the higher ranks. Thus, we allow our model to stop at a higher rank in the tree during the top-down classification. Towards this goal, we still apply SoftMax threshold in our early stop function, which stops reads from further classification by the child classifiers. This early stop function also serves as a quality control measure for preventing wrong classifications if the confidence is low. For example, when a read arrives, the model may be able to identify its order and family labels with high reliability but cannot decide which genus it belongs to. In this case, the model should only assign order and family label to this read rather than classifying it into a potentially wrong genus. To achieve this goal, SoftMax threshold is applied for every classifier in the tree. Only reads with a score higher than the threshold will be assigned to the child class.\\nAn example of early stop is highlighted by the yellow path in Fig. 1 . The model can classify the read into order 2 (O2) and then family 4 (F4) with high reliability but cannot decide which genus it belongs to.\\nAll the viruses used in the experiments were obtained from the RefSeq in the NCBI virus database. The The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.26.009001 doi: bioRxiv preprint viruses' taxonomic classification was downloaded from Virus Taxonomy 2018b Release [30] published by the International Committee on Taxonomic of Virus (ICVT). Because training deep learning classifiers required enough samples, we removed the orders with only one family, families with only one genus, and genera with less than 3 species from our dataset. Finally, there are 6 orders, 23 families, and 55 genera remained. The RefSeq genomes are available at the following URL:https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/fi nd-data/virus. The information of the taxa can be found in [Supplementary file 1]. As CHEER conducts read-level taxonomic classification, the training is also conducted using reads. In order to train our model on each complete virus genome, we generated simulated reads by uniformly extracting substrings of 250 bp with 200bp overlaps. All these reads formed the training set so that the model can learn the features from the whole genome. The validation set, however, is generated by read simulation tool WgSim [31] with specified error rate so that we can have a better estimation of the model's performance on real sequencing data. Some existing deep learning tools for taxonomic classification, like [21, 23] , generate reads first, then randomly choose some of them as training set and others as testing set. This strategy cannot guarantee that the training and testing data sets have no overlaps from the same species. In order to test whether our model can learn the label of new species, we use test species masking in the training set. Specifically, we split our RNA virus database into two independent sets. For any test species, their reads are not used in the training data. Thus, the model will only use the knowledge of known species for the new species classification.\\nIn this section, we will first give details of the hyperparameters of CHEER. Then we present the results of applying CHEER to both simulated and mock sequencing data. For the simulated data by WgSim, we evaluated how sequencing error rate affects the performance of CHEER and compared the taxonomic classification results with state-of-the-art. For the mock data, we tested CHEER's performance in both taxonomic classification and its ability in distinguishing viral reads from others.\\nIn our experiments, the minimum number of species within a genus is 3. We thus constructed three test sets, each containing reads from one species that is randomly chosen from each genus. The three test sets contain three different species, respectively. In total, 165 species from 55 genera are chosen to be \"new\" species for testing and the information of these species can be found in [Supplementary file 1].\\nWe used WgSim to simulate reads from testing species. In order to evaluate the impact of sequencing error on CHEER, we simulated reads with an error rate of 0, 0.01, 0.02, and 0.05, respectively. For each classifier (CNN), we computed the classification accuracy, which is defined in equation (1). To ensure that the computation of the accuracy is not affected by the number of reads from different species, we constructed a balanced test read set by simulating the same number of reads for each test species (1,000 reads for each). (1) For each component classifier in the tree model, we record the average accuracy for multiple test species. For each taxa level, we calculate the average accuracy of all the classifiers in the same level.\\nWe present the results of CHEER at each level in this section. Because CHEER is designed for new species, we benchmark it against alignment-free models. Based on the results shown by Wood et al. [7] , Naïve Bayes Classifier (NBC) outperformed other alignment-free models. Thus, we compared the performance of CHEER and NBC. NBC in [14] was designed for bacteria taxonomic classification and thus, we retrained NBC on our RNA virus dataset. We implemented the NBC model following [14] with the same parameters. Since each family has a separately trained classifier, which conducts genus classification within that family, there are totally 14 classifiers at this level. Fig. 5 shows the accuracy for labeling families. There are totally 6 classifiers that assign reads to families within each of the 6 orders. Fig. 6 shows the mean accuracy at each rank. The family and genus level accuracy are the average accuracy of all classifiers in Fig. 4 and Fig. 5 . There is only one classifier for order level classification.\\nAs shown in Fig. 4, Fig. 5, and Fig. 6 , CHEER has better performance than NBC across different ranks. We also compared the performance of using one-hot encoding and embedding layer in CHEER. The performance comparison at each rank shows that for most of the classifiers, the embedding layer improves the learning ability. Fig. 4, Fig. 5, and Fig. 6 also revealed the robustness of CHEER. Although the classification accuracy decreases as the error rate increases, the decrease is smaller than 2%. The accuracy is almost identical for many classifiers. Thus, CHEER is not very sensitive to sequencing errors. We also conducted experiments for reads with higher error rates (0.02 and 0.05). The results can be found in [Supplementary file 1].\\nThe next experiment is to evaluate our model on a real sequencing dataset and test CHEER's performance in both taxonomic classification and non-viral read rejection. We built a mock dataset by mixing Ebolavirus reads with gut bacterial reads from a gut amplicon sequencing dataset, which is chosen to ensure that we only use reads from bacteria. The taxonomic classification of Zaire Ebolavirus in the tree model is highlighted in Fig. 7 . We chose one of the Ebola species named Zaire Ebolavirus as our testing species because it is the most dangerous of the five Ebolavirus species and was responsible for the 2014 West African Ebola outbreak. We downloaded the dataset from NCBI with accession number SRR1930021. These reads were sequenced by Illumina using whole-genome amplification strategy. Since Zaire Ebolavirus usually causes fatal hemorrhagic fever and abdominal pain, it is possible that this Ebola virus occurs with other gut bacteria. Thus, we created a mock viral metagenomic data by mixing these Ebola reads with a gut bacteria amplicon sequencing data in order to test whether CHEER can reject the bacterial reads at the top rejection layer. The gut bacteria dataset is downloaded from NCBI with accession number: SRR10714026. There are totally 2,737 Ebola virus reads and 51,280 gut bacterial reads in the mock data. After the rejection layer, the Ebola reads will be used to evaluate our tree model.\\nTo treat Zaire Ebolavirus as a \"new\" species, we removed its genome from our training set and re-trained the whole pipeline. As shown in Fig. 7 , after removing this species, we still have four Ebola species in our training set. However, our read mapping experiment using Bowtie2 [32] showed that none of the reads from Zaire Ebolavirus can be mapped to other Ebola species with the default parameters, indicating that using usual alignment-based methods will have difficulty to conduct genus-level assignment.\\nIn our work, CHEER will first use rejection layer to remove the reads not belonging to the RNA virus. Second, the \"Order classifier\" will be applied to evaluate how many reads can be classified into Mononegavirales order. Third, \"Mononegavirales classifier\" takes all these Mononegavirales reads as input to predict how many reads will be assigned to the Filoviridae family. Finally, \"Filoviridae classifier\" is adopted to check how many reads belong to the Ebola genus. Fig. 8 show the ROC curve of the rejection layer. True positive rate reveals how many Ebolavirus reads pass the rejection layer. The false positive rate reveals how many gut bacterial reads pass the rejection layer. We generated thresholds from 0 to 1 with step size 0.01 and used each threshold for the rejection layer. Then for each threshold, we recorded the true positive rate and false positive rate for plotting the ROC curve. As shown in Fig. 8 , with the increase of the threshold, more bacterial reads are filtered by the rejection layer, at the expense of missing more real Ebola reads. The default threshold for the rejection layer is 0.6 in CHEER. User can adjust the threshold according to their needs.\\nAfter evaluating the rejection layer, we continued to test the hierarchical classification performance at each rank. We first compared our tree model with NBC. Then we also benchmarked CHEER with the method using protein-level alignment in Section 3.3.3. The performance of the complete hierarchical classification pipeline of NBC and CHEER is shown in Fig. 9 . From order to genus, each bar shows the accuracy of correctly classified reads by the parent classifier. For example, NBC correctly classified 1580 at the order level. Then, 1564 out of the 1580 reads are correctly labeled at the family level. While Fig. 4 , 5, and 6 have shown the performance of CHEER and NBC at each taxonomic rank using the same set of reads as input for each level of classifiers, this experiment focused on comparing the hierarchical classification pipeline using only correctly predicted reads by the parent classifier. The results demonstrated that CHEER can classify Ebola reads with better performance. NBC can only classify 1,564 (57.1%) reads into Ebolavirus. CHEER, however, had totally 2407 (87.9%) reads classified correctly.\\nAlthough nucleotide-level alignment models are not ideal choices for predicting new species, as mentioned in [8] , protein-level alignment such as BLASTx is still frequently used for homology search because of the conservation among homologous protein sequences. We followed VirusSeeker [10] , which is a BLASTx based model, to align translated reads to virus-only protein database (VIRUSDB NR) provided by NCBI. Then, we kept alignments with E-value cutoff 0.001, which is used in VirusSeeker. The output of BLASTx shows that a read can often align to multiple proteins, making the phylogenetic classification ambiguous (see Fig. 11 ). Also, existing tools such as VirusSeeker often use the best alignment only, which does not always yield the highest accuracy. To optimize the usage of the alignment-based method, we implemented k-Nearest Neighbor (kNN) algorithm to record the best k alignments rather than only using the best one. The label of the input read is determined by the majority vote in the labels of the top k alignments. According to Fig. 10 , most of the reads have hit numbers between 11 to 15 and 6 to10. Thus, we chose 1, 5, 10, 15 as our k when running kNN. There are also 512 reads that cannot be aligned with any VIRUSDB protein sequence. All these reads will be regarded as wrong prediction. Fig. 11 shows the classification results of kNN. When we use the best hit as the prediction the accuracy is only 64%. The best performance of kNN will be 74% with 2,026 correct predictions when k is 5. But the accuracy sharply decreases if k is larger than 5. Fig. 10 and Fig. 11 also revealed that it is hard for users to choose a proper k in real classification tasks. The most common value of the hit number is 11-15 shown in Fig. 11 . The best kNN result, however, has a k equal to 5. Thus, comparing to this alignment-based method, CHEER, with 2,407 (87.9%) reads classified correctly, is more practical and reliable.\\nThe two main purposes of the early stop function are: 1) to handle the cases of a new species from an unseen genus or higher ranks; 2) to serve as a quality control measure for preventing wrong classifications in child classifiers. As Fig. 6 shows, the higher rank classification tends to be more accurate than lower ranks. Because the genus of Ebolavirus exists in our classification tree, the evaluation of the early stop function focuses on the second purpose. Fig. 12 shows the percentage of how many reads are classified into Ebolavirus genus correctly for different classifiers. By allowing early stop, the number of wrong predictions decreased sharply. In short, although there are fewer reads being labelled at the genus level, the total number of reads with wrong classification is decreased. Thus, this is still a useful feature provided by CHEER.\\nAnother significant advantage of CHEER is the high classification speed. Although training the model uses heavy computing resources, the inference procedure of this algorithm is fast. And with the help of GPU acceleration, the training time also decreases a lot.\\nIn the experiments shown in Section 3.4, it takes around 46 minutes to run BLASTx against the virus protein database. Our method, however, only takes 7 second for the preprocessing step to convert the raw sequence into a vector, and then takes less than 35 second to make the prediction for all 2,737 reads.\\nIn this work, all training data are RNA virus genomes downloaded from the NCBI RefSeq database. The training reads are extracted uniformly from the genome to make sure that classifiers can learn the features from the whole genome. To evaluate the model on new species detection, all species in the test set are excluded from the training data. Error rates are also added to evaluate the robustness of the model. The results reveal that the model is tolerant to the sequencing error. In addition, a SoftMax threshold rejection layer is applied to filter reads from non-RNA viruses. The SoftMax threshold in each classifier will also help early stop classification with low confidence and thus decreases the wrong predication rate. We show a case study on real-world data to identify how this threshold will influence the rejection classifier and early stop function. The results show that CHEER competes favorably with state-of-the-art tools.\\n   \n",
       "\n",
       "   clusters  \n",
       "0  4         \n",
       "1  0         \n",
       "2  2         \n",
       "3  14        \n",
       "4  0         "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['clusters'] = clusters\n",
    "#save the dataframe\n",
    "df.to_csv('clusters_df.csv')\n",
    "df.head() #display the head"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h1 style=\"color:red\">MISSING PLOTTING, TSNE, KEYWORD (THEME) CATEGORIZATION</h1> "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# taged_abs=[]\n",
    "# nouns=[]\n",
    "# # tagging the clean data\n",
    "# for val in clean_abs:\n",
    "#     taged_abs.append(pos_tag(val.split()))\n",
    "# # extracting the nouns\n",
    "# for i in range(len(taged_abs)):\n",
    "#     doc=[]\n",
    "#     for j in range(len(taged_abs[i])):\n",
    "#         if(taged_abs[i][j][1]=='NN' or taged_abs[i][j][1]=='NNS'):\n",
    "#             doc.append(taged_abs[i][j][0])\n",
    "#     nouns.append(doc)\n",
    "        \n",
    "# print(nouns)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Above we dcided to calculate the tf-idf so that we can represent every word that is present in the abstract quatitavily. By doing so we can further use the results in order model the topic according to the abstract that we just quatified.Furthermore we will use (Non-negative Matrix Factorization) NMF in order to come up with topic's that carry most weight in the abstract. To accomplish this we are going to filter all the nouns that are avaliable in the abstract and use them to represent the different topis that are avaliable."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
