{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package stopwords to\n",
      "[nltk_data]     C:\\Users\\HP\\AppData\\Roaming\\nltk_data...\n",
      "[nltk_data]   Package stopwords is already up-to-date!\n",
      "[nltk_data] Downloading package punkt to\n",
      "[nltk_data]     C:\\Users\\HP\\AppData\\Roaming\\nltk_data...\n",
      "[nltk_data]   Package punkt is already up-to-date!\n",
      "[nltk_data] Downloading package wordnet to\n",
      "[nltk_data]     C:\\Users\\HP\\AppData\\Roaming\\nltk_data...\n",
      "[nltk_data]   Package wordnet is already up-to-date!\n"
     ]
    }
   ],
   "source": [
    "import nltk\n",
    "import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\n",
    "import re #import the regular expression library\n",
    "import string\n",
    "import os\n",
    "import json\n",
    "from sklearn.feature_extraction.text import CountVectorizer, TfidfVectorizer\n",
    "from sklearn.metrics.pairwise import cosine_similarity\n",
    "from sklearn.datasets import load_files\n",
    "from nltk import pos_tag\n",
    "from nltk.tokenize import RegexpTokenizer\n",
    "from tqdm import tqdm\n",
    "from nltk.corpus import stopwords #import the stopwords from the ntlk.corpus library\n",
    "nltk.download('stopwords')\n",
    "from nltk.tokenize import word_tokenize #import the word_tokenize method, which is used to turn sentences into words\"\n",
    "nltk.download('punkt')\n",
    "from collections import Counter\n",
    "from nltk.stem import WordNetLemmatizer, PorterStemmer\n",
    "nltk.download('wordnet')\n",
    "import numpy as np\n",
    "import spacy\n",
    "from spacy.lang.en.stop_words import STOP_WORDS\n",
    "import en_core_sci_lg  # model downloaded in previous step"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {
    "_cell_guid": "b1076dfc-b9ad-4769-8c92-a6c4dae69d19",
    "_uuid": "8f2839f25d086af736a60e9eeb907d3b93b6e0e5"
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 1934/1934 [00:04<00:00, 437.77it/s]\n"
     ]
    }
   ],
   "source": [
    "FullPaper=[]#an array which will save all the papers \n",
    "directories = [\"biorxiv_medrxiv\"] #the folder name which contains the papers\n",
    "for directory in directories: #for each of the folders carrying the json format of different research papers\n",
    "    for file in tqdm(os.listdir(f\"{directory}/{directory}/pdf_json\")): #for every json file\n",
    "        file_path=f\"{directory}/{directory}/pdf_json/{file}\" #set the file path to the file_path variable \n",
    "        paper = json.load(open(file_path,\"rb\")) #load the json version of the file\n",
    "        title = paper['metadata']['title'] #set the title variable to the file's title\n",
    "        try:\n",
    "            abstract = paper['abstract']#set the abstract variable to the file's abstract\n",
    "        except:\n",
    "            abstarct=\"\"   #set the abstract variable to the an empty string if the file does not have an abstract             \n",
    "        full_text=\"\" #create an empty string which will hold the body of the paper (file)    \n",
    "        \n",
    "        for text in paper['body_text']: #for every word in the body of the file\n",
    "            full_text += (text['text'] +'\\n') #concatenate the word to the variable full_text\n",
    "        FullPaper.append([title,abstract,full_text]) #since the 3 variable: title, abstract, full_text, now have a value, a paper can be created and added to the FullPaper list\n",
    "        \n",
    "FullPaperDataframe=pd.DataFrame(FullPaper,columns=['title','abstract','full_text'])#create a dataframe which will hold all the papers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "> Our aim is to address these points:\n",
    "* Capabilities to discover a therapeutic (not vaccine) for the disease, and clinical effectiveness studies to discover therapeutics, to include antiviral agents."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 6/6 [00:30<00:00,  5.07s/it]\n"
     ]
    }
   ],
   "source": [
    "full_text = FullPaperDataframe['full_text'].head(6)#fetch all the papers\n",
    "clean_text = [] #a list which will hold all the bodies of the papers after being stripped out of stopwords\n",
    "wnl = WordNetLemmatizer()#word lemmatizer object\n",
    "porter = PorterStemmer()#word stemmer object\n",
    "# punctuation_regex = string.punctuation\n",
    "# punctuation_regex = punctuation_regex.replace(\"-\", \"\") # keep the hyphens\n",
    "# pattern = r\"[{}]\".format(punctuation_regex) # generate the regex pattern\n",
    "pattern = \"\"\"!\"#$%&'()*+,.:;<=>?@[\\]^`{|}~\"\"\" #the pattern which will account for punctuation, and will later be used to remove them\n",
    "#cleaning the data and removing stop words\n",
    "for val in  tqdm(full_text):\n",
    "        body_tokens = word_tokenize(val)\n",
    "        paper_body_without_stopwords = [token.lower() for token in body_tokens if not token.lower() in stopwords.words('english') and nltk.pos_tag([token.lower()])!='PRP'] #remove the stop words in the body and return a list\n",
    "        clean_string = ' '.join(paper_body_without_stopwords).lower() #convert the list into string\n",
    "        clean_string = re.sub(rf\"[{pattern}]\", '', clean_string)#remove punctuation except for hyphens\n",
    "        clean_string = re.sub(r\"\\bthe\\b\", r\"\", clean_string) #remove the\n",
    "        clean_string = re.sub(r'\\bwe\\b', '',clean_string ) #remove the pronoun we\n",
    "        clean_string = re.sub(r'\\bit\\b', '',clean_string )#remove the pronoun it\n",
    "        clean_string = re.sub(r'\\bthey\\b', '',clean_string )#remove the pronoun they\n",
    "        clean_string = re.sub(r'\\bcopyright\\b', '',clean_string )#remove the word copyright\n",
    "        clean_string = re.sub(r'\\bhttps\\b', '',clean_string )#remove the word https\n",
    "        clean_string = re.sub(r'\\bet\\b', '',clean_string )#remove the word et\n",
    "        clean_string = re.sub(r'\\bal\\b', '',clean_string )#remove the word al\n",
    "        clean_string = re.sub(r'\\bpreprint\\b', '',clean_string )#remove the word preprint\n",
    "        clean_string = re.sub(r'\\bthis\\b', '',clean_string )#remove the word this\n",
    "        clean_string = re.sub(r'\\bthese\\b', '',clean_string )#remove the word these\n",
    "        clean_string = re.sub(r'\\balso\\b', '',clean_string )#remove the word also\n",
    "        clean_string = re.sub(r'\\bhowever\\b', '',clean_string )#remove the word however\n",
    "        clean_string = re.sub(r'\\balthough\\b', '',clean_string )#remove the word although\n",
    "        clean_string = re.sub(r'\\bamong\\b', '',clean_string )#remove the word among\n",
    "        clean_string = re.sub(r'\\bin\\b', '',clean_string )#remove the preposition in\n",
    "        clean_string = re.sub(r'\\bmedrxiv\\b', '',clean_string )#remove the word medrxiv\n",
    "        clean_string = re.sub(r'\\bbiorxiv\\b', '',clean_string )#remove the word mbiorxiv\n",
    "        clean_string = re.sub(r'\\blicense\\b', '',clean_string )#remove the word license\n",
    "        clean_string = re.sub(r'\\bwithout\\b', '',clean_string )#remove the word without\n",
    "        clean_string = re.sub(r'\\bfig\\b', '',clean_string )#remove the word fig\n",
    "        clean_string = re.sub(r'\\bfigure\\b', '', clean_string)#remove the word figure\n",
    "        clean_string = re.sub(r'\\b[a-zA-Z]\\b', '', clean_string)#remove all the single letters      \n",
    "\n",
    "        clean_string = re.sub(r'^\\d+\\s|\\s\\d+\\s|\\s\\d+$', '', clean_string) #remove any digit that is not part of a word\n",
    "        clean_text.append(clean_string) #add the string to the list \n",
    "stemmed_text=[] # a list which will contain the stemmed bodies of all docs      \n",
    "for val in clean_text: #for the body (which was stripped out of stop words) of each paper\n",
    "    stemmed_val=\"\" #create an empty string\n",
    "    words = val.split(' ') #split body of paper on spaces\n",
    "    for word in words: #for each word in the body\n",
    "        if(word!=\"\"): #if the word is not an empty string\n",
    "            word = word.lower()\n",
    "            if (word==\"coronaviruses\"): #since the stem and lemmatize functions do not recognize the word coronaviruses as the plural of coronavirus, a conidition was created\n",
    "                word =\"coronavirus\"\n",
    "            stem_lemma = porter.stem(word)\n",
    "            stemmed_val+=stem_lemma+\" \" #stem the word and concat it to the string stemmed_val\n",
    "    stemmed_text.append(stemmed_val) #add the string stemmed_val to the list which will contain the stemmed bodies of all docs  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [],
   "source": [
    "# this cell is no longer needed and is kept for reference\n",
    "\n",
    "\n",
    "text = ' '.join(stemmed_text) #convert all the stemmed bodies into one string containing all the stemmed bodies of all papers \n",
    "s=text.split(' ')#split the combination of papers body on space\n",
    "vocab_count = Counter(s)# count the number of vocab (unique words)\n",
    "most_occur = vocab_count.most_common(200) #get the most common 200 words\n",
    "#print(most_occur) #print the most common 200 words"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[('001_2020', 0), ('002_2020', 1), ('003_2020', 2), ('004_2020', 3), ('005', 4), ('005_2020', 5), ('012_2020', 6), ('01_2019', 7), ('01_2020', 8), ('020_2020', 9), ('027_2020', 10), ('029_2020', 11), ('02_2019', 12), ('02_2020', 13), ('03021_2020', 14), ('03_2019', 15), ('03_2020', 16), ('04_2019', 17), ('04_2020', 18), ('05', 19), ('05_2020', 20), ('06_2020', 21), ('07', 22), ('07_2020', 23), ('09_2020', 24), ('100', 25), ('101101', 26), ('1024', 27), ('11', 28), ('110', 29), ('131', 30), ('1311', 31), ('133', 32), ('1331', 33), ('15', 34), ('1580', 35), ('16', 36), ('1645', 37), ('17', 38), ('18', 39), ('19', 40), ('1929', 41), ('19day', 42), ('1_2019', 43), ('1_2020', 44), ('1a', 45), ('1b', 46), ('1c', 47), ('20', 48), ('200bp', 49), ('2010', 50), ('2018b', 51), ('2019', 52), ('2020', 53), ('20200110901801', 54), ('202003', 55), ('20200326009001', 56), ('202004', 57), ('2020040820058180', 58), ('2080ti', 59), ('20_2020', 60), ('21_2020', 61), ('22', 62), ('256', 63), ('25_2020', 64), ('28', 65), ('29', 66), ('2_2020', 67), ('2a', 68), ('2b', 69), ('2c', 70), ('33', 71), ('38142_2020', 72), ('3a', 73), ('3b', 74), ('3c', 75), ('3d', 76), ('467', 77), ('501_2020', 78), ('521_2020', 79), ('5b', 80), ('675', 81), ('99comcn', 82), ('a12_2020', 83), ('aatii', 84), ('aatiiinsert', 85), ('aba', 86), ('abba', 87), ('abdomin', 88), ('abil', 89), ('abl', 90), ('absenc', 91), ('abstract', 92), ('abund', 93), ('acceler', 94), ('accept', 95), ('access', 96), ('accommod', 97), ('accord', 98), ('account', 99), ('accumul', 100), ('accur', 101), ('accuraci', 102), ('achiev', 103), ('acid', 104), ('across', 105), ('acrosscountri', 106), ('activ', 107), ('actual', 108), ('ad', 109), ('adam', 110), ('add', 111), ('addit', 112), ('address', 113), ('adequ', 114), ('adjust', 115), ('administ', 116), ('admiss', 117), ('adopt', 118), ('advanc', 119), ('advantag', 120), ('advers', 121), ('affect', 122), ('aflii', 123), ('african', 124), ('against', 125), ('age', 126), ('agent', 127), ('aggreg', 128), ('aglidjian', 129), ('agre', 130), ('aid', 131), ('aii', 132), ('aim', 133), ('air', 134), ('algorithm', 135), ('alighn', 136), ('align', 137), ('alignment', 138), ('alik', 139), ('alleg', 140), ('allow', 141), ('almost', 142), ('along', 143), ('alreadi', 144), ('altern', 145), ('alway', 146), ('ambigu', 147), ('american', 148), ('amid', 149), ('amongst', 150), ('amount', 151), ('amplicon', 152), ('amplif', 153), ('amplifi', 154), ('analog', 155), ('analys', 156), ('analysi', 157), ('analysiscountri', 158), ('analyt', 159), ('analyz', 160), ('ancestr', 161), ('anecdot', 162), ('angel', 163), ('anim', 164), ('announc', 165), ('anoth', 166), ('anova', 167), ('answer', 168), ('appear', 169), ('appli', 170), ('applic', 171), ('approach', 172), ('appropri', 173), ('approxim', 174), ('approximately71', 175), ('arbitrarili', 176), ('archaeal', 177), ('architectur', 178), ('area', 179), ('aroundminut', 180), ('arriv', 181), ('arrow', 182), ('art', 183), ('articl', 184), ('artifici', 185), ('ashley', 186), ('asian', 187), ('aspect', 188), ('assay', 189), ('assembl', 190), ('assembledauthor', 191), ('assess', 192), ('assessmentpati', 193), ('assign', 194), ('associ', 195), ('assum', 196), ('assumpt', 197), ('attent', 198), ('audienc', 199), ('auroc', 200), ('australia', 201), ('author', 202), ('automat', 203), ('avail', 204), ('averag', 205), ('avoid', 206), ('bacteri', 207), ('bacteria', 208), ('bag', 209), ('baidu', 210), ('balanc', 211), ('bandar', 212), ('bar', 213), ('barcelona', 214), ('bas', 215), ('base', 216), ('baselin', 217), ('basic', 218), ('bay', 219), ('bayesian', 220), ('becom', 221), ('bed', 222), ('began', 223), ('begin', 224), ('behavior', 225), ('beij', 226), ('believ', 227), ('belong', 228), ('benchmark', 229), ('besid', 230), ('best', 231), ('better', 232), ('bi', 233), ('bia', 234), ('bic', 235), ('bin', 236), ('binari', 237), ('bioinformat', 238), ('blast', 239), ('blastx', 240), ('boast', 241), ('bodi', 242), ('bolduc', 243), ('bootstrap', 244), ('bottleneck', 245), ('bottom', 246), ('boundari', 247), ('bowtie2', 248), ('branch', 249), ('brazil', 250), ('breath', 251), ('brian', 252), ('brief', 253), ('build', 254), ('built', 255), ('bukhari', 256), ('by', 257), ('c0', 258), ('c11', 259), ('calcul', 260), ('calibr', 261), ('can', 262), ('captur', 263), ('care', 264), ('carri', 265), ('cascella', 266), ('case', 267), ('categor', 268), ('caus', 269), ('caution', 270), ('cc', 271), ('cell', 272), ('cent', 273), ('center', 274), ('certain', 275), ('cgu', 276), ('challeng', 277), ('chan', 278), ('chang', 279), ('changing20052018', 280), ('changsha', 281), ('character', 282), ('characterist', 283), ('check', 284), ('cheer', 285), ('chen', 286), ('chi', 287), ('child', 288), ('chimer', 289), ('china', 290), ('chinacountri', 291), ('chinalat', 292), ('choic', 293), ('chongq', 294), ('choos', 295), ('choosefilt', 296), ('chose', 297), ('chosen', 298), ('circul', 299), ('citi', 300), ('clade', 301), ('cladogram', 302), ('claim', 303), ('clair', 304), ('class', 305), ('classif', 306), ('classifi', 307), ('classifiedord', 308), ('clear', 309), ('clearli', 310), ('climat', 311), ('clinic', 312), ('clock', 313), ('clon', 314), ('close', 315), ('closer', 316), ('cluster', 317), ('cnn', 318), ('co', 319), ('cod', 320), ('cohort', 321), ('colin', 322), ('collabor', 323), ('collect', 324), ('combin', 325), ('commiss', 326), ('committe', 327), ('common', 328), ('commonli', 329), ('commun', 330), ('compar', 331), ('comparedauthor', 332), ('comparison', 333), ('compet', 334), ('competit', 335), ('competitivedisadvantag', 336), ('complet', 337), ('complex', 338), ('compon', 339), ('composit', 340), ('composition', 341), ('compris', 342), ('comput', 343), ('concaten', 344), ('concern', 345), ('concert', 346), ('conclus', 347), ('condit', 348), ('conduct', 349), ('confer', 350), ('confid', 351), ('confirm', 352), ('conflict', 353), ('conga', 354), ('connect', 355), ('consensu', 356), ('consequ', 357), ('conserv', 358), ('consid', 359), ('consider', 360), ('consist', 361), ('consistedgenomescountri', 362), ('consistentrequir', 363), ('construct', 364), ('consult', 365), ('contagi', 366), ('contagion', 367), ('contain', 368), ('contained29564', 369), ('containingsingl', 370), ('contamin', 371), ('context', 372), ('contig', 373), ('conting', 374), ('continu', 375), ('contribut', 376), ('contributed63', 377), ('control', 378), ('controversi', 379), ('conveni', 380), ('convent', 381), ('convers', 382), ('convert', 383), ('convolut', 384), ('copi', 385), ('corner', 386), ('coronaviru', 387), ('correct', 388), ('correctli', 389), ('correl', 390), ('correspond', 391), ('cost', 392), ('cough', 393), ('could', 394), ('count', 395), ('countri', 396), ('cours', 397), ('cov', 398), ('cover', 399), ('coverag', 400), ('covid', 401), ('cq', 402), ('creat', 403), ('crisi', 404), ('crit', 405), ('criterion', 406), ('critic', 407), ('criticized', 408), ('crossentropyloss', 409), ('crucial', 410), ('cruz', 411), ('cs_2020', 412), ('cultur', 413), ('cumul', 414), ('current', 415), ('curv', 416), ('cutoff', 417), ('cycl', 418), ('daili', 419), ('damag', 420), ('dambe6', 421), ('danger', 422), ('dashboard', 423), ('data', 424), ('databas', 425), ('dataset', 426), ('datasuggest', 427), ('date', 428), ('day', 429), ('de', 430), ('death', 431), ('debat', 432), ('decemb', 433), ('decembernovel', 434), ('decid', 435), ('decis', 436), ('decker', 437), ('declin', 438), ('decreas', 439), ('deep', 440), ('deepfam', 441), ('default', 442), ('defect', 443), ('defin', 444), ('definit', 445), ('degener', 446), ('degre', 447), ('delay', 448), ('delet', 449), ('demograph', 450), ('demonstr', 451), ('depend', 452), ('deploy', 453), ('deposit', 454), ('describ', 455), ('design', 456), ('despit', 457), ('detail', 458), ('detect', 459), ('determin', 460), ('develop', 461), ('deviat', 462), ('devic', 463), ('diagnosi', 464), ('differ', 465), ('differencescould', 466), ('difficulti', 467), ('digest', 468), ('digestedspei', 469), ('dimens', 470), ('dimension', 471), ('diment', 472), ('diminish', 473), ('direct', 474), ('directli', 475), ('disabl', 476), ('disappear', 477), ('discard', 478), ('discharg', 479), ('discov', 480), ('discoverednew', 481), ('discoveri', 482), ('discrimin', 483), ('discuss', 484), ('diseas', 485), ('display', 486), ('disrupt', 487), ('dissemin', 488), ('distanc', 489), ('distance', 490), ('distanceshow', 491), ('distinct', 492), ('distinguish', 493), ('distress', 494), ('distribut', 495), ('diverg', 496), ('divers', 497), ('diversif', 498), ('divid', 499), ('dna', 500), ('dnadigest', 501), ('dnase', 502), ('doi', 503), ('doiorg', 504), ('domain', 505), ('done', 506), ('donna', 507), ('dot', 508), ('doubt', 509), ('down', 510), ('download', 511), ('downstream', 512), ('dr', 513), ('dramat', 514), ('drawn', 515), ('dropout', 516), ('drosophilid', 517), ('drsaverag', 518), ('due', 519), ('duplic', 520), ('dynam', 521), ('earli', 522), ('earlier', 523), ('easi', 524), ('easili', 525), ('ebola', 526), ('ebolaviru', 527), ('edinburgh', 528), ('editori', 529), ('effect', 530), ('effectreplicon', 531), ('effort', 532), ('eg', 533), ('eicu', 534), ('either', 535), ('electron', 536), ('element', 537), ('embed', 538), ('emerg', 539), ('emphasis', 540), ('empir', 541), ('employe', 542), ('enabl', 543), ('encod', 544), ('end', 545), ('enough', 546), ('ensu', 547), ('ensur', 548), ('entir', 549), ('envelop', 550), ('envf13', 551), ('environ', 552), ('environment', 553), ('epi_isl_402119_betacov_wuhan_ivdc', 554), ('epi_isl_402120_betacov_wuhan_ivdc', 555), ('epi_isl_402123_betacov_wuhan_ipbcams', 556), ('epi_isl_402125_betacov_wuhan', 557), ('epi_isl_402127_betacov_wuhan_wiv02_2019', 558), ('epi_isl_402129_betacov_wuhan_wiv06_2019', 559), ('epi_isl_402132_betacov_wuhan_hbcdc', 560), ('epi_isl_403928_betacov_wuhan_ipbcams', 561), ('epi_isl_403929_betacov_wuhan_ipbcams', 562), ('epi_isl_403930_betacov_wuhan_ipbcams', 563), ('epi_isl_403931_betacov_wuhan_ipbcams', 564), ('epi_isl_403932_betacov_guangdong_20sf012_2020', 565), ('epi_isl_403933_betacov_guangdong_20sf013_2020', 566), ('epi_isl_403934_betacov_guangdong_20sf014_2020', 567), ('epi_isl_403935_betacov_guangdong_20sf025_2020', 568), ('epi_isl_403936_betacov_guangdong_20sf028_2020', 569), ('epi_isl_403937_betacov_guangdong_20sf040_2020', 570), ('epi_isl_403962_betacov_nonthaburi_61_2020', 571), ('epi_isl_403963_betacov_nonthaburi_74_2020', 572), ('epi_isl_404227_betacov_zhejiang_wz', 573), ('epi_isl_404228_betacov_zhejiang_wz', 574), ('epi_isl_404253_betacov_usa_il1_2020', 575), ('epi_isl_404895_betacov_usa_wa1_2020', 576), ('epi_isl_405839_betacov_shenzhen_hku', 577), ('epi_isl_406030_betacov_shenzhen_hku', 578), ('epi_isl_406031_betacov_taiwan_2_2020', 579), ('epi_isl_406034_betacov_usa_ca1_2020', 580), ('epi_isl_406036_betacov_usa_ca2_2020', 581), ('epi_isl_406223_betacov_usa_az1_2020', 582), ('epi_isl_406531_betacov_guangdong_20sf174_2020', 583), ('epi_isl_406533_betacov_guangzhou_20sf206_2020', 584), ('epi_isl_406534_betacov_foshan_20sf207_2020', 585), ('epi_isl_406535_betacov_foshan_20sf210_2020', 586), ('epi_isl_406536_betacov_foshan_20sf211_2020', 587), ('epi_isl_406538_betacov_guangdong_20sf201_2020', 588), ('epi_isl_406593_betacov_shenzhen_szth', 589), ('epi_isl_406594_betacov_shenzhen_szth', 590), ('epi_isl_406596_betacov_france_idf0372_2020', 591), ('epi_isl_406597_betacov_france_idf0373_2020', 592), ('epi_isl_406716_betacov_china_whu01_2020', 593), ('epi_isl_406717_betacov_china_whu02_2020', 594), ('epi_isl_406798_betacov_wuhan_wh01_2019', 595), ('epi_isl_406801_betacov_wuhan_wh04_2020', 596), ('epi_isl_406844_betacov_australia_vic01_2020', 597), ('epi_isl_406862', 598), ('epi_isl_406862_betacov_germany_bavpat1_2020', 599), ('epi_isl_406970_betacov_hangzhou_hz', 600), ('epi_isl_406973_betacov_singapore_1_2020', 601), ('epi_isl_407071_betacov_england_01_2020', 602), ('epi_isl_407073_betacov_england_02_2020', 603), ('epi_isl_407079_betacov_finland_1_2020', 604), ('epi_isl_407084_betacov_japan_ai_i', 605), ('epi_isl_407193_betacov_south_korea_kcdc03_2020', 606), ('epi_isl_407214_betacov_usa_wa1', 607), ('epi_isl_407215_betacov_usa_wa1', 608), ('epi_isl_407313_betacov_hangzhou_hzcdc0001_2020', 609), ('epi_isl_407893_betacov_australia_nsw01_2020', 610), ('epi_isl_407894_betacov_australia_qld01_2020', 611), ('epi_isl_407896_betacov_australia_qld02_2020', 612), ('epi_isl_407976_betacov_belgium_ghb', 613), ('epi_isl_407987_betacov_singapore_2_2020', 614), ('epi_isl_407988_betacov_singapore_3_2020', 615), ('epi_isl_408008_betacov_usa_ca3_2020', 616), ('epi_isl_408009_betacov_usa_ca4_2020', 617), ('epi_isl_408010_betacov_usa_ca5_2020', 618), ('epi_isl_408430_betacov_france_idf0515_2020', 619), ('epi_isl_408431_betacov_france_idf0626_2020', 620), ('epi_isl_408478_betacov_chongqing_yc01_2020', 621), ('epi_isl_408479_betacov_chongqing_zx01_2020', 622), ('epi_isl_408480_betacov_yunnan_ivdc', 623), ('epi_isl_408481_betacov_chongqing_ivdc', 624), ('epi_isl_408484_betacov_sichuan_ivdc', 625), ('epi_isl_408486_betacov_jiangxi_ivdc', 626), ('epi_isl_408488_betacov_jiangsu_ivdc', 627), ('epi_isl_408489_betacov_taiwan_ntu01_2020', 628), ('epi_isl_408514_betacov_wuhan_ivdc', 629), ('epi_isl_408515_betacov_wuhan_ivdc', 630), ('epi_isl_408665_betacov_japan_ty', 631), ('epi_isl_408666_betacov_japan_ty', 632), ('epi_isl_408667_betacov_japan_ty', 633), ('epi_isl_408668_betacov_vietnam_vr03', 634), ('epi_isl_408669_betacov_japan_ky', 635), ('epi_isl_408670_betacov_usa_wi1_2020', 636), ('epi_isl_408976_betacov_sydney_2_2020', 637), ('epi_isl_408977_betacov_sydney_3_2020', 638), ('epi_isl_408978_betacov_wuhan_wh05_2020', 639), ('epi_isl_409067_betacov_usa_ma1_2020', 640), ('epi_isl_410044_betacov_usa_ca6_2020', 641), ('epi_isl_410045_betacov_usa_il2_2020', 642), ('epi_isl_410218_betacov_taiwan_ntu02_2020', 643), ('epi_isl_410301_betacov_nepal_61_2020', 644), ('epi_isl_410486_betacov_france_ra739_2020', 645), ('epi_isl_410531_betacov_japan_na', 646), ('epi_isl_410532_betacov_japan_os', 647), ('epi_isl_410535_betacov_singapore_4_2020', 648), ('epi_isl_410536_betacov_singapore_5_2020', 649), ('epi_isl_410537_betacov_singapore_6_2020', 650), ('epi_isl_410545_betacov_italy_inmi1', 651), ('epi_isl_410546_betacov_italy_inmi1', 652), ('epi_isl_410713_betacov_singapore_7_2020', 653), ('epi_isl_410714_betacov_singapore_8_2020', 654), ('epi_isl_410715_betacov_singapore_9_2020', 655), ('epi_isl_410716_betacov_singapore_10_2020', 656), ('epi_isl_410717_betacov_australia_qld03_2020', 657), ('epi_isl_410718_betacov_australia_qld04_2020', 658), ('epi_isl_410720_betacov_france_idf0372', 659), ('epi_isl_410984_betacov_france_idf0515', 660), ('epi_isl_411060_betacov_fujian_8_2020', 661), ('epi_isl_411066_betacov_fujian_13_2020', 662), ('epi_isl_411218_betacov_france_idf0571_2020', 663), ('epi_isl_411219_betacov_france_idf0386', 664), ('epi_isl_411220_betacov_france_idf0386', 665), ('epi_isl_411902_betacov_cambodia_0012_2020', 666), ('epi_isl_411915_betacov_taiwan_cgmh', 667), ('epi_isl_411926_betacov_taiwan_3_2020', 668), ('epi_isl_411927_betacov_taiwan_4_2020', 669), ('epi_isl_411950_betacov_jiangsu_js01_2020', 670), ('epi_isl_411951_betacov_sweden_01_2020', 671), ('epi_isl_411952_betacov_jiangsu_js02_2020', 672), ('epi_isl_411953_betacov_jiangsu_js03_2020', 673), ('epi_isl_411954_betacov_usa_ca7_2020', 674), ('epi_isl_411955_betacov_usa_ca8_2020', 675), ('epi_isl_411956_betacov_usa_tx1_2020', 676), ('epi_isl_411957_betacov_china_wh', 677), ('epi_isl_412026_betacov_hefei_2_2020', 678), ('epi_isl_412028_betacov_hong_kong_vm20001061_2020', 679), ('epi_isl_412029_betacov_hong_kong_vm20001988_2020', 680), ('epi_isl_412030_betacov_hong_kong_vb20026565_2020', 681), ('epi_isl_412459_betacov_jingzhou_hbcdc', 682), ('epi_isl_412862_betacov_usa_ca9_2020', 683), ('epi_isl_412869_betacov_korea_kcdc05_2020', 684), ('epi_isl_412870_betacov_korea_kcdc06_2020', 685), ('epi_isl_412871_betacov_korea_kcdc07_2020', 686), ('epi_isl_412872_betacov_korea_kcdc12_2020', 687), ('epi_isl_412873_betacov_korea_kcdc24_2020', 688), ('epi_isl_412898__betacov_wuhan_hbcdc', 689), ('epi_isl_412899__betacov_wuhan_hbcdc', 690), ('epi_isl_412912_betacov_germany_baden', 691), ('epi_isl_412964', 692), ('epi_isl_412965_betacov_canada_bc_37_0', 693), ('epi_isl_412966_betacov_china_iqtc01_2020', 694), ('epi_isl_412967_betacov_china_iqtc02_2020', 695), ('epi_isl_412968_betacov_japan_hu_dp_kng_19', 696), ('epi_isl_412969_betacov_japan_hu_dp_kng_19', 697), ('epi_isl_412970_betacov_usa_wa2_2020', 698), ('epi_isl_412971_betacov_finland_fin', 699), ('epi_isl_412972_betacov_mexico_cdmx_indre_01_2020', 700), ('epi_isl_412973', 701), ('epi_isl_412973_betacov_italy_cdg1_2020', 702), ('epi_isl_412974_betacov_italy_spl1_2020', 703), ('epi_isl_412975_betacov_australia_nsw05_2020', 704), ('epi_isl_412978__betacov_wuhan_hbcdc', 705), ('epi_isl_412979__betacov_wuhan_hbcdc', 706), ('epi_isl_412980__betacov_wuhan_hbcdc', 707), ('epi_isl_412981__betacov_wuhan_hbcdc', 708), ('epi_isl_412982__betacov_wuhan_hbcdc', 709), ('epi_isl_412983_betacov_tianmen_hbcdc', 710), ('epi_isl_413014_betacov_canada_on', 711), ('epi_isl_413015_betacov_canada_on_vido', 712), ('epi_isl_413016_betacov_brazil_spbr', 713), ('epi_isl_413017_betacov_south_korea_kumc01_2020', 714), ('epi_isl_413018_betacov_south_korea_kumc02_2020', 715), ('epi_isl_413019_betacov_switzerland_1000477102_2020', 716), ('epi_isl_413020_betacov_switzerland_1000477377_2020', 717), ('epi_isl_413555', 718), ('epi_isl_413648', 719), ('epicent', 720), ('epidem', 721), ('epidemiolog', 722), ('equal', 723), ('equat', 724), ('equilater', 725), ('error', 726), ('especi', 727), ('essenti', 728), ('estim', 729), ('etc', 730), ('ethnic', 731), ('etiolog', 732), ('european', 733), ('evalu', 734), ('even', 735), ('event', 736), ('ever', 737), ('everi', 738), ('evid', 739), ('evidence', 740), ('evidenceloop', 741), ('evolut', 742), ('evolutionari', 743), ('evolv', 744), ('exampl', 745), ('exclud', 746), ('exercis', 747), ('exist', 748), ('expans', 749), ('expect', 750), ('expens', 751), ('experi', 752), ('experienc', 753), ('expert', 754), ('explicitli', 755), ('explor', 756), ('exponenti', 757), ('exposed', 758), ('extens', 759), ('extern', 760), ('extra', 761), ('extract', 762), ('extrem', 763), ('f1', 764), ('f4', 765), ('f6_2020', 766), ('f7', 767), ('face', 768), ('facilit', 769), ('factor', 770), ('failur', 771), ('fairli', 772), ('faith', 773), ('fake', 774), ('fals', 775), ('famili', 776), ('far', 777), ('fast', 778), ('faster', 779), ('fastq', 780), ('fastqc', 781), ('fatal', 782), ('favor', 783), ('featur', 784), ('featuresseri', 785), ('feb', 786), ('februari', 787), ('february15', 788), ('februaryth', 789), ('fed', 790), ('feed', 791), ('fever', 792), ('fewer', 793), ('fi', 794), ('field', 795), ('fifteen', 796), ('fig1', 797), ('fight', 798), ('figur', 799), ('figures6', 800), ('file', 801), ('filovirida', 802), ('filter', 803), ('final', 804), ('find', 805), ('fine', 806), ('finland', 807), ('first', 808), ('firstnucleotid', 809), ('fit', 810), ('five', 811), ('flip', 812), ('flu', 813), ('fmdv', 814), ('fmdvlibrari', 815), ('fmdvread', 816), ('focu', 817), ('focus', 818), ('fold', 819), ('follow', 820), ('food', 821), ('forc', 822), ('form', 823), ('format', 824), ('formul', 825), ('forth', 826), ('forthcom', 827), ('forum', 828), ('found', 829), ('four', 830), ('fragment', 831), ('framework', 832), ('fre', 833), ('free', 834), ('freedom', 835), ('frequenc', 836), ('frequent', 837), ('fuel', 838), ('full', 839), ('fulli', 840), ('function', 841), ('fund', 842), ('fungal', 843), ('furthermor', 844), ('furthest', 845), ('futur', 846), ('gap', 847), ('gender', 848), ('gene', 849), ('gener', 850), ('genera', 851), ('genet', 852), ('genom', 853), ('genomerepl', 854), ('genu', 855), ('genus', 856), ('genushighlight', 857), ('genuslevel', 858), ('geograph', 859), ('germani', 860), ('get', 861), ('giovannucci', 862), ('gisaid', 863), ('give', 864), ('given', 865), ('glasgow', 866), ('global', 867), ('glut', 868), ('gnn', 869), ('goal', 870), ('gottfri', 871), ('govern', 872), ('gpu', 873), ('grain', 874), ('gram', 875), ('grant', 876), ('granular', 877), ('great', 878), ('greater', 879), ('greatest', 880), ('group', 881), ('grouppandem', 882), ('growingproduc', 883), ('guarante', 884), ('gut', 885), ('h0', 886), ('h1', 887), ('handl', 888), ('happen', 889), ('hard', 890), ('hasegawa', 891), ('hasgawa', 892), ('hazard', 893), ('hb', 894), ('health', 895), ('healthcar', 896), ('healthcare', 897), ('healthcareresource', 898), ('heat', 899), ('heavi', 900), ('help', 901), ('hemorrhag', 902), ('henc', 903), ('hereaft', 904), ('heterogen', 905), ('hf', 906), ('hidden', 907), ('hierarch', 908), ('high', 909), ('higher', 910), ('highest', 911), ('highli', 912), ('highlight', 913), ('hindiii', 914), ('hindranc', 915), ('hispan', 916), ('histor', 917), ('hit', 918), ('hiv', 919), ('hl7', 920), ('hold', 921), ('holder', 922), ('homolog', 923), ('hopkin', 924), ('hospit', 925), ('host', 926), ('hot', 927), ('hotspot', 928), ('hour', 929), ('hourli', 930), ('hpcc', 931), ('http', 932), ('hu', 933), ('hubei', 934), ('human', 935), ('humid', 936), ('hybridis', 937), ('hype', 938), ('hyper', 939), ('hyperparamet', 940), ('hypothes', 941), ('hypothesi', 942), ('hypothesis', 943), ('icu', 944), ('icvt', 945), ('idba', 946), ('idea', 947), ('ideal', 948), ('ident', 949), ('identifi', 950), ('idi', 951), ('ie', 952), ('ijk', 953), ('ill', 954), ('illumina', 955), ('imag', 956), ('imm', 957), ('immedi', 958), ('immunodefici', 959), ('impact', 960), ('implement', 961), ('implic', 962), ('import', 963), ('importantli', 964), ('improv', 965), ('inadequ', 966), ('includ', 967), ('incorpor', 968), ('increas', 969), ('incub', 970), ('inde', 971), ('independ', 972), ('independent', 973), ('index', 974), ('indic', 975), ('individu', 976), ('infect', 977), ('infecti', 978), ('infectious', 979), ('infer', 980), ('influenc', 981), ('influenza', 982), ('inform', 983), ('inhomogen', 984), ('initi', 985), ('input', 986), ('insert', 987), ('insid', 988), ('insight', 989), ('instal', 990), ('instead', 991), ('instruct', 992), ('insuffici', 993), ('integr', 994), ('intellig', 995), ('intend', 996), ('interact', 997), ('interactauthor', 998), ('interest', 999), ('interfac', 1000), ('intern', 1001), ('internationally', 1002), ('interpol', 1003), ('interpret', 1004), ('interval14', 1005), ('introduc', 1006), ('introduct', 1007), ('invas', 1008), ('invertebr', 1009), ('investig', 1010), ('involv', 1011), ('iq', 1012), ('iran', 1013), ('irrelev', 1014), ('irrespons', 1015), ('isl_2020', 1016), ('islp1_2020', 1017), ('islp3_2020', 1018), ('isol', 1019), ('isolatescontain', 1020), ('itali', 1021), ('italian', 1022), ('januari', 1023), ('januaryaccommod', 1024), ('januaryproduc', 1025), ('jiujiang', 1026), ('john', 1027), ('js', 1028), ('jx', 1029), ('kampf', 1030), ('keep', 1031), ('kept', 1032), ('key', 1033), ('kit', 1034), ('kmer', 1035), ('knn', 1036), ('knncorrect', 1037), ('knowledg', 1038), ('known', 1039), ('korea', 1040), ('kpni', 1041), ('kurt', 1042), ('label', 1043), ('labels', 1044), ('laboratori', 1045), ('labs', 1046), ('lack', 1047), ('languag', 1048), ('larg', 1049), ('large', 1050), ('larger', 1051), ('largescal', 1052), ('largest', 1053), ('lasergene229', 1054), ('last', 1055), ('lastli', 1056), ('later', 1057), ('latitud', 1058), ('layer', 1059), ('layerch', 1060), ('layerhidden', 1061), ('lazio', 1062), ('lead', 1063), ('leaf', 1064), ('learn', 1065), ('learning', 1066), ('least', 1067), ('leav', 1068), ('led', 1069), ('length', 1070), ('lengthkb', 1071), ('less', 1072), ('lessconfirm', 1073), ('lesssecond', 1074), ('lessspeci', 1075), ('let', 1076), ('level', 1077), ('levelconsid', 1078), ('levelrepl', 1079), ('librari', 1080), ('lie', 1081), ('life', 1082), ('lifecyclerecapitul', 1083), ('ligat', 1084), ('ligatedpbluescript', 1085), ('light', 1086), ('lik', 1087), ('like', 1088), ('likelihood', 1089), ('limit', 1090), ('line', 1091), ('lineag', 1092), ('linear', 1093), ('linearis', 1094), ('link', 1095), ('list', 1096), ('liter', 1097), ('littl', 1098), ('lo', 1099), ('loc', 1100), ('local', 1101), ('localis', 1102), ('locat', 1103), ('lockdown', 1104), ('logit', 1105), ('lombardi', 1106), ('long', 1107), ('loop', 1108), ('lot', 1109), ('low', 1110), ('lower', 1111), ('lowest', 1112), ('lytl', 1113), ('machin', 1114), ('made', 1115), ('main', 1116), ('mainland', 1117), ('mainli', 1118), ('maintain', 1119), ('major', 1120), ('make', 1121), ('maker', 1122), ('male', 1123), ('manag', 1124), ('mani', 1125), ('map', 1126), ('mar', 1127), ('march18', 1128), ('march19', 1129), ('marchalreadi', 1130), ('marchapril', 1131), ('marchrd', 1132), ('marchrdgenom', 1133), ('marchrdth', 1134), ('marchroutin', 1135), ('marchth', 1136), ('marchthgenom', 1137), ('marker', 1138), ('markov', 1139), ('mask', 1140), ('mass', 1141), ('match', 1142), ('materi', 1143), ('mathemat', 1144), ('matric', 1145), ('matrix', 1146), ('max', 1147), ('maximum', 1148), ('may', 1149), ('mayreflect', 1150), ('mckenna', 1151), ('mean', 1152), ('measur', 1153), ('mechan', 1154), ('media', 1155), ('median', 1156), ('medic', 1157), ('medium', 1158), ('mengo', 1159), ('mention', 1160), ('mer', 1161), ('metagenom', 1162), ('metapopul', 1163), ('method', 1164), ('metric', 1165), ('metropolis', 1166), ('mexico', 1167), ('michigancould', 1168), ('microb', 1169), ('middl', 1170), ('might', 1171), ('migrat', 1172), ('milan', 1173), ('million', 1174), ('mind', 1175), ('minim', 1176), ('minimalrequir', 1177), ('minimum', 1178), ('mislead', 1179), ('miss', 1180), ('misunderstand', 1181), ('mitig', 1182), ('mix', 1183), ('ml', 1184), ('mobil', 1185), ('mock', 1186), ('model', 1187), ('modifi', 1188), ('molecular', 1189), ('moment', 1190), ('monitor', 1191), ('mononegaviral', 1192), ('month', 1193), ('moreov', 1194), ('mortal', 1195), ('mostli', 1196), ('motif', 1197), ('motiv', 1198), ('movement', 1199), ('much', 1200), ('multipl', 1201), ('multiplepk', 1202), ('municip', 1203), ('must', 1204), ('mutagenesi', 1205), ('mutagenesisperform', 1206), ('mutantcontrol', 1207), ('mutat', 1208), ('name', 1209), ('nanchang', 1210), ('napl', 1211), ('napoli', 1212), ('nation', 1213), ('natur', 1214), ('naïv', 1215), ('nbc', 1216), ('nc', 1217), ('ncbi', 1218), ('ncov', 1219), ('nd', 1220), ('ndintern', 1221), ('near', 1222), ('nearest', 1223), ('nearli', 1224), ('neb', 1225), ('necess', 1226), ('necessarili', 1227), ('need', 1228), ('neg', 1229), ('neighbor', 1230), ('netherland', 1231), ('network', 1232), ('networkhospit', 1233), ('neural', 1234), ('new', 1235), ('newli', 1236), ('news', 1237), ('next', 1238), ('nextgener', 1239), ('nextstrain', 1240), ('nextstrainteam', 1241), ('nhcc', 1242), ('nhei', 1243), ('nmia', 1244), ('node', 1245), ('nois', 1246), ('non', 1247), ('none', 1248), ('noninvas', 1249), ('normal', 1250), ('northwel', 1251), ('notabl', 1252), ('note', 1253), ('notic', 1254), ('notori', 1255), ('novel', 1256), ('novelti', 1257), ('novo', 1258), ('nowaday', 1259), ('nr', 1260), ('nucleic', 1261), ('nucleotid', 1262), ('nucleotide', 1263), ('number', 1264), ('numbers15to10', 1265), ('nyc', 1266), ('o1', 1267), ('o2', 1268), ('o3', 1269), ('ob', 1270), ('object', 1271), ('observ', 1272), ('obtain', 1273), ('occup', 1274), ('occupi', 1275), ('occur', 1276), ('occurr', 1277), ('of', 1278), ('offer', 1279), ('often', 1280), ('one', 1281), ('onehot', 1282), ('onepk', 1283), ('ongo', 1284), ('onli', 1285), ('onset', 1286), ('open', 1287), ('oper', 1288), ('oppos', 1289), ('optim', 1290), ('optimizerlearn', 1291), ('option', 1292), ('order', 1293), ('organ', 1294), ('orient', 1295), ('origin', 1296), ('other', 1297), ('outbreak', 1298), ('outcom', 1299), ('outperform', 1300), ('output', 1301), ('outsid', 1302), ('overal', 1303), ('overfit', 1304), ('overlap', 1305), ('oxford', 1306), ('p2p3', 1307), ('pa', 1308), ('packag', 1309), ('paid', 1310), ('pain', 1311), ('pair', 1312), ('pairwis', 1313), ('pandem', 1314), ('paper', 1315), ('paramet', 1316), ('paraphylet', 1317), ('parent', 1318), ('parsimoni', 1319), ('part', 1320), ('partial', 1321), ('particular', 1322), ('particularli', 1323), ('pass', 1324), ('passag', 1325), ('passagereplicon', 1326), ('path', 1327), ('pathogen', 1328), ('patient', 1329), ('pattern', 1330), ('pbluescript', 1331), ('pbluescriptc11', 1332), ('pcr', 1333), ('peer', 1334), ('peopl', 1335), ('per', 1336), ('percentag', 1337), ('perform', 1338), ('performedpurifi', 1339), ('performedresampl', 1340), ('perhap', 1341), ('period', 1342), ('permiss', 1343), ('perpetu', 1344), ('persist', 1345), ('phage', 1346), ('phasepandem', 1347), ('phenomenon', 1348), ('philip', 1349), ('phl2445_2020', 1350), ('phylogenet', 1351), ('phylogeni', 1352), ('phylogeny', 1353), ('phylogeograph', 1354), ('phylogeographi', 1355), ('phymm', 1356), ('phymmbl', 1357), ('picornaviru', 1358), ('picornavirus', 1359), ('picornavirusesauthor', 1360), ('pipelin', 1361), ('pirouz', 1362), ('pk', 1363), ('pk1', 1364), ('pk3', 1365), ('pk4', 1366), ('pkfound', 1367), ('pkrevers', 1368), ('pks4', 1369), ('pks4delet', 1370), ('pksreplication', 1371), ('place', 1372), ('plan', 1373), ('plaqu', 1374), ('plasmid', 1375), ('plot', 1376), ('pneumonia', 1377), ('point', 1378), ('polici', 1379), ('poly', 1380), ('polymeras', 1381), ('pool', 1382), ('poor', 1383), ('popul', 1384), ('popular', 1385), ('portug', 1386), ('pose', 1387), ('posit', 1388), ('positionspecif', 1389), ('possibl', 1390), ('post', 1391), ('posterior', 1392), ('potenti', 1393), ('practic', 1394), ('pre', 1395), ('preand', 1396), ('preced', 1397), ('precis', 1398), ('predic', 1399), ('predict', 1400), ('predictedconsist', 1401), ('predictionread', 1402), ('prefer', 1403), ('preferenti', 1404), ('preliminari', 1405), ('prep', 1406), ('prepar', 1407), ('prepared', 1408), ('preprocess', 1409), ('presenc', 1410), ('presencemultipl', 1411), ('present', 1412), ('prevent', 1413), ('previou', 1414), ('previous', 1415), ('primari', 1416), ('primarili', 1417), ('primer', 1418), ('prior', 1419), ('priorhour', 1420), ('pro', 1421), ('probabl', 1422), ('problem', 1423), ('procedur', 1424), ('process', 1425), ('produc', 1426), ('product', 1427), ('profession', 1428), ('program', 1429), ('project', 1430), ('pronounc', 1431), ('proofread', 1432), ('proper', 1433), ('proport', 1434), ('propos', 1435), ('prospect', 1436), ('prostructur', 1437), ('protect', 1438), ('protein', 1439), ('protocol', 1440), ('proven', 1441), ('provid', 1442), ('provinc', 1443), ('provinci', 1444), ('proxi', 1445), ('proxim', 1446), ('ptgfp', 1447), ('public', 1448), ('publicli', 1449), ('publish', 1450), ('pulmonari', 1451), ('purelink', 1452), ('purif', 1453), ('purpos', 1454), ('put', 1455), ('putativepackag', 1456), ('pytorch', 1457), ('qianxibaiducom', 1458), ('qualit', 1459), ('qualiti', 1460), ('quantit', 1461), ('quarantin', 1462), ('quartet', 1463), ('queri', 1464), ('question', 1465), ('quit', 1466), ('rais', 1467), ('raisedpre', 1468), ('ramsar', 1469), ('randomli', 1470), ('rang', 1471), ('rank', 1472), ('rapid', 1473), ('rapidli', 1474), ('rare', 1475), ('rate', 1476), ('rather', 1477), ('raw', 1478), ('ray', 1479), ('rdp', 1480), ('re', 1481), ('reaction', 1482), ('reactiv', 1483), ('read', 1484), ('readlevel', 1485), ('readsgut', 1486), ('real', 1487), ('receiv', 1488), ('recent', 1489), ('reciproc', 1490), ('recombin', 1491), ('recommend', 1492), ('reconstruct', 1493), ('record', 1494), ('recov', 1495), ('reduc', 1496), ('reduct', 1497), ('refer', 1498), ('reflect', 1499), ('reform', 1500), ('refram', 1501), ('refseq', 1502), ('regard', 1503), ('region', 1504), ('regions', 1505), ('regress', 1506), ('reinforc', 1507), ('reject', 1508), ('rel', 1509), ('relat', 1510), ('relationship', 1511), ('releas', 1512), ('relev', 1513), ('reliabl', 1514), ('rell', 1515), ('remain', 1516), ('remainday', 1517), ('remot', 1518), ('remov', 1519), ('removedsub', 1520), ('renown', 1521), ('repeat', 1522), ('repeats86', 1523), ('replac', 1524), ('replic', 1525), ('replication', 1526), ('replicativeadvantag', 1527), ('replicon', 1528), ('repliconscontain', 1529), ('report', 1530), ('repres', 1531), ('represent', 1532), ('representbedsicushospit', 1533), ('representingc', 1534), ('requir', 1535), ('rescu', 1536), ('research', 1537), ('resembl', 1538), ('reserv', 1539), ('residuespk3', 1540), ('resolut', 1541), ('resolv', 1542), ('resourc', 1543), ('resource', 1544), ('respect', 1545), ('respiratori', 1546), ('respons', 1547), ('responsiblewest', 1548), ('restor', 1549), ('restrict', 1550), ('result', 1551), ('retrain', 1552), ('retriev', 1553), ('reus', 1554), ('reveal', 1555), ('revers', 1556), ('reverseprim', 1557), ('review', 1558), ('right', 1559), ('rigor', 1560), ('risk', 1561), ('rna', 1562), ('robust', 1563), ('roc', 1564), ('role', 1565), ('rome', 1566), ('root', 1567), ('rout', 1568), ('routin', 1569), ('run', 1570), ('rush', 1571), ('s1', 1572), ('s2', 1573), ('sagripanti', 1574), ('sampl', 1575), ('sample', 1576), ('santa', 1577), ('sar', 1578), ('sars', 1579), ('satisfi', 1580), ('sc', 1581), ('scal', 1582), ('scale', 1583), ('scatter', 1584), ('scenario', 1585), ('scienc', 1586), ('scientif', 1587), ('score', 1588), ('screen', 1589), ('search', 1590), ('seattl', 1591), ('second', 1592), ('secondari', 1593), ('section', 1594), ('secur', 1595), ('see', 1596), ('seem', 1597), ('seen', 1598), ('seir', 1599), ('select', 1600), ('sem', 1601), ('semant', 1602), ('sensit', 1603), ('sentenc', 1604), ('separ', 1605), ('sepsi', 1606), ('seqman', 1607), ('sequenc', 1608), ('sequencepotenti', 1609), ('sequenceremov', 1610), ('sequencescountri', 1611), ('sequenti', 1612), ('sequentiallypassag', 1613), ('seriou', 1614), ('serv', 1615), ('serval', 1616), ('set', 1617), ('sever', 1618), ('shanghai', 1619), ('shape', 1620), ('share', 1621), ('sharpli', 1622), ('shed', 1623), ('sheer', 1624), ('shi', 1625), ('shimodaira', 1626), ('short', 1627), ('shortag', 1628), ('show', 1629), ('shown', 1630), ('sicker', 1631), ('side', 1632), ('sig', 1633), ('sign', 1634), ('signal', 1635), ('signific', 1636), ('significantli', 1637), ('similar', 1638), ('similarli', 1639), ('simul', 1640), ('sinc', 1641), ('singapor', 1642), ('singl', 1643), ('single', 1644), ('singleauthor', 1645), ('singleton', 1646), ('site', 1647), ('site3', 1648), ('situat', 1649), ('six', 1650), ('size', 1651), ('sizeus', 1652), ('sketch', 1653), ('skew', 1654), ('skip', 1655), ('skipgram', 1656), ('slower', 1657), ('small', 1658), ('smaller', 1659), ('snapshot', 1660), ('social', 1661), ('socioeconom', 1662), ('softmax', 1663), ('softwar', 1664), ('solid', 1665), ('solv', 1666), ('soon', 1667), ('sound', 1668), ('south', 1669), ('space', 1670), ('spain', 1671), ('spatiotempor', 1672), ('spear', 1673), ('spearman', 1674), ('speci', 1675), ('species', 1676), ('speciesgenera', 1677), ('specif', 1678), ('specifi', 1679), ('speculatedform', 1680), ('speed', 1681), ('spei', 1682), ('split', 1683), ('spread', 1684), ('spss', 1685), ('squar', 1686), ('srr10714026', 1687), ('srr1930021', 1688), ('ssrna', 1689), ('standard', 1690), ('star', 1691), ('start', 1692), ('state', 1693), ('statist', 1694), ('stay', 1695), ('stem', 1696), ('step', 1697), ('steric', 1698), ('stigmat', 1699), ('still', 1700), ('stochast', 1701), ('stop', 1702), ('strain', 1703), ('strainidentifi', 1704), ('strand', 1705), ('strategi', 1706), ('stress', 1707), ('stretch', 1708), ('strong', 1709), ('structur', 1710), ('studi', 1711), ('sub', 1712), ('subclad', 1713), ('subject', 1714), ('suboptim', 1715), ('substanti', 1716), ('substitut', 1717), ('substringsbp', 1718), ('subtl', 1719), ('success', 1720), ('such', 1721), ('sudden', 1722), ('suffici', 1723), ('suggest', 1724), ('suitabl', 1725), ('sum', 1726), ('summar', 1727), ('sun', 1728), ('sunshin', 1729), ('superior', 1730), ('supplementari', 1731), ('suppli', 1732), ('support', 1733), ('supportsfind', 1734), ('sure', 1735), ('surfac', 1736), ('surround', 1737), ('survey', 1738), ('surviv', 1739), ('survivor', 1740), ('susceptible', 1741), ('sustain', 1742), ('switch', 1743), ('switzerland', 1744), ('symptom', 1745), ('syntact', 1746), ('synthesi', 1747), ('synthet', 1748), ('system', 1749), ('sz', 1750), ('t7', 1751), ('tabl', 1752), ('tabledescrib', 1753), ('take', 1754), ('taken', 1755), ('takessecond', 1756), ('tandem', 1757), ('target', 1758), ('task', 1759), ('taxa', 1760), ('taxonom', 1761), ('taxonomi', 1762), ('techniqu', 1763), ('technolog', 1764), ('tehran', 1765), ('tele', 1766), ('temperatur', 1767), ('tempest', 1768), ('tempor', 1769), ('ten', 1770), ('tend', 1771), ('term', 1772), ('test', 1773), ('th', 1774), ('thank', 1775), ('therefor', 1776), ('third', 1777), ('though', 1778), ('threat', 1779), ('threaten', 1780), ('three', 1781), ('threefold', 1782), ('threshold', 1783), ('thresholds1', 1784), ('thu', 1785), ('till', 1786), ('tim', 1787), ('time', 1788), ('timefram', 1789), ('tip', 1790), ('tissu', 1791), ('to', 1792), ('togeth', 1793), ('toler', 1794), ('tool', 1795), ('top', 1796), ('topolog', 1797), ('total', 1798), ('totallyclassifi', 1799), ('totallyebola', 1800), ('toward', 1801), ('town', 1802), ('trace', 1803), ('track', 1804), ('tract', 1805), ('train', 1806), ('transcript', 1807), ('transfect', 1808), ('transfer', 1809), ('transit', 1810), ('transition', 1811), ('transitions', 1812), ('translat', 1813), ('transmiss', 1814), ('transpar', 1815), ('transvers', 1816), ('travel', 1817), ('tre', 1818), ('treat', 1819), ('treatment', 1820), ('tree', 1821), ('treetim', 1822), ('tremend', 1823), ('trend', 1824), ('triag', 1825), ('triangl', 1826), ('trigger', 1827), ('trna', 1828), ('tropic', 1829), ('true', 1830), ('truncat', 1831), ('turn', 1832), ('twenti', 1833), ('twice', 1834), ('two', 1835), ('typ', 1836), ('type', 1837), ('ud', 1838), ('uk', 1839), ('unbalanc', 1840), ('uncertainti', 1841), ('unclear', 1842), ('uncontrol', 1843), ('uncorrel', 1844), ('under', 1845), ('understand', 1846), ('uneven', 1847), ('uniform', 1848), ('uniformli', 1849), ('uniqu', 1850), ('unit', 1851), ('univers', 1852), ('unknown', 1853), ('unlik', 1854), ('unpreced', 1855), ('unreli', 1856), ('unresolv', 1857), ('unsampl', 1858), ('unseen', 1859), ('unsupport', 1860), ('untrac', 1861), ('unveil', 1862), ('up', 1863), ('upcom', 1864), ('updat', 1865), ('upload', 1866), ('upstream', 1867), ('urban', 1868), ('urgenc', 1869), ('urgent', 1870), ('url', 1871), ('us', 1872), ('usa', 1873), ('usag', 1874), ('use', 1875), ('usedhiscrib', 1876), ('user', 1877), ('usingsickl', 1878), ('usual', 1879), ('usually5', 1880), ('util', 1881), ('utr', 1882), ('uv', 1883), ('vaccin', 1884), ('valencia', 1885), ('valid', 1886), ('valu', 1887), ('vari', 1888), ('variabl', 1889), ('varianc', 1890), ('variat', 1891), ('variou', 1892), ('vector', 1893), ('vehicl', 1894), ('ventil', 1895), ('vernon', 1896), ('vertebr', 1897), ('vi', 1898), ('via', 1899), ('viabl', 1900), ('view', 1901), ('violini', 1902), ('vir', 1903), ('viral', 1904), ('viru', 1905), ('virus', 1906), ('virusdb', 1907), ('virusseek', 1908), ('vision', 1909), ('vital', 1910), ('vitamin', 1911), ('vitro', 1912), ('vote', 1913), ('vp0', 1914), ('vp2', 1915), ('vp3', 1916), ('vp4', 1917), ('vpg', 1918), ('vs', 1919), ('vssi', 1920), ('vulner', 1921), ('wale', 1922), ('way', 1923), ('weaker', 1924), ('weather', 1925), ('web', 1926), ('websit', 1927), ('week', 1928), ('weigh', 1929), ('weight', 1930), ('well', 1931), ('wgsim', 1932), ('wh', 1933), ('whether', 1934), ('whole', 1935), ('wide', 1936), ('wilcoxon', 1937), ('wild', 1938), ('within', 1939), ('withinhour', 1940), ('withinord', 1941), ('wk', 1942), ('wood', 1943), ('woolhous', 1944), ('word', 1945), ('work', 1946), ('world', 1947), ('worldwid', 1948), ('worth', 1949), ('would', 1950), ('wrong', 1951), ('wt', 1952), ('wu', 1953), ('wuerttemberg', 1954), ('wuhan', 1955), ('wwwgisaidorg', 1956), ('wwwncbinlmnihgov', 1957), ('xinyang', 1958), ('xl', 1959), ('year', 1960), ('yeast', 1961), ('yellow', 1962), ('yet', 1963), ('yield', 1964), ('yiyang', 1965), ('yn', 1966), ('york', 1967), ('zair', 1968), ('zone', 1969), ('ºc', 1970), ('δpkforward', 1971), ('δpkforwardprim', 1972), ('μg', 1973)]\n",
      "[[0.         0.         0.         ... 0.03719836 0.03719836 0.03719836]\n",
      " [0.05925081 0.05925081 0.04780318 ... 0.         0.         0.        ]\n",
      " [0.         0.         0.         ... 0.         0.         0.        ]\n",
      " [0.         0.         0.         ... 0.         0.         0.        ]\n",
      " [0.         0.         0.         ... 0.         0.         0.        ]\n",
      " [0.         0.         0.         ... 0.         0.         0.        ]]\n"
     ]
    }
   ],
   "source": [
    "#create a tf-idf matrix for the words above\n",
    "v = TfidfVectorizer(sublinear_tf = True, min_df = 0.05)\n",
    "#For the above vectorizer, the first paramater specifies that the 1+log(tf) is going to be used instead of tf.\n",
    "#The second parameter specifies that if a term appears in less than 5% of the docs, ignore it\n",
    "#The third parameter, considers word that do not occur in more than 80% of the corpus\n",
    "tfidf = v.fit_transform(stemmed_text) #fit and transform the stemmed texts which were previously cleaned.\n",
    "print(sorted(v.vocabulary_.items(), key=lambda x : x[1]))\n",
    "print(tfidf.toarray())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Enter your query: quarantine\n",
      "[0.         0.         0.05503208 0.         0.         0.        ]\n",
      "1\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#Next, the search engine will be implemented with the help of the tf-idf matrix created above and the cosine similarity\n",
    "flag = False # a flag to indicate whether or not a result is found\n",
    "pd.set_option('display.max_colwidth', -1)#to avoid dataframe column's data truncation\n",
    "query = input(\"Enter your query: \") #get the query from the user\n",
    "finalQuery=\"\" #create an empty string\n",
    "words = query.split(' ') #split the query on spaces\n",
    "for word in words: #for each word in the query\n",
    "    word = word.lower()\n",
    "    if(word!=\"\"): #if the word is not an empty string\n",
    "        if (word==\"coronaviruses\"):#since the stem and lemmatize functions do not recognize the word coronaviruses as the plural of coronavirus, a conidition was created\n",
    "            word =\"coronavirus\"\n",
    "        stem_query = porter.stem(word)#stem the word\n",
    "        finalQuery+=stem_query+\" \" #stem the word and concat it to the string finalQuery\n",
    "query = [finalQuery.strip()]# remove any trailing spaces from the query\n",
    "query_tfidf = v.transform(query)\n",
    "cosineSimilarities = cosine_similarity(query_tfidf, tfidf).flatten() #get the cosine similarity list \n",
    "#IF NO RESULT\n",
    "print(cosineSimilarities)\n",
    "countzero_in2 = np.count_nonzero(cosineSimilarities) #get the number of non zero values in cosine similarity\n",
    "print(countzero_in2)\n",
    "if (countzero_in2==0):#if no cosine similarity which is =/= 0\n",
    "    flag = True\n",
    "\n",
    "else:\n",
    "    if (countzero_in2>=5):#if there are more than 5 values greater than 5 output them\n",
    "        answer_indicies = sorted(range(len(cosineSimilarities)), key=lambda i: cosineSimilarities[i])[-5:] #get the top 5 indices with the highest cosine similarity\n",
    "    else:\n",
    "        answer_indicies = sorted(range(len(cosineSimilarities)), key=lambda i: cosineSimilarities[i])[-countzero_in2:] #get the top countzero_in2 indices with the highest cosine "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Healthcare-resource-adjusted vulnerabilities towards the 2019-nCoV epidemic across China</td>\n",
       "      <td>[]</td>\n",
       "      <td>The 2019-nCoV epidemic has spread across China and 24 other countries 1-3 as of February 8, 2020 . The mass quarantine measure implemented in the epidemic's epicenter, Wuhan, Hubei Province, China, on January 23, 2020, has stopped the human migration between Wuhan and other places. However, according to the local authorities, over five million people had already traveled from Wuhan to other parts of China in the weeks leading to January 23, 2020, with the majority (79.96%) to the other cities within Hubei.\\nHealthcare resources have become a major bottleneck in the response to this public health crisis. The sudden onset of the epidemic and the sheer size of the infection have caused significant shortage in healthcare resources across Hubei Province. So far, there have been over 11,000 non-local healthcare professionals, mobilized by the public health authorities from other parts of China, fighting against the epidemic in Wuhan. However, this is far from adequate for Wuhan given the severity of the epidemic. In addition to Wuhan, other cities in Hubei are in need of urgent external help as well with respect to health resources, as multiple localized outbreaks are taking place. On February 7, 2020, the National Health Commission of China (NHCC) initiated the \"One City Covered by One Province\" program, explicitly assigning a specific province to aid one city in Hubei. Two critical questions of operational nature remain, which cities, both in Hubei and other provinces, are most vulnerable to localized outbreaks, and how the public health authorities should balance the healthcare resources under stress to best fight the epidemic and minimize the risks? This study is aimed to shed light on these questions by developing a vulnerability analysis framework integrating the human movement and healthcare resource data.\\nThis study makes use of four public datasets: (a) the daily counts of confirmed 2019-nCoV cases, collected from NHCC 4 ; (b) the list of designated hospitals with the number of reserved beds, retrieved from the websites of the provincial and municipal health commissions; (c) the information of the hospitals designated for 2019-nCoV, retrieved from 99.com.cn, a public hospital database; and (d) the human movement data, provided by the Baidu Migration (http://qianxi.baidu.com/), a public site offering real-time migration information across mainland China.\\nRecent studies indicated that the confirmed case data for Hubei Province, in particular, Wuhan, was subject to the under-reporting bias, while the data from places outside of Hubei is more reliable for inferring the actual epidemic size 5 . Following Wu et al. 5 , we developed a susceptible-exposed-infectious-recovered (SEIR)-based metapopulation model, which has 100 sub-populations representing the 100 cities that have the greatest number of people traveling from Wuhan. Note that we adopted the same epidemiological parameters as Wu et al. 5 . This model was then used to estimate IDi, the number of infected individuals in each city. Lastly, healthcare-resource-adjusted vulnerability indexes to the 2019-nCoV outbreak for city i, Vi were obtained as follows: Vi = IDi / Bi, where Bi represents the total number of reserved beds in the designated hospitals.\\nThe Figure presents the estimated number of infected individuals, as well as the healthcareresource-adjusted vulnerability to the 2019-nCoV outbreak across mainland China, as of January 29, 2020. The epidemiological parameters may have changed after January 23, 2020, when the lockdown measure was implemented. Thus, we only use the data by January 23, 2020 and produce the estimation till January 29, 2020 to accommodate the six-day incubation period identified by Wu et al. 5 .\\nThe ten most vulnerable non-Hubei cities are also listed in the figure. Note that the cities in Hubei (which are being aided by other provinces) are not presented here because of their significantly higher vulnerability as compared to others. Among the ten most vulnerable non-Hubei cities, Changsha, Beijing, Chongqing and Shanghai are major metropolises with a large number of estimated infected individuals. Xinyang, Jiujiang, Nanchang and Yiyang are medium-sized cities neighboring Hubei. In particular, Xinyang appears to be in a particular vulnerable situation because of the large number of people traveling from Wuhan and its insufficient healthcare resources.\\nThe reported vulnerability analysis informs public health response to the 2019-nCoV epidemic in multiple ways. For the identified cities with high vulnerability, it is important to increase the reserve of healthcare resources or have contingent plans. In particular, some of these cities already contributed significant healthcare resources to Hubei, leaving themselves inadequately prepared for the localized outbreak. Our analysis revealed that several medium-sized cities, particularly those neighboring Hubei, are highly vulnerable due to the lack of healthcare resources. \\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                      title  \\\n",
       "2  Healthcare-resource-adjusted vulnerabilities towards the 2019-nCoV epidemic across China   \n",
       "\n",
       "  abstract  \\\n",
       "2  []        \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               full_text  \n",
       "2  The 2019-nCoV epidemic has spread across China and 24 other countries 1-3 as of February 8, 2020 . The mass quarantine measure implemented in the epidemic's epicenter, Wuhan, Hubei Province, China, on January 23, 2020, has stopped the human migration between Wuhan and other places. However, according to the local authorities, over five million people had already traveled from Wuhan to other parts of China in the weeks leading to January 23, 2020, with the majority (79.96%) to the other cities within Hubei.\\nHealthcare resources have become a major bottleneck in the response to this public health crisis. The sudden onset of the epidemic and the sheer size of the infection have caused significant shortage in healthcare resources across Hubei Province. So far, there have been over 11,000 non-local healthcare professionals, mobilized by the public health authorities from other parts of China, fighting against the epidemic in Wuhan. However, this is far from adequate for Wuhan given the severity of the epidemic. In addition to Wuhan, other cities in Hubei are in need of urgent external help as well with respect to health resources, as multiple localized outbreaks are taking place. On February 7, 2020, the National Health Commission of China (NHCC) initiated the \"One City Covered by One Province\" program, explicitly assigning a specific province to aid one city in Hubei. Two critical questions of operational nature remain, which cities, both in Hubei and other provinces, are most vulnerable to localized outbreaks, and how the public health authorities should balance the healthcare resources under stress to best fight the epidemic and minimize the risks? This study is aimed to shed light on these questions by developing a vulnerability analysis framework integrating the human movement and healthcare resource data.\\nThis study makes use of four public datasets: (a) the daily counts of confirmed 2019-nCoV cases, collected from NHCC 4 ; (b) the list of designated hospitals with the number of reserved beds, retrieved from the websites of the provincial and municipal health commissions; (c) the information of the hospitals designated for 2019-nCoV, retrieved from 99.com.cn, a public hospital database; and (d) the human movement data, provided by the Baidu Migration (http://qianxi.baidu.com/), a public site offering real-time migration information across mainland China.\\nRecent studies indicated that the confirmed case data for Hubei Province, in particular, Wuhan, was subject to the under-reporting bias, while the data from places outside of Hubei is more reliable for inferring the actual epidemic size 5 . Following Wu et al. 5 , we developed a susceptible-exposed-infectious-recovered (SEIR)-based metapopulation model, which has 100 sub-populations representing the 100 cities that have the greatest number of people traveling from Wuhan. Note that we adopted the same epidemiological parameters as Wu et al. 5 . This model was then used to estimate IDi, the number of infected individuals in each city. Lastly, healthcare-resource-adjusted vulnerability indexes to the 2019-nCoV outbreak for city i, Vi were obtained as follows: Vi = IDi / Bi, where Bi represents the total number of reserved beds in the designated hospitals.\\nThe Figure presents the estimated number of infected individuals, as well as the healthcareresource-adjusted vulnerability to the 2019-nCoV outbreak across mainland China, as of January 29, 2020. The epidemiological parameters may have changed after January 23, 2020, when the lockdown measure was implemented. Thus, we only use the data by January 23, 2020 and produce the estimation till January 29, 2020 to accommodate the six-day incubation period identified by Wu et al. 5 .\\nThe ten most vulnerable non-Hubei cities are also listed in the figure. Note that the cities in Hubei (which are being aided by other provinces) are not presented here because of their significantly higher vulnerability as compared to others. Among the ten most vulnerable non-Hubei cities, Changsha, Beijing, Chongqing and Shanghai are major metropolises with a large number of estimated infected individuals. Xinyang, Jiujiang, Nanchang and Yiyang are medium-sized cities neighboring Hubei. In particular, Xinyang appears to be in a particular vulnerable situation because of the large number of people traveling from Wuhan and its insufficient healthcare resources.\\nThe reported vulnerability analysis informs public health response to the 2019-nCoV epidemic in multiple ways. For the identified cities with high vulnerability, it is important to increase the reserve of healthcare resources or have contingent plans. In particular, some of these cities already contributed significant healthcare resources to Hubei, leaving themselves inadequately prepared for the localized outbreak. Our analysis revealed that several medium-sized cities, particularly those neighboring Hubei, are highly vulnerable due to the lack of healthcare resources. \\n  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#loop on the top 5 indicies and display the title, abstract, and fulltext of the paper that was fetched from the search\n",
    "if (flag == True):\n",
    "    print(\"Sorry, no results\")\n",
    "else: \n",
    "    for index in answer_indicies:\n",
    "        display(FullPaperDataframe.iloc[[index]] )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.decomposition import PCA #import PCA which will be used to reduce the dimensions of the dataset\n",
    "\n",
    "pca = PCA(n_components=0.90,random_state=30)#keep 90% of the variance, and start random centroid @ 30\n",
    "tf_idf_norm_kmeans= pca.fit_transform(tfidf.toarray())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 4/4 [00:00<00:00, 23.67it/s]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 1440x1440 with 0 Axes>"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYUAAAEaCAYAAAD+E0veAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAgAElEQVR4nOzdd1QU19sH8O+dpfceFEVFBcUaQQUrUTQaYzTRYBe7BnuaxqhYIxaMDUvUSEQTMfaYaBJib7EbFRuWxC6IiFKkzPP+wc99XWkLsswCz+ccz3FnZme+ywUeZu6dO4KICIwxxhgASekAjDHG9AcXBcYYY2pcFBhjjKlxUWCMMabGRYExxpgaFwXGGGNqXBRKufPnz6NRo0YwMTFB5cqVlY6jiPDwcBgYGBTZ/vr16wd/f/8i2x8rGvv27YMQAnfu3FE6SonGRUFLKSkpmDRpEqpXrw5TU1PY29ujYcOGWLRokdLR8vTll1/CysoKly9fxokTJ3Ldbs2aNfDy8oKVlRUsLS1Rs2ZNDB48uBiTKsvPzw9CCAghYGhoCCcnJ7Rs2RLz589HSkqKxrYLFy7Ezz//rPW+DQwMEB4eXsSJlfHtt9/C1NQU8fHxOa7v2LEjmjVrBgCQZRnz5s1D7dq1YW5uDhsbG9SrVw8TJ04s1LHXrVuHFi1awNraGubm5qhduza+/PJL3L17t9CfJz8zZswoc39McVHQ0ieffIK1a9di7ty5iI6Oxp49ezB8+HAkJCQoHS1P165dQ8uWLVG5cmU4OjrmuE14eDiGDRuG/v374+TJkzh16hRCQkKQmZlZzGlzRkRIT0/X+XF69uyJ+/fv49atW/jzzz/RtWtXhIaGokGDBnj06JF6O2tra9ja2uo8jz4KDAwEAERERGRbd/fuXezatQtDhgwBAEybNg0zZ87E+PHj8c8//+Dw4cOYMGECkpKSCnzcgQMHYuDAgWjRogV27dqF6OhoLFq0CA8ePEBoaOibfahikpaWpnQE7RDTirW1NS1evDjPbQIDA6l169YayyIiIujVL3NwcDBVrVqVIiMjqVq1amRqakqdOnWip0+f0ubNm8nd3Z0sLCyoS5culJCQkOfx7t27R926dSNra2syMTGhli1b0okTJ4iI6ObNmwRA419wcHCO++nUqRN16dIl369BZGQkVa1alYyNjcnX15e2b99OAOjgwYNERLR3714CQLdv39Z4n0qlojVr1qhfT5gwgWrUqEGmpqZUoUIFGjp0qMZnXbNmDalUKtqzZw/Vr1+fDA0N6ZdffiEioj/++IOaNGlCJiYmVL58eerXrx/FxcWp3yvLMk2cOJEcHR3J3NycunXrRvPnzyeVSpXnZ2vZsiUNHDgw2/Lbt2+TjY0N9e/fX73s9Xa+cOECtW3blqytrcnMzIxq1KhBa9euJSKiSpUqZWsHIqL4+Hjq1asXVaxYkUxMTMjd3Z3mzZtHsixnO86KFSvI1dWVLC0t6YMPPqBHjx5pZPzzzz+pWbNmZGpqSlZWVtSiRQuKiYlRr//pp5+oXr16ZGxsTJUqVaKxY8fS8+fP1esPHjxITZo0IQsLC7KwsKC6devS7t27c/1a9e7dm2rVqpVt+bRp08jGxoaSk5OJiKhevXr02Wef5bofbW3atIkA0E8//ZTj+vj4eCLK/v2n7ffjzJkzqUqVKmRkZEQODg7Utm1bSk5OpjVr1uT6M5Senk7BwcFUuXJlMjY2Jk9PT1q+fLnGcQDQwoULqUePHmRlZUVdu3bN83j6gouClmrUqEEdOnSgx48f57qNtkXBzMyM3nvvPTp37hzt27ePHBwcqE2bNtS+fXs6e/YsHThwgJycnOjLL7/M9ViyLFOjRo2oXr16dPDgQfrnn38oICCAbGxsKDY2ljIyMuj+/ftUoUIFGjduHN2/f5+ePXuW476GDRtGlSpVoitXruR6vNOnT5MQgsaPH0+XL1+mzZs3U+XKlQtVFKZPn04HDhygmzdvUlRUFHl4eFDfvn3V69esWUNCCPL29qa//vqLrl+/To8ePaK//vqLTE1NadGiRXT16lU6fvw4+fn5UfPmzdW/TBcsWEBmZmYUHh5OV65codmzZ5O1tXWhiwIR0fDhw8nKyooyMzOJKHs716lTh3r06EEXL16k69ev02+//aYuYo8ePSKVSkULFiyg+/fv0/3794mI6P79+xQSEkKnTp2iGzduUEREBJmbm9P333+v3m9gYCBZWVlR9+7d6fz583T48GFydXXV+Fr9+eefJEkSjR49ms6ePUuXLl2iVatW0aVLl9RfSxsbG1q7di1dv36d9u/fT3Xq1KHevXsTEVFGRgbZ2trS2LFj6erVq3T16lXasmULHThwINev1cGDBwkAHTlyRL0sMzOTKlWqRCNHjlQva9euHXl7e9OdO3fy/Nrnp1OnTlStWrV8tytMUdi8eTNZWlrSjh076N9//6UzZ87Qt99+S8nJyZScnEzjxo2jChUqqNvu5c9QYGAg1alTh37//Xe6ceMGbdiwgaytrWnVqlXq4wAgOzs7WrRoEcXExNCVK1fyPJ6+4KKgpUOHDpGrqytJkkR16tShwYMH07Zt23L8y+5VORUFlUpFsbGx6mVBQUEkSZLGX4CjRo0iLy+vXPNERUURALp48aJ6WWpqKjk7O9PUqVPVyypVqkTTp0/P87Pdv3+fmjZtSgCoUqVKFBAQQCtWrKCkpCT1Nr169SJfX1+N9y1evLhQReF1W7ZsISMjI/Uv3Zd/ob3+i6lly5Y0btw4jWX//vsvAaAzZ84QEZGLiwtNmDBBY5suXbq8UVFYtmwZAaCHDx8SUfZ2trKyyvPz5ff5Xxo1ahT5+/urXwcGBpKDgwOlpqaql82aNYucnZ3Vr5s1a0YdOnTIdZ+VKlWiZcuWaSzbv38/AaD4+HiKj48nALR37958873K09NT4+xp9+7dBIDOnz+vXnbp0iWqVasWCSHI3d2d+vbtS+vWraP09PQCHatmzZrUsWPHfLcrTFGYP38+Va9endLS0nLc5/Tp06lSpUoay27cuEFCCHXhfWnq1KlUr1499WsANGDAAI1t8juePuA+BS01bdoU169fx8GDBxEYGIiHDx+iS5cu+OCDD0AFnFPQxcUFDg4O6tfOzs5wdnbWuObv7OyscR37dRcvXoS9vT08PT3Vy4yNjdG4cWNcvHixQHmcnZ1x6NAhREdH46uvvoK5uTm+/PJL1KpVS50hOjoaTZs21Xjfyw7FgtqyZQtatGiB8uXLw8LCAr169UJaWhoePHigsV3Dhg01Xp84cQILFiyAhYWF+t/Lz3/t2jUkJibi7t27aNKkSZHkfOll+wohclz/+eefY9CgQfDz88OUKVNw+vTpfPcpyzJCQkJQv359ODg4wMLCAsuXL8e///6rsV3NmjVhbGysfu3i4oKHDx+qX586dQpt27bN8RixsbH4999/8emnn2p8zdq3bw8AiImJga2tLQYNGoR3330X7du3R0hICK5cuZJv/iFDhiAyMhKJiYkAgJUrV8LX1xe1a9dWb1OjRg2cP38ep06dwogRI5CWloZBgwbBx8cnW+d9Xogo16/9mwoICEB6ejoqVaqEfv36ISIiAs+ePcvzPSdPngQRwdvbW+Pr+s033+DatWsa2zZq1OiNj1fcuCgUgIGBAZo0aYLPPvsM27dvR3h4OHbu3IkDBw4AACRJylYgcuogNTQ01Hj9csTL68tkWc4zT04/KG/yA1SzZk0MHToU33//Pc6cOYM7d+5g2bJlWu9XkiT1ti9lZmZqfI6///4bH3/8MVq0aIGtW7fi9OnTWL58OQDNjjiVSgUTExON/cuyjHHjxuHs2bMa/65du4b27dvn+8u7sC5cuAAbGxvY29vnuH7SpEm4evUqAgICcOHCBfj4+OQ7wiY0NBSzZs3CyJEj8eeff+Ls2bMYNGhQts5IIyMjjddCiGzfY7l93pdf94ULF2p8vc6dO4dr166hTp06ALJ+oZ86dQpt2rTB/v37Ubt2baxYsSLP/H379gURYf369Xj48CF27Nih7mB+Pdvbb7+NkSNH4qeffsKff/6JU6dOYePGjXnu/1UeHh6Ijo7WevuXtPl+dHFxweXLl/H999/DyckJ06dPh4eHB27fvp3rfl++/8iRIxpf1wsXLuCff/7R2Nbc3FzjdWGOV9y4KLyBmjVrAoD6r2knJyfcu3dPYxtt/mosjFq1aiEuLk7jh+XFixc4fvw4atWq9cb7r1y5MszMzNSfrVatWjh8+LDGNq+/dnJyAgCNr8HZs2c1figPHToEBwcHzJgxA40bN4a7u7vW48q9vb1x8eJFVKtWLds/CwsLWFtbw8XFJd+cBXHnzh2sX78eXbt2Vf+SyYmbmxuCgoKwadMmTJs2TV1Mgaxf7K+P5Dpw4ADatWuHgQMH4u2330a1atWy/ZWpDS8vL/z+++85rnvrrbdQsWJFXLlyJcev2atFt3bt2vj000+xa9cuDBw4EN99912ex7W1tcXHH3+MlStXIjw8HGZmZggICMg37+s/M9ro3bs3YmJisGHDhhzXP3nyJMfl2nw/Alln2O3atcOcOXNw/vx5JCcnY9u2bQBybjsvLy8AwH///Zfta1q1atV8P09ex9MHRXdHTynXsmVL9OjRA97e3nB0dERMTAwmTJgAGxsbvPPOOwAAf39/zJ49G0uWLEH79u2xZ8+eAv1FVBCtWrVCo0aN0LNnT4SFhcHa2hrTp09HamoqPvnkkwLt65NPPoGzszNatWoFV1dXxMXFYeHChUhMTETnzp0BAGPHjkXDhg3x9ddfIzAwEBcvXsw2FLBatWqoVKkSpkyZgm+//RZxcXGYMGGCxl+yHh4eiI2NxerVq/HOO+/g0KFDWLp0qVY5p02bhrZt22Ls2LEIDAyEpaUlrl27hp9//hlLliyBqakpPvvsM0yaNAk1atSAj48PduzYgaioKK32n5KSggcPHkCWZcTFxeHAgQOYNWsWXFxcMGvWrBzf8/z5c4wbNw5dunRBlSpVkJCQgN27d2tc1qtSpQr27t2L9u3bw8jICA4ODvDw8EBERAT27t0LFxcXrF27Fn///XeBh7pOmjQJ7du3x5gxYzBgwAAYGxvj6NGj8PX1hYeHB2bOnImBAwfCxsYGnTt3hqGhIS5duoRdu3ZhxYoViImJwcqVK9GxY0dUrFgR9+7dw8GDB9GgQYN8jz106FA0bdoUt27dQu/evWFmZqaxvkuXLmjSpAmaNGmC8uXL4+7du5gxYwYMDQ3RoUMHAMDx48fRt29frF27Ntullpe6du2Kvn37qr/v3nvvPbi4uODmzZsIDw+Hra0t5s+fn+192nw/rl69GrIso1GjRrCxscFff/2FZ8+eqduvSpUqePDgAY4ePYrq1avDzMwM1apVw4ABAzB48GDMmTMHvr6+SEpKwqlTpxAbG4tx48bl+jXL73h6QYmOjJJo1qxZ1KxZM3J0dCRjY2OqWLEi9erVS6Ojl4hoxowZVL58eTI3N6fu3bvTkiVLchyS+qqcOrNmzZpFLi4ueWZ6fUhqixYt1ENSX9Kmo3nz5s3UsWNHcnFxISMjI3JyciJ/f3/67bffNLb76aefyM3NjYyMjKhRo0a0bds2jY5mIqJjx45RgwYNyMTEhOrWrUsHDhzI1tE6ceJEcnJyIjMzM2rfvj39+OOPBIBu3rxJRP8/JDUnBw4coNatW5OFhYV6+Ofo0aPVnZeZmZn01Vdfkb29PZmZmVGXLl20HpKK/w07NDAwIHt7e2rRogWFhoZmGxnyakdzSkoK9ejRQz000dHRkQICAui///5Tb79r1y6qUaMGGRkZqb8XEhIS6OOPPyZLS0uys7OjoKAgmjhxosb3gTYDF4iyOnl9fHzIxMSErKysyM/Pj65fv65ev3XrVvLx8SFTU1OytLSkevXqqQcj3Lt3jz788EN125crV44GDRqU73Dol2rXrk0A6Ny5c9nWfffdd+Tv70/Ozs5kZGRE5cuXp06dOmmMWnrZGaxNR3d4eDg1bdqULC0tyczMjGrVqkXjxo2je/fuaezr1Y7l/L4fN2/eTL6+vmRjY0OmpqZUq1YtjRFEaWlp1KNHD7K1tdUYkpqRkUGzZ88mDw8PMjQ0VH+/bNy4Uf1eABQREaHxGfI7nj4QRPzkNVY4t27dQpUqVXDw4ME37sxljOkH7lNgjDGmxkWBMcaYGl8+YowxpsZnCowxxtS4KDDGGFMr8fcpvH6zmLYcHBwQFxdXxGnYm+J20T/cJvrpTdqlfPnyua7jMwXGGGNqXBQYY4ypFcvlo7S0NAQHByMjIwOZmZnw8fHJNk/Kvn37EBERATs7OwBAu3bt0Lp16+KIxxhj7H+KpSgYGhoiODgYJiYmyMjIwOTJk1G/fn24u7trbNekSRMMHDiwOCIxxhjLQbFcPhJCqGdkzMzMRGZmps7mR2eMMVZ4xTb66OVc+A8ePMC7776L6tWrZ9vm77//xqVLl1CuXDkEBgZqPIjmpaioKPWslyEhITluk5eU/b/j+frleBj3CJKDEyx6DYNpy3cL96FYkTMwMChwmzLd4jbRT7pql2K/ozkpKQnz5s1D//794erqql7+7NkzmJiYwNDQEH/88QeOHj2K4ODgfPdXkCGp8rF9oIgwIO3F/y80MoboMxySj19BPgbTER7+qH+4TfRTqRmSam5uDk9PT5w9e1ZjuaWlpfrpY/7+/rhx40aRH5u2RmgWBABIe5G1nDHGWPEUhcTERCQlJQHIGol0/vx5uLi4aGzz6tOTTp48iQoVKhR9kPhcqmpuyxljrIwplj6FJ0+eICwsDLIsg4jg6+sLLy8vREZGomrVqvD29sauXbtw8uRJqFQqWFhYICgoqOiD2DkA8bHZl5uYgjLSIQwMs69jjLEypMTPkvrGfQqSBMgyULEKpIGfQbi45r4DpnN8/Vr/cJvop1LTp6AkyccPos9wwM4REAKwc4ToPwbS8AlAQjzkGWMhR+0AybLSURljTBElfkK8gpJ8/AAfv2xVVnLzgPzDElDkKtA/JyD1Gw1hx8PwGGNlS5k6U8iLsLKFNGJi1pnEjSuQp46EfPyA0rEYY6xYcVF4hRACUot3IU1eADhXAK2cB3nlPFDSc6WjMcZYseCikAPhVB7SlyEQnXqBTh2GPHUU6NI5pWMxxpjOcVHIhVCpIL3fDdL4OYCxMeT5kyBHrgalpykdjTHGdIaLQj5E5eqQJi6AeKcDKGo75OljQf8V/d3WjDGmD7goaEEYG0PqORTS6ClAchLkbz6HvGszSM5UOhpjjBUpLgoFIGo3gDRlEVC/EWjLD5Dnfg2KfaB0LMYYKzJcFApIWFhBGjoOYsBY4O4tyFNHQz4chRJ+YzhjjAHgolAoQghIvu9ACl4EVKoKCl8Eedks0LOnSkdjjLE3wkXhDQh7J0ifzYDo2h84fxLylJGgf04oHYsxxgqNi8IbEpIE6d0PIX0dClhaQ148HfK6paAXqUpHY4yxAuOiUEREhSqQvp4P0fZD0IHfIU8bA7pxRelYjDFWIFwUipAwNIT0cX9In80AMtIhzx4HecePoIwMpaMxxphWuCjogPCoAyl4EUSjlqBfNkCePQ704I7SsRhjLF9cFHREmJlDGjgW0tAvgdgHkKePgbzvNx66yhjTa1wUdEx4N8u64a1aLdD65ZAXTQMlxCsdizHGcsRFoRgIG3tIY6ZA9BwKXDkPeepI0OkjSsdijLFstCoKFy5cwKNHjwAAT548wZIlS7B06VIkJCToNFxpIoSA9E4HSJMWAPZvQV4WAvn7BaCUZKWjMcaYmlZFYfXq1ZCkrE3Xrl2LzMxMCCGwYsUKnYYrjUS5CpDGz4F4vxvo2L6sZzVcvaB0LMYYA6BlUYiPj4eDgwMyMzNx7tw5DB06FIMHD8bVq1d1na9UEgYGkDr1gjQuBJAkyPO+hrwpHJSernQ0xlgZp1VRMDU1RUJCAqKjo1GhQgWYmJgAADJ4/P0bEVVrQJq8EKJ5W9DvWyB/8zno7r9Kx2KMlWEG2mzUrl07fPXVV8jIyEC/fv0AAJcvX4aLi4sus5UJwsQUos9wUN1GkH9YBHnGWIgP+0L4fwAh8TgAxljxEqTlwPl79+5BkiQ4OzurX2dkZMDV1VWnAbXJVRgODg6Ii4sr4jRvhhITIK9dApw7DnjUgdR/DIS9o9KxipU+tktZx22in96kXcqXL5/rOq3/FHVycsKTJ09w5EjWUEo7Ozs4OTkVKhDLmbCygTT8a4i+I4Bb1yBPHQX52D6+4Y0xVmy0unz033//Yfbs2TA0NMTjx4/RpEkTREdHY//+/Rg7dqyuM5YpQoisPgaPOpC//xa0ej7wzwmg1zAIc0ul4zHGSjmtzhRWrlyJbt26YcGCBTAwyKojnp6euHz5sk7DlWXCqRykL2dBdO4NOn0k61kN0WeUjsUYK+W0OlO4c+cOmjdvrrHMxMQEaWlpWh0kLS0NwcHByMjIQGZmJnx8fBAQEKCxTXp6OpYsWYIbN27A0tISY8aMKfOXp4SkgugQAKrtBXn1fMjfBkO07gjxUV8II2Ol4zHGSiGtzhQcHR1x48YNjWUxMTHqTuf8GBoaIjg4GHPnzsWcOXNw9uzZbPc47NmzB+bm5li8eDE6dOiA9evXa/kRSj9RqSqkifMhWncE/fUL5Bmfgv69rnQsxlgppFVR6NatG0JCQrBx40ZkZGRg69atmD9/Prp3767VQYQQ6nsbMjMz1XdEv+rkyZPw8/MDAPj4+ODChQvcwfoKYWQMqftgSGOnAilJkGd9DvnXjSA5U+lojLFSRKvLR15eXvjqq6+wZ88eeHp6IjY2Fp9//jnc3Ny0PpAsyxg3bhwePHiAd999F9WrV9dYHx8fD3t7ewCASqWCmZkZnj17BisrK43toqKiEBUVBQAICQmBg4OD1hleZWBgUOj3KqpFG8hvN0bid3PxYts6GFw+B6vRk2HgXDruGSmx7VKKcZvoJ121i1ZFAQDc3NwKVAReJ0kS5s6di6SkJMybNw///fefxj0OOZ0VvH42AQD+/v7w9/dXvy7sON2SPvaa+o6CqFEf6euX4/GYvhDdBkI0a5Pj16wkKentUhpxm+gnRe9TmDdvHi5duqSx7NKlSwgNDS1wGHNzc3h6euLs2bMay+3t7fH48WMAWZeYkpOTYWFhUeD9lxVCCEiNW2Y9q6FyNdDaJZCXfgNK5JlrGWOFp1VRiI6OhoeHh8Yyd3d3XLx4UauDJCYmIikpCUDWSKTz589nmyLDy8sL+/btAwAcO3YMtWrVKvF/9RYHYecI6dPpEAEDgQuns4aunjuudCzGWAml1eUjQ0NDpKamwszMTL0sNTUVKpVKq4M8efIEYWFhkGUZRARfX194eXkhMjISVatWhbe3N1q1aoUlS5Zg5MiRsLCwwJgxYwr3icogIUkQbTqBPOtDXjUf8pIZEM3bQgQMhDAxVToeY6wE0Wruo6VLlyItLQ1DhgyBmZkZkpOTsWrVKqhUKgwfPrw4cuaqNM19VBQoPR20fT3oj62Aw1uQBn4KUbWG0rG0VlrbpSTjNtFPivYp9O3bFykpKRgwYAAGDRqEAQMGIDk5WT1jKtMfwtAQUtd+kD6fCcgy5NnjIW9bB+JpzhljWtB6llQASEhIQFxcHBwcHGBjY6PLXFrjM4XcUUoy6KfvQEf3AJWqZZ01lKugdKw8lYV2KWm4TfST4rOkAlkjXiwtLfHixQs8fPgQDx8+LFQgVjyEqRmkAWMgDRsPPH4IefoYyHt28k2BjLFcadXRfPbsWSxbtgwJCdmHO0ZGRhZ5KFa0hFcTSFVrQP5hcdaZw7kTkPqPgrCxVzoaY0zPaFUUVq9ejS5dusDPzw9GRka6zsR0QNjYQRo1GbR/F+jn7yFPGQWpTxCEV1OlozHG9IhWl4+eP3+ONm3acEEo4YQQkPzegzRpIeDoDHn5bMirvwUlJykdjTGmJ7QqCq1atcLevXt1nYUVE+HsAmncbIiO3UHH90OeOgp05YLSsRhjekCry0fXrl3Drl27sH379myjjqZOnaqTYEy3hIEBxAc9//eshm8hh34N0aYzROfeEIaGSsdjjClEq6LQqlUrtGrVStdZmAKEmwekyQtAP38P+mMr6OJpSIM+hahQRelojDEFaFUUXj7ngJVOwtgEoncQqG5DyD8shjzzM4jOfSDafAAhaTeVCWOsdNB66uyEhATExMTg2bNnGuPc+Qyi9BB1G0Kashjy2jDQpjWgf05AGjAGwr5sPxaVsbJEq6Jw/PhxLF68GOXKlcPt27dRsWJF3L59GzVq1OCiUMoIS2tIQV+BjuwBbfgO8tRRED2GQvj48ay1jJUBWo0+ioyMRFBQEObMmQMTExPMmTMHQ4YMQZUqfN25NBJCQGraGtLkhYBLZdD330JeMRv0PFHpaIwxHdOqKMTFxcHX11djWcuWLXHgwAGdhGL6QTg6Q/piJsRHgcDZ45CnjAJdOK10LMaYDmlVFKysrNRTXDg6OuLq1at4+PAhZFnWaTimPCGpILXvAmnCXMDMHPLCKZB/XA568ULpaIwxHdCqT6F169a4fPkyfHx80KFDB0ydOhVCCLz//vu6zsf0hHCtCmnSt6Ata0FRO0CXzmXNulq5utLRGGNFqEBTZ78UFxeH1NRUVKig/DTMPHV28aNL5yCvWQgkPoF4vxtE+48htHwKX364XfQPt4l+UnTq7Dlz5mQLU6FCBcybN69QgVjJJmrWgxS8CMKrKWj7j5DnjAc9KlxxZozpF62KwsWLFwu0nJV+wtwC0uDPIQZ9Bjy4A3naGMgHfudnNTBWwuXZp/DyWQkZGRnZnpvw8OFDODo66i4ZKxGkxi1B1WtBDl8IiggDnTsOKXAEhJWt0tEYY4WQZ1F4/PgxAECWZfX/X3JwcEBAQIDukrESQ9g5QBozFbRnJ2jzD1nPaug7HKK+j9LRGGMFlGdRCAoKAgC4u7vD39+/WAKxkklIEoT/ByDP+pBXhUIO+waiWRuIbgMhTMyUjscY05JWfQo1atRQ36eQmpqKjRs3YtOmTXjBY9XZa0R5V0gT5kG07wo6/A8+u3YAACAASURBVBfkaWNAMdFKx2KMaUmrorBw4UIkJycDANauXYtLly7h6tWr+O6773QajpVMwsAQ0kd9IX3xDUAEec4EyFsjQBnpSkdjjOVDq5vXYmNjUb58eRARTpw4gdDQUBgZGWHEiBG6zsdKMFHdE9LkhaDIVaDffgZdOJV1w1t5V6WjMcZyodWZgqGhIVJSUhATEwN7e3tYWVnB0NAQ6en8lx/LmzA1g9RvFKRPvgLi4yDP+BTyX7+AeIoUxvSSVmcKTZs2xbRp05CSkoJ27doBAG7evAknJ55nn2lHNPCFVLUG5B8WgzaszBq62m80hJ2D0tEYY6/QepqLc+fOQaVSoXbt2gCA69evIyUlRf1aKTzNRclCRKADv4M2rgYMDCB6B0Fq2Fy9nttF/3Cb6CddTXNRqLmPCiouLg5hYWFISEiAEAL+/v547733NLa5ePEi5syZoz77aNy4Mbp27ZrvvrkolEz08B7k1fOBm1chGrUEudcCfvsZeBIH2DpAfNgHko+f0jEZ+GdFX+mqKOR6+WjmzJn4+uuvAQCTJ0/O9albU6dOzTeASqVCnz594ObmhpSUFIwfPx5169bNNqFezZo1MX78+Hz3x0o+8VZ5SONmg3b9DNr+I3B8//+vjI8FRYRBBrgwMFbMci0KLVu2VP//TR+5aWtrC1vbrGkPTE1N4eLigvj4eL2YZZUpR6hUEO93R+beXUDiE82VaS9AWyMALgqMFatci0KzZs3U//fz8yuyAz569Ag3b95EtWrVsq27evUqvvjiC9ja2qJPnz6oWLFikR2X6bHEhJyXx/MlC8aKW759Cmlpadi7dy8uXbqEpKQkmJubw9PTE35+fjAyMirQwVJTUxEcHIyPPvoIjRs31liXnJwMSZJgYmKC06dPIzw8HIsWLcq2j6ioKERFRQEAQkJCkJaWVqAMLxkYGCAjI6NQ72VFK3bIh5BjH2ZfYWQMh2UbobLjiReVxD8r+ulN2iWv3915FoXk5GRMmjQJz58/R506dWBnZ4f4+HicP38eFhYWmD59OszMtJvXJiMjA7Nnz0a9evW0emLb8OHDMWvWLFhZWeW5HXc0l3zysX2giDAg7ZVpU1QqQCbAxATio0CIFu9CSFrdVsOKGP+s6Kdi72gGgG3btsHKygozZ86EiYmJenlqairmzp2Lbdu2oWfPnvkGICIsX74cLi4uuRaEhIQEWFtbQwiBmJgYyLIMS0vLfPfNSj7Jxw8ykNWH8MroI1HFHfK6paD1y0DH9kLqMwLChe+GZkyX8jxT+PzzzxEUFAQ3N7ds627cuIGwsDCEhobme5DLly9j8uTJcHV1VY9i6tGjh7rKtW3bFrt378Yff/wBlUoFIyMj9O3bFx4eHvnum88USpfX24WIQEf3gDZ+D6SmQLT7CKJDAIRhwS5dssLjnxX9pMiZQmxsLFxdc/7LzNXVFbGxsVoFqFGjBjZu3JjnNu3atVPfLc3YS0IIiCatQXW8QRtXg37dCDpxKOt5DR51lI7HWKmT70VaA4Oc64aBgUGu9y4wVtSEpTWkgZ9CGjsVIBnyvK8hhy8CJT1TOhpjpUqeZwrp6enZHsP5Kh6RwIqb8HwbUvBi0M4NoD+2gv45AdFtEESjFvxHCmNFIM+i0KxZs2yP4XxV06ZNizwQY/kRxsYQXQJBjVpAXrsEtCoUdHQPpF6fQDg6Kx2PsRKtWOY+0iXuaC5dCtouJGeC9v4G2roOoEyID3pB+H8AoVLpMGXZwj8r+klXHc088JuVaEJSQWrdEdK0JUDN+qBNayDP/BR065rS0RgrkbgosFJB2DlCGv41pGHjgcSnkL/5AnLkalBqitLRGCtRtHrIDmMlgRAC8GoCqWZd0Ja1oKjtoNNHIPUaBlG3odLxGCsR+EyBlTrCzAJS7yBI40IAYxPIi6dDXjEH9PRJ/m9mrIzL9UzhwoULWu1A6SevMZYbUc0T0uQFoN1bQL9GgqLPQHTpB9GsDc+jxFguci0Ky5Yt03gdHx8PIQQsLS3x7NkzEBHs7e2xZMkSnYdkrLCEgSHE+91A3k0hRywFRYT9bx6l4RDleGp2xl6Xa1EICwtT/3/Lli14/vw5unXrBmNjY7x48QKRkZE8YR0rMYRzBUifzwQdjgL9vAbytNEQ7btCtP8YwtBQ6XiM6Q2tzqF//fVX9OzZE8bGxgAAY2Nj9OzZEzt37tRpOMaKkhACUrM2kKYvhWjQFPTLBsjTRoOuaneplLGyQKuiYGJigpiYGI1l169fVxcJxkoSYWUDafBnkEYHA+lpkOdOyLozOum50tEYU5xWQ1K7deuGb775Bl5eXrC3t8fjx49x+vRpDBw4UNf5GNMZUdsL0tQloB0/ZQ1fPXccovtgCO9mPI8SK7O0nubizp07OHbsGJ48eQJbW1v4+PigQoUKus6XL57monRRql3ov+uQ14YB/8YAdbyz7m2wdyr2HPqIf1b0k66muSjQ3EeyLOPp06ewtbUtVBBd4KJQuijZLpSZCdq7E7RtPUAE0bk3RKv3y/w8Svyzop8UecjOS0lJSVi1ahWOHTsGAwMDRERE4OTJk4iJiUH37t0LFYoxfSNUKgj/TqC3fSGvX571UJ+/92c90Me1qtLxGCsWWnU0r1y5EmZmZli6dKn6oTvu7u44cuSITsMxpgRh7wRp5CSIIV8CT+Igz/wM8s/fg16kKh2NMZ3T6kzh/PnzWLFihcZT2KysrPD06VOdBWNMSUIIiIbNQJ71QVt+AP2xDXTqSNYzG+p4KR2PMZ3R6kzBzMwMz55pPvYwLi5Or/oWGNMFYW4Bqc9wSF/MAgyNIC+aCnnlPFAiz6PESietikLr1q0RGhqKCxcugIhw9epVhIWFoU2bNrrOx5heEO61IE1eCNGxB+j0EciThkM++AdK+DOqGMtGq9FHRITffvsNUVFRiIuLg4ODA/z9/fHee+8pPp6bRx+VLiWhXej+HcgRS4Br0YB7bUh9giCclR+erSsloU3KIsWGpMqyjH379qF58+Yw1MM5YrgolC4lpV1IlrPmUdq0Bkh7AdEhAKJdFwgD/fsZeVMlpU3KGsUexylJEtauXauXBYExpQhJgtS8LaRpSyHe9gVt/xHytDGgmGilozH2RrTqU/Dy8sLJkyd1nYWxEkdY20Ia8gWkUZOBF6mQZ4/PmqI7medRYiWTVkNS09PTMX/+fLi7u8Pe3l6jH2HEiBE6C8dYSSHqeP9vHqUfQVG/gM79DanHEKBBE8X73RgrCK2KQsWKFVGxIj+QhLG8CBNTiICBoMYtIa9dAnn5bKBeI0g9h0LYOSodjzGtFGjuI33EHc2lS2lpF8rMBP21A7T9R0BIEJ17QbTqACGVvHmUSkublDaKzn0EABkZGbh37x4SExM1lmvzjOa4uDiEhYUhISEBQgj1cNZXERHWrFmDM2fOwNjYGEFBQXBzc9M2HmN6RahUEG0/BDVoAnn9MlDkqqx5lPoMh3Dl72umv7QqCpcvX8b8+fORnp6OlJQUmJqaIjU1VetnNKtUKvTp0wdubm5ISUnB+PHjUbduXY2pt8+cOYMHDx5g0aJFuHbtGlatWoVvvvmm8J+MMT0gHN6CNCoYdOIgaMNKyDM/hWjTGaJjDwh+SBXTQ1qNPvrhhx/wwQcfYM2aNTA1NcWaNWvQpUsXtG3bVquD2Nraqv/qNzU1hYuLC+Lj4zW2OXnyJFq0aAEhBNzd3ZGUlIQnT3gqAVbyCSEgNWqR9RjQJq1Bv2+BPGUE6OIZpaMxlo1WReHevXvZLvd07twZv/76a4EP+OjRI9y8eRPVqlXTWB4fHw8HBwf1a3t7+2yFg7GSTJhbQgocCenzbwCVAeQFwZBXhYISE5SOxpiaVpePzMzMkJKSAnNzc9jY2ODOnTuwsLBAamrBphJOTU1FaGgo+vXrBzMzM411OfV35zSULyoqClFRUQCAkJAQjUJSEAYGBoV+L9OdMtEuDn6ghr5I2hyBpC1rgYtnYNF/JEzeUX7amJyUiTYpgXTVLloVhcaNG+PMmTNo1qwZWrVqhalTp0KlUsHX11frA2VkZCA0NBTNmzdH48aNs623t7fX6El//PhxjrOw+vv7w9/fX/26sL3vPKJCP5WpdmnTGVKtBpAjwpC4eCYS/9iR1RH9Vu4jQ5RQptqkBFF09FG/fv3U/+/YsSOqVauG1NRU1KtXT6sARITly5fDxcUF77//fo7beHt7Y/fu3WjatCmuXbsGMzMznpqblXqivCukL2aBDv4B2vwD5CkjId7vBvHuh6VyHiWm/4rlPoXLly9j8uTJcHV1VZ8e9+jRQ13l2rZtCyLC6tWrce7cORgZGSEoKAhVq+b/CES+T6F0KcvtQgnxoA0rQacOA+VdIfUdAVG1htKxynSb6DPFZkkFgMmTJ+d6rXPq1KmFClVUuCiULtwuAJ07DvnH5cCTxxAt20N82AfCzFyxPNwm+knRy0etWrXSeJ2QkIC9e/eiefPmhQrEGMudqNcIkkdt0Lb1oD07QWePQeoxFKKB9n14jBWWVkXBz88v2zIfHx8sXboUXbt2LepMjJV5wsQMovtgUGO/rHmUls0C6jfOKg52PBKI6Y5W9ynkxM7ODv/++29RZmGMvUZUqQ7p61CIrv2A6DOQg4dD3rMTJGcqHY2VUlqdKezZs0fjdVpaGv7++2+4u7vrJBRj7P8JAwOIdz/Kmkdp3TLQT9+Bju2D1Hc4RIUqSsdjpYxWReHgwYMar42NjeHh4YEOHTroJBRjLDvh6AxpzBTQ3/tBG1dDnvEpRNvOEO93hzDieZRY0eCps5le4XbRDj1PBG1aAzr8F+DoDKl3EIRnfZ0ci9tEPyk6+ujhw4daHeitt97SLhFj7I0ICyuIfqOzOqLXLYX87WQI33cgPh4IYWmldDxWgmlVFEaNGqXVziIjI98oDGOsYETNepCCF4F+/Rn0+2bQ+ZNZhcH3Hb2cR4npP62KwrBhw3D+/Hl8/PHHcHR0RGxsLDZt2oQ6derkOFyVMVZ8hJExxIe9QY2aQ44IA61ZADq2F1LvTyCc9GseJab/tBqSGhkZiWHDhqFcuXIwMDBAuXLlMGTIEGzYsEHX+RhjWhIulSB9GQLRaxhw6xrkKaMg79oEyshQOhorQbQqCkSER48eaSyLjY2FLMs6CcUYKxwhSZD83oM0NQyo4wXashbyjLGgG1eUjsZKCK0uH3Xo0AHTpk2Dn5+fusd7//79PCSVMT0lbO2h+uQr0NljkNevgBzyJYTfe1nzKJma5b8DVmZpVRQ++OADuLq64ujRo7h16xZsbGzwySefoH593QyBY4wVDVHfB5JHXdC2daC9v4LOHIPUayhEfR+lozE9xfcpML3C7aI7dOMK5LVLgLv/Ag18IXUfAmFrn+/7uE30kyL3KZw9exampqbw8PAAADx48ABhYWH477//4O7ujqCgIH4QDmMlhHDzgDTxW9Cf20C/bIB8aTjEh30hWraDkAo9DRorZfL8ToiMjNQY67xs2TKYmZlh9OjRMDY2RkREhM4DMsaKjjAwgNS+K6Qpi4HK1UE/Loc8ZzzoLk9uybLkWRQePHigfvrZ06dPceXKFQwdOhQNGjTAkCFDcPHixWIJyRgrWsKpHKSx0yAGjAUe3oU8fQzkretA6WlKR2MK0/qc8erVq3BycoKdnR0AwNLSEqmpqToLxhjTLSEEJN93IE1bBtGoBei3jZCnjAJd/kfpaExBefYpVKtWDbt27UKrVq3w119/aYw2evjwISwtLXUekDGmW8LSCmLAWJDPO1nzKIVOhGjSGlTFHdi1CQ+fxAG2DhAf9oHk46d0XKZjeZ4pBAYG4vfff0f//v1x//59dO7cWb3uwIEDqFmzps4DMsaKh/CsD2nKYoj2XUFH9wDrlwHxsQAREB8LigiDfGyf0jGZjmk1JPXZs2fZzgqSkpJgYGAAY2Nl53HnIamlC7eLfsj8PBB4+iT7CjtHqGavLv5ALBtFp87O6TKRubl5ocIwxkqApwk5L4/ngl3a8eBkxlh2dg45L5ck0JULxZuFFSsuCoyxbMSHfYDXH/FpYAiYmkGeNwHy9wtAzxKVCcd0KteisHv3bvX/Hzx4UCxhGGP6QfLxg+gzHLBzBIQA7BwhAkdCmv19Vkf08f2QJ30C+dCfIJ4tuVTJtaM5MDAQP/zwQ7b/6xvuaC5duF30T05tQnf/g7xuKRATDVT3hNQrCMLFVaGEZVOxdzQ7Oztj7dq1qFChAjIyMrBnz54ct2vVqlWhQjHGSi7h4grpi29AR/4CbQqHPH00RNsPITp0g1B4RCJ7M7kWhdGjR2PHjh04fPgwMjMzcfDgwRy346LAWNkkJAmiWRtQvcagTWtAuzaBThyE1HMYRB0vpeOxQtLqPoVp06Zh8uTJxZGnwPjyUenC7aJ/tG0TunIh65LSgzuAVxNI3QdD2OQ/NTcrHF1dPtL6eQqZmZm4cuUK4uPjYW9vD3d3d6hUKq0CLF26FKdPn4a1tTVCQ0Ozrb948SLmzJkDJycnAEDjxo3RtWtXrfbNRaF04XbRPwVpE0pPB/2xFfTrRkClgujcB+Kd9hCSdr8rmPYUvXnt3r17CAkJQVpaGuzt7fH48WMYGhpi3LhxqFChQr7v9/PzQ7t27RAWFpbrNjVr1sT48eO1icMY01PC0BCiQwCoYXPI65eDNnwHOroHUp8giErVlI7HtKBVUVi5ciX8/f3RsWNH9fMVduzYgdWrVyM4ODjf93t6euLRo0dvlpQxVmIIp3KQxkwBnTwEilwFeebnEK06QHTqxc+I1nNaFYVbt25h0qRJGg/c6dChA7Zu3VpkQa5evYovvvgCtra26NOnDypWrJjjdlFRUYiKigIAhISEwMEhlzsv82FgYFDo9zLd4XbRP2/UJu0/hNyiDZ6vX4GU3VsgzhyFxcAxMPZ9R+P3CSs4Xf2saFUU7OzsEB0djdq1a6uXXbp0qcgexVmlShUsXboUJiYmOH36NObOnYtFixbluK2/vz/8/f3Vrwt7TY2vXesnbhf9UyRt8lEgpLd9IUeE4enciUBtL0g9h0I4OhdNyDJI0T6FHj16YPbs2fDy8lIHOX36NEaOHFmoQK8zM/v/08kGDRpg9erVSExMhJWVVZHsnzGmPFHFHdLX80F7d4K2/Qh5ygiI97tDtOkEYWCodDz2P1rNfeTt7Y3Zs2ejYsWKSE1NRcWKFRESEoKGDRsWSYiEhAS8HAQVExMDWZb5AT6MlUJCpYLk3wnStDCgthdoy1rI08eCrkUrHY39j9ZDUt/EggULEB0djWfPnsHa2hoBAQHIyMgAALRt2xa7d+/GH3/8AZVKBSMjI/Tt2xceHh5a7ZuHpJYu3C76R5dtQudOQP5xORAfC9GsDUSXQAgLvkKgDcXvU9BXXBRKF24X/aPrNqEXqaBfNoCitgOm5hAf94fwbcUd0fnQVVHgqbMZY4oSxiaQuvaDNPFb4K3yoDULIc/7GnT/ttLRyiQuCowxvSAqVIb0ZUjWlN13bkGeOhry1nWgtBdKRytTuCgwxvSGkCRILd6FNH0pRMPmoN82Qp4yEnTxjNLRygythqTGxcXh559/xq1bt5CamqqxbuHChToJxhgru4SVDcTAsaAmrSCvXw55QTBEw+YQAQMhbOyUjleqaVUU5s+fj/LlyyMgIABGRka6zsQYYwAAUbMepOBFoN2bQb/9DLpwCuLDvhAt3+VJ9nREq6Jw9+5dzJgxA5LEV5sYY8VLGBpCdOwOatQC8vploB+XZ02y1/sTCNeqSscrdbT6Le/l5YXoaL65hDGmHPFWeUhjp0EM+gyIewh5xmeQI1eBUpOVjlaqaHWmMGDAAEycOBFvvfUWrK2tNdYFBQXpJBhjjL1OCAHRuCWothdo61rQX7+ATh2B1H0w8LYP39tQBLQqCkuXLoUkSXBxceE+BcaY4oS5BUTvIJBvK8jrlkJeNguo2zBrkj17J6XjlWhaFYULFy5gxYoVMDU11XUexhjTmqhaA9LEb0F/7QBt/xHy5OEQH/SAaP0BhIFWv97Ya7TqU6hUqRKePXum6yyMMVZgQqWC1PZDSNOWAjXrgTaFQ54xFhRzSeloJZJWpbRWrVqYOXMm/Pz8svUptGrVSifBGGOsIIS9I1QjJoLOHoP803eQZ4+DaN42a5I9c551WVtaFYUrV67Azs4O//zzT7Z1XBQYY/pE1PeBVKMe6JefQFE7QGf/hvh4AISPH3dEa4FnSWV6hdtF/5TkNqHbNyFHhAE3rwI16kLqNQzCuYLSsYqEorOkyrKc6z/GGNNXomIVSOPnQPT6BPj3OuSpoyBv/xGUnqZ0NL2l9eM4cxMZGVlkYRhjrKgJSYLwaw962we08XvQzg2g4weyzho86ysdT+9oVRSWLFmi8frJkyfYtm0bvL29dRKKMcaKmrC2hRj8Gajp/ybZ+3YyRKOWEN0GQFjZKh1Pb2h1+cjR0VHjn7u7O0aMGIHt27frOh9jjBUp4fk2pCmLId7vDjp9GPLEIMj7doH4cjiAN3ieQnJyMhITE4syC2OMFQthaASpU09IwYsAVzfQ+mWQZ48D3b6pdDTFaXX5aPHixRpDuV68eIFLly6hefPmOgvGGGO6JpwrQPpsBujYPtDP30OeMRbC/wOIjj0gTMrmDA5aFQVnZ2eN18bGxmjTpg3q1q2rk1CMMVZchBAQvu+A6nqDNv8A+mMb6OQhSD2GQtRvrHS8Ysf3KTC9wu2if8pam1BMNOR1y4C7/wL1G0PqPgTC3lHpWNno6j6FPM8Ubty4AQMDA7i6ugIAEhMTER4ejtu3b6N69ero27cvTExMChWKMcb0kajmmTXJXtR20C8/QQ5+ZZI9Vel/2lueHc3h4eFISEhQv16+fDnu37+P1q1b4/bt21i3bp3OAzLGWHETBgaQ2nWBNDUMcK8N+nkN5Bmfgq5fVjqazuVZFO7evYuaNWsCAJKSknDmzBmMHDkS7dq1w+jRo3Hq1KliCckYY0oQDm9BGjkJ0ifjgeeJkGePg7xuKSj5udLRdCbPy0eZmZkw+N+c5NeuXYONjY36WpSDgwOSkpJ0n5AxxhQkhAAaNIHkWR+0/UfQXztBZ45BBAyEaNSi1E2yl+eZQsWKFXH06FEAwOHDh1GnTh31uvj4eJiZmek2HWOM6QlhYgap2yBIX4cCtg6gVaGQFwSDHhZusIu+yrMo9OrVCytXrkT//v1x+vRpdO7cWb3uyJEj8PDw0HlAxhjTJ6JSVUgT5kL0HArcvAp5ykjIv2wApacrHa1I5DskNSUlBffv30e5cuU0Hsd57949mJiYwM7OLt+DLF26FKdPn4a1tTVCQ0OzrScirFmzBmfOnIGxsTGCgoLg5uam1QfgIamlC7eL/uE2yR0lPM6aZO/EQeAtF0i9P4GoUTz3byk2dbapqSnc3NyyPZ+5fPnyWhUEAPDz88OECRNyXX/mzBk8ePAAixYtwpAhQ7Bq1Sqt9ssYY0oSNvaQhnwBaXQwIGdCDp0IefW3oMSE/N+spwo991FBeHp6wsLCItf1J0+eRIsWWR027u7uSEpKwpMnT4ojGmOMvTFR2ytrkr33AkAnDkKeFAT5wO8lcpI9raa50LX4+Hg4ODioX9vb2yM+Ph62ttmns42KikJUVBQAICQkRON9BWFgYFDo9zLd4XbRP9wmBTB4DDLadUbi8jlIjwiDwYmDsBr2BQwqVS3yQ+mqXfSiKOTUrZHbMC9/f3/4+/urXxf2mhpfJ9VP3C76h9ukgEwtQGOmQhzZg/RN3+PxZ/0g/DtBdOwOYVx0M0AoMs1FcbG3t9f4cI8fP87xLIExxkoCIQRE09agug1Bm8NBv2/5/0n26jVUOl6eiqVPIT/e3t44cOAAiAhXr16FmZkZFwXGWIknLK0g9RsF6YtvACNjyEumI3PZLFC8/p55FcssqQsWLEB0dDSePXsGa2trBAQEICMjAwDQtm1bEBFWr16Nc+fOwcjICEFBQahaVbtrcDwktXThdtE/3CZFgzLSs6bl3hkJSCqIzj0h3nm/0JPs6eryEU+dzfQKt4v+4TYpWhT7APKPK4ALpwBXN0i9h0NUqV7g/Sh2nwJjjLGiIxydIY2aDGnol8DTBMizPof843JQsn7MJacXHc2MMVaWCCEA72aQajUAbVsH2vsr6PRRiG6DILybKTrJHp8pMMaYQoSpGaQeQyBNmAdY24G+mwt54RTQo/uKZeKiwBhjChOVq0P6eh5E98HA9ctZk+z9uhGUUfyT7HFRYIwxPSAkFaTWHSFNWwrU9QZtWwd52hjQlQvFmoOLAmOM6RFhaw/VsPGQRk0G0l5AnjcB8pqFoGeJxXJ87mhmjDE9JOp4Q5oaBvp1Q9b9DeeOQ3TtB1IZANvW4eGTOMDWAeLDPpB8/IrsuFwUGGNMTwljY4iPAkGN/bKeDf3DYkAI4OXtZfGxoIgwyECRFQa+fMQYY3pOuFSC9MUswNzy/wvCS2kvQFsjiuxYXBQYY6wEEJIEJD3PeWURzqXERYExxkoKu1yen5Db8kLgosAYYyWE+LAPYGSsudDIOGt5EeGOZsYYKyEkHz/IQFYfAo8+YowxJvn4AT5+Opu9li8fMcYYU+OiwBhjTI2LAmOMMTUuCowxxtS4KDDGGFMr8c9oZowxVnTK7JnC+PHjlY7AcsDton+4TfSTrtqlzBYFxhhj2XFRYIwxplZmi4K/v7/SEVgOuF30D7eJftJVu3BHM2OMMbUye6bAGGMsOy4KjDHG1MrcLKlxcXEICwtDQkIChBDw9/fHe++9p3SsMi0tLQ3BwcHIyMhAZmYmfHx8EBAQoHQs9j+yLGP8+PGws7Pj4al6YPjw4TAxMYEkSVCpVAgJCSnS/Ze5oqBSqdCnTx+4ubkhJSUF48ePR926dVGhQgWlo5VZhoaGCA4OhomJDWLZcwAACJVJREFUCTIyMjB58mTUr18f7u7uSkdjAH777Te4uLggJSVF6Sjsf4KDg2FlZaWTfZe5y0e2trZwc3MDAJiamsLFxQXx8fEKpyrbhBAwMTEBAGRmZiIzMxNCCIVTMQB4/PgxTp8+jdatWysdhRWTMnem8KpHjx7h5s2bqFatmtJRyjxZljFu3Dg8ePAA7777LqpXr650JAYgPDwcvXv35rMEPTNz5kwAQJs2bYp8aGqZLQqpqakIDQ1Fv379YGZmpnScMk+SJMydOxdJSUmYN28e/vvvP7i6uiodq0w7deoUrK2t4ebmhosXLyodh/3P9OnTYWdnh6dPn2LGjBkoX748PD09i2z/ZbIoZGRkIDQ0FM2bN0fjxo2VjsNeYW5uDk9PT5w9e5aLgsKuXLmCkydP4syZM0hLS0NKSgoWLVqEUaNGKR2tTLOzswMAWFtbo2HDhoiJieGi8CaICMuXL4eLiwvef/99peMwAImJiVCpVDA3N0daWhrOnz+PTp06KR2rzOvZsyd69uwJALh48SJ++eUXLggKS01NBRHB1NQUqamp+Oeff9C1a9ciPUaZKwpXrlzBgQMH4Orqii+++AIA0KNHDzRo0EDhZGXXkydPEBYWBlmWQUTw9fWFl5eX0rEY0ztPnz7FvHnzAGQNymjWrBnq169fpMfgaS4YY4yplbkhqYwxxnLHRYExxpgaFwXGGGNqXBQYY4ypcVFgjDGmxkWBlVgbN27EokWLlI5RIn3zzTfYt29fjusePXqEgIAAZGZmFm8ophfK3H0KrGQ5dOgQdu7cibt378LU1BSVK1fGRx99hBo1ahTJ/h89eoQRI0bgp59+gkqlKpJ9All3zW/ZsgWHDh3CkydPYGVlhVq1aqFr165wcnJ6o31PmTIFzZs3f6NJ6iZMmPBGGVjpxUWB6a2dO3di27ZtGDx4MOrVqwcDAwOcPXsWJ06cKLKi8KYyMzNzLCahoaGIj4/HqFGjUKVKFbx48QL/1979hTTVxgEc/zpN53DqcpaOrsLsyoUgkuSf/glhf2gSRlctC4sS+kckgUQQsaB/asKgxkyM8CIbBAa1rCQcTVIxKyyJqFDDmWuKOnTrvYj34F6nr7wEr9Tvc7Vzzu95zu/s5rfnOeN5Wltb6enpYePGjf9LTkIshBQFsSiNj4/T2NjI4cOHQ9anysrKIisra1b869evqampwWq1KueOHDnCwYMHMRqN9PX1cfPmTQYGBoiOjiY3N5e9e/dy9uxZAMxmMwCVlZWkp6fT0tLC/fv38Xq9pKWlUVZWRnJyMgAlJSWUlpbS3NxMIBCgtrY2JJfu7m66u7upqqpCr9cDoNFo2LJlS8jz3bp1i87OTiIiItiwYQMlJSWoVCqePn3K48ePWbVqFU+ePEGj0XDgwAEyMzO5c+cOb9++5f3799TV1bF+/Xr2798fNqfe3l7q6uro7+/HYDBgNptZvXo1EDraCAaDNDQ08OzZM2JjY2X5lz+cFAWxKL17946pqSmys7N/SX92u52ioiLy8/OZnJzk06dPAJw7d47y8nLq6uqUX9dut5t79+5x+vRpUlNTcTgcVFVVcf78eaW/9vZ2Lly4QHR09Kx7vXr1irS0NKUghHP9+nUSExOprq7G7/djsVhISkqisLAQgL6+PgoKCrDZbDidTqxWK1arlT179tDb2xt2+mhmTmNjY1gsFvbt28e6detwuVxYLBaqq6vRarUh7ZxOJx0dHVy8eBG1Ws3ly5f/25csfgvyolksSqOjo2i12l82DRIVFcXg4CA+nw+1Wj3vrm5OpxOTycSKFSuIjIzEZDLx8eNHhoaGlBiTyURcXFzYojA6OopOp5uzf6/XS1dXF2azGbVaTUJCAlu3bqWtrU2J0ev1bN68GZVKRUFBASMjI3z//n3eZ5yZU0dHBykpKeTn5xMZGUlubi4Gg4GXL1/OaudyuSgqKkKv1xMXF8fOnTvnvY/4vclIQSxKWq2W0dHRXzY/fujQIRobGzl+/DjLli1j165dcy66NzQ0hN1up76+Xjn348cPvn37pkwhJSUlzZv7wMDAnNc9Hg+BQICysrKQ/mf2mZiYqHyOiYkBfq6QOZ+Z7Wfm+rfk5OSwuwyOjIyEjGr+2U78WaQoiEUpPT2dJUuW0N7eztq1a/81PiYmBr/frxwHg0F8Pp9ynJqayrFjxwgGg7jdbq5cuYLNZgu77ader6e4uJi8vLw57zffdqEZGRk0NzczPDwctngkJSURFRWFzWb7TwVvrnvPPL906VJevHgRct3j8YRdUVOn0+HxeELixJ9Lpo/EoqTRaNi9ezc2mw23243f72d6eprOzk4aGhpmxRsMBqampujo6GB6epq7d+8yNTWlXG9tbcXn86FSqZSd9lQqFfHx8URERPD161cltrCwEIfDwefPn4GfL4VdLteCczcajRiNRi5dusSHDx8IBAJMTEzw8OFDWlpa0Ol0rFmzhvr6esbHxwkGgwwODvLmzZsF9Z+QkBCSbziZmZkMDAzw/PlzAoEAbW1tfPnyJewS8Tk5OTx48IDh4WHGxsZwOBwLflbx+5GRgli0tm3bRkJCAk1NTdTU1KBWq1m5ciXFxcWzYv/+h47VaiUYDLJjx46QX+ldXV3U19fj9/tJTk7m6NGjyvuA4uJiKisrCQQCnDlzhuzsbCYnJ7l27RoejweNRkNGRgY5OTkLzv3kyZM0NTVx9epVvF4vWq0Wo9GobIhSXl7O7du3OXHiBBMTEyxfvnzBGwsVFRVRW1vLo0ePyMvLo7S0dFaMVquloqICu93OjRs3SElJoaKigvj4+FmxmzZtor+/n1OnThEbG8v27dvp6elZ8LOK34vspyCEEEIh00dCCCEUUhSEEEIopCgIIYRQSFEQQgihkKIghBBCIUVBCCGEQoqCEEIIhRQFIYQQir8A9+mHZb0MSVUAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 1440x1440 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from sklearn.cluster import KMeans #import kmeans which will be used to cluster the data\n",
    "\n",
    "import matplotlib.pyplot as plt \n",
    "plt.style.use('ggplot')\n",
    "\n",
    "max_number_of_clusters = 5 #clusters can range from 2-max_number_of_clusters\n",
    "max_number_of_clusters = max_number_of_clusters+1 # since the range method is exclusive, a 1 should be added.\n",
    "Ks = range(2, max_number_of_clusters, 1) #Hold the range of clusters k (from 2-max_number_of_clusters)\n",
    "    \n",
    "sum_of_squared_distances = [] #a list which will carry the sum of squared distances at each k\n",
    "for k in tqdm(Ks):\n",
    "    sum_of_squared_distances.append(KMeans(n_clusters=k,random_state=30).fit(tf_idf_norm_kmeans).inertia_)#calculate the sum of squared distance for each k.\n",
    "        \n",
    "  #plot the elbow graph      \n",
    "f, ax = plt.subplots(1, 1)\n",
    "ax.plot(Ks, sum_of_squared_distances, marker='o')\n",
    "ax.set_xlabel('Cluster Centroid')\n",
    "ax.set_xticks(Ks)\n",
    "ax.set_xticklabels(Ks)\n",
    "ax.set_ylabel('Sum of Squared Distances')\n",
    "ax.set_title('Sum of Squared Distances VS. Clusters')\n",
    "plt.figure(figsize=(20,20))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [],
   "source": [
    "clusters = KMeans(n_clusters=2,random_state=30).fit_predict(tf_idf_norm_kmeans) #from the elbow graph we get the n_clusters\n",
    "df =FullPaperDataframe.head(6).copy(deep=True)  #create a new df which will hold the papers attribute along with the cluster number"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>full_text</th>\n",
       "      <th>clusters</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>The RNA pseudoknots in foot-and-mouth disease virus are dispensable for genome replication but essential for the production of infectious virus. 2 3</td>\n",
       "      <td>[{'text': 'word count: 194 22 Text word count: 5168 23 24 25 author/funder. All rights reserved. No reuse allowed without permission. Abstract 27 The positive stranded RNA genomes of picornaviruses comprise a single large open reading 28 frame flanked by 5′ and 3′ untranslated regions (UTRs). Foot-and-mouth disease virus (FMDV) 29 has an unusually large 5′ UTR (1.3 kb) containing five structural domains. These include the 30 internal ribosome entry site (IRES), which facilitates initiation of translation, and the cis-acting 31 replication element (cre). Less well characterised structures are a 5′ terminal 360 nucleotide 32 stem-loop, a variable length poly-C-tract of approximately 100-200 nucleotides and a series of 33 two to four tandemly repeated pseudoknots (PKs). We investigated the structures of the PKs 34 by selective 2′ hydroxyl acetylation analysed by primer extension (SHAPE) analysis and 35 determined their contribution to genome replication by mutation and deletion experiments. 36 SHAPE and mutation experiments confirmed the importance of the previously predicted PK 37 structures for their function. Deletion experiments showed that although PKs are not essential 38', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'for replication, they provide genomes with a competitive advantage. However, although 39 replicons and full-length genomes lacking all PKs were replication competent, no infectious 40 virus was rescued from genomes containing less than one PK copy. This is consistent with our 41 earlier report describing the presence of putative packaging signals in the PK region. 42 43 author/funder. All rights reserved. No reuse allowed without permission.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>VP3, and VP0 (which is further processed to VP2 and VP4 during virus assembly) (6). The P2 64 and P3 regions encode the non-structural proteins 2B and 2C and 3A, 3B (1-3) (VPg), 3C pro and 4 structural protein-coding region is replaced by reporter genes, allow the study of genome 68 replication without the requirement for high containment (9, 10) ( figure 1A ).\\nThe FMDV 5′ UTR is the largest known picornavirus UTR, comprising approximately 1300 71 nucleotides and containing several highly structured regions. The first 360 nucleotides at the 5′ 72 end are predicted to fold into a single large stem loop termed the S-fragment, followed by a The PKs were originally predicted in 1987 and consist of two to four tandem repeats of a ~48 86 nucleotide region containing a small stem loop and downstream interaction site (figure 1B) 87 (12). Due to the sequence similarity between the PKs (figure 1C), it is speculated that they 88 were formed by duplication events during viral replication, probably involving recombination. 89 Between two and four PKs are present in different virus isolates but no strain has been 90 identified with less than two PKs, emphasising their potential importance in the viral life cycle 91 (19, 20) . The presence of PKs has been reported in the 5′ UTR of other picornaviruses such as 92 author/funder. All rights reserved. No reuse allowed without permission. can occur in the absence of PKs at least one is required for wild-type (wt) replication. 104 Furthermore, competition experiments showed that extra copies of PKs conferred a replicative 105 advantage to genomes. Although replicons and full-length genomes lacking PKs were 106 replication-competent, no infectious virus was rescued from genomes containing less than one 107 PK copy. This is consistent with our earlier report describing the presence of putative 108 packaging signals in the PK region (22). 109 110 author/funder. All rights reserved. No reuse allowed without permission. Plasmid construction. 117 The FMDV replicon plasmids, pRep-ptGFP, and the replication-defective polymerase mutant 118 control, 3D-GNN, have already been described (10).\\nTo introduce mutations into the PK region, the pRep-ptGFP replicon plasmid was digested 121 with SpeI and KpnI and the resulting fragment inserted into a sub-cloning vector (pBluescript) 122 to create the pBluescript PK. PKs 3 and 4 were removed by digestion with HindIII and AatII 123 before insertion of a synthetic DNA sequence with PK 3 and 4 deleted. PKs 2, 3 and 4 were 124 deleted by PCR amplification using ΔPK 234 Forward primer and FMDV 1331-1311 reverse 125 primer, the resultant product was digested with HindIII and AatII and ligated into the 126 pBluescript PK vector. Complete PK deletion was achieved by introduction of an AflII site at 127 the 3′ end of the poly-C tract by PCR mutagenesis to create the sub-cloning vector, pBluescript 128 C11, which was then used to remove all the PKs by PCR mutagenesis using ΔPK 1234 forward 129 primer and FMDV 1331-1311 reverse primer. The modified PK sequences were removed from 130 the sub-cloning vectors and inserted into the pRep-ptGFP plasmid using NheI-HF and KpnI-131 HF.\\n132 133 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 7 Mutations to disrupt and reform PK structure were introduced using synthetic DNA by 134 digestion with AflII and AatII and ligation into a similarly digested pBluescript PK vector.\\nMutations were then introduced into the replicon plasmid as described above.\\nTo assess the effects of truncation of the poly-C-tract on replication the entire sequence was 137 removed. This was performed by PCR mutagenesis using primers C0 SpeI, and FMDV 1331- In vitro transcription. 143 In vitro transcription reactions for replicon assays were performed as described previously (28).\\nTranscription reactions to produce large amounts of RNA for SHAPE analysis were performed 145 with purified linear DNA as described above, and 1 μg of linearised DNA was then used in a 146 HiScribe T7 synthesis kit (NEB), before DNase treatment and purification using a PureLink FastQ files were quality checked using FastQC with poor quality reads filtered using the 225 Sickle algorithm. Host cell reads were removed using FastQ Screen algorithm and FMDV 226 reads assembled de novo into contigs using IDBA-UD (35). Contigs that matched the FMDV 227 library (identified using Basic Local ALighnment Search Tool (BLAST)) were assembled 228 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint into consensus sequences using SeqMan Pro software in the DNA STAR Lasergene 13 229 package (DNA STAR) (36). The SHAPE data largely agreed with the predicted structures with the stems of PK 1, 2 and 3, interacting nucleotides showed little to no reactivity, suggesting NMIA could not interact with 300 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 14 these nucleotides either due to the predicted base pairing or steric hindrance (figure 2B). The\\nNMIA reactivity for the interacting nucleotides in the stem-loops with downstream residues of 302 PK 1, 2 and 3 again largely agreed with the predicted structure, although the SHAPE data 303 suggests that there might be fewer interactions than previously predicted. However, differences 304 here could be due to heterogeneity in the formation of PKs in this experiment. The evidence 305 for loop-downstream interaction was weaker for PK4. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint\\norientation. 351 Since removal of all four PKs resulted in a significant decrease in replication, the minimal 352 requirements to maintain wt levels of replication were investigated. As near wt level of 353 replication was observed when only one PK was present, all further mutagenesis was 354 performed in a C11 replicon plasmid containing only PK 1. In addition, the orientation of PK 1 was reversed by \"flipping\" the nucleotide sequence to 367 potentially facilitate hybridisation of the loop with upstream rather than downstream sequences.\\nChanging the orientation of the PK reduced replicon replication to a similar level seen in the replication decreased until at passage three there is a 2.5 fold reduction compared to that of 398 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint passage 0 (figure 5B). Therefore, it appears that replicons with a single PK are at a competitive 399 disadvantage compared to those with two or more. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 20 of infectious virus despite being able to replicate after transfection into cells, is consistent with 448 a requirement for RNA structure within the PK region being required for virus assembly. The 5′ UTR of FMDV is unique amongst picornaviruses due to its large size and the presence 454 of multiple RNA elements, some of which still have unknown function. One of these features 455 is a series of repeated PKs varying in number from 2-4, depending on virus strain. In this study, 456 we sequentially deleted or mutated the PKs to help understand their role in the viral life cycle. 457 We also confirmed the predicted PK structures by SHAPE mapping, although there may be Although all viruses isolated to date contain at least two PKs, replicons or viruses containing a 464 single PK were still replication competent. However, replicons with more than a single PK 465 were found to have a competitive advantage over replicons with a single PK when sequentially 466 passaged. Replicons lacking all PKs displayed poor passaging potential even when co-467 transfected with yeast tRNA, reinforcing the observation of a significant impact in replication.\\nMoreover, viruses recovered from genomes with reduced numbers of PKs were slower growing 469 and produced smaller plaques. In addition, these differences were more pronounced in more PKs is functionally competent as no differences was seen between replicons congaing a single 472 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 21 copy of PK1 or PK4. This observation is consistent with a previous report of deletion of PK1, 473 along with the Poly-C-tract, with no adverse effect in viral replication (37). This also supports 474 our findings that the truncation of the Poly-C-tract to create the C11 construct had no effect on 475 replicon replication in the cell lines tested. As has been described with Mengo virus, it is 476 possible that the role of the poly-C-tract is essential in other aspects of the viral lifecycle which 477 cannot be recapitulated in a standard tissue culture system (39).\\nThe presence of at least two PKs in all viral isolates sequenced so far suggests that multiple 480 PKs confer a competitive advantage in replication. Here we showed by sequential passage that 481 replicons containing at least two PKs were maintained at a level similar to wt, but replicons 482 containing only one PK showed a persistent decline. It is unclear why some viral isolates 483 contain two, three or four PKs is still unknown, but this may be stochastic variation or may 484 reflect subtle effects of host range or geographical localisation. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint Significance is shown comparing the replication of C11 PK disrupt and C11 PK restore (Aii). Significance shown is compared to wt replicon. Error bars are calculated by SEM, n = 3, * P 673 &lt; 0.05, **** P &lt; 0.0001. 674 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 33 675 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint \\n</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Analysis Title: Regaining perspective on SARS-CoV-2 molecular tracing and its implications</td>\n",
       "      <td>[{'text': 'During the past three months, a new coronavirus (SARS-CoV-2) epidemic has been growing exponentially, affecting over 100 thousand people worldwide, and causing enormous distress to economies and societies of affected countries. A plethora of analyses based on viral sequences has already been published, in scientific journals as well as through non-peer reviewed channels, to investigate SARS-CoV-2 genetic heterogeneity and spatiotemporal dissemination. We examined all full genome sequences currently available to assess the presence of sufficient information for reliable phylogenetic and phylogeographic studies. Our analysis clearly shows severe limitations in the present data, in light of which any finding should be considered, at the very best, preliminary and hypothesis-generating. Hence the need for avoiding stigmatization based on partial information, and for continuing concerted efforts to increase number and quality of the sequences required for robust tracing of the epidemic.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': '. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'is the (which was not peer-reviewed) The copyright holder for this preprint .', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>In December 2019, a novel coronavirus, SARS-CoV-2, was identified in Wuhan, China, as the etiologic agent of coronavirus disease 2019 , which by March 2020 has already spread across more than 80 countries 1 . Common symptoms of infection include fever, cough, and shortness of breath, while severe cases are characterized by advanced respiratory distress and pneumonia, often resulting in death 2 . Soon after the first epidemiological data together with SARS-CoV-2 genetic sequences were made available, a glut of phylogeny-based analyses began to circulate discussing, in scientific papers as well as (social) media, countries that might have been fueling the spread. The implications of misunderstanding the real dynamic of the COVID-19 pandemic are extremely dangerous. Ethnic or social discrimination resulting from unsupported assumptions on viral contagion -often amplified by irresponsible, uncontrollable communications -can be highly damaging for people and countries.\\nDespite (social) media are often vehicle for fake news and boast news hype, it is also worth noting the tremendous effort of the scientific community to provide free, up-to-date information on ongoing studies, as well as critical evaluations. In particular, the US-based NextStrain 3 team has been posting real-time updates on the epidemic tracing by molecular analyses. Several discussions and evidence-based debates on controversial hypotheses on the epidemic have ensued -such as the number of untraced infections in the US, the putative virus introduction in Italy through Germany, and the alleged lineage diversification in China 4 later criticized 5 -aided by a concerted and productive effort of internationally-renowned research groups in Edinburgh, Glasgow, and Oxford, UK, among others. Recently, an editorial published on Science 6 has also highlighted how unsupported or misleading claims circulating in forums, social media, and even peer-reviewed articles, have been led by a substantial over interpretation of the available data.\\nHence, the urgency to reframe the current debate in more rigorous scientific terms, and quantitatively evaluate whether sufficient information for reliable phylogenetic and phylogeographic studies currently exists, or which gaps need to be addressed.\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint Before carrying out any phylogeny-based analysis of virus evolution and spatiotemporal spread, it is crucial to test the quality of sequence data, since uneven sampling, presence of phylogenetic noise, and absence of temporal signal can affect reliability of the results (e.g. ancestral state reconstructions, molecular clock calibrations) 7 . We downloaded all SARS-CoV-2 full genome sequences deposited on GISAID (https://www.gisaid.org/) 8 to compare the number of high quality genomes sampled per country with the number of confirmed cases at the time of sampling, as well as the country's total population ( Figure 1 ). As March 10 th 2020, 331 genome sequences from 29 countries were available, almost twice as much than just one week earlier (March 3 rd 2020, 169 genome sequences from 22 countries). Yet, the sampling per country was inhomogeneous, with confirmed cases clearly uncorrelated to the number of genomes (Spearman correlation = 0.03). Moreover, correlation could only be investigated with confirmed cases (as proxy), since not all affected countries have made publicly available the total number of coronavirus testing performed. Even within the same country, sequenced genomes were usually sampled from few hotspots, not necessarily representative of the whole epidemic in that country.\\nItaly, for example, the second country per confirmed cases, has uploaded so far only two genomes from Lombardy and three from Lazio (Table S1) , i.e. only two of the twenty Italian regions, all with confirmed cases 9 . China and USA contributed with 123 and 63 genomes respectively (56% combined), while 13 countries contribute with only one or two samples (6.3% combined). Notably, three countries with less than 400 confirmed cases have provided a substantial number of genomes (Netherlands 28, Australia 18, Singapore 11; 17% combined) ( Figure 1 ).\\nThe effect of inhomogeneous sampling and missing data on phylogeography reconstructions, like the ones recently rushed through news and (social) media to claim specific dissemination routes of SARS-CoV-2 among countries, can be quite dramatic. An instructive example is the putative introduction of SARS-CoV-2 in Italy from Germany. The maximum likelihood (ML) tree, inferred from the full genome viral sequences available on March 3 rd , 2020 (Supplementary Figure S1 ), showed a well-supported cluster of European and Asian sequences (reported in Figure 2a ), which contained a subclade (subclade A, Figure 2a ) including a sequence isolated in Germany that . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint appears to be paraphyletic (with strong bootstrap support) to an Italian sequence clustering, in turn, with sequences from Finland, Mexico, Germany and Switzerland (Figure 2a ).\\nBased on this observation (which was available on NextStrain), a heated discussion circulated on social media about a transmission event from Germany to Italy followed by further spread from Italy the other countries. However, in a new tree inferred just one week later, when more than 135 new full genome sequences were made available on GISAID 8 ( Figure S2 ), the direct link between Germany and Italy has disappeared due to the additional clustering of previously unsampled sequences from Portugal, Brazil, Wales and Netherland ( Figure 2b ). Most importantly, there is no sufficient evidence of a higher likelihood (Shimodaira-Hasgawa test, Table 1 ) for tree topologies generated arbitrarily switching branches in the tree (see arrows in Figure 2b) compared to the ML tree inferred from the real data. In other words, it is not possible with the present data to decide which branching pattern (and, therefore, phylogeographic reconstruction)\\nis the most likely.\\nLack of resolution and uncertainty in the SARS-CoV-2 phylogenetic tree is to be expected, considering that relatively little genetic diversity can be accumulated during the first three months of an epidemic, even for an exponentially spreading and fast-evolving RNA virus. Plots of pairwise transitions/transversions vs. genetic distance 10 showed, indeed, low genetic diversity among SARS-CoV-2 genomes, considerable scattering, and overlap in transitions and transversions among closely related sequences (Figure 3a ). The 331 full genomes available contained 477 out of 29,564 variable sites (1.6%), of which 377 were singleton and only 100 (0.34%) parsimony informative, indicating a substantial lack of phylogenetic information, and raising the possibility of a number of sequencing errors. Further assessment of phylogenetic signal in the dataset was carried out using likelihood mapping analysis 11 , which estimates the likelihood of each of possible tree topology for any group of four sequences (quartet), randomly . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint chosen from an alignment, and reports them inside an equilateral triangle (the likelihood map)\\nwhere the corners represent distinct tree topologies and the center star-like trees. Quartets are considered \"resolved\" when the three likelihoods are significantly different (phylogenetic signal, most dots equally distributed in the corners, i.e. data are suitable for robust phylogeny inference), unresolved or partially resolved, when all three likelihood values or two of them are not significantly different (phylogenetic noise, most dots in sides or center areas, i.e. data may not be sufficient for robust phylogeny inference). Extensive simulation studies have shown that, for sequences to be considered robust in terms of phylogenetic signal, side/center areas of the likelihood mapping must include &lt;40% of the unresolved quartets 12 , way lower than the 52.9% unresolved quartets detected in the SARS-CoV-2 full genomes alignment ( Figure 3b ). It is important to keep in mind that such a lack of phylogenetic signal makes the overall topology of any SARS-CoV-2 tree unreliable, and even clades with high bootstrap values should be interpreted with extreme caution. Finally, the mean genetic distance of each sequence from the root of a ML tree indicated that the available genomic data lack temporal signal (no significant linear correlation between accumulated genetic distance and sampling time, Figure 3c ). This is, again, expected in genomes obtained over a very short period of time (~ three months) since the beginning of the outbreak. Bayesian analysis 13 , which infers phylogenetic and phylogeographic patterns from a posterior distribution of trees, might facilitate comparisons about different evolutionary scenarios, help in retrieving the correct topology, and estimate an accurate evolutionary rate. With more genome sequences sampled from different time points and geographic areas made available, literally, every day, Bayesian analysis may soon be a viable option. However, as long the new data do not increase phylogenetic and temporal signal, Bayesian phylogenetic and phylogeography results would remain equally questionable.\\nTaken together, our observations suggest that any conclusion drawn, at present, about existing lineages and direction of viral spread, based on phylogenetic analysis of SARS-CoV-2 sequence data, is potentially misleading. Nowadays, published scientific data and media are very easily . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint .\\naccessible to a worldwide audience, thus it is important more than ever to weigh the information being shared. We believe that current molecular epidemiology data are not solid enough to provide a scientifically sound analysis of SARS-CoV-2 spread. Any inference based on phylogenetic trees should be considered at the very best preliminary, and hypothesis-generating.\\nHence the need for 1) avoiding stigmatization based on partial information and 2) continuing the concerted scientific effort to increase the number and quality of available data. The evolutionary dynamics of SARS-CoV-2 spread will unveil an unprecedented amount of information, which is essential to make policy decisions. The whole of humanity is threatened by the current pandemic, and policy makers need to adjust their mitigation measures while the pandemic itself is developing. Some of the urgent answers required lie in the timely availability of abundant, high quality sequence data not only from countries experiencing a high number of reported cases, such as Italy and South Korea, but also from others that seem to be experiencing, at least for now, a lower number of infections.\\nGISAID was accessed on March 3 rd and 10 th , 2020 (final dataset information reported in Table   S1 ). After quality control of sequences that were not full genomes or contained extensive stretches of unknown nucleotides, the first dataset (March 3 rd 2020) consisted of 169 genomes from 22 countries, while the second (March 10 th , 2020) 331 from 29 countries. Confirmed cases were retrieved for the same day from interactive web-based dashboard developed by Johns Hopkins University 14 .\\nTransition/transversions vs. genetic distance plot were calculated using DAMBE6 10 . Evaluation of the presence of phylogenetic signal satisfying resolved phylogenetic relationships among sequences was carried out with IQ-TREE, allowing the software to search for all possible quartets . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint using the best-fitting nucleotide substitution model 11 . ML tree reconstruction was performed in IQ-TREE based on the best-fit model chosen according to Bayesian Information Criterion (BIC) 15, 16 . Topology was tested performing the Shimodaira-Hasegawa 17 on \"Subclade A\" by comparing the inferred tree to three trees where two branches: (Tree 1) Italy (EPI_ISL_412973) switched with Wales (EPI_ISL_413555), (Tree 2) Germany (EPI_ISL_406862 with) Brazil (EPI_ISL_412964), (Tree 3) and Portugal (EPI_ISL_413648) with Brazil (EPI_ISL_412964), (Tree 4) Germany (EPI_ISL_406862) with Portugal (EPI_ISL_413648). All tests performed 10,000 resamplings using the using the RELL approximation method 18 . Position of these genomes within \"Subclade A\" is indicated in Figure 1 with an arrow.\\nExploration of temporal structure, i.e. presence of molecular clock in the data, was assessed by regression of divergence -root-to-tip genetic distance-against sampling time using TempEst 19 . In this case, absence of a linear trend indicates that the data does not contain temporal signal and that the data is not appropriate for phylogenetic inference using molecular clock models.\\nAncestral state reconstruction of full datasets reported in Figures S1 and S2 were based on the ML tree obtained with IQ-TREE and performed using TreeTime 20 . We choose to show the \"European subclade\" and \"Subclade A\" as cladograms in Figure 1 as there are no phylogenetic and temporal signals in the datasets: scaling phylogenies in time or nucleotide substitution in this case is misleading. . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint Each bi-directional arrow, and corresponding number, connects two tip branches that were switched to generate an alternative tree topology to be tested (see Table 1 , Methods). CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity. EPI_ISL_412971_BetaCoV_Finland_FIN-25_2020 EPI_ISL_406533_BetaCoV_Guangzhou_20SF206_2020 EPI_ISL_407896_BetaCoV_Australia_QLD02_2020 EPI_ISL_408478_BetaCoV_Chongqing_YC01_2020\\nEPI_ISL_412979__BetaCoV_Wuhan_HBCDC-HB-03_2020\\nEPI_ISL_411220_BetaCoV_France_IDF0386-islP3_2020 EPI_ISL_410044_BetaCoV_USA_CA6_2020 EPI_ISL_413020_BetaCoV_Switzerland_1000477377_2020\\nEPI_ISL_406030_BetaCoV_Shenzhen_HKU-SZ-002_2020\\nEPI_ISL_408669_BetaCoV_Japan_KY-V-029_2020\\nEPI_ISL_408515_BetaCoV_Wuhan_IVDC-HB-envF13-21_2020 EPI_ISL_410713_BetaCoV_Singapore_7_2020 EPI_ISL_412873_BetaCoV_Korea_KCDC24_2020\\nEPI_ISL_402120_BetaCoV_Wuhan_IVDC-HB-04_2020\\nEPI_ISL_406970_BetaCoV_Hangzhou_HZ-1_2020 EPI_ISL_406973_BetaCoV_Singapore_1_2020 EPI_ISL_412029_BetaCoV_Hong_Kong_VM20001988_2020\\nEPI_ISL_410720_BetaCoV_France_IDF0372-isl_2020 EPI_ISL_411951_BetaCoV_Sweden_01_2020 EPI_ISL_408488_BetaCoV_Jiangsu_IVDC-JS-001_2020\\nEPI_ISL_402123_BetaCoV_Wuhan_IPBCAMS-WH-01_2019 EPI_ISL_407894_BetaCoV_Australia_QLD01_2020 EPI_ISL_411955_BetaCoV_USA_CA8_2020 EPI_ISL_403962_BetaCoV_Nonthaburi_61_2020 EPI_ISL_406801_BetaCov_Wuhan_WH04_2020 EPI_ISL_408978_BetaCoV_Wuhan_WH05_2020\\nEPI_ISL_407084_BetaCoV_Japan_AI_I-004_2020 EPI_ISL_407215_BetaCoV_USA_WA1-F6_2020 EPI_ISL_411060_BetaCoV_Fujian_8_2020 EPI_ISL_410984_BetaCoV_France_IDF0515-isl_2020\\nEPI_ISL_412980__BetaCoV_Wuhan_HBCDC-HB-04_2020 EPI_ISL_410546_BetaCoV_Italy_INMI1-cs_2020 EPI_ISL_408977_BetaCoV_Sydney_3_2020 EPI_ISL_412982__BetaCoV_Wuhan_HBCDC-HB-06_2020 EPI_ISL_407071_BetaCoV_England_01_2020 EPI_ISL_412912_BetaCoV_Germany_Baden-Wuerttemberg-1_2020 EPI_ISL_411927_BetaCoV_Taiwan_4_2020 EPI_ISL_410536_BetaCoV_Singapore_5_2020 EPI_ISL_406538_BetaCoV_Guangdong_20SF201_2020 EPI_ISL_407988_BetaCoV_Singapore_3_2020 EPI_ISL_406036_BetaCoV_USA_CA2_2020 EPI_ISL_411954_BetaCoV_USA_CA7_2020 EPI_ISL_412028_BetaCoV_Hong_Kong_VM20001061_2020 EPI_ISL_406862_BetaCoV_Germany_BavPat1_2020 EPI_ISL_406597_BetaCoV_France_IDF0373_2020 EPI_ISL_408009_BetaCoV_USA_CA4_2020 EPI_ISL_410716_BetaCoV_Singapore_10_2020 EPI_ISL_403933_BetaCoV_Guangdong_20SF013_2020 EPI_ISL_402129_BetaCoV_Wuhan_WIV06_2019\\nEPI_ISL_406594_BetaCoV_Shenzhen_SZTH-003_2020 EPI_ISL_407987_BetaCoV_Singapore_2_2020 EPI_ISL_413015_BetaCoV_Canada_ON_VIDO-01_2020 EPI_ISL_407976_BetaCoV_Belgium_GHB-03021_2020 EPI_ISL_406531_BetaCoV_Guangdong_20SF174_2020 EPI_ISL_410715_BetaCoV_Singapore_9_2020\\nEPI_ISL_404227_BetaCoV_Zhejiang_WZ-01_2020 EPI_ISL_408479_BetaCoV_Chongqing_ZX01_2020 EPI_ISL_406535_BetaCoV_Foshan_20SF210_2020 EPI_ISL_408976_BetaCoV_Sydney_2_2020 EPI_ISL_411066_BetaCoV_Fujian_13_2020 EPI_ISL_407073_BetaCoV_England_02_2020\\nEPI_ISL_403929_BetaCoV_Wuhan_IPBCAMS-WH-04_2019 EPI_ISL_410301_BetaCoV_Nepal_61_2020 EPI_ISL_403931_BetaCoV_Wuhan_IPBCAMS-WH-02_2019\\nEPI_ISL_410531_BetaCoV_Japan_NA-20-05-1_2020 EPI_ISL_403963_BetaCoV_Nonthaburi_74_2020 EPI_ISL_402125_BetaCoV_Wuhan-Hu-1_2019\\nEPI_ISL_408514_BetaCoV_Wuhan_IVDC-HB-envF13-20_2020\\nEPI_ISL_402119_BetaCoV_Wuhan_IVDC-HB-01_2019 EPI_ISL_411956_BetaCoV_USA_TX1_2020 EPI_ISL_406596_BetaCoV_France_IDF0372_2020 EPI_ISL_412967_BetaCoV_China_IQTC02_2020 EPI_ISL_408431_BetaCov_France_IDF0626_2020\\nEPI_ISL_410532_BetaCoV_Japan_OS-20-07-1_2020\\nEPI_ISL_411957_BetaCoV_China_WH-09_2020 EPI_ISL_412970_BetaCoV_USA_WA2_2020 EPI_ISL_412978__BetaCoV_Wuhan_HBCDC-HB-02_2020 EPI_ISL_412869_BetaCoV_Korea_KCDC05_2020 EPI_ISL_412965_BetaCoV_Canada_BC_37_0-2_2020 EPI_ISL_412871_BetaCoV_Korea_KCDC07_2020 EPI_ISL_408008_BetaCoV_USA_CA3_2020\\nEPI_ISL_406593_BetaCoV_Shenzhen_SZTH-002_2020 EPI_ISL_406536_BetaCoV_Foshan_20SF211_2020 EPI_ISL_412459_BetaCoV_Jingzhou_HBCDC-HB-01_2020 EPI_ISL_410718_BetaCoV_Australia_QLD04_2020 EPI_ISL_412862_BetaCoV_USA_CA9_2020 EPI_ISL_413019_BetaCoV_Switzerland_1000477102_2020\\nEPI_ISL_412981__BetaCoV_Wuhan_HBCDC-HB-05_2020\\nEPI_ISL_408668_BetaCoV_Vietnam_VR03-38142_2020 EPI_ISL_410535_BetaCoV_Singapore_4_2020 EPI_ISL_408665_BetaCoV_Japan_TY-WK-012_2020 EPI_ISL_406716_BetaCoV_China_WHU01_2020 EPI_ISL_402127_BetaCoV_Wuhan_WIV02_2019 EPI_ISL_411926_BetaCoV_Taiwan_3_2020 EPI_ISL_406223_BetaCoV_USA_AZ1_2020 EPI_ISL_407214_BetaCoV_USA_WA1-A12_2020 EPI_ISL_406717_BetaCoV_China_WHU02_2020 EPI_ISL_404228_BetaCoV_Zhejiang_WZ-02_2020\\nEPI_ISL_412898__BetaCoV_Wuhan_HBCDC-HB-02_2019 EPI_ISL_413017_BetaCoV_South_Korea_KUMC01_2020 EPI_ISL_408480_BetaCoV_Yunnan_IVDC-YN-003_2020 EPI_ISL_403936_BetaCoV_Guangdong_20SF028_2020 EPI_ISL_408667_BetaCoV_Japan_TY-WK-521_2020 EPI_ISL_411218_BetaCoV_France_IDF0571_2020 EPI_ISL_405839_BetaCoV_Shenzhen_HKU-SZ-005_2020 EPI_ISL_408010_BetaCoV_USA_CA5_2020 EPI_ISL_406031_BetaCoV_Taiwan_2_2020 EPI_ISL_406844_BetaCoV_Australia_VIC01_2020 EPI_ISL_411950_BetaCoV_Jiangsu_JS01_2020 EPI_ISL_412974_BetaCoV_Italy_SPL1_2020\\nEPI_ISL_403928_BetaCoV_Wuhan_IPBCAMS-WH-05_2020 EPI_ISL_412870_BetaCoV_Korea_KCDC06_2020 EPI_ISL_404253_BetaCoV_USA_IL1_2020 EPI_ISL_410045_BetaCoV_USA_IL2_2020 EPI_ISL_403932_BetaCoV_Guangdong_20SF012_2020\\nEPI_ISL_408484_BetaCoV_Sichuan_IVDC-SC-001_2020 EPI_ISL_410537_BetaCoV_Singapore_6_2020 EPI_ISL_412872_BetaCoV_Korea_KCDC12_2020 EPI_ISL_407079_BetaCoV_Finland_1_2020\\nEPI_ISL_410545_BetaCoV_Italy_INMI1-isl_2020 EPI_ISL_408670_BetaCoV_USA_WI1_2020 EPI_ISL_403937_BetaCoV_Guangdong_20SF040_2020 EPI_ISL_403934_BetaCoV_Guangdong_20SF014_2020\\nEPI_ISL_413014_BetaCoV_Canada_ON-PHL2445_2020\\nEPI_ISL_411915_BetaCoV_Taiwan_CGMH-CGU-01_2020 EPI_ISL_410717_BetaCoV_Australia_QLD03_2020 EPI_ISL_412026_BetaCoV_Hefei_2_2020 EPI_ISL_407313_BetaCoV_Hangzhou_HZCDC0001_2020 EPI_ISL_412972_BetaCoV_Mexico_CDMX_InDRE_01_2020 EPI_ISL_406034_BetaCoV_USA_CA1_2020 EPI_ISL_411952_BetaCoV_Jiangsu_JS02_2020 EPI_ISL_411902_BetaCoV_Cambodia_0012_2020\\nEPI_ISL_413016_BetaCov_Brazil_SPBR-02_2020 EPI_ISL_413018_BetaCoV_South_Korea_KUMC02_2020 EPI_ISL_403930_BetaCoV_Wuhan_IPBCAMS-WH-03_2019\\nEPI_ISL_412899__BetaCoV_Wuhan_HBCDC-HB-03_2019 EPI_ISL_408489_BetaCoV_Taiwan_NTU01_2020 EPI_ISL_409067_BetaCoV_USA_MA1_2020 EPI_ISL_404895_BetaCoV_USA_WA1_2020\\nEPI_ISL_408666_BetaCoV_Japan_TY-WK-501_2020 EPI_ISL_410218_BetaCov_Taiwan_NTU02_2020 EPI_ISL_407893_BetaCoV_Australia_NSW01_2020 EPI_ISL_407193_BetaCoV_South_Korea_KCDC03_2020\\nEPI_ISL_412983_BetaCoV_Tianmen_HBCDC-HB-07_2020 EPI_ISL_410714_BetaCoV_Singapore_8_2020 EPI_ISL_403935_BetaCoV_Guangdong_20SF025_2020 EPI_ISL_406798_BetaCov_Wuhan_WH01_2019 EPI_ISL_411953_BetaCoV_Jiangsu_JS03_2020 EPI_ISL_412030_BetaCoV_Hong_Kong_VB20026565_2020 EPI_ISL_412968_BetaCoV_Japan_Hu_DP_Kng_19-020_2020 EPI_ISL_412973_BetaCoV_Italy_CDG1_2020 EPI_ISL_402132_BetaCoV_Wuhan_HBCDC-HB-01_2019\\nEPI_ISL_408486_BetaCoV_Jiangxi_IVDC-JX-002_2020 EPI_ISL_408430_BetaCoV_France_IDF0515_2020 EPI_ISL_410486_BetaCoV_France_RA739_2020 EPI_ISL_412966_BetaCoV_China_IQTC01_2020\\nEPI_ISL_412969_BetaCoV_Japan_Hu_DP_Kng_19-027_2020\\nEPI_ISL_411219_BetaCoV_France_IDF0386-islP1_2020\\nEPI_ISL_408481_BetaCoV_Chongqing_IVDC-CQ-001_2020 EPI_ISL_412975_BetaCoV_Australia_NSW05_2020 EPI_ISL_406534_BetaCoV_Foshan_20SF207_2020 . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint \\n</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Healthcare-resource-adjusted vulnerabilities towards the 2019-nCoV epidemic across China</td>\n",
       "      <td>[]</td>\n",
       "      <td>The 2019-nCoV epidemic has spread across China and 24 other countries 1-3 as of February 8, 2020 . The mass quarantine measure implemented in the epidemic's epicenter, Wuhan, Hubei Province, China, on January 23, 2020, has stopped the human migration between Wuhan and other places. However, according to the local authorities, over five million people had already traveled from Wuhan to other parts of China in the weeks leading to January 23, 2020, with the majority (79.96%) to the other cities within Hubei.\\nHealthcare resources have become a major bottleneck in the response to this public health crisis. The sudden onset of the epidemic and the sheer size of the infection have caused significant shortage in healthcare resources across Hubei Province. So far, there have been over 11,000 non-local healthcare professionals, mobilized by the public health authorities from other parts of China, fighting against the epidemic in Wuhan. However, this is far from adequate for Wuhan given the severity of the epidemic. In addition to Wuhan, other cities in Hubei are in need of urgent external help as well with respect to health resources, as multiple localized outbreaks are taking place. On February 7, 2020, the National Health Commission of China (NHCC) initiated the \"One City Covered by One Province\" program, explicitly assigning a specific province to aid one city in Hubei. Two critical questions of operational nature remain, which cities, both in Hubei and other provinces, are most vulnerable to localized outbreaks, and how the public health authorities should balance the healthcare resources under stress to best fight the epidemic and minimize the risks? This study is aimed to shed light on these questions by developing a vulnerability analysis framework integrating the human movement and healthcare resource data.\\nThis study makes use of four public datasets: (a) the daily counts of confirmed 2019-nCoV cases, collected from NHCC 4 ; (b) the list of designated hospitals with the number of reserved beds, retrieved from the websites of the provincial and municipal health commissions; (c) the information of the hospitals designated for 2019-nCoV, retrieved from 99.com.cn, a public hospital database; and (d) the human movement data, provided by the Baidu Migration (http://qianxi.baidu.com/), a public site offering real-time migration information across mainland China.\\nRecent studies indicated that the confirmed case data for Hubei Province, in particular, Wuhan, was subject to the under-reporting bias, while the data from places outside of Hubei is more reliable for inferring the actual epidemic size 5 . Following Wu et al. 5 , we developed a susceptible-exposed-infectious-recovered (SEIR)-based metapopulation model, which has 100 sub-populations representing the 100 cities that have the greatest number of people traveling from Wuhan. Note that we adopted the same epidemiological parameters as Wu et al. 5 . This model was then used to estimate IDi, the number of infected individuals in each city. Lastly, healthcare-resource-adjusted vulnerability indexes to the 2019-nCoV outbreak for city i, Vi were obtained as follows: Vi = IDi / Bi, where Bi represents the total number of reserved beds in the designated hospitals.\\nThe Figure presents the estimated number of infected individuals, as well as the healthcareresource-adjusted vulnerability to the 2019-nCoV outbreak across mainland China, as of January 29, 2020. The epidemiological parameters may have changed after January 23, 2020, when the lockdown measure was implemented. Thus, we only use the data by January 23, 2020 and produce the estimation till January 29, 2020 to accommodate the six-day incubation period identified by Wu et al. 5 .\\nThe ten most vulnerable non-Hubei cities are also listed in the figure. Note that the cities in Hubei (which are being aided by other provinces) are not presented here because of their significantly higher vulnerability as compared to others. Among the ten most vulnerable non-Hubei cities, Changsha, Beijing, Chongqing and Shanghai are major metropolises with a large number of estimated infected individuals. Xinyang, Jiujiang, Nanchang and Yiyang are medium-sized cities neighboring Hubei. In particular, Xinyang appears to be in a particular vulnerable situation because of the large number of people traveling from Wuhan and its insufficient healthcare resources.\\nThe reported vulnerability analysis informs public health response to the 2019-nCoV epidemic in multiple ways. For the identified cities with high vulnerability, it is important to increase the reserve of healthcare resources or have contingent plans. In particular, some of these cities already contributed significant healthcare resources to Hubei, leaving themselves inadequately prepared for the localized outbreak. Our analysis revealed that several medium-sized cities, particularly those neighboring Hubei, are highly vulnerable due to the lack of healthcare resources. \\n</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Relationship between Average Daily Temperature and Average Cumulative Daily Rate of Confirmed Cases of COVID-19</td>\n",
       "      <td>[{'text': 'The rapid outbreak of the new Coronavirus pandemic and the spread of the virus worldwide, especially in the Northern Hemisphere, have prompted various investigations about the impact of environmental factors on the rate of development of this epidemic. Different studies have called the attention to various parameters that may have influenced the spread of the virus, and in particular, the impact of climatic parameters has been emphasized. The main purpose of this study is to investigate the correlation between the average daily temperature and the rate of coronavirus epidemic growth in the infected regions. The main hypothesis object of our research is that between regions exhibiting a significant difference in the mean daily temperature, a significant difference is also observed in the average cumulative daily rate of confirmed cases, and that this does not happen if there is no significant difference in mean daily temperature. To test this research hypothesis, we carried on the case study of three regions in each of five countries and analyzed the correlation through F-test, and Independent-Samples T-Test. In all five selected countries, we found that when there is a significant difference in the daily mean temperature between two regions of a country, a significant difference exists also in the average cumulative daily rate of confirmed cases. Conversely, if there are no significant differences in the mean daily temperature of two regions in the same country, no significant difference is observed in the average cumulative daily rate of confirmed cases for these regions.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>The outbreak of infectious diseases has always been one of the most important health problems in the world. With the rapid spread of the new coronavirus (COVID- 19) , much attention has been paid to subjects like rate of an epidemic, transition ways, prevention methods, and remaining time of the virus in the environment [1] [2] [3] [4] [5] . Therefore, the behavior of the virus in relation to environmental factors is critical. The coronavirus is one of the epidemic diseases that many countries have experienced in the new type of it, COVID-19, and can become a serious challenge and affect the previous efforts on sustainable development [6] [7] [8] [9] .\\nIt has been found that there are some delays between the detection of confirmed cases and the actual infection date due to the laboratory test, the time of the announcement of confirmed cases in media, and the time for the patients showing initial symptoms for a total usually of 3 to 5 days [10] [11] [12] . The analysis of Kampf et al. (2020) demonstrated that COVID-19 could remain up to 9 days on different materials based on temperature and humidity [13] . Cascella et al. (2020) conducted research on the new coronavirus structure. Their analysis shows that the virus belongs to the family of single-stranded RNA viruses (+ ssRNA), which has a length of about 30 kb and an envelope with spear structures and they tested the sensitivity of the virus to UV and temperature [14] . investigated the correlations between environmental factors and the number of positive cases of COVID-19 by the use of artificial intelligence techniques. The results of this study put evidence on the role of urban and weather conditions on the epidemic rate [15] . Chen et al. (2020) developed a time-dependent mathematical model for the prediction of the total number of positive cases [16] . Lytle and Sagripanti (2005) studied the impact of UV rays, and their results determined that it can affect the structure of coronaviruses because UV rays could damage the nucleic acids of viral RNA that is vital for the survival of the virus [17] . developed an assessment method for investigating the impact of climate and urban factors in confirmed cases of COVID-19 [18] . Bukhari et al., (2020) , studied the epidemic of COVID-19 on tropical zones and their results show that the highest number of infection happened in temperature among 3 to 17 ºC [19] .\\nThe analysis of Grant and Giovannucci (2009) shows that by increasing the UV, the level of vitamin D in the body increases and makes the infection rates decrease. Therefore, in areas on lower latitudes and closer to the equator line, the epidemic on COVID-19 might be different [20] . In the study of Chan et al. (2011) , the effect of temperature and humidity on SARS coronavirus was analyzed, and they found that increasing temperature decreases the epidemic rate by faster disabling virus structure on contaminated surfaces [21] . Pirouz and Violini (2020) , analyzed the outbreak patterns of COVID-19 by analysis of 117 countries. They suggested two mechanisms of triggering the development of the infection, namely the normal spread of an infection from one parent case and the consequence of a unique phenomenon of accumulation of contagious people [22] . Analysis of the previous studies demonstrates the necessity of further research about the effect of environmental factors on the COVID-19. Since some of the previous researches have shown the effect of weather conditions, especially temperature, on coronavirus viruses, the main purpose of this study is to investigate the better correlation of daily mean air temperature on the epidemic rate of the COVID-19.\\nIn this study, we analysed the number of positive contagions of coronavirus COVID-19, from February 29 th to March 19 th . The selection of February 29 th is due to existence on this date of at least one positive case of COVID-19 in all case studies studied. The analysis of the weather data started from February15 th and ended on March 18 th because the impact of the weather parameter would occur in the upcoming days. Thus, the considered average daily temperature is with an interval of 1 to 14 days before the definitive cases of COVID-19.\\n In order to minimize the impact of other parameters on the results, such as the policy of each government, cultural differences between countries, health care systems, or the protocols in getting coronavirus test, comparative analysis have been done for different regions of the same country, without comparing with each other the results of different countries;  The analysis period for average daily temperature ranges from February15 th , 2020 to March18 th , 2020 (33 days);  The analysis period for confirmed cases of coronavirus COVID-19 is from February 29 th to March19 th ;  For what concerns the cumulative daily rate of confirmed cases, the total number of confirmed cases in a region was divided into the population of that region in order to take into account the different size of the population in each region.\\nTo carry out the analysis, the datasets of five countries and in each country, of three regions/provinces have been considered. The selected case studies are presented in Table 1 . and the location of the three regions shown in Figures 1. \\nMathematical models have been successfully used for the analysis of many hazards [32, 33] . In complex subjects, different Mathematical models can be used to solve the problem [34] [35] [36] . Statistical tests are an appropriate method for the investigation of possible correlations between different datasets [37] . Many statistical tests can be used for the analysis of significant differences between the datasets, among which F-test and Independent-Samples T-Test [38] .\\nIn order to investigate the hypothesis about a correlation between significant differences (Sig.) of average daily temperatures and average cumulative daily rate of confirmed cases in two regions, the following assumptions, and statistical tests have been done:\\n The variances of the two groups are equal: H0  The variances of the two groups are not equal: H1\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10.20059337 doi: medRxiv preprint When using F-test, if the significance level (sig) is less than 0.05, hypothesis H0 is rejected, and then hypothesis H1 is accepted, i.e., the variances of the datasets in the two selected regions are not equal. Conversely, if the significance level (sig.) is larger than 0.05, the hypothesis H1 is rejected, and the hypothesis H0 is accepted, i.e., the variances of the datasets in the two selected regions are equal.\\nThrough F-test, one determines the presence or absence of differences in the variances of the two regions. Through another statistical test, the Independent-Samples T-Test, one can determine whether the difference is significant, has been used. When using Independent-Samples T-Test, a significance level (sig.) lower (higher) than 0.05, means that the difference is (is not) significant.\\nIn our analysis, we used SPSS software. The critical factors in the SPSS Model set up are as follow:\\n The datasets in different regions of the same countries were analyzed with respect to the average daily air temperature during the 33-day period from 15-Feb to 18-Mar and the cumulative daily rate of confirmed cases of COVID-19 during the 19 days from 29-Feb to 19-Mar.  In all statistical tests, the confidence level of 95% is considered.  First, the F-test was used to test whether there is a significant equality of variances for each pair of case studies, and then, depending on the F-test result, by the T-Test the existence of a significant difference between the data of the two cases was investigated.\\nThe average daily temperature [39, 40] , and cumulative daily rate of confirmed cases [41] , are shown in Figures 2 to 6.\\na. Average daily temperature b. Cumulative daily rate of confirmed cases Correlations between the average temperatures and the cumulative daily rate of confirmed cases were found especially for towns with significant differences, with respect to the other two towns of the same country, in the average daily temperature. This was the case for Bandar Abas in Iran, Milan in Italy, Santa Cruz in Spain, and Los Angeles in the US. The average daily temperature and cumulative daily rate of confirmed cases are listed in Table 2 . As a general feature of Table 2 , we notice that  In 11 out of the possible fifteen comparisons, the average cumulative rate of confirmed cases is higher when the average daily temperature is lower, and in the four remaining cases the differences in either variable are small.\\nA more specific analysis per country shows that  In Iran, the average daily temperature in Bandar Abbas was the highest (21.54 ºC), and the average cumulative rate of daily of confirmed cases was the lowest (0.000027). . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10.20059337 doi: medRxiv preprint  In Italy, the average daily temperature in Milan was the lowest (7.99 ºC), and the average cumulative rate of daily of confirmed cases was the highest (0.000690).  In Germany, the average daily air temperatures of the selected towns have similar values, as well as the spread of the COVID-19 infection is similar.  In Spain, the average daily air temperature in Barcelona and Valencia are similar, and the spread of the infection is also similar. However, the comparison of Santa Cruz with higher average daily temperatures shows a slower spread of COVID-19.  In the United States, the average daily air temperature in both New York and Seattle are lower than Los Angeles, and the average cumulative rate of confirmed cases shows the spread of the virus in both is faster than in Los Angeles.\\nThese qualitative results reflect in the statistical analysis of the data by F-test and Independent-Samples T-Test presented in Table 3 . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10.20059337 doi: medRxiv preprint In Table 3 . t values or p-values can be used to check the existence of a significant correlation between the data for two cities.\\nLet us discuss in detail one such comparison, that between Ramsar and Tehran. From the F-test, the F value is equal to 20.4. The criterion for variance equality in the F-test is that the F value must smaller than the critical F value. The statistical analysis has been done on a statistical confidence level of 95%, and therefore the error level is 0.05, and considering the degrees of freedom, the critical F value of the test that can be found through F distribution tables [42] , in this case, is equal to F = 1.57. Since the F value we found is 20.4, larger than the critical F value, the assumption of the equality of variances is rejected. It means that the average temperature variance of the Ramsar and Tehran are not equal.\\nAccording to the t-test, the t value for these two cities is equal to -1.929. Based on an error level of 0.05 and by considering the degrees of freedom, the critical values of the test indicated in the T distribution tables [43] , is in this case equal to t = ± 1.645. Since the t value is out of the critical boundary (-1.645 ≤ acceptable t ≤ +1.645), the assumption of equality in the averages is rejected, and the existence of difference in averages is confirmed. Thus from two tests, it is evident that there is a significant difference between the average temperature of the two cities of Ramsar and Tehran.\\nAn analogous result holds for the average cumulative rate of confirmed cases.\\nThe main results of Table 4 , with a confidence level of 95% are as follow:\\n In Iran, the city of Bandar Abbas is significantly different from Ramsar and Tehran in terms of both average daily temperature and average cumulative rate of confirmed cases.  In Italy, the city of Milan is significantly different from Rome and Naples in terms of both average daily temperature and average cumulative rate of confirmed cases. Instead, the two cities of Rome and Napoli are not significantly different in terms of these variables.  In Germany, there are no significant differences in average daily temperature in all three cities, and the average cumulative rate of confirmed cases is not significantly different.  In Spain, Barcelona and Valencia are not significantly different in terms of average daily temperature, and the average cumulative rate of confirmed cases is not significantly different. However, in Barcelona and Santa Cruz, the average daily temperature is significantly different, as well as the average cumulative rate of confirmed cases, and the same trend exists between Valencia and Santa Cruz.  In the United States, the two cities of New York and Seattle are not significantly different in terms of average daily temperature, and of the average cumulative rate of confirmed cases, the number of patients is not significantly different. Instead, the city of Los Angeles is significantly different in terms of average daily temperature with two cities of New York and Seattle, and the average cumulative rate of confirmed cases are different significantly.\\nThe results of our analysis of daily air temperature for three case studies from five countries showed the existence of correlations between the average temperatures and the epidemic rate, and this is especially evident when the differences in average daily temperature are larger. For example, in Iran and Italy, the regions with lower average daily temperature had a higher epidemic level of the new coronavirus (COVID- 19) , and the same trend exists in all other case studies.\\nThe statistical analysis showed that in all the case studies we considered, referring to five different countries, if there is a significant difference in daily mean temperature, there is also a significant difference in the average cumulative daily rate of confirmed cases, supporting the proposed research hypothesis about the correlation between air temperature and the epidemic rates of COVID-19. Due to the correlations among temperature and hours of sunshine for future research, it is recommended to study the effects of the sun's UV rays as well as the average hours of sunshine on the epidemic rates of COVID-19.\\nThis study analyzed the possible correlation of daily average air temperature with the epidemic rate of the coronavirus (COVID-19) using significant differences analysis between the datasets. The research hypothesis is that in regions with a significant difference in the average daily temperature, the average cumulative daily rate of confirmed cases would also is different significantly. Conversely, to the absence of a significant difference in daily temperature, would correspond no significant difference in the rate of confirmed cases. The hypothesis was investigated through statistical analysis. In order to minimize the impact on the results of factors like the policy of the government or cultural differences among countries (food, exercise, weight, etc.), three case studies within five countries, namely Iran, Italy, Germany, Spain, and United States were compared separately. This comparison using statistical analysis through F-test and Independent-Samples T-Test of the daily temperatures and daily rate of confirmed cases shows that there is a correlation between the average temperature and the epidemic rate, and this is especially evident when differences in average daily temperature are significantly larger, as it happens for Bandar Abbas in Iran, Milan in Italy, Santa Cruz in Spain, and Los Angeles in the US. Besides, the analysis of the average air temperatures shows that the epidemic rates of COVID-19 were higher in the case studies with a lower average temperature. Instead, when no significant differences exist in the average daily temperature of two cities in the same country, there is no significant difference in the average cumulative daily rate of confirmed cases. In conclusion, the results of this study support the research hypothesis and confirm the effectiveness of the proposed method for analysis of the epidemic rates.\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10 . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10.20059337 doi: medRxiv preprint \\n</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>CHEER: hierarCHical taxonomic classification for viral mEtagEnomic data via deep leaRning</td>\n",
       "      <td>[{'text': 'The fast accumulation of viral metagenomic data has contributed significantly to new RNA virus discovery. However, the short read size, complex composition, and large data size can all make taxonomic analysis difficult. In particular, commonly used alignment-based methods are not ideal choices for detecting new viral species. In this work, we present a novel hierarchical classification model named CHEER, which can conduct read-level taxonomic classification from order to genus for new species. By combining k-mer embedding-based encoding, hierarchically organized CNNs, and carefully trained rejection layer, CHEER is able to assign correct taxonomic labels for reads from new species. We tested CHEER on both simulated and real sequencing data. The results show that CHEER can achieve higher accuracy than popular alignmentbased and alignment-free taxonomic assignment tools. The source code, scripts, and pre-trained parameters for CHEER are available via GitHub: https://github.com/KennthShang/CHEER.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]</td>\n",
       "      <td>Metagenomic sequencing, which allows us to directly obtain total genomic DNAs from host-associated and environmental samples, has led to important findings in many areas, such as digestive health [1] . While bacteria are the main focus of most metagenomic sequencing projects, there are fast accumulation of viral metagenomic data with sequencing viruses as the main purpose [2] . There are different types of viruses. We are mainly concerned with RNA viruses because many RNA viruses are notorious human pathogens, such as Influenza A, Human immunodeficiency virus (HIV), Ebola, SARS-CoV, and recently identified 2019-nCoV causing Wuhan pneumonia. Unlike DNA viruses, RNA viruses, which contain RNA genomes, lack faithful proofreading mechanisms during replication and thus can produce a group of related but different viral strains infecting the same host. This high genetic diversity within and across different hosts poses a great challenge for designing long-term protection strategies against these infectious diseases. For example, as the circulating strains can change every year, flu vaccine has to be administered every year.\\nAdvances in viral metagenomics have contributed significantly to new RNA virus discovery. According to a survey by Woolhouse et. al, the number of newly identified RNA viruses is changing from 1,899 in 2005 to 5,561 in 2018, which is 3 times of increase [3] . Largescale RNA virus sequencing projects using nextgeneration sequencing technologies have been conducted for different species. For example, Shi et al. have discovered a large number of new RNA viruses by sequencing samples from invertebrate and vertebrate animals [4, 5] . Claire et al. have discovered 25 new RNA viruses by sequencing samples associated with different drosophilid [6] . Bolduc et al. sequenced RNA viruses from Archaeal and bacterial samples [7] . Given fast accumulation of viral metagenomic data and expected discovery of new viruses, a key step is to conduct composition analysis for these data and assign taxonomic groups for possibly new species.\\nComposition analysis can be conducted at the read level or contig level. There is no doubt that contigs can provide more information than short reads in taxonomic classification. However, metagenomic assembly is still one of the most challenging computational problems in bioinformatics. Unlike single genome assembly, metagenomic assembly is more likely to produce chimeric contigs that can contain regions from different species. Thus, metagenomic assembly is often preceded by read binning that groups reads with the same or similar taxonomic labels [8] in order to achieve better performance.\\nIn this work, we design a read level taxonomic classification tool (named as \"CHEER\") for labeling new species in viral metagenomic data. To detect known species with available reference genomes, alignment-based methods can provide sufficient information for both classification and abundance estimation. This function has been incorporated in existing pipelines [9, 10, 11] and is thus not the focus of our tool. Instead, our tool is designed to handle the challenging cases of assigning taxonomic labels for reads of new species, which have not been observed before. This problem was clearly formulated in metagenomic phylogenetic classification tools such as Phymm and PhymmBL [8] . We first define the classification problem following the formulation in Phymm: the classification problem is the assignment of a phylogenetic group label to each read in input data sets. As we are only interested in the hard case of classifying reads of new species, which have not been sequenced and thus no label exists for these species, the expected labels for these reads are the higher-rank taxonomic groups such as genus, family, or order. All our data and experiments are designed so that the test species are masked in the reference database and the trained models.\\nPrevious experiments have shown that nucleotide-level homology search is not sensitive enough for assigning higher rank taxonomic groups to reads from new species [9, 10] . Protein-based homology search is thus preferred for phylogenetic classification [11, 12] . In particular, a recent tool, VirusSeeker [13] utilized BLASTx to classify reads into bacteria, phages, and other viruses. It is not clear whether the protein homology search is able to generate accurate classification at lower ranks. We will evaluate this in our experiments.\\nThere are also alignment-free methods for phylogenetic classification. Many of those tools implemented different machine learning models and were mainly designed for bacteria. For example, RDP [14] adopted a Naïve Bayes based classifier that can automatically learn features from the reads and assigned taxonomic labels to sequences from bacterial and archaeal 16S genes and fungal 28S genes. The NBC classifier [15, 16] encoded metagenomic reads using k-mers and implemented a bag-of-word model to achieve the readlevel taxonomic classification. Phymm [8] utilized interpolated Markov models (IMMs) for metagenomic read classification. These learning-based models enable users to label reads originating from new species.\\nThe previous works have shown that sequence composition is still an important and informative feature for phylogenetic classification. Recent applications of deep learning models have achieved better performance than conventional machine learning methods in different sequence classification problems [17] . In particular, the convolution filters in Convolutional Neural Network models (CNN) can represent degenerated sequence motifs and thus help CNN models to learn abstract composition-based features. For example, frequently activated convolution filters learned by DeepFam often represented the conserved motifs in different protein domain families [18] . The successful applications of deep learning in sequence classification motivated us to design a novel deep learning-based classification model for assigning taxonomic groups for new species in viral metagenomic data.\\nOur main contributions are summarized below. First, our deep learning model can assign higher rank phylogenetic group labels (such as genus) for reads from new species. To achieve this goal, our model organizes multiple CNN-based classifiers in a tree for hierarchical taxonomic group assignment. Second, to enable our model to capture as much information as possible from short reads, we implemented and compared two encoding methods: one hot vs. embedding. Third, as viral metagenomic data usually contains contaminations from either the host genomes or other microbes, we formulated the pre-processing step as an open set problem in targeted image classification and rejected non-viral reads by choosing appropriate negative training sets. We tested our model on both simulated metagenomic data and also real sequencing data. The results show that our model competes favorably with other popular methods.\\nIn this section, we will introduce our method for viral metagenomic taxonomic classification. First, we will show the architecture of CHEER, which is a hierarchical classification model from order to genus. Second, classifiers at each level will be described. Third, we will introduce DNA sequence encoding using skipgram based word embedding and compare it with onehot encoding. Forth, we will describe the rejection layer, following the idea of the open set problem, which is adopted to filter reads not belonging to RNA viruses. Finally, we will detail our dataset used in training and validation.\\nThe architecture of our model is sketched in Fig. 1. The key component is a tree model that consists of multiple classifiers from order to genus. In order to conduct phylogenetic classification for reads from new species, or even new genus, our classification is conducted using a top-down approach from the root to the leaf node. The top layer is a trained CNN that can reject reads not belonging to RNA viruses, which is an important preprocessing step for viral metagenomic data because of the contamination from the host genome or other microbes. Then, the filtered reads (i.e. mostly RNA viral reads) will be fed into the hierarchical classification model, which is referred to as the tree model hereafter. Specifically, the order-level classifier is responsible for classifying the reads into their originating orders. Then each order has a separately trained classifier for assigning input reads to families within that order (O1 to O3 in Fig. 1 ). For all the reads assigned into one family, the family classifier (F1-F7) will assign the reads to different genus within that family. As an example, a path for classifying a read from a species in genus 7 is highlighted in Fig. 1 . CHEER also implemented early stop functions at each level for stopping the classification path at higher ranks. This function can accommodate taxonomic classification for species from new genus or even higher ranks, which is possible for RNA viruses. It is convenient to add fine-grained classifiers within each genus so that we can assign species-level labels. However, as our focus is to conduct phylogenetic classification for reads from new species that do not have a species labels in the training data, we did not include the species-level assignment in this work.\\nAlternative designs of hierarchical classification work, like [19, 20] , either build a classifier for each taxonomic rank or build a binary classifier for each class. We implemented a structure that built just one classifier for each taxonomic rank and compared different structures' performance. According to the result shown in [Supplementary file 1], the structure shown in Fig.1 is better than one classifier per rank. Although one classifier per rank requires fewer classifiers, it needs to train many more parameters because the number of labels for each classifier increases by times. As a result, we observe overfitting more frequently.\\nEach classifier in the tree model is implemented using CNN, which has achieved superior performance in various sequence classification problems [18] . The convolution filters used in CNN resemble positionspecific weight matrix of motifs in genomic sequences. Frequently activated convolution filters can represent well-conserved motifs among sequences in the same class. Thus, a large number of convolution filters will be used in our CNN in order to learn well-conserved sequence features in different classes. Deep CNNs in fields such as computer vision often have multiple convolution layers in order to obtain more abstract features from training images. Here, we use a wide convolution design rather than a deep one, following the idea in [18] . Fig. 2 shows the classifier with the default hyper-parameters. First, reads will be encoded in matrices as input to the convolution layer using two methods. We will describe and compare these two encoding methods later in the Section 2.3. Then, multiple convolution filters of different sizes are utilized to learn the conserved sequence features. According to our empirical study, we choose 3, 7, 11, 15 as the filter sizes, with 256 filters for each size respectively (see Fig. 2 ). Max pooling is then applied to each convolution filter's output so that only the highest convolution value is kept. After the max-pooling layer, we have a 256-dimentional vector as the output of each convolution layer. We concatenate these four vectors into one 1024-dimentional vector and feed it to one fully connected layer with 512 hidden units. Dropout is applied to mitigate overfitting. The CrossEntropyLoss function supplied by Pytorch is adopted as an error estimation function. Due to the unbalanced size of each training class, we use the reciprocal of the proportion of each class as their error weight to help the model fairly learn from each class. And we utilize Adam optimizer with a 0.001 learning rate to update the parameters. The model is trained on HPCC with the 2080Ti GPU unit. \\nIn previous deep learning-based sequence classification models [21, 22] , one hot encoding was commonly adopted to convert a sequence into a matrix, which will then be used as input to convolution layers. There also exist some methods using k-mer composition [23] and frequency [15, 16] to encode the DNA sequence, similar to the bag-of-words model in the field of natural language processing. However, these k-mers representation cannot give complete sequential information about the original reads.\\nTo incorporate both the k-mer composition and also their ordering information, we build a Skip-Gram [24] based embedding layer that can learn which k-mers tend to occur close to each other. Embedding is widely used in natural language processing to learn the semantic and syntactic relationships from sentences. A neural network with one hidden layer is trained to map a word to an n-dimensional vector so that the words that usually occur together will be closer in the n-dimension space. For our DNA read classification task, k-mers are the words and the embedding layer will map proximate kmers into vectors of high similarity. The main hyper-parameter when training the skip-gram model is the size of the training context m, which defines the maximum context location at which the furthest k-mer is taken for training. Specifically, for a kmer at position i as the input to the skip-gram model, the output k-mers are its neighbors located at i+jk, where −m ≤ ≤ m. Larger m results in more training samples with a cost of training time. In our model, the default value of m is 1. As shown in Fig. 3 , we will sample 3 k-mers as our dataset at a time with the middle one as the input and the surrounding two as the output.\\nAs proven in [24] , Skip-Gram is able to automatically learn the relationship between proximate words and thus is expected to help learn the order between k-mers in genomic data. We implemented a hidden layer with 100 hidden units shown in Fig. 3 . After training the model, the co-occurrence features are embedded in the output of the hidden layer. Thus, a k-mer in a read will be represented as a 100-dimension vector, where each element in the vector is the output of the hidden unit in the hidden layer.\\nWhen we use this pre-trained embedding layer to encode the reads, we first split each read into sequential k-mers. Second, each k-mer is fed to the embedding layer and the output is a 100-dimension vector. Third, all these vectors are combined into a large matrix according to the original position of each k-mers. Finally, this matrix will be fed to the convolutional layer.\\nBased on our large-scale experiments, we found that classifiers are easy to overfit with increase of k. Thus, we choose 100-dimensional feature vectors with k being 3 as our default embedding layer parameter set based on our empirical analysis.\\nDue to the small genome size and limitations in library construction, it is not rare that many viral metagenomic datasets are still contaminated with the genomes of the host and other microbes. Viral read screening is thus a key pre-processing step before the downstream composition analysis. Specifically, our tree model should only be applied to viral reads. However, the close set nature of the classifier forces it to assign a known label to each input without being able to distinguish viral reads from others. In this work, we will reject non-viral reads by formulating it as an open set problem in targeted image classification [25] .\\nIn a majority of neural network models [26, 27] , the logit of the last layer will be fed into the SoftMax function, which takes a vector of n real numbers as input and output a normalized probability vector for all labels/classes. However, for an irrelevant input from an unknown clade, all classes in the model tend to have low probabilities and thus applying a threshold on uncertainty can be used to reject unknown classes [25] . Along with this idea, serval approaches [28, 29] have been proposed to solve the open set problem.\\nIn this work, we incorporate the pre-processing step in our hierarchical classification model by integrating a trained CNN and a SoftMax threshold. Contaminations such as bacteria will be rejected using empirically chosen SoftMax threshold τ. Our experiments show that SoftMax outputs a nearly uniform probability distribution of all labels if the input is not relevant. Thus, the SoftMax threshold τ is used to reject reads not belonging to either DNA or RNA viruses. However, using this thresholding method is not able to distinguish DNA viral reads from RNA viral reads due to their similar sequence composition. To accommodate this challenge, we designed a rejection classifier in the top layer using DNA viral reads as the negative class in the training (see Fig. 1 ).\\nAs the focus of our work is to assign labels for new species, the hierarchical classification ends at genuslevel assignment, which is conducted by the bottom layer in Fig. 1 . Given the fast accumulation of viral metagenomic data, there are cases where a new species does not belong to any existing genus and thus the label assignment should stop at the higher ranks. Thus, we allow our model to stop at a higher rank in the tree during the top-down classification. Towards this goal, we still apply SoftMax threshold in our early stop function, which stops reads from further classification by the child classifiers. This early stop function also serves as a quality control measure for preventing wrong classifications if the confidence is low. For example, when a read arrives, the model may be able to identify its order and family labels with high reliability but cannot decide which genus it belongs to. In this case, the model should only assign order and family label to this read rather than classifying it into a potentially wrong genus. To achieve this goal, SoftMax threshold is applied for every classifier in the tree. Only reads with a score higher than the threshold will be assigned to the child class.\\nAn example of early stop is highlighted by the yellow path in Fig. 1 . The model can classify the read into order 2 (O2) and then family 4 (F4) with high reliability but cannot decide which genus it belongs to.\\nAll the viruses used in the experiments were obtained from the RefSeq in the NCBI virus database. The The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.26.009001 doi: bioRxiv preprint viruses' taxonomic classification was downloaded from Virus Taxonomy 2018b Release [30] published by the International Committee on Taxonomic of Virus (ICVT). Because training deep learning classifiers required enough samples, we removed the orders with only one family, families with only one genus, and genera with less than 3 species from our dataset. Finally, there are 6 orders, 23 families, and 55 genera remained. The RefSeq genomes are available at the following URL:https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/fi nd-data/virus. The information of the taxa can be found in [Supplementary file 1]. As CHEER conducts read-level taxonomic classification, the training is also conducted using reads. In order to train our model on each complete virus genome, we generated simulated reads by uniformly extracting substrings of 250 bp with 200bp overlaps. All these reads formed the training set so that the model can learn the features from the whole genome. The validation set, however, is generated by read simulation tool WgSim [31] with specified error rate so that we can have a better estimation of the model's performance on real sequencing data. Some existing deep learning tools for taxonomic classification, like [21, 23] , generate reads first, then randomly choose some of them as training set and others as testing set. This strategy cannot guarantee that the training and testing data sets have no overlaps from the same species. In order to test whether our model can learn the label of new species, we use test species masking in the training set. Specifically, we split our RNA virus database into two independent sets. For any test species, their reads are not used in the training data. Thus, the model will only use the knowledge of known species for the new species classification.\\nIn this section, we will first give details of the hyperparameters of CHEER. Then we present the results of applying CHEER to both simulated and mock sequencing data. For the simulated data by WgSim, we evaluated how sequencing error rate affects the performance of CHEER and compared the taxonomic classification results with state-of-the-art. For the mock data, we tested CHEER's performance in both taxonomic classification and its ability in distinguishing viral reads from others.\\nIn our experiments, the minimum number of species within a genus is 3. We thus constructed three test sets, each containing reads from one species that is randomly chosen from each genus. The three test sets contain three different species, respectively. In total, 165 species from 55 genera are chosen to be \"new\" species for testing and the information of these species can be found in [Supplementary file 1].\\nWe used WgSim to simulate reads from testing species. In order to evaluate the impact of sequencing error on CHEER, we simulated reads with an error rate of 0, 0.01, 0.02, and 0.05, respectively. For each classifier (CNN), we computed the classification accuracy, which is defined in equation (1). To ensure that the computation of the accuracy is not affected by the number of reads from different species, we constructed a balanced test read set by simulating the same number of reads for each test species (1,000 reads for each). (1) For each component classifier in the tree model, we record the average accuracy for multiple test species. For each taxa level, we calculate the average accuracy of all the classifiers in the same level.\\nWe present the results of CHEER at each level in this section. Because CHEER is designed for new species, we benchmark it against alignment-free models. Based on the results shown by Wood et al. [7] , Naïve Bayes Classifier (NBC) outperformed other alignment-free models. Thus, we compared the performance of CHEER and NBC. NBC in [14] was designed for bacteria taxonomic classification and thus, we retrained NBC on our RNA virus dataset. We implemented the NBC model following [14] with the same parameters. Since each family has a separately trained classifier, which conducts genus classification within that family, there are totally 14 classifiers at this level. Fig. 5 shows the accuracy for labeling families. There are totally 6 classifiers that assign reads to families within each of the 6 orders. Fig. 6 shows the mean accuracy at each rank. The family and genus level accuracy are the average accuracy of all classifiers in Fig. 4 and Fig. 5 . There is only one classifier for order level classification.\\nAs shown in Fig. 4, Fig. 5, and Fig. 6 , CHEER has better performance than NBC across different ranks. We also compared the performance of using one-hot encoding and embedding layer in CHEER. The performance comparison at each rank shows that for most of the classifiers, the embedding layer improves the learning ability. Fig. 4, Fig. 5, and Fig. 6 also revealed the robustness of CHEER. Although the classification accuracy decreases as the error rate increases, the decrease is smaller than 2%. The accuracy is almost identical for many classifiers. Thus, CHEER is not very sensitive to sequencing errors. We also conducted experiments for reads with higher error rates (0.02 and 0.05). The results can be found in [Supplementary file 1].\\nThe next experiment is to evaluate our model on a real sequencing dataset and test CHEER's performance in both taxonomic classification and non-viral read rejection. We built a mock dataset by mixing Ebolavirus reads with gut bacterial reads from a gut amplicon sequencing dataset, which is chosen to ensure that we only use reads from bacteria. The taxonomic classification of Zaire Ebolavirus in the tree model is highlighted in Fig. 7 . We chose one of the Ebola species named Zaire Ebolavirus as our testing species because it is the most dangerous of the five Ebolavirus species and was responsible for the 2014 West African Ebola outbreak. We downloaded the dataset from NCBI with accession number SRR1930021. These reads were sequenced by Illumina using whole-genome amplification strategy. Since Zaire Ebolavirus usually causes fatal hemorrhagic fever and abdominal pain, it is possible that this Ebola virus occurs with other gut bacteria. Thus, we created a mock viral metagenomic data by mixing these Ebola reads with a gut bacteria amplicon sequencing data in order to test whether CHEER can reject the bacterial reads at the top rejection layer. The gut bacteria dataset is downloaded from NCBI with accession number: SRR10714026. There are totally 2,737 Ebola virus reads and 51,280 gut bacterial reads in the mock data. After the rejection layer, the Ebola reads will be used to evaluate our tree model.\\nTo treat Zaire Ebolavirus as a \"new\" species, we removed its genome from our training set and re-trained the whole pipeline. As shown in Fig. 7 , after removing this species, we still have four Ebola species in our training set. However, our read mapping experiment using Bowtie2 [32] showed that none of the reads from Zaire Ebolavirus can be mapped to other Ebola species with the default parameters, indicating that using usual alignment-based methods will have difficulty to conduct genus-level assignment.\\nIn our work, CHEER will first use rejection layer to remove the reads not belonging to the RNA virus. Second, the \"Order classifier\" will be applied to evaluate how many reads can be classified into Mononegavirales order. Third, \"Mononegavirales classifier\" takes all these Mononegavirales reads as input to predict how many reads will be assigned to the Filoviridae family. Finally, \"Filoviridae classifier\" is adopted to check how many reads belong to the Ebola genus. Fig. 8 show the ROC curve of the rejection layer. True positive rate reveals how many Ebolavirus reads pass the rejection layer. The false positive rate reveals how many gut bacterial reads pass the rejection layer. We generated thresholds from 0 to 1 with step size 0.01 and used each threshold for the rejection layer. Then for each threshold, we recorded the true positive rate and false positive rate for plotting the ROC curve. As shown in Fig. 8 , with the increase of the threshold, more bacterial reads are filtered by the rejection layer, at the expense of missing more real Ebola reads. The default threshold for the rejection layer is 0.6 in CHEER. User can adjust the threshold according to their needs.\\nAfter evaluating the rejection layer, we continued to test the hierarchical classification performance at each rank. We first compared our tree model with NBC. Then we also benchmarked CHEER with the method using protein-level alignment in Section 3.3.3. The performance of the complete hierarchical classification pipeline of NBC and CHEER is shown in Fig. 9 . From order to genus, each bar shows the accuracy of correctly classified reads by the parent classifier. For example, NBC correctly classified 1580 at the order level. Then, 1564 out of the 1580 reads are correctly labeled at the family level. While Fig. 4 , 5, and 6 have shown the performance of CHEER and NBC at each taxonomic rank using the same set of reads as input for each level of classifiers, this experiment focused on comparing the hierarchical classification pipeline using only correctly predicted reads by the parent classifier. The results demonstrated that CHEER can classify Ebola reads with better performance. NBC can only classify 1,564 (57.1%) reads into Ebolavirus. CHEER, however, had totally 2407 (87.9%) reads classified correctly.\\nAlthough nucleotide-level alignment models are not ideal choices for predicting new species, as mentioned in [8] , protein-level alignment such as BLASTx is still frequently used for homology search because of the conservation among homologous protein sequences. We followed VirusSeeker [10] , which is a BLASTx based model, to align translated reads to virus-only protein database (VIRUSDB NR) provided by NCBI. Then, we kept alignments with E-value cutoff 0.001, which is used in VirusSeeker. The output of BLASTx shows that a read can often align to multiple proteins, making the phylogenetic classification ambiguous (see Fig. 11 ). Also, existing tools such as VirusSeeker often use the best alignment only, which does not always yield the highest accuracy. To optimize the usage of the alignment-based method, we implemented k-Nearest Neighbor (kNN) algorithm to record the best k alignments rather than only using the best one. The label of the input read is determined by the majority vote in the labels of the top k alignments. According to Fig. 10 , most of the reads have hit numbers between 11 to 15 and 6 to10. Thus, we chose 1, 5, 10, 15 as our k when running kNN. There are also 512 reads that cannot be aligned with any VIRUSDB protein sequence. All these reads will be regarded as wrong prediction. Fig. 11 shows the classification results of kNN. When we use the best hit as the prediction the accuracy is only 64%. The best performance of kNN will be 74% with 2,026 correct predictions when k is 5. But the accuracy sharply decreases if k is larger than 5. Fig. 10 and Fig. 11 also revealed that it is hard for users to choose a proper k in real classification tasks. The most common value of the hit number is 11-15 shown in Fig. 11 . The best kNN result, however, has a k equal to 5. Thus, comparing to this alignment-based method, CHEER, with 2,407 (87.9%) reads classified correctly, is more practical and reliable.\\nThe two main purposes of the early stop function are: 1) to handle the cases of a new species from an unseen genus or higher ranks; 2) to serve as a quality control measure for preventing wrong classifications in child classifiers. As Fig. 6 shows, the higher rank classification tends to be more accurate than lower ranks. Because the genus of Ebolavirus exists in our classification tree, the evaluation of the early stop function focuses on the second purpose. Fig. 12 shows the percentage of how many reads are classified into Ebolavirus genus correctly for different classifiers. By allowing early stop, the number of wrong predictions decreased sharply. In short, although there are fewer reads being labelled at the genus level, the total number of reads with wrong classification is decreased. Thus, this is still a useful feature provided by CHEER.\\nAnother significant advantage of CHEER is the high classification speed. Although training the model uses heavy computing resources, the inference procedure of this algorithm is fast. And with the help of GPU acceleration, the training time also decreases a lot.\\nIn the experiments shown in Section 3.4, it takes around 46 minutes to run BLASTx against the virus protein database. Our method, however, only takes 7 second for the preprocessing step to convert the raw sequence into a vector, and then takes less than 35 second to make the prediction for all 2,737 reads.\\nIn this work, all training data are RNA virus genomes downloaded from the NCBI RefSeq database. The training reads are extracted uniformly from the genome to make sure that classifiers can learn the features from the whole genome. To evaluate the model on new species detection, all species in the test set are excluded from the training data. Error rates are also added to evaluate the robustness of the model. The results reveal that the model is tolerant to the sequencing error. In addition, a SoftMax threshold rejection layer is applied to filter reads from non-RNA viruses. The SoftMax threshold in each classifier will also help early stop classification with low confidence and thus decreases the wrong predication rate. We show a case study on real-world data to identify how this threshold will influence the rejection classifier and early stop function. The results show that CHEER competes favorably with state-of-the-art tools.\\n</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                  title  \\\n",
       "0  The RNA pseudoknots in foot-and-mouth disease virus are dispensable for genome replication but essential for the production of infectious virus. 2 3   \n",
       "1  Analysis Title: Regaining perspective on SARS-CoV-2 molecular tracing and its implications                                                             \n",
       "2  Healthcare-resource-adjusted vulnerabilities towards the 2019-nCoV epidemic across China                                                               \n",
       "3  Relationship between Average Daily Temperature and Average Cumulative Daily Rate of Confirmed Cases of COVID-19                                        \n",
       "4  CHEER: hierarCHical taxonomic classification for viral mEtagEnomic data via deep leaRning                                                              \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      abstract  \\\n",
       "0  [{'text': 'word count: 194 22 Text word count: 5168 23 24 25 author/funder. All rights reserved. No reuse allowed without permission. Abstract 27 The positive stranded RNA genomes of picornaviruses comprise a single large open reading 28 frame flanked by 5′ and 3′ untranslated regions (UTRs). Foot-and-mouth disease virus (FMDV) 29 has an unusually large 5′ UTR (1.3 kb) containing five structural domains. These include the 30 internal ribosome entry site (IRES), which facilitates initiation of translation, and the cis-acting 31 replication element (cre). Less well characterised structures are a 5′ terminal 360 nucleotide 32 stem-loop, a variable length poly-C-tract of approximately 100-200 nucleotides and a series of 33 two to four tandemly repeated pseudoknots (PKs). We investigated the structures of the PKs 34 by selective 2′ hydroxyl acetylation analysed by primer extension (SHAPE) analysis and 35 determined their contribution to genome replication by mutation and deletion experiments. 36 SHAPE and mutation experiments confirmed the importance of the previously predicted PK 37 structures for their function. Deletion experiments showed that although PKs are not essential 38', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'for replication, they provide genomes with a competitive advantage. However, although 39 replicons and full-length genomes lacking all PKs were replication competent, no infectious 40 virus was rescued from genomes containing less than one PK copy. This is consistent with our 41 earlier report describing the presence of putative packaging signals in the PK region. 42 43 author/funder. All rights reserved. No reuse allowed without permission.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]   \n",
       "1  [{'text': 'During the past three months, a new coronavirus (SARS-CoV-2) epidemic has been growing exponentially, affecting over 100 thousand people worldwide, and causing enormous distress to economies and societies of affected countries. A plethora of analyses based on viral sequences has already been published, in scientific journals as well as through non-peer reviewed channels, to investigate SARS-CoV-2 genetic heterogeneity and spatiotemporal dissemination. We examined all full genome sequences currently available to assess the presence of sufficient information for reliable phylogenetic and phylogeographic studies. Our analysis clearly shows severe limitations in the present data, in light of which any finding should be considered, at the very best, preliminary and hypothesis-generating. Hence the need for avoiding stigmatization based on partial information, and for continuing concerted efforts to increase number and quality of the sequences required for robust tracing of the epidemic.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': '. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}, {'text': 'is the (which was not peer-reviewed) The copyright holder for this preprint .', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]                                                                                                                                                                                                                                                                                                                                         \n",
       "2  []                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
       "3  [{'text': 'The rapid outbreak of the new Coronavirus pandemic and the spread of the virus worldwide, especially in the Northern Hemisphere, have prompted various investigations about the impact of environmental factors on the rate of development of this epidemic. Different studies have called the attention to various parameters that may have influenced the spread of the virus, and in particular, the impact of climatic parameters has been emphasized. The main purpose of this study is to investigate the correlation between the average daily temperature and the rate of coronavirus epidemic growth in the infected regions. The main hypothesis object of our research is that between regions exhibiting a significant difference in the mean daily temperature, a significant difference is also observed in the average cumulative daily rate of confirmed cases, and that this does not happen if there is no significant difference in mean daily temperature. To test this research hypothesis, we carried on the case study of three regions in each of five countries and analyzed the correlation through F-test, and Independent-Samples T-Test. In all five selected countries, we found that when there is a significant difference in the daily mean temperature between two regions of a country, a significant difference exists also in the average cumulative daily rate of confirmed cases. Conversely, if there are no significant differences in the mean daily temperature of two regions in the same country, no significant difference is observed in the average cumulative daily rate of confirmed cases for these regions.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]                                                                                                       \n",
       "4  [{'text': 'The fast accumulation of viral metagenomic data has contributed significantly to new RNA virus discovery. However, the short read size, complex composition, and large data size can all make taxonomic analysis difficult. In particular, commonly used alignment-based methods are not ideal choices for detecting new viral species. In this work, we present a novel hierarchical classification model named CHEER, which can conduct read-level taxonomic classification from order to genus for new species. By combining k-mer embedding-based encoding, hierarchically organized CNNs, and carefully trained rejection layer, CHEER is able to assign correct taxonomic labels for reads from new species. We tested CHEER on both simulated and real sequencing data. The results show that CHEER can achieve higher accuracy than popular alignmentbased and alignment-free taxonomic assignment tools. The source code, scripts, and pre-trained parameters for CHEER are available via GitHub: https://github.com/KennthShang/CHEER.', 'cite_spans': [], 'ref_spans': [], 'section': 'Abstract'}]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               full_text  \\\n",
       "0  VP3, and VP0 (which is further processed to VP2 and VP4 during virus assembly) (6). The P2 64 and P3 regions encode the non-structural proteins 2B and 2C and 3A, 3B (1-3) (VPg), 3C pro and 4 structural protein-coding region is replaced by reporter genes, allow the study of genome 68 replication without the requirement for high containment (9, 10) ( figure 1A ).\\nThe FMDV 5′ UTR is the largest known picornavirus UTR, comprising approximately 1300 71 nucleotides and containing several highly structured regions. The first 360 nucleotides at the 5′ 72 end are predicted to fold into a single large stem loop termed the S-fragment, followed by a The PKs were originally predicted in 1987 and consist of two to four tandem repeats of a ~48 86 nucleotide region containing a small stem loop and downstream interaction site (figure 1B) 87 (12). Due to the sequence similarity between the PKs (figure 1C), it is speculated that they 88 were formed by duplication events during viral replication, probably involving recombination. 89 Between two and four PKs are present in different virus isolates but no strain has been 90 identified with less than two PKs, emphasising their potential importance in the viral life cycle 91 (19, 20) . The presence of PKs has been reported in the 5′ UTR of other picornaviruses such as 92 author/funder. All rights reserved. No reuse allowed without permission. can occur in the absence of PKs at least one is required for wild-type (wt) replication. 104 Furthermore, competition experiments showed that extra copies of PKs conferred a replicative 105 advantage to genomes. Although replicons and full-length genomes lacking PKs were 106 replication-competent, no infectious virus was rescued from genomes containing less than one 107 PK copy. This is consistent with our earlier report describing the presence of putative 108 packaging signals in the PK region (22). 109 110 author/funder. All rights reserved. No reuse allowed without permission. Plasmid construction. 117 The FMDV replicon plasmids, pRep-ptGFP, and the replication-defective polymerase mutant 118 control, 3D-GNN, have already been described (10).\\nTo introduce mutations into the PK region, the pRep-ptGFP replicon plasmid was digested 121 with SpeI and KpnI and the resulting fragment inserted into a sub-cloning vector (pBluescript) 122 to create the pBluescript PK. PKs 3 and 4 were removed by digestion with HindIII and AatII 123 before insertion of a synthetic DNA sequence with PK 3 and 4 deleted. PKs 2, 3 and 4 were 124 deleted by PCR amplification using ΔPK 234 Forward primer and FMDV 1331-1311 reverse 125 primer, the resultant product was digested with HindIII and AatII and ligated into the 126 pBluescript PK vector. Complete PK deletion was achieved by introduction of an AflII site at 127 the 3′ end of the poly-C tract by PCR mutagenesis to create the sub-cloning vector, pBluescript 128 C11, which was then used to remove all the PKs by PCR mutagenesis using ΔPK 1234 forward 129 primer and FMDV 1331-1311 reverse primer. The modified PK sequences were removed from 130 the sub-cloning vectors and inserted into the pRep-ptGFP plasmid using NheI-HF and KpnI-131 HF.\\n132 133 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 7 Mutations to disrupt and reform PK structure were introduced using synthetic DNA by 134 digestion with AflII and AatII and ligation into a similarly digested pBluescript PK vector.\\nMutations were then introduced into the replicon plasmid as described above.\\nTo assess the effects of truncation of the poly-C-tract on replication the entire sequence was 137 removed. This was performed by PCR mutagenesis using primers C0 SpeI, and FMDV 1331- In vitro transcription. 143 In vitro transcription reactions for replicon assays were performed as described previously (28).\\nTranscription reactions to produce large amounts of RNA for SHAPE analysis were performed 145 with purified linear DNA as described above, and 1 μg of linearised DNA was then used in a 146 HiScribe T7 synthesis kit (NEB), before DNase treatment and purification using a PureLink FastQ files were quality checked using FastQC with poor quality reads filtered using the 225 Sickle algorithm. Host cell reads were removed using FastQ Screen algorithm and FMDV 226 reads assembled de novo into contigs using IDBA-UD (35). Contigs that matched the FMDV 227 library (identified using Basic Local ALighnment Search Tool (BLAST)) were assembled 228 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint into consensus sequences using SeqMan Pro software in the DNA STAR Lasergene 13 229 package (DNA STAR) (36). The SHAPE data largely agreed with the predicted structures with the stems of PK 1, 2 and 3, interacting nucleotides showed little to no reactivity, suggesting NMIA could not interact with 300 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 14 these nucleotides either due to the predicted base pairing or steric hindrance (figure 2B). The\\nNMIA reactivity for the interacting nucleotides in the stem-loops with downstream residues of 302 PK 1, 2 and 3 again largely agreed with the predicted structure, although the SHAPE data 303 suggests that there might be fewer interactions than previously predicted. However, differences 304 here could be due to heterogeneity in the formation of PKs in this experiment. The evidence 305 for loop-downstream interaction was weaker for PK4. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint\\norientation. 351 Since removal of all four PKs resulted in a significant decrease in replication, the minimal 352 requirements to maintain wt levels of replication were investigated. As near wt level of 353 replication was observed when only one PK was present, all further mutagenesis was 354 performed in a C11 replicon plasmid containing only PK 1. In addition, the orientation of PK 1 was reversed by \"flipping\" the nucleotide sequence to 367 potentially facilitate hybridisation of the loop with upstream rather than downstream sequences.\\nChanging the orientation of the PK reduced replicon replication to a similar level seen in the replication decreased until at passage three there is a 2.5 fold reduction compared to that of 398 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint passage 0 (figure 5B). Therefore, it appears that replicons with a single PK are at a competitive 399 disadvantage compared to those with two or more. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 20 of infectious virus despite being able to replicate after transfection into cells, is consistent with 448 a requirement for RNA structure within the PK region being required for virus assembly. The 5′ UTR of FMDV is unique amongst picornaviruses due to its large size and the presence 454 of multiple RNA elements, some of which still have unknown function. One of these features 455 is a series of repeated PKs varying in number from 2-4, depending on virus strain. In this study, 456 we sequentially deleted or mutated the PKs to help understand their role in the viral life cycle. 457 We also confirmed the predicted PK structures by SHAPE mapping, although there may be Although all viruses isolated to date contain at least two PKs, replicons or viruses containing a 464 single PK were still replication competent. However, replicons with more than a single PK 465 were found to have a competitive advantage over replicons with a single PK when sequentially 466 passaged. Replicons lacking all PKs displayed poor passaging potential even when co-467 transfected with yeast tRNA, reinforcing the observation of a significant impact in replication.\\nMoreover, viruses recovered from genomes with reduced numbers of PKs were slower growing 469 and produced smaller plaques. In addition, these differences were more pronounced in more PKs is functionally competent as no differences was seen between replicons congaing a single 472 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 21 copy of PK1 or PK4. This observation is consistent with a previous report of deletion of PK1, 473 along with the Poly-C-tract, with no adverse effect in viral replication (37). This also supports 474 our findings that the truncation of the Poly-C-tract to create the C11 construct had no effect on 475 replicon replication in the cell lines tested. As has been described with Mengo virus, it is 476 possible that the role of the poly-C-tract is essential in other aspects of the viral lifecycle which 477 cannot be recapitulated in a standard tissue culture system (39).\\nThe presence of at least two PKs in all viral isolates sequenced so far suggests that multiple 480 PKs confer a competitive advantage in replication. Here we showed by sequential passage that 481 replicons containing at least two PKs were maintained at a level similar to wt, but replicons 482 containing only one PK showed a persistent decline. It is unclear why some viral isolates 483 contain two, three or four PKs is still unknown, but this may be stochastic variation or may 484 reflect subtle effects of host range or geographical localisation. The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint Significance is shown comparing the replication of C11 PK disrupt and C11 PK restore (Aii). Significance shown is compared to wt replicon. Error bars are calculated by SEM, n = 3, * P 673 < 0.05, **** P < 0.0001. 674 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint 33 675 author/funder. All rights reserved. No reuse allowed without permission.\\nThe copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.01.10.901801 doi: bioRxiv preprint \\n                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
       "1  In December 2019, a novel coronavirus, SARS-CoV-2, was identified in Wuhan, China, as the etiologic agent of coronavirus disease 2019 , which by March 2020 has already spread across more than 80 countries 1 . Common symptoms of infection include fever, cough, and shortness of breath, while severe cases are characterized by advanced respiratory distress and pneumonia, often resulting in death 2 . Soon after the first epidemiological data together with SARS-CoV-2 genetic sequences were made available, a glut of phylogeny-based analyses began to circulate discussing, in scientific papers as well as (social) media, countries that might have been fueling the spread. The implications of misunderstanding the real dynamic of the COVID-19 pandemic are extremely dangerous. Ethnic or social discrimination resulting from unsupported assumptions on viral contagion -often amplified by irresponsible, uncontrollable communications -can be highly damaging for people and countries.\\nDespite (social) media are often vehicle for fake news and boast news hype, it is also worth noting the tremendous effort of the scientific community to provide free, up-to-date information on ongoing studies, as well as critical evaluations. In particular, the US-based NextStrain 3 team has been posting real-time updates on the epidemic tracing by molecular analyses. Several discussions and evidence-based debates on controversial hypotheses on the epidemic have ensued -such as the number of untraced infections in the US, the putative virus introduction in Italy through Germany, and the alleged lineage diversification in China 4 later criticized 5 -aided by a concerted and productive effort of internationally-renowned research groups in Edinburgh, Glasgow, and Oxford, UK, among others. Recently, an editorial published on Science 6 has also highlighted how unsupported or misleading claims circulating in forums, social media, and even peer-reviewed articles, have been led by a substantial over interpretation of the available data.\\nHence, the urgency to reframe the current debate in more rigorous scientific terms, and quantitatively evaluate whether sufficient information for reliable phylogenetic and phylogeographic studies currently exists, or which gaps need to be addressed.\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint Before carrying out any phylogeny-based analysis of virus evolution and spatiotemporal spread, it is crucial to test the quality of sequence data, since uneven sampling, presence of phylogenetic noise, and absence of temporal signal can affect reliability of the results (e.g. ancestral state reconstructions, molecular clock calibrations) 7 . We downloaded all SARS-CoV-2 full genome sequences deposited on GISAID (https://www.gisaid.org/) 8 to compare the number of high quality genomes sampled per country with the number of confirmed cases at the time of sampling, as well as the country's total population ( Figure 1 ). As March 10 th 2020, 331 genome sequences from 29 countries were available, almost twice as much than just one week earlier (March 3 rd 2020, 169 genome sequences from 22 countries). Yet, the sampling per country was inhomogeneous, with confirmed cases clearly uncorrelated to the number of genomes (Spearman correlation = 0.03). Moreover, correlation could only be investigated with confirmed cases (as proxy), since not all affected countries have made publicly available the total number of coronavirus testing performed. Even within the same country, sequenced genomes were usually sampled from few hotspots, not necessarily representative of the whole epidemic in that country.\\nItaly, for example, the second country per confirmed cases, has uploaded so far only two genomes from Lombardy and three from Lazio (Table S1) , i.e. only two of the twenty Italian regions, all with confirmed cases 9 . China and USA contributed with 123 and 63 genomes respectively (56% combined), while 13 countries contribute with only one or two samples (6.3% combined). Notably, three countries with less than 400 confirmed cases have provided a substantial number of genomes (Netherlands 28, Australia 18, Singapore 11; 17% combined) ( Figure 1 ).\\nThe effect of inhomogeneous sampling and missing data on phylogeography reconstructions, like the ones recently rushed through news and (social) media to claim specific dissemination routes of SARS-CoV-2 among countries, can be quite dramatic. An instructive example is the putative introduction of SARS-CoV-2 in Italy from Germany. The maximum likelihood (ML) tree, inferred from the full genome viral sequences available on March 3 rd , 2020 (Supplementary Figure S1 ), showed a well-supported cluster of European and Asian sequences (reported in Figure 2a ), which contained a subclade (subclade A, Figure 2a ) including a sequence isolated in Germany that . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint appears to be paraphyletic (with strong bootstrap support) to an Italian sequence clustering, in turn, with sequences from Finland, Mexico, Germany and Switzerland (Figure 2a ).\\nBased on this observation (which was available on NextStrain), a heated discussion circulated on social media about a transmission event from Germany to Italy followed by further spread from Italy the other countries. However, in a new tree inferred just one week later, when more than 135 new full genome sequences were made available on GISAID 8 ( Figure S2 ), the direct link between Germany and Italy has disappeared due to the additional clustering of previously unsampled sequences from Portugal, Brazil, Wales and Netherland ( Figure 2b ). Most importantly, there is no sufficient evidence of a higher likelihood (Shimodaira-Hasgawa test, Table 1 ) for tree topologies generated arbitrarily switching branches in the tree (see arrows in Figure 2b) compared to the ML tree inferred from the real data. In other words, it is not possible with the present data to decide which branching pattern (and, therefore, phylogeographic reconstruction)\\nis the most likely.\\nLack of resolution and uncertainty in the SARS-CoV-2 phylogenetic tree is to be expected, considering that relatively little genetic diversity can be accumulated during the first three months of an epidemic, even for an exponentially spreading and fast-evolving RNA virus. Plots of pairwise transitions/transversions vs. genetic distance 10 showed, indeed, low genetic diversity among SARS-CoV-2 genomes, considerable scattering, and overlap in transitions and transversions among closely related sequences (Figure 3a ). The 331 full genomes available contained 477 out of 29,564 variable sites (1.6%), of which 377 were singleton and only 100 (0.34%) parsimony informative, indicating a substantial lack of phylogenetic information, and raising the possibility of a number of sequencing errors. Further assessment of phylogenetic signal in the dataset was carried out using likelihood mapping analysis 11 , which estimates the likelihood of each of possible tree topology for any group of four sequences (quartet), randomly . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint chosen from an alignment, and reports them inside an equilateral triangle (the likelihood map)\\nwhere the corners represent distinct tree topologies and the center star-like trees. Quartets are considered \"resolved\" when the three likelihoods are significantly different (phylogenetic signal, most dots equally distributed in the corners, i.e. data are suitable for robust phylogeny inference), unresolved or partially resolved, when all three likelihood values or two of them are not significantly different (phylogenetic noise, most dots in sides or center areas, i.e. data may not be sufficient for robust phylogeny inference). Extensive simulation studies have shown that, for sequences to be considered robust in terms of phylogenetic signal, side/center areas of the likelihood mapping must include <40% of the unresolved quartets 12 , way lower than the 52.9% unresolved quartets detected in the SARS-CoV-2 full genomes alignment ( Figure 3b ). It is important to keep in mind that such a lack of phylogenetic signal makes the overall topology of any SARS-CoV-2 tree unreliable, and even clades with high bootstrap values should be interpreted with extreme caution. Finally, the mean genetic distance of each sequence from the root of a ML tree indicated that the available genomic data lack temporal signal (no significant linear correlation between accumulated genetic distance and sampling time, Figure 3c ). This is, again, expected in genomes obtained over a very short period of time (~ three months) since the beginning of the outbreak. Bayesian analysis 13 , which infers phylogenetic and phylogeographic patterns from a posterior distribution of trees, might facilitate comparisons about different evolutionary scenarios, help in retrieving the correct topology, and estimate an accurate evolutionary rate. With more genome sequences sampled from different time points and geographic areas made available, literally, every day, Bayesian analysis may soon be a viable option. However, as long the new data do not increase phylogenetic and temporal signal, Bayesian phylogenetic and phylogeography results would remain equally questionable.\\nTaken together, our observations suggest that any conclusion drawn, at present, about existing lineages and direction of viral spread, based on phylogenetic analysis of SARS-CoV-2 sequence data, is potentially misleading. Nowadays, published scientific data and media are very easily . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint .\\naccessible to a worldwide audience, thus it is important more than ever to weigh the information being shared. We believe that current molecular epidemiology data are not solid enough to provide a scientifically sound analysis of SARS-CoV-2 spread. Any inference based on phylogenetic trees should be considered at the very best preliminary, and hypothesis-generating.\\nHence the need for 1) avoiding stigmatization based on partial information and 2) continuing the concerted scientific effort to increase the number and quality of available data. The evolutionary dynamics of SARS-CoV-2 spread will unveil an unprecedented amount of information, which is essential to make policy decisions. The whole of humanity is threatened by the current pandemic, and policy makers need to adjust their mitigation measures while the pandemic itself is developing. Some of the urgent answers required lie in the timely availability of abundant, high quality sequence data not only from countries experiencing a high number of reported cases, such as Italy and South Korea, but also from others that seem to be experiencing, at least for now, a lower number of infections.\\nGISAID was accessed on March 3 rd and 10 th , 2020 (final dataset information reported in Table   S1 ). After quality control of sequences that were not full genomes or contained extensive stretches of unknown nucleotides, the first dataset (March 3 rd 2020) consisted of 169 genomes from 22 countries, while the second (March 10 th , 2020) 331 from 29 countries. Confirmed cases were retrieved for the same day from interactive web-based dashboard developed by Johns Hopkins University 14 .\\nTransition/transversions vs. genetic distance plot were calculated using DAMBE6 10 . Evaluation of the presence of phylogenetic signal satisfying resolved phylogenetic relationships among sequences was carried out with IQ-TREE, allowing the software to search for all possible quartets . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint using the best-fitting nucleotide substitution model 11 . ML tree reconstruction was performed in IQ-TREE based on the best-fit model chosen according to Bayesian Information Criterion (BIC) 15, 16 . Topology was tested performing the Shimodaira-Hasegawa 17 on \"Subclade A\" by comparing the inferred tree to three trees where two branches: (Tree 1) Italy (EPI_ISL_412973) switched with Wales (EPI_ISL_413555), (Tree 2) Germany (EPI_ISL_406862 with) Brazil (EPI_ISL_412964), (Tree 3) and Portugal (EPI_ISL_413648) with Brazil (EPI_ISL_412964), (Tree 4) Germany (EPI_ISL_406862) with Portugal (EPI_ISL_413648). All tests performed 10,000 resamplings using the using the RELL approximation method 18 . Position of these genomes within \"Subclade A\" is indicated in Figure 1 with an arrow.\\nExploration of temporal structure, i.e. presence of molecular clock in the data, was assessed by regression of divergence -root-to-tip genetic distance-against sampling time using TempEst 19 . In this case, absence of a linear trend indicates that the data does not contain temporal signal and that the data is not appropriate for phylogenetic inference using molecular clock models.\\nAncestral state reconstruction of full datasets reported in Figures S1 and S2 were based on the ML tree obtained with IQ-TREE and performed using TreeTime 20 . We choose to show the \"European subclade\" and \"Subclade A\" as cladograms in Figure 1 as there are no phylogenetic and temporal signals in the datasets: scaling phylogenies in time or nucleotide substitution in this case is misleading. . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint Each bi-directional arrow, and corresponding number, connects two tip branches that were switched to generate an alternative tree topology to be tested (see Table 1 , Methods). CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity. EPI_ISL_412971_BetaCoV_Finland_FIN-25_2020 EPI_ISL_406533_BetaCoV_Guangzhou_20SF206_2020 EPI_ISL_407896_BetaCoV_Australia_QLD02_2020 EPI_ISL_408478_BetaCoV_Chongqing_YC01_2020\\nEPI_ISL_412979__BetaCoV_Wuhan_HBCDC-HB-03_2020\\nEPI_ISL_411220_BetaCoV_France_IDF0386-islP3_2020 EPI_ISL_410044_BetaCoV_USA_CA6_2020 EPI_ISL_413020_BetaCoV_Switzerland_1000477377_2020\\nEPI_ISL_406030_BetaCoV_Shenzhen_HKU-SZ-002_2020\\nEPI_ISL_408669_BetaCoV_Japan_KY-V-029_2020\\nEPI_ISL_408515_BetaCoV_Wuhan_IVDC-HB-envF13-21_2020 EPI_ISL_410713_BetaCoV_Singapore_7_2020 EPI_ISL_412873_BetaCoV_Korea_KCDC24_2020\\nEPI_ISL_402120_BetaCoV_Wuhan_IVDC-HB-04_2020\\nEPI_ISL_406970_BetaCoV_Hangzhou_HZ-1_2020 EPI_ISL_406973_BetaCoV_Singapore_1_2020 EPI_ISL_412029_BetaCoV_Hong_Kong_VM20001988_2020\\nEPI_ISL_410720_BetaCoV_France_IDF0372-isl_2020 EPI_ISL_411951_BetaCoV_Sweden_01_2020 EPI_ISL_408488_BetaCoV_Jiangsu_IVDC-JS-001_2020\\nEPI_ISL_402123_BetaCoV_Wuhan_IPBCAMS-WH-01_2019 EPI_ISL_407894_BetaCoV_Australia_QLD01_2020 EPI_ISL_411955_BetaCoV_USA_CA8_2020 EPI_ISL_403962_BetaCoV_Nonthaburi_61_2020 EPI_ISL_406801_BetaCov_Wuhan_WH04_2020 EPI_ISL_408978_BetaCoV_Wuhan_WH05_2020\\nEPI_ISL_407084_BetaCoV_Japan_AI_I-004_2020 EPI_ISL_407215_BetaCoV_USA_WA1-F6_2020 EPI_ISL_411060_BetaCoV_Fujian_8_2020 EPI_ISL_410984_BetaCoV_France_IDF0515-isl_2020\\nEPI_ISL_412980__BetaCoV_Wuhan_HBCDC-HB-04_2020 EPI_ISL_410546_BetaCoV_Italy_INMI1-cs_2020 EPI_ISL_408977_BetaCoV_Sydney_3_2020 EPI_ISL_412982__BetaCoV_Wuhan_HBCDC-HB-06_2020 EPI_ISL_407071_BetaCoV_England_01_2020 EPI_ISL_412912_BetaCoV_Germany_Baden-Wuerttemberg-1_2020 EPI_ISL_411927_BetaCoV_Taiwan_4_2020 EPI_ISL_410536_BetaCoV_Singapore_5_2020 EPI_ISL_406538_BetaCoV_Guangdong_20SF201_2020 EPI_ISL_407988_BetaCoV_Singapore_3_2020 EPI_ISL_406036_BetaCoV_USA_CA2_2020 EPI_ISL_411954_BetaCoV_USA_CA7_2020 EPI_ISL_412028_BetaCoV_Hong_Kong_VM20001061_2020 EPI_ISL_406862_BetaCoV_Germany_BavPat1_2020 EPI_ISL_406597_BetaCoV_France_IDF0373_2020 EPI_ISL_408009_BetaCoV_USA_CA4_2020 EPI_ISL_410716_BetaCoV_Singapore_10_2020 EPI_ISL_403933_BetaCoV_Guangdong_20SF013_2020 EPI_ISL_402129_BetaCoV_Wuhan_WIV06_2019\\nEPI_ISL_406594_BetaCoV_Shenzhen_SZTH-003_2020 EPI_ISL_407987_BetaCoV_Singapore_2_2020 EPI_ISL_413015_BetaCoV_Canada_ON_VIDO-01_2020 EPI_ISL_407976_BetaCoV_Belgium_GHB-03021_2020 EPI_ISL_406531_BetaCoV_Guangdong_20SF174_2020 EPI_ISL_410715_BetaCoV_Singapore_9_2020\\nEPI_ISL_404227_BetaCoV_Zhejiang_WZ-01_2020 EPI_ISL_408479_BetaCoV_Chongqing_ZX01_2020 EPI_ISL_406535_BetaCoV_Foshan_20SF210_2020 EPI_ISL_408976_BetaCoV_Sydney_2_2020 EPI_ISL_411066_BetaCoV_Fujian_13_2020 EPI_ISL_407073_BetaCoV_England_02_2020\\nEPI_ISL_403929_BetaCoV_Wuhan_IPBCAMS-WH-04_2019 EPI_ISL_410301_BetaCoV_Nepal_61_2020 EPI_ISL_403931_BetaCoV_Wuhan_IPBCAMS-WH-02_2019\\nEPI_ISL_410531_BetaCoV_Japan_NA-20-05-1_2020 EPI_ISL_403963_BetaCoV_Nonthaburi_74_2020 EPI_ISL_402125_BetaCoV_Wuhan-Hu-1_2019\\nEPI_ISL_408514_BetaCoV_Wuhan_IVDC-HB-envF13-20_2020\\nEPI_ISL_402119_BetaCoV_Wuhan_IVDC-HB-01_2019 EPI_ISL_411956_BetaCoV_USA_TX1_2020 EPI_ISL_406596_BetaCoV_France_IDF0372_2020 EPI_ISL_412967_BetaCoV_China_IQTC02_2020 EPI_ISL_408431_BetaCov_France_IDF0626_2020\\nEPI_ISL_410532_BetaCoV_Japan_OS-20-07-1_2020\\nEPI_ISL_411957_BetaCoV_China_WH-09_2020 EPI_ISL_412970_BetaCoV_USA_WA2_2020 EPI_ISL_412978__BetaCoV_Wuhan_HBCDC-HB-02_2020 EPI_ISL_412869_BetaCoV_Korea_KCDC05_2020 EPI_ISL_412965_BetaCoV_Canada_BC_37_0-2_2020 EPI_ISL_412871_BetaCoV_Korea_KCDC07_2020 EPI_ISL_408008_BetaCoV_USA_CA3_2020\\nEPI_ISL_406593_BetaCoV_Shenzhen_SZTH-002_2020 EPI_ISL_406536_BetaCoV_Foshan_20SF211_2020 EPI_ISL_412459_BetaCoV_Jingzhou_HBCDC-HB-01_2020 EPI_ISL_410718_BetaCoV_Australia_QLD04_2020 EPI_ISL_412862_BetaCoV_USA_CA9_2020 EPI_ISL_413019_BetaCoV_Switzerland_1000477102_2020\\nEPI_ISL_412981__BetaCoV_Wuhan_HBCDC-HB-05_2020\\nEPI_ISL_408668_BetaCoV_Vietnam_VR03-38142_2020 EPI_ISL_410535_BetaCoV_Singapore_4_2020 EPI_ISL_408665_BetaCoV_Japan_TY-WK-012_2020 EPI_ISL_406716_BetaCoV_China_WHU01_2020 EPI_ISL_402127_BetaCoV_Wuhan_WIV02_2019 EPI_ISL_411926_BetaCoV_Taiwan_3_2020 EPI_ISL_406223_BetaCoV_USA_AZ1_2020 EPI_ISL_407214_BetaCoV_USA_WA1-A12_2020 EPI_ISL_406717_BetaCoV_China_WHU02_2020 EPI_ISL_404228_BetaCoV_Zhejiang_WZ-02_2020\\nEPI_ISL_412898__BetaCoV_Wuhan_HBCDC-HB-02_2019 EPI_ISL_413017_BetaCoV_South_Korea_KUMC01_2020 EPI_ISL_408480_BetaCoV_Yunnan_IVDC-YN-003_2020 EPI_ISL_403936_BetaCoV_Guangdong_20SF028_2020 EPI_ISL_408667_BetaCoV_Japan_TY-WK-521_2020 EPI_ISL_411218_BetaCoV_France_IDF0571_2020 EPI_ISL_405839_BetaCoV_Shenzhen_HKU-SZ-005_2020 EPI_ISL_408010_BetaCoV_USA_CA5_2020 EPI_ISL_406031_BetaCoV_Taiwan_2_2020 EPI_ISL_406844_BetaCoV_Australia_VIC01_2020 EPI_ISL_411950_BetaCoV_Jiangsu_JS01_2020 EPI_ISL_412974_BetaCoV_Italy_SPL1_2020\\nEPI_ISL_403928_BetaCoV_Wuhan_IPBCAMS-WH-05_2020 EPI_ISL_412870_BetaCoV_Korea_KCDC06_2020 EPI_ISL_404253_BetaCoV_USA_IL1_2020 EPI_ISL_410045_BetaCoV_USA_IL2_2020 EPI_ISL_403932_BetaCoV_Guangdong_20SF012_2020\\nEPI_ISL_408484_BetaCoV_Sichuan_IVDC-SC-001_2020 EPI_ISL_410537_BetaCoV_Singapore_6_2020 EPI_ISL_412872_BetaCoV_Korea_KCDC12_2020 EPI_ISL_407079_BetaCoV_Finland_1_2020\\nEPI_ISL_410545_BetaCoV_Italy_INMI1-isl_2020 EPI_ISL_408670_BetaCoV_USA_WI1_2020 EPI_ISL_403937_BetaCoV_Guangdong_20SF040_2020 EPI_ISL_403934_BetaCoV_Guangdong_20SF014_2020\\nEPI_ISL_413014_BetaCoV_Canada_ON-PHL2445_2020\\nEPI_ISL_411915_BetaCoV_Taiwan_CGMH-CGU-01_2020 EPI_ISL_410717_BetaCoV_Australia_QLD03_2020 EPI_ISL_412026_BetaCoV_Hefei_2_2020 EPI_ISL_407313_BetaCoV_Hangzhou_HZCDC0001_2020 EPI_ISL_412972_BetaCoV_Mexico_CDMX_InDRE_01_2020 EPI_ISL_406034_BetaCoV_USA_CA1_2020 EPI_ISL_411952_BetaCoV_Jiangsu_JS02_2020 EPI_ISL_411902_BetaCoV_Cambodia_0012_2020\\nEPI_ISL_413016_BetaCov_Brazil_SPBR-02_2020 EPI_ISL_413018_BetaCoV_South_Korea_KUMC02_2020 EPI_ISL_403930_BetaCoV_Wuhan_IPBCAMS-WH-03_2019\\nEPI_ISL_412899__BetaCoV_Wuhan_HBCDC-HB-03_2019 EPI_ISL_408489_BetaCoV_Taiwan_NTU01_2020 EPI_ISL_409067_BetaCoV_USA_MA1_2020 EPI_ISL_404895_BetaCoV_USA_WA1_2020\\nEPI_ISL_408666_BetaCoV_Japan_TY-WK-501_2020 EPI_ISL_410218_BetaCov_Taiwan_NTU02_2020 EPI_ISL_407893_BetaCoV_Australia_NSW01_2020 EPI_ISL_407193_BetaCoV_South_Korea_KCDC03_2020\\nEPI_ISL_412983_BetaCoV_Tianmen_HBCDC-HB-07_2020 EPI_ISL_410714_BetaCoV_Singapore_8_2020 EPI_ISL_403935_BetaCoV_Guangdong_20SF025_2020 EPI_ISL_406798_BetaCov_Wuhan_WH01_2019 EPI_ISL_411953_BetaCoV_Jiangsu_JS03_2020 EPI_ISL_412030_BetaCoV_Hong_Kong_VB20026565_2020 EPI_ISL_412968_BetaCoV_Japan_Hu_DP_Kng_19-020_2020 EPI_ISL_412973_BetaCoV_Italy_CDG1_2020 EPI_ISL_402132_BetaCoV_Wuhan_HBCDC-HB-01_2019\\nEPI_ISL_408486_BetaCoV_Jiangxi_IVDC-JX-002_2020 EPI_ISL_408430_BetaCoV_France_IDF0515_2020 EPI_ISL_410486_BetaCoV_France_RA739_2020 EPI_ISL_412966_BetaCoV_China_IQTC01_2020\\nEPI_ISL_412969_BetaCoV_Japan_Hu_DP_Kng_19-027_2020\\nEPI_ISL_411219_BetaCoV_France_IDF0386-islP1_2020\\nEPI_ISL_408481_BetaCoV_Chongqing_IVDC-CQ-001_2020 EPI_ISL_412975_BetaCoV_Australia_NSW05_2020 EPI_ISL_406534_BetaCoV_Foshan_20SF207_2020 . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.03. 16.20034470 doi: medRxiv preprint \\n                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
       "2  The 2019-nCoV epidemic has spread across China and 24 other countries 1-3 as of February 8, 2020 . The mass quarantine measure implemented in the epidemic's epicenter, Wuhan, Hubei Province, China, on January 23, 2020, has stopped the human migration between Wuhan and other places. However, according to the local authorities, over five million people had already traveled from Wuhan to other parts of China in the weeks leading to January 23, 2020, with the majority (79.96%) to the other cities within Hubei.\\nHealthcare resources have become a major bottleneck in the response to this public health crisis. The sudden onset of the epidemic and the sheer size of the infection have caused significant shortage in healthcare resources across Hubei Province. So far, there have been over 11,000 non-local healthcare professionals, mobilized by the public health authorities from other parts of China, fighting against the epidemic in Wuhan. However, this is far from adequate for Wuhan given the severity of the epidemic. In addition to Wuhan, other cities in Hubei are in need of urgent external help as well with respect to health resources, as multiple localized outbreaks are taking place. On February 7, 2020, the National Health Commission of China (NHCC) initiated the \"One City Covered by One Province\" program, explicitly assigning a specific province to aid one city in Hubei. Two critical questions of operational nature remain, which cities, both in Hubei and other provinces, are most vulnerable to localized outbreaks, and how the public health authorities should balance the healthcare resources under stress to best fight the epidemic and minimize the risks? This study is aimed to shed light on these questions by developing a vulnerability analysis framework integrating the human movement and healthcare resource data.\\nThis study makes use of four public datasets: (a) the daily counts of confirmed 2019-nCoV cases, collected from NHCC 4 ; (b) the list of designated hospitals with the number of reserved beds, retrieved from the websites of the provincial and municipal health commissions; (c) the information of the hospitals designated for 2019-nCoV, retrieved from 99.com.cn, a public hospital database; and (d) the human movement data, provided by the Baidu Migration (http://qianxi.baidu.com/), a public site offering real-time migration information across mainland China.\\nRecent studies indicated that the confirmed case data for Hubei Province, in particular, Wuhan, was subject to the under-reporting bias, while the data from places outside of Hubei is more reliable for inferring the actual epidemic size 5 . Following Wu et al. 5 , we developed a susceptible-exposed-infectious-recovered (SEIR)-based metapopulation model, which has 100 sub-populations representing the 100 cities that have the greatest number of people traveling from Wuhan. Note that we adopted the same epidemiological parameters as Wu et al. 5 . This model was then used to estimate IDi, the number of infected individuals in each city. Lastly, healthcare-resource-adjusted vulnerability indexes to the 2019-nCoV outbreak for city i, Vi were obtained as follows: Vi = IDi / Bi, where Bi represents the total number of reserved beds in the designated hospitals.\\nThe Figure presents the estimated number of infected individuals, as well as the healthcareresource-adjusted vulnerability to the 2019-nCoV outbreak across mainland China, as of January 29, 2020. The epidemiological parameters may have changed after January 23, 2020, when the lockdown measure was implemented. Thus, we only use the data by January 23, 2020 and produce the estimation till January 29, 2020 to accommodate the six-day incubation period identified by Wu et al. 5 .\\nThe ten most vulnerable non-Hubei cities are also listed in the figure. Note that the cities in Hubei (which are being aided by other provinces) are not presented here because of their significantly higher vulnerability as compared to others. Among the ten most vulnerable non-Hubei cities, Changsha, Beijing, Chongqing and Shanghai are major metropolises with a large number of estimated infected individuals. Xinyang, Jiujiang, Nanchang and Yiyang are medium-sized cities neighboring Hubei. In particular, Xinyang appears to be in a particular vulnerable situation because of the large number of people traveling from Wuhan and its insufficient healthcare resources.\\nThe reported vulnerability analysis informs public health response to the 2019-nCoV epidemic in multiple ways. For the identified cities with high vulnerability, it is important to increase the reserve of healthcare resources or have contingent plans. In particular, some of these cities already contributed significant healthcare resources to Hubei, leaving themselves inadequately prepared for the localized outbreak. Our analysis revealed that several medium-sized cities, particularly those neighboring Hubei, are highly vulnerable due to the lack of healthcare resources. \\n                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
       "3  The outbreak of infectious diseases has always been one of the most important health problems in the world. With the rapid spread of the new coronavirus (COVID- 19) , much attention has been paid to subjects like rate of an epidemic, transition ways, prevention methods, and remaining time of the virus in the environment [1] [2] [3] [4] [5] . Therefore, the behavior of the virus in relation to environmental factors is critical. The coronavirus is one of the epidemic diseases that many countries have experienced in the new type of it, COVID-19, and can become a serious challenge and affect the previous efforts on sustainable development [6] [7] [8] [9] .\\nIt has been found that there are some delays between the detection of confirmed cases and the actual infection date due to the laboratory test, the time of the announcement of confirmed cases in media, and the time for the patients showing initial symptoms for a total usually of 3 to 5 days [10] [11] [12] . The analysis of Kampf et al. (2020) demonstrated that COVID-19 could remain up to 9 days on different materials based on temperature and humidity [13] . Cascella et al. (2020) conducted research on the new coronavirus structure. Their analysis shows that the virus belongs to the family of single-stranded RNA viruses (+ ssRNA), which has a length of about 30 kb and an envelope with spear structures and they tested the sensitivity of the virus to UV and temperature [14] . investigated the correlations between environmental factors and the number of positive cases of COVID-19 by the use of artificial intelligence techniques. The results of this study put evidence on the role of urban and weather conditions on the epidemic rate [15] . Chen et al. (2020) developed a time-dependent mathematical model for the prediction of the total number of positive cases [16] . Lytle and Sagripanti (2005) studied the impact of UV rays, and their results determined that it can affect the structure of coronaviruses because UV rays could damage the nucleic acids of viral RNA that is vital for the survival of the virus [17] . developed an assessment method for investigating the impact of climate and urban factors in confirmed cases of COVID-19 [18] . Bukhari et al., (2020) , studied the epidemic of COVID-19 on tropical zones and their results show that the highest number of infection happened in temperature among 3 to 17 ºC [19] .\\nThe analysis of Grant and Giovannucci (2009) shows that by increasing the UV, the level of vitamin D in the body increases and makes the infection rates decrease. Therefore, in areas on lower latitudes and closer to the equator line, the epidemic on COVID-19 might be different [20] . In the study of Chan et al. (2011) , the effect of temperature and humidity on SARS coronavirus was analyzed, and they found that increasing temperature decreases the epidemic rate by faster disabling virus structure on contaminated surfaces [21] . Pirouz and Violini (2020) , analyzed the outbreak patterns of COVID-19 by analysis of 117 countries. They suggested two mechanisms of triggering the development of the infection, namely the normal spread of an infection from one parent case and the consequence of a unique phenomenon of accumulation of contagious people [22] . Analysis of the previous studies demonstrates the necessity of further research about the effect of environmental factors on the COVID-19. Since some of the previous researches have shown the effect of weather conditions, especially temperature, on coronavirus viruses, the main purpose of this study is to investigate the better correlation of daily mean air temperature on the epidemic rate of the COVID-19.\\nIn this study, we analysed the number of positive contagions of coronavirus COVID-19, from February 29 th to March 19 th . The selection of February 29 th is due to existence on this date of at least one positive case of COVID-19 in all case studies studied. The analysis of the weather data started from February15 th and ended on March 18 th because the impact of the weather parameter would occur in the upcoming days. Thus, the considered average daily temperature is with an interval of 1 to 14 days before the definitive cases of COVID-19.\\n In order to minimize the impact of other parameters on the results, such as the policy of each government, cultural differences between countries, health care systems, or the protocols in getting coronavirus test, comparative analysis have been done for different regions of the same country, without comparing with each other the results of different countries;  The analysis period for average daily temperature ranges from February15 th , 2020 to March18 th , 2020 (33 days);  The analysis period for confirmed cases of coronavirus COVID-19 is from February 29 th to March19 th ;  For what concerns the cumulative daily rate of confirmed cases, the total number of confirmed cases in a region was divided into the population of that region in order to take into account the different size of the population in each region.\\nTo carry out the analysis, the datasets of five countries and in each country, of three regions/provinces have been considered. The selected case studies are presented in Table 1 . and the location of the three regions shown in Figures 1. \\nMathematical models have been successfully used for the analysis of many hazards [32, 33] . In complex subjects, different Mathematical models can be used to solve the problem [34] [35] [36] . Statistical tests are an appropriate method for the investigation of possible correlations between different datasets [37] . Many statistical tests can be used for the analysis of significant differences between the datasets, among which F-test and Independent-Samples T-Test [38] .\\nIn order to investigate the hypothesis about a correlation between significant differences (Sig.) of average daily temperatures and average cumulative daily rate of confirmed cases in two regions, the following assumptions, and statistical tests have been done:\\n The variances of the two groups are equal: H0  The variances of the two groups are not equal: H1\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10.20059337 doi: medRxiv preprint When using F-test, if the significance level (sig) is less than 0.05, hypothesis H0 is rejected, and then hypothesis H1 is accepted, i.e., the variances of the datasets in the two selected regions are not equal. Conversely, if the significance level (sig.) is larger than 0.05, the hypothesis H1 is rejected, and the hypothesis H0 is accepted, i.e., the variances of the datasets in the two selected regions are equal.\\nThrough F-test, one determines the presence or absence of differences in the variances of the two regions. Through another statistical test, the Independent-Samples T-Test, one can determine whether the difference is significant, has been used. When using Independent-Samples T-Test, a significance level (sig.) lower (higher) than 0.05, means that the difference is (is not) significant.\\nIn our analysis, we used SPSS software. The critical factors in the SPSS Model set up are as follow:\\n The datasets in different regions of the same countries were analyzed with respect to the average daily air temperature during the 33-day period from 15-Feb to 18-Mar and the cumulative daily rate of confirmed cases of COVID-19 during the 19 days from 29-Feb to 19-Mar.  In all statistical tests, the confidence level of 95% is considered.  First, the F-test was used to test whether there is a significant equality of variances for each pair of case studies, and then, depending on the F-test result, by the T-Test the existence of a significant difference between the data of the two cases was investigated.\\nThe average daily temperature [39, 40] , and cumulative daily rate of confirmed cases [41] , are shown in Figures 2 to 6.\\na. Average daily temperature b. Cumulative daily rate of confirmed cases Correlations between the average temperatures and the cumulative daily rate of confirmed cases were found especially for towns with significant differences, with respect to the other two towns of the same country, in the average daily temperature. This was the case for Bandar Abas in Iran, Milan in Italy, Santa Cruz in Spain, and Los Angeles in the US. The average daily temperature and cumulative daily rate of confirmed cases are listed in Table 2 . As a general feature of Table 2 , we notice that  In 11 out of the possible fifteen comparisons, the average cumulative rate of confirmed cases is higher when the average daily temperature is lower, and in the four remaining cases the differences in either variable are small.\\nA more specific analysis per country shows that  In Iran, the average daily temperature in Bandar Abbas was the highest (21.54 ºC), and the average cumulative rate of daily of confirmed cases was the lowest (0.000027). . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10.20059337 doi: medRxiv preprint  In Italy, the average daily temperature in Milan was the lowest (7.99 ºC), and the average cumulative rate of daily of confirmed cases was the highest (0.000690).  In Germany, the average daily air temperatures of the selected towns have similar values, as well as the spread of the COVID-19 infection is similar.  In Spain, the average daily air temperature in Barcelona and Valencia are similar, and the spread of the infection is also similar. However, the comparison of Santa Cruz with higher average daily temperatures shows a slower spread of COVID-19.  In the United States, the average daily air temperature in both New York and Seattle are lower than Los Angeles, and the average cumulative rate of confirmed cases shows the spread of the virus in both is faster than in Los Angeles.\\nThese qualitative results reflect in the statistical analysis of the data by F-test and Independent-Samples T-Test presented in Table 3 . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10.20059337 doi: medRxiv preprint In Table 3 . t values or p-values can be used to check the existence of a significant correlation between the data for two cities.\\nLet us discuss in detail one such comparison, that between Ramsar and Tehran. From the F-test, the F value is equal to 20.4. The criterion for variance equality in the F-test is that the F value must smaller than the critical F value. The statistical analysis has been done on a statistical confidence level of 95%, and therefore the error level is 0.05, and considering the degrees of freedom, the critical F value of the test that can be found through F distribution tables [42] , in this case, is equal to F = 1.57. Since the F value we found is 20.4, larger than the critical F value, the assumption of the equality of variances is rejected. It means that the average temperature variance of the Ramsar and Tehran are not equal.\\nAccording to the t-test, the t value for these two cities is equal to -1.929. Based on an error level of 0.05 and by considering the degrees of freedom, the critical values of the test indicated in the T distribution tables [43] , is in this case equal to t = ± 1.645. Since the t value is out of the critical boundary (-1.645 ≤ acceptable t ≤ +1.645), the assumption of equality in the averages is rejected, and the existence of difference in averages is confirmed. Thus from two tests, it is evident that there is a significant difference between the average temperature of the two cities of Ramsar and Tehran.\\nAn analogous result holds for the average cumulative rate of confirmed cases.\\nThe main results of Table 4 , with a confidence level of 95% are as follow:\\n In Iran, the city of Bandar Abbas is significantly different from Ramsar and Tehran in terms of both average daily temperature and average cumulative rate of confirmed cases.  In Italy, the city of Milan is significantly different from Rome and Naples in terms of both average daily temperature and average cumulative rate of confirmed cases. Instead, the two cities of Rome and Napoli are not significantly different in terms of these variables.  In Germany, there are no significant differences in average daily temperature in all three cities, and the average cumulative rate of confirmed cases is not significantly different.  In Spain, Barcelona and Valencia are not significantly different in terms of average daily temperature, and the average cumulative rate of confirmed cases is not significantly different. However, in Barcelona and Santa Cruz, the average daily temperature is significantly different, as well as the average cumulative rate of confirmed cases, and the same trend exists between Valencia and Santa Cruz.  In the United States, the two cities of New York and Seattle are not significantly different in terms of average daily temperature, and of the average cumulative rate of confirmed cases, the number of patients is not significantly different. Instead, the city of Los Angeles is significantly different in terms of average daily temperature with two cities of New York and Seattle, and the average cumulative rate of confirmed cases are different significantly.\\nThe results of our analysis of daily air temperature for three case studies from five countries showed the existence of correlations between the average temperatures and the epidemic rate, and this is especially evident when the differences in average daily temperature are larger. For example, in Iran and Italy, the regions with lower average daily temperature had a higher epidemic level of the new coronavirus (COVID- 19) , and the same trend exists in all other case studies.\\nThe statistical analysis showed that in all the case studies we considered, referring to five different countries, if there is a significant difference in daily mean temperature, there is also a significant difference in the average cumulative daily rate of confirmed cases, supporting the proposed research hypothesis about the correlation between air temperature and the epidemic rates of COVID-19. Due to the correlations among temperature and hours of sunshine for future research, it is recommended to study the effects of the sun's UV rays as well as the average hours of sunshine on the epidemic rates of COVID-19.\\nThis study analyzed the possible correlation of daily average air temperature with the epidemic rate of the coronavirus (COVID-19) using significant differences analysis between the datasets. The research hypothesis is that in regions with a significant difference in the average daily temperature, the average cumulative daily rate of confirmed cases would also is different significantly. Conversely, to the absence of a significant difference in daily temperature, would correspond no significant difference in the rate of confirmed cases. The hypothesis was investigated through statistical analysis. In order to minimize the impact on the results of factors like the policy of the government or cultural differences among countries (food, exercise, weight, etc.), three case studies within five countries, namely Iran, Italy, Germany, Spain, and United States were compared separately. This comparison using statistical analysis through F-test and Independent-Samples T-Test of the daily temperatures and daily rate of confirmed cases shows that there is a correlation between the average temperature and the epidemic rate, and this is especially evident when differences in average daily temperature are significantly larger, as it happens for Bandar Abbas in Iran, Milan in Italy, Santa Cruz in Spain, and Los Angeles in the US. Besides, the analysis of the average air temperatures shows that the epidemic rates of COVID-19 were higher in the case studies with a lower average temperature. Instead, when no significant differences exist in the average daily temperature of two cities in the same country, there is no significant difference in the average cumulative daily rate of confirmed cases. In conclusion, the results of this study support the research hypothesis and confirm the effectiveness of the proposed method for analysis of the epidemic rates.\\n. CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10 . CC-BY-NC-ND 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.04. 10.20059337 doi: medRxiv preprint \\n                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
       "4  Metagenomic sequencing, which allows us to directly obtain total genomic DNAs from host-associated and environmental samples, has led to important findings in many areas, such as digestive health [1] . While bacteria are the main focus of most metagenomic sequencing projects, there are fast accumulation of viral metagenomic data with sequencing viruses as the main purpose [2] . There are different types of viruses. We are mainly concerned with RNA viruses because many RNA viruses are notorious human pathogens, such as Influenza A, Human immunodeficiency virus (HIV), Ebola, SARS-CoV, and recently identified 2019-nCoV causing Wuhan pneumonia. Unlike DNA viruses, RNA viruses, which contain RNA genomes, lack faithful proofreading mechanisms during replication and thus can produce a group of related but different viral strains infecting the same host. This high genetic diversity within and across different hosts poses a great challenge for designing long-term protection strategies against these infectious diseases. For example, as the circulating strains can change every year, flu vaccine has to be administered every year.\\nAdvances in viral metagenomics have contributed significantly to new RNA virus discovery. According to a survey by Woolhouse et. al, the number of newly identified RNA viruses is changing from 1,899 in 2005 to 5,561 in 2018, which is 3 times of increase [3] . Largescale RNA virus sequencing projects using nextgeneration sequencing technologies have been conducted for different species. For example, Shi et al. have discovered a large number of new RNA viruses by sequencing samples from invertebrate and vertebrate animals [4, 5] . Claire et al. have discovered 25 new RNA viruses by sequencing samples associated with different drosophilid [6] . Bolduc et al. sequenced RNA viruses from Archaeal and bacterial samples [7] . Given fast accumulation of viral metagenomic data and expected discovery of new viruses, a key step is to conduct composition analysis for these data and assign taxonomic groups for possibly new species.\\nComposition analysis can be conducted at the read level or contig level. There is no doubt that contigs can provide more information than short reads in taxonomic classification. However, metagenomic assembly is still one of the most challenging computational problems in bioinformatics. Unlike single genome assembly, metagenomic assembly is more likely to produce chimeric contigs that can contain regions from different species. Thus, metagenomic assembly is often preceded by read binning that groups reads with the same or similar taxonomic labels [8] in order to achieve better performance.\\nIn this work, we design a read level taxonomic classification tool (named as \"CHEER\") for labeling new species in viral metagenomic data. To detect known species with available reference genomes, alignment-based methods can provide sufficient information for both classification and abundance estimation. This function has been incorporated in existing pipelines [9, 10, 11] and is thus not the focus of our tool. Instead, our tool is designed to handle the challenging cases of assigning taxonomic labels for reads of new species, which have not been observed before. This problem was clearly formulated in metagenomic phylogenetic classification tools such as Phymm and PhymmBL [8] . We first define the classification problem following the formulation in Phymm: the classification problem is the assignment of a phylogenetic group label to each read in input data sets. As we are only interested in the hard case of classifying reads of new species, which have not been sequenced and thus no label exists for these species, the expected labels for these reads are the higher-rank taxonomic groups such as genus, family, or order. All our data and experiments are designed so that the test species are masked in the reference database and the trained models.\\nPrevious experiments have shown that nucleotide-level homology search is not sensitive enough for assigning higher rank taxonomic groups to reads from new species [9, 10] . Protein-based homology search is thus preferred for phylogenetic classification [11, 12] . In particular, a recent tool, VirusSeeker [13] utilized BLASTx to classify reads into bacteria, phages, and other viruses. It is not clear whether the protein homology search is able to generate accurate classification at lower ranks. We will evaluate this in our experiments.\\nThere are also alignment-free methods for phylogenetic classification. Many of those tools implemented different machine learning models and were mainly designed for bacteria. For example, RDP [14] adopted a Naïve Bayes based classifier that can automatically learn features from the reads and assigned taxonomic labels to sequences from bacterial and archaeal 16S genes and fungal 28S genes. The NBC classifier [15, 16] encoded metagenomic reads using k-mers and implemented a bag-of-word model to achieve the readlevel taxonomic classification. Phymm [8] utilized interpolated Markov models (IMMs) for metagenomic read classification. These learning-based models enable users to label reads originating from new species.\\nThe previous works have shown that sequence composition is still an important and informative feature for phylogenetic classification. Recent applications of deep learning models have achieved better performance than conventional machine learning methods in different sequence classification problems [17] . In particular, the convolution filters in Convolutional Neural Network models (CNN) can represent degenerated sequence motifs and thus help CNN models to learn abstract composition-based features. For example, frequently activated convolution filters learned by DeepFam often represented the conserved motifs in different protein domain families [18] . The successful applications of deep learning in sequence classification motivated us to design a novel deep learning-based classification model for assigning taxonomic groups for new species in viral metagenomic data.\\nOur main contributions are summarized below. First, our deep learning model can assign higher rank phylogenetic group labels (such as genus) for reads from new species. To achieve this goal, our model organizes multiple CNN-based classifiers in a tree for hierarchical taxonomic group assignment. Second, to enable our model to capture as much information as possible from short reads, we implemented and compared two encoding methods: one hot vs. embedding. Third, as viral metagenomic data usually contains contaminations from either the host genomes or other microbes, we formulated the pre-processing step as an open set problem in targeted image classification and rejected non-viral reads by choosing appropriate negative training sets. We tested our model on both simulated metagenomic data and also real sequencing data. The results show that our model competes favorably with other popular methods.\\nIn this section, we will introduce our method for viral metagenomic taxonomic classification. First, we will show the architecture of CHEER, which is a hierarchical classification model from order to genus. Second, classifiers at each level will be described. Third, we will introduce DNA sequence encoding using skipgram based word embedding and compare it with onehot encoding. Forth, we will describe the rejection layer, following the idea of the open set problem, which is adopted to filter reads not belonging to RNA viruses. Finally, we will detail our dataset used in training and validation.\\nThe architecture of our model is sketched in Fig. 1. The key component is a tree model that consists of multiple classifiers from order to genus. In order to conduct phylogenetic classification for reads from new species, or even new genus, our classification is conducted using a top-down approach from the root to the leaf node. The top layer is a trained CNN that can reject reads not belonging to RNA viruses, which is an important preprocessing step for viral metagenomic data because of the contamination from the host genome or other microbes. Then, the filtered reads (i.e. mostly RNA viral reads) will be fed into the hierarchical classification model, which is referred to as the tree model hereafter. Specifically, the order-level classifier is responsible for classifying the reads into their originating orders. Then each order has a separately trained classifier for assigning input reads to families within that order (O1 to O3 in Fig. 1 ). For all the reads assigned into one family, the family classifier (F1-F7) will assign the reads to different genus within that family. As an example, a path for classifying a read from a species in genus 7 is highlighted in Fig. 1 . CHEER also implemented early stop functions at each level for stopping the classification path at higher ranks. This function can accommodate taxonomic classification for species from new genus or even higher ranks, which is possible for RNA viruses. It is convenient to add fine-grained classifiers within each genus so that we can assign species-level labels. However, as our focus is to conduct phylogenetic classification for reads from new species that do not have a species labels in the training data, we did not include the species-level assignment in this work.\\nAlternative designs of hierarchical classification work, like [19, 20] , either build a classifier for each taxonomic rank or build a binary classifier for each class. We implemented a structure that built just one classifier for each taxonomic rank and compared different structures' performance. According to the result shown in [Supplementary file 1], the structure shown in Fig.1 is better than one classifier per rank. Although one classifier per rank requires fewer classifiers, it needs to train many more parameters because the number of labels for each classifier increases by times. As a result, we observe overfitting more frequently.\\nEach classifier in the tree model is implemented using CNN, which has achieved superior performance in various sequence classification problems [18] . The convolution filters used in CNN resemble positionspecific weight matrix of motifs in genomic sequences. Frequently activated convolution filters can represent well-conserved motifs among sequences in the same class. Thus, a large number of convolution filters will be used in our CNN in order to learn well-conserved sequence features in different classes. Deep CNNs in fields such as computer vision often have multiple convolution layers in order to obtain more abstract features from training images. Here, we use a wide convolution design rather than a deep one, following the idea in [18] . Fig. 2 shows the classifier with the default hyper-parameters. First, reads will be encoded in matrices as input to the convolution layer using two methods. We will describe and compare these two encoding methods later in the Section 2.3. Then, multiple convolution filters of different sizes are utilized to learn the conserved sequence features. According to our empirical study, we choose 3, 7, 11, 15 as the filter sizes, with 256 filters for each size respectively (see Fig. 2 ). Max pooling is then applied to each convolution filter's output so that only the highest convolution value is kept. After the max-pooling layer, we have a 256-dimentional vector as the output of each convolution layer. We concatenate these four vectors into one 1024-dimentional vector and feed it to one fully connected layer with 512 hidden units. Dropout is applied to mitigate overfitting. The CrossEntropyLoss function supplied by Pytorch is adopted as an error estimation function. Due to the unbalanced size of each training class, we use the reciprocal of the proportion of each class as their error weight to help the model fairly learn from each class. And we utilize Adam optimizer with a 0.001 learning rate to update the parameters. The model is trained on HPCC with the 2080Ti GPU unit. \\nIn previous deep learning-based sequence classification models [21, 22] , one hot encoding was commonly adopted to convert a sequence into a matrix, which will then be used as input to convolution layers. There also exist some methods using k-mer composition [23] and frequency [15, 16] to encode the DNA sequence, similar to the bag-of-words model in the field of natural language processing. However, these k-mers representation cannot give complete sequential information about the original reads.\\nTo incorporate both the k-mer composition and also their ordering information, we build a Skip-Gram [24] based embedding layer that can learn which k-mers tend to occur close to each other. Embedding is widely used in natural language processing to learn the semantic and syntactic relationships from sentences. A neural network with one hidden layer is trained to map a word to an n-dimensional vector so that the words that usually occur together will be closer in the n-dimension space. For our DNA read classification task, k-mers are the words and the embedding layer will map proximate kmers into vectors of high similarity. The main hyper-parameter when training the skip-gram model is the size of the training context m, which defines the maximum context location at which the furthest k-mer is taken for training. Specifically, for a kmer at position i as the input to the skip-gram model, the output k-mers are its neighbors located at i+jk, where −m ≤ ≤ m. Larger m results in more training samples with a cost of training time. In our model, the default value of m is 1. As shown in Fig. 3 , we will sample 3 k-mers as our dataset at a time with the middle one as the input and the surrounding two as the output.\\nAs proven in [24] , Skip-Gram is able to automatically learn the relationship between proximate words and thus is expected to help learn the order between k-mers in genomic data. We implemented a hidden layer with 100 hidden units shown in Fig. 3 . After training the model, the co-occurrence features are embedded in the output of the hidden layer. Thus, a k-mer in a read will be represented as a 100-dimension vector, where each element in the vector is the output of the hidden unit in the hidden layer.\\nWhen we use this pre-trained embedding layer to encode the reads, we first split each read into sequential k-mers. Second, each k-mer is fed to the embedding layer and the output is a 100-dimension vector. Third, all these vectors are combined into a large matrix according to the original position of each k-mers. Finally, this matrix will be fed to the convolutional layer.\\nBased on our large-scale experiments, we found that classifiers are easy to overfit with increase of k. Thus, we choose 100-dimensional feature vectors with k being 3 as our default embedding layer parameter set based on our empirical analysis.\\nDue to the small genome size and limitations in library construction, it is not rare that many viral metagenomic datasets are still contaminated with the genomes of the host and other microbes. Viral read screening is thus a key pre-processing step before the downstream composition analysis. Specifically, our tree model should only be applied to viral reads. However, the close set nature of the classifier forces it to assign a known label to each input without being able to distinguish viral reads from others. In this work, we will reject non-viral reads by formulating it as an open set problem in targeted image classification [25] .\\nIn a majority of neural network models [26, 27] , the logit of the last layer will be fed into the SoftMax function, which takes a vector of n real numbers as input and output a normalized probability vector for all labels/classes. However, for an irrelevant input from an unknown clade, all classes in the model tend to have low probabilities and thus applying a threshold on uncertainty can be used to reject unknown classes [25] . Along with this idea, serval approaches [28, 29] have been proposed to solve the open set problem.\\nIn this work, we incorporate the pre-processing step in our hierarchical classification model by integrating a trained CNN and a SoftMax threshold. Contaminations such as bacteria will be rejected using empirically chosen SoftMax threshold τ. Our experiments show that SoftMax outputs a nearly uniform probability distribution of all labels if the input is not relevant. Thus, the SoftMax threshold τ is used to reject reads not belonging to either DNA or RNA viruses. However, using this thresholding method is not able to distinguish DNA viral reads from RNA viral reads due to their similar sequence composition. To accommodate this challenge, we designed a rejection classifier in the top layer using DNA viral reads as the negative class in the training (see Fig. 1 ).\\nAs the focus of our work is to assign labels for new species, the hierarchical classification ends at genuslevel assignment, which is conducted by the bottom layer in Fig. 1 . Given the fast accumulation of viral metagenomic data, there are cases where a new species does not belong to any existing genus and thus the label assignment should stop at the higher ranks. Thus, we allow our model to stop at a higher rank in the tree during the top-down classification. Towards this goal, we still apply SoftMax threshold in our early stop function, which stops reads from further classification by the child classifiers. This early stop function also serves as a quality control measure for preventing wrong classifications if the confidence is low. For example, when a read arrives, the model may be able to identify its order and family labels with high reliability but cannot decide which genus it belongs to. In this case, the model should only assign order and family label to this read rather than classifying it into a potentially wrong genus. To achieve this goal, SoftMax threshold is applied for every classifier in the tree. Only reads with a score higher than the threshold will be assigned to the child class.\\nAn example of early stop is highlighted by the yellow path in Fig. 1 . The model can classify the read into order 2 (O2) and then family 4 (F4) with high reliability but cannot decide which genus it belongs to.\\nAll the viruses used in the experiments were obtained from the RefSeq in the NCBI virus database. The The copyright holder for this preprint (which was not peer-reviewed) is the . https://doi.org/10.1101/2020.03.26.009001 doi: bioRxiv preprint viruses' taxonomic classification was downloaded from Virus Taxonomy 2018b Release [30] published by the International Committee on Taxonomic of Virus (ICVT). Because training deep learning classifiers required enough samples, we removed the orders with only one family, families with only one genus, and genera with less than 3 species from our dataset. Finally, there are 6 orders, 23 families, and 55 genera remained. The RefSeq genomes are available at the following URL:https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/fi nd-data/virus. The information of the taxa can be found in [Supplementary file 1]. As CHEER conducts read-level taxonomic classification, the training is also conducted using reads. In order to train our model on each complete virus genome, we generated simulated reads by uniformly extracting substrings of 250 bp with 200bp overlaps. All these reads formed the training set so that the model can learn the features from the whole genome. The validation set, however, is generated by read simulation tool WgSim [31] with specified error rate so that we can have a better estimation of the model's performance on real sequencing data. Some existing deep learning tools for taxonomic classification, like [21, 23] , generate reads first, then randomly choose some of them as training set and others as testing set. This strategy cannot guarantee that the training and testing data sets have no overlaps from the same species. In order to test whether our model can learn the label of new species, we use test species masking in the training set. Specifically, we split our RNA virus database into two independent sets. For any test species, their reads are not used in the training data. Thus, the model will only use the knowledge of known species for the new species classification.\\nIn this section, we will first give details of the hyperparameters of CHEER. Then we present the results of applying CHEER to both simulated and mock sequencing data. For the simulated data by WgSim, we evaluated how sequencing error rate affects the performance of CHEER and compared the taxonomic classification results with state-of-the-art. For the mock data, we tested CHEER's performance in both taxonomic classification and its ability in distinguishing viral reads from others.\\nIn our experiments, the minimum number of species within a genus is 3. We thus constructed three test sets, each containing reads from one species that is randomly chosen from each genus. The three test sets contain three different species, respectively. In total, 165 species from 55 genera are chosen to be \"new\" species for testing and the information of these species can be found in [Supplementary file 1].\\nWe used WgSim to simulate reads from testing species. In order to evaluate the impact of sequencing error on CHEER, we simulated reads with an error rate of 0, 0.01, 0.02, and 0.05, respectively. For each classifier (CNN), we computed the classification accuracy, which is defined in equation (1). To ensure that the computation of the accuracy is not affected by the number of reads from different species, we constructed a balanced test read set by simulating the same number of reads for each test species (1,000 reads for each). (1) For each component classifier in the tree model, we record the average accuracy for multiple test species. For each taxa level, we calculate the average accuracy of all the classifiers in the same level.\\nWe present the results of CHEER at each level in this section. Because CHEER is designed for new species, we benchmark it against alignment-free models. Based on the results shown by Wood et al. [7] , Naïve Bayes Classifier (NBC) outperformed other alignment-free models. Thus, we compared the performance of CHEER and NBC. NBC in [14] was designed for bacteria taxonomic classification and thus, we retrained NBC on our RNA virus dataset. We implemented the NBC model following [14] with the same parameters. Since each family has a separately trained classifier, which conducts genus classification within that family, there are totally 14 classifiers at this level. Fig. 5 shows the accuracy for labeling families. There are totally 6 classifiers that assign reads to families within each of the 6 orders. Fig. 6 shows the mean accuracy at each rank. The family and genus level accuracy are the average accuracy of all classifiers in Fig. 4 and Fig. 5 . There is only one classifier for order level classification.\\nAs shown in Fig. 4, Fig. 5, and Fig. 6 , CHEER has better performance than NBC across different ranks. We also compared the performance of using one-hot encoding and embedding layer in CHEER. The performance comparison at each rank shows that for most of the classifiers, the embedding layer improves the learning ability. Fig. 4, Fig. 5, and Fig. 6 also revealed the robustness of CHEER. Although the classification accuracy decreases as the error rate increases, the decrease is smaller than 2%. The accuracy is almost identical for many classifiers. Thus, CHEER is not very sensitive to sequencing errors. We also conducted experiments for reads with higher error rates (0.02 and 0.05). The results can be found in [Supplementary file 1].\\nThe next experiment is to evaluate our model on a real sequencing dataset and test CHEER's performance in both taxonomic classification and non-viral read rejection. We built a mock dataset by mixing Ebolavirus reads with gut bacterial reads from a gut amplicon sequencing dataset, which is chosen to ensure that we only use reads from bacteria. The taxonomic classification of Zaire Ebolavirus in the tree model is highlighted in Fig. 7 . We chose one of the Ebola species named Zaire Ebolavirus as our testing species because it is the most dangerous of the five Ebolavirus species and was responsible for the 2014 West African Ebola outbreak. We downloaded the dataset from NCBI with accession number SRR1930021. These reads were sequenced by Illumina using whole-genome amplification strategy. Since Zaire Ebolavirus usually causes fatal hemorrhagic fever and abdominal pain, it is possible that this Ebola virus occurs with other gut bacteria. Thus, we created a mock viral metagenomic data by mixing these Ebola reads with a gut bacteria amplicon sequencing data in order to test whether CHEER can reject the bacterial reads at the top rejection layer. The gut bacteria dataset is downloaded from NCBI with accession number: SRR10714026. There are totally 2,737 Ebola virus reads and 51,280 gut bacterial reads in the mock data. After the rejection layer, the Ebola reads will be used to evaluate our tree model.\\nTo treat Zaire Ebolavirus as a \"new\" species, we removed its genome from our training set and re-trained the whole pipeline. As shown in Fig. 7 , after removing this species, we still have four Ebola species in our training set. However, our read mapping experiment using Bowtie2 [32] showed that none of the reads from Zaire Ebolavirus can be mapped to other Ebola species with the default parameters, indicating that using usual alignment-based methods will have difficulty to conduct genus-level assignment.\\nIn our work, CHEER will first use rejection layer to remove the reads not belonging to the RNA virus. Second, the \"Order classifier\" will be applied to evaluate how many reads can be classified into Mononegavirales order. Third, \"Mononegavirales classifier\" takes all these Mononegavirales reads as input to predict how many reads will be assigned to the Filoviridae family. Finally, \"Filoviridae classifier\" is adopted to check how many reads belong to the Ebola genus. Fig. 8 show the ROC curve of the rejection layer. True positive rate reveals how many Ebolavirus reads pass the rejection layer. The false positive rate reveals how many gut bacterial reads pass the rejection layer. We generated thresholds from 0 to 1 with step size 0.01 and used each threshold for the rejection layer. Then for each threshold, we recorded the true positive rate and false positive rate for plotting the ROC curve. As shown in Fig. 8 , with the increase of the threshold, more bacterial reads are filtered by the rejection layer, at the expense of missing more real Ebola reads. The default threshold for the rejection layer is 0.6 in CHEER. User can adjust the threshold according to their needs.\\nAfter evaluating the rejection layer, we continued to test the hierarchical classification performance at each rank. We first compared our tree model with NBC. Then we also benchmarked CHEER with the method using protein-level alignment in Section 3.3.3. The performance of the complete hierarchical classification pipeline of NBC and CHEER is shown in Fig. 9 . From order to genus, each bar shows the accuracy of correctly classified reads by the parent classifier. For example, NBC correctly classified 1580 at the order level. Then, 1564 out of the 1580 reads are correctly labeled at the family level. While Fig. 4 , 5, and 6 have shown the performance of CHEER and NBC at each taxonomic rank using the same set of reads as input for each level of classifiers, this experiment focused on comparing the hierarchical classification pipeline using only correctly predicted reads by the parent classifier. The results demonstrated that CHEER can classify Ebola reads with better performance. NBC can only classify 1,564 (57.1%) reads into Ebolavirus. CHEER, however, had totally 2407 (87.9%) reads classified correctly.\\nAlthough nucleotide-level alignment models are not ideal choices for predicting new species, as mentioned in [8] , protein-level alignment such as BLASTx is still frequently used for homology search because of the conservation among homologous protein sequences. We followed VirusSeeker [10] , which is a BLASTx based model, to align translated reads to virus-only protein database (VIRUSDB NR) provided by NCBI. Then, we kept alignments with E-value cutoff 0.001, which is used in VirusSeeker. The output of BLASTx shows that a read can often align to multiple proteins, making the phylogenetic classification ambiguous (see Fig. 11 ). Also, existing tools such as VirusSeeker often use the best alignment only, which does not always yield the highest accuracy. To optimize the usage of the alignment-based method, we implemented k-Nearest Neighbor (kNN) algorithm to record the best k alignments rather than only using the best one. The label of the input read is determined by the majority vote in the labels of the top k alignments. According to Fig. 10 , most of the reads have hit numbers between 11 to 15 and 6 to10. Thus, we chose 1, 5, 10, 15 as our k when running kNN. There are also 512 reads that cannot be aligned with any VIRUSDB protein sequence. All these reads will be regarded as wrong prediction. Fig. 11 shows the classification results of kNN. When we use the best hit as the prediction the accuracy is only 64%. The best performance of kNN will be 74% with 2,026 correct predictions when k is 5. But the accuracy sharply decreases if k is larger than 5. Fig. 10 and Fig. 11 also revealed that it is hard for users to choose a proper k in real classification tasks. The most common value of the hit number is 11-15 shown in Fig. 11 . The best kNN result, however, has a k equal to 5. Thus, comparing to this alignment-based method, CHEER, with 2,407 (87.9%) reads classified correctly, is more practical and reliable.\\nThe two main purposes of the early stop function are: 1) to handle the cases of a new species from an unseen genus or higher ranks; 2) to serve as a quality control measure for preventing wrong classifications in child classifiers. As Fig. 6 shows, the higher rank classification tends to be more accurate than lower ranks. Because the genus of Ebolavirus exists in our classification tree, the evaluation of the early stop function focuses on the second purpose. Fig. 12 shows the percentage of how many reads are classified into Ebolavirus genus correctly for different classifiers. By allowing early stop, the number of wrong predictions decreased sharply. In short, although there are fewer reads being labelled at the genus level, the total number of reads with wrong classification is decreased. Thus, this is still a useful feature provided by CHEER.\\nAnother significant advantage of CHEER is the high classification speed. Although training the model uses heavy computing resources, the inference procedure of this algorithm is fast. And with the help of GPU acceleration, the training time also decreases a lot.\\nIn the experiments shown in Section 3.4, it takes around 46 minutes to run BLASTx against the virus protein database. Our method, however, only takes 7 second for the preprocessing step to convert the raw sequence into a vector, and then takes less than 35 second to make the prediction for all 2,737 reads.\\nIn this work, all training data are RNA virus genomes downloaded from the NCBI RefSeq database. The training reads are extracted uniformly from the genome to make sure that classifiers can learn the features from the whole genome. To evaluate the model on new species detection, all species in the test set are excluded from the training data. Error rates are also added to evaluate the robustness of the model. The results reveal that the model is tolerant to the sequencing error. In addition, a SoftMax threshold rejection layer is applied to filter reads from non-RNA viruses. The SoftMax threshold in each classifier will also help early stop classification with low confidence and thus decreases the wrong predication rate. We show a case study on real-world data to identify how this threshold will influence the rejection classifier and early stop function. The results show that CHEER competes favorably with state-of-the-art tools.\\n   \n",
       "\n",
       "   clusters  \n",
       "0  1         \n",
       "1  0         \n",
       "2  0         \n",
       "3  0         \n",
       "4  0         "
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['clusters'] = clusters\n",
    "df.head() #display the head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# taged_abs=[]\n",
    "# nouns=[]\n",
    "# # tagging the clean data\n",
    "# for val in clean_abs:\n",
    "#     taged_abs.append(pos_tag(val.split()))\n",
    "# # extracting the nouns\n",
    "# for i in range(len(taged_abs)):\n",
    "#     doc=[]\n",
    "#     for j in range(len(taged_abs[i])):\n",
    "#         if(taged_abs[i][j][1]=='NN' or taged_abs[i][j][1]=='NNS'):\n",
    "#             doc.append(taged_abs[i][j][0])\n",
    "#     nouns.append(doc)\n",
    "        \n",
    "# print(nouns)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Above we dcided to calculate the tf-idf so that we can represent every word that is present in the abstract quatitavily. By doing so we can further use the results in order model the topic according to the abstract that we just quatified.Furthermore we will use (Non-negative Matrix Factorization) NMF in order to come up with topic's that carry most weight in the abstract. To accomplish this we are going to filter all the nouns that are avaliable in the abstract and use them to represent the different topis that are avaliable."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
